#!/usr/bin/perl
END{

# check if required modules are installed on this machine and output intelligible error messages if not.
$missing_module="";
$load_module=eval{require File::Path;File::Path->import();1;};
if(!$load_module){print"\nPerl module File::Path (in core since 1995-03-14) is not installed on this machine!";$missing_module="File::Path";}
else{eval'use File::Path qw( make_path );';}
$load_module=eval{require File::Copy;File::Copy->import();1;};
if(!$load_module){print"\nPerl module File::Copy (in core since 1996-02-29) is not installed on this machine!";$missing_module="File::Copy";}
$load_module=eval{require File::Basename;File::Basename->import();1;};
if(!$load_module){print"\nPerl module File::Basename (in core since 1994-10-17) is not installed on this machine!";$missing_module="File::Basename";}
$load_module=eval{require Getopt::Long;Getopt::Long->import();1;};
if(!$load_module){print"\nPerl module Getopt::Long (in core since 1994-10-17) is not installed on this machine!";$missing_module="Getopt::Long";}
$load_module=eval{require LWP::Simple;LWP::Simple->import();1;};
if(!$load_module){print"\nPerl module LWP::Simple is not installed on this machine!";$missing_module="LWP::Simple";}
$load_module=eval{require LWP::UserAgent;LWP::UserAgent->import();1;};
if(!$load_module){print"\nPerl module LWP::UserAgent is not installed on this machine!";$missing_module="LWP::UserAgent";}
$load_module=eval{require LWP::Protocol::https;LWP::Protocol::https->import();1;};
if(!$load_module){print"\nPerl module LWP::Protocol::https is not installed on this machine!";$missing_module="LWP::Protocol::https";}
$load_module=eval{require Archive::Extract;Archive::Extract->import();1;};
if(!$load_module){print"\nPerl module Archive::Extract (in core since 2007-07-07) is not installed on this machine!";$missing_module="Archive::Extract";}
$load_module=eval{require Scalar::Util;Scalar::Util->import();1;};
if(!$load_module){print"\nPerl module Scalar::Util (in core since 2002-03-05) is not installed on this machine!";$missing_module="Scalar::Util";}
else{eval'use Scalar::Util qw( looks_like_number );';}
if(length$missing_module>0)
	{
	print"
Perl modules are available for download at http://www.cpan.org/\n\nIf you have CPAN installed try the following:
 cpan
 install $missing_module
 exit

If you are on MAC or UNIX you should run cpan with sudo:
 sudo cpan
 install $missing_module
 exit

";
	exit;
	}

$|=1;
$unitas_version="1.7.6";
$last_modified="19. February 2020"; # Wrong version number. Error when importing user RefSeq =tr/ instead of =~tr/
$t0=time;

if($^O eq'MSWin32')
	{
	$usage_command="unitas.exe -input path/to/sequence.file -species genus_species";
	}
else
	{
	$usage_command="./unitas -input path/to/sequence.file -species genus_species";
	}

########## USAGE ##########
$usage="

                                                           X
                                                          oXX.o
                                                         .XXXX
                                                        .XXXXX
                                                       .XXXXX.
                                                      .XXXXXXXXXo
                                                     oXXXXXX' 'XXX.
            XXX                                   .oXXXXXXXX. .XXXXo
           XXXXX.  .oXXXXoo..                 ...oXXXXXXXXXXXXXXXXXX
            XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXo'''
               XXXXXXXXXXXXXXX------------------XXXXXXXXXXXX'
           .X.   'XXXXXXXXXXXX      unitas      XXXXXXXXXXXXX
         XXXXXXooXXXXXXXXXXXXX------------------XXXXXXXXXXXXXoooooo...XXXX
       .XXXX'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'''XXXXXXXXXXXXXXX''
    .oXXXX'    ''XXXXXXXXXXX' 'XXXXXXXXXXXXXXX'''''        '''       ''
   oXXXX'          'XXXXX''
   XXXX'


 Version: ..... $unitas_version
 Date: ........ $last_modified


                              ### Usage ###
 All you need to start is a file that contains your sequence data in FASTQ,
 FASTA, SAM or ELAND3 format and a connection to the internet. Start the unitas
 Perl script from the command line (terminal) using the following command:

    perl unitas.pl -input path/to/sequence.file -species genus_species

 When you run the standalone executable file use this command instead:

    $usage_command

 unitas uses ncRNA (and cDNA) sequences from Ensembl and thus supports species
 included in the latest Ensembl release. Use the option -supp_spec to display a
 list of all supported species/strains. If you want to annotate sRNA sequences
 from a different species chose a closely related species or use the option
 -species like this:

    -species x

 In the latter case you must provide your own reference sequence file(s) using
 the option

    -refseq sequence_1.fas [-refseq sequence_2.fas ...]

 Use the option -show_options to display a list with all available command line
 options.


                          ### How to cite unitas ###
 Gebert D, Hewel C, Rosenkranz D. unitas: the universal tool for annotation of
 small RNAs. BMC Genomics 2017 18:644.


                               ### Contact ###
 David Rosenkranz
 Institute of Organismic and Molecular Evolutionary Biology, Anthropology
 Johannes Gutenberg University Mainz, Germany
 email: rosenkranz\@uni-mainz.de
 web: www.smallRNAgroup.uni-mainz.de

 You can find a more detailed documentation here:
 www.smallRNAgroup.uni-mainz.de/software/unitas_documentation_$unitas_version.pdf


                               ### Twitter ###
 Follow us on Twitter to get information on latest software updates:
 https://twitter.com/smallRNA_Mainz

 ==============================================================================

 ";

 $usage_all_options="

                                ### Options ###
 Allowed values for options: [s]=string, [i]=integer, [f]=floating point number

 -i OR -input [s]          Name of the input file. You can use multiple input
                           files at once, e.g.:
                           -i file1.fas -i file2.fq -i file3.sam
                           Alternatively you can provide the name of a
                           directory that contains all of your input files,
                           e.g:
                           -i path/to/your/input/files
 -s OR -species [s]        Species name. Use the binominal nomenclature with
                           genus and species seperated by an underline, e.g.:
                           -s mus_musculus
 -latest_ref               Force unitaz to download the latest database
                           sequences. Otherwise unitas will use previously
                           downloaded sequences (if available) from the
                           UNITAS_refdump folder.
 -refdump                  Create a reference sequence dump folder, download
                           species-specific reference sequences and QUIT.
                           This folder and sequences herein can be used for
                           later offline unitas runs.
 -refseq [s]               Use an arbitrary number of additional files that
                           contain reference sequeces in FASTA format. FASTA
                           headers should have the following format:
                           >ncRNA_type|ncRNA_name
 -riborase                 Find all rRNA-like sequences in your input file
                           by using a complete NCBI rRNA nucleotide collection
                           in addition to the available ncRNA data for your
                           species.
 -trim                     Use this option to search for and remove any 3'
                           adapter sequence in input files. There is no need
                           to provide a specific sequence.
 -trim_minlength [i]       Minimum allowed read length [nt] after removal of
                           adapter sequence. Default=10.
 -trim_maxlength [i]       Maximum allowed read length [nt] after removal of
                           adapter sequence. Default=50.
 -trim_ignore_5p [i]       Ignore first n basepairs for 3� adapter prediction.
                           Default=22.
 -trim_skip_reads [i]      Skip the first n reads when searching for frequent
                           motifs. Default=10000
 -trim_check_reads [i]     Check n reads for 3� adapter prediction.
                           Deafult=500000
 -trim_polyA               Allow polyA tails to be recognized as 3' adapter.
                           Useful when library was prepared with polyA-tailing
                           rather than 3' adapter ligation.
 -trim_maxfinalbias [f]    Maximum allowed 3' nucleotide bias after trimming.
                           If this value is exceeded, unitas will trim further
                           3' positions. Default=0.8.
 -quick                    Allow less mismatch. Equivalent to:
                           -tail 1 -intmod 0 -mismatch 0 -insdel 0
                           Faster, less memory usage, less sensitive.
 -slow                     Allow more mismatch. Equivalent to:
                           -tail 3 -intmod 2 -mismatch 2 -insdel 1
                           Slower, more memory usage, more sensitive.
 -tail [i]                 Maximum number of allowed non-template 3'
                           nucleotides when mapping user sequences to reference
                           miRNA (miRNA-precursor) and piRNA cluster sequences.
                           Default=2. Summed values for -tail and -intmod must
                           not exceed 5.
 -intmod [i]               Maximum number of allowed internal mismatches when
                           mapping user sequences to reference miRNA (miRNA-
                           precursor). Default=1. Summed values for -tail and
                           -intmod must not exceed 5.
 -species_miR_only         Do not use miRNA (-precursor) sequences from other
                           species when searching for miRNAs in input dataset.
                           Default=off.
 -mismatch [i]             Maximum number of allowed mismatches when mapping
                           user sequences to reference ncRNA sequences (does
                           not apply to miRNAs and putative piRNAs). Default=1.
 -insdel [i]               Maximum number of insertions/deletions when mapping
                           user sequences to reference ncRNA sequences (does
                           not apply to miRNAs and putative piRNAs). Default=0.
 -threads [i]              Maximum number of parallel threads. Applies only if
                           using more than one input file. Default=1.
 -memory [i]               Limit memory usage [GB] for SeqMap mapping. Default
                           is no limit. When using a limit, unitas will quit
                           before the limit will be exceeded.
 -less_memory              Will use modified parameters for SeqMap mapping to
                           reduce memory usage. Will increase running time.
 -even_less_memory         Will use modified parameters for SeqMap mapping to
                           reduce memory usage even more. Will increase running
                           time.
 -min_tRNA_half_size       Minimum length for a tRNA fragment to be annotated
                           as tRNA-half insted of tRF. Default=30.
 -max_tRNA_end_dist        Maximum distance of a tRNA fragment from the ends
                           of a source tRNA to be annotated as 5' or 3'
                           fragment. Otherwise it will be annotated as
                           misc-tRF (internal fragment). Default=0.
 -phasi [i]                Search for phased siRNAs (phasiRNAs or tasiRNAs)
                           with specified length = [integer]. When using this
                           option, the input file must be a map file in either
                           SAM or ELAND3 format. You can use this option
                           multiple times, e.g: -phasi 21 -phasi 24.
 -na_for_phasi             Only sequences without annotation will be considered
                           for phasiRNA prediction.
 -pp                       Analyze 5' overlaps of mapped reads to identify so-
                           called ping-pong signatures (preference for 10 bp
                           overlaps) which is a footprint of secondary piRNA
                           biogenesis. When using this option, the input file
                           must be a map file in SAM or ELAND3 format.
 -keep_temp                Do not remove temporary files from data dump folder.
 -skip_mapping             Skip mapping to reference sequences. Equivalent to:
                           -skip_miR -skip_cDNA -skip_ncRNA -skip_piR
                           Use this when you want to trim 3' adapters and/or
                           want to search for phasiRNAs only.
 -skip_miR                 Skip search for miRNA reads, search for other ncRNAs
                           only.
 -skip_cDNA                Skip search for reads that map to Ensembl cDNA
                           sequences.
 -skip_ncRNA               Skip search for reads that map to ncRNA collection.
 -skip_piR                 Skip search for reads that map to known piRNA
                           producing loci.
 -skip_dust                Skip filtering low complexity sequences.
 -dust_limit [f]           Threshold for fraction of a sequence that consists
                           of simple sequence repeats When filtering low
                           complexity reads from input files. Default=0.75.
 -dust_max [i]             Maximum motif size for simple sequence repeats.
                           Default=5.
 -dust_maxN [f]            Maximum fraction of Ns in a read. Default=0.5.
 -noCCAcheck               Mature tRNAs carry a CCA-3' which is enzymatically
                           added to tRNA transcripts unless it it present
                           already. unitas will check 3' termini of tRNAs and
                           add the CCA-3' unless this option is turned on.
 -CC_is_CCA                Treat 3' tRFs with CC-3' as CCA-3'. Makes sense
                           when protocols for library construction use poly-A
                           tailing with subsequent A(n)-3' trimming.
 -no_html                  Do not output html files with visualization of
                           sncRNA annotation and analysis.
 -diffexpr                 Search for differentially expressed sRNAs accross
                           different probes. Define 2 groups with a recommended
                           minimum of 2 probes per group using the options -g1
                           and -g2 as described below. When using this option
                           unitas will do the comparison and quit. Change the
                           name of the output file using the option
                           -diffexpr_out.
 -diffexpr_t               Perform standard t-test (unpaired, two-sided).
                           By default unitas will calculate the Shared Area of
                           two Normal Distributions (SAND) derived from the
                           read counts of the two distinct groups and use it
                           as a proxy for p.
 -diffexpr_p [f]           Critical p value for differential expression
                           analysis. Default=0.05.
 -diffexpr_acorrection [i] Correction of alpha error (multiple testing).
                           Allowed values are 0 (=no correction), 1
                           (=Bonferroni-Holm, default) and 2 (=Bonferroni).
 -diffexpr_estdev [f]      If one group comprises only one replicate, the
                           variance can be estimated by means of this factor
                           which is 0.05 by default. That means if sequence A
                           has 100 reads in the given replicate, unitas
                           virtually adds two replicates with read count values
                           105 and 95 (+- 5%).
 -diffexpr_nostrays        Only consider sequences that are present in all
                           replicates.
 -g1 [s]                   unitas output folder for probe assigned to group 1.
                           Use this option multiple times for every probe that
                           you want to assign to group 1. E.g:
                           -g1 UNITAS_04-01-2017_SRR123456.fasta_#1
 -g2 [s]                   Probes assigned to group 2 (see -g1).
 -diffexpr_out [s]         Specifies file names were the results of the
                           differentially expressed RNAs analysis will be
                           written to. Extensions .up1 and .up2 will be added
                           to the name. Default file names are:
                           'unitas.diff-expr.up1' and
                           'unitas.diff-expr.up2'.
                           (~.up1->higher expression in group 1, ~.up2->higher
                           expression in group 2).
 -silent                   Less console (STDOUT) output during annotation.
 -supp_spec                Prints the complete list of supported species and
                           quits.
 -h OR -help               Shows usage information and quits.
 -show_options             Shows available options and quits.

 ==============================================================================

";

### SET DEFAULTS ###

# trimming
$ignore_5p=22;
$skip_first_X_reads=10000;
$check_seq=500000; # set to -1 if all sequences should be checked
$check_seq_for_size=100000;
$min_size=10;
$max_size=50;
$adapter_mismatch=2;
$clipped_fraction_warning=0.5;
$post_check_3p_ends=50;
$refinement_threshold=0.8;

# main
$max_miRtail_length=2;
$intmod_miR=1;
$seqmap_max_mm=1;
$seqmap_max_insdel=0;
$threads=1;
$memory=99999;

# low complexity filter
$dust_limit=0.75;
$dust_min=1;
$dust_max=5;
$dust_maxN=0.5;

# diff. expr.
$diffexpr_outfile="unitas.diff-expr";
$diff_expr_p=0.05;
$alpha_correction=1; # 0 = no correction / 1 = Bonferroni-Holm / 2 = Bonferroni
$estimated_deviation=0.05;

# tRNA
$tRNA_half_min_size=30;
$tRNA_max_dist_from_end=0; # maximum allowed distance from the source tRNA 5' or 3' ends to get annotated as 5' fragments or 3' fragments

# phasiRNA
$max_factorials_stored=100; # maximum  number of factorials stored in memory
$min_phasi_fraction=0.25; # minimum fraction of phased reads in a given sliding window
$min_phasi_reads=0; # minimum number of normalized phased reads
$min_phasi_loci=5; # minimum number of phased loci within the sliding window
$max_allowed_strandbias=0.95; # maximum strand bias of mapped reads. Avoid false positive phasiRNA prediction (e.g. rRNA fragments)
$min_in_line=0; # minimum number of loci directly adjacent to each other
$min_distance_between_clusters=1000; # minimum distance of loci to allocate different cluster ID
$phasi_sw_size=1000; # must be an even number
$phasi_p=0.05;

# miscellaneous
$dump_folder_prefix="UNITAS";
$outfile_prefix="unitas";
%miRtails=('A'=>1,'T'=>1,'G'=>1,'C'=>1);

# Set user agent
$useragent = LWP::UserAgent->new(ssl_opts => { verify_hostname => 0 });

# collect and check command line arguments
GetOptions
	(
	# mandatory
	"i|input=s"=>\@input,
	"s|species=s"=>\$species,

	# optional
	"refseq=s"=>\@user_refseq, # use additional reference sequences provided by the user
	"riborase"=>\$riborase, # use NCBI rRNA sequence collection to strictly filter rRNA sequences
	"trim=s"=>\$do_trimming, # search and remove 3' adapter sequences
	"trim_ignore_5p=i"=>\$ignore_5p, # ignore 5p sequence [nt] when searching for frequent motifs
	"trim_skip_reads=i"=>\$skip_first_X_reads, # skip first n reads in input file (often low quality and lots of N)
	"trim_check_reads=i"=>\$check_seq, # check only a specified number of reads (will make it faster)
	"trim_polyA"=>\$trim_polyA, # allow polyA tailings to be recognized as 3' adapter sequence
	"trim_minlength=i"=>\$min_size, # minimum length of a read after adapter removal
	"trim_maxlength=i"=>\$max_size, # maximum length of a read after adapter removal
	"trim_maxfinalbias=f"=>\$refinement_threshold, # maximum allowed 3' nucleotide bias after trimming
	"quick"=>\$quick_annotation, # maximum number of nontemplate 3' nucleotides
	"slow"=>\$slow_annotation, # maximum number of nontemplate 3' nucleotides
	"tail=i"=>\$max_miRtail_length, # maximum number of nontemplate 3' nucleotides
	"intmod=i"=>\$intmod_miR, # maximum number mismatches to ncRNA (not miR)
	"species_miR_only"=>\$search_species_miRs_only, # skip mapping to miRNA (-precursor) sequences from other species
	"mismatch=i"=>\$seqmap_max_mm, # maximum number mismatches to ncRNA (not miR)
	"insdel=i"=>\$seqmap_max_insdel, # maximum number of indels to ncRNA (not miR)
	"memory=i"=>\$memory, # limit memory usage for SeqMap
	"less_memory"=>\$less_memory, # use special options for SeqMap to reduce memory usage
	"even_less_memory"=>\$even_less_memory, # use special options for SeqMap to reduce memory usage
	"phasi=i"=>\@phasi, # search for phased siRNAs with length [integer]
	"min_phasi_fraction=f"=>\$min_phasi_fraction, # minimum fraction of phased reads in a given sliding window
	"min_phasi_reads=f"=>\$min_phasi_reads, # minimum number of normalized phased reads
	"min_phasi_loci=i"=>\$min_phasi_loci, # minimum number of phased loci within the sliding window
	"min_phasi_in_line=i"=>\$min_in_line, # minimum number of loci directly adjacent to each other
	"min_distance_between_clusters=i"=>\$min_distance_between_clusters, # minimum distance of loci to allocate different cluster ID
	"max_allowed_strandbias=f"=>\$max_allowed_strandbias, # maximum strand bias of mapped reads. Avoid false positive phasiRNA prediction (e.g. rRNA fragments)
	"min_tRNA_half_size=i"=>\$tRNA_half_min_size, # minimum size for a tRNA half
	"max_tRNA_end_dist=i"=>\$tRNA_max_dist_from_end, # maximum distance for 3' or 5' fragments from the ends of a source tRNA
	"phasi_sw_size=i"=>\$phasi_sw_size, # must be an even number
	"phasi_p=f"=>\$phasi_p, # critical p-value for significant enrichment of phasiRNAs
	"na_for_phasi"=>\$na_for_phasi, # only sequences without annotation will be considered for phasiRNA prediction
	"calc_big_factorials"=>\$calc_big_factorials, # calculate big (>100) factorials. Without this option reads will be downscaled to n=100 which may decrease sensitivity
	"pp"=>\$check_pp_signature, # analyze 5' overlaps for total dataset and ncRNA sub-fractions
	"threads=i"=>\$threads, # maximum number of parallel threads
	"silent"=>\$silent, # less output to console
	"diffexpr"=>\$diffexpr, # do differential expression analysis and quit
	"diffexpr_t"=>\$standard_Ttest, # choose test
	"diffexpr_p=f"=>\$diff_expr_p, # critical p value
	"diffexpr_out"=>\$diffexpr_outfile, # name of the output file for differentially expressed RNAs
	"diffexpr_acorrection=i"=>\$alpha_correction, # 0 = no correction / 1 = Bonferroni-Holm / 2 = Bonferroni
	"diffexpr_estdev=f"=>\$estimated_deviation, # estimated deviation (+ and -) when using only one replicate
	"diffexpr_nostrays"=>\$diffexpr_nostrays, # only consider sequences that are present in all replicates
	"g1=s"=>\@diffex_g1, # unitas results folders for group 1
	"g2=s"=>\@diffex_g2, # unitas results  folders for group 2
	"latest_ref"=>\$latestref, # force program to download latest reference sequences
	"refdump"=>\$refdump_and_quit, # download reference sequences and quit
	"dump_prefix=s"=>\$dump_folder_prefix, # prefix for dump folders
	"output_prefix=s"=>\$outfile_prefix, # prefix for output files
	"keep_temp"=>\$keep_temp, # do not remove temporary files from data dump folder
	"skip_mapping"=>\$skip_mapping, # skip all mapping steps
	"skip_miR"=>\$skip_miR, # skip search for sequences derived from known piRNA producing loci
	"skip_cDNA"=>\$skip_cDNA, # skip search for sequences derived from annotated mRNA transcripts
	"skip_ncDNA"=>\$skip_ncRNA, # skip search for ncRNA fragments other than miRNA
	"skip_piR"=>\$skip_piR, # skip search for sequences derived from known piRNA producing loci
	"skip_dust"=>\$skip_dust, # skip low complexitiy filter
	"dust_limit=f"=>\$dust_limit, # threshold for dusty fraction of a sequence
	"dust_max=i"=>\$dust_max, # maximum motif size for simple sequence repeat
	"dust_maxN=f"=>\$dust_maxN, # maximum number of Ns within a sequence
	"dust_ignore_2bases_bias"=>\$no_overrepresentation_check, # do not check for overrepresentation of 2 bases
	"CC_is_CCA"=>\$CC_is_CCA, # treat 3'tRFs with CC-3' as CCA-3', makes sense when library construction was done with poly-A tailing
	"noCCAcheck"=>\$noCCAcheck, # do not check tRNAs in order to add CCA to the 3' end
	"no_html"=>\$no_html, # do not output html files
	"supp_spec"=>\$print_supported_species, # print supported species
	"h|help"=>\$print_help, # print help desk and quit
	"show_options"=>\$show_all_options, # print available options and quit
	)||die print"\nError in command line arguments.\n";

# print help / options and quit
if($print_help)
	{
	print$usage;
	exit;
	}
elsif($show_all_options)
	{
	print$usage_all_options;
	exit;
	}

# optionally make differentiall expression analysis and quit
if($diffexpr)
	{
	if($alpha_correction!~/^[012]$/)
		{
		print"\nERROR! Allowed values for alpha correction are 0 (no correction), 1 (Bonferroni-Holm) and 2 (Bonferroni).\n\n";
		exit;
		}
	open(DIFFEX_OUT_1,">${diffexpr_outfile}.up1")||die print"\nERROR! Unable to open output file ${diffexpr_outfile}.up1.\n$!\n\n";
	open(DIFFEX_OUT_2,">${diffexpr_outfile}.up2")||die print"\nERROR! Unable to open output file ${diffexpr_outfile}.up2.\n$!\n\n";
	diff_expr();
	exit;
	}
$species=lc$species;
$species=~s/^\s*//;
$species=~s/\s*$//;
foreach(@phasi)
	{
	$phasi{$_}=1;
	}

if(@input==0)
	{
	if(!$refdump_and_quit&&!$print_supported_species) # continue if the user only wants to safe reference sequences or print supported species
		{
		print"\nError. No input files specified. Use the option -input [file_or_folder.name] to specify input files.\n\n";
		exit;
		}
	}

# Check for latest unitas version
$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/software/current_versions');
$response = $useragent->request($request);
$current_versions = $response->content;
if(length$current_versions>0)
	{
	@current_versions=split("\t",$current_versions);
	foreach(@current_versions)
		{
		@info=split(':',$_);
		if($info[0]eq"unitas")
			{
			if($info[1]eq$unitas_version)
				{
				print"\n Congrats! You are using the latest unitas version.\n";
				$update_available="";
				}
			else
				{
				if(!$silent){print"\n There is an updated version of unitas (v.$info[1]) available for download.\n Check it out at https://www.smallrnagroup.uni-mainz.de/software.html\n";}
				$update_available=" There is an updated version of unitas (v.$info[1]) available for download.\n Check it out at https://www.smallrnagroup.uni-mainz.de/software.html\n\n";
				}
			}
		}
	}
load_species_dictionary();

# use SeqMap options that will lead to less memory usage
$less_memory_options_miR="";
$less_memory_options_ncR="";
if($even_less_memory&&$max_miRtail_length+$intmod_miR>0)
	{
	$max_num_part=int(((($max_miRtail_length+$intmod_miR)**2)/3)+0.5);
	$less_memory_options_miR="/limit_num_part:$max_num_part "; #/no_fast_index /no_store_key /shift_mask
	}
elsif($less_memory&&$max_miRtail_length+$intmod_miR>0)
	{
	$max_num_part=int(((($max_miRtail_length+$intmod_miR)**2)/2)+0.5);
	$less_memory_options_miR="/limit_num_part:$max_num_part ";
	}
if($even_less_memory&&$seqmap_max_mm+$seqmap_max_insdel>0)
	{
	$max_num_part=int(((($seqmap_max_mm+$seqmap_max_insdel)**2)/3)+0.5);
	$less_memory_options_ncR="/limit_num_part:$max_num_part "; #/no_fast_index /no_store_key /shift_mask
	}
elsif($less_memory&&$seqmap_max_mm+$seqmap_max_insdel>0)
	{
	$max_num_part=int(((($seqmap_max_mm+$seqmap_max_insdel)**2)/2)+0.5);
	$less_memory_options_ncR="/limit_num_part:$max_num_part ";
	}

# override mismatch settings when using the options -quick or -slow
if($quick_annotation&&$slow_annotation)
	{
	print"\nError. Do not use the options -quick and -slow at the same time.\n\n";
	exit;
	}
elsif($quick_annotation)
	{
	$max_miRtail_length=1;
	$intmod_miR=0;
	$seqmap_max_mm=0;
	$seqmap_max_insdel=0;
	}
elsif($slow_annotation)
	{
	$max_miRtail_length=3;
	$intmod_miR=2;
	$seqmap_max_mm=2;
	$seqmap_max_insdel=1;
	}

# optionally skip all mapping steps
if($skip_mapping)
	{
	$skip_ncRNA=1;
	$skip_cDNA=1;
	$skip_miR=1;
	$skip_piR=1;
	}

# check command line arguments
if($max_miRtail_length>5)
	{
	print"\nError. Maximum number of non-template 3' nucleotides =5.\n\n";
	exit;
	}
if($max_miRtail_length+$intmod_miR>5)
	{
	print"\nError. Maximum sum for non-template 3' nucleotides and internal modifications must not exceed 5.\n\n";
	exit;
	}
if($seqmap_max_mm>5)
	{
	print"\nError. Maximum number of mismatches for ncRNA annotation =5.\n\n";
	exit;
	}
if($seqmap_max_insdel>3)
	{
	print"\nError. Maximum number of indels for ncRNA annotation =3.\n\n";
	exit;
	}
if($dust_limit<0.25||$dust_limit>1)
	{
	print"\nError. -dust_limit must be >=0.25 and <=1.";
	exit;
	}
if($dust_min<1)
	{
	print"\nError. -dust_min must be >0";
	exit;
	}
if($dust_min>$dust_max)
	{
	print"\nError. -dust_min must not be larger than -dust_max";
	exit;
	}

$total_number_of_supported_species=keys%species;
if($print_supported_species)
	{
	print"\nSupported species:";
	foreach(sort{$a cmp $b}keys%species)
		{
		next if($_ eq'x'||$_ eq'other');
		print"\n - $_";
		}
	print"\n\nTotal number of supported species (or strains): $total_number_of_supported_species.\nIf your species is not listed here use a closely related species instead.\nAlternatively start unitas with '-species x' or '-species other'\ntogether with your own input reference sequences: '-refseq reference.fasta'.\n\n";
	exit;
	}

# check input files/folders (exist, redundant, identical file names)
%input=();
foreach(@input)
	{
	$input{$_}=1;
	}
@input=keys%input;
@input=sort@input;
@files_only_input=();
%input=();
foreach$file(@input)
	{
	if(-d $file)
		{
		opendir(DIR,$file)||print"\nError. Unable to open directory $file.\n$!\n\n";
		while(readdir DIR)
			{
			if(-f "$file/$_")
				{
				if(!$input{$_})
					{
					$input{$_}=1;
					}
				else
					{
					print"\nError. Do not use files with identical names (from different directories): /$_.\n\n";
					exit;
					}
				push(@files_only_input,"$file/$_");
				}
			}
		close DIR;
		}
	elsif(!-e $file)
		{
		die print"\nError! Input file or folder $file does not exist.\n\n";
		}
	else
		{
		$parsed_filename=$file;
		$parsed_filename=~s/^.*[\\\/]//;
		if(!$input{$parsed_filename})
			{
			$input{$parsed_filename}=1;
			push(@files_only_input,$file);
			}
		else
			{
			print"\nError. Do not use files with identical names (from different directories): /$parsed_filename.\n\n";
			exit;
			}
		}
	}
@input=@files_only_input;
undef%input;
undef@files_only_input;

#check user refseq files (exist, redundant, correct format)
if(@user_refseq>0)
	{
	%user_refseq=();
	foreach(@user_refseq)
		{
		$user_refseq{$_}=1;
		}
	@user_refseq=keys%user_refseq;
	@user_refseq=sort@user_refseq;
	%user_refseq=();
	%refseq_headers=();
	foreach$file(@user_refseq)
		{
		if(!-e$file)
			{
			die print"\nError! Refseq file $file does not exist.\n\n";
			}
		else
			{
			# check format
			$printed_info1=0;
			$printed_info2=0;
			open(REFSEQ,$file)||die print"\nError! Unable to read Refseq file $file.\n$!\n\n";
			$i=0;
			while(<REFSEQ>)
				{
				$_=~s/\s*$//;
				if($_=~s/^>//)
					{
					$i++;
					$head=$_;
					if(length$head>30)
						{
						$head=substr($head,0,27)."...";
						}
					$refseq_headers{$head}++;
					if($refseq_headers{$head}>1)
						{
						$head.="_[$refseq_headers{$head}]";
						if(!$silent&&$printed_info1<10){print"\nNote: Multiple use of FASTA header:\n>$_\nWill change it to: >$head\n";}
						$printed_info1++;
						if($printed_info1==10){print"\n...skip further reports.";}
						}
					if($head!~/^[^\| ]+|[^\| ]+$/)
						{
						if(!$silent&&$printed_info2<10){print"\nError! Bad format in FASTA header of Refseq file $file.\n>$_\n\nYou should use the following format (avoid white space):\n >ncRNA_class|ncRNA_name\n";}
						$new_head=$head;
						$new_head=~s/\|//g;
						$new_head=~s/ /_/g;
						$new_head="refseq|$head";
						if(!$silent){print"Changed FASTA header to: >$new_head\n";}
						$printed_info2++;
						if($printed_info2==10){print"\n...skip further reports.";}
						}
					}
				else
					{
					$_=~tr/uU/tT/;
					$user_refseq{$head}.=uc$_;
					}
				}
			close REFSEQ;
			if(!$silent&&$printed_info1+$printed_info2>0){print"\nMultiple use of FASTA header: $printed_info1 times.\nChanges FASTA header because of bad format: $printed_info2 times.\n";}
			if($i==0)
				{
				if(!$silent){print"\nWarning! No FASTA headers found in Refseq file $file.\n\n";}
				}
			}
		}
	}

# use scientific species name as trivial name for those species that do not have a trivial name.
foreach$species(keys%species)
	{
	if($species{$species}eq'')
		{
		$species{$species}=$species;
		}
	}

# search for supported species
if(!$species{$species})
	{
	print"\n\nERROR: Typo or $species is not a supported species!";
	%suggested_species=();
	$clip=-1;
	while(1)
		{
		$clip++;
		%name_fragments=();
		last if(length$species<6);
		foreach$p(0..$clip)
			{
			$name_fragments{substr($species,$p,(length$species)-$clip)}=1;
			}
		foreach$fragment(keys%name_fragments)
			{
			foreach$sspec(keys%species)
				{
				if($sspec=~/$fragment/)
					{
					$suggested_species{$sspec}=1;
					}
				}
			}
		last if (keys%suggested_species>0);
		undef%name_fragments;
		}
	if(keys%suggested_species>0)
		{
		print"\n\nDid you mean:";
		foreach(keys%suggested_species)
			{
			print"\n  $_";
			}
		}
	else
		{
		print"\nNo fuzzy match to any supported species found.";
		}
	print"\n\nYou can start unitas with the option -supp_spec to print a complete\nlist of supported species.\n\n";
	exit;
	}

if(!$silent){print"

                                                          X
                                                          oXX.o
                                                         .XXXX
                                                        .XXXXX
                                                       .XXXXX.
                                                      .XXXXXXXXXo
                                                     oXXXXXX' 'XXX.
            XXX                                   .oXXXXXXXX. .XXXXo
           XXXXX.  .oXXXXoo..                 ...oXXXXXXXXXXXXXXXXXX
            XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXo'''
               XXXXXXXXXXXXXXX------------------XXXXXXXXXXXX'
           .X.   'XXXXXXXXXXXX      unitas      XXXXXXXXXXXXX
         XXXXXXooXXXXXXXXXXXXX------------------XXXXXXXXXXXXXoooooo...XXXX
       .XXXX'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'''XXXXXXXXXXXXXXX''
    .oXXXX'    ''XXXXXXXXXXX' 'XXXXXXXXXXXXXXX'''''        '''       ''
   oXXXX'          'XXXXX''
   XXXX'


 Version: ..... $unitas_version
 Date: ........ $last_modified

 Input file(s):";}
foreach$file(@input)
	{
	if(!$silent){print"\n   - $file";}
	}
if(!$silent){print
"

 Parallel threads: ................ $threads
 Species: ......................... $species -> $species{$species}
 Max. miR-tail length: ............ $max_miRtail_length
 Max. internal miR modifications .. $intmod_miR
 Max. mismatch (not miR): ......... $seqmap_max_mm
 Max. indels (not miR): ........... $seqmap_max_insdel

";}

# prepare information on data base release or date
$db_versions_info="";
($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst)=localtime(time);
$year+=1900;
$mon++;
$mday=substr("0".$mday,-2);
$mon=substr("0".$mon,-2);

# load reference sequences and quit if the user only wants to download reference sequences for later offline runs
if($refdump_and_quit)
	{
	if(!-d"UNITAS_refdump_$species"&&$species{$species} ne 'undefined')
		{
		mkdir("UNITAS_refdump_$species")||die print"\nError. Unable to create dump folder for reference sequences.\n$!\n\n";
		}
	$download_reference=1;
	load_reference_sequences();
	exit;
	}

# download reference sequences to reference dump folder if option latestref is used, this folder does not exist or does not contain all required files
if($species{$species} eq 'undefined')
	{
	if(@user_refseq==0)
		{
		print"\nError: Use -species x only together with -refseq [your_reference.file]\n\n";exit;
		}
	$skip_miR=1;
	$skip_cDNA=1;
	if(!-e"UNITAS_refdump_$species")
		{
		mkdir("UNITAS_refdump_$species")||die print"\nError. Unable to create dump folder for reference sequences.\n$!\n\n";
		}
	print"\nCreate file with complete user refseq data...";
	open(USERREFSEQ,">UNITAS_refdump_$species/ncRNA.fas")||die print"\nUnable to create reference file UNITAS_refdump_$species/ncRNA.fas.\n$!\n\n";
	foreach$head(sort{$a cmp $b}keys%user_refseq)
		{
		print USERREFSEQ">$head\n$user_refseq{$head}\n";
		}
	close USERREFSEQ;
	print" done.";
	}
elsif(!-d"UNITAS_refdump_$species"&&$species{$species} ne 'undefined')
	{
	mkdir("UNITAS_refdump_$species")||die print"\nError. Unable to create dump folder for reference sequences.\n$!\n\n";
	$download_reference=1;
	}
elsif($latestref)
	{
	$download_reference=1;
	}
else
	{
	if(!-e"UNITAS_refdump_$species/db_versions.info"||!-e"UNITAS_refdump_$species/ncRNA.fas"||!-e"UNITAS_refdump_$species/tRF-1.fas"||!-e"UNITAS_refdump_$species/tRNA_leader.fas"||!-e"UNITAS_refdump_$species/miRBase_mature"||!-e"UNITAS_refdump_$species/miRBase_hairpin_$species{$species}.fas"||!-e"UNITAS_refdump_$species/miRBase_hairpin_non-$species{$species}.fas")
		{
		print"\nAttention: At least one reference file is missing in folder UNITAS_refdump_$species.\nWill download latest reference sequences.\n";
		$download_reference=1;
		}
	elsif(!$skip_cDNA&&!-e"UNITAS_refdump_$species/cDNA.fas")
		{
		print"\nAttention: Reference file cDNA.fas missing in folder UNITAS_refdump_$species.\nWill download latest reference sequences.\n";
		$download_reference=1;
		}
	else
		{
		# UNITAS_refdump folder contains the required files, no download necessary
		open(DBVERSIONS,"UNITAS_refdump_$species/db_versions.info");
		while(<DBVERSIONS>)
			{
			$db_versions_info.=$_;
			}
		$download_reference=0;
		if(!$silent){print"\nFound valid UNITAS_refdump folder. No downloads required.\nUsing the option -latest_ref will force unitas to download latest\nreference sequences.\n";}
		}
	}

# download NCBI rRNA sequence collection (if it does not exist in the current working directory)
if($riborase&&!-e"NCBI_rRNA_collection.fasta")
	{
	$NCBI_rRNA_collection='https://www.smallrnagroup.uni-mainz.de/data/UNITAS/NCBI_rRNA_collection.zip';
	if(!$silent){print"\nDownload NCBI rRNA sequence collection...";}
	$request = HTTP::Request->new(GET => "$NCBI_rRNA_collection");
	$response = $useragent->request($request);
	$content = $response->content;
	open(my $out, '>', "NCBI_rRNA_collection.zip");
	print $out "$content";
	close $out;
	if(-e"NCBI_rRNA_collection.zip")
		{
		if(!$silent){print" OK.";}
		if(!$silent){print"\nExtract file...";}
		$ae=Archive::Extract->new(archive=>"NCBI_rRNA_collection.zip");
		$ae->extract||print"\nERROR! Unable to decompress NCBI_rRNA_collection.zip.\n$!\n\n";
		unlink"NCBI_rRNA_collection.zip"||print"\nERROR! Unable to remove compressed file NCBI_rRNA_collection.zip.\n$!\n\n";
		if(!-e"NCBI_rRNA_collection.fasta")
			{
			if(!$silent){print"\nFailed to download and extract NCBI rRNA sequence collection!";}
			undef$riborase;
			}
		else
			{
			if(!$silent){print" OK.";}
			push(@user_refseq,"NCBI_rRNA_collection.fasta");
			}
		}
	else
		{
		if(!$silent){print" FAILED! $status\n\n";}
		}
	}

# download SeqMap source and compile it (if executable is not present in current working directory)
$SeqMap_source='https://www.smallrnagroup.uni-mainz.de/data/UNITAS/SeqMap/seqmap-1.0.13-src.zip';
$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/SeqMap/release');
$response = $useragent->request($request);
$version = $response->content;
$db_versions_info.="SeqMap version/date: ...................... $version\n";
if(!-e"seqmap.exe")
	{
	if(!$silent){print"\nDownload SeqMap source code...";}
	$request = HTTP::Request->new(GET => "$SeqMap_source");
	$response = $useragent->request($request);
	$content = $response->content;
	open(my $out, '>', "UNITAS_refdump_$species/SeqMap.zip");
	print $out "$content";
	close $out;
	if(-e"UNITAS_refdump_$species/SeqMap.zip")
		{
		print" OK.";
		}
	else
		{
		print" FAILED! $status\n\n";
		}
	if(!$silent){print"\nExtract file...";}
	$ae=Archive::Extract->new(archive=>"UNITAS_refdump_$species/SeqMap.zip");
	$ae->extract(to=>"UNITAS_refdump_$species")||die print"\nERROR! Unable to decompress UNITAS_refdump_$species/SeqMap.zip.\n$!\n\n";
	if(!$silent){print" OK.\nRemove compressed file...";}
	unlink"UNITAS_refdump_$species/SeqMap.zip"||print"\nERROR! Unable to remove compressed file UNITAS_refdump_$species/SeqMap.zip.\n$!\n\n";
	if(!$silent){print" OK.\nCompile SeqMap source code... ";}
	system("g++ -O3 -m64 -o seqmap.exe UNITAS_refdump_$species/seqmap-1.0.13-src/match.cpp");
	if(-e'seqmap.exe')
		{
		if(!$silent){print" OK.";}
		}
	else
		{
		print"\nERROR! No success when building SeqMap from source. Will download pre-compiled SeqMap version from smallRNAgroup-SERVER.\n";
		if($^O eq'linux')
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/software/Linux/seqmap.exe');
			$response = $useragent->request($request);
			$content = $response->content;
			open(my $out, '>', "seqmap.exe");
			print $out "$content";
			close $out;
			if(-e"seqmap.exe")
				{
				# set file permission to executable
				chmod(0755,"seqmap.exe");
				print" OK.\nCheck executability (file permission)...";
				if(-x"seqmap.exe")
					{
					print" OK.";
					}
				else
					{
					print" Permission denied.\nYou have to change file permission for file 'seqmap.exe' manually. Try the following command:\n\n chmod 755 seqmap.exe\n\nThen restart unitas.\n\n";
					exit;
					}
				}
			else
				{
				die print" FAILED! $status\n\n";
				}
			}
		elsif($^O eq'darwin')
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/software/MacOS/seqmap.exe');
			$response = $useragent->request($request);
			$content = $response->content;
			open(my $out, '>', "seqmap.exe");
			print $out "$content";
			close $out;
			if(-e"seqmap.exe")
				{
				# remove apple quarantine and set file permission to executable
				system("xattr -d com.apple.quarantine seqmap.exe");
				chmod(0755,"seqmap.exe");
				print" OK.\nCheck executability (file permission)...";
				if(-x"seqmap.exe")
					{
					print" OK.";
					}
				else
					{
					print" Permission denied.\nYou have to change file permission for file 'seqmap.exe' manually. Try the following command:\n\n chmod 755 seqmap.exe\n\nThen restart unitas.\n\n";
					exit;
					}
				}
			else
				{
				die print" FAILED! $status\n\n";
				}
			}
		elsif($^O eq'MSWin32')
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/software/Windows/seqmap.exe');
			$response = $useragent->request($request);
			$content = $response->content;
			open(my $out, '>', "seqmap.exe");
			print $out "$content";
			close $out;
			if(-e"seqmap.exe")
				{
				print" OK.";
				}
			else
				{
				die print" FAILED! $status\n\n";
				}
			}
		}
	}

load_reference_sequences();

# prepare a list with refseq files provided by the user
$list_user_refseq="";
if(@user_refseq>0)
	{
	$list_user_refseq="\nAdditional files with reference sequences:";
	foreach(@user_refseq)
		{
		$list_user_refseq.="\n -$_";
		}
	$list_user_refseq.="\n";
	}
elsif($species{$species}eq'undefined')
	{
	print"\nError! You must provide reference sequence file(s) when using undefined species.\n\n";
	exit;
	}


if(!$silent){print"

####################################################################

The following sources will be used for small RNA annotation:

$db_versions_info
$list_user_refseq
PLEASE MAKE SURE TO CITE THE ACCORDING DATABASE PAPERS WHEN
PUBLISHING THESE RESULTS!

####################################################################

";}

@childs=();
$parent_process_id=$$;
foreach$input(@input)
	{
	# fork only when less than $threads pid-files exist in the current directory.
	while(1)
		{
		sleep(1);
		@pid_files=glob("*.$parent_process_id.unitas_pid");
		if(@pid_files<$threads)
			{
			$pid=fork;
			if($pid)
				{
				push(@childs,$pid);
				}
			elsif($pid==0)
				{
				open(WIZ_PID,">$$.$parent_process_id.unitas_pid");
				close WIZ_PID;
				process_file();
				unlink("$$.$parent_process_id.unitas_pid")||print"\nError. Unable to remove $$.$parent_process_id.unitas_pid.\n$!\n\n";
				exit 0;
				}
			else
				{
				die print"\nError. Unable to fork annotation of $input.\n\n";
				}
			last;
			}
		}
	}
foreach$pid(@childs)
	{
	waitpid($pid,0);
	}

sub process_file
	{
	# declare hashes that are valid only in this subroutine
	my%input;
	my%extracted_sequences;
	my%input_map_seq;
	my%miR_species;
	my%miR_nonspecies;
	my%tails_per_miR_species;
	my%tails_per_miR_nonspecies;
	my%miRBase_species;
	my%miRBase_nonspecies;
	my%miR_canonical_species;
	my%miR_canonical_nonspecies;
	my%output_miR_species;
	my%output_miR_nonspecies;
	my%count_internal_mods_species;
	my%count_internal_mods_nonspecies;
	my%non_annotated;
	my%valid_species_hits;
	my%tails_species;
	my%tails_nonspecies;
	my%tails_per_miR_species;
	my%tails_per_miR_nonspecies;
	my%miR_table_out_species;
	my%miR_table_out_nonspecies;
	my%miR_noncanonical_species;
	my%miR_noncanonical_nonspecies;
	my%miR_spec_seq;
	my%miR_nonspec_seq;
	my%nonmiR_seq;
	my%tRF_seqs;
	my%tRNA_half5;
	my%tRF5;
	my%tRNA_half3;
	my%tRF3_CCA;
	my%tRF3;
	my%misc_tRF;
	my%ncRNA_counts_1; # class only
	my%ncRNA_counts_2; # class and id
	my%sorted_sequences;
	my%phasiRNA_readcount;
	my%phasi_positions;
	my%sense_antisense_counts;
	my%global_annotation_matrix;

	# create data dump folder and copy reference sequences
	$i=0;
	$parsed_input=$input;
	$parsed_input=~s/^.*[\\\/]//;
	while(1)
		{
		$i++;
		$dump_folder=$dump_folder_prefix."_$mday-$mon-$year"."_$parsed_input"."_#$i";
		if(-d$dump_folder)
			{
			next;
			}
		else
			{
			if(!$silent){print"\nCreate data dump folder $dump_folder...";}
			mkdir($dump_folder)||die print"\nERROR! Unable to create folder.\n$!\n\n";
			copy("UNITAS_refdump_$species/ncRNA.fas","$dump_folder/ncRNA.fas")||die print"\nERROR! Unable to copy UNITAS_refdump_$species/ncRNA.fas to dump folder.\n$!\n\n";
			if(!$skip_cDNA&&$species{$species}ne'undefined')
				{
				copy("UNITAS_refdump_$species/cDNA.fas","$dump_folder/cDNA.fas")||die print"\nERROR! Unable to copy UNITAS_refdump_$species/cDNA.fas to dump folder.\n$!\n\n";
				}
			# add user RefSeq files to ncRNA.fas
			if(@user_refseq>0)
				{
				$user_refseq_added=0;
				if(!$silent){print"\nAdd sequences from files defined by user...";}
				open(NCRNA,">>$dump_folder/ncRNA.fas")||print"\nError! Failed to add user defined RefSeq to file $dump_folder/ncRNA.fas.\n$!\n\n";
				foreach$header(keys%user_refseq)
					{
					$user_refseq_added++;
					print NCRNA">$header\n$user_refseq{$header}\n";
					}
				close NCRNA;
				if(!$silent){print" OK. Added $user_refseq_added sequences.";}
				}
			if($species{$species}ne'undefined')
				{
				copy("UNITAS_refdump_$species/miRBase_mature","$dump_folder/miRBase_mature")||die print"\nERROR! Unable to copy UNITAS_refdump_$species/miRBase_mature to dump folder.\n$!\n\n";
				copy("UNITAS_refdump_$species/miRBase_hairpin_$species{$species}.fas","$dump_folder/miRBase_hairpin_$species{$species}.fas")||die print"\nERROR! Unable to copy UNITAS_refdump_$species/miRBase_hairpin_$species{$species}.fas to dump folder.\n$!\n\n";
				copy("UNITAS_refdump_$species/miRBase_hairpin_non-$species{$species}.fas","$dump_folder/miRBase_hairpin_non-$species{$species}.fas")||die print"\nERROR! Unable to copy UNITAS_refdump_$species/miRBase_hairpin_non-$species{$species}.fas to dump folder.\n$!\n\n";
				}
			if(-e"UNITAS_refdump_$species/piRclDB")
				{
				copy("UNITAS_refdump_$species/piRclDB","$dump_folder/piRclDB")||die print"\nERROR! Unable to copy UNITAS_refdump_$species/piRclDB to dump folder.\n$!\n\n";
				}
			if(!$silent){print" OK.";}
			last;
			}
		}

	# check total amount of reference sequence to estimate time to be used for SeqMap mapping
	$total_ncRNA_reference_bp=0;
	if(!$silent){print"\nCheck total amount of ncRNA reference sequence (excl. miRs)...";}
	open(REF,"$dump_folder/ncRNA.fas")||die print"\nERROR! Unable to open $dump_folder/ncRNA.fas.\n$!\n\n";
	while(<REF>)
		{
		if($_!~/^>/)
			{
			$_=~s/\s*$//;
			$total_ncRNA_reference_bp+=length$_;
			}
		}
	close REF;
	if(!$silent){print" $total_ncRNA_reference_bp bp.";}

	# check input file
	if(!$silent){print"\nCheck input file format...";}
	if(!-e$input)
		{
		die print"\nERROR! Input file $input not found.\n\n";
		}
	else
		{
		open(INPUT,$input)||die print"\nERROR! Unable to open input file $input.\n$!\n\n";
		$i=0;
		$fasta_heads=0;
		$fastq_heads=0;
		$sam_lines=0;
		$eland3_lines=0;
		$collapsed='collapsed ';
		$head_is_over=0;
		while(<INPUT>)
			{
			if($_!~/^trans_id/&&$_!~/^[#\@]/)# skip all the header lines in map files
				{
				$head_is_over=1;
				}
			next if($head_is_over==0);
			$_=~s/\s*$//;
			@d=split("\t",$_);
			$i++;
			last if($i==1000); # check 1000 lines after the header section
			if($_=~/^>/)
				{
				if($_!~/^>\s*\d+\s*$/)
					{
					$collapsed='uncollapsed ';
					}
				$fasta_heads++;
				}
			elsif($_=~/^@/)
				{
				if($_!~/^@\s*\d+\s*$/)
					{
					$collapsed='uncollapsed ';
					}
				$fastq_heads++;
				}
			elsif(looks_like_number($d[1])&&$d[1]>-1&&$d[1]<2**16&&$d[3]=~/^\d+$/&&$d[9]=~/^[ATGCUN]+$/)
				{
				$sam_lines++;
				}
			elsif($d[1]=~/^\d+$/&&$d[2]=~/^[ATGCUN]+$/&&$d[4]=~/^[ATGCUN]+$/&&$d[5]=~/^\d+$/&&$d[6]=~/^[\+-]$/)
				{
				$eland3_lines++;
				}
			}
		close INPUT;
		if($fasta_heads>$fastq_heads&&$fasta_heads>$sam_lines&&$fasta_heads>$eland3_lines)
			{
			$format='FASTA';
			}
		elsif($fastq_heads>$fasta_heads&&$fastq_heads>$sam_lines&&$fastq_heads>$eland3_lines)
			{
			$format='FASTQ';
			}
		elsif($sam_lines>$eland3_lines)
			{
			$format='SAM';
			$collapsed='';
			}
		elsif($eland3_lines>$sam_lines)
			{
			$format='ELAND3';
			$collapsed='';
			}
		else
			{
			print"\nERROR! Unknown file format of input file $input.\nUse FASTA, FASTQ, SAM or ELAND3 (from sRNAmapper: https://www.smallrnagroup.uni-mainz.de/software.html).\n\n";
			exit;
			}
		if(!$silent){print" ${collapsed}$format.";}
		close INPUT;

		if(@phasi>0&&$format ne'SAM'&&$format ne'ELAND3')
			{
			print"\nERROR! When searching for phased RNAs the input file must be a map file in SAM or ELAND3 format (ELAND3 is standard output format of sRNAmapper: https://www.smallrnagroup.uni-mainz.de/software.html).\n\n";
			exit;
			}
		if($check_pp_signature)
			{
			if($format ne'SAM'&&$format ne'ELAND3')
				{
				print"\nERROR! When searching for ping pong signatures the input file must be a map file in SAM or ELAND3 format (ELAND3 is standard output format of sRNAmapper: https://www.smallrnagroup.uni-mainz.de/software.html).\n\n";
				exit;
				}
			}
		if($do_trimming)
			{
			if($format eq'SAM'||$format eq'ELAND3')
				{
				print"\nNOTE! Do not use the trimming option (-trim) when the input file is a map file.\n\n";
				undef$do_trimming;
				}
			}
		}

	# convert FASTQ/SAM/ELAND3 to FASTA (SeqMap accepts FASTA only)
	if($format eq'FASTQ')
		{
		if(!$silent){print"\nConvert input file to FASTA...";}
		open(FASTQ,$input)||die print"\nUnable to open input file $input.\n$!\n\n";
		open(FASTA,">$dump_folder/$parsed_input.fas")||die print"\nError! Unable to create output file $dump_folder/$parsed_input.fas.\n$!\n\n";
		$i=0;
		@seq_data=();
		while(<FASTQ>)
			{
			$_=~s/\s*$//;
			push(@seq_data,$_);
			$i+=0.25;
			if($i==1)
				{
				$i=0;
				$seq_data[0]=~s/^\@/>/;
				if(length$seq_data[1]>0)
					{
					print FASTA"$seq_data[0]\n$seq_data[1]\n";
					}
				@seq_data=();
				}
			}
		close FASTA;
		close FASTQ;
		$parsed_input.=".fas";
		$input_for_DNApi=$input;
		$input="$dump_folder/$parsed_input";
		if(!$silent){print" OK.";}
		}
	elsif($format eq'SAM')
		{
		$input_map=$input;
		if(!$silent){print"\nExtract sequences from $input...";}
		# save flags that refer to '-' or '+' strand and mapped/unmapped reads in SAM files
		foreach$bit1(0,1)
		{foreach$bit2(0,2)
		{foreach$bit3(0,4)
		{foreach$bit4(0,8)
		{foreach$bit5(0,16)
		{foreach$bit6(0,32)
		{foreach$bit7(0,64)
		{foreach$bit8(0,128)
		{foreach$bit9(0,256)
		{foreach$bit10(0,512)
		{foreach$bit11(0,1024)
		{foreach$bit12(0,2048)
			{
			$flag=$bit1+$bit2+$bit3+$bit4+$bit5+$bit6+$bit7+$bit8+$bit9+$bit10+$bit11+$bit12;
			if($bit3==4)
				{
				$sam_mapped{$flag}=0;
				}
			else
				{
				$sam_mapped{$flag}=1;
				}
			if($bit5==16)
				{
				$sam_strand{$flag}='-';
				}
			else
				{
				$sam_strand{$flag}='+';
			}}}}}}}}}}}}}
		open(SAM,$input)||die print"\nERROR! Unable to open input file $input.\n$!\n\n";
		$extracted_sequences_nonid=0;
		$extracted_sequence_reads=0;
		$all_identifiers_are_numeric=1;
		while(<SAM>)
			{
			next if($_=~/^@/); # skip SAM header lines
			@sam=split("\t",$_);
			next if($sam_mapped{$sam[1]}==0||$sam[3]==0); # skip unmapped reads
			if(!looks_like_number($sam[0]))
				{
				$all_identifiers_are_numeric=0;
				}

			# reverse complement if sequence was mapped to the minus strand (SAM gives sequence of the locus, not of the mapped sequence)
			if($sam_strand{$sam[1]}eq'-')
				{
				$sam[9]=reverse$sam[9];
				$sam[9]=~tr/ATGC/TACG/;
				}

			$extracted_sequences{$sam[9]}{$sam[0]}++;
			}
		close SAM;

		# make a collapsed FASTA file from map file
		fasta_from_map_file();
		sub fasta_from_map_file
			{
			open(EXTRACTED,">$dump_folder/$parsed_input.mapped_sequences")||die print"\nError! Unable to create output file $dump_folder/$parsed_input.mapped_sequences.\n$!\n\n";
			if($all_identifiers_are_numeric==0)
				{
				foreach$seq(keys%extracted_sequences)
					{
					next if(length$seq==0);
					$read_counts=keys%{$extracted_sequences{$seq}};
					print EXTRACTED">$read_counts\n$seq\n";
					$input_map_seq{$seq}=$read_counts;
					$extracted_sequence_reads+=$read_counts;
					$extracted_sequences_nonid++;
					}
				}
			elsif($all_identifiers_are_numeric==1)
				{
				foreach$seq(keys%extracted_sequences)
					{
					next if(length$seq==0);
					foreach$val(keys%{$extracted_sequences{$seq}}) # there will be only one key!
						{
						$read_counts=$val;
						}
					print EXTRACTED">$read_counts\n$seq\n";
					$input_map_seq{$seq}=$read_counts;
					$extracted_sequence_reads+=$read_counts;
					$extracted_sequences_nonid++;
					}
				}
			$input="$dump_folder/$parsed_input.mapped_sequences";
			$collapsed='collapsed ';
			if(!$silent){print" done. $extracted_sequences_nonid non-identical sequences represented by $extracted_sequence_reads reads.";}
			if(@phasi==0&&!$check_pp_signature)
				{
				undef%extracted_sequences;
				undef%input_map_seq;
				}
			close EXTRACTED;
			}
		}
	elsif($format eq'ELAND3')
		{
		$input_map=$input;
		if(!$silent){print"\nExtract sequences from $input...";}
		open(ELAND3,$input)||die print"\nERROR! Unable to open input file $input.\n$!\n\n";
		%extracted_sequences=();
		$all_identifiers_are_numeric=1;
		while(<ELAND3>)
			{
			next if($_=~/^trans_id/||$_=~/^#/);
			@eland3=split("\t",$_);
			if(!looks_like_number($eland3[3]))
				{
				$all_identifiers_are_numeric=0;
				}
			$extracted_sequences{$eland3[4]}{$eland3[3]}++;
			}
		close ELAND3;
		fasta_from_map_file(); # see above (SAM)
		}

	# remove 3' adapter sequence
	if($do_trimming)
		{
		if($do_trimming=~/^[ATGCatgc]{5,}$/) # use pre-defined adapter sequence at least 5 nt in length
			{
			$adapter_to_clip=uc$do_trimming;
			}
		elsif($do_trimming eq'?') # search putative adapter sequence in input file
			{
			if(!$silent){print"\nSearch and remove 3' adapter sequences.";}
			my %motifs=();
			my %length=();
			my %new_3p=();
			my %trimmed_out=();

			# check for read length in input file, adjust adapter motif length
			open(TRIM,$input)||die print"\nERROR! Unable to open input file $input for trimming.\n$!\n\n";
			if(!$silent){print"\nCheck read length in input file (first $check_seq_for_size reads)...";}
			$i=0;
			%length=();
			while(<TRIM>)
				{
				if($_!~/^>/)
					{
					$i++;
					$_=~s/\s*$//;
					$length{length$_}++;
					last if($i==$check_seq_for_size);
					}
				}
			close TRIM;
			if(!$silent){print" OK.";}
			foreach$l(sort{$length{$a}<=>$length{$b}}keys%length)
				{
				$length=$l;
				if(!$silent){print"\n - length:$l nt -> $length{$l} reads.";}
				}
			if(keys%length==1)
				{
				if(!$silent){print"\nLooks good (untrimmed). All reads have equal length.\n";}
				$init_motif_length=$length-$ignore_5p;
				$min_motif_length=6;
				if($init_motif_length>12)
					{
					$init_motif_length=12;
					}
				if($init_motif_length<6)
					{
					$init_motif_length=5;
					$min_motif_length=4;
					$ignore_5p=$length-8;
					print"\n
  ------------------------------------------------------------------------
  WARNING: Reads in input file are very short. Searching for 3' adapter
  sequences is challenging, considering the typical miRNA length of 22 nt.
  Parameters are modified to search for frequent $init_motif_length nt motifs in last 8 nt
  of the reads. The final minimum motif length at 3' ends is set to $min_motif_length nt.

  NOTE:
  You may also consider to map your reads using a popular aligner such as
  STAR, bowtie or bowtie2, allowing softclipping at the 3' end. Then you
  can use the map file (SAM format) as input for unitas. We recommend this
  procedure for all unclipped sRNA datasets with sequence length <=26 nt.
  ------------------------------------------------------------------------

	";
					}
				if($init_motif_length<$min_motif_length)
					{
					$min_motif_length=$init_motif_length;
					}
				}
			else
				{
				if(!$silent){print"\nNOTE: Input file looks trimmed (found sequences with different length)!\nContinue nevertheless.\n";}
				$init_motif_length=12;
				$min_motif_length=6;
				}

			# search adapter sequence with DNApi
			$dnapi_success=0;
			if($format eq "FASTQ" && -e "dnapi.py" && -x "dnapi.py")
				{
				if(!$silent){print"\nSearch for adapter sequence with DNApi...";}
				if($^O eq'linux'||$^O eq'darwin')
					{
					qx/python dnapi.py $input_for_DNApi > $dump_folder\/adapter.dnapi/;
					}
				elsif($^O eq'MSWin32')
					{
					system("python dnapi.py $input_for_DNApi > $dump_folder/adapter.dnapi");
					}
				else
					{
					if(!$silent){print" WARNING! Unknown operating system.";}
					# try unix command
					qx/python dnapi.py $input_for_DNApi > $dump_folder\/adapter.dnapi/;
					if(!-e"$dump_folder/adapter.dnapi")
						{
						# try windows command
						system("python dnapi.py $input_for_DNApi > $dump_folder/adapter.dnapi");
						}
					}
				if(!-e"$dump_folder/adapter.dnapi")
					{
					if(!$silent){print"\nERROR. Tried to execute DNApi but cannot find its output file.\nWill use internal algorithm to search for 3' adapter sequence.";}
					}
				else
					{
					open(DNAPI,"$dump_folder/adapter.dnapi");
					while(<DNAPI>)
						{
						$_=~s/\s*$//;
						if($_=~/^[ATGC]+$/)
							{
							$dnapi_success=1;
							$adapter_abundance="??";
							$adapter_to_clip=$_;
							last;
							}
						}
					close DNAPI;
					if($dnapi_success==0)
						{
						if(!$silent){print"\nERROR. DNApi did not run properly (missing file?).\nWill use internal algorithm to search for 3' adapter sequence.";}
						}
					}
				}

			# check for motif (internal algorithm)
			if($dnapi_success==0)
				{
				search_motif_to_trim();
				sub search_motif_to_trim
					{
					$i=0;
					$j=0;
					open(TRIM,$input)||die print"\nERROR. Unable to open input file $input.\n$!\n\n";
					if(!$silent){print"\nSearch for frequent 3' motifs ($init_motif_length nt) in input file...";}
					while(<TRIM>)
						{
						if($_!~/^>/)
							{
							$i++;
							$_=~s/\s*$//;
							next if($i<=$skip_first_X_reads);
							$j++;
							last if($i==$check_seq+$skip_first_X_reads);
							$_=~s/^.{$ignore_5p}//;
							while(length$_>$init_motif_length)
								{
								$motifs{substr$_,0,$init_motif_length}++;
								$_=~s/.//;
								}
							}
						}
					close TRIM;
					}

				# repeat search without skipping first X reads if there are not enough reads in the input file
				if($j==0)
					{
					$skip_first_X_reads=0;
					search_motif_to_trim();
					}

				foreach$seq(sort{$motifs{$b}<=>$motifs{$a}}keys%motifs)
					{
					next if($seq=~/N/);
					next if($seq=~/^A+$/&&!$trim_polyA);
					next if($seq=~/^T+$/);
					next if($seq=~/^G+$/);
					next if($seq=~/^C+$/);
					$adapter_to_clip=$seq;
					$adapter_abundance=(int((($motifs{$seq}/$i)*10000)+0.5))/100;
					last;
					}
				if(!$silent){print" OK.";}
				if($adapter_abundance<$clipped_fraction_warning*100)
					{
					if(!$silent){print"\nNOTE: Fraction of reads containing putative adapter sequence is low!\nContinue nevertheless.\n";}
					}
				}
			}
		else{print"\n\nERROR. Predefined adapter sequence should not be shorter than 5 nt.\nUse '?' to let unitas search for unknown adapter sequences.\n\n";exit}

		# 1st round of motif clipping ("KNIPSER" by Daniel Gebert)
		open(TRIM,$input)||die print"\nERROR. Unable to open input file $input.\n$!\n\n";
		open(OUT,">$dump_folder/$parsed_input.trimmed")||die print"\nERROR. Unable to open output file $dump_folder/$parsed_input.trimmed.\n$!\n\n";
		if(!$silent&&$do_trimming eq'?'){print"\nClip most abundant sequence ($adapter_abundance%): $adapter_to_clip-(N)x-3'...";}
		elsif(!$silent&&$do_trimming=~/^[ATGCatgc]{5,}$/){print"\nClip pre-defined adapter sequence: $adapter_to_clip-(N)x-3'...";}
		$j=0; # fasta heads
		$k=0; # trimmed
		$k_ok=0; # trimmed, length ok
		$win_start=$min_size;
		while($line=<TRIM>)
			{
			$line=~s/\s*$//;
			if($line=~s/^>//)
				{
				$j++;
				$header = $line;
				}
			else
				{
				# Get sequence, adapter motif and max. Hamming Distance
				my $seq = $line;
				my $motif = $adapter_to_clip;
				my $hd_max = $adapter_mismatch;
				# Slide window through sequence and search for motif
				for (my $i=$win_start; $i <= length($seq)-$min_motif_length; $i++)
					{
						# Chop motif if window hits end of sequence
						if ($i > length($seq)-length($motif)) {
							chop($motif);
						}
						# Get window sequence
						my $window = substr($seq, $i, length $motif);
						# Check if motif matches window
						if ($window eq $motif) {
							$k++;
							if($i<=$max_size){
							$k_ok++;
							# Save clipped sequence and leave loop
							$clip_seq = substr($seq, 0, $i);
							$trimmed_out{$clip_seq}.="##$header";
							check_new_3p();
							last;
							}
						}
						# If no exact match found and mismatches allowed
						elsif ($hd_max > 0) {
							# Reduce max. Hamming Distance if motif chopped
							if ($motif ne $adapter_to_clip) { $hd_max = 1 }
							if (length($motif) < 9) { $hd_max = 0 }
							# Get Hamming Distance
							my $hd = ($window ^ $motif) =~ tr/\0//c;
							if ($hd <= $hd_max) {
								$k++;
								if($i<=$max_size){
								$k_ok++;
								# Save clipped sequence and leave loop
								$clip_seq = substr($seq, 0, $i);
								$trimmed_out{$clip_seq}.="##$header";
								check_new_3p();
								last;
								}
							}
						}
					}
				}
			}
		close TRIM;
		sub check_new_3p
			{
			foreach$p(1..$post_check_3p_ends)
				{
				$new_3p{-$p}{substr($clip_seq,-$p,1)}++;
				}
			}

		if(!$silent){print" OK.\n\nTotal processed reads: $j\nClipped reads: $k\n$min_size-$max_size nt: $k_ok\n";}
		if($k/$j<$clipped_fraction_warning)
			{
			if(!$silent){print"\nNOTE: Fraction of clipped reads is low!\nContinue nevertheless.\n";}
			}

		# refinement
		$i=0;
		$nibble=0;
		$nibble_motif="";
		if(!$silent){print"\nCheck new 3' ends for bias in positional nucleotide composition.\nNucleotide compositions of new 3' ends [%]:";}
		foreach$p(1..$post_check_3p_ends)
			{
			$i++;
			$N=$new_3p{-$p}{'A'}+$new_3p{-$p}{'T'}+$new_3p{-$p}{'G'}+$new_3p{-$p}{'C'};
			$A_perc=(int((($new_3p{-$p}{'A'}/$N)*10000)+0.5))/100;
			$T_perc=(int((($new_3p{-$p}{'T'}/$N)*10000)+0.5))/100;
			$G_perc=(int((($new_3p{-$p}{'G'}/$N)*10000)+0.5))/100;
			$C_perc=(int((($new_3p{-$p}{'C'}/$N)*10000)+0.5))/100;
			if(!$silent){print"\nPosition -$p: A:$A_perc T:$T_perc G:$G_perc C:$C_perc";}
			if($new_3p{-$p}{'A'}/$N>$refinement_threshold)
				{
				$nibble++;
				$nibble_motif="A".$nibble_motif;
				if(!$silent){print"\tBAD! Too much 'A'!";}
				}
			elsif($new_3p{-$p}{'T'}/$N>$refinement_threshold)
				{
				$nibble++;
				$nibble_motif="T".$nibble_motif;
				if(!$silent){print"\tBAD! Too much 'T'!";}
				}
			elsif($new_3p{-$p}{'G'}/$N>$refinement_threshold)
				{
				$nibble++;
				$nibble_motif="G".$nibble_motif;
				if(!$silent){print"\tBAD! Too much 'G'!";}
				}
			elsif($new_3p{-$p}{'C'}/$N>$refinement_threshold)
				{
				$nibble++;
				$nibble_motif="C".$nibble_motif;
				if(!$silent){print"\tBAD! Too much 'C'!";}
				}
			else
				{
				if(!$silent){print"\tOK.";}
				last;
				}
			}
		if($i==$post_check_3p_ends)
			{
			if(!$silent){print"\nNOTE: Sequence file needs attention!\nLow heterogenety in last $post_check_3p_ends nucleotides!\nTrimming is presumably not correct!\nContinue nevertheless.\n\n";}
			}
			$i=0;
		if($nibble==0)
			{
			if(!$silent){print"\n\n3' ends of processed sequences look fine.\nPrint sequences to $input.trimmed...";}
			foreach$seq(keys%trimmed_out)
				{
				$trimmed_out{$seq}=~s/^##//;
				@titles=split('##',$trimmed_out{$seq});
				foreach$title(@titles)
					{
					$i++;
					print OUT">$title\n$seq\n";
					}
				}
			if(!$silent){print" OK.";}
			}
		else
			{
			if(!$silent){print"\n\n3' ends of processed sequences need further refinement.\nRemove $nibble_motif-3' and print sequences to $input.trimmed...";}
			foreach$seq(keys%trimmed_out)
				{
				$trimmed_seq=$seq;
				if($trimmed_seq=~s/$nibble_motif$//)
					{
					if(length$trimmed_seq>=$min_size)
						{
						$trimmed_out{$seq}=~s/^##//;
						@titles=split('##',$trimmed_out{$seq});
						foreach$title(@titles)
							{
							$i++;
							print OUT">$title\n$trimmed_seq\n";
							}
						}
					}
				}
			if(!$silent){print"\nRemaining reads $min_size-$max_size nt: $i";}
			}
		close OUT;
		if(!$silent){print"\n\nDetermined adapter sequence: ${nibble_motif}${adapter_to_clip}-(N)x-3'.\n";}
		$input=$trimmmed_file;
		$parsed_input.=".trimmed";
		$input="$dump_folder/$parsed_input";
		}

	# collapse if input is uncollapsed FASTA or FASTQ (or map file from uncollapsed FASTA or FASTQ)
	if($collapsed eq'uncollapsed ')
		{
		if(!$silent){print"\nCollapse input file...";}
		open(INPUT,$input)||die print"\nError! Unable to open input file $input.\n$!\n\n";
		open(COLLAPSED,">$dump_folder/$parsed_input.collapsed")||die print"\nError! Unable to create input file $dump_folder/$parsed_input.\n$!\n\n";
		while(<INPUT>)
			{
			$_=~s/\s*$//;
			if($_!~/^>/)
				{
				$input{$_}++;
				}
			}
		foreach$seq(sort{$input{$b}<=>$input{$a}}keys%input)
			{
			next if(length$seq==0);
			print COLLAPSED">$input{$seq}\n$seq\n";
			}
		close INPUT;
		close COLLAPSED;
		$parsed_input.=".collapsed";
		$input="$dump_folder/$parsed_input";
		undef%input;
		if(!$silent){print" OK.";}
		}

	# low complexity filter | use default parameters from the duster tool (NGS TOOLBOX)
	if(!$skip_dust)
		{
		if(!$silent){print"\nFilter low complexity reads from $parsed_input...";}
		open(IN,$input)||die print"\nError! Unable to open input file $input.\n$!\n\n";
		open(OK,">$dump_folder/$parsed_input.no-dust")||die print"\nError! Unable to create file $dump_folder/$parsed_input.no-dust.\n$!\n\n";
		open(DUST,">$dump_folder/$parsed_input.dust")||die print"\nError! Unable to create file $dump_folder/$parsed_input.dust.\n$!\n\n";
		$i=0;
		$simple_reads=0;
		$complex_reads=0;
		$threshold_2nt=$dust_limit+((1-$dust_limit)*$dust_limit);
		$simple_1nt=0;
		$simple_2nt=0;
		my%simple_motif=();
		@seq_data=();
		while(<IN>)
			{
			$_=~s/\s*$//;
			if($_=~s/^>//)
				{
				$reads=$_;
				}
			else
				{
				$ok=1;
				next if(length$_==0);
				$read_length=length$_;

				# simple monomer over-representation check
				$monomer_check=$_;
				$amount_N=($monomer_check=~tr/N//);
				$non_N=length$monomer_check;
				$amount_A=($monomer_check=~tr/A//);
				$amount_T=($monomer_check=~tr/T//);
				$amount_G=($monomer_check=~tr/G//);
				$amount_C=($monomer_check=~tr/C//);
				if($non_N/$read_length<$dust_maxN||$amount_A/$non_N>$dust_limit||$amount_T/$non_N>$dust_limit||$amount_G/$non_N>$dust_limit||$amount_C/$non_N>$dust_limit)
					{
					$ok=0;
					$simple_1nt++;
					}

				# motif enrichment check
				if($ok==1)
					{
					my%fragments=();
					foreach$frag_size(2..$dust_max)
						{
						$gnaw=$_;
						while(length$gnaw>=$frag_size)
							{
							$gnaw=~s/^.{$frag_size}//;
							$fragments{$&}=$frag_size;
							}
						}
					foreach$fragment(sort{$fragments{$a}<=>$fragments{$b}}keys%fragments) # make it faster by searching for short motifs first
						{
						$check=$_;
						$check=~s/$fragment//g;
						if(length$check<=$read_length*(1-$dust_limit))
							{
							$ok=0;
							$simple_motif{length$fragment}++;
							last;
							}
						}
					undef%fragments;
					}

				# 2 bases over-representation check
				if($ok==1&&!$no_overrepresentation_check)
					{
					if(($amount_A+$amount_T)/$non_N>$threshold_2nt||($amount_A+$amount_G)/$non_N>$threshold_2nt||($amount_A+$amount_C)/$non_N>$threshold_2nt||($amount_T+$amount_G)/$non_N>$threshold_2nt||($amount_T+$amount_C)/$non_N>$threshold_2nt||($amount_G+$amount_C)/$non_N>$threshold_2nt)
						{
						$ok=0;
						$simple_2nt++;
						}
					}

				$output=">$reads\n$_\n";
				if($ok==1)
					{
					$complex_reads+=$reads;
					print OK$output;
					$global_annotation_matrix{$_}{$reads}.="";
					}
				else
					{
					$simple_reads+=$reads;
					print DUST$output;
					$global_annotation_matrix{$_}{$reads}="low_complexity\t";
					}
				}
			}
		$parsed_input.=".no-dust";
		$input="$dump_folder/$parsed_input";
		if(!$silent){print" done.\n$complex_reads sequences are ok.\n$simple_reads sequences are dusty (low-complexity).\n  Monomer:             $simple_1nt\n  Biased for 2 bases:  $simple_2nt";}
		if(!$silent)
			{
			foreach$i(2..$dust_max)
				{
				if(!$simple_motif{$i})
					{
					$simple_motif{$i}=0;
					}
				print"\n  $i nt motif enriched: $simple_motif{$i}";
				}
			}
		if($complex_reads==0)
			{
			print"\nNo reads passed low complexity filter!";
			exit;
			}
		}
	close IN;
	close OK;
	close DUST;

	# load mature miRNA sequences to memory
	$no_nonmiRs=0;
	if(!$skip_miR)
		{
		if(!$silent){print"\nLoad mature miRNA sequences for $species{$species} to memory...";}
		open(MIR_MAT,"$dump_folder/miRBase_mature")||die print"\nERROR! Unable to open miRBase file $dump_folder/miRBase_mature.\n$!\n\n";
		$i=0;
		while(<MIR_MAT>)
			{
			$_=~s/\s*$//;
			if($_=~s/^>//)
				{
				@head=split(' ',$_);
				}
			else
				{
				$_=~s/U/T/g;
				if(lc$head[2].'_'.$head[3]eq$species)
					{
					$i++;
					$miRBase_species{$_}=$head[4];
					}
				}
			}
		close MIR_MAT;
		if(!$silent){print" OK ($i sequences loaded).\nLoad mature miRNA sequences for other species to memory...";}
		open(MIR_MAT,"$dump_folder/miRBase_mature")||die print"\nERROR! Unable to open miRBase file $dump_folder/miRBase_mature.\n$!\n\n";
		$i=0;
		while(<MIR_MAT>)
			{
			$_=~s/\s*$//;
			if($_=~s/^>//)
				{
				@head=split(' ',$_);
				}
			else
				{
				$_=~s/U/T/g;
				if(!$miRBase_species{$_})
					{
					$i++;
					$miRBase_nonspecies{$_}.="$head[0] ";
					}
				}
			}
		close MIR_MAT;
		foreach(keys%miRBase_nonspecies)
			{
			$miRBase_nonspecies{$_}=~s/ $//;
			}
		if(!$silent){print" OK ($i sequences loaded).";}

		# search mature canonical miRNAs, perfect full-length hits.
		if(!$silent){print"\nSearch for mature, canonical miRNAs...";}
		open(OUT_MIR_SPECIES,">$dump_folder/$outfile_prefix.miR.$species{$species}.fas")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.mature-miR.$species{$species}.fas.\n$!\n\n";
		open(OUT_MIR_NONSPECIES,">$dump_folder/$outfile_prefix.miR.non-$species{$species}.fas")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.mature-miR.non-$species{$species}.fas.\n$!\n\n";
		open(OUT_NONMIR,">$dump_folder/$outfile_prefix.non-miR.fas")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.non-miR.fas.\n$!\n\n";
		open(INPUT,$input)||die print" ERROR! Unable to open input file $input.\n$!\n\n";
		while(<INPUT>)
			{
			$_=~s/\s*$//;
			if($_=~s/^>//)
				{
				$reads=$_;
				}
			elsif($miRBase_species{$_})
				{
				$miR_canonical_species{$_}=$reads;
				$miR_table_out_species{$_}{0}=$reads;
				$output_miR_species{$_}=1;
				if($global_annotation_matrix{$_}{$reads}!~/\Q$miRBase_species{$_}/){$global_annotation_matrix{$_}{$reads}.="$miRBase_species{$_}\t";}
				}
			elsif($miR_nonspecies{$_})
				{
				$miR_canonical_nonspecies{$_}=$reads;
				$miR_table_out_nonspecies{$_}{0}=$reads;
				$output_miR_nonspecies{$_}=1;
				if($global_annotation_matrix{$_}{$reads}!~/\Q$miR_nonspecies{$_}/){$global_annotation_matrix{$_}{$reads}.="$miR_nonspecies{$_}\t";}
				}
			else
				{
				$non_annotated{$_}=$reads;
				}
			}
		close INPUT;
		if(!$silent){print" OK.\nSearch for non-canonical and 3' tailed miRNAs ($species{$species})...";}

		# create SeqMap input file for SPECIES run
		open(INPUT,$input)||die print"\nERROR! Unable to open input file $input.\n$!\n\n";
		open(NEW_INPUT,">$dump_folder/SeqMap_input1.temp")||die print"\nERROR! Unable to create temporary SeqMap input file $dump_folder/SeqMap_input1.temp.\n$!\n\n";
		while($input_line=<INPUT>)
			{
			$input_line=~s/\s*$//;
			if($input_line=~s/^>//)
				{
				$reads=$input_line;
				}
			elsif(!$miR_canonical_species{$input_line})
				{
				print NEW_INPUT">$reads\n$input_line\n";
				}
			}
		close INPUT;
		close NEW_INPUT;

		# check length of miRNA precursors
		%miR_hairpin_length=();
		foreach$index("$species{$species}","non-$species{$species}")
			{
			open(MIR_PREC,"$dump_folder/miRBase_hairpin_$index.fas")||die print"\nERROR! Unable to check miR-precursor lengths from $dump_folder/miRBase_hairpin_$index.fas.\n$!\n\n";
			while(<MIR_PREC>)
				{
				$_=~s/\s*$//;
				if($_=~s/^>//)
					{
					$head=$_;
					}
				else
					{
					$miR_hairpin_length{$head}=length$_;
					}
				}
			close MIR_PREC;
			}

		# run SeqMap on SPECIES miRNA hairpins
		$mm_for_SeqMap_run=$intmod_miR+$max_miRtail_length;
		if($^O eq'linux'||$^O eq'darwin')
			{
			qx/.\/seqmap.exe $mm_for_SeqMap_run $dump_folder\/SeqMap_input1.temp $dump_folder\/miRBase_hairpin_$species{$species}.fas $dump_folder\/SeqMap_species.temp \/output_all_matches \/forward_strand \/available_memory:${memory}000 $less_memory_options_miR> $dump_folder\/SeqMap_report1.txt/;
			}
		elsif($^O eq'MSWin32')
			{
			system("seqmap.exe $mm_for_SeqMap_run $dump_folder/SeqMap_input1.temp $dump_folder/miRBase_hairpin_$species{$species}.fas $dump_folder/SeqMap_species.temp /output_all_matches /forward_strand /available_memory:${memory}000 ${less_memory_options_miR}2>&1 > $dump_folder/SeqMap_report1.txt");
			}
		else
			{
			if(!$silent){print" WARNING! Unknown operating system.";}
			# try unix command
			qx/.\/seqmap.exe $mm_for_SeqMap_run $dump_folder\/SeqMap_input1.temp $dump_folder\/miRBase_hairpin_$species{$species}.fas $dump_folder\/SeqMap_species.temp \/output_all_matches \/forward_strand \/available_memory:${memory}000 $less_memory_options_miR> $dump_folder\/SeqMap_report1.txt/;
			if(!-e"$dump_folder/SeqMap_species.temp")
				{
				# try windows command
				system("seqmap.exe $mm_for_SeqMap_run $dump_folder/SeqMap_input1.temp $dump_folder/miRBase_hairpin_$species{$species}.fas $dump_folder/SeqMap_species.temp /output_all_matches /forward_strand /available_memory:${memory}000 ${less_memory_options_miR}2>&1 > $dump_folder/SeqMap_report1.txt");
				}
			}
		# check SeqMap report
		open(SMR,"$dump_folder/SeqMap_report1.txt");
		while(<SMR>)
			{
			if($_=~/not enough memory/)
				{
				print"\n\nERROR. Not enough memory available for SeqMap. Check $dump_folder/SeqMap_report1.txt for more details. Try less mismatch (-mismatch [n]), less insertions/deletions (-insdel [n]), fewer non-template 3' nucleotides (-tail [n]). Alternatively you may want to use the options -less_memory or -even_less_memory.\n\n";
				exit;
				}
			}
		close SMR;

		# clean up SeqMap output for SPECIES
		$file_id="species";
		clean_SeqMap_miRNA();
		open(SEQMAP,"$dump_folder/SeqMap_species.clean-2.temp");
		%hits_per_miRspecies=();
		while(<SEQMAP>)
			{
			$_=~s/\s*$//;
			@seqmap=split("\t",$_);
			$hits_per_miRspecies{$seqmap[4]}++;

			# save targets of this sequence (without the tail)
			if(!$targets_per_miRspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])})
				{
				$targets_per_miRspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}.="$seqmap[0] ";
				}

			# save valid hits to exclude these from non-species search
			$valid_species_hits{$seqmap[4]}=1;
			}
		close SEQMAP;

		# search non-canonical (tailed and untailed) SPECIES miRNAs
		unless(open(SEQMAP,"$dump_folder/SeqMap_species.clean-2.temp"))
			{
			print"\nERROR! Unable to open temporary SeqMap output file.\n$!\n\n";
			open(REPORT,"$dump_folder/SeqMap_report1.txt");
			while(<REPORT>)
				{
				print$_;
				}
			exit;
			}
		while(<SEQMAP>)
			{
			$_=~s/\s*$//;
			@seqmap=split("\t",$_);

			# count internal modifications
			if($seqmap[0]=~/\(.+\)/)
				{
				$mod=$&;
				$mod=~s/[\(\)]//g;
				$mod=~s/\d+//;
				$modpos=$&;
				$mod=~s/^-/?-/;
				$count_internal_mods_species{$mod}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				$count_internal_mods_species{$modpos}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				$mods_per_pos_species{$modpos}{$mod}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				}

			if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/\Q$seqmap[0]/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="$seqmap[0]\t";}

			# perfect hit to species precursor but not to canonical miRNA according to miRBase
			if($seqmap[6]==0&&!$miRBase_species{$seqmap[4]})
				{
				$miR_noncanonical_species{$seqmap[4]}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				$miR_table_out_species{$seqmap[4]}{0}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				$output_miR_species{$seqmap[4]}=1;
				delete$miR_canonical_nonspecies{$seqmap[4]};
				delete$miR_table_out_nonspecies{$seqmap[4]};
				delete$output_miR_nonspecies{$seqmap[4]};
				}

			# not tailed, only internal modification
			elsif($seqmap[5]==0&&!$miRBase_species{$seqmap[4]})
				{
				$miR_noncanonical_species{$seqmap[4]}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				$miR_table_out_species{$seqmap[4]}{0}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				$output_miR_species{$seqmap[4]}=1;
				delete$miR_canonical_nonspecies{$seqmap[4]};
				delete$miR_table_out_nonspecies{$seqmap[4]};
				delete$output_miR_nonspecies{$seqmap[4]};
				}

			# check for 3' tailed hit to species precursor
			else
				{
				$output_miR_species{$seqmap[4]}=1;
				$tail=substr($seqmap[4],-$seqmap[5]);
				$tails_species{$tail}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
				$tails_per_miR_species{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}{$tail}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};

				# check if this is a tailed canonical miRNA according to miRBase
				if($miRBase_species{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])})
					{
					$miR_table_out_species{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}{length$tail}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
					$miR_canonical_species{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
					}

				# this is a tailed non-canonical miRNA according to miRBase
				else
					{
					$miR_table_out_species{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}{length$tail}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
					$miR_noncanonical_species{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}+=$seqmap[3]/$hits_per_miRspecies{$seqmap[4]};
					}

				# remove tailed and non-tailed versions of this miRNA from non-species and non-annotated hashes
				delete$miR_canonical_nonspecies{$seqmap[4]};
				delete$miR_table_out_nonspecies{$seqmap[4]};
				delete$output_miR_nonspecies{$seqmap[4]};
				delete$miR_canonical_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])};
				delete$miR_table_out_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])};
				delete$output_miR_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])};
				}
			}
		close SEQMAP;
		undef%hits_per_miRspecies;

		if(!$search_species_miRs_only)
			{
			# create SeqMap input file for NON-SPECIES run
			if(!$silent){print" OK.\nSearch for non-canonical and 3' tailed miRNAs (non-$species{$species})...";}
			open(INPUT,"$dump_folder/SeqMap_input1.temp")||die print"\nERROR! Unable to open temporary SeqMap input file $dump_folder/SeqMap_input1.temp.\n$!\n\n";
			open(NEW_INPUT,">$dump_folder/SeqMap_input2.temp")||die print"\nERROR! Unable to create temporary SeqMap input file $dump_folder/SeqMap_input2.temp.\n$!\n\n";
			$n_remaining_seqs=0;
			while(<INPUT>)
				{
				$_=~s/\s*$//;
				if($_=~s/^>//)
					{
					$reads=$_;
					}
				elsif(!$valid_species_hits{$_}&&!$miR_canonical_nonspecies{$_})
					{
					$n_remaining_seqs++;
					print NEW_INPUT">$reads\n$_\n";
					}
				}
			close INPUT;
			close NEW_INPUT;

			if($n_remaining_seqs>0)
				{
				# run SeqMap on NON-SPECIES miRNA hairpins IF THERE ARE SEQUENCES THAT COULD NOT BE ANNOTATED AS SPECIES MIR!!!
				if($^O eq'linux'||$^O eq'darwin')
					{
					qx/.\/seqmap.exe $mm_for_SeqMap_run $dump_folder\/SeqMap_input2.temp $dump_folder\/miRBase_hairpin_non-$species{$species}.fas $dump_folder\/SeqMap_nonspecies.temp \/output_all_matches \/forward_strand \/available_memory:${memory}000 $less_memory_options_miR> $dump_folder\/SeqMap_report2.txt/;
					}
				elsif($^O eq'MSWin32')
					{
					system("seqmap.exe $mm_for_SeqMap_run $dump_folder/SeqMap_input2.temp $dump_folder/miRBase_hairpin_non-$species{$species}.fas $dump_folder/SeqMap_nonspecies.temp /output_all_matches /forward_strand /available_memory:${memory}000 ${less_memory_options_miR}2>&1 > $dump_folder/SeqMap_report2.txt");
					}
				else
					{
					if(!$silent){print" WARNING! Unknown operating system.";}
					# try unix command
					qx/.\/seqmap.exe $mm_for_SeqMap_run $dump_folder\/SeqMap_input2.temp $dump_folder\/miRBase_hairpin_non-$species{$species}.fas $dump_folder\/SeqMap_nonspecies.temp \/output_all_matches \/forward_strand \/available_memory:${memory}000 $less_memory_options_miR> $dump_folder\/SeqMap_report2.txt/;
					if(!-e"$dump_folder/SeqMap_nonspecies.temp")
						{
						# try windows command
						system("seqmap.exe $mm_for_SeqMap_run $dump_folder/SeqMap_input2.temp $dump_folder/miRBase_hairpin_non-$species{$species}.fas $dump_folder/SeqMap_nonspecies.temp /output_all_matches /forward_strand /available_memory:${memory}000 ${less_memory_options_miR}2>&1 > $dump_folder/SeqMap_report2.txt");
						}
					}
				# check SeqMap report
				open(SMR,"$dump_folder/SeqMap_report2.txt");
				while(<SMR>)
					{
					if($_=~/not enough memory/)
						{
						print"\n\nERROR. Not enough memory available for SeqMap. Check $dump_folder/SeqMap_report2.txt for more details. Try less mismatch (-mismatch [n]), less insertions/deletions (-insdel [n]), fewer non-template 3' nucleotides (-tail [n]). Alternatively you may want to use the options -less_memory or -even_less_memory.\n\n";
						exit;
						}
					}
				close SMR;

				# clean up SeqMap output
				$file_id="nonspecies";
				clean_SeqMap_miRNA();
				open(SEQMAP,"$dump_folder/SeqMap_nonspecies.clean-2.temp");
				%hits_per_miRnonspecies=();
				while(<SEQMAP>)
					{
					$_=~s/\s*$//;
					@seqmap=split("\t",$_);
					$hits_per_miRnonspecies{$seqmap[4]}++;

					# save targets of this sequence (without the tail)
					if(!$targets_per_miRnonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])})
						{
						$targets_per_miRnonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}.="$seqmap[0] ";
						}
					}
				close SEQMAP;

				# search non-canonical (tailed and untailed) NON-SPECIES miRNAs
				unless(open(SEQMAP,"$dump_folder/SeqMap_nonspecies.clean-2.temp"))
					{
					print"\nERROR! Unable to open temporary SeqMap output file.\n$!\n\n";
					open(REPORT,"$dump_folder/SeqMap_report2.txt");
					while(<REPORT>)
						{
						print$_;
						}
					exit;
					}
				while(<SEQMAP>)
					{
					$_=~s/\s*$//;
					@seqmap=split("\t",$_);

					# count internal modifications
					if($seqmap[0]=~/\(.+\)/)
						{
						$mod=$&;
						$mod=~s/[\(\)]//g;
						$mod=~s/\d+//;
						$modpos=$&;
						$mod=~s/^-/?-/;
						$count_internal_mods_nonspecies{$mod}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						$count_internal_mods_nonspecies{$modpos}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						$mods_per_pos_nonspecies{$modpos}{$mod}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						}

					if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/\Q$seqmap[0]/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="$seqmap[0]\t";}

					# perfect hit to non-species precursor but not to canonical non-species miRNA according to miRBase
					if($seqmap[6]==0&&!$miRBase_nonspecies{$seqmap[4]})
						{
						$miR_noncanonical_nonspecies{$seqmap[4]}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						$miR_table_out_nonspecies{$seqmap[4]}{0}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						$output_miR_nonspecies{$seqmap[4]}=1;
						}

					# not tailed, only internal modification
					elsif($seqmap[5]==0&&!$miRBase_nonspecies{$seqmap[4]})
						{
						$miR_noncanonical_nonspecies{$seqmap[4]}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						$miR_table_out_nonspecies{$seqmap[4]}{0}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						$output_miR_nonspecies{$seqmap[4]}=1;
						}

					# check for 3' tailed hit to non-species precursor
					else
						{
						$output_miR_nonspecies{$seqmap[4]}=1;
						$tail=substr($seqmap[4],-$seqmap[5]);
						$tails_nonspecies{$tail}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
						$tails_per_miR_nonspecies{substr($seqmap[4],0,$p)}{$tail}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};

						# check if this is a tailed canonical non-species miRNA
						if($miRBase_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])})
							{
							$miR_table_out_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}{length$tail}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
							$miR_canonical_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
							}
						# this must be a tailed non-canonical non-species miRNA
						else
							{
							$miR_table_out_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}{length$tail}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
							$miR_noncanonical_nonspecies{substr($seqmap[4],0,(length$seqmap[4])-$seqmap[5])}+=$seqmap[3]/$hits_per_miRnonspecies{$seqmap[4]};
							}
						}
					}
				close SEQMAP;
				undef%hits_per_miRnonspecies;
				}
			}

		# make miRNA output files
		$i_miRspecies=0;
		$i_miRnonspecies=0;
		$nonid_miRspecies=0;
		$nonid_miRnonspecies=0;
		open(INPUT,$input)||die print"\nERROR! Unable to open input file $input.\n$!\n\n";
		while(<INPUT>)
			{
			$_=~s/\s*$//;
			if($_=~s/^>//)
				{
				$reads=$_;
				}
			else
				{
				if($output_miR_species{$_})
					{
					$nonid_miRspecies++;
					$i_miRspecies+=$reads;
					$miR_spec_seq{$_}=$reads;
					}
				elsif($output_miR_nonspecies{$_})
					{
					$nonid_miRnonspecies++;
					$i_miRnonspecies+=$reads;
					$miR_nonspec_seq{$_}=$reads;
					}
				else
					{
					$nonmiR_seq{$_}=$reads;
					}
				}
			}
		close INPUT;
		foreach(sort{$miR_spec_seq{$b}<=>$miR_spec_seq{$a} or $a cmp $b}keys%miR_spec_seq)
			{
			print OUT_MIR_SPECIES">$miR_spec_seq{$_}\n$_\n";
			}
		foreach(sort{$miR_nonspec_seq{$b}<=>$miR_nonspec_seq{$a} or $a cmp $b}keys%miR_nonspec_seq)
			{
			print OUT_MIR_NONSPECIES">$miR_nonspec_seq{$_}\n$_\n";
			}
		foreach(sort{$nonmiR_seq{$b}<=>$nonmiR_seq{$a} or $a cmp $b}keys%nonmiR_seq)
			{
			print OUT_NONMIR">$nonmiR_seq{$_}\n$_\n";
			}

		if(keys%nonmiR_seq==0)
			{
			$no_nonmiRs=1; # skip ncRNA annotation when non-miR fasta file is empty
			}
		undef%miR_spec_seq;
		undef%miR_nonspec_seq;
		undef%nonmiR_seq;

		close OUT_MIR_SPECIES;
		close OUT_MIR_NONSPECIES;
		close OUT_NONMIR;
		$ba_in="$dump_folder/$outfile_prefix.miR.non-$species{$species}.fas";
		$ba_out="$dump_folder/$outfile_prefix.miR.non-$species{$species}.info";
		basic_analysis();
		$ba_in="$dump_folder/$outfile_prefix.miR.$species{$species}.fas";
		$ba_out="$dump_folder/$outfile_prefix.miR.$species{$species}.info";
		basic_analysis();

		# make miRNA-modifications files
		open(OUT_MODS_SPECIES,">$dump_folder/$outfile_prefix.miR-modifications_$species{$species}.txt")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.miR-modifications_$species{$species}.txt.\n$!\n\n";
		open(OUT_MODS_NONSPECIES,">$dump_folder/$outfile_prefix.miR-modifications_non-$species{$species}.txt")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.miR-modifications_non-$species{$species}.txt.\n$!\n\n";
		print OUT_MODS_SPECIES"3'-tailings (non-template nucleotides)\n";
		print OUT_MODS_NONSPECIES"3'-tailings (non-template nucleotides)\n";
		%new=();
		%old=%miRtails;
		foreach(1..$max_miRtail_length-1)
			{
			%new=%old;
			foreach$nt(keys%miRtails)
				{
				foreach(keys%old)
					{
					$new{$_.$nt}=1;
					}
				}
			%old=(%old,%new)
			}
		@combs=keys%old;
		@combs=sort@combs;
		foreach$tail(@combs)
			{
			if(!$tails_species{$tail})
				{
				$tails_species{$tail}=0;
				}
			if(!$tails_nonspecies{$tail})
				{
				$tails_nonspecies{$tail}=0;
				}
			print OUT_MODS_SPECIES"$tail\t$tails_species{$tail}\n";
			print OUT_MODS_NONSPECIES"$tail\t$tails_nonspecies{$tail}\n";
			}

		print OUT_MODS_SPECIES"\nInternal modifications (?->[ATGCN] = mapped sequence exceeds precursor sequence)\n";
		print OUT_MODS_NONSPECIES"\nInternal modifications (?->[ATGCN] = mapped sequence exceeds precursor sequence)\n";
		foreach$mod(sort{$count_internal_mods_species{$b}<=>$count_internal_mods_species{$a}}keys%count_internal_mods_species)
			{
			if($mod!~/\d/){print OUT_MODS_SPECIES"$mod\t$count_internal_mods_species{$mod}\n";}
			}
		foreach$mod(sort{$count_internal_mods_nonspecies{$b}<=>$count_internal_mods_nonspecies{$a}}keys%count_internal_mods_nonspecies)
			{
			if($mod!~/\d/){print OUT_MODS_NONSPECIES"$mod\t$count_internal_mods_nonspecies{$mod}\n";}
			}
		print OUT_MODS_SPECIES"\nPositions were internal modifications occur\n";
		print OUT_MODS_NONSPECIES"\nPositions were internal modifications occur\n";
		foreach$mod(1..100)
			{
			if($count_internal_mods_species{$mod}>0){print OUT_MODS_SPECIES"$mod\t$count_internal_mods_species{$mod}\n";}
			}
		foreach$mod(1..100)
			{
			if($count_internal_mods_species{$mod}>0){print OUT_MODS_NONSPECIES"$mod\t$count_internal_mods_nonspecies{$mod}\n";}
			}

		print OUT_MODS_SPECIES"\nModifications per position\n";
		print OUT_MODS_NONSPECIES"\nModifications per position\n";
		foreach$mod(sort{$count_internal_mods_species{$b}<=>$count_internal_mods_species{$a}}keys%count_internal_mods_species)
			{
			if($mod!~/\d/){print OUT_MODS_SPECIES"\t$mod";}
			}
		foreach$pos(1..100)
			{
			if($count_internal_mods_species{$pos}>0)
				{
				print OUT_MODS_SPECIES"\n$pos";
				foreach$mod(sort{$count_internal_mods_species{$b}<=>$count_internal_mods_species{$a}}keys%count_internal_mods_species)
					{
					next if($mod=~/\d/);
					if(!$mods_per_pos_species{$pos}{$mod})
						{
						$mods_per_pos_species{$pos}{$mod}=0;
						}
					print OUT_MODS_SPECIES"\t$mods_per_pos_species{$pos}{$mod}";
					}
				}
			}
		foreach$mod(sort{$count_internal_mods_nonspecies{$b}<=>$count_internal_mods_nonspecies{$a}}keys%count_internal_mods_nonspecies)
			{
			if($mod!~/\d/){print OUT_MODS_NONSPECIES"\t$mod";}
			}
		foreach$pos(1..100)
			{
			if($count_internal_mods_nonspecies{$pos}>0)
				{
				print OUT_MODS_NONSPECIES"\n$pos";
				foreach$mod(sort{$count_internal_mods_nonspecies{$b}<=>$count_internal_mods_nonspecies{$a}}keys%count_internal_mods_nonspecies)
					{
					next if($mod=~/\d/);
					if(!$mods_per_pos_nonspecies{$pos}{$mod})
						{
						$mods_per_pos_nonspecies{$pos}{$mod}=0;
						}
					print OUT_MODS_NONSPECIES"\t$mods_per_pos_nonspecies{$pos}{$mod}";
					}
				}
			}
		close OUT_MODS_SPECIES;
		close OUT_MODS_NONSPECIES;

		# make miRNA tables
		open(OUT_TABLE_SPECIES,">$dump_folder/$outfile_prefix.miR-table_$species{$species}.txt")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.miR-table_$species{$species}.txt.\n$!\n\n";
		open(OUT_TABLE_NONSPECIES,">$dump_folder/$outfile_prefix.miR-table_non-$species{$species}.txt")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.miR-table_non-$species{$species}.txt.\n$!\n\n";
		print OUT_TABLE_SPECIES"miR-name\tsequence\ttotal_reads";
		print OUT_TABLE_NONSPECIES"miR-name\tsequence\ttotal_reads";
		foreach$n(0..$max_miRtail_length)
			{
			print OUT_TABLE_SPECIES"\t+N$n-3'";
			print OUT_TABLE_NONSPECIES"\t+N$n-3'";
			}
		foreach$tail(@combs)
			{
			print OUT_TABLE_SPECIES"\t$tail-tailed_reads";
			print OUT_TABLE_NONSPECIES"\t$tail-tailed_reads";
			}
		print OUT_TABLE_SPECIES"\n";
		print OUT_TABLE_NONSPECIES"\n";

		foreach$seq(sort{$a cmp $b}keys%miR_table_out_species)
			{
			# get miRNA name
			if($miRBase_species{$seq})
				{
				$miR_name=$miRBase_species{$seq};
				}
			else
				{
				$miR_name=$targets_per_miRspecies{$seq};
				}

			# get total miRNA counts
			$miR_counts=0;
			foreach$i(0..$max_miRtail_length)
				{
				$miR_counts+=$miR_table_out_species{$seq}{$i};
				}

			print OUT_TABLE_SPECIES"$miR_name\t$seq\t$miR_counts\t";
			foreach$n(0..$max_miRtail_length)
				{
				if(!$miR_table_out_species{$seq}{$n})
					{
					$miR_table_out_species{$seq}{$n}=0;
					}
				print OUT_TABLE_SPECIES"$miR_table_out_species{$seq}{$n}\t";
				}
			foreach$tail(@combs)
				{
				if(!$tails_per_miR_species{$seq}{$tail})
					{
					$tails_per_miR_species{$seq}{$tail}=0;
					}
				print OUT_TABLE_SPECIES"$tails_per_miR_species{$seq}{$tail}\t";
				}
			print OUT_TABLE_SPECIES"\n";
			}

		foreach$seq(sort{$a cmp $b}keys%miR_table_out_nonspecies)
			{
			# get miRNA name
			if($miRBase_nonspecies{$seq})
				{
				$miR_name=$miRBase_nonspecies{$seq};
				}
			else
				{
				$miR_name=$targets_per_miRnonspecies{$seq};
				}

			# get total miRNA counts
			$miR_counts=0;
			foreach$i(0..$max_miRtail_length)
				{
				$miR_counts+=$miR_table_out_nonspecies{$seq}{$i};
				}

			print OUT_TABLE_NONSPECIES"$miR_name\t$seq\t$miR_counts\t";
			foreach$n(0..$max_miRtail_length)
				{
				if(!$miR_table_out_nonspecies{$seq}{$n})
					{
					$miR_table_out_nonspecies{$seq}{$n}=0;
					}
				print OUT_TABLE_NONSPECIES"$miR_table_out_nonspecies{$seq}{$n}\t";
				}
			foreach$tail(@combs)
				{
				if(!$tails_per_miR_nonspecies{$seq}{$tail})
					{
					$tails_per_miR_nonspecies{$seq}{$tail}=0;
					}
				print OUT_TABLE_NONSPECIES"$tails_per_miR_nonspecies{$seq}{$tail}\t";
				}
			print OUT_TABLE_NONSPECIES"\n";
			}
		close OUT_TABLE_SPECIES;
		close OUT_TABLE_NONSPECIES;
		}
	else # make a non-miR file to search for other ncRNA classes
		{
		if(!$silent){print" \nSkip miRNA annotation. Prepare file for ncRNA annotation...";}
		$i_miRspecies=0;
		$i_miRnonspecies=0;
		$nonid_miRspecies=0;
		$nonid_miRnonspecies=0;
		copy($input,"$dump_folder/$outfile_prefix.non-miR.fas")||die print"\nERROR! Unable to create input file for ncRNA search.\n$!\n\n";
		}

	# search ncRNAs other than miRNAs or mRNA fragments (if non-miR.fas is not empty)
	if(!$skip_cDNA||!$skip_ncRNA)
		{
		if($no_nonmiRs==0)
			{
			$map_to_what="";
			# add cDNA sequences to ncRNA file
			if(!$skip_cDNA&&!$skip_ncRNA)
				{
				$map_to_what="ncRNA and cDNA";
				open(SEQMAP_REF,">>$dump_folder\/ncRNA.fas")||die print"ERROR! Unable to open $dump_folder/ncRNA.fas to add cDNA sequences.\n$!\n\n";
				open(CDNA,"$dump_folder\/cDNA.fas")||die print"ERROR! Unable to open $dump_folder/cDNA.fas to add cDNA sequences.\n$!\n\n";
				$total_cDNA_reference_bp=0;
				while(<CDNA>)
					{
					print SEQMAP_REF$_;
					if($_!~/^>/)
						{
						$_=~s/\s*$//;
						$total_cDNA_reference_bp+=length$_;
						}
					}
				close SEQMAP_REF;
				close CDNA;
				$reference_file="$dump_folder\/ncRNA.fas";
				}
			# map to ncRNA only
			elsif(!$skip_ncRNA)
				{
				$map_to_what="ncRNA";
				$reference_file="$dump_folder\/ncRNA.fas";
				}
			# map to cDNA only
			elsif(!$skip_cDNA)
				{
				$map_to_what="cDNA";
				$reference_file="$dump_folder\/cDNA.fas";
				}

			# get lengths of tRNAs to identify 3' fragments, optionally add CCA to 3' end and save added sequences in a new file: .CCAcheck
			open(NCRNA,"$dump_folder/ncRNA.fas")||die print"\nERROR! Unable to open $dump_folder/ncRNA.fas.\n$!\n\n";
			if(!$noCCAcheck)
				{
				open(CCACHECK,">$dump_folder/ncRNA.fas.CCAcheck")||die print"\nERROR! Unable to create file $dump_folder/ncRNA.fas.CCAcheck.\n$!\n\n";
				}
			%tRNA_length=();
			while(<NCRNA>)
				{
				$_=~s/\s*$//;
				if($_=~s/^>//)
					{
					if($_=~/tRNA\|/)
						{
						$head=$_;
						# add CCA to 3' terminal end of tRNAs (opt-out)
						if($_!~/tRF-1_/&&$_!~/tRNA-leader_/&&!$noCCAcheck)
							{
							$doCCAcheck=1;
							}
						else
							{
							$doCCAcheck=0;
							}
						}
					else
						{
						$head="X";
						}
					if(!$noCCAcheck)
						{
						print CCACHECK">$_\n";
						}
					}
				elsif($head ne"X")
					{
					#print"\ndatabase_version:\n$_";
					if($_=~/[^C]C$/&&$doCCAcheck==1)
						{
						$_.="CA";
						}
					elsif($_=~/CC$/&&$doCCAcheck==1)
						{
						$_.="A";
						}
					elsif($_!~/CCA$/&&$doCCAcheck==1)
						{
						$_.="CCA";
						}
					$tRNA_length{$head}=length$_;
					print CCACHECK"$_\n";
					#print"\nedited_version:\n$_\n";
					}
				else
					{
					print CCACHECK"$_\n";
					}
				}
			close NCRNA;
			if(!$noCCAcheck)
				{
				close CCACHECK;
				$reference_file="$dump_folder/ncRNA.fas.CCAcheck";
				}

			$total_reference_bp=$total_cDNA_reference_bp+$total_ncRNA_reference_bp;
			if(!$silent){print" OK.\nCompare non-miR sequences with $map_to_what sequences (total reference sequence: $total_reference_bp bp)...";}
			if($^O eq'linux'||$^O eq'darwin')
				{
				qx/.\/seqmap.exe $seqmap_max_mm $dump_folder\/$outfile_prefix.non-miR.fas $reference_file $dump_folder\/SeqMap_ncRNA.temp \/allow_insdel:$seqmap_max_insdel \/output_all_matches \/forward_strand \/available_memory:${memory}000 $less_memory_options_ncR> $dump_folder\/SeqMap_report3.txt/;
				}
			elsif($^O eq'MSWin32')
				{
				system("seqmap.exe $seqmap_max_mm $dump_folder/$outfile_prefix.non-miR.fas $reference_file $dump_folder/SeqMap_ncRNA.temp /allow_insdel:$seqmap_max_insdel /output_all_matches \/forward_strand /available_memory:${memory}000 ${less_memory_options_ncR}2>&1 > $dump_folder/SeqMap_report3.txt");
				}
			else
				{
				print" WARNING! Unknown operating system.";
				#try unix command
				qx/.\/seqmap.exe $seqmap_max_mm $dump_folder\/$outfile_prefix.non-miR.fas $reference_file $dump_folder\/SeqMap_ncRNA.temp \/allow_insdel:$seqmap_max_insdel \/output_all_matches \/forward_strand \/available_memory:${memory}000 $less_memory_options_ncR> $dump_folder\/SeqMap_report3.txt/;
				if(!-e"$dump_folder\/SeqMap_ncRNA.temp")
					{
					# try windows command
					system("seqmap.exe $seqmap_max_mm $dump_folder/$outfile_prefix.non-miR.fas $reference_file $dump_folder/SeqMap_ncRNA.temp /allow_insdel:$seqmap_max_insdel /output_all_matches \/forward_strand /available_memory:${memory}000 ${less_memory_options_ncR}2>&1 > $dump_folder/SeqMap_report3.txt");
					}
				}

			# check SeqMap report
			open(SMR,"$dump_folder/SeqMap_report3.txt");
			while(<SMR>)
				{
				if($_=~/not enough memory/)
					{
					print"\n\nERROR. Not enough memory available for SeqMap. Check $dump_folder/SeqMap_report3.txt for more details. Try less mismatch (-mismatch [n]), less insertions/deletions (-insdel [n]), fewer non-template 3' nucleotides (-tail [n]). Alternatively you may want to use the options -less_memory or -even_less_memory.\n\n";
					exit;
					}
				}
			close SMR;

			if($seqmap_max_mm+$seqmap_max_insdel>0)
				{
				clean_SeqMap_ncRNA();
				open(SEQMAP,"$dump_folder/SeqMap_ncRNA.clean.temp")||die print"\nERROR! Unable to open $dump_folder/SeqMap_ncRNA.clean.temp.\n$!\n\n";
				}
			else
				{
				open(SEQMAP,"$dump_folder/SeqMap_ncRNA.temp")||die print"\nERROR! Unable to open $dump_folder/SeqMap_ncRNA.temp.\n$!\n\n";
				$head=<SEQMAP>;
				}
			%hits_per_seq=();
			while(<SEQMAP>)
				{
				$_=~s/\s*$//;
				@seqmap=split("\t",$_);
				$hits_per_seq{$seqmap[4]}++;
				}
			close SEQMAP;

			$tRNA_half5{"genomic"}=0;
			$tRF5{"genomic"}=0;
			$tRNA_half3{"genomic"}=0;
			$tRF3{"genomic"}=0;
			$tRF3_CCA{"genomic"}=0;
			$tRF1{"genomic"}=0;
			$tRNAleader{"genomic"}=0;
			$misc_tRF{"genomic"}=0;
			$tRNA_half5{"Mt"}=0;
			$tRF5{"Mt"}=0;
			$tRNA_half3{"Mt"}=0;
			$tRF3{"Mt"}=0;
			$tRF3_CCA{"Mt"}=0;
			$tRF1{"Mt"}=0;
			$tRNAleader{"Mt"}=0;
			$misc_tRF{"Mt"}=0;

			if($seqmap_max_mm+$seqmap_max_insdel>0)
				{
				open(SEQMAP,"$dump_folder/SeqMap_ncRNA.clean.temp")||die print"\nERROR! Unable to open $dump_folder/SeqMap_ncRNA.clean.temp.\n$!\n\n";
				}
			else
				{
				open(SEQMAP,"$dump_folder/SeqMap_ncRNA.temp")||die print"\nERROR! Unable to open $dump_folder/SeqMap_ncRNA.temp.\n$!\n\n";
				$head=<SEQMAP>;
				}
			while(<SEQMAP>)
				{
				$_=~s/\s*$//;
				@seqmap=split("\t",$_);

				# save alignment start and end coordinates
				$map_start_pos=$seqmap[1];
				$map_end_pos=($seqmap[1]+length$seqmap[2])-1;
				$alignment_info="($map_start_pos-$map_end_pos)";

				# avoid nested quantifiers in sequence names such as 'trna3197-Undet???'
				$seqmap[0]=~s/\?/X/g;
				$seqmap[0]=~s/\*/x/g;
				$seqmap[0]=~s/\+/plus/g;

				@id=split(/\|/,$seqmap[0]);
				$sorted_sequences{$id[0]}{$seqmap[4]}=$seqmap[3];
				$ncRNA_counts_1{$id[0]}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
				$ncRNA_counts_2{$id[0]}{$id[1]}+=$seqmap[3]/$hits_per_seq{$seqmap[4]}; # hits per transcript table

				# check tRNA derived sequences
				if($seqmap[0]=~/^tRNA\|/||$seqmap[0]=~/^Mt_tRNA\|/)
					{
					$origin="genomic";
					if($seqmap[0]=~/^Mt_tRNA\|/)
						{
						$origin="Mt";
						}
					$source_tRNA=$seqmap[0];
					$source_tRNA=~s/^[^\|]+\|//;

					# tRF-1 (derived from tRNA trailer)
					if($seqmap[0]=~/tRNA\|tRF-1_/)
						{
						$source_tRNA=~s/^tRF-1_\[*\d*\]*_*//;
						$tRF1{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
						$tRF_seqs{"tRF-1"}{$origin}{$seqmap[4]}=$seqmap[3];
						$tRF_data_table{$source_tRNA}{"tRF-1"}{"total"}+=$seqmap[3];
						$tRF_data_table{$source_tRNA}{"tRF-1"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
						if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/tRF-1 from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="tRF-1 from $source_tRNA $alignment_info\t";}
						}
					# tRNA leader sequence
					elsif($seqmap[0]=~/tRNA\|tRNA-leader_/)
						{
						$source_tRNA=~s/^tRNA-leader_\[*\d*\]*_*//;
						$tRNAleader{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
						$tRF_seqs{"tRNA-leader"}{$origin}{$seqmap[4]}=$seqmap[3];
						$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"total"}+=$seqmap[3];
						$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
						if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/tRF-leader from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="tRF-leader from $source_tRNA $alignment_info\t";}
						}
					# 5' fragment
					elsif($seqmap[1]==1+$tRNA_max_dist_from_end)
						{
						if(length$seqmap[4]>=$tRNA_half_min_size)
							{
							$tRNA_half5{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
							$tRF_seqs{"5p-tR-halves"}{$origin}{$seqmap[4]}=$seqmap[3];
							$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"total"}+=$seqmap[3];
							$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
							if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/5p-tR-half from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="5p-tR-half from $source_tRNA $alignment_info\t";}
							}
						else
							{
							$tRF5{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
							$tRF_seqs{"5p-tRFs"}{$origin}{$seqmap[4]}=$seqmap[3];
							$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"total"}+=$seqmap[3];
							$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
							if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/5p-tRF from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="5p-tRF from $source_tRNA $alignment_info\t";}
							}
						}
					# 3' fragment (not necessarily the very 3' end, tRNA may have up to 3bp 3' overhang in the alignment [-3])
					elsif($seqmap[1]+(length$seqmap[4])-1>=($tRNA_length{$seqmap[0]})-(3+$tRNA_max_dist_from_end))
						{
						if(length$seqmap[4]>=$tRNA_half_min_size)
							{
							$tRNA_half3{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
							$tRF_seqs{"3p-tR-halves"}{$origin}{$seqmap[4]}=$seqmap[3];
							$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"total"}+=$seqmap[3];
							$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
							if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/3p-tR-half from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="3p-tR-half from $source_tRNA $alignment_info\t";}
							}
						else
							{
							# CCA fragments have to match the very 3' end of the tRNA
							if($seqmap[4]=~/CCA$/i&&$seqmap[1]+(length$seqmap[4])-1==$tRNA_length{$seqmap[0]})
								{
								CCA();
								sub CCA
									{
									$tRF3_CCA{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
									$tRF_seqs{"3p-CCA-tRFs"}{$origin}{$seqmap[4]}=$seqmap[3];
									$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"total"}+=$seqmap[3];
									$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
									if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/3p-CCA-tRF from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="3p-CCA-tRF from $source_tRNA $alignment_info\t";}
									}
								}
							elsif($seqmap[4]=~/CC$/i&&$CC_is_CCA&&$seqmap[1]+(length$seqmap[4])==$tRNA_length{$seqmap[0]}) # allows to identify 3p-CCA-tRFs if poly-A tails were trimmed
								{
								CCA();
								}
							else
								{
								$tRF3{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
								$tRF_seqs{"3p-tRFs"}{$origin}{$seqmap[4]}=$seqmap[3];
								$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"total"}+=$seqmap[3];
								$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
								if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/3p-tRF from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="3p-tRF from $source_tRNA $alignment_info\t";}
								}
							}
						}
					else
						{
						$misc_tRF{$origin}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
						$tRF_seqs{"misc-tRFs"}{$origin}{$seqmap[4]}=$seqmap[3];
						$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"total"}+=$seqmap[3];
						$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"normalized"}+=$seqmap[3]/$hits_per_seq{$seqmap[4]};
						if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/misc-tRF from $source_tRNA $alignment_info/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="misc-tRF from $source_tRNA $alignment_info\t";}
						}
					}
				else
					{
					if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/\Q$seqmap[0] $alignment_info/i){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="$seqmap[0] $alignment_info\t";}
					}
				}
			close SEQMAP;
			}
		}

	# create tRF fasta files
	foreach$tRF_type("5p-tR-halves","5p-tRFs","3p-tR-halves","3p-CCA-tRFs","3p-tRFs","tRF-1","tRNA-leader","misc-tRFs")
		{
		foreach$origin("genomic","Mt")
			{
			if(keys%{$tRF_seqs{$tRF_type}{$origin}}>0)
				{
				open(TRF,">$dump_folder/$outfile_prefix.${origin}_$tRF_type.fas")||die print"\nERROR! Unable to create file $dump_folder/$outfile_prefix.${origin}_$tRF_type.fas.\n$!\n\n";
				foreach$seq(sort{$tRF_seqs{$tRF_type}{$origin}{$b}<=>$tRF_seqs{$tRF_type}{$origin}{$a} or $a cmp $b} keys%{$tRF_seqs{$tRF_type}{$origin}})
					{
					print TRF">$tRF_seqs{$tRF_type}{$origin}{$seq}\n$seq\n";
					}
				close TRF;
				$ba_in="$dump_folder/$outfile_prefix.${origin}_$tRF_type.fas";
				$ba_out="$dump_folder/$outfile_prefix.${origin}_$tRF_type.info";
				basic_analysis();
				}
			}
		}

	# create tRF table
	if(!$skip_ncRNA)
		{
		open(TRF_TABLE,">$dump_folder/${outfile_prefix}.tRF-table.txt")||die print"\nERROR! Unable to create file $dump_folder/$outfile_prefix.${origin}_tRF-table.txt.\n$!\n\n";
		print TRF_TABLE"NOTE: SEQUENCE READS MAY MAP TO MORE THAN ONE REFERENCE SEQUENCE.\nFRACTIONATED READ COUNTS ACCOUNT FOR MULTIPLE HITS OF ONE SEQUENCE.\n\nsource_tRNA\t5p-tR-halves (fractionated)\t5p-tR-halves (absolute)\t5p-tRFs (fractionated)\t5p-tRFs (absolute)\t3p-tR-halves (fractionated)\t3p-tR-halves (absolute)\t3p-CCA-tRFs (fractionated)\t3p-CCA-tRFs (absolute)\t3p-tRFs (fractionated)\t3p-tRFs (absolute)\ttRF-1 (fractionated)\ttRF-1 (absolute)\ttRNA-leader (fractionated)\ttRNA-leader (absolute)\tmisc-tRFs (fractionated)\tmisc-tRFs (absolute)";
		foreach$source_tRNA(sort{$a cmp $b}keys%tRF_data_table)
			{
			if(!$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"total"}){$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"normalized"}){$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"normalized"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"total"}){$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"normalized"}){$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"normalized"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"total"}){$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"normalized"}){$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"normalized"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"total"}){$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"normalized"}){$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"normalized"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"total"}){$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"normalized"}){$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"normalized"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"tRF-1"}{"total"}){$tRF_data_table{$source_tRNA}{"tRF-1"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"tRF-1"}{"normalized"}){$tRF_data_table{$source_tRNA}{"tRF-1"}{"normalized"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"total"}){$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"normalized"}){$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"normalized"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"total"}){$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"total"}=0;}
			if(!$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"normalized"}){$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"normalized"}=0;}
			print TRF_TABLE"\n$source_tRNA\t".$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"5p-tR-halves"}{"total"};
			print TRF_TABLE"\t".$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"5p-tRFs"}{"total"};
			print TRF_TABLE"\t".$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"3p-tR-halves"}{"total"};
			print TRF_TABLE"\t".$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"3p-CCA-tRFs"}{"total"};
			print TRF_TABLE"\t".$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"3p-tRFs"}{"total"};
			print TRF_TABLE"\t".$tRF_data_table{$source_tRNA}{"tRF-1"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"tRF-1"}{"total"};
			print TRF_TABLE"\t".$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"tRNA-leader"}{"total"};
			print TRF_TABLE"\t".$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"normalized"}."\t".$tRF_data_table{$source_tRNA}{"misc-tRFs"}{"total"};
			}
		close TRF_TABLE;
		}

	# create ncRNA/cDNA fasta files, ncRNA/cDNA count table and non-annotated file
	open(NCRNA_TABLE,">$dump_folder/$outfile_prefix.annotation_summary.txt")||die print"\nERROR! Unable to create file $dump_folder/$outfile_prefix.annotation_summary.txt.\n$!\n\n";
	if(!$skip_dust)
		{
		print NCRNA_TABLE"low_complexity\t$simple_reads\n";
		}
	if(!$ncRNA_counts_1{"tRNA"}){$ncRNA_counts_1{"tRNA"}=0;}
	if(!$ncRNA_counts_1{"Mt_tRNA"}){$ncRNA_counts_1{"Mt_tRNA"}=0;}
	if(!$ncRNA_counts_1{"rRNA"}){$ncRNA_counts_1{"rRNA"}=0;}
	if(!$ncRNA_counts_1{"Mt_rRNA"}){$ncRNA_counts_1{"Mt_rRNA"}=0;}
	$total_miR_reads=$i_miRspecies+$i_miRnonspecies;
	$total_tRNA_reads=$ncRNA_counts_1{"tRNA"}+$ncRNA_counts_1{"Mt_tRNA"};
	$total_rRNA_reads=$ncRNA_counts_1{"rRNA"}+$ncRNA_counts_1{"Mt_rRNA"};
	print NCRNA_TABLE"miRNA\t$total_miR_reads\n   miRNA:$species\t$i_miRspecies\n   miRNA:other\t$i_miRnonspecies\n";
	print NCRNA_TABLE"rRNA\t$total_rRNA_reads\n   genomic_rRNA\t$ncRNA_counts_1{rRNA}\n   Mt_rRNA\t$ncRNA_counts_1{Mt_rRNA}\n";
	print NCRNA_TABLE"tRNA\t$total_tRNA_reads\n   genomic_tRNA\t$ncRNA_counts_1{tRNA}\n      5'tR-halves\t$tRNA_half5{genomic}\n      5'tRFs\t$tRF5{genomic}\n      3'tR-halves\t$tRNA_half3{genomic}\n      3'tRFs\t$tRF3{genomic}\n      3'CCA-tRFs\t$tRF3_CCA{genomic}\n      tRF-1\t$tRF1{genomic}\n      tRNA-leader\t$tRNAleader{genomic}\n      misc-tRFs\t$misc_tRF{genomic}\n   Mt_tRNA\t$ncRNA_counts_1{Mt_tRNA}\n      5'tR-halves\t$tRNA_half5{Mt}\n      5'tRFs\t$tRF5{Mt}\n      3'tR-halves\t$tRNA_half3{Mt}\n      3'tRFs\t$tRF3{Mt}\n      3'CCA-tRFs\t$tRF3_CCA{Mt}\n      tRF-1\t$tRF1{Mt}\n      tRNA-leader\t$tRNAleader{Mt}\n      misc-tRFs\t$misc_tRF{Mt}\n";

	my%ncRNA_annotated;
	foreach$biotype(keys%sorted_sequences)
		{
		if($biotype ne"tRNA"&&$biotype ne"Mt_tRNA"&&$biotype ne"rRNA"&&$biotype ne"Mt_rRNA")
			{
			print NCRNA_TABLE"$biotype\t$ncRNA_counts_1{$biotype}\n";
			}
		$biotype_filename=$biotype;
		$biotype_filename=~s/[\\\/\*\.\"\[\]:;\|=\,]/_/g;
		open(NCRNA_OUT,">$dump_folder/$outfile_prefix.$biotype_filename.fas")||die print"\nERROR! Unable to create file $dump_folder/$outfile_prefix.$biotype_filename.fas.\n$!\n\n";
		foreach$seq(sort{$sorted_sequences{$biotype}{$b}<=>$sorted_sequences{$biotype}{$a} or $a cmp $b}keys%{$sorted_sequences{$biotype}})
			{
			$ncRNA_annotated{$seq}=1;
			print NCRNA_OUT">$sorted_sequences{$biotype}{$seq}\n$seq\n";
			}
		close NCRNA_OUT;
		$ba_in="$dump_folder/$outfile_prefix.$biotype_filename.fas";
		$ba_out="$dump_folder/$outfile_prefix.$biotype_filename.info";
		basic_analysis();
		}

	open(NO_ANNOTATION,">$dump_folder/$outfile_prefix.no-annotation.fas")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.no-annotation.fas.\n$!\n\n";
	open(INPUT,$input)||die print"\nERROR! Unable to open input file $input.\n$!\n\n";
	$no_annotation=0;
	my%no_annotation_seqs;
	while(<INPUT>)
		{
		$_=~s/\s*$//;
		if($_=~s/^>//)
			{
			$reads=$_;
			}
		elsif(!$ncRNA_annotated{$_}&&!$output_miR_species{$_}&&!$output_miR_nonspecies{$_})
			{
			$no_annotation+=$reads;
			$no_annotation_seqs{$_}=$reads;
			}
		}
	close INPUT;
	foreach(sort{$no_annotation_seqs{$b}<=>$no_annotation_seqs{$a}}keys%no_annotation_seqs)
		{
		print NO_ANNOTATION">$no_annotation_seqs{$_}\n$_\n";
		}
	close NO_ANNOTATION;
	$ba_in="$dump_folder/$outfile_prefix.no-annotation.fas";
	$ba_out="$dump_folder/$outfile_prefix.no-annotation.info";
	basic_analysis();

	print NCRNA_TABLE"no annotation\t$no_annotation\n";
	# NCRNA_TABLE is finished when piRNA cluster matching reads are added
	if(!$silent){print" OK.";}

	# map reads without annotation to piRNA cluster sequences (if available and if $no_annotation!=0)
	if(-e"$dump_folder/piRclDB"&&!$skip_piR&&$no_annotation!=0)
		{
		if(!$silent){print"\nMap sequences without annotation to $species{$species} piRNA cluster sequences...";}
		if($^O eq'linux'||$^O eq'darwin')
			{
			qx/.\/seqmap.exe $max_miRtail_length $dump_folder\/$outfile_prefix.no-annotation.fas $dump_folder\/piRclDB $dump_folder\/SeqMap_piR.temp \/output_all_matches \/available_memory:${memory}000 $less_memory_options_ncR> $dump_folder\/SeqMap_report4.txt/;
			}
		elsif($^O eq'MSWin32')
			{
			system("seqmap.exe $max_miRtail_length $dump_folder/$outfile_prefix.no-annotation.fas $dump_folder/piRclDB $dump_folder/SeqMap_piR.temp /output_all_matches /available_memory:${memory}000 ${less_memory_options_ncR}2>&1 > $dump_folder/SeqMap_report4.txt");
			}
		else
			{
			if(!$silent){print" WARNING! Unknown operating system.";}
			# try unix command
			qx/.\/seqmap.exe $max_miRtail_length $dump_folder\/$outfile_prefix.no-annotation.fas piRclDB $dump_folder\/SeqMap_piR.temp \/output_all_matches \/available_memory:${memory}000 $less_memory_options_ncR> $dump_folder\/SeqMap_report4.txt/;
			if(!-e"$dump_folder/SeqMap_species.temp")
				{
				# try windows command
				system("seqmap.exe $max_miRtail_length $dump_folder/$outfile_prefix.no-annotation.fas piRclDB $dump_folder/SeqMap_piR.temp /output_all_matches /available_memory:${memory}000 ${less_memory_options_ncR}2>&1 > $dump_folder/SeqMap_report4.txt");
				}
			}

		# check SeqMap report
		open(SMR,"$dump_folder/SeqMap_report4.txt");
		while(<SMR>)
			{
			if($_=~/not enough memory/)
				{
				print"\n\nERROR. Not enough memory available for SeqMap. Check $dump_folder/SeqMap_report4.txt for more details. Try less mismatch (-mismatch [n]), less insertions/deletions (-insdel [n]), fewer non-template 3' nucleotides (-tail [n]). Alternatively you may want to use the options -less_memory or -even_less_memory.\n\n";
				exit;
				}
			}
		close SMR;

		# clean up SeqMap output (same as for miR)
		$file_id="piR";
		clean_SeqMap_miRNA();

		my%hits_per_piR;
		open(SEQMAP,"$dump_folder/SeqMap_piR.clean-2.temp")||die print"\nERROR! Unable to open $dump_folder/SeqMap_piR.clean-2.temp.\n$!\n\n";
		while(<SEQMAP>)
			{
			$_=~s/\s*$//;
			@seqmap=split("\t",$_);
			$hits_per_piR{$seqmap[4]}++;
			}
		close SEQMAP;

		open(SEQMAP,"$dump_folder/SeqMap_piR.clean-2.temp")||die print"\nERROR! Unable to open $dump_folder/SeqMap_piR.clean-2.temp.\n$!\n\n";
		my%pi_tails;
		my%piRNA_candidates;
		my%piRNA_header;
		while(<SEQMAP>)
			{
			$_=~s/\s*$//;
			@seqmap=split("\t",$_);

			$parsed_loc=$seqmap[0];
			$parsed_loc=~s/ /_/g;
			if($global_annotation_matrix{$seqmap[4]}{$seqmap[3]}!~/piR-cluster:$parsed_loc/){$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="piR-cluster:$parsed_loc\t";}

			# 3' non-template hit to piRNA cluster
			if($seqmap[5]>0)
				{
				foreach$p((length$seqmap[2])-$max_miRtail_length..(length$seqmap[2])-1)
					{
					if(substr($seqmap[2],$p,1) ne substr($seqmap[4],$p,1))
						{
						$tail=substr($seqmap[4],$p);
						$pi_tails{$tail}+=$seqmap[3]/$hits_per_piR{$seqmap[4]};
						last;
						}
					}
				}
			$piRNA_candidates{$seqmap[4]}=$seqmap[3];
			$piRNA_header{$seqmap[4]}.="$seqmap[0]|";
			}
		close SEQMAP;

		# make piRNA candidates tail file
		open(OUT_PITAILS,">$dump_folder/$outfile_prefix.piRcandidates-tails_$species{$species}.txt")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.piRcandidates-tails_$species{$species}.txt.\n$!\n\n";
		foreach$tail(@combs)
			{
			if(!$pi_tails{$tail})
				{
				$pi_tails{$tail}=0;
				}
			print OUT_PITAILS"$tail\t$pi_tails{$tail}\n";
			}
		close OUT_PITAILS;

		# make piRNA candidates fasta file
		open(PIR_CANDIDATES,">$dump_folder/$outfile_prefix.piRcandidates.fas")||die print"\nERROR! Unable to create output file $dump_folder/$outfile_prefix.piRcandidates.fas.\n$!\n\n";
		$piRc=0;
		foreach$piR(sort{$piRNA_candidates{$b}<=>$piRNA_candidates{$a}}keys%piRNA_candidates)
			{
			$piRc+=$piRNA_candidates{$piR};
			$piRNA_header{$piR}=~s/\|$//;
			print PIR_CANDIDATES">$piRNA_candidates{$piR}:$piRNA_header{$piR}\n$piR\n";
			}
		close PIR_CANDIDATES;
		$ba_in="$dump_folder/$outfile_prefix.piRcandidates.fas";
		$ba_out="$dump_folder/$outfile_prefix.piRcandidates.info";
		basic_analysis();
		print NCRNA_TABLE"   mapped to piRNA producing loci\t$piRc\n";
		print" done.";
		}

	# search for phased RNAs
	if(@phasi>0)
		{
		if(!$silent){print"\nSearch for phased RNAs in $input_map:";}
		if($na_for_phasi)
			{
			$info_for_phasi_out="This is a subset of mapped reads without annotation.";
			}
		else
			{
			$info_for_phasi_out="This is a subset of all mapped reads.";
			}
		reformat_input_map();
		sub reformat_input_map
			{
			$lines_in_phasi_map=0;
			# create new input map file (and save reads and hits per sequence) if input map is not ELAND3 from collapsed data
			if($format eq'SAM')
				{
				if(!$silent){print"\n -check input map file...";}
				%lines_in_map=();
				open(INPUT_MAP,$input_map)||die print"\nERROR! Unable to open input map file $input_map.\n$!\n\n";
				while(<INPUT_MAP>)
					{
					next if($_=~/^@/); # skip SAM header lines
					@sam=split("\t",$_);
					next if ($sam_mapped{$sam[1]}==0||$sam[3]==0); # skip unmapped reads

					# reverse complement if sequence was mapped to the minus strand (SAM gives sequence of the locus, not of the mapped sequence)
					if($sam_strand{$sam[1]}eq'-')
						{
						$sam[9]=reverse$sam[9];
						$sam[9]=~tr/ATGC/TACG/;
						}

					#print"\n$sam[9]:$input_map_seq{$sam[9]}";
					$lines_in_map{$sam[2]}{$sam[3]}{$input_map_seq{$sam[9]}}{$sam[9]}=$sam_strand{$sam[1]};
					}
				close INPUT_MAP;
				create_phasi_map();
				}
			elsif($format eq'ELAND3'&&$all_identifiers_are_numeric==0)
				{
				if(!$silent){print"\n -check input map file...";}
				%lines_in_map=();
				open(INPUT_MAP,$input_map)||die print"\nERROR! Unable to open input map file $input_map.\n$!\n\n";
				while(<INPUT_MAP>)
					{
					next if($_=~/^trans_id/||$_=~/^#/); # skip ELAND3 header lines
					$_=~s/\s*$//;
					@eland3=split("\t",$_);
					$lines_in_map{$eland3[0]}{$eland3[1]}{$input_map_seq{$eland3[4]}}{$eland3[4]}=$eland3[6];
					}
				close INPUT_MAP;
				create_phasi_map();
				}
			else
				{
				if(!$silent){print"\n -check input map file...";}
				%hits_per_seq=();
				open(INPUT_MAP,$input_map);
				while(<INPUT_MAP>)
					{
					next if($_=~/^trans_id/);
					@map=split("\t",$_);
					$hits_per_seq{$map[4]}++;
					$lines_in_phasi_map++;
					}
				close INPUT_MAP;
				if(!$silent){print" OK. No reformat required.";}
				}

			sub create_phasi_map
				{
				if(!$silent){print" OK.\n -reformat input map file...";}
				%hits_per_seq=();
				open(REFORMAT_MAP,">$dump_folder/eland3collapsed.map")||die print"\nERROR! Unable to create sorted map file for phasiRNA detection.\n$!\n\n";
				foreach$chr(sort{$a cmp $b}keys%lines_in_map)
					{
					foreach$pos(sort{$a<=>$b}keys%{$lines_in_map{$chr}})
						{
						foreach$reads(keys%{$lines_in_map{$chr}{$pos}})
							{
							foreach$seq(keys%{$lines_in_map{$chr}{$pos}{$reads}})
								{
								$hits_per_seq{$seq}++;
								print REFORMAT_MAP"$chr\t$pos\t-\t$reads\t$seq\t-\t$lines_in_map{$chr}{$pos}{$reads}{$seq}\n";
								$lines_in_phasi_map++;
								}
							}
						}
					}
				close REFORMAT_MAP;
				undef%lines_in_map;
				$input_map="$dump_folder/eland3collapsed.map";
				if(!$silent){print" OK.";}
				}
			}

		if(!$silent){print"\n -read map file with sliding window (sw size=$phasi_sw_size bp).\n
 0%                                  50%                                  100%
 |------------------------------------|------------------------------------|
 ";}
		open(INPUT_MAP,$input_map);
		$phasi_progress_lines=0;
		$progress=0;
		while(<INPUT_MAP>)
			{
			next if($_=~/^trans_id/);
			$_=~s/\s*$//;
			@map=split("\t",$_);

			# give information on progress
			$phasi_progress_lines++;
			if(!$silent&&$phasi_progress_lines>=$lines_in_phasi_map/75)
				{
				if(!$silent){print".";}
				$phasi_progress_lines=0;
				$progress++;
				}

			# consider only reads with correct length for phasiRNA prediction
			if($phasi{length$map[4]}==1)
				{
				if($no_annotation_seqs{$map[4]}||!$na_for_phasi)# optionally consider only reads without ncRNA annotation
					{
					$coordinate_unified=$map[1];
					if($map[6]eq'-')
						{
						$coordinate_unified=$map[1]+2; # combine + and - strand mapped reads
						}
					$pos_offset=int(((($coordinate_unified/(length$map[4]))-(int($coordinate_unified/(length$map[4]))))*length$map[4])+0.5);

					# each locus belongs to two sliding windows
					$phasiRNA_readcount{$map[0]}{length$map[4]}{$pos_offset}{(int($coordinate_unified/$phasi_sw_size))+1}+=$map[3]/$hits_per_seq{$map[4]};
					$phasiRNA_readcount{$map[0]}{length$map[4]}{$pos_offset}{(int(($coordinate_unified+($phasi_sw_size/2))/$phasi_sw_size))+1}+=$map[3]/$hits_per_seq{$map[4]};
					$phasi_positions{$map[0]}{length$map[4]}{$pos_offset}{(int($coordinate_unified/$phasi_sw_size))+1}{$coordinate_unified}+=$map[3]/$hits_per_seq{$map[4]};
					$phasi_positions{$map[0]}{length$map[4]}{$pos_offset}{(int(($coordinate_unified+($phasi_sw_size/2))/$phasi_sw_size))+1}{$coordinate_unified}+=$map[3]/$hits_per_seq{$map[4]};

					# additionally save read counts from one strand only to exclude loci with suspicious strand bias
					if($map[6]eq'+')
						{
						$phasiRNA_readcount_plus{$map[0]}{length$map[4]}{$pos_offset}{(int($coordinate_unified/$phasi_sw_size))+1}+=$map[3]/$hits_per_seq{$map[4]};
						$phasiRNA_readcount_plus{$map[0]}{length$map[4]}{$pos_offset}{(int(($coordinate_unified+($phasi_sw_size/2))/$phasi_sw_size))+1}+=$map[3]/$hits_per_seq{$map[4]};
						}

					# save that for calculation of p with binomialdistriburion
					$total_readcount_phasilength{$map[0]}{length$map[4]}{(int($coordinate_unified/$phasi_sw_size))+1}+=$map[3]/$hits_per_seq{$map[4]};
					$total_readcount_phasilength{$map[0]}{length$map[4]}{(int(($coordinate_unified+($phasi_sw_size/2))/$phasi_sw_size))+1}+=$map[3]/$hits_per_seq{$map[4]};

					# count total number of sliding windows for Bonferroni correction of multiple testing
					$total_sw_count{$map[0]}{(int($coordinate_unified/$phasi_sw_size))+1}=1;
					}
				}
			$total_counts{$map[0]}{(int($coordinate_unified/500))+1}+=$map[3]/$hits_per_seq{$map[4]};
			$total_counts{$map[0]}{(int(($coordinate_unified+250)/500))+1}+=$map[3]/$hits_per_seq{$map[4]};
			}
		if(!$silent)
			{
			foreach($progress..74)
				{
				if(!$silent){print".";}
				}
			if(!$silent){print"\n\n";}
			}
		close INPUT_MAP;

		# create hash with factorials
		use bignum;
		$val=1;
		$fac{0}=1;
		foreach(1..$max_factorials_stored)
		    {
		    $val=$val*$_;
		    $fac{$_}=$val;
		    }

		# calculate Bonferroni-corrected p-value
		$total_sw_count=0;
		foreach$loc(keys%total_sw_count)
			{
			$total_sw_count+=keys%{$total_sw_count{$loc}};
			}
		$bonferroni_p=$phasi_p/$total_sw_count;
		$bonferroni_p_print=(int(($bonferroni_p*(10**15))+0.5))/(10**15);

		if(!$silent){print"\n\n -check for significant phasing signature.\n   -> Initial p set by user: $phasi_p\n   -> After corr. for multiple testing ($total_sw_count windows): $bonferroni_p_print\n
 0%                                  50%                                  100%
 |------------------------------------|------------------------------------|
 ";}
		# go through sliding windows
		my%getphasi=();
		my%sufficient_k=();
		my%insufficient_k=();
		$number_of_tests=0;
		$phasi_progress_windows=0;
		foreach$location(keys%phasiRNA_readcount)
			{
			foreach$phasi_length(@phasi)
				{
				foreach$frame(1..$phasi_length)
					{
					$number_of_tests+=keys%{$phasiRNA_readcount{$location}{$phasi_length}{$frame}};
					}
				}
			}
		$progress=0;
		foreach$location(keys%phasiRNA_readcount)
			{
			foreach$phasi_length(@phasi)
				{
				$pphased=1/$phasi_length;
				$pnotphased=1-$pphased;
				foreach$frame(1..$phasi_length)
					{
					foreach$sw_id(keys%{$phasiRNA_readcount{$location}{$phasi_length}{$frame}})
						{
						# skip windows with suspicious strand bias if they have only few phased loci
						$bad_strand_bias=0;
						if($phasiRNA_readcount_plus{$location}{$phasi_length}{$frame}{$sw_id}/$phasiRNA_readcount{$location}{$phasi_length}{$frame}{$sw_id}>$max_allowed_strandbias||$phasiRNA_readcount_plus{$location}{$phasi_length}{$frame}{$sw_id}/$phasiRNA_readcount{$location}{$phasi_length}{$frame}{$sw_id}<1-$max_allowed_strandbias)
							{
							$bad_strand_bias=1;
							}

						# optionally apply further thresholds: minimum number of loci, minimum fraction of phased reads in sw
						if(keys%{$phasi_positions{$location}{$phasi_length}{$frame}{$sw_id}}>=$min_phasi_loci&&$phasiRNA_readcount{$location}{$phasi_length}{$frame}{$sw_id}>=$total_counts{$location}{$sw_id}*$min_phasi_fraction)
							{
							# check p from binomialdistribution / round phased reads and all reads to nearest integer
							$k_observed=int($phasiRNA_readcount{$location}{$phasi_length}{$frame}{$sw_id}+0.5);
							$n=int($total_readcount_phasilength{$location}{$phasi_length}{$sw_id}+0.5);
							next if($k_observed<$min_phasi_reads);
							if($n>$max_factorials_stored)
								{
								if($calc_big_factorials&&!$fac{$n}) # calculate huge factorial if it was not calculated before
									{
									$fac_huge=$fac{$max_factorials_stored};
									foreach($max_factorials_stored+1..$n)
										{
										$fac_huge=$fac_huge*$_;
										}
									$fac{$n}=$fac_huge;
									}
								else # scale down if n is bigger than the highest factorial available
									{
									$k_observed=int(($k_observed/($n/$max_factorials_stored))+0.5);
									$n=$max_factorials_stored;
									}
								}

							# skip calculation if significant k for n was calculated before AND no need for calculation of loci in line
							if($sufficient_k{$n}&&$sufficient_k{$n}<=$k_observed&&$min_in_line<2&&$bad_strand_bias==0)
								{
								save_phasi_positions();
								}
							# skip calculation if higher k was found to be insufficient for this n before
							elsif(!$insufficient_k{$n}||$insufficient_k{$n}<$k_observed)
								{
								$k=-1;
								$p=0;
								while(1)
									{
									$k++;
									$p+=($fac{$n}/($fac{$k}*$fac{$n-$k}))*($pphased**$k)*($pnotphased**($n-$k));
									if($p>1-$bonferroni_p&&$k<$n) # last, sufficient (significant) k is $k+1
										{
										$sufficient_k{$n}=$k+1; # safe sufficient number of k for n total reads
										$max_in_line=1;
										$current_in_line=0;
										if($min_in_line>1)
											{
											foreach$pos(sort{$a<=>$b}keys%{$phasi_positions{$location}{$phasi_length}{$frame}{$sw_id}})
												{
												if($current_in_line==0||$prev_pos+$phasi_length==$pos)
													{
													$current_in_line++;
													if($current_in_line>$max_in_line)
														{
														$max_in_line=$current_in_line;
														}
													}
												else
													{
													$current_in_line=0;
													}
												$prev_pos=$pos;
												}
											}
										if($max_in_line>=$min_in_line&&$bad_strand_bias==0)
											{
											save_phasi_positions();
											sub save_phasi_positions
												{
												foreach$pos(sort{$a<=>$b}keys%{$phasi_positions{$location}{$phasi_length}{$frame}{$sw_id}})
													{
													$getphasi{$location}{$phasi_length}{$pos}=$frame;
													}
												}
											}
										elsif($bad_strand_bias==1)
											{
											if($min_in_line>=5||$min_phasi_loci>=25)
												{
												save_phasi_positions();
												}
											}
										last;
										}
									elsif($k==$k_observed) # last with non-significant p
										{
										# safe in-sufficient number of k for n total reads
										if(!$insufficient_k{$n}||$insufficient_k{$n}<$k)
											{
											$insufficient_k{$n}=$k;
											}
										$p_out_test=1-((int(($p*(10**10))+0.5))/(10**10));
										$bonferroni_out_test=(int(($bonferroni_p*(10**10))+0.5))/(10**10);
										last;
										}
									}
								}
							}
						$phasi_progress_windows++;
						if($phasi_progress_windows>=$number_of_tests/75)
							{
							if(!$silent){print".";}
							$progress++;
							$phasi_progress_windows=0;
							}
						}
					}
				}
			}
		if(!$silent)
			{
			foreach($progress..74)
				{
				if(!$silent){print".";}
				}
			if(!$silent){print"\n\n";}
			}

		$phasi_cluster_id=0;
		$prev_phasi_location="";
		$prev_phasi_coordinate=0;
		$prev_phasi_seq="";
		$loci_occupied=0;
		$cluster_reads_total=0;
		$cluster_reads_normalized=0;
		if(keys%getphasi>0)
			{
			open(PHASI_SORTED,">$dump_folder/$outfile_prefix.phasiRNA.sorted.fas")||print"\nERROR! Unable to open output file $dump_folder/$outfile_prefix.phasiRNA.sorted.fas.\n$!\n\n";
			open(PHASI_ALL,">$dump_folder/$outfile_prefix.phasiRNA.fas")||print"\nERROR! Unable to open output file $dump_folder/$outfile_prefix.phasiRNA.sorted.fas.\n$!\n\n";
			open(PHASI_ALIGN,">$dump_folder/$outfile_prefix.phasiRNA.align")||print"\nERROR! Unable to open output file $dump_folder/$outfile_prefix.phasiRNA.align.\n$!\n\n";
			print PHASI_ALIGN"CLUSTERS OF PHASED RNAs.\nGenomic sequence is represented by the center line.\nReads above the center line are aligned to the + strand and stated in 5'-3' orientation.\nReads below the center line are aligned to the - strand and stated in 3'-5' orientation.\nInformation on sequence read counts and genomic hits per sequence are given in the following format at the beginning of each sequence line: [sequence read count/genomic hits]\n\n";
			my%phasiRNAs=();
			$phasiRNA_readcount=0;
			if(!$silent){print"\n\n -collect putative phasiRNAs...";}
			$phasi_align_out_plus="";
			$phasi_align_out_minus="";
			open(INPUT_MAP,$input_map);
			while(<INPUT_MAP>)
				{
				$_=~s/\s*$//;
				@map=split("\t",$_);
				if($map[6]eq'+'&&$getphasi{$map[0]}{length$map[4]}{$map[1]})
					{
					report_phasi_cluster();
					$loci_occupied++;
					$cluster_reads_total+=$map[3];
					$cluster_reads_normalized+=$map[3]/$hits_per_seq{$map[4]};
					$read_info_head=substr("                  $map[3]/$hits_per_seq{$map[4]} ",-20);
					$whitespace=" " x ($map[1]-$current_cluster_start);
					$phasi_align_out_plus="${read_info_head}${whitespace}$map[4]\n".$phasi_align_out_plus;
					$print_frame=$getphasi{$map[0]}{length$map[4]}{$map[1]};
					report_phasi_sequence();
					}
				elsif($map[6]eq'-'&&$getphasi{$map[0]}{length$map[4]}{$map[1]+2})
					{
					report_phasi_cluster();
					$loci_occupied++;
					$cluster_reads_total+=$map[3];
					$cluster_reads_normalized+=$map[3]/$hits_per_seq{$map[4]};
					$read_info_head=substr("                  $map[3]/$hits_per_seq{$map[4]} ",-20);
					$seq3to5=reverse$map[4];
					$whitespace=" " x ($map[1]-$current_cluster_start);
					$phasi_align_out_minus.="\n${read_info_head}${whitespace}$seq3to5";
					$print_frame=$getphasi{$map[0]}{length$map[4]}{($map[1])+2};
					report_phasi_sequence();
					}

				# REPORT THE CLUSTER TO OUTFILE
				sub report_phasi_cluster
					{
					if($map[0]ne$prev_phasi_location||$map[1]-$min_distance_between_clusters>=$prev_phasi_coordinate||$print_frame!=$prev_frame) # new cluster if chromosome/scaffold changes, distance > threshold, different phasing frame
						{
						if($phasi_cluster_id>0)
							{
							$current_cluster_end=$prev_phasi_coordinate+(length$prev_phasi_seq)-1;
							$cluster_size=$current_cluster_end-$current_cluster_start+1;
							$center_line="                    ".">" x ($current_cluster_end-$current_cluster_start+1);
							$center_line.="\n                    "."<" x ($current_cluster_end-$current_cluster_start+1);
							print PHASI_ALIGN"cluster_ID:$phasi_cluster_id\tlocation:$current_cluster_location\tstart:$current_cluster_start\tend:$current_cluster_end\tsize:$cluster_size bp\tnon-identical_sequences:$loci_occupied\treads:$cluster_reads_total\treads_normalized:$cluster_reads_normalized\n${phasi_align_out_plus}${center_line}$phasi_align_out_minus\n\n";
							$loci_occupied=0;
							$cluster_reads_total=0;
							$cluster_reads_normalized=0;
							}
						$phasi_align_out_plus="";
						$phasi_align_out_minus="";
						$phasi_cluster_id++;
						$current_cluster_location=$map[0];
						$current_cluster_start=$map[1];
						}
					}

				# REPORT EACH SEQUENCE TO SORTED FASTA FILE
				sub report_phasi_sequence
					{
					$l=length$map[4];
					print PHASI_SORTED">length[nt]:$l, location:$map[0], position:$map[1], cluster_ID:$phasi_cluster_id, strand:$map[6], frame:$print_frame, read_count:$map[3], genomic_hits:$hits_per_seq{$map[4]}\n$map[4]\n";
					$phasiRNAs{$map[4]}=$map[3];
					$prev_phasi_location=$map[0];
					$prev_phasi_coordinate=$map[1];
					$prev_frame=$print_frame;
					$prev_phasi_seq=$map[4];
					}
				}
			close INPUT_MAP;
			foreach$seq(sort{$phasiRNAs{$b}<=>$phasiRNAs{$a} or $a cmp $b} keys%phasiRNAs)
				{
				$phasiRNA_readcount+=$phasiRNAs{$seq};
				print PHASI_ALL">$phasiRNAs{$seq}\n$seq\n";
				$global_annotation_matrix{$seqmap[4]}{$seqmap[3]}.="phasiRNA\t";
				}
			close PHASI_SORTED;
			close PHASI_ALL;
			close PHASI_ALIGN;
			undef%phasiRNAs;
			undef%phasiRNA_readcount;
			undef%getphasi;
			undef%sufficient_k;
			undef%insufficient_k;
			$ba_in="$dump_folder/$outfile_prefix.phasiRNA.fas";
			$ba_out="$dump_folder/$outfile_prefix.phasiRNA.info";
			basic_analysis();
			if($phasiRNA_readcount>0)
				{
				print NCRNA_TABLE"phasiRNAs\t$phasiRNA_readcount\tNOTE: $info_for_phasi_out\n";
				}
			if(!$silent){print" OK.\nFound $phasiRNA_readcount putative phasiRNAs organized in $phasi_cluster_id distinct clusters.\n";}
			}
		else
			{
			if(!$silent){print" no phased reads found.";}
			}
		no bignum;
		}
	undef%no_annotation_seqs;

	open(TARGET_COUNTS,">$dump_folder/$outfile_prefix.hits_per_target.txt")||print"\nError. Unable to create output file $dump_folder/$outfile_prefix.hits_per_target.txt.\n$!\n\n";
	print TARGET_COUNTS"TRANSCRIPT_CLASS\tTRANSCRIPT_NAME\tREAD_COUNT\n";
	foreach$biotype(sort{$a cmp $b}keys%ncRNA_counts_2)
		{
		foreach$id(sort{$ncRNA_counts_2{$biotype}{$b}<=>$ncRNA_counts_2{$biotype}{$a}}keys%{$ncRNA_counts_2{$biotype}})
			{
			print TARGET_COUNTS"$biotype\t$id\t$ncRNA_counts_2{$biotype}{$id}\n";
			}
		}
	close TARGET_COUNTS;
	close NCRNA_TABLE;

	# check ping-pong signature
	if($check_pp_signature)
		{
		if(!$silent){print"\nCheck for ping-pong signature in sequence datasets:";}
		if(!-e"$dump_folder/eland3collapsed.map") # reformat mapfile if that was not done before (phasiRNA prediction)
			{
			reformat_input_map();
			}
		opendir(WIZDIR,$dump_folder);
		%output=();
		while($out_file=readdir(WIZDIR))
			{
			next if($out_file!~/^$outfile_prefix\..+\.fas$/);
			$class=$out_file;
			$class=~s/^$outfile_prefix\.//;
			$class=~s/\.fas$//;
			open(OUTFASTA,"$dump_folder/$out_file")||print"\nERROR! Unable to open output file $out_file.\n$!\n\n";
			while(<OUTFASTA>)
				{
				if($_!~/^>/)
					{
					$_=~s/\s*$//;
					push(@{$output{$_}},$class);
					}
				}
			close OUTFASTA;
			}
		closedir WIZDIR;

		$max_length=0;
		$mapfile_hits=0;
		my%hits_per_seq=();
		open(INPUT_MAP,$input_map);
		while(<INPUT_MAP>)
			{
			next if($_=~/^trans_id/||$_=~/^#/);
			$_=~s/\s*$//;
			@d=split("\t",$_);
			$mapfile_hits++;
			$hits_per_seq{$d[4]}++;
			$reads_per_seq{$d[4]}=$d[3];
			if(length$d[4]>$max_length) # save max. sequence length (-> maximum overlap)
				{
				$max_length=length$d[4];
				}
			}
		close INPUT_MAP;
		if(!$silent){print"

 Scan input map file for 5' overlaps:
 0%                                  50%                                  100%
 |------------------------------------|------------------------------------|
 "};
		@sw=();
		$progress=0;
		$prev_loc="";
		$processed_hits=0;
		$min_length=-1;
		%overlap=();
		%ppmatrix=();
		%ppsequences=();
		open(INPUT_MAP,$input_map)||print"\nERROR! Unable to open input map file $input_map.\n$!\n\n";
		while(<INPUT_MAP>)
			{
			next if($_=~/^trans_id/||$_=~/^#/);
			@d=split("\t",$_);
			if($min_length==-1||length$d[4]<$min_length)
				{
				$min_length=length$d[4];
				}
			if($d[0]ne$prev_loc) # empty sliding window when location changes
				{
				@sw=();
				}
			if($d[6]=~/-/) # add hit to sliding window if current hit is on the minus strand
				{
				push(@sw,[$d[1],$d[4],$d[8]]); # sliding window entries comprise coordinate|sequence|allocated reads (if available)
				while(1) # check if hit(s) must be removed from sliding window
					{
					if(@sw>1&&$d[1]>($sw[0][0])+$max_length)
						{
						shift@sw;
						}
					else
						{
						last;
						}
					}
				}
			else # count overlaps in the sliding window if current hit is on the plus strand
				{
				foreach$i(0..@sw-1)
					{
					if($sw[$i][0]+(length$sw[$i][1])>$d[1]&&$sw[$i][0]+(length$sw[$i][1])<$d[1]+length$d[4]) # check whether hits overlap
						{
						# global pp signature
						$overlap{'global'}{$sw[$i][0]+(length$sw[$i][1])-$d[1]}+=(($reads_per_seq{$d[4]}/$hits_per_seq{$d[4]})*($reads_per_seq{$sw[$i][1]}/$hits_per_seq{$sw[$i][1]}));

						if(($sw[$i][0]+(length$sw[$i][1])-$d[1])==10)
							{
							# ping pong matrix (which length fraction plays ping pong with which length fraction)
							$pp_length_1=length$d[4];
							$pp_length_2=length$sw[$i][1];
							$ppmatrix{'global'}{$pp_length_1}{$pp_length_2}+=(($reads_per_seq{$d[4]}/$hits_per_seq{$d[4]})*($reads_per_seq{$sw[$i][1]}/$hits_per_seq{$sw[$i][1]}));
							$ppmatrix{'global'}{$pp_length_2}{$pp_length_1}+=(($reads_per_seq{$d[4]}/$hits_per_seq{$d[4]})*($reads_per_seq{$sw[$i][1]}/$hits_per_seq{$sw[$i][1]}));

							# save ping pong players
							$ppsequences{$d[4]}=$reads_per_seq{$d[4]};
							$ppsequences{$sw[$i][1]}=$reads_per_seq{$sw[$i][1]};
							}

						# pp signature per RNA class
						foreach$class_seq1(@{$output{$d[4]}})
							{
							foreach$class_seq2(@{$output{$sw[$i][1]}})
								{
								if($class_seq1 eq $class_seq2)
									{
									$overlap{$class_seq1}{$sw[$i][0]+(length$sw[$i][1])-$d[1]}+=(($reads_per_seq{$d[4]}/$hits_per_seq{$d[4]})*($reads_per_seq{$sw[$i][1]}/$hits_per_seq{$sw[$i][1]}));

									if(($sw[$i][0]+(length$sw[$i][1])-$d[1])==10)
										{
										# ping pong matrix per RNA class (which length fraction plays ping pong with which length fraction)
										$ppmatrix{$class_seq1}{$pp_length_1}{$pp_length_2}+=(($reads_per_seq{$d[4]}/$hits_per_seq{$d[4]})*($reads_per_seq{$sw[$i][1]}/$hits_per_seq{$sw[$i][1]}));
										$ppmatrix{$class_seq1}{$pp_length_2}{$pp_length_1}+=(($reads_per_seq{$d[4]}/$hits_per_seq{$d[4]})*($reads_per_seq{$sw[$i][1]}/$hits_per_seq{$sw[$i][1]}));
										}
									}
								}
							}
						}
					}
				}
			$processed_hits++;
			if($processed_hits>=$mapfile_hits/75&&!$silent)
				{
				$progress++;
				$processed_hits=0;
				if(!$silent){print".";}
				}
			$prev_loc=$d[0];
			}
		close INPUT_MAP;
		if(!$silent)
			{
			foreach($progress..74)
				{
				if(!$silent){print".";}
				}
			if(!$silent){print"\n\n";}
			}

		foreach$RNAclass(keys%overlap)
			{
			$pp_results="";
			$pp_results.="Total scores for 5' overlap:\noverlap [nt]\tread pairs\n";
			$average_background=0;
			foreach(1..($max_length-1))
				{
				if($_>=1&&$_<=9)
					{
					$average_background+=$overlap{$RNAclass}{$_};
					}
				elsif($_>=11&&$_<=20)
					{
					$average_background+=$overlap{$RNAclass}{$_};
					}
				$pp_results.="$_\t$overlap{$RNAclass}{$_}\n";
				}
			$average_background=$average_background/19;
			$variance=0;
			foreach(1..($max_length-1))
				{
				if($_>=1&&$_<=9)
					{
					$variance+=($overlap{$RNAclass}{$_}-$average_background)**2;
					}
				elsif($_>=11&&$_<=20)
					{
					$variance+=($overlap{$RNAclass}{$_}-$average_background)**2;
					}
				}
			$variance=$variance/19;
			$stddeviation=sqrt($variance);
			if($stddeviation>0)
				{
				$Zscore=($overlap{$RNAclass}{10}-$average_background)/$stddeviation;
				}
			else
				{
				if($overlap{$RNAclass}{10}>0)
					{
					$Zscore="infinite";
					}
				else
					{
					$Zscore=0;
					}
				}
			$pp_results.="
Z-score calculation:
Average background (overlaps 1-9+11-20): $average_background
Variance background: $variance
Standard deviation: $stddeviation
Ping-Pong Z-Score: $Zscore

Z-score >= 1.6449 (one-tailed hypothesis) -> p < 0.05
Z-score >= 2.3264 (one-tailed hypothesis) -> p < 0.01
";

			open(PP_OUT,">$dump_folder/$outfile_prefix.$RNAclass.pp")||print"\nERROR! Unable to open $dump_folder/$outfile_prefix.$RNAclass.pp.\n$!\n\n";
			print PP_OUT $pp_results;
			close PP_OUT;

			open(PPMATRIX,">$dump_folder/$outfile_prefix.$RNAclass.ppmatrix")||print"\nERROR! Unable to open $dump_folder/$outfile_prefix.$RNAclass.ppmatrix.\n$!\n\n";
			print PPMATRIX"This matrix shows the 10 nt 5' overlap scores produced by each sequence length pair.\nExample: A peak at 29x29 suggests homotypic ping-pong incl. one PIWI paralog, a peak at 24x29 suggests heterotypic ping-pong incl. two PIWI paralogs.\n\n";
			foreach$l($min_length..($max_length-1))
				{
				print PPMATRIX"\t$l";
				}
			foreach$l1($min_length..($max_length-1))
				{
				print PPMATRIX"\n$l1";
				foreach$l2($min_length..($max_length-1))
					{
					if(!$ppmatrix{$RNAclass}{$l1}{$l2})
						{
						$ppmatrix{$RNAclass}{$l1}{$l2}="n.a.";
						}
					print PPMATRIX"\t$ppmatrix{$RNAclass}{$l1}{$l2}";
					}
				}
			close PPMATRIX;
			}

		open(PPSEQUENCES,">$dump_folder/$outfile_prefix.ping-pong.fas")||print"\nERROR! Unable to open $dump_folder/$outfile_prefix.ping-pong.fasta.\n$!\n\n";
		foreach$seq(sort{$ppsequences{$b}<=>$ppsequences{$a}}keys%ppsequences)
			{
			print PPSEQUENCES">$ppsequences{$seq}\n$seq\n"
			}
		close PPSEQUENCES;

		undef%overlap;
		undef%ppmatrix;
		undef%ppsequences;
		}

	# remove temporary files from dump folder
	if(!$keep_temp)
		{
		foreach$i(1..4)
			{
			unlink("$dump_folder/SeqMap_report$i.txt");
			}
		unlink("$dump_folder/SeqMap_species.temp");
		unlink("$dump_folder/SeqMap_nonspecies.temp");
		unlink("$dump_folder/SeqMap_ncRNA.temp");
		unlink("$dump_folder/SeqMap_ncRNA.clean.temp");
		unlink("$dump_folder/SeqMap_piR.temp");
		foreach$i(1..2)
			{
			unlink("$dump_folder/SeqMap_species.clean-$i.temp");
			unlink("$dump_folder/SeqMap_nonspecies.clean-$i.temp");
			unlink("$dump_folder/SeqMap_piR.clean-$i.temp");
			unlink("$dump_folder/SeqMap_ncRNA.clean-$i.temp");
			unlink("$dump_folder/SeqMap_input$i.temp");
			}
		unlink("$dump_folder/piRclDB");
		unlink("$dump_folder/miRBase_mature");
		unlink("$dump_folder/miRBase_hairpin_$species{$species}.fas");
		unlink("$dump_folder/miRBase_hairpin_non-$species{$species}.fas");
		unlink("$dump_folder/ncRNA.fas");
		unlink("$dump_folder/ncRNA.fas.CCAcheck");
		unlink("$dump_folder/cDNA.fas");
		}

	# create global annotation matrix
	open(MATRIX,">$dump_folder/$outfile_prefix.full_annotation_matrix.txt")||print"\nError. Unable to create annotation matrix file.\n$!\n\n";
	print MATRIX"sequence\treads\tannotation(s)";
	foreach$seq(sort{$a cmp $b}keys%global_annotation_matrix)
		{
		print MATRIX"\n$seq";
		foreach$reads(keys%{$global_annotation_matrix{$seq}})
			{
			print MATRIX"\t$reads\t$global_annotation_matrix{$seq}{$reads}";
			}
		}
	close MATRIX;

	# make simplified miRNA count table
	open(MIR_TABLE,"$dump_folder/unitas.miR-table_$species{$species}.txt")||print$!;
	open(MIR_TABLE_SIMPLIFIED,">$dump_folder/miR-table_$species{$species}.simplified.txt")||print$!;
	$head=<MIR_TABLE>;
	%simple_miR_counts=();
	while(<MIR_TABLE>)
		{
		@d=split("\t",$_);
		$d[0]=~s/3p.*$/3p/; #remove information on internal mismatches
		$d[0]=~s/5p.*$/5p/; #remove information on internal mismatches
		$simple_miR_counts{$d[0]}+=$d[2];
		}
	close MIR_TABLE;
	print MIR_TABLE_SIMPLIFIED"miR_name\tabsolute_counts";
	foreach$miR_ID(sort{$simple_miR_counts{$b}<=>$simple_miR_counts{$a}}keys%simple_miR_counts)
		{
		print MIR_TABLE_SIMPLIFIED"\n$miR_ID\t$simple_miR_counts{$miR_ID}";
		}
	close MIR_TABLE_SIMPLIFIED;

	# make simplified tRNA count table
	open(TRNA_TABLE,"$dump_folder/unitas.tRF-table.txt")||print$!;
	open(TRNA_TABLE_SIMPLIFIED,">$dump_folder/unitas.tRF-table.simplified.txt")||print$!;
	%simple_tRNA_counts=();
	%simple_tRNA_counts_mature_only=();
	while(<TRNA_TABLE>)
		{
		@d=split("\t",$_);
		if($d[0]=~s/tRNA-[^-]+-...//)
			{
			$simple_tRNA_counts{$&}+=$d[1]+$d[3]+$d[5]+$d[7]+$d[9]+$d[11]+$d[13]+$d[15]; # including leader and trailer of pre-mature tRNA
			$simple_tRNA_counts_mature_only{$&}+=$d[1]+$d[3]+$d[5]+$d[7]+$d[9]+$d[15]; # excluding leader and trailer of pre-mature tRNA
			}
		}
	close TRNATABLE;
	print TRNA_TABLE_SIMPLIFIED"tRNA_name\tabsolute_counts\tabsolute_counts_including_trailer_and_leader_of_pre-tRNA";
	foreach$tRNA_ID(sort{$simple_tRNA_counts{$b}<=>$simple_tRNA_counts{$a}}keys%simple_tRNA_counts)
		{
		print TRNA_TABLE_SIMPLIFIED"\n$tRNA_ID\t$simple_tRNA_counts_mature_only{$tRNA_ID}\t$simple_tRNA_counts{$tRNA_ID}";
		}
	close TRNA_TABLE_SIMPLIFIED;

	# MAKE  HTML FIGURES
	if(!$no_html)
		{
		print"\nPrepare HTML image files...";
		$storage="html";
		make_path("$dump_folder/$storage")||die print$!;

		# ANNOTATION SUMMARY PIE CHART
		$PieHole=0.4;
		$VisibilityThreshold=0.01;
		$pie_width=800;
		$pie_height=500;
		%data=();
		$data{"map to piRNA clusters"}=0;
		open(SUMMARY,"$dump_folder/unitas.annotation_summary.txt")||die print$!;
		while(<SUMMARY>)
			{
			$_=~s/\s*$//;
			if($_=~s/   mapped to piRNA producing loci\t//)
				{
				$data{"map to piRNA clusters"}=$_;
				next;
				}
			next if($_=~/^ /); # skip other sub-classes
			@d=split("\t",$_);
			$d[0]=~s/_/ /g;
			$data{$d[0]}=$d[1];
			$total+=$d[1];
			}
		$total=int($total+0.5);
		$data{"no annotation"}=$data{"no annotation"}-$data{"map to piRNA clusters"};
		close SUMMARY;
		open(CHART,">$dump_folder/$storage/summary_pie.html")||die print$!;
		print CHART
		'<!doctype html>
		<html>
		  <head>
		    <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
		    <script type="text/javascript">
		      google.charts.load("current", {packages:["corechart"]});
		      google.charts.setOnLoadCallback(drawChart);
		      function drawChart() {
			var data = google.visualization.arrayToDataTable([
			  [\'RNA class\', \'fractionated reads\'],
		';
		foreach$class(sort{$data{$b}<=>$data{$a}}keys%data)
			{
			print CHART"          ['$class', $data{$class}],\n";
			}
		print CHART
		"        ]);

			var options = {
			  title: 'sRNA annotation overview. Total reads: $total',
			  pieHole: $PieHole,
			sliceVisibilityThreshold: $VisibilityThreshold
			};

			var chart = new google.visualization.PieChart(document.getElementById('sRNA annotation summary'));
			chart.draw(data, options);
		      }
		    </script>
		  </head>
		  <body>
		    <div id='sRNA annotation summary' style='width: ${pie_width}px; height: ${pie_height}px;'></div>
		  </body>
		</html>";
		close CHART;

		# ANNOTATION SUMMARY TABLE
		@colors=
			(
			"#4881CB",
			"#BECADB",
			"#D3E0F2",
			);

		open(TABLE,">$dump_folder/$storage/summary_table.html")||die print$!;
		print TABLE
		'<!doctype html>
		<html>
		  <head>
		    <meta charset="utf-8">
		    <title>sRNA annotation summary</title>
		    <style>
		       table, th, td {border: 1px solid; border-collapse: collapse;}
		    </style>
		  </head>
		  <body>
		  <table>
		    <tbody>
		';
		open(SUMMARY,"$dump_folder/unitas.annotation_summary.txt")||die print$!;
		while(<SUMMARY>)
			{
			$_=~s/\s*$//;
			@d=split("\t",$_);
			$color_ID=0;
			if($d[0]=~/^      /)
				{
				$color_ID=2;
				}
			elsif($d[0]=~/^   /)
				{
				$color_ID=1;
				}
			$d[0]=~s/_/ /g;
			$d[0]=~s/   /&emsp;/g;
			$d[1]=int($d[1]+0.5);

			print TABLE"
		      <tr>
			<td bgcolor=\"$colors[$color_ID]\">$d[0]</td>
			<td style = \"text-align: right;\">$d[1]</td>
		      </tr>";
			}
		close SUMMARY;
		print TABLE"
		    </tbody>
		  </table>
		  </body>
		</html>";
		close TABLE;

		# PING PONG SIGNATURE
		$pp_width=500;
		$pp_height=300;
		opendir(DIR,$dump_folder)||die print$!;
		while($file=readdir(DIR))
			{
			if($file=~/\.pp$/)
				{
				next if($file=~/non-miR/);
				$total_read_pairs=0;
				%pp_signature=();
				$class=$file;
				$class=~s/^unitas\.//;
				$class=~s/\.pp$//;
				$class=~s/global/total/;
				open(PP,"$dump_folder/$file")||die print$!;
				$head=<PP>;
				$head=<PP>;
				while(<PP>)
					{
					last if($_=~/^\s*$/);
					$_=~s/\s*$//;
					@d=split("\t",$_);
					$max_overlap=$d[0];
					if($d[1]!~/\d/)
						{
						$d[1]=0;
						}
					$pp_signature{$d[0]}=$d[1];
					$total_read_pairs+=$d[1];
					}
				close PP;
				next if($total_read_pairs<1);
				open(PP_HTML,">$dump_folder/$storage/pingpong_$class.html")||die print$!;
				print PP_HTML
		"<html>
		  <head>
		    <script type=\"text/javascript\" src=\"https://www.gstatic.com/charts/loader.js\"></script>
		    <script type=\"text/javascript\">
		      google.charts.load('current', {'packages':['bar']});
		      google.charts.setOnLoadCallback(drawChart);

		      function drawChart() {
			var data = google.visualization.arrayToDataTable([
			  ['5 prime overlap', 'read pairs', { role: \"style\" }],";

				foreach$i(1..$max_overlap)
					{
					if($i==10)
						{
						print PP_HTML"\n          ['$i', $pp_signature{$i}, \"color: #6E95DA\"],";
						}
					else
						{
						print PP_HTML"\n          ['$i', $pp_signature{$i}, \"color: #4A4A4A\"],";
						}
					}

				print PP_HTML"
			]);

			var options = {
			  chart: {
			    title: 'Check for ping-pong signature (bias for 10 bp 5 prime overlaps)',
			    subtitle: 'number of read pairs for any possible 5 prime overlap',
			  }
			};

			var chart = new google.charts.Bar(document.getElementById('columnchart_material'));

			chart.draw(data, google.charts.Bar.convertOptions(options));
		      }
		    </script>
		  </head>
		  <body>
		    <div id=\"columnchart_material\" style=\"width: ${pp_width}px; height: ${pp_height}px;\"></div>
		  </body>
		</html>";

				close PP_HTML;
				}
			}
		closedir DIR;

		# PING PONG MATRIX
		$pp_width=500;
		$pp_height=300;
		opendir(DIR,$dump_folder)||die print$!;
		while($file=readdir(DIR))
			{
			if($file=~/\.ppmatrix$/)
				{
				next if($file=~/non-miR/);
				$class=$file;
				$class=~s/^unitas\.//;
				$class=~s/\.ppmatrix$//;
				$class=~s/global/total/;
				next if(!-e"$dump_folder/$storage/pingpong_$class.html"); # do not make a matrix for classes with <1 read pairs (fractionated)
				open(MATRIX,"$dump_folder/$file")||die print$!;
				$head=<MATRIX>;$head=<MATRIX>;$head=<MATRIX>;$head=<MATRIX>;
				$max_value=0;
				while(<MATRIX>)
					{
					$_=~s/\s*$//;
					@values=split("\t",$_);
					foreach$i(1..@values)
						{
						if($values[$i]>$max_value)
							{
							$max_value=$values[$i];
							}
						}
					}
				close MATRIX;
				next if($max_value==0); # no 10 bp overlap
				open(MATRIX,"$dump_folder/$file")||die print$!;
				$head=<MATRIX>;$head=<MATRIX>;$head=<MATRIX>;$head=<MATRIX>;
				$head=~s/\s*$//;
				@head=split("\t",$head);
				open(MATRIX_HTML,">$dump_folder/$storage/pingpong-matrix_$class.html")||die print$!;
				print MATRIX_HTML
		"<html>
		<b>ping-pong matrix</b><br>
		<small>sequence length combinations within ping-pong pairs</small><br>
		<br>
		<table style=\"border-spacing: 1px; font-size: 10px;\";>
		   <tr>
		      <td> </td>";
				foreach$i(1..@head-1) # head of the matrix
					{
					print MATRIX_HTML"\n      <td> <small>$head[$i]</small></td>";
					}
				print MATRIX_HTML"\n   </tr>";

				while(<MATRIX>)
					{
					@values=split("\t",$_);
					print MATRIX_HTML"\n   <tr>\n      <td><small>$values[0]</small></td>";
					foreach$i(1..@values-1)
						{
						$rel_value=((sqrt($values[$i]))/(sqrt($max_value)));
						$red=0;
						$green=0;
						$blue=0;
						if($rel_value<=0.5)
							{
							$blue=(int(($rel_value*255)+0.5))*2;
							}
						else
							{
							$blue=255-((int((($rel_value-0.5)*255)+0.5))*2);
							$red=(int((($rel_value-0.5)*255)+0.5))*2;
							}
						$x="";
						if($values[$i]==$max_value)
							{
							$x="x"
							}
						print MATRIX_HTML"\n      <td style=\"background-color: rgb($red,$green,$blue); width:12px; height:12px; text-align:center\"><small>$x</small></td>";
						}
					print MATRIX_HTML"\n   </tr>";
					}
				print MATRIX_HTML"</table>";

				# color coding
				print MATRIX_HTML"<br><table style=\"border-spacing: 0px;\"><tr><td><small>min</small></td>";
				$steps=100;
				foreach$i(1..$steps)
					{
					$red=0;
					$green=0;
					$blue=0;
					$rel_value=$i/$steps;
					if($rel_value<=0.5)
						{
						$blue=(int(($rel_value*255)+0.5))*2;
						}
					else
						{
						$blue=255-((int((($rel_value-0.5)*255)+0.5))*2);
						$red=(int((($rel_value-0.5)*255)+0.5))*2;
						}
					print MATRIX_HTML"<td style=\"background-color: rgb($red,$green,$blue); height: 8px;\"> </td>";
					}
				print MATRIX_HTML"<td><small>max</small></td></tr></table>";
				close MATRIX_HTML;
				}
			}
		closedir DIR;

		# POSITIONAL NUCLEOTIDE COMPOSITION / LENGTH DISTRIBUTION
		opendir(DIR,$dump_folder)||die print$!;
		$min_length=0;
		$max_length=0;
		%seqs=();
		%seqs_per_class=();
		%readscounts_per_class=();
		while($file=readdir(DIR))
			{
			if($file=~/\.fas$/)
				{
				next if($file=~/non-miR/);
				open(FAS,"$dump_folder/$file")||die print$!;
				$class=$file;
				$class=~s/^unitas\.//;
				$class=~s/\.fas$//;
				while(<FAS>)
					{
					$_=~s/\s*$//;
					if($_=~s/^>//)
						{
						$reads=$_;
						$reads=~s/:.+$//; # file with piR candidates contains additional information
						}
					else
						{
						if(length$_>$max_length)
							{
							$max_length=length$_;
							}
						if($min_length==0||length$_<$min_length)
							{
							$min_length=length$_;
							}
						$seqs{$_}=$reads; # save reads for each sequence
						$seqs_per_class{$class}{$_}=$reads;
						$readscounts_per_class{$class}+=$reads;
						}
					}
				close FAS;
				}
			}
		closedir DIR;

		# global
		%length=();
		%pos_nucl_comp=();
		$width=500;
		$height=500;
		$stacked_height=300;
		$curve_type="none"; # use 'function' for smoth curves
		@nucl_order=('A','T','G','C','N');
		foreach$seq(keys%seqs)
			{
			$length{length$seq}+=$seqs{$seq};
			foreach$p(1..length$seq)
				{
				$pos_nucl_comp{$p}{substr($seq,$p-1,1)}+=$seqs{$seq};
				}
			}

		$html_string="";
		$class="total";
		$class_for_print="all input reads";
		$reads_in_this_class_absolut=$total;
		generate_length_comp_htlm();
		sub generate_length_comp_htlm
			{
			open(NUCL_COMP,">$dump_folder/$storage/nucl-comp_$class.html")||die print$!;
			print NUCL_COMP
			"<html>
			  <head>
			    <script type=\"text/javascript\" src=\"https://www.gstatic.com/charts/loader.js\"></script>
			    <script type=\"text/javascript\">
			      google.charts.load(\'current\', {'packages\':[\'bar\']});
			      google.charts.setOnLoadCallback(drawChart);

			      function drawChart() {
				var data = google.visualization.arrayToDataTable([
				['Position', 'A', 'T', 'G', 'C','N',{ role: 'annotation' } ],";
			foreach$p(1..$max_length)
				{
				$total=0;
				foreach$n(@nucl_order)
					{
					$total+=$pos_nucl_comp{$p}{$n};
					}
				last if($total==0);
				foreach$n(@nucl_order)
					{
					$pos_nucl_comp{$p}{$n}=$pos_nucl_comp{$p}{$n}/$total;
					}
				print NUCL_COMP"\n        ['$p', $pos_nucl_comp{$p}{A}, $pos_nucl_comp{$p}{T}, $pos_nucl_comp{$p}{G}, $pos_nucl_comp{$p}{C}, $pos_nucl_comp{$p}{N}, ''],";
				}
			print NUCL_COMP
			"      ]);
			      var options = {
				  isStacked: 'true',
				  height: $stacked_height,
				  legend: {position: 'top', maxLines: 3},
				  title: 'Positional Nucleotide Frequencies',
				  vAxis: {
				    minValue: 0,
				    ticks: [0, .25, .5, .75, 1]
				  },
				  series: {
				    0:{color:'#109618'},
				    1:{color:'#DC3912'},
				    2:{color:'#3C3C3C'},
				    3:{color:'#3366CC'},
				    4:{color:'#FF9900'},
				  }
				};

				var chart = new google.charts.Bar(document.getElementById('pos_nucl_comp'));

				chart.draw(data, google.charts.Bar.convertOptions(options));
			      }
			    </script>
			  </head>
			  <body>
			    <div id=\"pos_nucl_comp\" style=\"width: ${width}px; height: ${height}px;\"></div>
			  </body>
			</html>
			";
			close NUCL_COMP;

			$width=500;
			$height=300;
			open(LENGTH_DISTR,">$dump_folder/$storage/length-distr_$class.html")||die print$!;
			print LENGTH_DISTR
			"
			  <html>
			  <head>
			    <script type=\"text/javascript\" src=\"https://www.gstatic.com/charts/loader.js\"></script>
			    <script type=\"text/javascript\">
			      google.charts.load('current', {'packages':['corechart']});
			      google.charts.setOnLoadCallback(drawChart);

			      function drawChart() {
				var data = google.visualization.arrayToDataTable([
				  ['nucleotides', 'reads',],";

			foreach$l($min_length..$max_length)
				{
				if(!$length{$l})
					{
					$length{$l}=0;
					}
				print LENGTH_DISTR"\n          ['$l', $length{$l}],";
				}

			print LENGTH_DISTR
			"
				]);

				var options = {
				  title: 'Length Distribution',
				  curveType: '$curve_type',
				  legend: { position: 'bottom' },
				  pointSize: 3,
				};

				var chart = new google.visualization.LineChart(document.getElementById('curve_chart'));

				chart.draw(data, options);
			      }
			    </script>
			  </head>
			  <body>
			    <div id=\"curve_chart\" style=\"width: ${width}px; height: ${height}px;\"></div>
			  </body>
			</html>
			";
			close LENGTH_DISTR;

			$html_string.="
		  <div style=\"width:1250px\"><br><big><b>$class_for_print.</b></big> Absolute reads matching this class: $reads_in_this_class_absolut (values with fractionated multimappers are given in the table at the top)</div>

		  <div style=\"float:left;\">
		  <iframe src=\"$storage/length-distr_$class.html\" frameborder=\"1\"  style=\"width:625px; height:350px\"></iframe>
		  </div>

		    <div style=\"\">
		  <iframe src=\"$storage/nucl-comp_$class.html\" frameborder=\"1\" scrolling=\"no\" style=\"width:625px; height:350px; overflow:hidden;\"></iframe>
		  </div>
		 ";

			if(-e"$dump_folder/$storage/pingpong_$class.html") # add ping pong signature if file exists
				{
				$html_string.="

		  <div style=\"float:left;\">
		  <iframe src=\"$storage/pingpong_$class.html\" frameborder=\"1\"  style=\"width:625px; height:380px\"></iframe>
		  </div>
		";
				}

			if(-e"$dump_folder/$storage/pingpong-matrix_$class.html") # add ping pong matrix if file exists
				{
				$html_string.="
		    <div style=\"\">
		  <iframe src=\"$storage/pingpong-matrix_$class.html\" frameborder=\"1\" scrolling=\"no\" style=\"width:625px; height:380px; overflow:hidden;\"></iframe>
		  </div>

		 ";
				}
			elsif(-e"$dump_folder/$storage/pingpong_$class.html") # no ping pong matrix but 5 prime overlaps
				{
				$html_string.="	<div style=\"width:700px; height:380px;\"><br>no read pairs with 10 bp 5 prime overlap</div>";
				}
			}

		# iterate through remaining ncRNA classes
		foreach$class(sort{$readscounts_per_class{$b}<=>$readscounts_per_class{$a}}keys%readscounts_per_class)
			{
			%length=();
			%pos_nucl_comp=();
			$reads_in_this_class_absolut=0;
			$reads_in_this_class_fractionated=0;
			$class_for_print=$class;
			$class_for_print=~s/_/ /g;
			foreach$seq(keys%{$seqs_per_class{$class}})
				{
				$length{length$seq}+=$seqs{$seq};
				$reads_in_this_class_absolut+=$seqs{$seq};
				foreach$p(1..length$seq)
					{
					$pos_nucl_comp{$p}{substr($seq,$p-1,1)}+=$seqs{$seq};
					}
				}
			$reads_in_this_class_fractionated=int($reads_in_this_class_fractionated+0.5);
			generate_length_comp_htlm();
			}

		# MERGE HTML OUTPUTS
		open(MERGE,">$dump_folder/results.html")||die print$!;
		print MERGE
		"<html>
		  <head>
		    <meta charset=\"utf-8\">
		    <title>UNITAS results</title>
		  </head>
		  <body>

		  <div style=\"text-align:center; background-color:#DCDCDC; padding:5px; max-width:1248px; margin-bottom:1px;\">
		  <a>
		  <h2>This is a summary of what <a style=\"color: #164F81\"><i>unitas</i></a> found in your dataset</h2>
		  </a>
		  </div>

		  <div style=\"float:left;\">
		  <iframe src=\"$storage/summary_table.html\" frameborder=\"1\"  style=\"width:400px; height:450px\"></iframe>
		  </div>

		  <div style=\"\">
		  <iframe src=\"$storage/summary_pie.html\" frameborder=\"1\" scrolling=\"no\" style=\"width:850px; height:450px\"></iframe>
		  </div>

		  <div style=\"text-align:center; background-color:#DCDCDC; padding:5px; max-width:1248px; margin-top:10px; margin-bottom:1px;\">
		  <a>
		  <h2>Here you see positional nucleotide frequencies and length distribution (global and separated by class)</h2>
		  If you used the otion -pp you will also see number of read pairs with specific 5 prime overlaps and a ping-pong matrix for read pairs with 10 bp 5 prime overlap.
		  </a>
		  </div>

		".$html_string.'

		  <div>
		  </body>
		</html>';
		close MERGE;
		print" done.\n";
		}

	# move .fas and .info files into final directories
	make_path("$dump_folder/fasta")||die print$!;
	make_path("$dump_folder/info")||die print$!;
	opendir(DUMP_FOLDER,$dump_folder)||die print$!;
	while($file=readdir(DUMP_FOLDER))
		{
		if($file=~/\.fas$/)
			{
			move("$dump_folder/$file","$dump_folder/fasta/$file")||print"\nFailed to move $dump_folder/$file to $dump_folder/fasta/$file.\n$!\n\n";
			}
		elsif($file=~/\.info$/)
			{
			move("$dump_folder/$file","$dump_folder/info/$file")||print"\nFailed to move $dump_folder/$file to $dump_folder/info/$file.\n$!\n\n";
			}
		}
	closedir DUMP_FOLDER;


	if(!$silent){print"\n Finished annotation for file $parsed_input.\n";}
	}

$computation_time=time-$t0;
$days=int($computation_time/86400);
$hours=int(($computation_time-$days*86400)/3600);
$minutes=int((($computation_time-$days*86400)-($hours*3600))/60);
$seconds=$computation_time-($days*86400)-($hours*3600)-($minutes*60);
print"\n\n Computation finished.\n Elapsed time: $days days, $hours hours, $minutes minutes, $seconds seconds.\n\n$update_available


                         ### How to cite unitas ###
 Gebert D, Hewel C, Rosenkranz D. unitas: the universal tool for annotation of
 small RNAs. BMC Genomics 2017 18:644.


                              ### Twitter ###
 Follow us on Twitter to get information on latest software updates:
 https://twitter.com/smallRNA_Mainz


";


#################################
#                       SUBROUTINES
#################################

sub clean_SeqMap_miRNA
	{
	# step 1: keep perferct hits or hits with non-template 3' ends
	open(SEQMAP,"$dump_folder/SeqMap_$file_id.temp")||die print"\nERROR! Unable to open temporary SeqMap output file $dump_folder/SeqMap_$file_id.temp.\n$!\n\n";
	open(SEQMAP_CLEAN1,">$dump_folder/SeqMap_$file_id.clean-1.temp")||die print"\nERROR! Unable to create clean SeqMap output file.\n$!\n\n";
	$head=<SEQMAP>;
	%best_hits=();
	while(<SEQMAP>)
		{
		$_=~s/\s*$//;
		@seqmap=split("\t",$_);
		if($seqmap[5]==0)
			{
			print SEQMAP_CLEAN1"$_\t0\n"; # 5th column will refer to tail length, 6th column to total mismatch counts!
			# save best alignment
			$best_hits{$seqmap[4]}=0;
			}

		# check number of internal modifications
		else
			{
			$current_internal_mismatch=0;
			foreach$p(1..(length$seqmap[2])-$max_miRtail_length)
				{
				if(substr($seqmap[2],$p-1,1)ne substr($seqmap[4],$p-1,1))
					{
					$current_internal_mismatch++;
					}
				last if($current_internal_mismatch>$intmod_miR);
				}
			}

		### relaxed tailing defenition = mm in one of the last n positions means: tail starts here ###
		if($current_internal_mismatch<=$intmod_miR)
			{
			$tail_present=0;
			foreach$p((length$seqmap[2])-$max_miRtail_length..(length$seqmap[2])-1)
				{
				if(substr($seqmap[2],$p,1) ne substr($seqmap[4],$p,1))
					{
					# check whether tail-nucleotides represent allowed 3' modification, otherwise count it as internal mismatch
					$tail=substr($seqmap[4],$p);
					@tail=split('',$tail);
					$all_ok=1;
					foreach$nt(@tail)
						{
						if(!$miRtails{$nt})
							{
							$all_ok=0;
							last;
							}
						}
					if($all_ok==0)
						{
						$current_internal_mismatch++;
						last if($current_internal_mismatch>$intmod_miR);
						next;
						}

					$tail_present=1;
					$seqmap[5]=length(substr($seqmap[2],$p)); #5th column will refer to tail length, 6th column to total mismatch counts!
					$total_mm_counts=$seqmap[5]+$current_internal_mismatch;
					print SEQMAP_CLEAN1"$seqmap[0]\t$seqmap[1]\t$seqmap[2]\t$seqmap[3]\t$seqmap[4]\t$seqmap[5]\t$total_mm_counts\n";

					# save best alignment
					if(!$best_hits{$seqmap[4]})
						{
						$best_hits{$seqmap[4]}=$total_mm_counts;
						}
					elsif($seqmap[5]<$best_hits{$seqmap[4]})
						{
						$best_hits{$seqmap[4]}=$total_mm_counts;
						}
					last;
					}
				}
			# no (valid) tail present, only internal modifications, yet not printet to clean seqmap file. 5th column will refer to tail length, 6th column to total mismatch counts!
			if($tail_present==0&&$current_internal_mismatch<=$intmod_miR)
				{
				$best_hits{$seqmap[4]}=$seqmap[5];
				print SEQMAP_CLEAN1"$seqmap[0]\t$seqmap[1]\t$seqmap[2]\t$seqmap[3]\t$seqmap[4]\t0\t$seqmap[5]\n";
				}
			}
		}
	close SEQMAP;
	close SEQMAP_CLEAN1;

	# step 2: keep best hits only, replace '-stem-loop' by '-5p/-3p', check internal modificationss
	open(SEQMAP_CLEAN1,"$dump_folder/SeqMap_$file_id.clean-1.temp")||die print"\nERROR! Unable to open temporary SeqMap output file $dump_folder/SeqMap_$file_id.clean-1.temp.\n$!\n\n";
	open(SEQMAP_CLEAN2,">$dump_folder/SeqMap_$file_id.clean-2.temp")||die print"\nERROR! Unable to create clean SeqMap output file $dump_folder/SeqMap_$file_id.clean-2.temp.\n$!\n\n";
	while(<SEQMAP_CLEAN1>)
		{
		$_=~s/\s*$//;
		@seqmap=split("\t",$_);
		if($best_hits{$seqmap[4]}==$seqmap[6])
			{
			# check -5p or -3p
			if($seqmap[1]+((length$seqmap[4])/2)<=$miR_hairpin_length{$seqmap[0]}/2)
				{
				$_=~s/_stem-loop/-5p/
				}
			else
				{
				$_=~s/_stem-loop/-3p/
				}

			# check for internal modifications
			if($seqmap[6]>$seqmap[5])
				{
				$modification_cigar_string="";
				foreach$p(1..(length$seqmap[4])-$max_miRtail_length)
					{
					if(substr($seqmap[2],$p-1,1)ne substr($seqmap[4],$p-1,1))
						{
						$modification_cigar_string.=$p.substr($seqmap[2],$p-1,1)."->".substr($seqmap[4],$p-1,1)."|";
						}
					}
				if(length$modification_cigar_string>0)
					{
					$modification_cigar_string=~s/\|$//;
					$_=~s/\t/($modification_cigar_string)\t/;
					}
				}
			print SEQMAP_CLEAN2"$_\n";
			}
		}
	close SEQMAP_CLEAN1;
	close SEQMAP_CLEAN2;
	undef%best_hits;
	}

sub clean_SeqMap_ncRNA
	{
	open(SEQMAP,"$dump_folder/SeqMap_ncRNA.temp")||die print"\nERROR! Unable to open temporary SeqMap output file $dump_folder/SeqMap_ncRNA.temp.\n$!\n\n";
	$head=<SEQMAP>;
	%best_hits=();
	while(<SEQMAP>)
		{
		@seqmap=split("\t",$_);
		if(!$best_hits{$seqmap[4]}||$seqmap[5]<$best_hits{$seqmap[4]})
			{
			$best_hits{$seqmap[4]}=$seqmap[5];
			}
		}
	close SEQMAP;
	open(SEQMAP,"$dump_folder/SeqMap_ncRNA.temp")||die print"\nERROR! Unable to open temporary SeqMap output file $dump_folder/SeqMap_ncRNA.temp.\n$!\n\n";
	open(SEQMAP_CLEAN,">$dump_folder/SeqMap_ncRNA.clean.temp")||die print"\nERROR! Unable to create clean SeqMap output file $dump_folder/SeqMap_ncRNA.clean.temp.\n$!\n\n";
	$head=<SEQMAP>;
	while(<SEQMAP>)
		{
		@seqmap=split("\t",$_);
		if($seqmap[5]==$best_hits{$seqmap[4]})
			{
			print SEQMAP_CLEAN$_;
			}
		}
	close SEQMAP;
	close SEQMAP_CLEAN;
	undef%best_hits;
	}

sub basic_analysis
	{
	open(IN,$ba_in)||print"\nERROR! Unable to analyze output file $ba_in.\n$!\n\n";
	open(OUT,">$ba_out")||print"\nERROR! Unable to create output file $ba_out.\n$!\n\n";
	%length=();
	%comp=();
	%nt=();
	while(<IN>)
		{
		$_=~s/\s*$//;
		if($_=~s/^>//)
			{
			$reads=$_;
			}
		else
			{
			$length{length$_}+=$reads;
			foreach$p(1..length$_)
				{
				$comp{$p}{substr($_,$p-1,1)}+=$reads;
				$nt{substr($_,$p-1,1)}=1;
				}
			}
		}
	close IN;

	@size=keys%length;
	@size=sort{$a<=>$b}@size;
	@nt=keys%nt;
	@nt=sort@nt;
	print OUT"LENGTH DISTRIBUTION\nlength\treads";
	foreach$l($size[0]..$size[-1])
		{
		print OUT"\n$l\t$length{$l}";
		}

	print OUT"\n\nPOSITIONAL NUCLEOTIDE COMPOSITION\npos.";
	foreach$nt(@nt)
		{
		print OUT"\t$nt";
		}
	foreach$p(1..$size[-1])
		{
		print OUT"\n$p";
		foreach$nt(@nt)
			{
			if(!$comp{$p}{$nt})
				{
				$comp{$p}{$nt}=0;
				}
			print OUT"\t$comp{$p}{$nt}";
			}
		}
	undef%length;
	undef%comp;
	undef%nt;
	close IN;
	close OUT;
	}

sub load_species_dictionary
	{
	$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/URLs/species');
	$response = $useragent->request($request);
	$eval_URL="";
	$eval_URL = $response->content;
	if(length$eval_URL>0)
		{
		eval$eval_URL;
		}
	else
		{
		if(!$silent){print"\nFailed to load current species list from smallRNAgroup-SERVER. Use internal list.";}

### SPECIES DICTONARY START ###
%species=
	(
	'_candida_arabinofermentans_nrrl_yb_2248_gca_001661425'=>'_candida_arabinofermentans_nrrl_yb_2248_gca_001661425',
	'_candida_auris_gca_001189475'=>'_candida_auris_gca_001189475',
	'_candida_auris_gca_002759435'=>'_candida_auris_gca_002759435',
	'_candida_auris_gca_002775015'=>'_candida_auris_gca_002775015',
	'_candida_auris_gca_003013715'=>'_candida_auris_gca_003013715',
	'_candida_auris_gca_003014415'=>'_candida_auris_gca_003014415',
	'_candida_duobushaemulonis_gca_002926085'=>'_candida_duobushaemulonis_gca_002926085',
	'_candida_glabrata_gca_000002545'=>'_candida_glabrata_gca_000002545',
	'_candida_glabrata_gca_002219185'=>'_candida_glabrata_gca_002219185',
	'_candida_glabrata_gca_002219195'=>'_candida_glabrata_gca_002219195',
	'_candida_haemulonis_gca_002926055'=>'_candida_haemulonis_gca_002926055',
	'_candida_pseudohaemulonis_gca_003013735'=>'_candida_pseudohaemulonis_gca_003013735',
	'_candida_tenuis_atcc_10573_gca_000223465'=>'_candida_tenuis_atcc_10573_gca_000223465',
	'absidia_glauca_gca_900079185'=>'Absidia_glauca_gca_900079185',
	'absidia_repens_gca_002105175'=>'Absidia_repens_gca_002105175',
	'acanthochromis_polyacanthus'=>'Spiny_chromis',
	'acaromyces_ingoldii_gca_003144295'=>'Acaromyces_ingoldii_gca_003144295',
	'acidomyces_richmondensis_bfw_gca_001592465'=>'Acidomyces_richmondensis_bfw_gca_001592465',
	'acremonium_chrysogenum_atcc_11550_gca_000769265'=>'Acremonium_chrysogenum_atcc_11550_gca_000769265',
	'acyrthosiphon_pisum'=>'Acyrthosiphon_pisum',
	'adineta_vaga'=>'Adineta_vaga',
	'aedes_aegypti'=>'Aedes_aegypti',
	'aegilops_tauschii'=>'Aegilops_tauschii',
	'agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555'=>'Agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555',
	'ailuropoda_melanoleuca'=>'Panda',
	'albugo_laibachii'=>'Albugo_laibachii',
	'allomyces_macrogynus_atcc_38327_gca_000151295'=>'Allomyces_macrogynus_atcc_38327_gca_000151295',
	'alternaria_alternata_gca_001642055'=>'Alternaria_alternata_gca_001642055',
	'amanita_muscaria_koide_bx008_gca_000827485'=>'Amanita_muscaria_koide_bx008_gca_000827485',
	'amborella_trichopoda'=>'Amborella_trichopoda',
	'amphiamblys_sp_wsbs2006_gca_001875675'=>'Amphiamblys_sp_wsbs2006_gca_001875675',
	'amphilophus_citrinellus'=>'Midas_cichlid',
	'amphimedon_queenslandica'=>'Amphimedon_queenslandica',
	'amphiprion_ocellaris'=>'Clown_anemonefish',
	'amphiprion_percula'=>'Orange_clownfish',
	'anabas_testudineus'=>'Climbing_perch',
	'anaeromyces_robustus_gca_002104895'=>'Anaeromyces_robustus_gca_002104895',
	'anas_platyrhynchos_platyrhynchos'=>'Duck',
	'angomonas_deanei'=>'Angomonas_deanei',
	'anncaliia_algerae_pra109_gca_000385855'=>'Anncaliia_algerae_pra109_gca_000385855',
	'anncaliia_algerae_pra339_gca_000385875'=>'Anncaliia_algerae_pra339_gca_000385875',
	'anolis_carolinensis'=>'Anole_lizard',
	'anopheles_darlingi'=>'Anopheles_darlingi',
	'anopheles_gambiae'=>'Anopheles_gambiae',
	'anoplophora_glabripennis'=>'Anoplophora_glabripennis',
	'anser_brachyrhynchus'=>'Pink-footed_goose',
	'anthracocystis_flocculosa_pf_1_gca_000417875'=>'Anthracocystis_flocculosa_pf_1_gca_000417875',
	'aotus_nancymaae'=>'Ma\'s_night_monkey',
	'aphanomyces_astaci'=>'Aphanomyces_astaci',
	'aphanomyces_invadans'=>'Aphanomyces_invadans',
	'apis_mellifera'=>'Apis_mellifera',
	'apteryx_haastii'=>'Great_spotted_kiwi',
	'apteryx_owenii'=>'Little_spotted_kiwi',
	'apteryx_rowi'=>'Okarito_brown_kiwi',
	'arabidopsis_halleri'=>'Arabidopsis_halleri',
	'arabidopsis_lyrata'=>'Arabidopsis_lyrata',
	'arabidopsis_thaliana'=>'Arabidopsis_thaliana',
	'armillaria_gallica_gca_002307695'=>'Armillaria_gallica_gca_002307695',
	'armillaria_ostoyae_gca_900157425'=>'Armillaria_ostoyae_gca_900157425',
	'armillaria_solidipes_gca_002307675'=>'Armillaria_solidipes_gca_002307675',
	'arthrobotrys_oligospora_atcc_24927_gca_000225545'=>'Arthrobotrys_oligospora_atcc_24927_gca_000225545',
	'ascochyta_rabiei_gca_001630375'=>'Ascochyta_rabiei_gca_001630375',
	'ascoidea_rubescens_dsm_1968_gca_001661345'=>'Ascoidea_rubescens_dsm_1968_gca_001661345',
	'ashbya_gossypii'=>'Ashbya_gossypii',
	'aspergillus_aculeatinus_cbs_121060_gca_003184765'=>'Aspergillus_aculeatinus_cbs_121060_gca_003184765',
	'aspergillus_aculeatus_atcc_16872_gca_001890905'=>'Aspergillus_aculeatus_atcc_16872_gca_001890905',
	'aspergillus_brasiliensis_cbs_101740_gca_001889945'=>'Aspergillus_brasiliensis_cbs_101740_gca_001889945',
	'aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695'=>'Aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695',
	'aspergillus_calidoustus_gca_001511075'=>'Aspergillus_calidoustus_gca_001511075',
	'aspergillus_campestris_ibt_28561_gca_002847485'=>'Aspergillus_campestris_ibt_28561_gca_002847485',
	'aspergillus_candidus_gca_002847045'=>'Aspergillus_candidus_gca_002847045',
	'aspergillus_carbonarius_item_5010_gca_001990825'=>'Aspergillus_carbonarius_item_5010_gca_001990825',
	'aspergillus_clavatus'=>'Aspergillus_clavatus',
	'aspergillus_cristatus_gca_001717485'=>'Aspergillus_cristatus_gca_001717485',
	'aspergillus_ellipticus_cbs_707_79_gca_003184645'=>'Aspergillus_ellipticus_cbs_707_79_gca_003184645',
	'aspergillus_eucalypticola_cbs_122712_gca_003184535'=>'Aspergillus_eucalypticola_cbs_122712_gca_003184535',
	'aspergillus_flavus_af70_gca_000952835'=>'Aspergillus_flavus_af70_gca_000952835',
	'aspergillus_fumigatus'=>'Aspergillus_fumigatus',
	'aspergillus_fumigatus_gca_002234985'=>'Aspergillus_fumigatus_gca_002234985',
	'aspergillus_fumigatusa1163'=>'Aspergillus_fumigatusa1163',
	'aspergillus_glaucus_cbs_516_65_gca_001890805'=>'Aspergillus_glaucus_cbs_516_65_gca_001890805',
	'aspergillus_heteromorphus_cbs_117_55_gca_003184545'=>'Aspergillus_heteromorphus_cbs_117_55_gca_003184545',
	'aspergillus_indologenus_cbs_114_80_gca_003184685'=>'Aspergillus_indologenus_cbs_114_80_gca_003184685',
	'aspergillus_japonicus_cbs_114_51_gca_003184785'=>'Aspergillus_japonicus_cbs_114_51_gca_003184785',
	'aspergillus_lacticoffeatus_cbs_101883_gca_003184595'=>'Aspergillus_lacticoffeatus_cbs_101883_gca_003184595',
	'aspergillus_lentulus_gca_001445615'=>'Aspergillus_lentulus_gca_001445615',
	'aspergillus_luchuensis_cbs_106_47_gca_001890685'=>'Aspergillus_luchuensis_cbs_106_47_gca_001890685',
	'aspergillus_luchuensis_gca_001602395'=>'Aspergillus_luchuensis_gca_001602395',
	'aspergillus_neoniger_cbs_115656_gca_003184625'=>'Aspergillus_neoniger_cbs_115656_gca_003184625',
	'aspergillus_nidulans'=>'Aspergillus_nidulans',
	'aspergillus_nidulans_fgsc_a4_gca_000149205'=>'Aspergillus_nidulans_fgsc_a4_gca_000149205',
	'aspergillus_niger'=>'Aspergillus_niger',
	'aspergillus_niger_atcc_1015_gca_000230395'=>'Aspergillus_niger_atcc_1015_gca_000230395',
	'aspergillus_niger_gca_001515345'=>'Aspergillus_niger_gca_001515345',
	'aspergillus_nomius_nrrl_13137_gca_001204775'=>'Aspergillus_nomius_nrrl_13137_gca_001204775',
	'aspergillus_novofumigatus_ibt_16806_gca_002847465'=>'Aspergillus_novofumigatus_ibt_16806_gca_002847465',
	'aspergillus_ochraceoroseus_ibt_24754_gca_002846915'=>'Aspergillus_ochraceoroseus_ibt_24754_gca_002846915',
	'aspergillus_oryzae'=>'Aspergillus_oryzae',
	'aspergillus_oryzae_3_042_gca_000269785'=>'Aspergillus_oryzae_3_042_gca_000269785',
	'aspergillus_parasiticus_su_1_gca_000956085'=>'Aspergillus_parasiticus_su_1_gca_000956085',
	'aspergillus_piperis_cbs_112811_gca_003184755'=>'Aspergillus_piperis_cbs_112811_gca_003184755',
	'aspergillus_rambellii_gca_000986645'=>'Aspergillus_rambellii_gca_000986645',
	'aspergillus_ruber_cbs_135680_gca_000600275'=>'Aspergillus_ruber_cbs_135680_gca_000600275',
	'aspergillus_saccharolyticus_jop_1030_1_gca_003184585'=>'Aspergillus_saccharolyticus_jop_1030_1_gca_003184585',
	'aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635'=>'Aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635',
	'aspergillus_sclerotioniger_cbs_115572_gca_003184525'=>'Aspergillus_sclerotioniger_cbs_115572_gca_003184525',
	'aspergillus_steynii_ibt_23096_gca_002849105'=>'Aspergillus_steynii_ibt_23096_gca_002849105',
	'aspergillus_sydowii_cbs_593_65_gca_001890705'=>'Aspergillus_sydowii_cbs_593_65_gca_001890705',
	'aspergillus_taichungensis_gca_002850765'=>'Aspergillus_taichungensis_gca_002850765',
	'aspergillus_terreus'=>'Aspergillus_terreus',
	'aspergillus_thermomutatus_gca_002237265'=>'Aspergillus_thermomutatus_gca_002237265',
	'aspergillus_tubingensis_cbs_134_48_gca_001890745'=>'Aspergillus_tubingensis_cbs_134_48_gca_001890745',
	'aspergillus_turcosus_gca_002234965'=>'Aspergillus_turcosus_gca_002234965',
	'aspergillus_turcosus_gca_002234975'=>'Aspergillus_turcosus_gca_002234975',
	'aspergillus_udagawae_gca_001078395'=>'Aspergillus_udagawae_gca_001078395',
	'aspergillus_ustus_gca_000812125'=>'Aspergillus_ustus_gca_000812125',
	'aspergillus_uvarum_cbs_121591_gca_003184745'=>'Aspergillus_uvarum_cbs_121591_gca_003184745',
	'aspergillus_vadensis_cbs_113365_gca_003184925'=>'Aspergillus_vadensis_cbs_113365_gca_003184925',
	'aspergillus_versicolor_cbs_583_65_gca_001890125'=>'Aspergillus_versicolor_cbs_583_65_gca_001890125',
	'aspergillus_violaceofuscus_cbs_115571_gca_003184705'=>'Aspergillus_violaceofuscus_cbs_115571_gca_003184705',
	'aspergillus_wentii_dto_134e9_gca_001890725'=>'Aspergillus_wentii_dto_134e9_gca_001890725',
	'astatotilapia_calliptera'=>'Eastern_happy',
	'astyanax_mexicanus'=>'Mexican_tetra',
	'atta_cephalotes'=>'Atta_cephalotes',
	'aureobasidium_melanogenum_cbs_110374_gca_000721775'=>'Aureobasidium_melanogenum_cbs_110374_gca_000721775',
	'aureobasidium_namibiae_cbs_147_97_gca_000721765'=>'Aureobasidium_namibiae_cbs_147_97_gca_000721765',
	'aureobasidium_pullulans_exf_150_gca_000721785'=>'Aureobasidium_pullulans_exf_150_gca_000721785',
	'aureobasidium_subglaciale_exf_2481_gca_000721755'=>'Aureobasidium_subglaciale_exf_2481_gca_000721755',
	'babesia_bovis'=>'Babesia_bovis',
	'babesia_microti_strain_ri'=>'Babesia_microti_strain_ri',
	'babjeviella_inositovora_nrrl_y_12698_gca_001661335'=>'Babjeviella_inositovora_nrrl_y_12698_gca_001661335',
	'basidiobolus_meristosporus_cbs_931_73_gca_002104905'=>'Basidiobolus_meristosporus_cbs_931_73_gca_002104905',
	'batrachochytrium_dendrobatidis_jam81_gca_000203795'=>'Batrachochytrium_dendrobatidis_jam81_gca_000203795',
	'batrachochytrium_dendrobatidis_jel423_gca_000149865'=>'Batrachochytrium_dendrobatidis_jel423_gca_000149865',
	'batrachochytrium_salamandrivorans_gca_002006685'=>'Batrachochytrium_salamandrivorans_gca_002006685',
	'baudoinia_panamericana_uamh_10762_gca_000338955'=>'Baudoinia_panamericana_uamh_10762_gca_000338955',
	'beauveria_bassiana'=>'Beauveria_bassiana',
	'beauveria_bassiana_arsef_2860_gca_000280675'=>'Beauveria_bassiana_arsef_2860_gca_000280675',
	'beauveria_bassiana_d1_5_gca_000770705'=>'Beauveria_bassiana_d1_5_gca_000770705',
	'beauveria_bassiana_gca_002871155'=>'Beauveria_bassiana_gca_002871155',
	'belgica_antarctica'=>'Belgica_antarctica',
	'beta_vulgaris'=>'Beta_vulgaris',
	'bifiguratus_adelaidae_gca_002261195'=>'Bifiguratus_adelaidae_gca_002261195',
	'bigelowiella_natans'=>'Bigelowiella_natans',
	'bigelowiella_natans_gca_000002455'=>'Bigelowiella_natans_gca_000002455',
	'bipolaris_maydis_atcc_48331_gca_000354255'=>'Bipolaris_maydis_atcc_48331_gca_000354255',
	'bipolaris_maydis_c5_gca_000338975'=>'Bipolaris_maydis_c5_gca_000338975',
	'bipolaris_sorokiniana_nd90pr_gca_000338995'=>'Bipolaris_sorokiniana_nd90pr_gca_000338995',
	'bipolaris_zeicola_26_r_13_gca_000523435'=>'Bipolaris_zeicola_26_r_13_gca_000523435',
	'bison_bison_bison'=>'American_bison',
	'blastomyces_dermatitidis_atcc_18188_gca_000151595'=>'Blastomyces_dermatitidis_atcc_18188_gca_000151595',
	'blastomyces_dermatitidis_atcc_26199_gca_000166155'=>'Blastomyces_dermatitidis_atcc_26199_gca_000166155',
	'blastomyces_dermatitidis_er_3_gca_000003525'=>'Blastomyces_dermatitidis_er_3_gca_000003525',
	'blastomyces_gilchristii_slh14081_gca_000003855'=>'Blastomyces_gilchristii_slh14081_gca_000003855',
	'blastomyces_percursus_gca_001883805'=>'Blastomyces_percursus_gca_001883805',
	'blumeria_graminis'=>'Blumeria_graminis',
	'blumeria_graminis_f_sp_tritici_96224_gca_000418435'=>'Blumeria_graminis_f_sp_tritici_96224_gca_000418435',
	'bodo_saltans'=>'Bodo_saltans',
	'bombus_impatiens'=>'Bombus_impatiens',
	'bombus_terrestris'=>'Bombus_terrestris',
	'bombyx_mori'=>'Bombyx_mori',
	'bos_mutus'=>'Wild_yak',
	'bos_taurus'=>'Cow',
	'botryobasidium_botryosum_fd_172_ss1_gca_000697705'=>'Botryobasidium_botryosum_fd_172_ss1_gca_000697705',
	'botrytis_cinerea'=>'Botrytis_cinerea',
	'botrytis_cinerea_bcdw1_gca_000349525'=>'Botrytis_cinerea_bcdw1_gca_000349525',
	'botrytis_cinerea_t4_gca_000227075'=>'Botrytis_cinerea_t4_gca_000227075',
	'brachypodium_distachyon'=>'Brachypodium_distachyon',
	'brassica_napus'=>'Brassica_napus',
	'brassica_oleracea'=>'Brassica_oleracea',
	'brassica_rapa'=>'Brassica_rapa',
	'brettanomyces_bruxellensis_awri1499_gca_000259595'=>'Brettanomyces_bruxellensis_awri1499_gca_000259595',
	'brugia_malayi'=>'Brugia_malayi',
	'byssochlamys_spectabilis_no_5_gca_000497085'=>'Byssochlamys_spectabilis_no_5_gca_000497085',
	'cadophora_sp_dse1049_gca_003073865'=>'Cadophora_sp_dse1049_gca_003073865',
	'caenorhabditis_brenneri'=>'Caenorhabditis_brenneri',
	'caenorhabditis_briggsae'=>'Caenorhabditis_briggsae',
	'caenorhabditis_elegans'=>'Caenorhabditis_elegans',
	'caenorhabditis_japonica'=>'Caenorhabditis_japonica',
	'caenorhabditis_remanei'=>'Caenorhabditis_remanei',
	'calidris_pugnax'=>'Ruff',
	'calidris_pygmaea'=>'Spoon-billed_sandpiper',
	'callithrix_jacchus'=>'Marmoset',
	'calocera_cornea_hhb12733_gca_001632435'=>'Calocera_cornea_hhb12733_gca_001632435',
	'calocera_viscosa_tufc12733_gca_001630345'=>'Calocera_viscosa_tufc12733_gca_001630345',
	'candida_albicans_12c_gca_000773845'=>'Candida_albicans_12c_gca_000773845',
	'candida_albicans_19f_gca_000775445'=>'Candida_albicans_19f_gca_000775445',
	'candida_albicans_ca6_gca_000784695'=>'Candida_albicans_ca6_gca_000784695',
	'candida_albicans_gc75_gca_000773735'=>'Candida_albicans_gc75_gca_000773735',
	'candida_albicans_l26_gca_000775455'=>'Candida_albicans_l26_gca_000775455',
	'candida_albicans_p34048_gca_000775465'=>'Candida_albicans_p34048_gca_000775465',
	'candida_albicans_p37005_gca_000773745'=>'Candida_albicans_p37005_gca_000773745',
	'candida_albicans_p37037_gca_000773825'=>'Candida_albicans_p37037_gca_000773825',
	'candida_albicans_p37039_gca_000784515'=>'Candida_albicans_p37039_gca_000784515',
	'candida_albicans_p57055_gca_000775505'=>'Candida_albicans_p57055_gca_000775505',
	'candida_albicans_p57072_gca_000773805'=>'Candida_albicans_p57072_gca_000773805',
	'candida_albicans_p60002_gca_000784525'=>'Candida_albicans_p60002_gca_000784525',
	'candida_albicans_p75010_gca_000784575'=>'Candida_albicans_p75010_gca_000784575',
	'candida_albicans_p75016_gca_000784595'=>'Candida_albicans_p75016_gca_000784595',
	'candida_albicans_p75063_gca_000775525'=>'Candida_albicans_p75063_gca_000775525',
	'candida_albicans_p76055_gca_000784505'=>'Candida_albicans_p76055_gca_000784505',
	'candida_albicans_p76067_gca_000784495'=>'Candida_albicans_p76067_gca_000784495',
	'candida_albicans_p78042_gca_000784615'=>'Candida_albicans_p78042_gca_000784615',
	'candida_albicans_p78048_gca_000773725'=>'Candida_albicans_p78048_gca_000773725',
	'candida_albicans_p87_gca_000774085'=>'Candida_albicans_p87_gca_000774085',
	'candida_albicans_p94015_gca_000773755'=>'Candida_albicans_p94015_gca_000773755',
	'candida_albicans_sc5314_gca_000784635'=>'Candida_albicans_sc5314_gca_000784635',
	'candida_albicans_sc5314_gca_000784655'=>'Candida_albicans_sc5314_gca_000784655',
	'candida_albicans_wo_1_gca_000149445'=>'Candida_albicans_wo_1_gca_000149445',
	'candida_dubliniensis_cd36_gca_000026945'=>'Candida_dubliniensis_cd36_gca_000026945',
	'candida_maltosa_xu316_gca_000344705'=>'Candida_maltosa_xu316_gca_000344705',
	'candida_orthopsilosis_co_90_125_gca_000315875'=>'Candida_orthopsilosis_co_90_125_gca_000315875',
	'candida_parapsilosis_gca_000182765'=>'Candida_parapsilosis_gca_000182765',
	'candida_tropicalis_mya_3404_gca_000006335'=>'Candida_tropicalis_mya_3404_gca_000006335',
	'canis_familiaris'=>'Dog',
	'canis_lupus_dingo'=>'Dingo',
	'capitella_teleta'=>'Capitella_teleta',
	'capra_hircus'=>'Goat',
	'capronia_coronata_cbs_617_96_gca_000585585'=>'Capronia_coronata_cbs_617_96_gca_000585585',
	'capronia_epimyces_cbs_606_96_gca_000585565'=>'Capronia_epimyces_cbs_606_96_gca_000585565',
	'capsaspora_owczarzaki_atcc_30864'=>'Capsaspora_owczarzaki_atcc_30864',
	'carlito_syrichta'=>'Tarsier',
	'castor_canadensis'=>'American_beaver',
	'catenaria_anguillulae_pl171_gca_002102555'=>'Catenaria_anguillulae_pl171_gca_002102555',
	'cavia_aperea'=>'Brazilian_guinea_pig',
	'cavia_porcellus'=>'Guinea_Pig',
	'cebus_capucinus'=>'Capuchin',
	'ceraceosorus_bombacis_gca_900000165'=>'Ceraceosorus_bombacis_gca_900000165',
	'ceraceosorus_guamensis_gca_003144195'=>'Ceraceosorus_guamensis_gca_003144195',
	'ceratocystis_platani_gca_000978885'=>'Ceratocystis_platani_gca_000978885',
	'cercocebus_atys'=>'Sooty_mangabey',
	'chaetomium_globosum_cbs_148_51_gca_000143365'=>'Chaetomium_globosum_cbs_148_51_gca_000143365',
	'chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225'=>'Chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225',
	'chelonoidis_abingdonii'=>'Abingdon_island_giant_tortoise',
	'chinchilla_lanigera'=>'Long-tailed_chinchilla',
	'chlamydomonas_reinhardtii'=>'Chlamydomonas_reinhardtii',
	'chlorocebus_sabaeus'=>'Vervet-AGM',
	'choanephora_cucurbitarum_gca_001683725'=>'Choanephora_cucurbitarum_gca_001683725',
	'choloepus_hoffmanni'=>'Sloth',
	'chondrus_crispus'=>'Chondrus_crispus',
	'chroomonas_mesostigmatica_ccmp1168'=>'Chroomonas_mesostigmatica_ccmp1168',
	'chrysemys_picta_bellii'=>'Painted_turtle',
	'ciona_intestinalis'=>'C.intestinalis',
	'ciona_savignyi'=>'C.savignyi',
	'cladophialophora_bantiana_cbs_173_52_gca_000835475'=>'Cladophialophora_bantiana_cbs_173_52_gca_000835475',
	'cladophialophora_carrionii_cbs_160_54_gca_000365165'=>'Cladophialophora_carrionii_cbs_160_54_gca_000365165',
	'cladophialophora_carrionii_gca_001700775'=>'Cladophialophora_carrionii_gca_001700775',
	'cladophialophora_immunda_gca_000835495'=>'Cladophialophora_immunda_gca_000835495',
	'cladophialophora_psammophila_cbs_110553_gca_000585535'=>'Cladophialophora_psammophila_cbs_110553_gca_000585535',
	'cladophialophora_yegresii_cbs_114405_gca_000585515'=>'Cladophialophora_yegresii_cbs_114405_gca_000585515',
	'claviceps_purpurea_20_1_gca_000347355'=>'Claviceps_purpurea_20_1_gca_000347355',
	'clavispora_lusitaniae_atcc_42720_gca_000003835'=>'Clavispora_lusitaniae_atcc_42720_gca_000003835',
	'clavispora_lusitaniae_gca_001673695'=>'Clavispora_lusitaniae_gca_001673695',
	'clohesyomyces_aquaticus_gca_002105025'=>'Clohesyomyces_aquaticus_gca_002105025',
	'coccidioides_immitis_h538_4_gca_000149815'=>'Coccidioides_immitis_h538_4_gca_000149815',
	'coccidioides_immitis_rmscc_2394_gca_000149895'=>'Coccidioides_immitis_rmscc_2394_gca_000149895',
	'coccidioides_immitis_rmscc_3703_gca_000150085'=>'Coccidioides_immitis_rmscc_3703_gca_000150085',
	'coccidioides_immitis_rs_gca_000149335'=>'Coccidioides_immitis_rs_gca_000149335',
	'coccidioides_posadasii_c735_delta_sowgp_gca_000151335'=>'Coccidioides_posadasii_c735_delta_sowgp_gca_000151335',
	'coccidioides_posadasii_rmscc_3488_gca_000150055'=>'Coccidioides_posadasii_rmscc_3488_gca_000150055',
	'coccidioides_posadasii_str_silveira_gca_000170175'=>'Coccidioides_posadasii_str_silveira_gca_000170175',
	'colletotrichum_fioriniae_pj7_gca_000582985'=>'Colletotrichum_fioriniae_pj7_gca_000582985',
	'colletotrichum_gloeosporioides'=>'Colletotrichum_gloeosporioides',
	'colletotrichum_gloeosporioides_cg_14_gca_000446055'=>'Colletotrichum_gloeosporioides_cg_14_gca_000446055',
	'colletotrichum_graminicola'=>'Colletotrichum_graminicola',
	'colletotrichum_higginsianum'=>'Colletotrichum_higginsianum',
	'colletotrichum_incanum_gca_001625285'=>'Colletotrichum_incanum_gca_001625285',
	'colletotrichum_nymphaeae_sa_01_gca_001563115'=>'Colletotrichum_nymphaeae_sa_01_gca_001563115',
	'colletotrichum_orbiculare'=>'Colletotrichum_orbiculare',
	'colletotrichum_salicis_gca_001563125'=>'Colletotrichum_salicis_gca_001563125',
	'colletotrichum_sublineola_gca_000696135'=>'Colletotrichum_sublineola_gca_000696135',
	'colletotrichum_tofieldiae_gca_001625265'=>'Colletotrichum_tofieldiae_gca_001625265',
	'colobus_angolensis_palliatus'=>'Angola_colobus',
	'conidiobolus_coronatus_nrrl_28638_gca_001566745'=>'Conidiobolus_coronatus_nrrl_28638_gca_001566745',
	'coniella_lustricola_gca_003019895'=>'Coniella_lustricola_gca_003019895',
	'coniochaeta_ligniaria_nrrl_30616_gca_001879275'=>'Coniochaeta_ligniaria_nrrl_30616_gca_001879275',
	'coniophora_puteana_rwd_64_598_ss2_gca_000271625'=>'Coniophora_puteana_rwd_64_598_ss2_gca_000271625',
	'coniosporium_apollinis_cbs_100218_gca_000281105'=>'Coniosporium_apollinis_cbs_100218_gca_000281105',
	'coprinopsis_cinerea_okayama7_130_gca_000182895'=>'Coprinopsis_cinerea_okayama7_130_gca_000182895',
	'corchorus_capsularis'=>'Corchorus_capsularis',
	'cordyceps_brongniartii_rcef_3172_gca_001636735'=>'Cordyceps_brongniartii_rcef_3172_gca_001636735',
	'cordyceps_confragosa_rcef_1005_gca_001636795'=>'Cordyceps_confragosa_rcef_1005_gca_001636795',
	'cordyceps_fumosorosea_arsef_2679_gca_001636725'=>'Cordyceps_fumosorosea_arsef_2679_gca_001636725',
	'cordyceps_militaris_cm01_gca_000225605'=>'Cordyceps_militaris_cm01_gca_000225605',
	'corynespora_cassiicola_philippines_gca_003016335'=>'Corynespora_cassiicola_philippines_gca_003016335',
	'coturnix_japonica'=>'Japanese_quail',
	'crassostrea_gigas'=>'Crassostrea_gigas',
	'cricetulus_griseus_chok1gshd'=>'Chinese_hamster_CHOK1GS',
	'cricetulus_griseus_crigri'=>'Chinese_hamster_CriGri',
	'cricetulus_griseus_picr'=>'Chinese_hamster_PICR',
	'crocodylus_porosus'=>'Australian_saltwater_crocodile',
	'cryptococcus_amylolentus_cbs_6039_gca_001720205'=>'Cryptococcus_amylolentus_cbs_6039_gca_001720205',
	'cryptococcus_amylolentus_cbs_6273_gca_001720235'=>'Cryptococcus_amylolentus_cbs_6273_gca_001720235',
	'cryptococcus_depauperatus_cbs_7841_gca_001720195'=>'Cryptococcus_depauperatus_cbs_7841_gca_001720195',
	'cryptococcus_depauperatus_cbs_7855_gca_001720245'=>'Cryptococcus_depauperatus_cbs_7855_gca_001720245',
	'cryptococcus_gattii_ca1280_gca_000836335'=>'Cryptococcus_gattii_ca1280_gca_000836335',
	'cryptococcus_gattii_ca1873_gca_000855695'=>'Cryptococcus_gattii_ca1873_gca_000855695',
	'cryptococcus_gattii_e566_gca_000875815'=>'Cryptococcus_gattii_e566_gca_000875815',
	'cryptococcus_gattii_ejb2_gca_000835745'=>'Cryptococcus_gattii_ejb2_gca_000835745',
	'cryptococcus_gattii_nt_10_gca_000935105'=>'Cryptococcus_gattii_nt_10_gca_000935105',
	'cryptococcus_gattii_ru294_gca_000836355'=>'Cryptococcus_gattii_ru294_gca_000836355',
	'cryptococcus_gattii_vgii_2001_935_1_gca_000835815'=>'Cryptococcus_gattii_vgii_2001_935_1_gca_000835815',
	'cryptococcus_gattii_vgii_99_473_gca_000836455'=>'Cryptococcus_gattii_vgii_99_473_gca_000836455',
	'cryptococcus_gattii_vgii_ca1014_gca_000875795'=>'Cryptococcus_gattii_vgii_ca1014_gca_000875795',
	'cryptococcus_gattii_vgii_cbs_10090_gca_000835765'=>'Cryptococcus_gattii_vgii_cbs_10090_gca_000835765',
	'cryptococcus_gattii_vgii_la55_gca_000836315'=>'Cryptococcus_gattii_vgii_la55_gca_000836315',
	'cryptococcus_gattii_vgii_mmrl2647_gca_000875855'=>'Cryptococcus_gattii_vgii_mmrl2647_gca_000875855',
	'cryptococcus_gattii_vgii_r265_gca_000149475'=>'Cryptococcus_gattii_vgii_r265_gca_000149475',
	'cryptococcus_gattii_vgii_ram5_gca_000836375'=>'Cryptococcus_gattii_vgii_ram5_gca_000836375',
	'cryptococcus_gattii_vgiv_ind107_gca_000835755'=>'Cryptococcus_gattii_vgiv_ind107_gca_000835755',
	'cryptococcus_gattii_wm276_gca_000185945'=>'Cryptococcus_gattii_wm276_gca_000185945',
	'cryptococcus_neoformans'=>'Cryptococcus_neoformans',
	'cryptococcus_neoformans_var_grubii_h99_gca_000149245'=>'Cryptococcus_neoformans_var_grubii_h99_gca_000149245',
	'cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385'=>'Cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385',
	'cryptomonas_paramecium'=>'Cryptomonas_paramecium',
	'cryptosporidium_muris_rn66'=>'Cryptosporidium_muris_rn66',
	'cryptosporidium_parvum'=>'Cryptosporidium_parvum',
	'cryptosporidium_parvum_iowa_ii'=>'Cryptosporidium_parvum_iowa_ii',
	'cucumis_sativus'=>'Cucumis_sativus',
	'culex_quinquefasciatus'=>'Culex_quinquefasciatus',
	'cutaneotrichosporon_oleaginosum_gca_001027345'=>'Cutaneotrichosporon_oleaginosum_gca_001027345',
	'cyanistes_caeruleus'=>'Blue_tit',
	'cyberlindnera_fabianii_gca_003205855'=>'Cyberlindnera_fabianii_gca_003205855',
	'cyberlindnera_jadinii_gca_001245095'=>'Cyberlindnera_jadinii_gca_001245095',
	'cyberlindnera_jadinii_nrrl_y_1542_gca_001661405'=>'Cyberlindnera_jadinii_nrrl_y_1542_gca_001661405',
	'cyclospora_cayetanensis'=>'Cyclospora_cayetanensis',
	'cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385'=>'Cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385',
	'cynoglossus_semilaevis'=>'Tongue_sole',
	'cyphellophora_europaea_cbs_101466_gca_000365145'=>'Cyphellophora_europaea_cbs_101466_gca_000365145',
	'cyprinodon_variegatus'=>'Sheepshead_minnow',
	'dacryopinax_primogenitus_gca_000292625'=>'Dacryopinax_primogenitus_gca_000292625',
	'dactylellina_haptotyla_cbs_200_50_gca_000441935'=>'Dactylellina_haptotyla_cbs_200_50_gca_000441935',
	'daedalea_quercina_l_15889_gca_001632345'=>'Daedalea_quercina_l_15889_gca_001632345',
	'daldinia_sp_ec12_gca_002120325'=>'Daldinia_sp_ec12_gca_002120325',
	'danaus_plexippus'=>'Danaus_plexippus',
	'danio_rerio'=>'Zebrafish',
	'daphnia_magna'=>'Daphnia_magna',
	'daphnia_pulex'=>'Daphnia_pulex',
	'dasypus_novemcinctus'=>'Armadillo',
	'daucus_carota'=>'Daucus_carota',
	'debaryomyces_fabryi_gca_001447935'=>'Debaryomyces_fabryi_gca_001447935',
	'debaryomyces_hansenii_cbs767_gca_000006445'=>'Debaryomyces_hansenii_cbs767_gca_000006445',
	'dendroctonus_ponderosae'=>'Dendroctonus_ponderosae',
	'diaporthe_ampelina_gca_001006365'=>'Diaporthe_ampelina_gca_001006365',
	'dichomitus_squalens_lyad_421_ss1_gca_000275845'=>'Dichomitus_squalens_lyad_421_ss1_gca_000275845',
	'dictyostelium_discoideum'=>'Dictyostelium_discoideum',
	'dictyostelium_fasciculatum'=>'Dictyostelium_fasciculatum',
	'dictyostelium_lacteum'=>'Dictyostelium_lacteum',
	'dioscorea_rotundata'=>'Dioscorea_rotundata',
	'diplodia_seriata_gca_001006355'=>'Diplodia_seriata_gca_001006355',
	'dipodomys_ordii'=>'Kangaroo_rat',
	'dothistroma_septosporum'=>'Dothistroma_septosporum',
	'drechmeria_coniospora_gca_001625195'=>'Drechmeria_coniospora_gca_001625195',
	'drechslerella_stenobrocha_248_gca_000525045'=>'Drechslerella_stenobrocha_248_gca_000525045',
	'dromaius_novaehollandiae'=>'Emu',
	'drosophila_ananassae'=>'Drosophila_ananassae',
	'drosophila_erecta'=>'Drosophila_erecta',
	'drosophila_grimshawi'=>'Drosophila_grimshawi',
	'drosophila_melanogaster'=>'Drosophila_melanogaster',
	'drosophila_mojavensis'=>'Drosophila_mojavensis',
	'drosophila_persimilis'=>'Drosophila_persimilis',
	'drosophila_pseudoobscura'=>'Drosophila_pseudoobscura',
	'drosophila_sechellia'=>'Drosophila_sechellia',
	'drosophila_simulans'=>'Drosophila_simulans',
	'drosophila_virilis'=>'Drosophila_virilis',
	'drosophila_willistoni'=>'Drosophila_willistoni',
	'drosophila_yakuba'=>'Drosophila_yakuba',
	'echinops_telfairi'=>'Lesser_hedgehog_tenrec',
	'edhazardia_aedis_usnm_41457_gca_000230595'=>'Edhazardia_aedis_usnm_41457_gca_000230595',
	'eimeria_brunetti'=>'Eimeria_brunetti',
	'eimeria_maxima'=>'Eimeria_maxima',
	'eimeria_mitis'=>'Eimeria_mitis',
	'eimeria_praecox'=>'Eimeria_praecox',
	'eimeria_tenella'=>'Eimeria_tenella',
	'eimeria_tenella_gca_000499545'=>'Eimeria_tenella_gca_000499545',
	'emergomyces_pasteuriana_ep9510_gca_001883825'=>'Emergomyces_pasteuriana_ep9510_gca_001883825',
	'emiliania_huxleyi'=>'Emiliania_huxleyi',
	'emmonsia_crescens_uamh_3008_gca_001008285'=>'Emmonsia_crescens_uamh_3008_gca_001008285',
	'emmonsia_parva_uamh_139_gca_001014755'=>'Emmonsia_parva_uamh_139_gca_001014755',
	'emmonsia_sp_cac_2015a_gca_001660665'=>'Emmonsia_sp_cac_2015a_gca_001660665',
	'encephalitozoon_cuniculi_ecuniii_l_gca_001078035'=>'Encephalitozoon_cuniculi_ecuniii_l_gca_001078035',
	'encephalitozoon_hellem_atcc_50504_gca_000277815'=>'Encephalitozoon_hellem_atcc_50504_gca_000277815',
	'encephalitozoon_intestinalis_atcc_50506_gca_000146465'=>'Encephalitozoon_intestinalis_atcc_50506_gca_000146465',
	'encephalitozoon_romaleae_sj_2008_gca_000280035'=>'Encephalitozoon_romaleae_sj_2008_gca_000280035',
	'endocarpon_pusillum_z07020_gca_000464535'=>'Endocarpon_pusillum_z07020_gca_000464535',
	'entamoeba_histolytica'=>'Entamoeba_histolytica',
	'entamoeba_histolytica_gca_001662325'=>'Entamoeba_histolytica_gca_001662325',
	'entamoeba_histolytica_hm_3_imss'=>'Entamoeba_histolytica_hm_3_imss',
	'entamoeba_histolytica_ku27'=>'Entamoeba_histolytica_ku27',
	'entamoeba_nuttalli_p19'=>'Entamoeba_nuttalli_p19',
	'enterocytozoon_bieneusi_h348_gca_000209485'=>'Enterocytozoon_bieneusi_h348_gca_000209485',
	'epicoccum_nigrum_gca_002116315'=>'Epicoccum_nigrum_gca_002116315',
	'eptatretus_burgeri'=>'Hagfish',
	'equus_asinus_asinus'=>'Donkey',
	'equus_caballus'=>'Horse',
	'eremothecium_cymbalariae_dbvpg_7215_gca_000235365'=>'Eremothecium_cymbalariae_dbvpg_7215_gca_000235365',
	'eremothecium_gossypii_fdag1_gca_000968835'=>'Eremothecium_gossypii_fdag1_gca_000968835',
	'erinaceus_europaeus'=>'Hedgehog',
	'erysiphe_necator_gca_000798715'=>'Erysiphe_necator_gca_000798715',
	'escovopsis_weberi_gca_001278495'=>'Escovopsis_weberi_gca_001278495',
	'esox_lucius'=>'Northern_pike',
	'eutypa_lata_ucrel1_gca_000349385'=>'Eutypa_lata_ucrel1_gca_000349385',
	'exidia_glandulosa_hhb12029_gca_001632375'=>'Exidia_glandulosa_hhb12029_gca_001632375',
	'exophiala_aquamarina_cbs_119918_gca_000709125'=>'Exophiala_aquamarina_cbs_119918_gca_000709125',
	'exophiala_dermatitidis_nih_ut8656_gca_000230625'=>'Exophiala_dermatitidis_nih_ut8656_gca_000230625',
	'exophiala_mesophila_gca_000836275'=>'Exophiala_mesophila_gca_000836275',
	'exophiala_oligosperma_gca_000835515'=>'Exophiala_oligosperma_gca_000835515',
	'exophiala_sideris_gca_000835395'=>'Exophiala_sideris_gca_000835395',
	'exophiala_spinifera_gca_000836115'=>'Exophiala_spinifera_gca_000836115',
	'exophiala_xenobiotica_gca_000835505'=>'Exophiala_xenobiotica_gca_000835505',
	'exserohilum_turcica_et28a_gca_000359705'=>'Exserohilum_turcica_et28a_gca_000359705',
	'felis_catus'=>'Cat',
	'fibroporia_radiculosa_gca_000313525'=>'Fibroporia_radiculosa_gca_000313525',
	'fibularhizoctonia_sp_cbs_109695_gca_001630335'=>'Fibularhizoctonia_sp_cbs_109695_gca_001630335',
	'ficedula_albicollis'=>'Flycatcher',
	'fistulina_hepatica_atcc_64428_gca_000934395'=>'Fistulina_hepatica_atcc_64428_gca_000934395',
	'fomitiporia_mediterranea_mf3_22_gca_000271605'=>'Fomitiporia_mediterranea_mf3_22_gca_000271605',
	'fomitopsis_pinicola_fp_58527_ss1_gca_000344655'=>'Fomitopsis_pinicola_fp_58527_ss1_gca_000344655',
	'fonsecaea_erecta_gca_001651985'=>'Fonsecaea_erecta_gca_001651985',
	'fonsecaea_monophora_gca_001642475'=>'Fonsecaea_monophora_gca_001642475',
	'fonsecaea_multimorphosa_cbs_102226_gca_000836435'=>'Fonsecaea_multimorphosa_cbs_102226_gca_000836435',
	'fonsecaea_multimorphosa_gca_001646985'=>'Fonsecaea_multimorphosa_gca_001646985',
	'fonsecaea_nubica_gca_001646965'=>'Fonsecaea_nubica_gca_001646965',
	'fonsecaea_pedrosoi_cbs_271_37_gca_000835455'=>'Fonsecaea_pedrosoi_cbs_271_37_gca_000835455',
	'fonticula_alba_gca_000388065'=>'Fonticula_alba_gca_000388065',
	'fukomys_damarensis'=>'Damara_mole_rat',
	'fundulus_heteroclitus'=>'Mummichog',
	'fusarium_culmorum'=>'Fusarium_culmorum',
	'fusarium_fujikuroi'=>'Fusarium_fujikuroi',
	'fusarium_fujikuroi_gca_001023035'=>'Fusarium_fujikuroi_gca_001023035',
	'fusarium_fujikuroi_gca_001023045'=>'Fusarium_fujikuroi_gca_001023045',
	'fusarium_fujikuroi_gca_001023065'=>'Fusarium_fujikuroi_gca_001023065',
	'fusarium_fujikuroi_gca_900096505'=>'Fusarium_fujikuroi_gca_900096505',
	'fusarium_fujikuroi_gca_900096685'=>'Fusarium_fujikuroi_gca_900096685',
	'fusarium_fujikuroi_gca_900096705'=>'Fusarium_fujikuroi_gca_900096705',
	'fusarium_graminearum'=>'Fusarium_graminearum',
	'fusarium_graminearum_gca_000240135'=>'Fusarium_graminearum_gca_000240135',
	'fusarium_langsethiae_gca_001292635'=>'Fusarium_langsethiae_gca_001292635',
	'fusarium_mangiferae_gca_900044065'=>'Fusarium_mangiferae_gca_900044065',
	'fusarium_oxysporum'=>'Fusarium_oxysporum',
	'fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215'=>'Fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215',
	'fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345'=>'Fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345',
	'fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365'=>'Fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365',
	'fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195'=>'Fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195',
	'fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955'=>'Fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955',
	'fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975'=>'Fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975',
	'fusarium_oxysporum_f_sp_melonis_26406_gca_000260495'=>'Fusarium_oxysporum_f_sp_melonis_26406_gca_000260495',
	'fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075'=>'Fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075',
	'fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155'=>'Fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155',
	'fusarium_oxysporum_f_sp_raphani_54005_gca_000260235'=>'Fusarium_oxysporum_f_sp_raphani_54005_gca_000260235',
	'fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175'=>'Fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175',
	'fusarium_oxysporum_fo47_gca_000271705'=>'Fusarium_oxysporum_fo47_gca_000271705',
	'fusarium_oxysporum_fosc_3_a_gca_000271745'=>'Fusarium_oxysporum_fosc_3_a_gca_000271745',
	'fusarium_oxysporum_gca_900096695'=>'Fusarium_oxysporum_gca_900096695',
	'fusarium_poae_gca_001675295'=>'Fusarium_poae_gca_001675295',
	'fusarium_proliferatum_et1_gca_900067095'=>'Fusarium_proliferatum_et1_gca_900067095',
	'fusarium_proliferatum_gca_900029915'=>'Fusarium_proliferatum_gca_900029915',
	'fusarium_pseudograminearum'=>'Fusarium_pseudograminearum',
	'fusarium_solani'=>'Fusarium_solani',
	'fusarium_venenatum_gca_900007375'=>'Fusarium_venenatum_gca_900007375',
	'fusarium_verticillioides'=>'Fusarium_verticillioides',
	'gadus_morhua'=>'Cod',
	'gaeumannomyces_graminis'=>'Gaeumannomyces_graminis',
	'galdieria_sulphuraria'=>'Galdieria_sulphuraria',
	'galerina_marginata_cbs_339_88_gca_000697645'=>'Galerina_marginata_cbs_339_88_gca_000697645',
	'gallus_gallus'=>'Chicken',
	'gambusia_affinis'=>'Western_mosquitofish',
	'gasterosteus_aculeatus'=>'Stickleback',
	'gelatoporia_subvermispora_b_gca_000320605'=>'Gelatoporia_subvermispora_b_gca_000320605',
	'giardia_intestinalis'=>'Giardia_intestinalis',
	'giardia_intestinalis_assemblage_b'=>'Giardia_intestinalis_assemblage_b',
	'giardia_intestinalis_atcc_50581'=>'Giardia_intestinalis_atcc_50581',
	'giardia_intestinalis_gca_000498715'=>'Giardia_intestinalis_gca_000498715',
	'giardia_lamblia'=>'Giardia_lamblia',
	'giardia_lamblia_p15'=>'Giardia_lamblia_p15',
	'glarea_lozoyensis_atcc_20868_gca_000409485'=>'Glarea_lozoyensis_atcc_20868_gca_000409485',
	'gloeophyllum_trabeum_atcc_11539_gca_000344685'=>'Gloeophyllum_trabeum_atcc_11539_gca_000344685',
	'glycine_max'=>'Glycine_max',
	'gonapodya_prolifera_jel478_gca_001574975'=>'Gonapodya_prolifera_jel478_gca_001574975',
	'gopherus_agassizii'=>'Agassiz\'s_desert_tortoise',
	'gorilla_gorilla'=>'Gorilla',
	'gossypium_raimondii'=>'Gossypium_raimondii',
	'gregarina_niphandrodes'=>'Gregarina_niphandrodes',
	'grifola_frondosa_gca_001683735'=>'Grifola_frondosa_gca_001683735',
	'grosmannia_clavigera_kw1407_gca_000143105'=>'Grosmannia_clavigera_kw1407_gca_000143105',
	'guillardia_theta'=>'Guillardia_theta',
	'guillardia_theta_gca_000002975'=>'Guillardia_theta_gca_000002975',
	'gymnopus_luxurians_fd_317_m1_gca_000827265'=>'Gymnopus_luxurians_fd_317_m1_gca_000827265',
	'hammondia_hammondi'=>'Hammondia_hammondi',
	'hanseniaspora_guilliermondii_gca_900119595'=>'Hanseniaspora_guilliermondii_gca_900119595',
	'hanseniaspora_opuntiae_gca_001749795'=>'Hanseniaspora_opuntiae_gca_001749795',
	'hanseniaspora_osmophila_gca_001747045'=>'Hanseniaspora_osmophila_gca_001747045',
	'hanseniaspora_uvarum_dsm_2768_gca_000968475'=>'Hanseniaspora_uvarum_dsm_2768_gca_000968475',
	'hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025'=>'Hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025',
	'haplochromis_burtoni'=>'Burton\'s_mouthbrooder',
	'hebeloma_cylindrosporum_h7_gca_000827355'=>'Hebeloma_cylindrosporum_h7_gca_000827355',
	'helianthus_annuus'=>'Helianthus_annuus',
	'heliconius_melpomene'=>'Heliconius_melpomene',
	'helobdella_robusta'=>'Helobdella_robusta',
	'hesseltinella_vesiculosa_gca_002104935'=>'Hesseltinella_vesiculosa_gca_002104935',
	'heterobasidion_irregulare_tc_32_1_gca_000320585'=>'Heterobasidion_irregulare_tc_32_1_gca_000320585',
	'heterocephalus_glaber_female'=>'Naked_mole-rat_female',
	'heterocephalus_glaber_male'=>'Naked_mole-rat_male',
	'hippocampus_comes'=>'Tiger_tail_seahorse',
	'hirsutella_minnesotensis_3608_gca_000956045'=>'Hirsutella_minnesotensis_3608_gca_000956045',
	'histoplasma_capsulatum_g186ar_gca_000150115'=>'Histoplasma_capsulatum_g186ar_gca_000150115',
	'histoplasma_capsulatum_h143_gca_000151035'=>'Histoplasma_capsulatum_h143_gca_000151035',
	'histoplasma_capsulatum_h88_gca_000151005'=>'Histoplasma_capsulatum_h88_gca_000151005',
	'histoplasma_capsulatum_nam1_gca_000149585'=>'Histoplasma_capsulatum_nam1_gca_000149585',
	'homo_sapiens'=>'Human',
	'hordeum_vulgare'=>'Hordeum_vulgare',
	'hortaea_werneckii_exf_2000_gca_002127715'=>'Hortaea_werneckii_exf_2000_gca_002127715',
	'hyaloperonospora_arabidopsidis'=>'Hyaloperonospora_arabidopsidis',
	'hydnomerulius_pinastri_md_312_gca_000827185'=>'Hydnomerulius_pinastri_md_312_gca_000827185',
	'hypholoma_sublateritium_fd_334_ss_4_gca_000827495'=>'Hypholoma_sublateritium_fd_334_ss_4_gca_000827495',
	'hyphopichia_burtonii_nrrl_y_1933_gca_001661395'=>'Hyphopichia_burtonii_nrrl_y_1933_gca_001661395',
	'hypoxylon_sp_ci_4a_gca_002120315'=>'Hypoxylon_sp_ci_4a_gca_002120315',
	'hypoxylon_sp_co27_5_gca_002120305'=>'Hypoxylon_sp_co27_5_gca_002120305',
	'hypoxylon_sp_ec38_gca_002120335'=>'Hypoxylon_sp_ec38_gca_002120335',
	'hypsizygus_marmoreus_gca_001605315'=>'Hypsizygus_marmoreus_gca_001605315',
	'ichthyophthirius_multifiliis'=>'Ichthyophthirius_multifiliis',
	'ictalurus_punctatus'=>'Channel_catfish',
	'ictidomys_tridecemlineatus'=>'Squirrel',
	'ixodes_scapularis'=>'Ixodes_scapularis',
	'jaapia_argillacea_mucl_33604_gca_000697665'=>'Jaapia_argillacea_mucl_33604_gca_000697665',
	'jaculus_jaculus'=>'Lesser_Egyptian_jerboa',
	'jaminaea_rosea_gca_003144245'=>'Jaminaea_rosea_gca_003144245',
	'junco_hyemalis'=>'Dark-eyed_junco',
	'kalmanozyma_brasiliensis_ghg001_gca_000497045'=>'Kalmanozyma_brasiliensis_ghg001_gca_000497045',
	'kazachstania_africana_cbs_2517_gca_000304475'=>'Kazachstania_africana_cbs_2517_gca_000304475',
	'kazachstania_naganishii_cbs_8797_gca_000348985'=>'Kazachstania_naganishii_cbs_8797_gca_000348985',
	'kluyveromyces_dobzhanskii_cbs_2104_gca_000820885'=>'Kluyveromyces_dobzhanskii_cbs_2104_gca_000820885',
	'kluyveromyces_lactis_gca_000002515'=>'Kluyveromyces_lactis_gca_000002515',
	'kluyveromyces_marxianus_dmku3_1042_gca_001417885'=>'Kluyveromyces_marxianus_dmku3_1042_gca_001417885',
	'kluyveromyces_marxianus_gca_001417835'=>'Kluyveromyces_marxianus_gca_001417835',
	'kockovaella_imperatae_gca_002102565'=>'Kockovaella_imperatae_gca_002102565',
	'komagataella_pastoris'=>'Komagataella_pastoris',
	'komagataella_pastoris_gca_001708105'=>'Komagataella_pastoris_gca_001708105',
	'komagataella_phaffii_cbs_7435_gca_000223565'=>'Komagataella_phaffii_cbs_7435_gca_000223565',
	'komagataella_phaffii_gca_001708085'=>'Komagataella_phaffii_gca_001708085',
	'komagataella_phaffii_gs115_gca_001746955'=>'Komagataella_phaffii_gs115_gca_001746955',
	'kryptolebias_marmoratus'=>'Mangrove_rivulus',
	'kuraishia_capsulata_cbs_1993_gca_000576695'=>'Kuraishia_capsulata_cbs_1993_gca_000576695',
	'kwoniella_bestiolae_cbs_10118_gca_000512585'=>'Kwoniella_bestiolae_cbs_10118_gca_000512585',
	'kwoniella_dejecticola_cbs_10117_gca_000512565'=>'Kwoniella_dejecticola_cbs_10117_gca_000512565',
	'kwoniella_heveanensis_bcc8398_gca_000507405'=>'Kwoniella_heveanensis_bcc8398_gca_000507405',
	'kwoniella_heveanensis_cbs_569_gca_000507425'=>'Kwoniella_heveanensis_cbs_569_gca_000507425',
	'kwoniella_mangroviensis_cbs_10435_gca_000507885'=>'Kwoniella_mangroviensis_cbs_10435_gca_000507885',
	'kwoniella_mangroviensis_cbs_8507_gca_000507465'=>'Kwoniella_mangroviensis_cbs_8507_gca_000507465',
	'kwoniella_mangroviensis_cbs_8886_gca_000507485'=>'Kwoniella_mangroviensis_cbs_8886_gca_000507485',
	'kwoniella_pini_cbs_10737_gca_000512605'=>'Kwoniella_pini_cbs_10737_gca_000512605',
	'labrus_bergylta'=>'Ballan_wrasse',
	'laccaria_amethystina_laam_08_1_gca_000827195'=>'Laccaria_amethystina_laam_08_1_gca_000827195',
	'laccaria_bicolor_s238n_h82_gca_000143565'=>'Laccaria_bicolor_s238n_h82_gca_000143565',
	'lachancea_dasiensis_cbs_10888_gca_900074725'=>'Lachancea_dasiensis_cbs_10888_gca_900074725',
	'lachancea_fermentati_gca_900074765'=>'Lachancea_fermentati_gca_900074765',
	'lachancea_lanzarotensis_gca_000938715'=>'Lachancea_lanzarotensis_gca_000938715',
	'lachancea_meyersii_cbs_8951_gca_900074715'=>'Lachancea_meyersii_cbs_8951_gca_900074715',
	'lachancea_mirantina_gca_900074745'=>'Lachancea_mirantina_gca_900074745',
	'lachancea_nothofagi_cbs_11611_gca_900074755'=>'Lachancea_nothofagi_cbs_11611_gca_900074755',
	'lachancea_quebecensis_gca_002900925'=>'Lachancea_quebecensis_gca_002900925',
	'lachancea_sp_cbs_6924_gca_900074735'=>'Lachancea_sp_cbs_6924_gca_900074735',
	'lachancea_thermotolerans_cbs_6340_gca_000142805'=>'Lachancea_thermotolerans_cbs_6340_gca_000142805',
	'laetiporus_sulphureus_93_53_gca_001632365'=>'Laetiporus_sulphureus_93_53_gca_001632365',
	'latimeria_chalumnae'=>'Coelacanth',
	'leersia_perrieri'=>'Leersia_perrieri',
	'leishmania_braziliensis_mhom_br_75_m2904'=>'Leishmania_braziliensis_mhom_br_75_m2904',
	'leishmania_donovani_bpk282a1'=>'Leishmania_donovani_bpk282a1',
	'leishmania_infantum_jpcm5'=>'Leishmania_infantum_jpcm5',
	'leishmania_major'=>'Leishmania_major',
	'leishmania_mexicana_mhom_gt_2001_u1103'=>'Leishmania_mexicana_mhom_gt_2001_u1103',
	'leishmania_panamensis'=>'Leishmania_panamensis',
	'lepeophtheirus_salmonis'=>'Lepeophtheirus_salmonis',
	'lepidothrix_coronata'=>'Blue-crowned_manakin',
	'lepisosteus_oculatus'=>'Spotted_gar',
	'leptomonas_pyrrhocoris'=>'Leptomonas_pyrrhocoris',
	'leptomonas_seymouri'=>'Leptomonas_seymouri',
	'leptosphaeria_maculans'=>'Leptosphaeria_maculans',
	'leucoagaricus_sp_symc_cos_gca_001563735'=>'Leucoagaricus_sp_symc_cos_gca_001563735',
	'leucosporidium_creatinivorum_gca_002105055'=>'Leucosporidium_creatinivorum_gca_002105055',
	'lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665'=>'Lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665',
	'lingula_anatina'=>'Lingula_anatina',
	'lipomyces_starkeyi_nrrl_y_11557_gca_001661325'=>'Lipomyces_starkeyi_nrrl_y_11557_gca_001661325',
	'loa_loa'=>'Loa_loa',
	'lobosporangium_transversale_gca_002105155'=>'Lobosporangium_transversale_gca_002105155',
	'lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685'=>'Lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685',
	'lomentospora_prolificans_gca_002276285'=>'Lomentospora_prolificans_gca_002276285',
	'lonchura_striata_domestica'=>'Bengalese_finch',
	'lotharella_oceanica'=>'Lotharella_oceanica',
	'lottia_gigantea'=>'Lottia_gigantea',
	'loxodonta_africana'=>'Elephant',
	'lucilia_cuprina'=>'Lucilia_cuprina',
	'lupinus_angustifolius'=>'Lupinus_angustifolius',
	'macaca_fascicularis'=>'Crab-eating_macaque',
	'macaca_mulatta'=>'Macaque',
	'macaca_nemestrina'=>'Pig-tailed_macaque',
	'macrophomina_phaseolina_ms6_gca_000302655'=>'Macrophomina_phaseolina_ms6_gca_000302655',
	'madurella_mycetomatis_gca_001275765'=>'Madurella_mycetomatis_gca_001275765',
	'magnaporthe_oryzae'=>'Magnaporthe_oryzae',
	'magnaporthe_poae'=>'Magnaporthe_poae',
	'malassezia_globosa_cbs_7966_gca_000181695'=>'Malassezia_globosa_cbs_7966_gca_000181695',
	'malassezia_pachydermatis_gca_001278385'=>'Malassezia_pachydermatis_gca_001278385',
	'malassezia_sympodialis_atcc_42132_gca_000349305'=>'Malassezia_sympodialis_atcc_42132_gca_000349305',
	'malassezia_sympodialis_atcc_42132_gca_900149145'=>'Malassezia_sympodialis_atcc_42132_gca_900149145',
	'malassezia_vespertilionis_gca_002818225'=>'Malassezia_vespertilionis_gca_002818225',
	'manacus_vitellinus'=>'Golden-collared_manakin',
	'mandrillus_leucophaeus'=>'Drill',
	'manihot_esculenta'=>'Manihot_esculenta',
	'marmota_marmota_marmota'=>'Alpine_marmot',
	'marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775'=>'Marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775',
	'mastacembelus_armatus'=>'Zig-zag_eel',
	'mayetiola_destructor'=>'Mayetiola_destructor',
	'maylandia_zebra'=>'Zebra_mbuna',
	'medicago_truncatula'=>'Medicago_truncatula',
	'megaselia_scalaris'=>'Megaselia_scalaris',
	'meira_miltonrushii_gca_003144205'=>'Meira_miltonrushii_gca_003144205',
	'melampsora_laricipopulina'=>'Melampsora_laricipopulina',
	'meleagris_gallopavo'=>'Turkey',
	'meliniomyces_bicolor_e_gca_002865645'=>'Meliniomyces_bicolor_e_gca_002865645',
	'meliniomyces_variabilis_f_gca_002865655'=>'Meliniomyces_variabilis_f_gca_002865655',
	'melitaea_cinxia'=>'Melitaea_cinxia',
	'melopsittacus_undulatus'=>'Budgerigar',
	'meriones_unguiculatus'=>'Mongolian_gerbil',
	'mesocricetus_auratus'=>'Golden_Hamster',
	'metarhizium_acridum_cqma_102_gca_000187405'=>'Metarhizium_acridum_cqma_102_gca_000187405',
	'metarhizium_album_arsef_1941_gca_000804445'=>'Metarhizium_album_arsef_1941_gca_000804445',
	'metarhizium_anisopliae_arsef_23_gca_000187425'=>'Metarhizium_anisopliae_arsef_23_gca_000187425',
	'metarhizium_anisopliae_brip_53293_gca_000426965'=>'Metarhizium_anisopliae_brip_53293_gca_000426965',
	'metarhizium_anisopliae_gca_000739145'=>'Metarhizium_anisopliae_gca_000739145',
	'metarhizium_majus_arsef_297_gca_000814945'=>'Metarhizium_majus_arsef_297_gca_000814945',
	'metarhizium_robertsii_gca_000591435'=>'Metarhizium_robertsii_gca_000591435',
	'metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035'=>'Metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035',
	'meyerozyma_guilliermondii_atcc_6260_gca_000149425'=>'Meyerozyma_guilliermondii_atcc_6260_gca_000149425',
	'microbotryum_violaceum'=>'Microbotryum_violaceum',
	'microcebus_murinus'=>'Mouse_Lemur',
	'microdochium_bolleyi_gca_001566295'=>'Microdochium_bolleyi_gca_001566295',
	'microsporum_canis_cbs_113480_gca_000151145'=>'Microsporum_canis_cbs_113480_gca_000151145',
	'microtus_ochrogaster'=>'Prairie_vole',
	'millerozyma_farinosa_cbs_7064_gca_000315895'=>'Millerozyma_farinosa_cbs_7064_gca_000315895',
	'mitosporidium_daphniae_gca_000760515'=>'Mitosporidium_daphniae_gca_000760515',
	'mixia_osmundae_iam_14324_gca_000241205'=>'Mixia_osmundae_iam_14324_gca_000241205',
	'mnemiopsis_leidyi'=>'Mnemiopsis_leidyi',
	'moelleriella_libera_rcef_2490_gca_001636675'=>'Moelleriella_libera_rcef_2490_gca_001636675',
	'moesziomyces_antarcticus_gca_000747765'=>'Moesziomyces_antarcticus_gca_000747765',
	'moesziomyces_antarcticus_t_34_gca_000334475'=>'Moesziomyces_antarcticus_t_34_gca_000334475',
	'mola_mola'=>'Ocean_sunfish',
	'moniliophthora_perniciosa_fa553_gca_000183025'=>'Moniliophthora_perniciosa_fa553_gca_000183025',
	'moniliophthora_roreri_gca_001466705'=>'Moniliophthora_roreri_gca_001466705',
	'moniliophthora_roreri_mca_2997_gca_000488995'=>'Moniliophthora_roreri_mca_2997_gca_000488995',
	'monodelphis_domestica'=>'Opossum',
	'monopterus_albus'=>'Swamp_eel',
	'mortierella_elongata_ag_77_gca_001651415'=>'Mortierella_elongata_ag_77_gca_001651415',
	'mortierella_verticillata_nrrl_6337_gca_000739165'=>'Mortierella_verticillata_nrrl_6337_gca_000739165',
	'mucor_ambiguus_gca_000950595'=>'Mucor_ambiguus_gca_000950595',
	'mucor_circinelloides_f_circinelloides_1006phl_gca_000401635'=>'Mucor_circinelloides_f_circinelloides_1006phl_gca_000401635',
	'mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945'=>'Mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945',
	'mus_caroli'=>'Ryukyu_mouse',
	'mus_musculus'=>'Mouse',
	'mus_musculus_129s1svimj'=>'mus_musculus_129s1svimj',
	'mus_musculus_aj'=>'mus_musculus_aj',
	'mus_musculus_akrj'=>'mus_musculus_akrj',
	'mus_musculus_balbcj'=>'mus_musculus_balbcj',
	'mus_musculus_c3hhej'=>'mus_musculus_c3hhej',
	'mus_musculus_c57bl6nj'=>'mus_musculus_c57bl6nj',
	'mus_musculus_casteij'=>'mus_musculus_casteij',
	'mus_musculus_cbaj'=>'mus_musculus_cbaj',
	'mus_musculus_dba2j'=>'mus_musculus_dba2j',
	'mus_musculus_fvbnj'=>'mus_musculus_fvbnj',
	'mus_musculus_lpj'=>'mus_musculus_lpj',
	'mus_musculus_nodshiltj'=>'mus_musculus_nodshiltj',
	'mus_musculus_nzohlltj'=>'mus_musculus_nzohlltj',
	'mus_musculus_pwkphj'=>'mus_musculus_pwkphj',
	'mus_musculus_wsbeij'=>'mus_musculus_wsbeij',
	'mus_pahari'=>'Shrew_mouse',
	'mus_spicilegus'=>'Steppe_mouse',
	'mus_spretus'=>'Algerian_mouse',
	'musa_acuminata'=>'Musa_acuminata',
	'mustela_putorius_furo'=>'Ferret',
	'mycosphaerella_eumusae_gca_001578235'=>'Mycosphaerella_eumusae_gca_001578235',
	'myotis_lucifugus'=>'Microbat',
	'nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315'=>'Nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315',
	'naegleria_gruberi'=>'Naegleria_gruberi',
	'naematelia_encephala_gca_002105065'=>'Naematelia_encephala_gca_002105065',
	'nannizzia_gypsea_cbs_118893_gca_000150975'=>'Nannizzia_gypsea_cbs_118893_gca_000150975',
	'nannochloropsis_gaditana'=>'Nannochloropsis_gaditana',
	'nannospalax_galili'=>'Upper_Galilee_mountains_blind_mole_rat',
	'nasonia_vitripennis'=>'Nasonia_vitripennis',
	'naumovozyma_castellii_cbs_4309_gca_000237345'=>'Naumovozyma_castellii_cbs_4309_gca_000237345',
	'naumovozyma_dairenensis_cbs_421_gca_000227115'=>'Naumovozyma_dairenensis_cbs_421_gca_000227115',
	'nematocida_parisii_ertm1_gca_000250985'=>'Nematocida_parisii_ertm1_gca_000250985',
	'nematocida_parisii_ertm3_gca_000190615'=>'Nematocida_parisii_ertm3_gca_000190615',
	'nematocida_sp_1_ertm2_gca_000250695'=>'Nematocida_sp_1_ertm2_gca_000250695',
	'nematocida_sp_1_ertm6_gca_000738915'=>'Nematocida_sp_1_ertm6_gca_000738915',
	'nematocida_sp_ertm5_gca_001642415'=>'Nematocida_sp_ertm5_gca_001642415',
	'nematostella_vectensis'=>'Nematostella_vectensis',
	'neocallimastix_californiae_gca_002104975'=>'Neocallimastix_californiae_gca_002104975',
	'neofusicoccum_parvum_ucrnp2_gca_000385595'=>'Neofusicoccum_parvum_ucrnp2_gca_000385595',
	'neolamprologus_brichardi'=>'Lyretail_cichlid',
	'neolecta_irregularis_dah_3_gca_001929475'=>'Neolecta_irregularis_dah_3_gca_001929475',
	'neolentinus_lepideus_hhb14362_ss_1_gca_001632425'=>'Neolentinus_lepideus_hhb14362_ss_1_gca_001632425',
	'neonectria_ditissima_gca_001306435'=>'Neonectria_ditissima_gca_001306435',
	'neosartorya_fischeri'=>'Neosartorya_fischeri',
	'neovison_vison'=>'American_mink',
	'neurospora_crassa'=>'Neurospora_crassa',
	'neurospora_crassa_gca_000786625'=>'Neurospora_crassa_gca_000786625',
	'neurospora_tetrasperma_fgsc_2509_gca_000213195'=>'Neurospora_tetrasperma_fgsc_2509_gca_000213195',
	'nicotiana_attenuata'=>'Nicotiana_attenuata',
	'nomascus_leucogenys'=>'Gibbon',
	'nosema_apis_brl_01_gca_000447185'=>'Nosema_apis_brl_01_gca_000447185',
	'nosema_bombycis_cq1_gca_000383075'=>'Nosema_bombycis_cq1_gca_000383075',
	'nosema_ceranae_brl01_gca_000182985'=>'Nosema_ceranae_brl01_gca_000182985',
	'nosema_ceranae_gca_000988165'=>'Nosema_ceranae_gca_000988165',
	'notamacropus_eugenii'=>'Wallaby',
	'notechis_scutatus'=>'Mainland_tiger_snake',
	'nothoprocta_perdicaria'=>'Chilean_tinamou',
	'numida_meleagris'=>'Helmeted_guineafowl',
	'ochotona_princeps'=>'Pika',
	'octodon_degus'=>'Degu',
	'octopus_bimaculoides'=>'Octopus_bimaculoides',
	'ogataea_parapolymorpha_dl_1_gca_000187245'=>'Ogataea_parapolymorpha_dl_1_gca_000187245',
	'ogataea_polymorpha_gca_001664045'=>'Ogataea_polymorpha_gca_001664045',
	'oidiodendron_maius_zn_gca_000827325'=>'Oidiodendron_maius_zn_gca_000827325',
	'onchocerca_volvulus'=>'Onchocerca_volvulus',
	'ophiocordyceps_sinensis_co18_gca_000448365'=>'Ophiocordyceps_sinensis_co18_gca_000448365',
	'ophiostoma_piceae_uamh_11346_gca_000410735'=>'Ophiostoma_piceae_uamh_11346_gca_000410735',
	'ordospora_colligata_oc4_gca_000803265'=>'Ordospora_colligata_oc4_gca_000803265',
	'oreochromis_niloticus'=>'Tilapia',
	'ornithorhynchus_anatinus'=>'Platypus',
	'oryctolagus_cuniculus'=>'Rabbit',
	'oryza_barthii'=>'Oryza_barthii',
	'oryza_brachyantha'=>'Oryza_brachyantha',
	'oryza_glaberrima'=>'Oryza_glaberrima',
	'oryza_glumipatula'=>'Oryza_glumipatula',
	'oryza_indica'=>'Oryza_indica',
	'oryza_longistaminata'=>'Oryza_longistaminata',
	'oryza_meridionalis'=>'Oryza_meridionalis',
	'oryza_nivara'=>'Oryza_nivara',
	'oryza_punctata'=>'Oryza_punctata',
	'oryza_rufipogon'=>'Oryza_rufipogon',
	'oryza_sativa'=>'Oryza_sativa',
	'oryzias_latipes'=>'Japanese_medaka_HdrR',
	'oryzias_latipes_hni'=>'Japanese_medaka_HNI',
	'oryzias_latipes_hsok'=>'Japanese_medaka_HSOK',
	'oryzias_melastigma'=>'Indian_medaka',
	'ostreococcus_lucimarinus'=>'Ostreococcus_lucimarinus',
	'otolemur_garnettii'=>'Bushbaby',
	'ovis_aries'=>'Sheep',
	'pachysolen_tannophilus_nrrl_y_2460_gca_001661245'=>'Pachysolen_tannophilus_nrrl_y_2460_gca_001661245',
	'pan_paniscus'=>'Bonobo',
	'pan_troglodytes'=>'Chimpanzee',
	'panthera_pardus'=>'Leopard',
	'panthera_tigris_altaica'=>'Tiger',
	'papio_anubis'=>'Olive_baboon',
	'paracoccidioides_brasiliensis_gca_001713645'=>'Paracoccidioides_brasiliensis_gca_001713645',
	'paracoccidioides_brasiliensis_gca_001713695'=>'Paracoccidioides_brasiliensis_gca_001713695',
	'paracoccidioides_brasiliensis_pb03_gca_000150475'=>'Paracoccidioides_brasiliensis_pb03_gca_000150475',
	'paracoccidioides_brasiliensis_pb18_gca_000150735'=>'Paracoccidioides_brasiliensis_pb18_gca_000150735',
	'paracoccidioides_sp_lutzii_pb01_gca_000150705'=>'Paracoccidioides_sp_lutzii_pb01_gca_000150705',
	'paramecium_tetraurelia'=>'Paramecium_tetraurelia',
	'paramicrosporidium_saccamoebae_gca_002794465'=>'Paramicrosporidium_saccamoebae_gca_002794465',
	'paramormyrops_kingsleyae'=>'Paramormyrops_kingsleyae',
	'paraphaeosphaeria_sporulosa_gca_001642045'=>'Paraphaeosphaeria_sporulosa_gca_001642045',
	'parus_major'=>'Great_Tit',
	'paulinella_chromatophora'=>'Paulinella_chromatophora',
	'paxillus_involutus_atcc_200175_gca_000827475'=>'Paxillus_involutus_atcc_200175_gca_000827475',
	'paxillus_rubicundulus_ve08_2h10_gca_000827395'=>'Paxillus_rubicundulus_ve08_2h10_gca_000827395',
	'pediculus_humanus'=>'Pediculus_humanus',
	'pelodiscus_sinensis'=>'Chinese_softshell_turtle',
	'penicilliopsis_zonata_cbs_506_65_gca_001890105'=>'Penicilliopsis_zonata_cbs_506_65_gca_001890105',
	'penicillium_brasilianum_gca_001048715'=>'Penicillium_brasilianum_gca_001048715',
	'penicillium_brasilianum_gca_002016555'=>'Penicillium_brasilianum_gca_002016555',
	'penicillium_camemberti_fm_013_gca_000513335'=>'Penicillium_camemberti_fm_013_gca_000513335',
	'penicillium_chrysogenum_gca_000710275'=>'Penicillium_chrysogenum_gca_000710275',
	'penicillium_digitatum_pd1_gca_000315645'=>'Penicillium_digitatum_pd1_gca_000315645',
	'penicillium_digitatum_phi26_gca_000315665'=>'Penicillium_digitatum_phi26_gca_000315665',
	'penicillium_expansum_gca_000769745'=>'Penicillium_expansum_gca_000769745',
	'penicillium_freii_gca_001513925'=>'Penicillium_freii_gca_001513925',
	'penicillium_griseofulvum_gca_001561935'=>'Penicillium_griseofulvum_gca_001561935',
	'penicillium_italicum_gca_000769765'=>'Penicillium_italicum_gca_000769765',
	'penicillium_nordicum_gca_001278595'=>'Penicillium_nordicum_gca_001278595',
	'penicillium_oxalicum_114_2_gca_000346795'=>'Penicillium_oxalicum_114_2_gca_000346795',
	'penicillium_roqueforti_fm164_gca_000513255'=>'Penicillium_roqueforti_fm164_gca_000513255',
	'penicillium_rubens_wisconsin_54_1255_gca_000226395'=>'Penicillium_rubens_wisconsin_54_1255_gca_000226395',
	'penicillium_subrubescens_gca_001908125'=>'Penicillium_subrubescens_gca_001908125',
	'peniophora_sp_cont_gca_001632445'=>'Peniophora_sp_cont_gca_001632445',
	'periconia_macrospinosa_gca_003073855'=>'Periconia_macrospinosa_gca_003073855',
	'periophthalmus_magnuspinnatus'=>'Periophthalmus_magnuspinnatus',
	'perkinsus_marinus_atcc_50983'=>'Perkinsus_marinus_atcc_50983',
	'peromyscus_maniculatus_bairdii'=>'Northern_American_deer_mouse',
	'pestalotiopsis_fici_w106_1_gca_000516985'=>'Pestalotiopsis_fici_w106_1_gca_000516985',
	'petromyzon_marinus'=>'Lamprey',
	'pezoloma_ericae_gca_002865625'=>'Pezoloma_ericae_gca_002865625',
	'phaeoacremonium_minimum_ucrpa7_gca_000392275'=>'Phaeoacremonium_minimum_ucrpa7_gca_000392275',
	'phaeodactylum_tricornutum'=>'Phaeodactylum_tricornutum',
	'phaeomoniella_chlamydospora_gca_001006345'=>'Phaeomoniella_chlamydospora_gca_001006345',
	'phaeosphaeria_nodorum'=>'Phaeosphaeria_nodorum',
	'phanerochaete_carnosa_hhb_10118_sp_gca_000300595'=>'Phanerochaete_carnosa_hhb_10118_sp_gca_000300595',
	'phascolarctos_cinereus'=>'Koala',
	'phaseolus_vulgaris'=>'Phaseolus_vulgaris',
	'phialocephala_scopiformis_gca_001500285'=>'Phialocephala_scopiformis_gca_001500285',
	'phialocephala_subalpina_gca_900073065'=>'Phialocephala_subalpina_gca_900073065',
	'phialophora_americana_gca_000835435'=>'Phialophora_americana_gca_000835435',
	'phialophora_attae_gca_001299255'=>'Phialophora_attae_gca_001299255',
	'phlebia_centrifuga_gca_001913855'=>'Phlebia_centrifuga_gca_001913855',
	'phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265'=>'Phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265',
	'physcomitrella_patens'=>'Physcomitrella_patens',
	'phytophthora_infestans'=>'Phytophthora_infestans',
	'phytophthora_kernoviae'=>'Phytophthora_kernoviae',
	'phytophthora_lateralis'=>'Phytophthora_lateralis',
	'phytophthora_parasitica'=>'Phytophthora_parasitica',
	'phytophthora_parasitica_cj01a1'=>'Phytophthora_parasitica_cj01a1',
	'phytophthora_parasitica_gca_000509465'=>'Phytophthora_parasitica_gca_000509465',
	'phytophthora_parasitica_gca_000509485'=>'Phytophthora_parasitica_gca_000509485',
	'phytophthora_parasitica_gca_000509505'=>'Phytophthora_parasitica_gca_000509505',
	'phytophthora_parasitica_gca_000509525'=>'Phytophthora_parasitica_gca_000509525',
	'phytophthora_parasitica_inra_310'=>'Phytophthora_parasitica_inra_310',
	'phytophthora_parasitica_p10297'=>'Phytophthora_parasitica_p10297',
	'phytophthora_parasitica_p1976'=>'Phytophthora_parasitica_p1976',
	'phytophthora_ramorum'=>'Phytophthora_ramorum',
	'phytophthora_sojae'=>'Phytophthora_sojae',
	'pichia_kudriavzevii_gca_000764455'=>'Pichia_kudriavzevii_gca_000764455',
	'pichia_kudriavzevii_gca_002166775'=>'Pichia_kudriavzevii_gca_002166775',
	'pichia_kudriavzevii_gca_003054445'=>'Pichia_kudriavzevii_gca_003054445',
	'pichia_membranifaciens_gca_001950575'=>'Pichia_membranifaciens_gca_001950575',
	'pichia_membranifaciens_nrrl_y_2026_gca_001661235'=>'Pichia_membranifaciens_nrrl_y_2026_gca_001661235',
	'piliocolobus_tephrosceles'=>'Ugandan_red_Colobus',
	'piloderma_croceum_f_1598_gca_000827315'=>'Piloderma_croceum_f_1598_gca_000827315',
	'piromyces_finnis_gca_002104945'=>'Piromyces_finnis_gca_002104945',
	'pisolithus_microcarpus_441_gca_000827275'=>'Pisolithus_microcarpus_441_gca_000827275',
	'pisolithus_tinctorius_marx_270_gca_000827335'=>'Pisolithus_tinctorius_marx_270_gca_000827335',
	'plasmodium_berghei'=>'Plasmodium_berghei',
	'plasmodium_berghei_gca_000005395'=>'Plasmodium_berghei_gca_000005395',
	'plasmodium_berghei_gca_900044335'=>'Plasmodium_berghei_gca_900044335',
	'plasmodium_chabaudi'=>'Plasmodium_chabaudi',
	'plasmodium_chabaudi_chabaudi_gca_000003075'=>'Plasmodium_chabaudi_chabaudi_gca_000003075',
	'plasmodium_coatneyi'=>'Plasmodium_coatneyi',
	'plasmodium_cynomolgi_strain_b'=>'Plasmodium_cynomolgi_strain_b',
	'plasmodium_falciparum'=>'Plasmodium_falciparum',
	'plasmodium_falciparum_7g8'=>'Plasmodium_falciparum_7g8',
	'plasmodium_falciparum_camp_malaysia'=>'Plasmodium_falciparum_camp_malaysia',
	'plasmodium_falciparum_dd2'=>'Plasmodium_falciparum_dd2',
	'plasmodium_falciparum_fch_4'=>'Plasmodium_falciparum_fch_4',
	'plasmodium_falciparum_hb3'=>'Plasmodium_falciparum_hb3',
	'plasmodium_falciparum_igh_cr14'=>'Plasmodium_falciparum_igh_cr14',
	'plasmodium_falciparum_malips096_e11'=>'Plasmodium_falciparum_malips096_e11',
	'plasmodium_falciparum_nf135_5_c10'=>'Plasmodium_falciparum_nf135_5_c10',
	'plasmodium_falciparum_nf54'=>'Plasmodium_falciparum_nf54',
	'plasmodium_falciparum_palo_alto_uganda'=>'Plasmodium_falciparum_palo_alto_uganda',
	'plasmodium_falciparum_raj116'=>'Plasmodium_falciparum_raj116',
	'plasmodium_falciparum_santa_lucia'=>'Plasmodium_falciparum_santa_lucia',
	'plasmodium_falciparum_tanzania_2000708_'=>'Plasmodium_falciparum_tanzania_2000708_',
	'plasmodium_falciparum_ugt5_1'=>'Plasmodium_falciparum_ugt5_1',
	'plasmodium_falciparum_vietnam_oak_knoll_fvo_'=>'Plasmodium_falciparum_vietnam_oak_knoll_fvo_',
	'plasmodium_fragile'=>'Plasmodium_fragile',
	'plasmodium_gaboni'=>'Plasmodium_gaboni',
	'plasmodium_inui_san_antonio_1'=>'Plasmodium_inui_san_antonio_1',
	'plasmodium_knowlesi'=>'Plasmodium_knowlesi',
	'plasmodium_knowlesi_strain_h_gca_900004885'=>'Plasmodium_knowlesi_strain_h_gca_900004885',
	'plasmodium_malariae'=>'Plasmodium_malariae',
	'plasmodium_ovale_curtisi'=>'Plasmodium_ovale_curtisi',
	'plasmodium_reichenowi'=>'Plasmodium_reichenowi',
	'plasmodium_reichenowi_gca_001601855'=>'Plasmodium_reichenowi_gca_001601855',
	'plasmodium_vinckei_petteri'=>'Plasmodium_vinckei_petteri',
	'plasmodium_vinckei_vinckei'=>'Plasmodium_vinckei_vinckei',
	'plasmodium_vivax'=>'Plasmodium_vivax',
	'plasmodium_vivax_brazil_i'=>'Plasmodium_vivax_brazil_i',
	'plasmodium_vivax_india_vii'=>'Plasmodium_vivax_india_vii',
	'plasmodium_vivax_mauritania_i'=>'Plasmodium_vivax_mauritania_i',
	'plasmodium_vivax_north_korean'=>'Plasmodium_vivax_north_korean',
	'plasmodium_yoelii'=>'Plasmodium_yoelii',
	'plasmodium_yoelii_17x'=>'Plasmodium_yoelii_17x',
	'plasmodium_yoelii_gca_900002395'=>'Plasmodium_yoelii_gca_900002395',
	'plasmodium_yoelii_yoelii'=>'Plasmodium_yoelii_yoelii',
	'plasmopara_halstedii'=>'Plasmopara_halstedii',
	'pleurotus_ostreatus_pc15_gca_000697685'=>'Pleurotus_ostreatus_pc15_gca_000697685',
	'plicaturopsis_crispa_fd_325_ss_3_gca_000827205'=>'Plicaturopsis_crispa_fd_325_ss_3_gca_000827205',
	'pneumocystis_carinii_b80_gca_001477545'=>'Pneumocystis_carinii_b80_gca_001477545',
	'pneumocystis_jirovecii_ru7_gca_001477535'=>'Pneumocystis_jirovecii_ru7_gca_001477535',
	'pneumocystis_murina_b123_gca_000349005'=>'Pneumocystis_murina_b123_gca_000349005',
	'pochonia_chlamydosporia_170_gca_001653235'=>'Pochonia_chlamydosporia_170_gca_001653235',
	'podospora_anserina_s_mat__gca_000226545'=>'Podospora_anserina_s_mat__gca_000226545',
	'poecilia_formosa'=>'Amazon_molly',
	'poecilia_latipinna'=>'Sailfin_molly',
	'poecilia_mexicana'=>'Shortfin_molly',
	'poecilia_reticulata'=>'Guppy',
	'pogona_vitticeps'=>'Central_bearded_dragon',
	'polysphondylium_pallidum_pn500'=>'Polysphondylium_pallidum_pn500',
	'pongo_abelii'=>'Orangutan',
	'populus_trichocarpa'=>'Populus_trichocarpa',
	'postia_placenta_mad_698_r_gca_000006255'=>'Postia_placenta_mad_698_r_gca_000006255',
	'postia_placenta_mad_698_r_sb12_gca_002117355'=>'Postia_placenta_mad_698_r_sb12_gca_002117355',
	'procavia_capensis'=>'Hyrax',
	'prolemur_simus'=>'Greater_bamboo_lemur',
	'propithecus_coquereli'=>'Coquerel\'s_sifaka',
	'protomyces_lactucaedebilis_gca_002105105'=>'Protomyces_lactucaedebilis_gca_002105105',
	'prunus_persica'=>'Prunus_persica',
	'pseudocercospora_fijiensis_cirad86_gca_000340215'=>'Pseudocercospora_fijiensis_cirad86_gca_000340215',
	'pseudocercospora_musae_gca_001578225'=>'Pseudocercospora_musae_gca_001578225',
	'pseudogymnoascus_destructans_20631_21_gca_000184105'=>'Pseudogymnoascus_destructans_20631_21_gca_000184105',
	'pseudogymnoascus_destructans_gca_001641265'=>'Pseudogymnoascus_destructans_gca_001641265',
	'pseudogymnoascus_sp_03vt05_gca_001662645'=>'Pseudogymnoascus_sp_03vt05_gca_001662645',
	'pseudogymnoascus_sp_05ny08_gca_001662605'=>'Pseudogymnoascus_sp_05ny08_gca_001662605',
	'pseudogymnoascus_sp_23342_1_i1_gca_001662575'=>'Pseudogymnoascus_sp_23342_1_i1_gca_001662575',
	'pseudogymnoascus_sp_24mn13_gca_001662595'=>'Pseudogymnoascus_sp_24mn13_gca_001662595',
	'pseudogymnoascus_sp_vkm_f_103_gca_000750895'=>'Pseudogymnoascus_sp_vkm_f_103_gca_000750895',
	'pseudogymnoascus_sp_vkm_f_3557_gca_000750665'=>'Pseudogymnoascus_sp_vkm_f_3557_gca_000750665',
	'pseudogymnoascus_sp_vkm_f_3775_gca_000750715'=>'Pseudogymnoascus_sp_vkm_f_3775_gca_000750715',
	'pseudogymnoascus_sp_vkm_f_3808_gca_000750675'=>'Pseudogymnoascus_sp_vkm_f_3808_gca_000750675',
	'pseudogymnoascus_sp_vkm_f_4246_gca_000750735'=>'Pseudogymnoascus_sp_vkm_f_4246_gca_000750735',
	'pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745'=>'Pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745',
	'pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755'=>'Pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755',
	'pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805'=>'Pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805',
	'pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815'=>'Pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815',
	'pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875'=>'Pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875',
	'pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925'=>'Pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925',
	'pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935'=>'Pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935',
	'pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995'=>'Pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995',
	'pseudogymnoascus_sp_wsf_3629_gca_001662585'=>'Pseudogymnoascus_sp_wsf_3629_gca_001662585',
	'pseudogymnoascus_verrucosus_gca_001662655'=>'Pseudogymnoascus_verrucosus_gca_001662655',
	'pseudoloma_neurophilia_gca_001432165'=>'Pseudoloma_neurophilia_gca_001432165',
	'pseudomassariella_vexata_gca_002105095'=>'Pseudomassariella_vexata_gca_002105095',
	'pseudomicrostroma_glucosiphilum_gca_003144135'=>'Pseudomicrostroma_glucosiphilum_gca_003144135',
	'pseudozyma_hubeiensis_sy62_gca_000403515'=>'Pseudozyma_hubeiensis_sy62_gca_000403515',
	'pteropus_vampyrus'=>'Megabat',
	'puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275'=>'Puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275',
	'puccinia_graminis'=>'Puccinia_graminis',
	'puccinia_graminisug99'=>'Puccinia_graminisug99',
	'puccinia_sorghi_gca_001263375'=>'Puccinia_sorghi_gca_001263375',
	'puccinia_striiformis'=>'Puccinia_striiformis',
	'puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645'=>'Puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645',
	'puccinia_triticina'=>'Puccinia_triticina',
	'punctularia_strigosozonata_hhb_11173_ss5_gca_000264995'=>'Punctularia_strigosozonata_hhb_11173_ss5_gca_000264995',
	'pundamilia_nyererei'=>'Makobe_Island_cichlid',
	'purpureocillium_lilacinum_gca_001653205'=>'Purpureocillium_lilacinum_gca_001653205',
	'purpureocillium_lilacinum_gca_001653265'=>'Purpureocillium_lilacinum_gca_001653265',
	'pygocentrus_nattereri'=>'Red-bellied_piranha',
	'pyrenochaeta_sp_ds3say3a_gca_001644535'=>'Pyrenochaeta_sp_ds3say3a_gca_001644535',
	'pyrenophora_teres'=>'Pyrenophora_teres',
	'pyrenophora_tritici_repentis_gca_003171545'=>'Pyrenophora_tritici_repentis_gca_003171545',
	'pyrenophora_tritici_repentis_gca_003231325'=>'Pyrenophora_tritici_repentis_gca_003231325',
	'pyrenophora_tritici_repentis_gca_003231365'=>'Pyrenophora_tritici_repentis_gca_003231365',
	'pyrenophora_tritici_repentis_gca_003231425'=>'Pyrenophora_tritici_repentis_gca_003231425',
	'pyrenophora_triticirepentis'=>'Pyrenophora_triticirepentis',
	'pythium_aphanidermatum'=>'Pythium_aphanidermatum',
	'pythium_arrhenomanes'=>'Pythium_arrhenomanes',
	'pythium_irregulare'=>'Pythium_irregulare',
	'pythium_iwayamai'=>'Pythium_iwayamai',
	'pythium_ultimum'=>'Pythium_ultimum',
	'pythium_vexans'=>'Pythium_vexans',
	'rachicladosporium_antarcticum_gca_002077065'=>'Rachicladosporium_antarcticum_gca_002077065',
	'rachicladosporium_sp_ccfee_5018_gca_002077045'=>'Rachicladosporium_sp_ccfee_5018_gca_002077045',
	'ramularia_collo_cygni_gca_900074925'=>'Ramularia_collo_cygni_gca_900074925',
	'rasamsonia_emersonii_cbs_393_64_gca_000968595'=>'Rasamsonia_emersonii_cbs_393_64_gca_000968595',
	'rattus_norvegicus'=>'Rat',
	'reticulomyxa_filosa'=>'Reticulomyxa_filosa',
	'rhinocladiella_mackenziei_cbs_650_93_gca_000835555'=>'Rhinocladiella_mackenziei_cbs_650_93_gca_000835555',
	'rhinopithecus_bieti'=>'Black_snub-nosed_monkey',
	'rhinopithecus_roxellana'=>'Golden_snub-nosed_monkey',
	'rhizoclosmatium_globosum_gca_002104985'=>'Rhizoclosmatium_globosum_gca_002104985',
	'rhizoctonia_solani_123e_gca_000715385'=>'Rhizoctonia_solani_123e_gca_000715385',
	'rhizoctonia_solani_ag_1_ia_gca_000334115'=>'Rhizoctonia_solani_ag_1_ia_gca_000334115',
	'rhizoctonia_solani_ag_1_ib_gca_000350255'=>'Rhizoctonia_solani_ag_1_ib_gca_000350255',
	'rhizoctonia_solani_ag_8_wac10335_gca_000695385'=>'Rhizoctonia_solani_ag_8_wac10335_gca_000695385',
	'rhizoctonia_solani_gca_001286725'=>'Rhizoctonia_solani_gca_001286725',
	'rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155'=>'Rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155',
	'rhizophagus_irregularis_daom_181602_gca_000439145'=>'Rhizophagus_irregularis_daom_181602_gca_000439145',
	'rhizophagus_irregularis_daom_197198w_gca_000597685'=>'Rhizophagus_irregularis_daom_197198w_gca_000597685',
	'rhizophagus_irregularis_gca_001593125'=>'Rhizophagus_irregularis_gca_001593125',
	'rhizophagus_irregularis_gca_001593145'=>'Rhizophagus_irregularis_gca_001593145',
	'rhizophagus_irregularis_gca_001593155'=>'Rhizophagus_irregularis_gca_001593155',
	'rhizophagus_irregularis_gca_001593205'=>'Rhizophagus_irregularis_gca_001593205',
	'rhizopogon_vinicolor_am_or11_026_gca_001658105'=>'Rhizopogon_vinicolor_am_or11_026_gca_001658105',
	'rhizopus_delemar_ra_99_880_gca_000149305'=>'Rhizopus_delemar_ra_99_880_gca_000149305',
	'rhizopus_microsporus_atcc_52813_gca_002708625'=>'Rhizopus_microsporus_atcc_52813_gca_002708625',
	'rhizopus_microsporus_gca_000825725'=>'Rhizopus_microsporus_gca_000825725',
	'rhizopus_microsporus_gca_002083735'=>'Rhizopus_microsporus_gca_002083735',
	'rhizopus_microsporus_var_microsporus_gca_002083745'=>'Rhizopus_microsporus_var_microsporus_gca_002083745',
	'rhodnius_prolixus'=>'Rhodnius_prolixus',
	'rhodotorula_graminis_wp1_gca_001329695'=>'Rhodotorula_graminis_wp1_gca_001329695',
	'rhodotorula_sp_jg_1b_gca_001541205'=>'Rhodotorula_sp_jg_1b_gca_001541205',
	'rhodotorula_taiwanensis_gca_002922495'=>'Rhodotorula_taiwanensis_gca_002922495',
	'rhodotorula_toruloides_atcc_204091_gca_000222205'=>'Rhodotorula_toruloides_atcc_204091_gca_000222205',
	'rhodotorula_toruloides_np11_gca_000320785'=>'Rhodotorula_toruloides_np11_gca_000320785',
	'rhynchosporium_agropyri_gca_900074905'=>'Rhynchosporium_agropyri_gca_900074905',
	'rhynchosporium_commune_gca_900074885'=>'Rhynchosporium_commune_gca_900074885',
	'rhynchosporium_secalis_gca_900074895'=>'Rhynchosporium_secalis_gca_900074895',
	'rosellinia_necatrix_gca_001445595'=>'Rosellinia_necatrix_gca_001445595',
	'rozella_allomycis_csf55_gca_000442015'=>'Rozella_allomycis_csf55_gca_000442015',
	'saccharomyces_arboricola_h_6_gca_000292725'=>'Saccharomyces_arboricola_h_6_gca_000292725',
	'saccharomyces_cerevisiae'=>'Saccharomyces_cerevisiae',
	'saccharomyces_cerevisiae_ec1118_gca_000218975'=>'Saccharomyces_cerevisiae_ec1118_gca_000218975',
	'saccharomyces_cerevisiae_fostersb_gca_000190255'=>'Saccharomyces_cerevisiae_fostersb_gca_000190255',
	'saccharomyces_cerevisiae_fosterso_gca_000326005'=>'Saccharomyces_cerevisiae_fosterso_gca_000326005',
	'saccharomyces_cerevisiae_gca_001634645'=>'Saccharomyces_cerevisiae_gca_001634645',
	'saccharomyces_cerevisiae_gca_002804325'=>'Saccharomyces_cerevisiae_gca_002804325',
	'saccharomyces_cerevisiae_kyokai_no_7_gca_000260735'=>'Saccharomyces_cerevisiae_kyokai_no_7_gca_000260735',
	'saccharomyces_cerevisiae_lalvin_qa23_gca_000325965'=>'Saccharomyces_cerevisiae_lalvin_qa23_gca_000325965',
	'saccharomyces_cerevisiae_p283_gca_000568295'=>'Saccharomyces_cerevisiae_p283_gca_000568295',
	'saccharomyces_cerevisiae_p301_gca_000568055'=>'Saccharomyces_cerevisiae_p301_gca_000568055',
	'saccharomyces_cerevisiae_r103_gca_000568365'=>'Saccharomyces_cerevisiae_r103_gca_000568365',
	'saccharomyces_cerevisiae_rm11_1a_gca_000149365'=>'Saccharomyces_cerevisiae_rm11_1a_gca_000149365',
	'saccharomyces_cerevisiae_vin13_gca_000190215'=>'Saccharomyces_cerevisiae_vin13_gca_000190215',
	'saccharomyces_cerevisiae_vl3_gca_000190235'=>'Saccharomyces_cerevisiae_vl3_gca_000190235',
	'saccharomyces_cerevisiae_yjm1078_gca_000975645'=>'Saccharomyces_cerevisiae_yjm1078_gca_000975645',
	'saccharomyces_cerevisiae_yjm1083_gca_000975675'=>'Saccharomyces_cerevisiae_yjm1083_gca_000975675',
	'saccharomyces_cerevisiae_yjm1129_gca_000975705'=>'Saccharomyces_cerevisiae_yjm1129_gca_000975705',
	'saccharomyces_cerevisiae_yjm1133_gca_000976695'=>'Saccharomyces_cerevisiae_yjm1133_gca_000976695',
	'saccharomyces_cerevisiae_yjm1190_gca_000976725'=>'Saccharomyces_cerevisiae_yjm1190_gca_000976725',
	'saccharomyces_cerevisiae_yjm1199_gca_000976755'=>'Saccharomyces_cerevisiae_yjm1199_gca_000976755',
	'saccharomyces_cerevisiae_yjm1202_gca_000976785'=>'Saccharomyces_cerevisiae_yjm1202_gca_000976785',
	'saccharomyces_cerevisiae_yjm1208_gca_000976815'=>'Saccharomyces_cerevisiae_yjm1208_gca_000976815',
	'saccharomyces_cerevisiae_yjm1242_gca_000976845'=>'Saccharomyces_cerevisiae_yjm1242_gca_000976845',
	'saccharomyces_cerevisiae_yjm1244_gca_000976875'=>'Saccharomyces_cerevisiae_yjm1244_gca_000976875',
	'saccharomyces_cerevisiae_yjm1248_gca_000976905'=>'Saccharomyces_cerevisiae_yjm1248_gca_000976905',
	'saccharomyces_cerevisiae_yjm1250_gca_000976935'=>'Saccharomyces_cerevisiae_yjm1250_gca_000976935',
	'saccharomyces_cerevisiae_yjm1252_gca_000976965'=>'Saccharomyces_cerevisiae_yjm1252_gca_000976965',
	'saccharomyces_cerevisiae_yjm1273_gca_000976995'=>'Saccharomyces_cerevisiae_yjm1273_gca_000976995',
	'saccharomyces_cerevisiae_yjm1304_gca_000977025'=>'Saccharomyces_cerevisiae_yjm1304_gca_000977025',
	'saccharomyces_cerevisiae_yjm1307_gca_000977055'=>'Saccharomyces_cerevisiae_yjm1307_gca_000977055',
	'saccharomyces_cerevisiae_yjm1311_gca_000977085'=>'Saccharomyces_cerevisiae_yjm1311_gca_000977085',
	'saccharomyces_cerevisiae_yjm1326_gca_000977115'=>'Saccharomyces_cerevisiae_yjm1326_gca_000977115',
	'saccharomyces_cerevisiae_yjm1332_gca_000977145'=>'Saccharomyces_cerevisiae_yjm1332_gca_000977145',
	'saccharomyces_cerevisiae_yjm1336_gca_000977175'=>'Saccharomyces_cerevisiae_yjm1336_gca_000977175',
	'saccharomyces_cerevisiae_yjm1338_gca_000977205'=>'Saccharomyces_cerevisiae_yjm1338_gca_000977205',
	'saccharomyces_cerevisiae_yjm1341_gca_000977235'=>'Saccharomyces_cerevisiae_yjm1341_gca_000977235',
	'saccharomyces_cerevisiae_yjm1342_gca_000977265'=>'Saccharomyces_cerevisiae_yjm1342_gca_000977265',
	'saccharomyces_cerevisiae_yjm1355_gca_000977295'=>'Saccharomyces_cerevisiae_yjm1355_gca_000977295',
	'saccharomyces_cerevisiae_yjm1356_gca_000977325'=>'Saccharomyces_cerevisiae_yjm1356_gca_000977325',
	'saccharomyces_cerevisiae_yjm1381_gca_000977355'=>'Saccharomyces_cerevisiae_yjm1381_gca_000977355',
	'saccharomyces_cerevisiae_yjm1383_gca_000977385'=>'Saccharomyces_cerevisiae_yjm1383_gca_000977385',
	'saccharomyces_cerevisiae_yjm1385_gca_000977415'=>'Saccharomyces_cerevisiae_yjm1385_gca_000977415',
	'saccharomyces_cerevisiae_yjm1386_gca_000977445'=>'Saccharomyces_cerevisiae_yjm1386_gca_000977445',
	'saccharomyces_cerevisiae_yjm1387_gca_000977475'=>'Saccharomyces_cerevisiae_yjm1387_gca_000977475',
	'saccharomyces_cerevisiae_yjm1388_gca_000977505'=>'Saccharomyces_cerevisiae_yjm1388_gca_000977505',
	'saccharomyces_cerevisiae_yjm1389_gca_000977535'=>'Saccharomyces_cerevisiae_yjm1389_gca_000977535',
	'saccharomyces_cerevisiae_yjm1399_gca_000977565'=>'Saccharomyces_cerevisiae_yjm1399_gca_000977565',
	'saccharomyces_cerevisiae_yjm1400_gca_000977595'=>'Saccharomyces_cerevisiae_yjm1400_gca_000977595',
	'saccharomyces_cerevisiae_yjm1401_gca_000977625'=>'Saccharomyces_cerevisiae_yjm1401_gca_000977625',
	'saccharomyces_cerevisiae_yjm1402_gca_000977655'=>'Saccharomyces_cerevisiae_yjm1402_gca_000977655',
	'saccharomyces_cerevisiae_yjm1415_gca_000977685'=>'Saccharomyces_cerevisiae_yjm1415_gca_000977685',
	'saccharomyces_cerevisiae_yjm1417_gca_000977715'=>'Saccharomyces_cerevisiae_yjm1417_gca_000977715',
	'saccharomyces_cerevisiae_yjm1418_gca_000977745'=>'Saccharomyces_cerevisiae_yjm1418_gca_000977745',
	'saccharomyces_cerevisiae_yjm1419_gca_000977775'=>'Saccharomyces_cerevisiae_yjm1419_gca_000977775',
	'saccharomyces_cerevisiae_yjm1433_gca_000977805'=>'Saccharomyces_cerevisiae_yjm1433_gca_000977805',
	'saccharomyces_cerevisiae_yjm1434_gca_000977835'=>'Saccharomyces_cerevisiae_yjm1434_gca_000977835',
	'saccharomyces_cerevisiae_yjm1439_gca_000977865'=>'Saccharomyces_cerevisiae_yjm1439_gca_000977865',
	'saccharomyces_cerevisiae_yjm1443_gca_000977895'=>'Saccharomyces_cerevisiae_yjm1443_gca_000977895',
	'saccharomyces_cerevisiae_yjm1444_gca_000977925'=>'Saccharomyces_cerevisiae_yjm1444_gca_000977925',
	'saccharomyces_cerevisiae_yjm1447_gca_000977955'=>'Saccharomyces_cerevisiae_yjm1447_gca_000977955',
	'saccharomyces_cerevisiae_yjm1450_gca_000977985'=>'Saccharomyces_cerevisiae_yjm1450_gca_000977985',
	'saccharomyces_cerevisiae_yjm1460_gca_000978015'=>'Saccharomyces_cerevisiae_yjm1460_gca_000978015',
	'saccharomyces_cerevisiae_yjm1463_gca_000978045'=>'Saccharomyces_cerevisiae_yjm1463_gca_000978045',
	'saccharomyces_cerevisiae_yjm1477_gca_000978075'=>'Saccharomyces_cerevisiae_yjm1477_gca_000978075',
	'saccharomyces_cerevisiae_yjm1478_gca_000978105'=>'Saccharomyces_cerevisiae_yjm1478_gca_000978105',
	'saccharomyces_cerevisiae_yjm1479_gca_000978135'=>'Saccharomyces_cerevisiae_yjm1479_gca_000978135',
	'saccharomyces_cerevisiae_yjm1526_gca_000978165'=>'Saccharomyces_cerevisiae_yjm1526_gca_000978165',
	'saccharomyces_cerevisiae_yjm1527_gca_000978195'=>'Saccharomyces_cerevisiae_yjm1527_gca_000978195',
	'saccharomyces_cerevisiae_yjm1549_gca_000978225'=>'Saccharomyces_cerevisiae_yjm1549_gca_000978225',
	'saccharomyces_cerevisiae_yjm1573_gca_000978255'=>'Saccharomyces_cerevisiae_yjm1573_gca_000978255',
	'saccharomyces_cerevisiae_yjm1574_gca_000978285'=>'Saccharomyces_cerevisiae_yjm1574_gca_000978285',
	'saccharomyces_cerevisiae_yjm1592_gca_000978315'=>'Saccharomyces_cerevisiae_yjm1592_gca_000978315',
	'saccharomyces_cerevisiae_yjm1615_gca_000978345'=>'Saccharomyces_cerevisiae_yjm1615_gca_000978345',
	'saccharomyces_cerevisiae_yjm189_gca_000975735'=>'Saccharomyces_cerevisiae_yjm189_gca_000975735',
	'saccharomyces_cerevisiae_yjm193_gca_000975765'=>'Saccharomyces_cerevisiae_yjm193_gca_000975765',
	'saccharomyces_cerevisiae_yjm195_gca_000975585'=>'Saccharomyces_cerevisiae_yjm195_gca_000975585',
	'saccharomyces_cerevisiae_yjm244_gca_000975615'=>'Saccharomyces_cerevisiae_yjm244_gca_000975615',
	'saccharomyces_cerevisiae_yjm248_gca_000975795'=>'Saccharomyces_cerevisiae_yjm248_gca_000975795',
	'saccharomyces_cerevisiae_yjm270_gca_000975825'=>'Saccharomyces_cerevisiae_yjm270_gca_000975825',
	'saccharomyces_cerevisiae_yjm271_gca_000975855'=>'Saccharomyces_cerevisiae_yjm271_gca_000975855',
	'saccharomyces_cerevisiae_yjm320_gca_000975885'=>'Saccharomyces_cerevisiae_yjm320_gca_000975885',
	'saccharomyces_cerevisiae_yjm326_gca_000975915'=>'Saccharomyces_cerevisiae_yjm326_gca_000975915',
	'saccharomyces_cerevisiae_yjm428_gca_000975945'=>'Saccharomyces_cerevisiae_yjm428_gca_000975945',
	'saccharomyces_cerevisiae_yjm450_gca_000975975'=>'Saccharomyces_cerevisiae_yjm450_gca_000975975',
	'saccharomyces_cerevisiae_yjm451_gca_000976005'=>'Saccharomyces_cerevisiae_yjm451_gca_000976005',
	'saccharomyces_cerevisiae_yjm453_gca_000976035'=>'Saccharomyces_cerevisiae_yjm453_gca_000976035',
	'saccharomyces_cerevisiae_yjm456_gca_000976065'=>'Saccharomyces_cerevisiae_yjm456_gca_000976065',
	'saccharomyces_cerevisiae_yjm470_gca_000976095'=>'Saccharomyces_cerevisiae_yjm470_gca_000976095',
	'saccharomyces_cerevisiae_yjm541_gca_000976125'=>'Saccharomyces_cerevisiae_yjm541_gca_000976125',
	'saccharomyces_cerevisiae_yjm554_gca_000976155'=>'Saccharomyces_cerevisiae_yjm554_gca_000976155',
	'saccharomyces_cerevisiae_yjm555_gca_000976185'=>'Saccharomyces_cerevisiae_yjm555_gca_000976185',
	'saccharomyces_cerevisiae_yjm627_gca_000976215'=>'Saccharomyces_cerevisiae_yjm627_gca_000976215',
	'saccharomyces_cerevisiae_yjm681_gca_000976245'=>'Saccharomyces_cerevisiae_yjm681_gca_000976245',
	'saccharomyces_cerevisiae_yjm682_gca_000976275'=>'Saccharomyces_cerevisiae_yjm682_gca_000976275',
	'saccharomyces_cerevisiae_yjm683_gca_000976305'=>'Saccharomyces_cerevisiae_yjm683_gca_000976305',
	'saccharomyces_cerevisiae_yjm689_gca_000976335'=>'Saccharomyces_cerevisiae_yjm689_gca_000976335',
	'saccharomyces_cerevisiae_yjm693_gca_000976365'=>'Saccharomyces_cerevisiae_yjm693_gca_000976365',
	'saccharomyces_cerevisiae_yjm789_gca_000181435'=>'Saccharomyces_cerevisiae_yjm789_gca_000181435',
	'saccharomyces_cerevisiae_yjm969_gca_000976395'=>'Saccharomyces_cerevisiae_yjm969_gca_000976395',
	'saccharomyces_cerevisiae_yjm972_gca_000976425'=>'Saccharomyces_cerevisiae_yjm972_gca_000976425',
	'saccharomyces_cerevisiae_yjm975_gca_000976455'=>'Saccharomyces_cerevisiae_yjm975_gca_000976455',
	'saccharomyces_cerevisiae_yjm978_gca_000976485'=>'Saccharomyces_cerevisiae_yjm978_gca_000976485',
	'saccharomyces_cerevisiae_yjm981_gca_000976515'=>'Saccharomyces_cerevisiae_yjm981_gca_000976515',
	'saccharomyces_cerevisiae_yjm984_gca_000976545'=>'Saccharomyces_cerevisiae_yjm984_gca_000976545',
	'saccharomyces_cerevisiae_yjm987_gca_000976575'=>'Saccharomyces_cerevisiae_yjm987_gca_000976575',
	'saccharomyces_cerevisiae_yjm990_gca_000976605'=>'Saccharomyces_cerevisiae_yjm990_gca_000976605',
	'saccharomyces_cerevisiae_yjm993_gca_000662435'=>'Saccharomyces_cerevisiae_yjm993_gca_000662435',
	'saccharomyces_cerevisiae_yjm996_gca_000976665'=>'Saccharomyces_cerevisiae_yjm996_gca_000976665',
	'saccharomyces_eubayanus_gca_001298625'=>'Saccharomyces_eubayanus_gca_001298625',
	'saccharomyces_kudriavzevii_ifo_1802_gca_000167075'=>'Saccharomyces_kudriavzevii_ifo_1802_gca_000167075',
	'saccharomycetaceae_sp_ashbya_aceri__gca_000412225'=>'Saccharomycetaceae_sp_ashbya_aceri__gca_000412225',
	'saimiri_boliviensis_boliviensis'=>'Bolivian_squirrel_monkey',
	'saitoella_complicata_nrrl_y_17804_gca_000227095'=>'Saitoella_complicata_nrrl_y_17804_gca_000227095',
	'saitoella_complicata_nrrl_y_17804_gca_001661265'=>'Saitoella_complicata_nrrl_y_17804_gca_001661265',
	'salpingoeca_rosetta'=>'Salpingoeca_rosetta',
	'salvator_merianae'=>'Argentine_black_and_white_tegu',
	'saprolegnia_diclina_vs20'=>'Saprolegnia_diclina_vs20',
	'saprolegnia_parasitica_cbs_223_65'=>'Saprolegnia_parasitica_cbs_223_65',
	'sarcophilus_harrisii'=>'Tasmanian_devil',
	'sarcoptes_scabiei'=>'Sarcoptes_scabiei',
	'scedosporium_apiospermum_gca_000732125'=>'Scedosporium_apiospermum_gca_000732125',
	'scheffersomyces_stipitis_cbs_6054_gca_000209165'=>'Scheffersomyces_stipitis_cbs_6054_gca_000209165',
	'schistosoma_mansoni'=>'Schistosoma_mansoni',
	'schizophyllum_commune_h4_8_gca_000143185'=>'Schizophyllum_commune_h4_8_gca_000143185',
	'schizopora_paradoxa_gca_001020605'=>'Schizopora_paradoxa_gca_001020605',
	'schizosaccharomyces_cryophilus'=>'Schizosaccharomyces_cryophilus',
	'schizosaccharomyces_japonicus'=>'Schizosaccharomyces_japonicus',
	'schizosaccharomyces_octosporus'=>'Schizosaccharomyces_octosporus',
	'schizosaccharomyces_pombe'=>'Schizosaccharomyces_pombe',
	'scleroderma_citrinum_foug_a_gca_000827425'=>'Scleroderma_citrinum_foug_a_gca_000827425',
	'scleropages_formosus'=>'Asian_bonytongue',
	'sclerotinia_borealis_f_4128_gca_000503235'=>'Sclerotinia_borealis_f_4128_gca_000503235',
	'sclerotinia_sclerotiorum'=>'Sclerotinia_sclerotiorum',
	'sclerotinia_sclerotiorum_1980_uf_70_gca_001857865'=>'Sclerotinia_sclerotiorum_1980_uf_70_gca_001857865',
	'scophthalmus_maximus'=>'Turbot',
	'selaginella_moellendorffii'=>'Selaginella_moellendorffii',
	'serendipita_indica_dsm_11827_gca_000313545'=>'Serendipita_indica_dsm_11827_gca_000313545',
	'serendipita_vermifera_maff_305830_gca_000827415'=>'Serendipita_vermifera_maff_305830_gca_000827415',
	'serinus_canaria'=>'Common_canary',
	'seriola_dumerili'=>'Greater_amberjack',
	'seriola_lalandi_dorsalis'=>'Yellowtail_amberjack',
	'serpula_lacrymans_var_lacrymans_s7_3_gca_000218725'=>'Serpula_lacrymans_var_lacrymans_s7_3_gca_000218725',
	'setaria_italica'=>'Setaria_italica',
	'sistotremastrum_niveocremeum_hhb9708_gca_001630475'=>'Sistotremastrum_niveocremeum_hhb9708_gca_001630475',
	'sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355'=>'Sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355',
	'solanum_lycopersicum'=>'Solanum_lycopersicum',
	'solanum_tuberosum'=>'Solanum_tuberosum',
	'solenopsis_invicta'=>'Solenopsis_invicta',
	'sordaria_macrospora_k_hell_gca_000182805'=>'Sordaria_macrospora_k_hell_gca_000182805',
	'sorex_araneus'=>'Shrew',
	'sorghum_bicolor'=>'Sorghum_bicolor',
	'spathaspora_passalidarum_nrrl_y_27907_gca_000223485'=>'Spathaspora_passalidarum_nrrl_y_27907_gca_000223485',
	'spermophilus_dauricus'=>'Daurian_ground_squirrel',
	'sphaerobolus_stellatus_ss14_gca_000827215'=>'Sphaerobolus_stellatus_ss14_gca_000827215',
	'sphaeroforma_arctica_jp610'=>'Sphaeroforma_arctica_jp610',
	'sphaerulina_musiva_so2202_gca_000320565'=>'Sphaerulina_musiva_so2202_gca_000320565',
	'sphenodon_punctatus'=>'Tuatara',
	'spironucleus_salmonicida'=>'Spironucleus_salmonicida',
	'spizellomyces_punctatus_daom_br117_gca_000182565'=>'Spizellomyces_punctatus_daom_br117_gca_000182565',
	'sporisorium_reilianum'=>'Sporisorium_reilianum',
	'sporisorium_reilianum_f_sp_reilianum_gca_900162835'=>'Sporisorium_reilianum_f_sp_reilianum_gca_900162835',
	'sporothrix_brasiliensis_5110_gca_000820605'=>'Sporothrix_brasiliensis_5110_gca_000820605',
	'sporothrix_insectorum_rcef_264_gca_001636815'=>'Sporothrix_insectorum_rcef_264_gca_001636815',
	'sporothrix_schenckii_1099_18_gca_000961545'=>'Sporothrix_schenckii_1099_18_gca_000961545',
	'sporothrix_schenckii_atcc_58251_gca_000474925'=>'Sporothrix_schenckii_atcc_58251_gca_000474925',
	'spraguea_lophii_42_110_gca_000430065'=>'Spraguea_lophii_42_110_gca_000430065',
	'stachybotrys_chartarum_ibt_40288_gca_000732765'=>'Stachybotrys_chartarum_ibt_40288_gca_000732765',
	'stachybotrys_chlorohalonata_ibt_40285_gca_000732775'=>'Stachybotrys_chlorohalonata_ibt_40285_gca_000732775',
	'stagonospora_sp_src1lsm3a_gca_001644525'=>'Stagonospora_sp_src1lsm3a_gca_001644525',
	'stegastes_partitus'=>'Bicolor_damselfish',
	'stegodyphus_mimosarum'=>'Stegodyphus_mimosarum',
	'stemphylium_lycopersici_gca_001191545'=>'Stemphylium_lycopersici_gca_001191545',
	'stereum_hirsutum_fp_91666_ss1_gca_000264905'=>'Stereum_hirsutum_fp_91666_ss1_gca_000264905',
	'strigamia_maritima'=>'Strigamia_maritima',
	'strigomonas_culicis'=>'Strigomonas_culicis',
	'strongylocentrotus_purpuratus'=>'Strongylocentrotus_purpuratus',
	'strongyloides_ratti'=>'Strongyloides_ratti',
	'sugiyamaella_lignohabitans_gca_001640025'=>'Sugiyamaella_lignohabitans_gca_001640025',
	'suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415'=>'Suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415',
	'suillus_luteus_uh_slu_lm8_n1_gca_000827255'=>'Suillus_luteus_uh_slu_lm8_n1_gca_000827255',
	'sus_scrofa'=>'Pig',
	'syncephalastrum_racemosum_gca_002105135'=>'Syncephalastrum_racemosum_gca_002105135',
	'taeniopygia_guttata'=>'Zebra_Finch',
	'takifugu_rubripes'=>'Fugu',
	'talaromyces_cellulolyticus_gca_000829775'=>'Talaromyces_cellulolyticus_gca_000829775',
	'talaromyces_islandicus_gca_000985935'=>'Talaromyces_islandicus_gca_000985935',
	'talaromyces_marneffei_atcc_18224_gca_000001985'=>'Talaromyces_marneffei_atcc_18224_gca_000001985',
	'talaromyces_marneffei_pm1_gca_000750115'=>'Talaromyces_marneffei_pm1_gca_000750115',
	'talaromyces_stipitatus_atcc_10500_gca_000003125'=>'Talaromyces_stipitatus_atcc_10500_gca_000003125',
	'teleopsis_dalmanni'=>'Teleopsis_dalmanni',
	'termitomyces_sp_j132_gca_001263195'=>'Termitomyces_sp_j132_gca_001263195',
	'testicularia_cyperi_gca_003144125'=>'Testicularia_cyperi_gca_003144125',
	'tetrahymena_thermophila'=>'Tetrahymena_thermophila',
	'tetranychus_urticae'=>'Tetranychus_urticae',
	'tetraodon_nigroviridis'=>'Tetraodon',
	'tetrapisispora_blattae_cbs_6284_gca_000315915'=>'Tetrapisispora_blattae_cbs_6284_gca_000315915',
	'tetrapisispora_phaffii_cbs_4417_gca_000236905'=>'Tetrapisispora_phaffii_cbs_4417_gca_000236905',
	'thalassiosira_oceanica'=>'Thalassiosira_oceanica',
	'thalassiosira_oceanica_ccmp1005'=>'Thalassiosira_oceanica_ccmp1005',
	'thalassiosira_pseudonana'=>'Thalassiosira_pseudonana',
	'thecamonas_trahens_atcc_50062'=>'Thecamonas_trahens_atcc_50062',
	'theileria_annulata'=>'Theileria_annulata',
	'theileria_equi_strain_wa'=>'Theileria_equi_strain_wa',
	'theileria_orientalis_strain_shintoku'=>'Theileria_orientalis_strain_shintoku',
	'theileria_parva'=>'Theileria_parva',
	'thelohanellus_kitauei'=>'Thelohanellus_kitauei',
	'theobroma_cacao'=>'Theobroma_cacao',
	'thermothelomyces_thermophila_atcc_42464_gca_000226095'=>'Thermothelomyces_thermophila_atcc_42464_gca_000226095',
	'theropithecus_gelada'=>'Gelada',
	'thielavia_terrestris_nrrl_8126_gca_000226115'=>'Thielavia_terrestris_nrrl_8126_gca_000226115',
	'thielaviopsis_punctulata_gca_000968615'=>'Thielaviopsis_punctulata_gca_000968615',
	'tilletia_controversa_gca_001645045'=>'Tilletia_controversa_gca_001645045',
	'tilletia_walkeri_gca_001645055'=>'Tilletia_walkeri_gca_001645055',
	'tilletiaria_anomala_ubc_951_gca_000711695'=>'Tilletiaria_anomala_ubc_951_gca_000711695',
	'tilletiopsis_washingtonensis_gca_003144115'=>'Tilletiopsis_washingtonensis_gca_003144115',
	'tolypocladium_ophioglossoides_cbs_100239_gca_001189435'=>'Tolypocladium_ophioglossoides_cbs_100239_gca_001189435',
	'torrubiella_hemipterigena_gca_000825705'=>'Torrubiella_hemipterigena_gca_000825705',
	'tortispora_caseinolytica_nrrl_y_17796_gca_001661475'=>'Tortispora_caseinolytica_nrrl_y_17796_gca_001661475',
	'torulaspora_delbrueckii_gca_000243375'=>'Torulaspora_delbrueckii_gca_000243375',
	'toxoplasma_gondii'=>'Toxoplasma_gondii',
	'toxoplasma_gondii_ari'=>'Toxoplasma_gondii_ari',
	'toxoplasma_gondii_fou'=>'Toxoplasma_gondii_fou',
	'toxoplasma_gondii_gab2_2007_gal_dom2'=>'Toxoplasma_gondii_gab2_2007_gal_dom2',
	'toxoplasma_gondii_gt1'=>'Toxoplasma_gondii_gt1',
	'toxoplasma_gondii_mas'=>'Toxoplasma_gondii_mas',
	'toxoplasma_gondii_p89'=>'Toxoplasma_gondii_p89',
	'toxoplasma_gondii_rub'=>'Toxoplasma_gondii_rub',
	'toxoplasma_gondii_tgcatprc2'=>'Toxoplasma_gondii_tgcatprc2',
	'toxoplasma_gondii_vand'=>'Toxoplasma_gondii_vand',
	'toxoplasma_gondii_veg'=>'Toxoplasma_gondii_veg',
	'trachipleistophora_hominis_gca_000316135'=>'Trachipleistophora_hominis_gca_000316135',
	'trametes_cinnabarina_gca_000765035'=>'Trametes_cinnabarina_gca_000765035',
	'trametes_coccinea_brfm310_gca_002092935'=>'Trametes_coccinea_brfm310_gca_002092935',
	'trametes_pubescens_gca_001895945'=>'Trametes_pubescens_gca_001895945',
	'trametes_versicolor_fp_101664_ss1_gca_000271585'=>'Trametes_versicolor_fp_101664_ss1_gca_000271585',
	'tremella_mesenterica_dsm_1558_gca_000271645'=>'Tremella_mesenterica_dsm_1558_gca_000271645',
	'tribolium_castaneum'=>'Tribolium_castaneum',
	'trichoderma_asperellum_cbs_433_97_gca_003025105'=>'Trichoderma_asperellum_cbs_433_97_gca_003025105',
	'trichoderma_atroviride_imi_206040_gca_000171015'=>'Trichoderma_atroviride_imi_206040_gca_000171015',
	'trichoderma_citrinoviride_gca_003025115'=>'Trichoderma_citrinoviride_gca_003025115',
	'trichoderma_harzianum_cbs_226_95_gca_003025095'=>'Trichoderma_harzianum_cbs_226_95_gca_003025095',
	'trichoderma_harzianum_gca_000988865'=>'Trichoderma_harzianum_gca_000988865',
	'trichoderma_longibrachiatum_atcc_18648_gca_003025155'=>'Trichoderma_longibrachiatum_atcc_18648_gca_003025155',
	'trichoderma_reesei'=>'Trichoderma_reesei',
	'trichoderma_reesei_rut_c_30_gca_000513815'=>'Trichoderma_reesei_rut_c_30_gca_000513815',
	'trichoderma_virens'=>'Trichoderma_virens',
	'trichomonas_vaginalis_g3'=>'Trichomonas_vaginalis_g3',
	'trichophyton_benhamiae_cbs_112371_gca_000151125'=>'Trichophyton_benhamiae_cbs_112371_gca_000151125',
	'trichophyton_equinum_cbs_127_97_gca_000151175'=>'Trichophyton_equinum_cbs_127_97_gca_000151175',
	'trichophyton_interdigitale_h6_gca_000616785'=>'Trichophyton_interdigitale_h6_gca_000616785',
	'trichophyton_interdigitale_mr816_gca_000622975'=>'Trichophyton_interdigitale_mr816_gca_000622975',
	'trichophyton_mentagrophytes_gca_003118255'=>'Trichophyton_mentagrophytes_gca_003118255',
	'trichophyton_rubrum_cbs_100081_gca_000616805'=>'Trichophyton_rubrum_cbs_100081_gca_000616805',
	'trichophyton_rubrum_cbs_118892_gca_000151425'=>'Trichophyton_rubrum_cbs_118892_gca_000151425',
	'trichophyton_rubrum_cbs_202_88_gca_000616985'=>'Trichophyton_rubrum_cbs_202_88_gca_000616985',
	'trichophyton_rubrum_cbs_288_86_gca_000616825'=>'Trichophyton_rubrum_cbs_288_86_gca_000616825',
	'trichophyton_rubrum_cbs_289_86_gca_000616845'=>'Trichophyton_rubrum_cbs_289_86_gca_000616845',
	'trichophyton_rubrum_cbs_735_88_gca_000616965'=>'Trichophyton_rubrum_cbs_735_88_gca_000616965',
	'trichophyton_rubrum_d6_gca_000622995'=>'Trichophyton_rubrum_d6_gca_000622995',
	'trichophyton_rubrum_mr1448_gca_000616905'=>'Trichophyton_rubrum_mr1448_gca_000616905',
	'trichophyton_rubrum_mr1459_gca_000616945'=>'Trichophyton_rubrum_mr1459_gca_000616945',
	'trichophyton_rubrum_mr850_gca_000616765'=>'Trichophyton_rubrum_mr850_gca_000616765',
	'trichophyton_soudanense_cbs_452_61_gca_000616865'=>'Trichophyton_soudanense_cbs_452_61_gca_000616865',
	'trichophyton_tonsurans_cbs_112818_gca_000151455'=>'Trichophyton_tonsurans_cbs_112818_gca_000151455',
	'trichophyton_verrucosum_hki_0517_gca_000151505'=>'Trichophyton_verrucosum_hki_0517_gca_000151505',
	'trichoplax_adhaerens'=>'Trichoplax_adhaerens',
	'trichosporon_asahii_var_asahii_cbs_2479_gca_000293215'=>'Trichosporon_asahii_var_asahii_cbs_2479_gca_000293215',
	'trichosporon_asahii_var_asahii_cbs_8904_gca_000299215'=>'Trichosporon_asahii_var_asahii_cbs_8904_gca_000299215',
	'trifolium_pratense'=>'Trifolium_pratense',
	'triticum_aestivum'=>'Triticum_aestivum',
	'triticum_dicoccoides'=>'Triticum_dicoccoides',
	'triticum_urartu'=>'Triticum_urartu',
	'trypanosoma_brucei'=>'Trypanosoma_brucei',
	'trypanosoma_cruzi'=>'Trypanosoma_cruzi',
	'trypanosoma_cruzi_dm28c'=>'Trypanosoma_cruzi_dm28c',
	'tsuchiyaea_wingfieldii_cbs_7118_gca_001720155'=>'Tsuchiyaea_wingfieldii_cbs_7118_gca_001720155',
	'tuber_borchii_gca_003070745'=>'Tuber_borchii_gca_003070745',
	'tuber_magnatum_gca_003182015'=>'Tuber_magnatum_gca_003182015',
	'tuber_melanosporum'=>'Tuber_melanosporum',
	'tulasnella_calospora_mut_4182_gca_000827465'=>'Tulasnella_calospora_mut_4182_gca_000827465',
	'tupaia_belangeri'=>'Tree_Shrew',
	'tursiops_truncatus'=>'Dolphin',
	'uncinocarpus_reesii_1704_gca_000003515'=>'Uncinocarpus_reesii_1704_gca_000003515',
	'urocitellus_parryii'=>'Arctic_ground_squirrel',
	'ursus_americanus'=>'American_black_bear',
	'ursus_maritimus'=>'Polar_bear',
	'ustilaginoidea_virens_gca_000687475'=>'Ustilaginoidea_virens_gca_000687475',
	'ustilaginomycotina_sp_sa_807_gca_003144235'=>'Ustilaginomycotina_sp_sa_807_gca_003144235',
	'ustilago_bromivora_gca_900080155'=>'Ustilago_bromivora_gca_900080155',
	'ustilago_hordei_gca_000286035'=>'Ustilago_hordei_gca_000286035',
	'ustilago_maydis'=>'Ustilago_maydis',
	'valsa_mali_gca_000818155'=>'Valsa_mali_gca_000818155',
	'valsa_mali_var_pyri_gca_000813385'=>'Valsa_mali_var_pyri_gca_000813385',
	'vanderwaltozyma_polyspora_dsm_70294_gca_000150035'=>'Vanderwaltozyma_polyspora_dsm_70294_gca_000150035',
	'vavraia_culicis_subsp_floridensis_gca_000192795'=>'Vavraia_culicis_subsp_floridensis_gca_000192795',
	'verruconis_gallopava_gca_000836295'=>'Verruconis_gallopava_gca_000836295',
	'verticillium_alfalfae_vams_102_gca_000150825'=>'Verticillium_alfalfae_vams_102_gca_000150825',
	'verticillium_dahliae'=>'Verticillium_dahliae',
	'verticillium_dahliaejr2'=>'Verticillium_dahliaejr2',
	'verticillium_longisporum_gca_001268145'=>'Verticillium_longisporum_gca_001268145',
	'vicugna_pacos'=>'Alpaca',
	'vigna_angularis'=>'Vigna_angularis',
	'vigna_radiata'=>'Vigna_radiata',
	'vitis_vinifera'=>'Vitis_vinifera',
	'vitrella_brassicaformis_ccmp3155'=>'Vitrella_brassicaformis_ccmp3155',
	'vittaforma_corneae_atcc_50505_gca_000231115'=>'Vittaforma_corneae_atcc_50505_gca_000231115',
	'vulpes_vulpes'=>'Red_fox',
	'wallemia_ichthyophaga_exf_994_gca_000400465'=>'Wallemia_ichthyophaga_exf_994_gca_000400465',
	'wallemia_mellicola_cbs_633_66_gca_000263375'=>'Wallemia_mellicola_cbs_633_66_gca_000263375',
	'wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255'=>'Wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255',
	'wickerhamomyces_ciferrii_gca_000313485'=>'Wickerhamomyces_ciferrii_gca_000313485',
	'wolfiporia_cocos_md_104_ss10_gca_000344635'=>'Wolfiporia_cocos_md_104_ss10_gca_000344635',
	'xenopus_tropicalis'=>'Xenopus',
	'xiphophorus_couchianus'=>'Monterrey_platyfish',
	'xiphophorus_maculatus'=>'Platyfish',
	'xylona_heveae_tc161_gca_001619985'=>'Xylona_heveae_tc161_gca_001619985',
	'yarrowia_lipolytica'=>'Yarrowia_lipolytica',
	'yarrowia_lipolytica_gca_001761485'=>'Yarrowia_lipolytica_gca_001761485',
	'yarrowia_lipolytica_gca_900087985'=>'Yarrowia_lipolytica_gca_900087985',
	'zea_mays'=>'Zea_mays',
	'zonotrichia_albicollis'=>'White-throated_sparrow',
	'zootermopsis_nevadensis'=>'Zootermopsis_nevadensis',
	'zygosaccharomyces_bailii_clib_213_gca_000442885'=>'Zygosaccharomyces_bailii_clib_213_gca_000442885',
	'zygosaccharomyces_bailii_gca_900162805'=>'Zygosaccharomyces_bailii_gca_900162805',
	'zygosaccharomyces_bailii_isa1307_gca_000530735'=>'Zygosaccharomyces_bailii_isa1307_gca_000530735',
	'zygosaccharomyces_parabailii_gca_001984395'=>'Zygosaccharomyces_parabailii_gca_001984395',
	'zygosaccharomyces_rouxii_gca_000026365'=>'Zygosaccharomyces_rouxii_gca_000026365',
	'zygosaccharomyces_rouxii_gca_001972345'=>'Zygosaccharomyces_rouxii_gca_001972345',
	'zymoseptoria_brevis_gca_000966595'=>'Zymoseptoria_brevis_gca_000966595',
	'zymoseptoria_tritici'=>'Zymoseptoria_tritici',
	'other'=>'undefined',
	'x'=>'undefined',
	);### SPECIES DICTONARY END ###
		}
	}

sub load_reference_sequences
	{
	if($download_reference==1&&$species{$species}ne'undefined')
		{
		# download and process Genomic tRNA Database sequences
		$request = HTTP::Request->new(GET => 'http://www.smallrnagroup.uni-mainz.de/data/UNITAS/URLs/gtRNAdb');
		$response = $useragent->request($request);
		$eval_URL="";
		$eval_URL = $response->content;
		if(length$eval_URL>0)
			{
			eval$eval_URL;
			}
		else
			{
			if(!$silent){print"\nFailed to load current gtRNAdb URLs from smallRNAgroup-SERVER. Use internal list.";}
### TRNA URLS START ###
%gtRNAdb_urls=
	(
	'_candida_arabinofermentans_nrrl_yb_2248_gca_001661425'=>'',
	'_candida_auris_gca_001189475'=>'',
	'_candida_auris_gca_002759435'=>'',
	'_candida_auris_gca_002775015'=>'',
	'_candida_auris_gca_003013715'=>'',
	'_candida_auris_gca_003014415'=>'',
	'_candida_duobushaemulonis_gca_002926085'=>'',
	'_candida_glabrata_gca_000002545'=>'',
	'_candida_glabrata_gca_002219185'=>'',
	'_candida_glabrata_gca_002219195'=>'',
	'_candida_haemulonis_gca_002926055'=>'',
	'_candida_pseudohaemulonis_gca_003013735'=>'',
	'_candida_tenuis_atcc_10573_gca_000223465'=>'',
	'absidia_glauca_gca_900079185'=>'',
	'absidia_repens_gca_002105175'=>'',
	'acanthochromis_polyacanthus'=>'',
	'acaromyces_ingoldii_gca_003144295'=>'',
	'acidomyces_richmondensis_bfw_gca_001592465'=>'',
	'acremonium_chrysogenum_atcc_11550_gca_000769265'=>'',
	'acyrthosiphon_pisum'=>'',
	'adineta_vaga'=>'',
	'aedes_aegypti'=>'',
	'aegilops_tauschii'=>'',
	'agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555'=>'',
	'ailuropoda_melanoleuca'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Amela1/ailMel1-tRNAs.fa',
	'albugo_laibachii'=>'',
	'allomyces_macrogynus_atcc_38327_gca_000151295'=>'',
	'alternaria_alternata_gca_001642055'=>'',
	'amanita_muscaria_koide_bx008_gca_000827485'=>'',
	'amborella_trichopoda'=>'',
	'amphiamblys_sp_wsbs2006_gca_001875675'=>'',
	'amphilophus_citrinellus'=>'',
	'amphimedon_queenslandica'=>'',
	'amphiprion_ocellaris'=>'',
	'amphiprion_percula'=>'',
	'anabas_testudineus'=>'',
	'anaeromyces_robustus_gca_002104895'=>'',
	'anas_platyrhynchos_platyrhynchos'=>'',
	'angomonas_deanei'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/bacteria/Cand_Kine_crithidii_ex_Angomonas_deanei_ATCC_30255/candKine_CRITHIDII_EX_ANGOMONA-tRNAs.fa',
	'anncaliia_algerae_pra109_gca_000385855'=>'',
	'anncaliia_algerae_pra339_gca_000385875'=>'',
	'anolis_carolinensis'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Acaro2/anoCar2-tRNAs.fa',
	'anopheles_darlingi'=>'',
	'anopheles_gambiae'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Agamb/Agamb-tRNAs.fa',
	'anoplophora_glabripennis'=>'',
	'anser_brachyrhynchus'=>'',
	'anthracocystis_flocculosa_pf_1_gca_000417875'=>'',
	'aotus_nancymaae'=>'',
	'aphanomyces_astaci'=>'',
	'aphanomyces_invadans'=>'',
	'apis_mellifera'=>'',
	'apteryx_haastii'=>'',
	'apteryx_owenii'=>'',
	'apteryx_rowi'=>'',
	'arabidopsis_halleri'=>'',
	'arabidopsis_lyrata'=>'',
	'arabidopsis_thaliana'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Athal10/araTha1-tRNAs.fa',
	'armillaria_gallica_gca_002307695'=>'',
	'armillaria_ostoyae_gca_900157425'=>'',
	'armillaria_solidipes_gca_002307675'=>'',
	'arthrobotrys_oligospora_atcc_24927_gca_000225545'=>'',
	'ascochyta_rabiei_gca_001630375'=>'',
	'ascoidea_rubescens_dsm_1968_gca_001661345'=>'',
	'ashbya_gossypii'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Ashb_goss_ATCC_10895/ashbGoss_ATCC10895-tRNAs.fa',
	'aspergillus_aculeatinus_cbs_121060_gca_003184765'=>'',
	'aspergillus_aculeatus_atcc_16872_gca_001890905'=>'',
	'aspergillus_brasiliensis_cbs_101740_gca_001889945'=>'',
	'aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695'=>'',
	'aspergillus_calidoustus_gca_001511075'=>'',
	'aspergillus_campestris_ibt_28561_gca_002847485'=>'',
	'aspergillus_candidus_gca_002847045'=>'',
	'aspergillus_carbonarius_item_5010_gca_001990825'=>'',
	'aspergillus_clavatus'=>'',
	'aspergillus_cristatus_gca_001717485'=>'',
	'aspergillus_ellipticus_cbs_707_79_gca_003184645'=>'',
	'aspergillus_eucalypticola_cbs_122712_gca_003184535'=>'',
	'aspergillus_flavus_af70_gca_000952835'=>'',
	'aspergillus_fumigatus'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Aspe_fumi_Af293/aspeFumi_AF293-tRNAs.fa',
	'aspergillus_fumigatus_gca_002234985'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Aspe_fumi_Af293/aspeFumi_AF293-tRNAs.fa',
	'aspergillus_fumigatusa1163'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Aspe_fumi_Af293/aspeFumi_AF293-tRNAs.fa',
	'aspergillus_glaucus_cbs_516_65_gca_001890805'=>'',
	'aspergillus_heteromorphus_cbs_117_55_gca_003184545'=>'',
	'aspergillus_indologenus_cbs_114_80_gca_003184685'=>'',
	'aspergillus_japonicus_cbs_114_51_gca_003184785'=>'',
	'aspergillus_lacticoffeatus_cbs_101883_gca_003184595'=>'',
	'aspergillus_lentulus_gca_001445615'=>'',
	'aspergillus_luchuensis_cbs_106_47_gca_001890685'=>'',
	'aspergillus_luchuensis_gca_001602395'=>'',
	'aspergillus_neoniger_cbs_115656_gca_003184625'=>'',
	'aspergillus_nidulans'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Aspe_nidu_FGSC_A4/aspeNidu_FGSC_A4-tRNAs.fa',
	'aspergillus_nidulans_fgsc_a4_gca_000149205'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Aspe_nidu_FGSC_A4/aspeNidu_FGSC_A4-tRNAs.fa',
	'aspergillus_niger'=>'',
	'aspergillus_niger_atcc_1015_gca_000230395'=>'',
	'aspergillus_niger_gca_001515345'=>'',
	'aspergillus_nomius_nrrl_13137_gca_001204775'=>'',
	'aspergillus_novofumigatus_ibt_16806_gca_002847465'=>'',
	'aspergillus_ochraceoroseus_ibt_24754_gca_002846915'=>'',
	'aspergillus_oryzae'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Aspe_oryz_RIB40/aspeOryz_RIB40-tRNAs.fa',
	'aspergillus_oryzae_3_042_gca_000269785'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Aspe_oryz_RIB40/aspeOryz_RIB40-tRNAs.fa',
	'aspergillus_parasiticus_su_1_gca_000956085'=>'',
	'aspergillus_piperis_cbs_112811_gca_003184755'=>'',
	'aspergillus_rambellii_gca_000986645'=>'',
	'aspergillus_ruber_cbs_135680_gca_000600275'=>'',
	'aspergillus_saccharolyticus_jop_1030_1_gca_003184585'=>'',
	'aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635'=>'',
	'aspergillus_sclerotioniger_cbs_115572_gca_003184525'=>'',
	'aspergillus_steynii_ibt_23096_gca_002849105'=>'',
	'aspergillus_sydowii_cbs_593_65_gca_001890705'=>'',
	'aspergillus_taichungensis_gca_002850765'=>'',
	'aspergillus_terreus'=>'',
	'aspergillus_thermomutatus_gca_002237265'=>'',
	'aspergillus_tubingensis_cbs_134_48_gca_001890745'=>'',
	'aspergillus_turcosus_gca_002234965'=>'',
	'aspergillus_turcosus_gca_002234975'=>'',
	'aspergillus_udagawae_gca_001078395'=>'',
	'aspergillus_ustus_gca_000812125'=>'',
	'aspergillus_uvarum_cbs_121591_gca_003184745'=>'',
	'aspergillus_vadensis_cbs_113365_gca_003184925'=>'',
	'aspergillus_versicolor_cbs_583_65_gca_001890125'=>'',
	'aspergillus_violaceofuscus_cbs_115571_gca_003184705'=>'',
	'aspergillus_wentii_dto_134e9_gca_001890725'=>'',
	'astatotilapia_calliptera'=>'',
	'astyanax_mexicanus'=>'',
	'atta_cephalotes'=>'',
	'aureobasidium_melanogenum_cbs_110374_gca_000721775'=>'',
	'aureobasidium_namibiae_cbs_147_97_gca_000721765'=>'',
	'aureobasidium_pullulans_exf_150_gca_000721785'=>'',
	'aureobasidium_subglaciale_exf_2481_gca_000721755'=>'',
	'babesia_bovis'=>'',
	'babesia_microti_strain_ri'=>'',
	'babjeviella_inositovora_nrrl_y_12698_gca_001661335'=>'',
	'basidiobolus_meristosporus_cbs_931_73_gca_002104905'=>'',
	'batrachochytrium_dendrobatidis_jam81_gca_000203795'=>'',
	'batrachochytrium_dendrobatidis_jel423_gca_000149865'=>'',
	'batrachochytrium_salamandrivorans_gca_002006685'=>'',
	'baudoinia_panamericana_uamh_10762_gca_000338955'=>'',
	'beauveria_bassiana'=>'',
	'beauveria_bassiana_arsef_2860_gca_000280675'=>'',
	'beauveria_bassiana_d1_5_gca_000770705'=>'',
	'beauveria_bassiana_gca_002871155'=>'',
	'belgica_antarctica'=>'',
	'beta_vulgaris'=>'',
	'bifiguratus_adelaidae_gca_002261195'=>'',
	'bigelowiella_natans'=>'',
	'bigelowiella_natans_gca_000002455'=>'',
	'bipolaris_maydis_atcc_48331_gca_000354255'=>'',
	'bipolaris_maydis_c5_gca_000338975'=>'',
	'bipolaris_sorokiniana_nd90pr_gca_000338995'=>'',
	'bipolaris_zeicola_26_r_13_gca_000523435'=>'',
	'bison_bison_bison'=>'',
	'blastomyces_dermatitidis_atcc_18188_gca_000151595'=>'',
	'blastomyces_dermatitidis_atcc_26199_gca_000166155'=>'',
	'blastomyces_dermatitidis_er_3_gca_000003525'=>'',
	'blastomyces_gilchristii_slh14081_gca_000003855'=>'',
	'blastomyces_percursus_gca_001883805'=>'',
	'blumeria_graminis'=>'',
	'blumeria_graminis_f_sp_tritici_96224_gca_000418435'=>'',
	'bodo_saltans'=>'',
	'bombus_impatiens'=>'',
	'bombus_terrestris'=>'',
	'bombyx_mori'=>'',
	'bos_mutus'=>'',
	'bos_taurus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Btaur8/bosTau8-tRNAs.fa',
	'botryobasidium_botryosum_fd_172_ss1_gca_000697705'=>'',
	'botrytis_cinerea'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Botr_cine_B05_10/botrCine_B05_10-tRNAs.fa',
	'botrytis_cinerea_bcdw1_gca_000349525'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Botr_cine_B05_10/botrCine_B05_10-tRNAs.fa',
	'botrytis_cinerea_t4_gca_000227075'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Botr_cine_B05_10/botrCine_B05_10-tRNAs.fa',
	'brachypodium_distachyon'=>'',
	'brassica_napus'=>'',
	'brassica_oleracea'=>'',
	'brassica_rapa'=>'',
	'brettanomyces_bruxellensis_awri1499_gca_000259595'=>'',
	'brugia_malayi'=>'',
	'byssochlamys_spectabilis_no_5_gca_000497085'=>'',
	'cadophora_sp_dse1049_gca_003073865'=>'',
	'caenorhabditis_brenneri'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Cbren2/caePb2-tRNAs.fa',
	'caenorhabditis_briggsae'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Cbrig3/cb3-tRNAs.fa',
	'caenorhabditis_elegans'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Celeg_WS220/ce10-tRNAs.fa',
	'caenorhabditis_japonica'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Cjapo1/caeJap1-tRNAs.fa',
	'caenorhabditis_remanei'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Crema3/caeRem3-tRNAs.fa',
	'calidris_pugnax'=>'',
	'calidris_pygmaea'=>'',
	'callithrix_jacchus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Cjacc3/calJac3-tRNAs.fa',
	'calocera_cornea_hhb12733_gca_001632435'=>'',
	'calocera_viscosa_tufc12733_gca_001630345'=>'',
	'candida_albicans_12c_gca_000773845'=>'',
	'candida_albicans_19f_gca_000775445'=>'',
	'candida_albicans_ca6_gca_000784695'=>'',
	'candida_albicans_gc75_gca_000773735'=>'',
	'candida_albicans_l26_gca_000775455'=>'',
	'candida_albicans_p34048_gca_000775465'=>'',
	'candida_albicans_p37005_gca_000773745'=>'',
	'candida_albicans_p37037_gca_000773825'=>'',
	'candida_albicans_p37039_gca_000784515'=>'',
	'candida_albicans_p57055_gca_000775505'=>'',
	'candida_albicans_p57072_gca_000773805'=>'',
	'candida_albicans_p60002_gca_000784525'=>'',
	'candida_albicans_p75010_gca_000784575'=>'',
	'candida_albicans_p75016_gca_000784595'=>'',
	'candida_albicans_p75063_gca_000775525'=>'',
	'candida_albicans_p76055_gca_000784505'=>'',
	'candida_albicans_p76067_gca_000784495'=>'',
	'candida_albicans_p78042_gca_000784615'=>'',
	'candida_albicans_p78048_gca_000773725'=>'',
	'candida_albicans_p87_gca_000774085'=>'',
	'candida_albicans_p94015_gca_000773755'=>'',
	'candida_albicans_sc5314_gca_000784635'=>'',
	'candida_albicans_sc5314_gca_000784655'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Cand_albi_WO_1/candAlbi_WO_1-tRNAs.fa',
	'candida_albicans_wo_1_gca_000149445'=>'',
	'candida_dubliniensis_cd36_gca_000026945'=>'',
	'candida_maltosa_xu316_gca_000344705'=>'',
	'candida_orthopsilosis_co_90_125_gca_000315875'=>'',
	'candida_parapsilosis_gca_000182765'=>'',
	'candida_tropicalis_mya_3404_gca_000006335'=>'',
	'canis_familiaris'=>'',
	'canis_lupus_dingo'=>'',
	'capitella_teleta'=>'',
	'capra_hircus'=>'',
	'capronia_coronata_cbs_617_96_gca_000585585'=>'',
	'capronia_epimyces_cbs_606_96_gca_000585565'=>'',
	'capsaspora_owczarzaki_atcc_30864'=>'',
	'carlito_syrichta'=>'',
	'castor_canadensis'=>'',
	'catenaria_anguillulae_pl171_gca_002102555'=>'',
	'cavia_aperea'=>'',
	'cavia_porcellus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Cporc3/cavPor3-tRNAs.fa',
	'cebus_capucinus'=>'',
	'ceraceosorus_bombacis_gca_900000165'=>'',
	'ceraceosorus_guamensis_gca_003144195'=>'',
	'ceratocystis_platani_gca_000978885'=>'',
	'cercocebus_atys'=>'',
	'chaetomium_globosum_cbs_148_51_gca_000143365'=>'',
	'chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225'=>'',
	'chelonoidis_abingdonii'=>'',
	'chinchilla_lanigera'=>'',
	'chlamydomonas_reinhardtii'=>'',
	'chlorocebus_sabaeus'=>'',
	'choanephora_cucurbitarum_gca_001683725'=>'',
	'choloepus_hoffmanni'=>'',
	'chondrus_crispus'=>'',
	'chroomonas_mesostigmatica_ccmp1168'=>'',
	'chrysemys_picta_bellii'=>'',
	'ciona_intestinalis'=>'',
	'ciona_savignyi'=>'',
	'cladophialophora_bantiana_cbs_173_52_gca_000835475'=>'',
	'cladophialophora_carrionii_cbs_160_54_gca_000365165'=>'',
	'cladophialophora_carrionii_gca_001700775'=>'',
	'cladophialophora_immunda_gca_000835495'=>'',
	'cladophialophora_psammophila_cbs_110553_gca_000585535'=>'',
	'cladophialophora_yegresii_cbs_114405_gca_000585515'=>'',
	'claviceps_purpurea_20_1_gca_000347355'=>'',
	'clavispora_lusitaniae_atcc_42720_gca_000003835'=>'',
	'clavispora_lusitaniae_gca_001673695'=>'',
	'clohesyomyces_aquaticus_gca_002105025'=>'',
	'coccidioides_immitis_h538_4_gca_000149815'=>'',
	'coccidioides_immitis_rmscc_2394_gca_000149895'=>'',
	'coccidioides_immitis_rmscc_3703_gca_000150085'=>'',
	'coccidioides_immitis_rs_gca_000149335'=>'',
	'coccidioides_posadasii_c735_delta_sowgp_gca_000151335'=>'',
	'coccidioides_posadasii_rmscc_3488_gca_000150055'=>'',
	'coccidioides_posadasii_str_silveira_gca_000170175'=>'',
	'colletotrichum_fioriniae_pj7_gca_000582985'=>'',
	'colletotrichum_gloeosporioides'=>'',
	'colletotrichum_gloeosporioides_cg_14_gca_000446055'=>'',
	'colletotrichum_graminicola'=>'',
	'colletotrichum_higginsianum'=>'',
	'colletotrichum_incanum_gca_001625285'=>'',
	'colletotrichum_nymphaeae_sa_01_gca_001563115'=>'',
	'colletotrichum_orbiculare'=>'',
	'colletotrichum_salicis_gca_001563125'=>'',
	'colletotrichum_sublineola_gca_000696135'=>'',
	'colletotrichum_tofieldiae_gca_001625265'=>'',
	'colobus_angolensis_palliatus'=>'',
	'conidiobolus_coronatus_nrrl_28638_gca_001566745'=>'',
	'coniella_lustricola_gca_003019895'=>'',
	'coniochaeta_ligniaria_nrrl_30616_gca_001879275'=>'',
	'coniophora_puteana_rwd_64_598_ss2_gca_000271625'=>'',
	'coniosporium_apollinis_cbs_100218_gca_000281105'=>'',
	'coprinopsis_cinerea_okayama7_130_gca_000182895'=>'',
	'corchorus_capsularis'=>'',
	'cordyceps_brongniartii_rcef_3172_gca_001636735'=>'',
	'cordyceps_confragosa_rcef_1005_gca_001636795'=>'',
	'cordyceps_fumosorosea_arsef_2679_gca_001636725'=>'',
	'cordyceps_militaris_cm01_gca_000225605'=>'',
	'corynespora_cassiicola_philippines_gca_003016335'=>'',
	'coturnix_japonica'=>'',
	'crassostrea_gigas'=>'',
	'cricetulus_griseus_chok1gshd'=>'',
	'cricetulus_griseus_crigri'=>'',
	'cricetulus_griseus_picr'=>'',
	'crocodylus_porosus'=>'',
	'cryptococcus_amylolentus_cbs_6039_gca_001720205'=>'',
	'cryptococcus_amylolentus_cbs_6273_gca_001720235'=>'',
	'cryptococcus_depauperatus_cbs_7841_gca_001720195'=>'',
	'cryptococcus_depauperatus_cbs_7855_gca_001720245'=>'',
	'cryptococcus_gattii_ca1280_gca_000836335'=>'',
	'cryptococcus_gattii_ca1873_gca_000855695'=>'',
	'cryptococcus_gattii_e566_gca_000875815'=>'',
	'cryptococcus_gattii_ejb2_gca_000835745'=>'',
	'cryptococcus_gattii_nt_10_gca_000935105'=>'',
	'cryptococcus_gattii_ru294_gca_000836355'=>'',
	'cryptococcus_gattii_vgii_2001_935_1_gca_000835815'=>'',
	'cryptococcus_gattii_vgii_99_473_gca_000836455'=>'',
	'cryptococcus_gattii_vgii_ca1014_gca_000875795'=>'',
	'cryptococcus_gattii_vgii_cbs_10090_gca_000835765'=>'',
	'cryptococcus_gattii_vgii_la55_gca_000836315'=>'',
	'cryptococcus_gattii_vgii_mmrl2647_gca_000875855'=>'',
	'cryptococcus_gattii_vgii_r265_gca_000149475'=>'',
	'cryptococcus_gattii_vgii_ram5_gca_000836375'=>'',
	'cryptococcus_gattii_vgiv_ind107_gca_000835755'=>'',
	'cryptococcus_gattii_wm276_gca_000185945'=>'',
	'cryptococcus_neoformans'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Cryp_neof_var_grubii_H99/crypNeof_VAR_GRUBII_H99-tRNAs.fa',
	'cryptococcus_neoformans_var_grubii_h99_gca_000149245'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Cryp_neof_var_grubii_H99/crypNeof_VAR_GRUBII_H99-tRNAs.fa',
	'cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Cryp_neof_var_grubii_H99/crypNeof_VAR_GRUBII_H99-tRNAs.fa',
	'cryptomonas_paramecium'=>'',
	'cryptosporidium_muris_rn66'=>'',
	'cryptosporidium_parvum'=>'',
	'cryptosporidium_parvum_iowa_ii'=>'',
	'cucumis_sativus'=>'',
	'culex_quinquefasciatus'=>'',
	'cutaneotrichosporon_oleaginosum_gca_001027345'=>'',
	'cyanistes_caeruleus'=>'',
	'cyberlindnera_fabianii_gca_003205855'=>'',
	'cyberlindnera_jadinii_gca_001245095'=>'',
	'cyberlindnera_jadinii_nrrl_y_1542_gca_001661405'=>'',
	'cyclospora_cayetanensis'=>'',
	'cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385'=>'',
	'cynoglossus_semilaevis'=>'',
	'cyphellophora_europaea_cbs_101466_gca_000365145'=>'',
	'cyprinodon_variegatus'=>'',
	'dacryopinax_primogenitus_gca_000292625'=>'',
	'dactylellina_haptotyla_cbs_200_50_gca_000441935'=>'',
	'daedalea_quercina_l_15889_gca_001632345'=>'',
	'daldinia_sp_ec12_gca_002120325'=>'',
	'danaus_plexippus'=>'',
	'danio_rerio'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Dreri_v8/danRer6-tRNAs.fa',
	'daphnia_magna'=>'',
	'daphnia_pulex'=>'',
	'dasypus_novemcinctus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Dnove3/dasNov3-tRNAs.fa',
	'daucus_carota'=>'',
	'debaryomyces_fabryi_gca_001447935'=>'',
	'debaryomyces_hansenii_cbs767_gca_000006445'=>'',
	'dendroctonus_ponderosae'=>'',
	'diaporthe_ampelina_gca_001006365'=>'',
	'dichomitus_squalens_lyad_421_ss1_gca_000275845'=>'',
	'dictyostelium_discoideum'=>'',
	'dictyostelium_fasciculatum'=>'',
	'dictyostelium_lacteum'=>'',
	'dioscorea_rotundata'=>'',
	'diplodia_seriata_gca_001006355'=>'',
	'dipodomys_ordii'=>'',
	'dothistroma_septosporum'=>'',
	'drechmeria_coniospora_gca_001625195'=>'',
	'drechslerella_stenobrocha_248_gca_000525045'=>'',
	'dromaius_novaehollandiae'=>'',
	'drosophila_ananassae'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Danan3/droAna3-tRNAs.fa',
	'drosophila_erecta'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Derec2/droEre2-tRNAs.fa',
	'drosophila_grimshawi'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dgrim2/droGri2-tRNAs.fa',
	'drosophila_melanogaster'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Dmela6/dm6-tRNAs.fa',
	'drosophila_mojavensis'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dmoja3/droMoj3-tRNAs.fa',
	'drosophila_persimilis'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dpers1/droPer1-tRNAs.fa',
	'drosophila_pseudoobscura'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dpseu4/dp4-tRNAs.fa',
	'drosophila_sechellia'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dsech1/droSec1-tRNAs.fa',
	'drosophila_simulans'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dsimu1/droSim1-tRNAs.fa',
	'drosophila_virilis'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dviri3/droVir3-tRNAs.fa',
	'drosophila_willistoni'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dwill1/droWil1-tRNAs.fa',
	'drosophila_yakuba'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Dyaku2/droYak2-tRNAs.fa',
	'echinops_telfairi'=>'',
	'edhazardia_aedis_usnm_41457_gca_000230595'=>'',
	'eimeria_brunetti'=>'',
	'eimeria_maxima'=>'',
	'eimeria_mitis'=>'',
	'eimeria_praecox'=>'',
	'eimeria_tenella'=>'',
	'eimeria_tenella_gca_000499545'=>'',
	'emergomyces_pasteuriana_ep9510_gca_001883825'=>'',
	'emiliania_huxleyi'=>'',
	'emmonsia_crescens_uamh_3008_gca_001008285'=>'',
	'emmonsia_parva_uamh_139_gca_001014755'=>'',
	'emmonsia_sp_cac_2015a_gca_001660665'=>'',
	'encephalitozoon_cuniculi_ecuniii_l_gca_001078035'=>'',
	'encephalitozoon_hellem_atcc_50504_gca_000277815'=>'',
	'encephalitozoon_intestinalis_atcc_50506_gca_000146465'=>'',
	'encephalitozoon_romaleae_sj_2008_gca_000280035'=>'',
	'endocarpon_pusillum_z07020_gca_000464535'=>'',
	'entamoeba_histolytica'=>'',
	'entamoeba_histolytica_gca_001662325'=>'',
	'entamoeba_histolytica_hm_3_imss'=>'',
	'entamoeba_histolytica_ku27'=>'',
	'entamoeba_nuttalli_p19'=>'',
	'enterocytozoon_bieneusi_h348_gca_000209485'=>'',
	'epicoccum_nigrum_gca_002116315'=>'',
	'eptatretus_burgeri'=>'',
	'equus_asinus_asinus'=>'',
	'equus_caballus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Ecaba2/equCab2-tRNAs.fa',
	'eremothecium_cymbalariae_dbvpg_7215_gca_000235365'=>'',
	'eremothecium_gossypii_fdag1_gca_000968835'=>'',
	'erinaceus_europaeus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Eeuro2/eriEur2-tRNAs.fa',
	'erysiphe_necator_gca_000798715'=>'',
	'escovopsis_weberi_gca_001278495'=>'',
	'esox_lucius'=>'',
	'eutypa_lata_ucrel1_gca_000349385'=>'',
	'exidia_glandulosa_hhb12029_gca_001632375'=>'',
	'exophiala_aquamarina_cbs_119918_gca_000709125'=>'',
	'exophiala_dermatitidis_nih_ut8656_gca_000230625'=>'',
	'exophiala_mesophila_gca_000836275'=>'',
	'exophiala_oligosperma_gca_000835515'=>'',
	'exophiala_sideris_gca_000835395'=>'',
	'exophiala_spinifera_gca_000836115'=>'',
	'exophiala_xenobiotica_gca_000835505'=>'',
	'exserohilum_turcica_et28a_gca_000359705'=>'',
	'felis_catus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Fcatu5/felCat5-tRNAs.fa',
	'fibroporia_radiculosa_gca_000313525'=>'',
	'fibularhizoctonia_sp_cbs_109695_gca_001630335'=>'',
	'ficedula_albicollis'=>'',
	'fistulina_hepatica_atcc_64428_gca_000934395'=>'',
	'fomitiporia_mediterranea_mf3_22_gca_000271605'=>'',
	'fomitopsis_pinicola_fp_58527_ss1_gca_000344655'=>'',
	'fonsecaea_erecta_gca_001651985'=>'',
	'fonsecaea_monophora_gca_001642475'=>'',
	'fonsecaea_multimorphosa_cbs_102226_gca_000836435'=>'',
	'fonsecaea_multimorphosa_gca_001646985'=>'',
	'fonsecaea_nubica_gca_001646965'=>'',
	'fonsecaea_pedrosoi_cbs_271_37_gca_000835455'=>'',
	'fonticula_alba_gca_000388065'=>'',
	'fukomys_damarensis'=>'',
	'fundulus_heteroclitus'=>'',
	'fusarium_culmorum'=>'',
	'fusarium_fujikuroi'=>'',
	'fusarium_fujikuroi_gca_001023035'=>'',
	'fusarium_fujikuroi_gca_001023045'=>'',
	'fusarium_fujikuroi_gca_001023065'=>'',
	'fusarium_fujikuroi_gca_900096505'=>'',
	'fusarium_fujikuroi_gca_900096685'=>'',
	'fusarium_fujikuroi_gca_900096705'=>'',
	'fusarium_graminearum'=>'',
	'fusarium_graminearum_gca_000240135'=>'',
	'fusarium_langsethiae_gca_001292635'=>'',
	'fusarium_mangiferae_gca_900044065'=>'',
	'fusarium_oxysporum'=>'',
	'fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215'=>'',
	'fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345'=>'',
	'fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365'=>'',
	'fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195'=>'',
	'fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955'=>'',
	'fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975'=>'',
	'fusarium_oxysporum_f_sp_melonis_26406_gca_000260495'=>'',
	'fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075'=>'',
	'fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155'=>'',
	'fusarium_oxysporum_f_sp_raphani_54005_gca_000260235'=>'',
	'fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175'=>'',
	'fusarium_oxysporum_fo47_gca_000271705'=>'',
	'fusarium_oxysporum_fosc_3_a_gca_000271745'=>'',
	'fusarium_oxysporum_gca_900096695'=>'',
	'fusarium_poae_gca_001675295'=>'',
	'fusarium_proliferatum_et1_gca_900067095'=>'',
	'fusarium_proliferatum_gca_900029915'=>'',
	'fusarium_pseudograminearum'=>'',
	'fusarium_solani'=>'',
	'fusarium_venenatum_gca_900007375'=>'',
	'fusarium_verticillioides'=>'',
	'gadus_morhua'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Gmorh1/gadMor1-tRNAs.fa',
	'gaeumannomyces_graminis'=>'',
	'galdieria_sulphuraria'=>'',
	'galerina_marginata_cbs_339_88_gca_000697645'=>'',
	'gallus_gallus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Ggall4/galGal4-tRNAs.fa',
	'gambusia_affinis'=>'',
	'gasterosteus_aculeatus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Gacul1/gasAcu1-tRNAs.fa',
	'gelatoporia_subvermispora_b_gca_000320605'=>'',
	'giardia_intestinalis'=>'',
	'giardia_intestinalis_assemblage_b'=>'',
	'giardia_intestinalis_atcc_50581'=>'',
	'giardia_intestinalis_gca_000498715'=>'',
	'giardia_lamblia'=>'',
	'giardia_lamblia_p15'=>'',
	'glarea_lozoyensis_atcc_20868_gca_000409485'=>'',
	'gloeophyllum_trabeum_atcc_11539_gca_000344685'=>'',
	'glycine_max'=>'',
	'gonapodya_prolifera_jel478_gca_001574975'=>'',
	'gopherus_agassizii'=>'',
	'gorilla_gorilla'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Ggori3/gorGor3-tRNAs.fa',
	'gossypium_raimondii'=>'',
	'gregarina_niphandrodes'=>'',
	'grifola_frondosa_gca_001683735'=>'',
	'grosmannia_clavigera_kw1407_gca_000143105'=>'',
	'guillardia_theta'=>'',
	'guillardia_theta_gca_000002975'=>'',
	'gymnopus_luxurians_fd_317_m1_gca_000827265'=>'',
	'hammondia_hammondi'=>'',
	'hanseniaspora_guilliermondii_gca_900119595'=>'',
	'hanseniaspora_opuntiae_gca_001749795'=>'',
	'hanseniaspora_osmophila_gca_001747045'=>'',
	'hanseniaspora_uvarum_dsm_2768_gca_000968475'=>'',
	'hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025'=>'',
	'haplochromis_burtoni'=>'',
	'hebeloma_cylindrosporum_h7_gca_000827355'=>'',
	'helianthus_annuus'=>'',
	'heliconius_melpomene'=>'',
	'helobdella_robusta'=>'',
	'hesseltinella_vesiculosa_gca_002104935'=>'',
	'heterobasidion_irregulare_tc_32_1_gca_000320585'=>'',
	'heterocephalus_glaber_female'=>'',
	'heterocephalus_glaber_male'=>'',
	'hippocampus_comes'=>'',
	'hirsutella_minnesotensis_3608_gca_000956045'=>'',
	'histoplasma_capsulatum_g186ar_gca_000150115'=>'',
	'histoplasma_capsulatum_h143_gca_000151035'=>'',
	'histoplasma_capsulatum_h88_gca_000151005'=>'',
	'histoplasma_capsulatum_nam1_gca_000149585'=>'',
	'homo_sapiens'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Hsapi19/hg19-tRNAs.fa',
	'hordeum_vulgare'=>'',
	'hortaea_werneckii_exf_2000_gca_002127715'=>'',
	'hyaloperonospora_arabidopsidis'=>'',
	'hydnomerulius_pinastri_md_312_gca_000827185'=>'',
	'hypholoma_sublateritium_fd_334_ss_4_gca_000827495'=>'',
	'hyphopichia_burtonii_nrrl_y_1933_gca_001661395'=>'',
	'hypoxylon_sp_ci_4a_gca_002120315'=>'',
	'hypoxylon_sp_co27_5_gca_002120305'=>'',
	'hypoxylon_sp_ec38_gca_002120335'=>'',
	'hypsizygus_marmoreus_gca_001605315'=>'',
	'ichthyophthirius_multifiliis'=>'',
	'ictalurus_punctatus'=>'',
	'ictidomys_tridecemlineatus'=>'',
	'ixodes_scapularis'=>'',
	'jaapia_argillacea_mucl_33604_gca_000697665'=>'',
	'jaculus_jaculus'=>'',
	'jaminaea_rosea_gca_003144245'=>'',
	'junco_hyemalis'=>'',
	'kalmanozyma_brasiliensis_ghg001_gca_000497045'=>'',
	'kazachstania_africana_cbs_2517_gca_000304475'=>'',
	'kazachstania_naganishii_cbs_8797_gca_000348985'=>'',
	'kluyveromyces_dobzhanskii_cbs_2104_gca_000820885'=>'',
	'kluyveromyces_lactis_gca_000002515'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Kluy_lact_NRRL_Y_1140/kluyLact_NRRL_Y_1140-tRNAs.fa',
	'kluyveromyces_marxianus_dmku3_1042_gca_001417885'=>'',
	'kluyveromyces_marxianus_gca_001417835'=>'',
	'kockovaella_imperatae_gca_002102565'=>'',
	'komagataella_pastoris'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Koma_past_CBS_7435/komaPast_CBS_7435-tRNAs.fa',
	'komagataella_pastoris_gca_001708105'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Koma_past_CBS_7435/komaPast_CBS_7435-tRNAs.fa',
	'komagataella_phaffii_cbs_7435_gca_000223565'=>'',
	'komagataella_phaffii_gca_001708085'=>'',
	'komagataella_phaffii_gs115_gca_001746955'=>'',
	'kryptolebias_marmoratus'=>'',
	'kuraishia_capsulata_cbs_1993_gca_000576695'=>'',
	'kwoniella_bestiolae_cbs_10118_gca_000512585'=>'',
	'kwoniella_dejecticola_cbs_10117_gca_000512565'=>'',
	'kwoniella_heveanensis_bcc8398_gca_000507405'=>'',
	'kwoniella_heveanensis_cbs_569_gca_000507425'=>'',
	'kwoniella_mangroviensis_cbs_10435_gca_000507885'=>'',
	'kwoniella_mangroviensis_cbs_8507_gca_000507465'=>'',
	'kwoniella_mangroviensis_cbs_8886_gca_000507485'=>'',
	'kwoniella_pini_cbs_10737_gca_000512605'=>'',
	'labrus_bergylta'=>'',
	'laccaria_amethystina_laam_08_1_gca_000827195'=>'',
	'laccaria_bicolor_s238n_h82_gca_000143565'=>'',
	'lachancea_dasiensis_cbs_10888_gca_900074725'=>'',
	'lachancea_fermentati_gca_900074765'=>'',
	'lachancea_lanzarotensis_gca_000938715'=>'',
	'lachancea_meyersii_cbs_8951_gca_900074715'=>'',
	'lachancea_mirantina_gca_900074745'=>'',
	'lachancea_nothofagi_cbs_11611_gca_900074755'=>'',
	'lachancea_quebecensis_gca_002900925'=>'',
	'lachancea_sp_cbs_6924_gca_900074735'=>'',
	'lachancea_thermotolerans_cbs_6340_gca_000142805'=>'',
	'laetiporus_sulphureus_93_53_gca_001632365'=>'',
	'latimeria_chalumnae'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Lchal1/latCha1-tRNAs.fa',
	'leersia_perrieri'=>'',
	'leishmania_braziliensis_mhom_br_75_m2904'=>'',
	'leishmania_donovani_bpk282a1'=>'',
	'leishmania_infantum_jpcm5'=>'',
	'leishmania_major'=>'',
	'leishmania_mexicana_mhom_gt_2001_u1103'=>'',
	'leishmania_panamensis'=>'',
	'lepeophtheirus_salmonis'=>'',
	'lepidothrix_coronata'=>'',
	'lepisosteus_oculatus'=>'',
	'leptomonas_pyrrhocoris'=>'',
	'leptomonas_seymouri'=>'',
	'leptosphaeria_maculans'=>'',
	'leucoagaricus_sp_symc_cos_gca_001563735'=>'',
	'leucosporidium_creatinivorum_gca_002105055'=>'',
	'lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665'=>'',
	'lingula_anatina'=>'',
	'lipomyces_starkeyi_nrrl_y_11557_gca_001661325'=>'',
	'loa_loa'=>'',
	'lobosporangium_transversale_gca_002105155'=>'',
	'lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685'=>'',
	'lomentospora_prolificans_gca_002276285'=>'',
	'lonchura_striata_domestica'=>'',
	'lotharella_oceanica'=>'',
	'lottia_gigantea'=>'',
	'loxodonta_africana'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Lafri3/loxAfr3-tRNAs.fa',
	'lucilia_cuprina'=>'',
	'lupinus_angustifolius'=>'',
	'macaca_fascicularis'=>'',
	'macaca_mulatta'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmula3/rheMac3-tRNAs.fa',
	'macaca_nemestrina'=>'',
	'macrophomina_phaseolina_ms6_gca_000302655'=>'',
	'madurella_mycetomatis_gca_001275765'=>'',
	'magnaporthe_oryzae'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Magn_oryz_70_15/magnOryz_70_15-tRNAs.fa',
	'magnaporthe_poae'=>'',
	'malassezia_globosa_cbs_7966_gca_000181695'=>'',
	'malassezia_pachydermatis_gca_001278385'=>'',
	'malassezia_sympodialis_atcc_42132_gca_000349305'=>'',
	'malassezia_sympodialis_atcc_42132_gca_900149145'=>'',
	'malassezia_vespertilionis_gca_002818225'=>'',
	'manacus_vitellinus'=>'',
	'mandrillus_leucophaeus'=>'',
	'manihot_esculenta'=>'',
	'marmota_marmota_marmota'=>'',
	'marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775'=>'',
	'mastacembelus_armatus'=>'',
	'mayetiola_destructor'=>'',
	'maylandia_zebra'=>'',
	'medicago_truncatula'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Mtrun/mt3-tRNAs.fa',
	'megaselia_scalaris'=>'',
	'meira_miltonrushii_gca_003144205'=>'',
	'melampsora_laricipopulina'=>'',
	'meleagris_gallopavo'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mgall1/melGal1-tRNAs.fa',
	'meliniomyces_bicolor_e_gca_002865645'=>'',
	'meliniomyces_variabilis_f_gca_002865655'=>'',
	'melitaea_cinxia'=>'',
	'melopsittacus_undulatus'=>'',
	'meriones_unguiculatus'=>'',
	'mesocricetus_auratus'=>'',
	'metarhizium_acridum_cqma_102_gca_000187405'=>'',
	'metarhizium_album_arsef_1941_gca_000804445'=>'',
	'metarhizium_anisopliae_arsef_23_gca_000187425'=>'',
	'metarhizium_anisopliae_brip_53293_gca_000426965'=>'',
	'metarhizium_anisopliae_gca_000739145'=>'',
	'metarhizium_majus_arsef_297_gca_000814945'=>'',
	'metarhizium_robertsii_gca_000591435'=>'',
	'metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035'=>'',
	'meyerozyma_guilliermondii_atcc_6260_gca_000149425'=>'',
	'microbotryum_violaceum'=>'',
	'microcebus_murinus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmuri1/micMur1-tRNAs.fa',
	'microdochium_bolleyi_gca_001566295'=>'',
	'microsporum_canis_cbs_113480_gca_000151145'=>'',
	'microtus_ochrogaster'=>'',
	'millerozyma_farinosa_cbs_7064_gca_000315895'=>'',
	'mitosporidium_daphniae_gca_000760515'=>'',
	'mixia_osmundae_iam_14324_gca_000241205'=>'',
	'mnemiopsis_leidyi'=>'',
	'moelleriella_libera_rcef_2490_gca_001636675'=>'',
	'moesziomyces_antarcticus_gca_000747765'=>'',
	'moesziomyces_antarcticus_t_34_gca_000334475'=>'',
	'mola_mola'=>'',
	'moniliophthora_perniciosa_fa553_gca_000183025'=>'',
	'moniliophthora_roreri_gca_001466705'=>'',
	'moniliophthora_roreri_mca_2997_gca_000488995'=>'',
	'monodelphis_domestica'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mdome5/monDom5-tRNAs.fa',
	'monopterus_albus'=>'',
	'mortierella_elongata_ag_77_gca_001651415'=>'',
	'mortierella_verticillata_nrrl_6337_gca_000739165'=>'',
	'mucor_ambiguus_gca_000950595'=>'',
	'mucor_circinelloides_f_circinelloides_1006phl_gca_000401635'=>'',
	'mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945'=>'',
	'mus_caroli'=>'',
	'mus_musculus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_129s1svimj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_aj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_akrj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_balbcj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_c3hhej'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_c57bl6nj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_casteij'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_cbaj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_dba2j'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_fvbnj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_lpj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_nodshiltj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_nzohlltj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_pwkphj'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_musculus_wsbeij'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mmusc10/mm10-tRNAs.fa',
	'mus_pahari'=>'',
	'mus_spicilegus'=>'',
	'mus_spretus'=>'',
	'musa_acuminata'=>'',
	'mustela_putorius_furo'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mputo1/musFur1-tRNAs.fa',
	'mycosphaerella_eumusae_gca_001578235'=>'',
	'myotis_lucifugus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Mluci2/myoLuc2-tRNAs.fa',
	'nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315'=>'',
	'naegleria_gruberi'=>'',
	'naematelia_encephala_gca_002105065'=>'',
	'nannizzia_gypsea_cbs_118893_gca_000150975'=>'',
	'nannochloropsis_gaditana'=>'',
	'nannospalax_galili'=>'',
	'nasonia_vitripennis'=>'',
	'naumovozyma_castellii_cbs_4309_gca_000237345'=>'',
	'naumovozyma_dairenensis_cbs_421_gca_000227115'=>'',
	'nematocida_parisii_ertm1_gca_000250985'=>'',
	'nematocida_parisii_ertm3_gca_000190615'=>'',
	'nematocida_sp_1_ertm2_gca_000250695'=>'',
	'nematocida_sp_1_ertm6_gca_000738915'=>'',
	'nematocida_sp_ertm5_gca_001642415'=>'',
	'nematostella_vectensis'=>'',
	'neocallimastix_californiae_gca_002104975'=>'',
	'neofusicoccum_parvum_ucrnp2_gca_000385595'=>'',
	'neolamprologus_brichardi'=>'',
	'neolecta_irregularis_dah_3_gca_001929475'=>'',
	'neolentinus_lepideus_hhb14362_ss_1_gca_001632425'=>'',
	'neonectria_ditissima_gca_001306435'=>'',
	'neosartorya_fischeri'=>'',
	'neovison_vison'=>'',
	'neurospora_crassa'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Neur_cras_OR74A/neurCras_OR74A-tRNAs.fa',
	'neurospora_crassa_gca_000786625'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Neur_cras_OR74A/neurCras_OR74A-tRNAs.fa',
	'neurospora_tetrasperma_fgsc_2509_gca_000213195'=>'',
	'nicotiana_attenuata'=>'',
	'nomascus_leucogenys'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Nleuc3/nomLeu3-tRNAs.fa',
	'nosema_apis_brl_01_gca_000447185'=>'',
	'nosema_bombycis_cq1_gca_000383075'=>'',
	'nosema_ceranae_brl01_gca_000182985'=>'',
	'nosema_ceranae_gca_000988165'=>'',
	'notamacropus_eugenii'=>'',
	'notechis_scutatus'=>'',
	'nothoprocta_perdicaria'=>'',
	'numida_meleagris'=>'',
	'ochotona_princeps'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Oprin3/ochPri3-tRNAs.fa',
	'octodon_degus'=>'',
	'octopus_bimaculoides'=>'',
	'ogataea_parapolymorpha_dl_1_gca_000187245'=>'',
	'ogataea_polymorpha_gca_001664045'=>'',
	'oidiodendron_maius_zn_gca_000827325'=>'',
	'onchocerca_volvulus'=>'',
	'ophiocordyceps_sinensis_co18_gca_000448365'=>'',
	'ophiostoma_piceae_uamh_11346_gca_000410735'=>'',
	'ordospora_colligata_oc4_gca_000803265'=>'',
	'oreochromis_niloticus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Onilo2/oreNil2-tRNAs.fa',
	'ornithorhynchus_anatinus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Oanat1/ornAna1-tRNAs.fa',
	'oryctolagus_cuniculus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Ocuni2/oryCun2-tRNAs.fa',
	'oryza_barthii'=>'',
	'oryza_brachyantha'=>'',
	'oryza_glaberrima'=>'',
	'oryza_glumipatula'=>'',
	'oryza_indica'=>'',
	'oryza_longistaminata'=>'',
	'oryza_meridionalis'=>'',
	'oryza_nivara'=>'',
	'oryza_punctata'=>'',
	'oryza_rufipogon'=>'',
	'oryza_sativa'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Osati/Osati-tRNAs.fa',
	'oryzias_latipes'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Olati2/',
	'oryzias_latipes_hni'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Olati2/',
	'oryzias_latipes_hsok'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Olati2/',
	'oryzias_melastigma'=>'',
	'ostreococcus_lucimarinus'=>'',
	'otolemur_garnettii'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Ogarn3/otoGar3-tRNAs.fa',
	'ovis_aries'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Oarie1/oviAri1-tRNAs.fa',
	'pachysolen_tannophilus_nrrl_y_2460_gca_001661245'=>'',
	'pan_paniscus'=>'',
	'pan_troglodytes'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Ptrog4/panTro4-tRNAs.fa',
	'panthera_pardus'=>'',
	'panthera_tigris_altaica'=>'',
	'papio_anubis'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Panub2/papAnu2-tRNAs.fa',
	'paracoccidioides_brasiliensis_gca_001713645'=>'',
	'paracoccidioides_brasiliensis_gca_001713695'=>'',
	'paracoccidioides_brasiliensis_pb03_gca_000150475'=>'',
	'paracoccidioides_brasiliensis_pb18_gca_000150735'=>'',
	'paracoccidioides_sp_lutzii_pb01_gca_000150705'=>'',
	'paramecium_tetraurelia'=>'',
	'paramicrosporidium_saccamoebae_gca_002794465'=>'',
	'paramormyrops_kingsleyae'=>'',
	'paraphaeosphaeria_sporulosa_gca_001642045'=>'',
	'parus_major'=>'',
	'paulinella_chromatophora'=>'',
	'paxillus_involutus_atcc_200175_gca_000827475'=>'',
	'paxillus_rubicundulus_ve08_2h10_gca_000827395'=>'',
	'pediculus_humanus'=>'',
	'pelodiscus_sinensis'=>'',
	'penicilliopsis_zonata_cbs_506_65_gca_001890105'=>'',
	'penicillium_brasilianum_gca_001048715'=>'',
	'penicillium_brasilianum_gca_002016555'=>'',
	'penicillium_camemberti_fm_013_gca_000513335'=>'',
	'penicillium_chrysogenum_gca_000710275'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Peni_chry_P2niaD18/peniChry_P2NIAD18-tRNAs.fa',
	'penicillium_digitatum_pd1_gca_000315645'=>'',
	'penicillium_digitatum_phi26_gca_000315665'=>'',
	'penicillium_expansum_gca_000769745'=>'',
	'penicillium_freii_gca_001513925'=>'',
	'penicillium_griseofulvum_gca_001561935'=>'',
	'penicillium_italicum_gca_000769765'=>'',
	'penicillium_nordicum_gca_001278595'=>'',
	'penicillium_oxalicum_114_2_gca_000346795'=>'',
	'penicillium_roqueforti_fm164_gca_000513255'=>'',
	'penicillium_rubens_wisconsin_54_1255_gca_000226395'=>'',
	'penicillium_subrubescens_gca_001908125'=>'',
	'peniophora_sp_cont_gca_001632445'=>'',
	'periconia_macrospinosa_gca_003073855'=>'',
	'periophthalmus_magnuspinnatus'=>'',
	'perkinsus_marinus_atcc_50983'=>'',
	'peromyscus_maniculatus_bairdii'=>'',
	'pestalotiopsis_fici_w106_1_gca_000516985'=>'',
	'petromyzon_marinus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Pmari2/petMar2-tRNAs.fa',
	'pezoloma_ericae_gca_002865625'=>'',
	'phaeoacremonium_minimum_ucrpa7_gca_000392275'=>'',
	'phaeodactylum_tricornutum'=>'',
	'phaeomoniella_chlamydospora_gca_001006345'=>'',
	'phaeosphaeria_nodorum'=>'',
	'phanerochaete_carnosa_hhb_10118_sp_gca_000300595'=>'',
	'phascolarctos_cinereus'=>'',
	'phaseolus_vulgaris'=>'',
	'phialocephala_scopiformis_gca_001500285'=>'',
	'phialocephala_subalpina_gca_900073065'=>'',
	'phialophora_americana_gca_000835435'=>'',
	'phialophora_attae_gca_001299255'=>'',
	'phlebia_centrifuga_gca_001913855'=>'',
	'phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265'=>'',
	'physcomitrella_patens'=>'',
	'phytophthora_infestans'=>'',
	'phytophthora_kernoviae'=>'',
	'phytophthora_lateralis'=>'',
	'phytophthora_parasitica'=>'',
	'phytophthora_parasitica_cj01a1'=>'',
	'phytophthora_parasitica_gca_000509465'=>'',
	'phytophthora_parasitica_gca_000509485'=>'',
	'phytophthora_parasitica_gca_000509505'=>'',
	'phytophthora_parasitica_gca_000509525'=>'',
	'phytophthora_parasitica_inra_310'=>'',
	'phytophthora_parasitica_p10297'=>'',
	'phytophthora_parasitica_p1976'=>'',
	'phytophthora_ramorum'=>'',
	'phytophthora_sojae'=>'',
	'pichia_kudriavzevii_gca_000764455'=>'',
	'pichia_kudriavzevii_gca_002166775'=>'',
	'pichia_kudriavzevii_gca_003054445'=>'',
	'pichia_membranifaciens_gca_001950575'=>'',
	'pichia_membranifaciens_nrrl_y_2026_gca_001661235'=>'',
	'piliocolobus_tephrosceles'=>'',
	'piloderma_croceum_f_1598_gca_000827315'=>'',
	'piromyces_finnis_gca_002104945'=>'',
	'pisolithus_microcarpus_441_gca_000827275'=>'',
	'pisolithus_tinctorius_marx_270_gca_000827335'=>'',
	'plasmodium_berghei'=>'',
	'plasmodium_berghei_gca_000005395'=>'',
	'plasmodium_berghei_gca_900044335'=>'',
	'plasmodium_chabaudi'=>'',
	'plasmodium_chabaudi_chabaudi_gca_000003075'=>'',
	'plasmodium_coatneyi'=>'',
	'plasmodium_cynomolgi_strain_b'=>'',
	'plasmodium_falciparum'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_7g8'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_camp_malaysia'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_dd2'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_fch_4'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_hb3'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_igh_cr14'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_malips096_e11'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_nf135_5_c10'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_nf54'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_palo_alto_uganda'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_raj116'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_santa_lucia'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_tanzania_2000708_'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_ugt5_1'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_falciparum_vietnam_oak_knoll_fvo_'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Pfalc/plasFalc1-tRNAs.fa',
	'plasmodium_fragile'=>'',
	'plasmodium_gaboni'=>'',
	'plasmodium_inui_san_antonio_1'=>'',
	'plasmodium_knowlesi'=>'',
	'plasmodium_knowlesi_strain_h_gca_900004885'=>'',
	'plasmodium_malariae'=>'',
	'plasmodium_ovale_curtisi'=>'',
	'plasmodium_reichenowi'=>'',
	'plasmodium_reichenowi_gca_001601855'=>'',
	'plasmodium_vinckei_petteri'=>'',
	'plasmodium_vinckei_vinckei'=>'',
	'plasmodium_vivax'=>'',
	'plasmodium_vivax_brazil_i'=>'',
	'plasmodium_vivax_india_vii'=>'',
	'plasmodium_vivax_mauritania_i'=>'',
	'plasmodium_vivax_north_korean'=>'',
	'plasmodium_yoelii'=>'',
	'plasmodium_yoelii_17x'=>'',
	'plasmodium_yoelii_gca_900002395'=>'',
	'plasmodium_yoelii_yoelii'=>'',
	'plasmopara_halstedii'=>'',
	'pleurotus_ostreatus_pc15_gca_000697685'=>'',
	'plicaturopsis_crispa_fd_325_ss_3_gca_000827205'=>'',
	'pneumocystis_carinii_b80_gca_001477545'=>'',
	'pneumocystis_jirovecii_ru7_gca_001477535'=>'',
	'pneumocystis_murina_b123_gca_000349005'=>'',
	'pochonia_chlamydosporia_170_gca_001653235'=>'',
	'podospora_anserina_s_mat__gca_000226545'=>'',
	'poecilia_formosa'=>'',
	'poecilia_latipinna'=>'',
	'poecilia_mexicana'=>'',
	'poecilia_reticulata'=>'',
	'pogona_vitticeps'=>'',
	'polysphondylium_pallidum_pn500'=>'',
	'pongo_abelii'=>'',
	'populus_trichocarpa'=>'',
	'postia_placenta_mad_698_r_gca_000006255'=>'',
	'postia_placenta_mad_698_r_sb12_gca_002117355'=>'',
	'procavia_capensis'=>'',
	'prolemur_simus'=>'',
	'propithecus_coquereli'=>'',
	'protomyces_lactucaedebilis_gca_002105105'=>'',
	'prunus_persica'=>'',
	'pseudocercospora_fijiensis_cirad86_gca_000340215'=>'',
	'pseudocercospora_musae_gca_001578225'=>'',
	'pseudogymnoascus_destructans_20631_21_gca_000184105'=>'',
	'pseudogymnoascus_destructans_gca_001641265'=>'',
	'pseudogymnoascus_sp_03vt05_gca_001662645'=>'',
	'pseudogymnoascus_sp_05ny08_gca_001662605'=>'',
	'pseudogymnoascus_sp_23342_1_i1_gca_001662575'=>'',
	'pseudogymnoascus_sp_24mn13_gca_001662595'=>'',
	'pseudogymnoascus_sp_vkm_f_103_gca_000750895'=>'',
	'pseudogymnoascus_sp_vkm_f_3557_gca_000750665'=>'',
	'pseudogymnoascus_sp_vkm_f_3775_gca_000750715'=>'',
	'pseudogymnoascus_sp_vkm_f_3808_gca_000750675'=>'',
	'pseudogymnoascus_sp_vkm_f_4246_gca_000750735'=>'',
	'pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745'=>'',
	'pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755'=>'',
	'pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805'=>'',
	'pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815'=>'',
	'pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875'=>'',
	'pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925'=>'',
	'pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935'=>'',
	'pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995'=>'',
	'pseudogymnoascus_sp_wsf_3629_gca_001662585'=>'',
	'pseudogymnoascus_verrucosus_gca_001662655'=>'',
	'pseudoloma_neurophilia_gca_001432165'=>'',
	'pseudomassariella_vexata_gca_002105095'=>'',
	'pseudomicrostroma_glucosiphilum_gca_003144135'=>'',
	'pseudozyma_hubeiensis_sy62_gca_000403515'=>'',
	'pteropus_vampyrus'=>'',
	'puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275'=>'',
	'puccinia_graminis'=>'',
	'puccinia_graminisug99'=>'',
	'puccinia_sorghi_gca_001263375'=>'',
	'puccinia_striiformis'=>'',
	'puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645'=>'',
	'puccinia_triticina'=>'',
	'punctularia_strigosozonata_hhb_11173_ss5_gca_000264995'=>'',
	'pundamilia_nyererei'=>'',
	'purpureocillium_lilacinum_gca_001653205'=>'',
	'purpureocillium_lilacinum_gca_001653265'=>'',
	'pygocentrus_nattereri'=>'',
	'pyrenochaeta_sp_ds3say3a_gca_001644535'=>'',
	'pyrenophora_teres'=>'',
	'pyrenophora_tritici_repentis_gca_003171545'=>'',
	'pyrenophora_tritici_repentis_gca_003231325'=>'',
	'pyrenophora_tritici_repentis_gca_003231365'=>'',
	'pyrenophora_tritici_repentis_gca_003231425'=>'',
	'pyrenophora_triticirepentis'=>'',
	'pythium_aphanidermatum'=>'',
	'pythium_arrhenomanes'=>'',
	'pythium_irregulare'=>'',
	'pythium_iwayamai'=>'',
	'pythium_ultimum'=>'',
	'pythium_vexans'=>'',
	'rachicladosporium_antarcticum_gca_002077065'=>'',
	'rachicladosporium_sp_ccfee_5018_gca_002077045'=>'',
	'ramularia_collo_cygni_gca_900074925'=>'',
	'rasamsonia_emersonii_cbs_393_64_gca_000968595'=>'',
	'rattus_norvegicus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Rnorv5/rn5-tRNAs.fa',
	'reticulomyxa_filosa'=>'',
	'rhinocladiella_mackenziei_cbs_650_93_gca_000835555'=>'',
	'rhinopithecus_bieti'=>'',
	'rhinopithecus_roxellana'=>'',
	'rhizoclosmatium_globosum_gca_002104985'=>'',
	'rhizoctonia_solani_123e_gca_000715385'=>'',
	'rhizoctonia_solani_ag_1_ia_gca_000334115'=>'',
	'rhizoctonia_solani_ag_1_ib_gca_000350255'=>'',
	'rhizoctonia_solani_ag_8_wac10335_gca_000695385'=>'',
	'rhizoctonia_solani_gca_001286725'=>'',
	'rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155'=>'',
	'rhizophagus_irregularis_daom_181602_gca_000439145'=>'',
	'rhizophagus_irregularis_daom_197198w_gca_000597685'=>'',
	'rhizophagus_irregularis_gca_001593125'=>'',
	'rhizophagus_irregularis_gca_001593145'=>'',
	'rhizophagus_irregularis_gca_001593155'=>'',
	'rhizophagus_irregularis_gca_001593205'=>'',
	'rhizopogon_vinicolor_am_or11_026_gca_001658105'=>'',
	'rhizopus_delemar_ra_99_880_gca_000149305'=>'',
	'rhizopus_microsporus_atcc_52813_gca_002708625'=>'',
	'rhizopus_microsporus_gca_000825725'=>'',
	'rhizopus_microsporus_gca_002083735'=>'',
	'rhizopus_microsporus_var_microsporus_gca_002083745'=>'',
	'rhodnius_prolixus'=>'',
	'rhodotorula_graminis_wp1_gca_001329695'=>'',
	'rhodotorula_sp_jg_1b_gca_001541205'=>'',
	'rhodotorula_taiwanensis_gca_002922495'=>'',
	'rhodotorula_toruloides_atcc_204091_gca_000222205'=>'',
	'rhodotorula_toruloides_np11_gca_000320785'=>'',
	'rhynchosporium_agropyri_gca_900074905'=>'',
	'rhynchosporium_commune_gca_900074885'=>'',
	'rhynchosporium_secalis_gca_900074895'=>'',
	'rosellinia_necatrix_gca_001445595'=>'',
	'rozella_allomycis_csf55_gca_000442015'=>'',
	'saccharomyces_arboricola_h_6_gca_000292725'=>'',
	'saccharomyces_cerevisiae'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_ec1118_gca_000218975'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_fostersb_gca_000190255'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_fosterso_gca_000326005'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_gca_001634645'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_gca_002804325'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_kyokai_no_7_gca_000260735'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_lalvin_qa23_gca_000325965'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_p283_gca_000568295'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_p301_gca_000568055'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_r103_gca_000568365'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_rm11_1a_gca_000149365'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_vin13_gca_000190215'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_vl3_gca_000190235'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1078_gca_000975645'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1083_gca_000975675'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1129_gca_000975705'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1133_gca_000976695'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1190_gca_000976725'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1199_gca_000976755'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1202_gca_000976785'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1208_gca_000976815'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1242_gca_000976845'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1244_gca_000976875'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1248_gca_000976905'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1250_gca_000976935'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1252_gca_000976965'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1273_gca_000976995'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1304_gca_000977025'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1307_gca_000977055'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1311_gca_000977085'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1326_gca_000977115'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1332_gca_000977145'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1336_gca_000977175'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1338_gca_000977205'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1341_gca_000977235'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1342_gca_000977265'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1355_gca_000977295'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1356_gca_000977325'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1381_gca_000977355'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1383_gca_000977385'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1385_gca_000977415'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1386_gca_000977445'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1387_gca_000977475'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1388_gca_000977505'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1389_gca_000977535'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1399_gca_000977565'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1400_gca_000977595'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1401_gca_000977625'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1402_gca_000977655'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1415_gca_000977685'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1417_gca_000977715'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1418_gca_000977745'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1419_gca_000977775'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1433_gca_000977805'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1434_gca_000977835'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1439_gca_000977865'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1443_gca_000977895'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1444_gca_000977925'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1447_gca_000977955'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1450_gca_000977985'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1460_gca_000978015'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1463_gca_000978045'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1477_gca_000978075'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1478_gca_000978105'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1479_gca_000978135'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1526_gca_000978165'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1527_gca_000978195'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1549_gca_000978225'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1573_gca_000978255'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1574_gca_000978285'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1592_gca_000978315'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm1615_gca_000978345'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm189_gca_000975735'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm193_gca_000975765'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm195_gca_000975585'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm244_gca_000975615'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm248_gca_000975795'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm270_gca_000975825'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm271_gca_000975855'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm320_gca_000975885'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm326_gca_000975915'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm428_gca_000975945'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm450_gca_000975975'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm451_gca_000976005'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm453_gca_000976035'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm456_gca_000976065'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm470_gca_000976095'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm541_gca_000976125'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm554_gca_000976155'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm555_gca_000976185'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm627_gca_000976215'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm681_gca_000976245'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm682_gca_000976275'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm683_gca_000976305'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm689_gca_000976335'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm693_gca_000976365'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm789_gca_000181435'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm969_gca_000976395'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm972_gca_000976425'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm975_gca_000976455'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm978_gca_000976485'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm981_gca_000976515'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm984_gca_000976545'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm987_gca_000976575'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm990_gca_000976605'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm993_gca_000662435'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_cerevisiae_yjm996_gca_000976665'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Scere3/sacCer3-tRNAs.fa',
	'saccharomyces_eubayanus_gca_001298625'=>'',
	'saccharomyces_kudriavzevii_ifo_1802_gca_000167075'=>'',
	'saccharomycetaceae_sp_ashbya_aceri__gca_000412225'=>'',
	'saimiri_boliviensis_boliviensis'=>'',
	'saitoella_complicata_nrrl_y_17804_gca_000227095'=>'',
	'saitoella_complicata_nrrl_y_17804_gca_001661265'=>'',
	'salpingoeca_rosetta'=>'',
	'salvator_merianae'=>'',
	'saprolegnia_diclina_vs20'=>'',
	'saprolegnia_parasitica_cbs_223_65'=>'',
	'sarcophilus_harrisii'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Sharr1/sarHar1-tRNAs.fa',
	'sarcoptes_scabiei'=>'',
	'scedosporium_apiospermum_gca_000732125'=>'',
	'scheffersomyces_stipitis_cbs_6054_gca_000209165'=>'',
	'schistosoma_mansoni'=>'',
	'schizophyllum_commune_h4_8_gca_000143185'=>'',
	'schizopora_paradoxa_gca_001020605'=>'',
	'schizosaccharomyces_cryophilus'=>'',
	'schizosaccharomyces_japonicus'=>'',
	'schizosaccharomyces_octosporus'=>'',
	'schizosaccharomyces_pombe'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Schi_pomb_972h/schiPomb_972H-tRNAs.fa',
	'scleroderma_citrinum_foug_a_gca_000827425'=>'',
	'scleropages_formosus'=>'',
	'sclerotinia_borealis_f_4128_gca_000503235'=>'',
	'sclerotinia_sclerotiorum'=>'',
	'sclerotinia_sclerotiorum_1980_uf_70_gca_001857865'=>'',
	'scophthalmus_maximus'=>'',
	'selaginella_moellendorffii'=>'',
	'serendipita_indica_dsm_11827_gca_000313545'=>'',
	'serendipita_vermifera_maff_305830_gca_000827415'=>'',
	'serinus_canaria'=>'',
	'seriola_dumerili'=>'',
	'seriola_lalandi_dorsalis'=>'',
	'serpula_lacrymans_var_lacrymans_s7_3_gca_000218725'=>'',
	'setaria_italica'=>'',
	'sistotremastrum_niveocremeum_hhb9708_gca_001630475'=>'',
	'sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355'=>'',
	'solanum_lycopersicum'=>'',
	'solanum_tuberosum'=>'',
	'solenopsis_invicta'=>'',
	'sordaria_macrospora_k_hell_gca_000182805'=>'',
	'sorex_araneus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Saran2/sorAra2-tRNAs.fa',
	'sorghum_bicolor'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Sbico/Sbico-tRNAs.fa',
	'spathaspora_passalidarum_nrrl_y_27907_gca_000223485'=>'',
	'spermophilus_dauricus'=>'',
	'sphaerobolus_stellatus_ss14_gca_000827215'=>'',
	'sphaeroforma_arctica_jp610'=>'',
	'sphaerulina_musiva_so2202_gca_000320565'=>'',
	'sphenodon_punctatus'=>'',
	'spironucleus_salmonicida'=>'',
	'spizellomyces_punctatus_daom_br117_gca_000182565'=>'',
	'sporisorium_reilianum'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Spor_reil_SRZ2/sporReil_SRZ2-tRNAs.fa',
	'sporisorium_reilianum_f_sp_reilianum_gca_900162835'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Spor_reil_SRZ2/sporReil_SRZ2-tRNAs.fa',
	'sporothrix_brasiliensis_5110_gca_000820605'=>'',
	'sporothrix_insectorum_rcef_264_gca_001636815'=>'',
	'sporothrix_schenckii_1099_18_gca_000961545'=>'',
	'sporothrix_schenckii_atcc_58251_gca_000474925'=>'',
	'spraguea_lophii_42_110_gca_000430065'=>'',
	'stachybotrys_chartarum_ibt_40288_gca_000732765'=>'',
	'stachybotrys_chlorohalonata_ibt_40285_gca_000732775'=>'',
	'stagonospora_sp_src1lsm3a_gca_001644525'=>'',
	'stegastes_partitus'=>'',
	'stegodyphus_mimosarum'=>'',
	'stemphylium_lycopersici_gca_001191545'=>'',
	'stereum_hirsutum_fp_91666_ss1_gca_000264905'=>'',
	'strigamia_maritima'=>'',
	'strigomonas_culicis'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/bacteria/Cand_Kine_blastocrithidii_ex_Strigomonas_culicis/candKine_BLASTOCRITHIDII_EX_ST-tRNAs.fa',
	'strongylocentrotus_purpuratus'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Spurp/Spurp-tRNAs.fa',
	'strongyloides_ratti'=>'',
	'sugiyamaella_lignohabitans_gca_001640025'=>'',
	'suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415'=>'',
	'suillus_luteus_uh_slu_lm8_n1_gca_000827255'=>'',
	'sus_scrofa'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Sscro3/susScr3-tRNAs.fa',
	'syncephalastrum_racemosum_gca_002105135'=>'',
	'taeniopygia_guttata'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Tgutt2/taeGut2-tRNAs.fa',
	'takifugu_rubripes'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Trubr3/fr3-tRNAs.fa',
	'talaromyces_cellulolyticus_gca_000829775'=>'',
	'talaromyces_islandicus_gca_000985935'=>'',
	'talaromyces_marneffei_atcc_18224_gca_000001985'=>'',
	'talaromyces_marneffei_pm1_gca_000750115'=>'',
	'talaromyces_stipitatus_atcc_10500_gca_000003125'=>'',
	'teleopsis_dalmanni'=>'',
	'termitomyces_sp_j132_gca_001263195'=>'',
	'testicularia_cyperi_gca_003144125'=>'',
	'tetrahymena_thermophila'=>'',
	'tetranychus_urticae'=>'',
	'tetraodon_nigroviridis'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Tnigr2/tetNig2-tRNAs.fa',
	'tetrapisispora_blattae_cbs_6284_gca_000315915'=>'',
	'tetrapisispora_phaffii_cbs_4417_gca_000236905'=>'',
	'thalassiosira_oceanica'=>'',
	'thalassiosira_oceanica_ccmp1005'=>'',
	'thalassiosira_pseudonana'=>'',
	'thecamonas_trahens_atcc_50062'=>'',
	'theileria_annulata'=>'',
	'theileria_equi_strain_wa'=>'',
	'theileria_orientalis_strain_shintoku'=>'',
	'theileria_parva'=>'',
	'thelohanellus_kitauei'=>'',
	'theobroma_cacao'=>'',
	'thermothelomyces_thermophila_atcc_42464_gca_000226095'=>'',
	'theropithecus_gelada'=>'',
	'thielavia_terrestris_nrrl_8126_gca_000226115'=>'',
	'thielaviopsis_punctulata_gca_000968615'=>'',
	'tilletia_controversa_gca_001645045'=>'',
	'tilletia_walkeri_gca_001645055'=>'',
	'tilletiaria_anomala_ubc_951_gca_000711695'=>'',
	'tilletiopsis_washingtonensis_gca_003144115'=>'',
	'tolypocladium_ophioglossoides_cbs_100239_gca_001189435'=>'',
	'torrubiella_hemipterigena_gca_000825705'=>'',
	'tortispora_caseinolytica_nrrl_y_17796_gca_001661475'=>'',
	'torulaspora_delbrueckii_gca_000243375'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Toru_delb_CBS_1146/toruDelb_CBS_1146-tRNAs.fa',
	'toxoplasma_gondii'=>'',
	'toxoplasma_gondii_ari'=>'',
	'toxoplasma_gondii_fou'=>'',
	'toxoplasma_gondii_gab2_2007_gal_dom2'=>'',
	'toxoplasma_gondii_gt1'=>'',
	'toxoplasma_gondii_mas'=>'',
	'toxoplasma_gondii_p89'=>'',
	'toxoplasma_gondii_rub'=>'',
	'toxoplasma_gondii_tgcatprc2'=>'',
	'toxoplasma_gondii_vand'=>'',
	'toxoplasma_gondii_veg'=>'',
	'trachipleistophora_hominis_gca_000316135'=>'',
	'trametes_cinnabarina_gca_000765035'=>'',
	'trametes_coccinea_brfm310_gca_002092935'=>'',
	'trametes_pubescens_gca_001895945'=>'',
	'trametes_versicolor_fp_101664_ss1_gca_000271585'=>'',
	'tremella_mesenterica_dsm_1558_gca_000271645'=>'',
	'tribolium_castaneum'=>'',
	'trichoderma_asperellum_cbs_433_97_gca_003025105'=>'',
	'trichoderma_atroviride_imi_206040_gca_000171015'=>'',
	'trichoderma_citrinoviride_gca_003025115'=>'',
	'trichoderma_harzianum_cbs_226_95_gca_003025095'=>'',
	'trichoderma_harzianum_gca_000988865'=>'',
	'trichoderma_longibrachiatum_atcc_18648_gca_003025155'=>'',
	'trichoderma_reesei'=>'',
	'trichoderma_reesei_rut_c_30_gca_000513815'=>'',
	'trichoderma_virens'=>'',
	'trichomonas_vaginalis_g3'=>'',
	'trichophyton_benhamiae_cbs_112371_gca_000151125'=>'',
	'trichophyton_equinum_cbs_127_97_gca_000151175'=>'',
	'trichophyton_interdigitale_h6_gca_000616785'=>'',
	'trichophyton_interdigitale_mr816_gca_000622975'=>'',
	'trichophyton_mentagrophytes_gca_003118255'=>'',
	'trichophyton_rubrum_cbs_100081_gca_000616805'=>'',
	'trichophyton_rubrum_cbs_118892_gca_000151425'=>'',
	'trichophyton_rubrum_cbs_202_88_gca_000616985'=>'',
	'trichophyton_rubrum_cbs_288_86_gca_000616825'=>'',
	'trichophyton_rubrum_cbs_289_86_gca_000616845'=>'',
	'trichophyton_rubrum_cbs_735_88_gca_000616965'=>'',
	'trichophyton_rubrum_d6_gca_000622995'=>'',
	'trichophyton_rubrum_mr1448_gca_000616905'=>'',
	'trichophyton_rubrum_mr1459_gca_000616945'=>'',
	'trichophyton_rubrum_mr850_gca_000616765'=>'',
	'trichophyton_soudanense_cbs_452_61_gca_000616865'=>'',
	'trichophyton_tonsurans_cbs_112818_gca_000151455'=>'',
	'trichophyton_verrucosum_hki_0517_gca_000151505'=>'',
	'trichoplax_adhaerens'=>'',
	'trichosporon_asahii_var_asahii_cbs_2479_gca_000293215'=>'',
	'trichosporon_asahii_var_asahii_cbs_8904_gca_000299215'=>'',
	'trifolium_pratense'=>'',
	'triticum_aestivum'=>'',
	'triticum_dicoccoides'=>'',
	'triticum_urartu'=>'',
	'trypanosoma_brucei'=>'',
	'trypanosoma_cruzi'=>'',
	'trypanosoma_cruzi_dm28c'=>'',
	'tsuchiyaea_wingfieldii_cbs_7118_gca_001720155'=>'',
	'tuber_borchii_gca_003070745'=>'',
	'tuber_magnatum_gca_003182015'=>'',
	'tuber_melanosporum'=>'',
	'tulasnella_calospora_mut_4182_gca_000827465'=>'',
	'tupaia_belangeri'=>'',
	'tursiops_truncatus'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Ttrun2/turTru2-tRNAs.fa',
	'uncinocarpus_reesii_1704_gca_000003515'=>'',
	'urocitellus_parryii'=>'',
	'ursus_americanus'=>'',
	'ursus_maritimus'=>'',
	'ustilaginoidea_virens_gca_000687475'=>'',
	'ustilaginomycotina_sp_sa_807_gca_003144235'=>'',
	'ustilago_bromivora_gca_900080155'=>'',
	'ustilago_hordei_gca_000286035'=>'',
	'ustilago_maydis'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Usti_mayd_521/ustiMayd_521-tRNAs.fa',
	'valsa_mali_gca_000818155'=>'',
	'valsa_mali_var_pyri_gca_000813385'=>'',
	'vanderwaltozyma_polyspora_dsm_70294_gca_000150035'=>'',
	'vavraia_culicis_subsp_floridensis_gca_000192795'=>'',
	'verruconis_gallopava_gca_000836295'=>'',
	'verticillium_alfalfae_vams_102_gca_000150825'=>'',
	'verticillium_dahliae'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Vert_dahl_VdLs_17/vertDahl_VDLS_17-tRNAs.fa',
	'verticillium_dahliaejr2'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Vert_dahl_JR2/vertDahl_JR2-tRNAs.fa',
	'verticillium_longisporum_gca_001268145'=>'',
	'vicugna_pacos'=>'',
	'vigna_angularis'=>'',
	'vigna_radiata'=>'',
	'vitis_vinifera'=>'',
	'vitrella_brassicaformis_ccmp3155'=>'',
	'vittaforma_corneae_atcc_50505_gca_000231115'=>'',
	'vulpes_vulpes'=>'',
	'wallemia_ichthyophaga_exf_994_gca_000400465'=>'',
	'wallemia_mellicola_cbs_633_66_gca_000263375'=>'',
	'wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255'=>'',
	'wickerhamomyces_ciferrii_gca_000313485'=>'',
	'wolfiporia_cocos_md_104_ss10_gca_000344635'=>'',
	'xenopus_tropicalis'=>'http://gtrnadb.ucsc.edu/genomes/eukaryota/Xtrop3/xenTro3-tRNAs.fa',
	'xiphophorus_couchianus'=>'',
	'xiphophorus_maculatus'=>'',
	'xylona_heveae_tc161_gca_001619985'=>'',
	'yarrowia_lipolytica'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Yarr_lipo_CLIB122/yarrLipo_CLIB122-tRNAs.fa',
	'yarrowia_lipolytica_gca_001761485'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Yarr_lipo_CLIB122/yarrLipo_CLIB122-tRNAs.fa',
	'yarrowia_lipolytica_gca_900087985'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Yarr_lipo_CLIB122/yarrLipo_CLIB122-tRNAs.fa',
	'zea_mays'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Zmays5/zeaMay5-tRNAs.fa',
	'zonotrichia_albicollis'=>'',
	'zootermopsis_nevadensis'=>'',
	'zygosaccharomyces_bailii_clib_213_gca_000442885'=>'',
	'zygosaccharomyces_bailii_gca_900162805'=>'',
	'zygosaccharomyces_bailii_isa1307_gca_000530735'=>'',
	'zygosaccharomyces_parabailii_gca_001984395'=>'',
	'zygosaccharomyces_rouxii_gca_000026365'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Zygo_roux_CBS_732/zygoRoux_CBS_732-tRNAs.fa',
	'zygosaccharomyces_rouxii_gca_001972345'=>'http://gtrnadb.ucsc.edu/GtRNAdb2/genomes/eukaryota/Zygo_roux_CBS_732/zygoRoux_CBS_732-tRNAs.fa',
	'zymoseptoria_brevis_gca_000966595'=>'',
	'zymoseptoria_tritici'=>'',
	);### TRNA URLS END ###
			}

		if(length$gtRNAdb_urls{$species}>0)
			{
			if(!$silent){print"\nDownload tRNA data for $species{$species} from Genomic tRNA database...";}
			$status=getstore("$gtRNAdb_urls{$species}","UNITAS_refdump_$species/gtRNAdb");
			if(is_success($status))
				{
				if(!$silent){print" OK. Database date: $mday.$mon.$year (dd.mm.yyyy)";}
				$db_versions_info.="Genomic tRNA database version/date: ....... $mday.$mon.$year (dd.mm.yyyy)\n";
				}
			else
				{
				if(!$silent){print" FAILED! $status. Download data from smallRNAgroup-SERVER...";}
				$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/gtRNAdb/gtRNAdb_$species.fasta");
				$response = $useragent->request($request);
				$content = $response->content;
				open(my $out, '>', "UNITAS_refdump_$species/gtRNAdb");
				print $out "$content";
				close $out;
				$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/gtRNAdb/release');
				$response = $useragent->request($request);
				$version = $response->content;
				$db_versions_info.="Genomic tRNA database version/date: ....... $version (dd.mm.yyyy)\n";
				if(-e"UNITAS_refdump_$species/gtRNAdb")
					{
					if(!$silent){print" OK.";}
					}
				else
					{
					if(!$silent){print" FAILED! $status.\n\n";}
					}
				}
			if(!$silent){print"\nProcess data from gtRNAdb...";}
			open(GTRNADB,"UNITAS_refdump_$species/gtRNAdb")||die print"\nERROR! Unable to read UNITAS_refdump_$species/gtRNAdb.\n$!\n\n";
			%gtRNAdb=();
			$i=0;
			while(<GTRNADB>)
				{
				$_=~s/\s*$//;
				if($_=~/^>/)
					{
					$_=~s/tRNA[^ ]+//i;
					$id=$&;
					$i++;
					}
				else
					{
					$gtRNAdb{$id}.=$_;
					}
				}
			close GTRNADB;
			if(!$silent){print" OK ($i sequences loaded).";}
			}
		else
			{
			if(!$silent){print"\nNo tRNA data for $species{$species} at Genomic tRNA database -> SKIP.";}
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/gtRNAdb/release');
			$response = $useragent->request($request);
			$version = $response->content;
			$db_versions_info.="Genomic tRNA database version/date: ....... $version (dd.mm.yyyy)\n";
			}

		# download piRNA cluster sequences from piRNA cluster database (value may contain more than one URL, separated by whitespace)
		if(!$skip_piR)
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/URLs/piRclDB');
			$response = $useragent->request($request);
			$eval_URL="";
			$eval_URL = $response->content;
			if(length$eval_URL>0)
				{
				eval$eval_URL;
				}
			else
				{
				if(!$silent){print"\nFailed to load current piRNA cluster database URLs from smallRNAgroup-SERVER. Use internal list.";}
### PIRCLDB URLS START ###
%piRclDB_urls=
	(
	'_candida_arabinofermentans_nrrl_yb_2248_gca_001661425'=>'',
	'_candida_auris_gca_001189475'=>'',
	'_candida_auris_gca_002759435'=>'',
	'_candida_auris_gca_002775015'=>'',
	'_candida_auris_gca_003013715'=>'',
	'_candida_auris_gca_003014415'=>'',
	'_candida_duobushaemulonis_gca_002926085'=>'',
	'_candida_glabrata_gca_000002545'=>'',
	'_candida_glabrata_gca_002219185'=>'',
	'_candida_glabrata_gca_002219195'=>'',
	'_candida_haemulonis_gca_002926055'=>'',
	'_candida_pseudohaemulonis_gca_003013735'=>'',
	'_candida_tenuis_atcc_10573_gca_000223465'=>'',
	'absidia_glauca_gca_900079185'=>'',
	'absidia_repens_gca_002105175'=>'',
	'acanthochromis_polyacanthus'=>'',
	'acaromyces_ingoldii_gca_003144295'=>'',
	'acidomyces_richmondensis_bfw_gca_001592465'=>'',
	'acremonium_chrysogenum_atcc_11550_gca_000769265'=>'',
	'acyrthosiphon_pisum'=>'',
	'adineta_vaga'=>'',
	'aedes_aegypti'=>'',
	'aegilops_tauschii'=>'',
	'agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555'=>'',
	'ailuropoda_melanoleuca'=>'',
	'albugo_laibachii'=>'',
	'allomyces_macrogynus_atcc_38327_gca_000151295'=>'',
	'alternaria_alternata_gca_001642055'=>'',
	'amanita_muscaria_koide_bx008_gca_000827485'=>'',
	'amborella_trichopoda'=>'',
	'amphiamblys_sp_wsbs2006_gca_001875675'=>'',
	'amphilophus_citrinellus'=>'',
	'amphimedon_queenslandica'=>'',
	'amphiprion_ocellaris'=>'',
	'amphiprion_percula'=>'',
	'anabas_testudineus'=>'',
	'anaeromyces_robustus_gca_002104895'=>'',
	'anas_platyrhynchos_platyrhynchos'=>'',
	'angomonas_deanei'=>'',
	'anncaliia_algerae_pra109_gca_000385855'=>'',
	'anncaliia_algerae_pra339_gca_000385875'=>'',
	'anolis_carolinensis'=>'',
	'anopheles_darlingi'=>'',
	'anopheles_gambiae'=>'',
	'anoplophora_glabripennis'=>'',
	'anser_brachyrhynchus'=>'',
	'anthracocystis_flocculosa_pf_1_gca_000417875'=>'',
	'aotus_nancymaae'=>'',
	'aphanomyces_astaci'=>'',
	'aphanomyces_invadans'=>'',
	'apis_mellifera'=>'',
	'apteryx_haastii'=>'',
	'apteryx_owenii'=>'',
	'apteryx_rowi'=>'',
	'arabidopsis_halleri'=>'',
	'arabidopsis_lyrata'=>'',
	'arabidopsis_thaliana'=>'',
	'armillaria_gallica_gca_002307695'=>'',
	'armillaria_ostoyae_gca_900157425'=>'',
	'armillaria_solidipes_gca_002307675'=>'',
	'arthrobotrys_oligospora_atcc_24927_gca_000225545'=>'',
	'ascochyta_rabiei_gca_001630375'=>'',
	'ascoidea_rubescens_dsm_1968_gca_001661345'=>'',
	'ashbya_gossypii'=>'',
	'aspergillus_aculeatinus_cbs_121060_gca_003184765'=>'',
	'aspergillus_aculeatus_atcc_16872_gca_001890905'=>'',
	'aspergillus_brasiliensis_cbs_101740_gca_001889945'=>'',
	'aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695'=>'',
	'aspergillus_calidoustus_gca_001511075'=>'',
	'aspergillus_campestris_ibt_28561_gca_002847485'=>'',
	'aspergillus_candidus_gca_002847045'=>'',
	'aspergillus_carbonarius_item_5010_gca_001990825'=>'',
	'aspergillus_clavatus'=>'',
	'aspergillus_cristatus_gca_001717485'=>'',
	'aspergillus_ellipticus_cbs_707_79_gca_003184645'=>'',
	'aspergillus_eucalypticola_cbs_122712_gca_003184535'=>'',
	'aspergillus_flavus_af70_gca_000952835'=>'',
	'aspergillus_fumigatus'=>'',
	'aspergillus_fumigatus_gca_002234985'=>'',
	'aspergillus_fumigatusa1163'=>'',
	'aspergillus_glaucus_cbs_516_65_gca_001890805'=>'',
	'aspergillus_heteromorphus_cbs_117_55_gca_003184545'=>'',
	'aspergillus_indologenus_cbs_114_80_gca_003184685'=>'',
	'aspergillus_japonicus_cbs_114_51_gca_003184785'=>'',
	'aspergillus_lacticoffeatus_cbs_101883_gca_003184595'=>'',
	'aspergillus_lentulus_gca_001445615'=>'',
	'aspergillus_luchuensis_cbs_106_47_gca_001890685'=>'',
	'aspergillus_luchuensis_gca_001602395'=>'',
	'aspergillus_neoniger_cbs_115656_gca_003184625'=>'',
	'aspergillus_nidulans'=>'',
	'aspergillus_nidulans_fgsc_a4_gca_000149205'=>'',
	'aspergillus_niger'=>'',
	'aspergillus_niger_atcc_1015_gca_000230395'=>'',
	'aspergillus_niger_gca_001515345'=>'',
	'aspergillus_nomius_nrrl_13137_gca_001204775'=>'',
	'aspergillus_novofumigatus_ibt_16806_gca_002847465'=>'',
	'aspergillus_ochraceoroseus_ibt_24754_gca_002846915'=>'',
	'aspergillus_oryzae'=>'',
	'aspergillus_oryzae_3_042_gca_000269785'=>'',
	'aspergillus_parasiticus_su_1_gca_000956085'=>'',
	'aspergillus_piperis_cbs_112811_gca_003184755'=>'',
	'aspergillus_rambellii_gca_000986645'=>'',
	'aspergillus_ruber_cbs_135680_gca_000600275'=>'',
	'aspergillus_saccharolyticus_jop_1030_1_gca_003184585'=>'',
	'aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635'=>'',
	'aspergillus_sclerotioniger_cbs_115572_gca_003184525'=>'',
	'aspergillus_steynii_ibt_23096_gca_002849105'=>'',
	'aspergillus_sydowii_cbs_593_65_gca_001890705'=>'',
	'aspergillus_taichungensis_gca_002850765'=>'',
	'aspergillus_terreus'=>'',
	'aspergillus_thermomutatus_gca_002237265'=>'',
	'aspergillus_tubingensis_cbs_134_48_gca_001890745'=>'',
	'aspergillus_turcosus_gca_002234965'=>'',
	'aspergillus_turcosus_gca_002234975'=>'',
	'aspergillus_udagawae_gca_001078395'=>'',
	'aspergillus_ustus_gca_000812125'=>'',
	'aspergillus_uvarum_cbs_121591_gca_003184745'=>'',
	'aspergillus_vadensis_cbs_113365_gca_003184925'=>'',
	'aspergillus_versicolor_cbs_583_65_gca_001890125'=>'',
	'aspergillus_violaceofuscus_cbs_115571_gca_003184705'=>'',
	'aspergillus_wentii_dto_134e9_gca_001890725'=>'',
	'astatotilapia_calliptera'=>'',
	'astyanax_mexicanus'=>'',
	'atta_cephalotes'=>'',
	'aureobasidium_melanogenum_cbs_110374_gca_000721775'=>'',
	'aureobasidium_namibiae_cbs_147_97_gca_000721765'=>'',
	'aureobasidium_pullulans_exf_150_gca_000721785'=>'',
	'aureobasidium_subglaciale_exf_2481_gca_000721755'=>'',
	'babesia_bovis'=>'',
	'babesia_microti_strain_ri'=>'',
	'babjeviella_inositovora_nrrl_y_12698_gca_001661335'=>'',
	'basidiobolus_meristosporus_cbs_931_73_gca_002104905'=>'',
	'batrachochytrium_dendrobatidis_jam81_gca_000203795'=>'',
	'batrachochytrium_dendrobatidis_jel423_gca_000149865'=>'',
	'batrachochytrium_salamandrivorans_gca_002006685'=>'',
	'baudoinia_panamericana_uamh_10762_gca_000338955'=>'',
	'beauveria_bassiana'=>'',
	'beauveria_bassiana_arsef_2860_gca_000280675'=>'',
	'beauveria_bassiana_d1_5_gca_000770705'=>'',
	'beauveria_bassiana_gca_002871155'=>'',
	'belgica_antarctica'=>'',
	'beta_vulgaris'=>'',
	'bifiguratus_adelaidae_gca_002261195'=>'',
	'bigelowiella_natans'=>'',
	'bigelowiella_natans_gca_000002455'=>'',
	'bipolaris_maydis_atcc_48331_gca_000354255'=>'',
	'bipolaris_maydis_c5_gca_000338975'=>'',
	'bipolaris_sorokiniana_nd90pr_gca_000338995'=>'',
	'bipolaris_zeicola_26_r_13_gca_000523435'=>'',
	'bison_bison_bison'=>'',
	'blastomyces_dermatitidis_atcc_18188_gca_000151595'=>'',
	'blastomyces_dermatitidis_atcc_26199_gca_000166155'=>'',
	'blastomyces_dermatitidis_er_3_gca_000003525'=>'',
	'blastomyces_gilchristii_slh14081_gca_000003855'=>'',
	'blastomyces_percursus_gca_001883805'=>'',
	'blumeria_graminis'=>'',
	'blumeria_graminis_f_sp_tritici_96224_gca_000418435'=>'',
	'bodo_saltans'=>'',
	'bombus_impatiens'=>'',
	'bombus_terrestris'=>'',
	'bombyx_mori'=>'',
	'bos_mutus'=>'',
	'bos_taurus'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/bos_taurus/proTRAC_IVF_embryo_generic/clusters.fasta https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/bos_taurus/proTRAC_normal_testis_generic/clusters.fasta https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/bos_taurus/proTRAC_oocytes_generic/clusters.fasta',
	'botryobasidium_botryosum_fd_172_ss1_gca_000697705'=>'',
	'botrytis_cinerea'=>'',
	'botrytis_cinerea_bcdw1_gca_000349525'=>'',
	'botrytis_cinerea_t4_gca_000227075'=>'',
	'brachypodium_distachyon'=>'',
	'brassica_napus'=>'',
	'brassica_oleracea'=>'',
	'brassica_rapa'=>'',
	'brettanomyces_bruxellensis_awri1499_gca_000259595'=>'',
	'brugia_malayi'=>'',
	'byssochlamys_spectabilis_no_5_gca_000497085'=>'',
	'cadophora_sp_dse1049_gca_003073865'=>'',
	'caenorhabditis_brenneri'=>'',
	'caenorhabditis_briggsae'=>'',
	'caenorhabditis_elegans'=>'',
	'caenorhabditis_japonica'=>'',
	'caenorhabditis_remanei'=>'',
	'calidris_pugnax'=>'',
	'calidris_pygmaea'=>'',
	'callithrix_jacchus'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/callithrix_jacchus/proTRAC_normal_testis_generic/clusters.fasta',
	'calocera_cornea_hhb12733_gca_001632435'=>'',
	'calocera_viscosa_tufc12733_gca_001630345'=>'',
	'candida_albicans_12c_gca_000773845'=>'',
	'candida_albicans_19f_gca_000775445'=>'',
	'candida_albicans_ca6_gca_000784695'=>'',
	'candida_albicans_gc75_gca_000773735'=>'',
	'candida_albicans_l26_gca_000775455'=>'',
	'candida_albicans_p34048_gca_000775465'=>'',
	'candida_albicans_p37005_gca_000773745'=>'',
	'candida_albicans_p37037_gca_000773825'=>'',
	'candida_albicans_p37039_gca_000784515'=>'',
	'candida_albicans_p57055_gca_000775505'=>'',
	'candida_albicans_p57072_gca_000773805'=>'',
	'candida_albicans_p60002_gca_000784525'=>'',
	'candida_albicans_p75010_gca_000784575'=>'',
	'candida_albicans_p75016_gca_000784595'=>'',
	'candida_albicans_p75063_gca_000775525'=>'',
	'candida_albicans_p76055_gca_000784505'=>'',
	'candida_albicans_p76067_gca_000784495'=>'',
	'candida_albicans_p78042_gca_000784615'=>'',
	'candida_albicans_p78048_gca_000773725'=>'',
	'candida_albicans_p87_gca_000774085'=>'',
	'candida_albicans_p94015_gca_000773755'=>'',
	'candida_albicans_sc5314_gca_000784635'=>'',
	'candida_albicans_sc5314_gca_000784655'=>'',
	'candida_albicans_wo_1_gca_000149445'=>'',
	'candida_dubliniensis_cd36_gca_000026945'=>'',
	'candida_maltosa_xu316_gca_000344705'=>'',
	'candida_orthopsilosis_co_90_125_gca_000315875'=>'',
	'candida_parapsilosis_gca_000182765'=>'',
	'candida_tropicalis_mya_3404_gca_000006335'=>'',
	'canis_familiaris'=>'',
	'canis_lupus_dingo'=>'',
	'capitella_teleta'=>'',
	'capra_hircus'=>'',
	'capronia_coronata_cbs_617_96_gca_000585585'=>'',
	'capronia_epimyces_cbs_606_96_gca_000585565'=>'',
	'capsaspora_owczarzaki_atcc_30864'=>'',
	'carlito_syrichta'=>'',
	'castor_canadensis'=>'',
	'catenaria_anguillulae_pl171_gca_002102555'=>'',
	'cavia_aperea'=>'',
	'cavia_porcellus'=>'',
	'cebus_capucinus'=>'',
	'ceraceosorus_bombacis_gca_900000165'=>'',
	'ceraceosorus_guamensis_gca_003144195'=>'',
	'ceratocystis_platani_gca_000978885'=>'',
	'cercocebus_atys'=>'',
	'chaetomium_globosum_cbs_148_51_gca_000143365'=>'',
	'chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225'=>'',
	'chelonoidis_abingdonii'=>'',
	'chinchilla_lanigera'=>'',
	'chlamydomonas_reinhardtii'=>'',
	'chlorocebus_sabaeus'=>'',
	'choanephora_cucurbitarum_gca_001683725'=>'',
	'choloepus_hoffmanni'=>'',
	'chondrus_crispus'=>'',
	'chroomonas_mesostigmatica_ccmp1168'=>'',
	'chrysemys_picta_bellii'=>'',
	'ciona_intestinalis'=>'',
	'ciona_savignyi'=>'',
	'cladophialophora_bantiana_cbs_173_52_gca_000835475'=>'',
	'cladophialophora_carrionii_cbs_160_54_gca_000365165'=>'',
	'cladophialophora_carrionii_gca_001700775'=>'',
	'cladophialophora_immunda_gca_000835495'=>'',
	'cladophialophora_psammophila_cbs_110553_gca_000585535'=>'',
	'cladophialophora_yegresii_cbs_114405_gca_000585515'=>'',
	'claviceps_purpurea_20_1_gca_000347355'=>'',
	'clavispora_lusitaniae_atcc_42720_gca_000003835'=>'',
	'clavispora_lusitaniae_gca_001673695'=>'',
	'clohesyomyces_aquaticus_gca_002105025'=>'',
	'coccidioides_immitis_h538_4_gca_000149815'=>'',
	'coccidioides_immitis_rmscc_2394_gca_000149895'=>'',
	'coccidioides_immitis_rmscc_3703_gca_000150085'=>'',
	'coccidioides_immitis_rs_gca_000149335'=>'',
	'coccidioides_posadasii_c735_delta_sowgp_gca_000151335'=>'',
	'coccidioides_posadasii_rmscc_3488_gca_000150055'=>'',
	'coccidioides_posadasii_str_silveira_gca_000170175'=>'',
	'colletotrichum_fioriniae_pj7_gca_000582985'=>'',
	'colletotrichum_gloeosporioides'=>'',
	'colletotrichum_gloeosporioides_cg_14_gca_000446055'=>'',
	'colletotrichum_graminicola'=>'',
	'colletotrichum_higginsianum'=>'',
	'colletotrichum_incanum_gca_001625285'=>'',
	'colletotrichum_nymphaeae_sa_01_gca_001563115'=>'',
	'colletotrichum_orbiculare'=>'',
	'colletotrichum_salicis_gca_001563125'=>'',
	'colletotrichum_sublineola_gca_000696135'=>'',
	'colletotrichum_tofieldiae_gca_001625265'=>'',
	'colobus_angolensis_palliatus'=>'',
	'conidiobolus_coronatus_nrrl_28638_gca_001566745'=>'',
	'coniella_lustricola_gca_003019895'=>'',
	'coniochaeta_ligniaria_nrrl_30616_gca_001879275'=>'',
	'coniophora_puteana_rwd_64_598_ss2_gca_000271625'=>'',
	'coniosporium_apollinis_cbs_100218_gca_000281105'=>'',
	'coprinopsis_cinerea_okayama7_130_gca_000182895'=>'',
	'corchorus_capsularis'=>'',
	'cordyceps_brongniartii_rcef_3172_gca_001636735'=>'',
	'cordyceps_confragosa_rcef_1005_gca_001636795'=>'',
	'cordyceps_fumosorosea_arsef_2679_gca_001636725'=>'',
	'cordyceps_militaris_cm01_gca_000225605'=>'',
	'corynespora_cassiicola_philippines_gca_003016335'=>'',
	'coturnix_japonica'=>'',
	'crassostrea_gigas'=>'',
	'cricetulus_griseus_chok1gshd'=>'',
	'cricetulus_griseus_crigri'=>'',
	'cricetulus_griseus_picr'=>'',
	'crocodylus_porosus'=>'',
	'cryptococcus_amylolentus_cbs_6039_gca_001720205'=>'',
	'cryptococcus_amylolentus_cbs_6273_gca_001720235'=>'',
	'cryptococcus_depauperatus_cbs_7841_gca_001720195'=>'',
	'cryptococcus_depauperatus_cbs_7855_gca_001720245'=>'',
	'cryptococcus_gattii_ca1280_gca_000836335'=>'',
	'cryptococcus_gattii_ca1873_gca_000855695'=>'',
	'cryptococcus_gattii_e566_gca_000875815'=>'',
	'cryptococcus_gattii_ejb2_gca_000835745'=>'',
	'cryptococcus_gattii_nt_10_gca_000935105'=>'',
	'cryptococcus_gattii_ru294_gca_000836355'=>'',
	'cryptococcus_gattii_vgii_2001_935_1_gca_000835815'=>'',
	'cryptococcus_gattii_vgii_99_473_gca_000836455'=>'',
	'cryptococcus_gattii_vgii_ca1014_gca_000875795'=>'',
	'cryptococcus_gattii_vgii_cbs_10090_gca_000835765'=>'',
	'cryptococcus_gattii_vgii_la55_gca_000836315'=>'',
	'cryptococcus_gattii_vgii_mmrl2647_gca_000875855'=>'',
	'cryptococcus_gattii_vgii_r265_gca_000149475'=>'',
	'cryptococcus_gattii_vgii_ram5_gca_000836375'=>'',
	'cryptococcus_gattii_vgiv_ind107_gca_000835755'=>'',
	'cryptococcus_gattii_wm276_gca_000185945'=>'',
	'cryptococcus_neoformans'=>'',
	'cryptococcus_neoformans_var_grubii_h99_gca_000149245'=>'',
	'cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385'=>'',
	'cryptomonas_paramecium'=>'',
	'cryptosporidium_muris_rn66'=>'',
	'cryptosporidium_parvum'=>'',
	'cryptosporidium_parvum_iowa_ii'=>'',
	'cucumis_sativus'=>'',
	'culex_quinquefasciatus'=>'',
	'cutaneotrichosporon_oleaginosum_gca_001027345'=>'',
	'cyanistes_caeruleus'=>'',
	'cyberlindnera_fabianii_gca_003205855'=>'',
	'cyberlindnera_jadinii_gca_001245095'=>'',
	'cyberlindnera_jadinii_nrrl_y_1542_gca_001661405'=>'',
	'cyclospora_cayetanensis'=>'',
	'cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385'=>'',
	'cynoglossus_semilaevis'=>'',
	'cyphellophora_europaea_cbs_101466_gca_000365145'=>'',
	'cyprinodon_variegatus'=>'',
	'dacryopinax_primogenitus_gca_000292625'=>'',
	'dactylellina_haptotyla_cbs_200_50_gca_000441935'=>'',
	'daedalea_quercina_l_15889_gca_001632345'=>'',
	'daldinia_sp_ec12_gca_002120325'=>'',
	'danaus_plexippus'=>'',
	'danio_rerio'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/danio_rerio/proTRAC_normal_testis_generic/clusters.fasta https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/danio_rerio/proTRAC_normal_ovary_generic/clusters.fasta',
	'daphnia_magna'=>'',
	'daphnia_pulex'=>'',
	'dasypus_novemcinctus'=>'',
	'daucus_carota'=>'',
	'debaryomyces_fabryi_gca_001447935'=>'',
	'debaryomyces_hansenii_cbs767_gca_000006445'=>'',
	'dendroctonus_ponderosae'=>'',
	'diaporthe_ampelina_gca_001006365'=>'',
	'dichomitus_squalens_lyad_421_ss1_gca_000275845'=>'',
	'dictyostelium_discoideum'=>'',
	'dictyostelium_fasciculatum'=>'',
	'dictyostelium_lacteum'=>'',
	'dioscorea_rotundata'=>'',
	'diplodia_seriata_gca_001006355'=>'',
	'dipodomys_ordii'=>'',
	'dothistroma_septosporum'=>'',
	'drechmeria_coniospora_gca_001625195'=>'',
	'drechslerella_stenobrocha_248_gca_000525045'=>'',
	'dromaius_novaehollandiae'=>'',
	'drosophila_ananassae'=>'',
	'drosophila_erecta'=>'',
	'drosophila_grimshawi'=>'',
	'drosophila_melanogaster'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/drosophila_melanogaster/proTRAC_normal_ovary_generic/clusters_rDNAremoved.fasta',
	'drosophila_mojavensis'=>'',
	'drosophila_persimilis'=>'',
	'drosophila_pseudoobscura'=>'',
	'drosophila_sechellia'=>'',
	'drosophila_simulans'=>'',
	'drosophila_virilis'=>'',
	'drosophila_willistoni'=>'',
	'drosophila_yakuba'=>'',
	'echinops_telfairi'=>'',
	'edhazardia_aedis_usnm_41457_gca_000230595'=>'',
	'eimeria_brunetti'=>'',
	'eimeria_maxima'=>'',
	'eimeria_mitis'=>'',
	'eimeria_praecox'=>'',
	'eimeria_tenella'=>'',
	'eimeria_tenella_gca_000499545'=>'',
	'emergomyces_pasteuriana_ep9510_gca_001883825'=>'',
	'emiliania_huxleyi'=>'',
	'emmonsia_crescens_uamh_3008_gca_001008285'=>'',
	'emmonsia_parva_uamh_139_gca_001014755'=>'',
	'emmonsia_sp_cac_2015a_gca_001660665'=>'',
	'encephalitozoon_cuniculi_ecuniii_l_gca_001078035'=>'',
	'encephalitozoon_hellem_atcc_50504_gca_000277815'=>'',
	'encephalitozoon_intestinalis_atcc_50506_gca_000146465'=>'',
	'encephalitozoon_romaleae_sj_2008_gca_000280035'=>'',
	'endocarpon_pusillum_z07020_gca_000464535'=>'',
	'entamoeba_histolytica'=>'',
	'entamoeba_histolytica_gca_001662325'=>'',
	'entamoeba_histolytica_hm_3_imss'=>'',
	'entamoeba_histolytica_ku27'=>'',
	'entamoeba_nuttalli_p19'=>'',
	'enterocytozoon_bieneusi_h348_gca_000209485'=>'',
	'epicoccum_nigrum_gca_002116315'=>'',
	'eptatretus_burgeri'=>'',
	'equus_asinus_asinus'=>'',
	'equus_caballus'=>'',
	'eremothecium_cymbalariae_dbvpg_7215_gca_000235365'=>'',
	'eremothecium_gossypii_fdag1_gca_000968835'=>'',
	'erinaceus_europaeus'=>'',
	'erysiphe_necator_gca_000798715'=>'',
	'escovopsis_weberi_gca_001278495'=>'',
	'esox_lucius'=>'',
	'eutypa_lata_ucrel1_gca_000349385'=>'',
	'exidia_glandulosa_hhb12029_gca_001632375'=>'',
	'exophiala_aquamarina_cbs_119918_gca_000709125'=>'',
	'exophiala_dermatitidis_nih_ut8656_gca_000230625'=>'',
	'exophiala_mesophila_gca_000836275'=>'',
	'exophiala_oligosperma_gca_000835515'=>'',
	'exophiala_sideris_gca_000835395'=>'',
	'exophiala_spinifera_gca_000836115'=>'',
	'exophiala_xenobiotica_gca_000835505'=>'',
	'exserohilum_turcica_et28a_gca_000359705'=>'',
	'felis_catus'=>'',
	'fibroporia_radiculosa_gca_000313525'=>'',
	'fibularhizoctonia_sp_cbs_109695_gca_001630335'=>'',
	'ficedula_albicollis'=>'',
	'fistulina_hepatica_atcc_64428_gca_000934395'=>'',
	'fomitiporia_mediterranea_mf3_22_gca_000271605'=>'',
	'fomitopsis_pinicola_fp_58527_ss1_gca_000344655'=>'',
	'fonsecaea_erecta_gca_001651985'=>'',
	'fonsecaea_monophora_gca_001642475'=>'',
	'fonsecaea_multimorphosa_cbs_102226_gca_000836435'=>'',
	'fonsecaea_multimorphosa_gca_001646985'=>'',
	'fonsecaea_nubica_gca_001646965'=>'',
	'fonsecaea_pedrosoi_cbs_271_37_gca_000835455'=>'',
	'fonticula_alba_gca_000388065'=>'',
	'fukomys_damarensis'=>'',
	'fundulus_heteroclitus'=>'',
	'fusarium_culmorum'=>'',
	'fusarium_fujikuroi'=>'',
	'fusarium_fujikuroi_gca_001023035'=>'',
	'fusarium_fujikuroi_gca_001023045'=>'',
	'fusarium_fujikuroi_gca_001023065'=>'',
	'fusarium_fujikuroi_gca_900096505'=>'',
	'fusarium_fujikuroi_gca_900096685'=>'',
	'fusarium_fujikuroi_gca_900096705'=>'',
	'fusarium_graminearum'=>'',
	'fusarium_graminearum_gca_000240135'=>'',
	'fusarium_langsethiae_gca_001292635'=>'',
	'fusarium_mangiferae_gca_900044065'=>'',
	'fusarium_oxysporum'=>'',
	'fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215'=>'',
	'fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345'=>'',
	'fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365'=>'',
	'fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195'=>'',
	'fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955'=>'',
	'fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975'=>'',
	'fusarium_oxysporum_f_sp_melonis_26406_gca_000260495'=>'',
	'fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075'=>'',
	'fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155'=>'',
	'fusarium_oxysporum_f_sp_raphani_54005_gca_000260235'=>'',
	'fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175'=>'',
	'fusarium_oxysporum_fo47_gca_000271705'=>'',
	'fusarium_oxysporum_fosc_3_a_gca_000271745'=>'',
	'fusarium_oxysporum_gca_900096695'=>'',
	'fusarium_poae_gca_001675295'=>'',
	'fusarium_proliferatum_et1_gca_900067095'=>'',
	'fusarium_proliferatum_gca_900029915'=>'',
	'fusarium_pseudograminearum'=>'',
	'fusarium_solani'=>'',
	'fusarium_venenatum_gca_900007375'=>'',
	'fusarium_verticillioides'=>'',
	'gadus_morhua'=>'',
	'gaeumannomyces_graminis'=>'',
	'galdieria_sulphuraria'=>'',
	'galerina_marginata_cbs_339_88_gca_000697645'=>'',
	'gallus_gallus'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/gallus_gallus/proTRAC_SRR772069/clusters.fasta',
	'gambusia_affinis'=>'',
	'gasterosteus_aculeatus'=>'',
	'gelatoporia_subvermispora_b_gca_000320605'=>'',
	'giardia_intestinalis'=>'',
	'giardia_intestinalis_assemblage_b'=>'',
	'giardia_intestinalis_atcc_50581'=>'',
	'giardia_intestinalis_gca_000498715'=>'',
	'giardia_lamblia'=>'',
	'giardia_lamblia_p15'=>'',
	'glarea_lozoyensis_atcc_20868_gca_000409485'=>'',
	'gloeophyllum_trabeum_atcc_11539_gca_000344685'=>'',
	'glycine_max'=>'',
	'gonapodya_prolifera_jel478_gca_001574975'=>'',
	'gopherus_agassizii'=>'',
	'gorilla_gorilla'=>'',
	'gossypium_raimondii'=>'',
	'gregarina_niphandrodes'=>'',
	'grifola_frondosa_gca_001683735'=>'',
	'grosmannia_clavigera_kw1407_gca_000143105'=>'',
	'guillardia_theta'=>'',
	'guillardia_theta_gca_000002975'=>'',
	'gymnopus_luxurians_fd_317_m1_gca_000827265'=>'',
	'hammondia_hammondi'=>'',
	'hanseniaspora_guilliermondii_gca_900119595'=>'',
	'hanseniaspora_opuntiae_gca_001749795'=>'',
	'hanseniaspora_osmophila_gca_001747045'=>'',
	'hanseniaspora_uvarum_dsm_2768_gca_000968475'=>'',
	'hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025'=>'',
	'haplochromis_burtoni'=>'',
	'hebeloma_cylindrosporum_h7_gca_000827355'=>'',
	'helianthus_annuus'=>'',
	'heliconius_melpomene'=>'',
	'helobdella_robusta'=>'',
	'hesseltinella_vesiculosa_gca_002104935'=>'',
	'heterobasidion_irregulare_tc_32_1_gca_000320585'=>'',
	'heterocephalus_glaber_female'=>'',
	'heterocephalus_glaber_male'=>'',
	'hippocampus_comes'=>'',
	'hirsutella_minnesotensis_3608_gca_000956045'=>'',
	'histoplasma_capsulatum_g186ar_gca_000150115'=>'',
	'histoplasma_capsulatum_h143_gca_000151035'=>'',
	'histoplasma_capsulatum_h88_gca_000151005'=>'',
	'histoplasma_capsulatum_nam1_gca_000149585'=>'',
	'homo_sapiens'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/homo_sapiens/proTRAC_normal_testis_generic/clusters.fasta https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/homo_sapiens/proTRAC_SRR1755255/clusters.fasta',
	'hordeum_vulgare'=>'',
	'hortaea_werneckii_exf_2000_gca_002127715'=>'',
	'hyaloperonospora_arabidopsidis'=>'',
	'hydnomerulius_pinastri_md_312_gca_000827185'=>'',
	'hypholoma_sublateritium_fd_334_ss_4_gca_000827495'=>'',
	'hyphopichia_burtonii_nrrl_y_1933_gca_001661395'=>'',
	'hypoxylon_sp_ci_4a_gca_002120315'=>'',
	'hypoxylon_sp_co27_5_gca_002120305'=>'',
	'hypoxylon_sp_ec38_gca_002120335'=>'',
	'hypsizygus_marmoreus_gca_001605315'=>'',
	'ichthyophthirius_multifiliis'=>'',
	'ictalurus_punctatus'=>'',
	'ictidomys_tridecemlineatus'=>'',
	'ixodes_scapularis'=>'',
	'jaapia_argillacea_mucl_33604_gca_000697665'=>'',
	'jaculus_jaculus'=>'',
	'jaminaea_rosea_gca_003144245'=>'',
	'junco_hyemalis'=>'',
	'kalmanozyma_brasiliensis_ghg001_gca_000497045'=>'',
	'kazachstania_africana_cbs_2517_gca_000304475'=>'',
	'kazachstania_naganishii_cbs_8797_gca_000348985'=>'',
	'kluyveromyces_dobzhanskii_cbs_2104_gca_000820885'=>'',
	'kluyveromyces_lactis_gca_000002515'=>'',
	'kluyveromyces_marxianus_dmku3_1042_gca_001417885'=>'',
	'kluyveromyces_marxianus_gca_001417835'=>'',
	'kockovaella_imperatae_gca_002102565'=>'',
	'komagataella_pastoris'=>'',
	'komagataella_pastoris_gca_001708105'=>'',
	'komagataella_phaffii_cbs_7435_gca_000223565'=>'',
	'komagataella_phaffii_gca_001708085'=>'',
	'komagataella_phaffii_gs115_gca_001746955'=>'',
	'kryptolebias_marmoratus'=>'',
	'kuraishia_capsulata_cbs_1993_gca_000576695'=>'',
	'kwoniella_bestiolae_cbs_10118_gca_000512585'=>'',
	'kwoniella_dejecticola_cbs_10117_gca_000512565'=>'',
	'kwoniella_heveanensis_bcc8398_gca_000507405'=>'',
	'kwoniella_heveanensis_cbs_569_gca_000507425'=>'',
	'kwoniella_mangroviensis_cbs_10435_gca_000507885'=>'',
	'kwoniella_mangroviensis_cbs_8507_gca_000507465'=>'',
	'kwoniella_mangroviensis_cbs_8886_gca_000507485'=>'',
	'kwoniella_pini_cbs_10737_gca_000512605'=>'',
	'labrus_bergylta'=>'',
	'laccaria_amethystina_laam_08_1_gca_000827195'=>'',
	'laccaria_bicolor_s238n_h82_gca_000143565'=>'',
	'lachancea_dasiensis_cbs_10888_gca_900074725'=>'',
	'lachancea_fermentati_gca_900074765'=>'',
	'lachancea_lanzarotensis_gca_000938715'=>'',
	'lachancea_meyersii_cbs_8951_gca_900074715'=>'',
	'lachancea_mirantina_gca_900074745'=>'',
	'lachancea_nothofagi_cbs_11611_gca_900074755'=>'',
	'lachancea_quebecensis_gca_002900925'=>'',
	'lachancea_sp_cbs_6924_gca_900074735'=>'',
	'lachancea_thermotolerans_cbs_6340_gca_000142805'=>'',
	'laetiporus_sulphureus_93_53_gca_001632365'=>'',
	'latimeria_chalumnae'=>'',
	'leersia_perrieri'=>'',
	'leishmania_braziliensis_mhom_br_75_m2904'=>'',
	'leishmania_donovani_bpk282a1'=>'',
	'leishmania_infantum_jpcm5'=>'',
	'leishmania_major'=>'',
	'leishmania_mexicana_mhom_gt_2001_u1103'=>'',
	'leishmania_panamensis'=>'',
	'lepeophtheirus_salmonis'=>'',
	'lepidothrix_coronata'=>'',
	'lepisosteus_oculatus'=>'',
	'leptomonas_pyrrhocoris'=>'',
	'leptomonas_seymouri'=>'',
	'leptosphaeria_maculans'=>'',
	'leucoagaricus_sp_symc_cos_gca_001563735'=>'',
	'leucosporidium_creatinivorum_gca_002105055'=>'',
	'lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665'=>'',
	'lingula_anatina'=>'',
	'lipomyces_starkeyi_nrrl_y_11557_gca_001661325'=>'',
	'loa_loa'=>'',
	'lobosporangium_transversale_gca_002105155'=>'',
	'lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685'=>'',
	'lomentospora_prolificans_gca_002276285'=>'',
	'lonchura_striata_domestica'=>'',
	'lotharella_oceanica'=>'',
	'lottia_gigantea'=>'',
	'loxodonta_africana'=>'',
	'lucilia_cuprina'=>'',
	'lupinus_angustifolius'=>'',
	'macaca_fascicularis'=>'',
	'macaca_mulatta'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/macaca_mulatta/proTRAC_normal_testis_generic/clusters.fasta',
	'macaca_nemestrina'=>'',
	'macrophomina_phaseolina_ms6_gca_000302655'=>'',
	'madurella_mycetomatis_gca_001275765'=>'',
	'magnaporthe_oryzae'=>'',
	'magnaporthe_poae'=>'',
	'malassezia_globosa_cbs_7966_gca_000181695'=>'',
	'malassezia_pachydermatis_gca_001278385'=>'',
	'malassezia_sympodialis_atcc_42132_gca_000349305'=>'',
	'malassezia_sympodialis_atcc_42132_gca_900149145'=>'',
	'malassezia_vespertilionis_gca_002818225'=>'',
	'manacus_vitellinus'=>'',
	'mandrillus_leucophaeus'=>'',
	'manihot_esculenta'=>'',
	'marmota_marmota_marmota'=>'',
	'marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775'=>'',
	'mastacembelus_armatus'=>'',
	'mayetiola_destructor'=>'',
	'maylandia_zebra'=>'',
	'medicago_truncatula'=>'',
	'megaselia_scalaris'=>'',
	'meira_miltonrushii_gca_003144205'=>'',
	'melampsora_laricipopulina'=>'',
	'meleagris_gallopavo'=>'',
	'meliniomyces_bicolor_e_gca_002865645'=>'',
	'meliniomyces_variabilis_f_gca_002865655'=>'',
	'melitaea_cinxia'=>'',
	'melopsittacus_undulatus'=>'',
	'meriones_unguiculatus'=>'',
	'mesocricetus_auratus'=>'',
	'metarhizium_acridum_cqma_102_gca_000187405'=>'',
	'metarhizium_album_arsef_1941_gca_000804445'=>'',
	'metarhizium_anisopliae_arsef_23_gca_000187425'=>'',
	'metarhizium_anisopliae_brip_53293_gca_000426965'=>'',
	'metarhizium_anisopliae_gca_000739145'=>'',
	'metarhizium_majus_arsef_297_gca_000814945'=>'',
	'metarhizium_robertsii_gca_000591435'=>'',
	'metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035'=>'',
	'meyerozyma_guilliermondii_atcc_6260_gca_000149425'=>'',
	'microbotryum_violaceum'=>'',
	'microcebus_murinus'=>'',
	'microdochium_bolleyi_gca_001566295'=>'',
	'microsporum_canis_cbs_113480_gca_000151145'=>'',
	'microtus_ochrogaster'=>'',
	'millerozyma_farinosa_cbs_7064_gca_000315895'=>'',
	'mitosporidium_daphniae_gca_000760515'=>'',
	'mixia_osmundae_iam_14324_gca_000241205'=>'',
	'mnemiopsis_leidyi'=>'',
	'moelleriella_libera_rcef_2490_gca_001636675'=>'',
	'moesziomyces_antarcticus_gca_000747765'=>'',
	'moesziomyces_antarcticus_t_34_gca_000334475'=>'',
	'mola_mola'=>'',
	'moniliophthora_perniciosa_fa553_gca_000183025'=>'',
	'moniliophthora_roreri_gca_001466705'=>'',
	'moniliophthora_roreri_mca_2997_gca_000488995'=>'',
	'monodelphis_domestica'=>'',
	'monopterus_albus'=>'',
	'mortierella_elongata_ag_77_gca_001651415'=>'',
	'mortierella_verticillata_nrrl_6337_gca_000739165'=>'',
	'mucor_ambiguus_gca_000950595'=>'',
	'mucor_circinelloides_f_circinelloides_1006phl_gca_000401635'=>'',
	'mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945'=>'',
	'mus_caroli'=>'',
	'mus_musculus'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_129s1svimj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_aj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_akrj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_balbcj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_c3hhej'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_c57bl6nj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_casteij'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_cbaj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_dba2j'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_fvbnj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_lpj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_nodshiltj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_nzohlltj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_pwkphj'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_musculus_wsbeij'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/mus_musculus/proTRAC_normal_testis_generic/clusters.fasta',
	'mus_pahari'=>'',
	'mus_spicilegus'=>'',
	'mus_spretus'=>'',
	'musa_acuminata'=>'',
	'mustela_putorius_furo'=>'',
	'mycosphaerella_eumusae_gca_001578235'=>'',
	'myotis_lucifugus'=>'',
	'nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315'=>'',
	'naegleria_gruberi'=>'',
	'naematelia_encephala_gca_002105065'=>'',
	'nannizzia_gypsea_cbs_118893_gca_000150975'=>'',
	'nannochloropsis_gaditana'=>'',
	'nannospalax_galili'=>'',
	'nasonia_vitripennis'=>'',
	'naumovozyma_castellii_cbs_4309_gca_000237345'=>'',
	'naumovozyma_dairenensis_cbs_421_gca_000227115'=>'',
	'nematocida_parisii_ertm1_gca_000250985'=>'',
	'nematocida_parisii_ertm3_gca_000190615'=>'',
	'nematocida_sp_1_ertm2_gca_000250695'=>'',
	'nematocida_sp_1_ertm6_gca_000738915'=>'',
	'nematocida_sp_ertm5_gca_001642415'=>'',
	'nematostella_vectensis'=>'',
	'neocallimastix_californiae_gca_002104975'=>'',
	'neofusicoccum_parvum_ucrnp2_gca_000385595'=>'',
	'neolamprologus_brichardi'=>'',
	'neolecta_irregularis_dah_3_gca_001929475'=>'',
	'neolentinus_lepideus_hhb14362_ss_1_gca_001632425'=>'',
	'neonectria_ditissima_gca_001306435'=>'',
	'neosartorya_fischeri'=>'',
	'neovison_vison'=>'',
	'neurospora_crassa'=>'',
	'neurospora_crassa_gca_000786625'=>'',
	'neurospora_tetrasperma_fgsc_2509_gca_000213195'=>'',
	'nicotiana_attenuata'=>'',
	'nomascus_leucogenys'=>'',
	'nosema_apis_brl_01_gca_000447185'=>'',
	'nosema_bombycis_cq1_gca_000383075'=>'',
	'nosema_ceranae_brl01_gca_000182985'=>'',
	'nosema_ceranae_gca_000988165'=>'',
	'notamacropus_eugenii'=>'',
	'notechis_scutatus'=>'',
	'nothoprocta_perdicaria'=>'',
	'numida_meleagris'=>'',
	'ochotona_princeps'=>'',
	'octodon_degus'=>'',
	'octopus_bimaculoides'=>'',
	'ogataea_parapolymorpha_dl_1_gca_000187245'=>'',
	'ogataea_polymorpha_gca_001664045'=>'',
	'oidiodendron_maius_zn_gca_000827325'=>'',
	'onchocerca_volvulus'=>'',
	'ophiocordyceps_sinensis_co18_gca_000448365'=>'',
	'ophiostoma_piceae_uamh_11346_gca_000410735'=>'',
	'ordospora_colligata_oc4_gca_000803265'=>'',
	'oreochromis_niloticus'=>'',
	'ornithorhynchus_anatinus'=>'',
	'oryctolagus_cuniculus'=>'',
	'oryza_barthii'=>'',
	'oryza_brachyantha'=>'',
	'oryza_glaberrima'=>'',
	'oryza_glumipatula'=>'',
	'oryza_indica'=>'',
	'oryza_longistaminata'=>'',
	'oryza_meridionalis'=>'',
	'oryza_nivara'=>'',
	'oryza_punctata'=>'',
	'oryza_rufipogon'=>'',
	'oryza_sativa'=>'',
	'oryzias_latipes'=>'',
	'oryzias_latipes_hni'=>'',
	'oryzias_latipes_hsok'=>'',
	'oryzias_melastigma'=>'',
	'ostreococcus_lucimarinus'=>'',
	'otolemur_garnettii'=>'',
	'ovis_aries'=>'',
	'pachysolen_tannophilus_nrrl_y_2460_gca_001661245'=>'',
	'pan_paniscus'=>'',
	'pan_troglodytes'=>'',
	'panthera_pardus'=>'',
	'panthera_tigris_altaica'=>'',
	'papio_anubis'=>'',
	'paracoccidioides_brasiliensis_gca_001713645'=>'',
	'paracoccidioides_brasiliensis_gca_001713695'=>'',
	'paracoccidioides_brasiliensis_pb03_gca_000150475'=>'',
	'paracoccidioides_brasiliensis_pb18_gca_000150735'=>'',
	'paracoccidioides_sp_lutzii_pb01_gca_000150705'=>'',
	'paramecium_tetraurelia'=>'',
	'paramicrosporidium_saccamoebae_gca_002794465'=>'',
	'paramormyrops_kingsleyae'=>'',
	'paraphaeosphaeria_sporulosa_gca_001642045'=>'',
	'parus_major'=>'',
	'paulinella_chromatophora'=>'',
	'paxillus_involutus_atcc_200175_gca_000827475'=>'',
	'paxillus_rubicundulus_ve08_2h10_gca_000827395'=>'',
	'pediculus_humanus'=>'',
	'pelodiscus_sinensis'=>'',
	'penicilliopsis_zonata_cbs_506_65_gca_001890105'=>'',
	'penicillium_brasilianum_gca_001048715'=>'',
	'penicillium_brasilianum_gca_002016555'=>'',
	'penicillium_camemberti_fm_013_gca_000513335'=>'',
	'penicillium_chrysogenum_gca_000710275'=>'',
	'penicillium_digitatum_pd1_gca_000315645'=>'',
	'penicillium_digitatum_phi26_gca_000315665'=>'',
	'penicillium_expansum_gca_000769745'=>'',
	'penicillium_freii_gca_001513925'=>'',
	'penicillium_griseofulvum_gca_001561935'=>'',
	'penicillium_italicum_gca_000769765'=>'',
	'penicillium_nordicum_gca_001278595'=>'',
	'penicillium_oxalicum_114_2_gca_000346795'=>'',
	'penicillium_roqueforti_fm164_gca_000513255'=>'',
	'penicillium_rubens_wisconsin_54_1255_gca_000226395'=>'',
	'penicillium_subrubescens_gca_001908125'=>'',
	'peniophora_sp_cont_gca_001632445'=>'',
	'periconia_macrospinosa_gca_003073855'=>'',
	'periophthalmus_magnuspinnatus'=>'',
	'perkinsus_marinus_atcc_50983'=>'',
	'peromyscus_maniculatus_bairdii'=>'',
	'pestalotiopsis_fici_w106_1_gca_000516985'=>'',
	'petromyzon_marinus'=>'',
	'pezoloma_ericae_gca_002865625'=>'',
	'phaeoacremonium_minimum_ucrpa7_gca_000392275'=>'',
	'phaeodactylum_tricornutum'=>'',
	'phaeomoniella_chlamydospora_gca_001006345'=>'',
	'phaeosphaeria_nodorum'=>'',
	'phanerochaete_carnosa_hhb_10118_sp_gca_000300595'=>'',
	'phascolarctos_cinereus'=>'',
	'phaseolus_vulgaris'=>'',
	'phialocephala_scopiformis_gca_001500285'=>'',
	'phialocephala_subalpina_gca_900073065'=>'',
	'phialophora_americana_gca_000835435'=>'',
	'phialophora_attae_gca_001299255'=>'',
	'phlebia_centrifuga_gca_001913855'=>'',
	'phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265'=>'',
	'physcomitrella_patens'=>'',
	'phytophthora_infestans'=>'',
	'phytophthora_kernoviae'=>'',
	'phytophthora_lateralis'=>'',
	'phytophthora_parasitica'=>'',
	'phytophthora_parasitica_cj01a1'=>'',
	'phytophthora_parasitica_gca_000509465'=>'',
	'phytophthora_parasitica_gca_000509485'=>'',
	'phytophthora_parasitica_gca_000509505'=>'',
	'phytophthora_parasitica_gca_000509525'=>'',
	'phytophthora_parasitica_inra_310'=>'',
	'phytophthora_parasitica_p10297'=>'',
	'phytophthora_parasitica_p1976'=>'',
	'phytophthora_ramorum'=>'',
	'phytophthora_sojae'=>'',
	'pichia_kudriavzevii_gca_000764455'=>'',
	'pichia_kudriavzevii_gca_002166775'=>'',
	'pichia_kudriavzevii_gca_003054445'=>'',
	'pichia_membranifaciens_gca_001950575'=>'',
	'pichia_membranifaciens_nrrl_y_2026_gca_001661235'=>'',
	'piliocolobus_tephrosceles'=>'',
	'piloderma_croceum_f_1598_gca_000827315'=>'',
	'piromyces_finnis_gca_002104945'=>'',
	'pisolithus_microcarpus_441_gca_000827275'=>'',
	'pisolithus_tinctorius_marx_270_gca_000827335'=>'',
	'plasmodium_berghei'=>'',
	'plasmodium_berghei_gca_000005395'=>'',
	'plasmodium_berghei_gca_900044335'=>'',
	'plasmodium_chabaudi'=>'',
	'plasmodium_chabaudi_chabaudi_gca_000003075'=>'',
	'plasmodium_coatneyi'=>'',
	'plasmodium_cynomolgi_strain_b'=>'',
	'plasmodium_falciparum'=>'',
	'plasmodium_falciparum_7g8'=>'',
	'plasmodium_falciparum_camp_malaysia'=>'',
	'plasmodium_falciparum_dd2'=>'',
	'plasmodium_falciparum_fch_4'=>'',
	'plasmodium_falciparum_hb3'=>'',
	'plasmodium_falciparum_igh_cr14'=>'',
	'plasmodium_falciparum_malips096_e11'=>'',
	'plasmodium_falciparum_nf135_5_c10'=>'',
	'plasmodium_falciparum_nf54'=>'',
	'plasmodium_falciparum_palo_alto_uganda'=>'',
	'plasmodium_falciparum_raj116'=>'',
	'plasmodium_falciparum_santa_lucia'=>'',
	'plasmodium_falciparum_tanzania_2000708_'=>'',
	'plasmodium_falciparum_ugt5_1'=>'',
	'plasmodium_falciparum_vietnam_oak_knoll_fvo_'=>'',
	'plasmodium_fragile'=>'',
	'plasmodium_gaboni'=>'',
	'plasmodium_inui_san_antonio_1'=>'',
	'plasmodium_knowlesi'=>'',
	'plasmodium_knowlesi_strain_h_gca_900004885'=>'',
	'plasmodium_malariae'=>'',
	'plasmodium_ovale_curtisi'=>'',
	'plasmodium_reichenowi'=>'',
	'plasmodium_reichenowi_gca_001601855'=>'',
	'plasmodium_vinckei_petteri'=>'',
	'plasmodium_vinckei_vinckei'=>'',
	'plasmodium_vivax'=>'',
	'plasmodium_vivax_brazil_i'=>'',
	'plasmodium_vivax_india_vii'=>'',
	'plasmodium_vivax_mauritania_i'=>'',
	'plasmodium_vivax_north_korean'=>'',
	'plasmodium_yoelii'=>'',
	'plasmodium_yoelii_17x'=>'',
	'plasmodium_yoelii_gca_900002395'=>'',
	'plasmodium_yoelii_yoelii'=>'',
	'plasmopara_halstedii'=>'',
	'pleurotus_ostreatus_pc15_gca_000697685'=>'',
	'plicaturopsis_crispa_fd_325_ss_3_gca_000827205'=>'',
	'pneumocystis_carinii_b80_gca_001477545'=>'',
	'pneumocystis_jirovecii_ru7_gca_001477535'=>'',
	'pneumocystis_murina_b123_gca_000349005'=>'',
	'pochonia_chlamydosporia_170_gca_001653235'=>'',
	'podospora_anserina_s_mat__gca_000226545'=>'',
	'poecilia_formosa'=>'',
	'poecilia_latipinna'=>'',
	'poecilia_mexicana'=>'',
	'poecilia_reticulata'=>'',
	'pogona_vitticeps'=>'',
	'polysphondylium_pallidum_pn500'=>'',
	'pongo_abelii'=>'',
	'populus_trichocarpa'=>'',
	'postia_placenta_mad_698_r_gca_000006255'=>'',
	'postia_placenta_mad_698_r_sb12_gca_002117355'=>'',
	'procavia_capensis'=>'',
	'prolemur_simus'=>'',
	'propithecus_coquereli'=>'',
	'protomyces_lactucaedebilis_gca_002105105'=>'',
	'prunus_persica'=>'',
	'pseudocercospora_fijiensis_cirad86_gca_000340215'=>'',
	'pseudocercospora_musae_gca_001578225'=>'',
	'pseudogymnoascus_destructans_20631_21_gca_000184105'=>'',
	'pseudogymnoascus_destructans_gca_001641265'=>'',
	'pseudogymnoascus_sp_03vt05_gca_001662645'=>'',
	'pseudogymnoascus_sp_05ny08_gca_001662605'=>'',
	'pseudogymnoascus_sp_23342_1_i1_gca_001662575'=>'',
	'pseudogymnoascus_sp_24mn13_gca_001662595'=>'',
	'pseudogymnoascus_sp_vkm_f_103_gca_000750895'=>'',
	'pseudogymnoascus_sp_vkm_f_3557_gca_000750665'=>'',
	'pseudogymnoascus_sp_vkm_f_3775_gca_000750715'=>'',
	'pseudogymnoascus_sp_vkm_f_3808_gca_000750675'=>'',
	'pseudogymnoascus_sp_vkm_f_4246_gca_000750735'=>'',
	'pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745'=>'',
	'pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755'=>'',
	'pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805'=>'',
	'pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815'=>'',
	'pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875'=>'',
	'pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925'=>'',
	'pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935'=>'',
	'pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995'=>'',
	'pseudogymnoascus_sp_wsf_3629_gca_001662585'=>'',
	'pseudogymnoascus_verrucosus_gca_001662655'=>'',
	'pseudoloma_neurophilia_gca_001432165'=>'',
	'pseudomassariella_vexata_gca_002105095'=>'',
	'pseudomicrostroma_glucosiphilum_gca_003144135'=>'',
	'pseudozyma_hubeiensis_sy62_gca_000403515'=>'',
	'pteropus_vampyrus'=>'',
	'puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275'=>'',
	'puccinia_graminis'=>'',
	'puccinia_graminisug99'=>'',
	'puccinia_sorghi_gca_001263375'=>'',
	'puccinia_striiformis'=>'',
	'puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645'=>'',
	'puccinia_triticina'=>'',
	'punctularia_strigosozonata_hhb_11173_ss5_gca_000264995'=>'',
	'pundamilia_nyererei'=>'',
	'purpureocillium_lilacinum_gca_001653205'=>'',
	'purpureocillium_lilacinum_gca_001653265'=>'',
	'pygocentrus_nattereri'=>'',
	'pyrenochaeta_sp_ds3say3a_gca_001644535'=>'',
	'pyrenophora_teres'=>'',
	'pyrenophora_tritici_repentis_gca_003171545'=>'',
	'pyrenophora_tritici_repentis_gca_003231325'=>'',
	'pyrenophora_tritici_repentis_gca_003231365'=>'',
	'pyrenophora_tritici_repentis_gca_003231425'=>'',
	'pyrenophora_triticirepentis'=>'',
	'pythium_aphanidermatum'=>'',
	'pythium_arrhenomanes'=>'',
	'pythium_irregulare'=>'',
	'pythium_iwayamai'=>'',
	'pythium_ultimum'=>'',
	'pythium_vexans'=>'',
	'rachicladosporium_antarcticum_gca_002077065'=>'',
	'rachicladosporium_sp_ccfee_5018_gca_002077045'=>'',
	'ramularia_collo_cygni_gca_900074925'=>'',
	'rasamsonia_emersonii_cbs_393_64_gca_000968595'=>'',
	'rattus_norvegicus'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/rattus_norvegicus/proTRAC_SRR035657/clusters.fasta https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/rattus_norvegicus/proTRAC_SRR035663/clusters.fasta',
	'reticulomyxa_filosa'=>'',
	'rhinocladiella_mackenziei_cbs_650_93_gca_000835555'=>'',
	'rhinopithecus_bieti'=>'',
	'rhinopithecus_roxellana'=>'',
	'rhizoclosmatium_globosum_gca_002104985'=>'',
	'rhizoctonia_solani_123e_gca_000715385'=>'',
	'rhizoctonia_solani_ag_1_ia_gca_000334115'=>'',
	'rhizoctonia_solani_ag_1_ib_gca_000350255'=>'',
	'rhizoctonia_solani_ag_8_wac10335_gca_000695385'=>'',
	'rhizoctonia_solani_gca_001286725'=>'',
	'rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155'=>'',
	'rhizophagus_irregularis_daom_181602_gca_000439145'=>'',
	'rhizophagus_irregularis_daom_197198w_gca_000597685'=>'',
	'rhizophagus_irregularis_gca_001593125'=>'',
	'rhizophagus_irregularis_gca_001593145'=>'',
	'rhizophagus_irregularis_gca_001593155'=>'',
	'rhizophagus_irregularis_gca_001593205'=>'',
	'rhizopogon_vinicolor_am_or11_026_gca_001658105'=>'',
	'rhizopus_delemar_ra_99_880_gca_000149305'=>'',
	'rhizopus_microsporus_atcc_52813_gca_002708625'=>'',
	'rhizopus_microsporus_gca_000825725'=>'',
	'rhizopus_microsporus_gca_002083735'=>'',
	'rhizopus_microsporus_var_microsporus_gca_002083745'=>'',
	'rhodnius_prolixus'=>'',
	'rhodotorula_graminis_wp1_gca_001329695'=>'',
	'rhodotorula_sp_jg_1b_gca_001541205'=>'',
	'rhodotorula_taiwanensis_gca_002922495'=>'',
	'rhodotorula_toruloides_atcc_204091_gca_000222205'=>'',
	'rhodotorula_toruloides_np11_gca_000320785'=>'',
	'rhynchosporium_agropyri_gca_900074905'=>'',
	'rhynchosporium_commune_gca_900074885'=>'',
	'rhynchosporium_secalis_gca_900074895'=>'',
	'rosellinia_necatrix_gca_001445595'=>'',
	'rozella_allomycis_csf55_gca_000442015'=>'',
	'saccharomyces_arboricola_h_6_gca_000292725'=>'',
	'saccharomyces_cerevisiae'=>'',
	'saccharomyces_cerevisiae_ec1118_gca_000218975'=>'',
	'saccharomyces_cerevisiae_fostersb_gca_000190255'=>'',
	'saccharomyces_cerevisiae_fosterso_gca_000326005'=>'',
	'saccharomyces_cerevisiae_gca_001634645'=>'',
	'saccharomyces_cerevisiae_gca_002804325'=>'',
	'saccharomyces_cerevisiae_kyokai_no_7_gca_000260735'=>'',
	'saccharomyces_cerevisiae_lalvin_qa23_gca_000325965'=>'',
	'saccharomyces_cerevisiae_p283_gca_000568295'=>'',
	'saccharomyces_cerevisiae_p301_gca_000568055'=>'',
	'saccharomyces_cerevisiae_r103_gca_000568365'=>'',
	'saccharomyces_cerevisiae_rm11_1a_gca_000149365'=>'',
	'saccharomyces_cerevisiae_vin13_gca_000190215'=>'',
	'saccharomyces_cerevisiae_vl3_gca_000190235'=>'',
	'saccharomyces_cerevisiae_yjm1078_gca_000975645'=>'',
	'saccharomyces_cerevisiae_yjm1083_gca_000975675'=>'',
	'saccharomyces_cerevisiae_yjm1129_gca_000975705'=>'',
	'saccharomyces_cerevisiae_yjm1133_gca_000976695'=>'',
	'saccharomyces_cerevisiae_yjm1190_gca_000976725'=>'',
	'saccharomyces_cerevisiae_yjm1199_gca_000976755'=>'',
	'saccharomyces_cerevisiae_yjm1202_gca_000976785'=>'',
	'saccharomyces_cerevisiae_yjm1208_gca_000976815'=>'',
	'saccharomyces_cerevisiae_yjm1242_gca_000976845'=>'',
	'saccharomyces_cerevisiae_yjm1244_gca_000976875'=>'',
	'saccharomyces_cerevisiae_yjm1248_gca_000976905'=>'',
	'saccharomyces_cerevisiae_yjm1250_gca_000976935'=>'',
	'saccharomyces_cerevisiae_yjm1252_gca_000976965'=>'',
	'saccharomyces_cerevisiae_yjm1273_gca_000976995'=>'',
	'saccharomyces_cerevisiae_yjm1304_gca_000977025'=>'',
	'saccharomyces_cerevisiae_yjm1307_gca_000977055'=>'',
	'saccharomyces_cerevisiae_yjm1311_gca_000977085'=>'',
	'saccharomyces_cerevisiae_yjm1326_gca_000977115'=>'',
	'saccharomyces_cerevisiae_yjm1332_gca_000977145'=>'',
	'saccharomyces_cerevisiae_yjm1336_gca_000977175'=>'',
	'saccharomyces_cerevisiae_yjm1338_gca_000977205'=>'',
	'saccharomyces_cerevisiae_yjm1341_gca_000977235'=>'',
	'saccharomyces_cerevisiae_yjm1342_gca_000977265'=>'',
	'saccharomyces_cerevisiae_yjm1355_gca_000977295'=>'',
	'saccharomyces_cerevisiae_yjm1356_gca_000977325'=>'',
	'saccharomyces_cerevisiae_yjm1381_gca_000977355'=>'',
	'saccharomyces_cerevisiae_yjm1383_gca_000977385'=>'',
	'saccharomyces_cerevisiae_yjm1385_gca_000977415'=>'',
	'saccharomyces_cerevisiae_yjm1386_gca_000977445'=>'',
	'saccharomyces_cerevisiae_yjm1387_gca_000977475'=>'',
	'saccharomyces_cerevisiae_yjm1388_gca_000977505'=>'',
	'saccharomyces_cerevisiae_yjm1389_gca_000977535'=>'',
	'saccharomyces_cerevisiae_yjm1399_gca_000977565'=>'',
	'saccharomyces_cerevisiae_yjm1400_gca_000977595'=>'',
	'saccharomyces_cerevisiae_yjm1401_gca_000977625'=>'',
	'saccharomyces_cerevisiae_yjm1402_gca_000977655'=>'',
	'saccharomyces_cerevisiae_yjm1415_gca_000977685'=>'',
	'saccharomyces_cerevisiae_yjm1417_gca_000977715'=>'',
	'saccharomyces_cerevisiae_yjm1418_gca_000977745'=>'',
	'saccharomyces_cerevisiae_yjm1419_gca_000977775'=>'',
	'saccharomyces_cerevisiae_yjm1433_gca_000977805'=>'',
	'saccharomyces_cerevisiae_yjm1434_gca_000977835'=>'',
	'saccharomyces_cerevisiae_yjm1439_gca_000977865'=>'',
	'saccharomyces_cerevisiae_yjm1443_gca_000977895'=>'',
	'saccharomyces_cerevisiae_yjm1444_gca_000977925'=>'',
	'saccharomyces_cerevisiae_yjm1447_gca_000977955'=>'',
	'saccharomyces_cerevisiae_yjm1450_gca_000977985'=>'',
	'saccharomyces_cerevisiae_yjm1460_gca_000978015'=>'',
	'saccharomyces_cerevisiae_yjm1463_gca_000978045'=>'',
	'saccharomyces_cerevisiae_yjm1477_gca_000978075'=>'',
	'saccharomyces_cerevisiae_yjm1478_gca_000978105'=>'',
	'saccharomyces_cerevisiae_yjm1479_gca_000978135'=>'',
	'saccharomyces_cerevisiae_yjm1526_gca_000978165'=>'',
	'saccharomyces_cerevisiae_yjm1527_gca_000978195'=>'',
	'saccharomyces_cerevisiae_yjm1549_gca_000978225'=>'',
	'saccharomyces_cerevisiae_yjm1573_gca_000978255'=>'',
	'saccharomyces_cerevisiae_yjm1574_gca_000978285'=>'',
	'saccharomyces_cerevisiae_yjm1592_gca_000978315'=>'',
	'saccharomyces_cerevisiae_yjm1615_gca_000978345'=>'',
	'saccharomyces_cerevisiae_yjm189_gca_000975735'=>'',
	'saccharomyces_cerevisiae_yjm193_gca_000975765'=>'',
	'saccharomyces_cerevisiae_yjm195_gca_000975585'=>'',
	'saccharomyces_cerevisiae_yjm244_gca_000975615'=>'',
	'saccharomyces_cerevisiae_yjm248_gca_000975795'=>'',
	'saccharomyces_cerevisiae_yjm270_gca_000975825'=>'',
	'saccharomyces_cerevisiae_yjm271_gca_000975855'=>'',
	'saccharomyces_cerevisiae_yjm320_gca_000975885'=>'',
	'saccharomyces_cerevisiae_yjm326_gca_000975915'=>'',
	'saccharomyces_cerevisiae_yjm428_gca_000975945'=>'',
	'saccharomyces_cerevisiae_yjm450_gca_000975975'=>'',
	'saccharomyces_cerevisiae_yjm451_gca_000976005'=>'',
	'saccharomyces_cerevisiae_yjm453_gca_000976035'=>'',
	'saccharomyces_cerevisiae_yjm456_gca_000976065'=>'',
	'saccharomyces_cerevisiae_yjm470_gca_000976095'=>'',
	'saccharomyces_cerevisiae_yjm541_gca_000976125'=>'',
	'saccharomyces_cerevisiae_yjm554_gca_000976155'=>'',
	'saccharomyces_cerevisiae_yjm555_gca_000976185'=>'',
	'saccharomyces_cerevisiae_yjm627_gca_000976215'=>'',
	'saccharomyces_cerevisiae_yjm681_gca_000976245'=>'',
	'saccharomyces_cerevisiae_yjm682_gca_000976275'=>'',
	'saccharomyces_cerevisiae_yjm683_gca_000976305'=>'',
	'saccharomyces_cerevisiae_yjm689_gca_000976335'=>'',
	'saccharomyces_cerevisiae_yjm693_gca_000976365'=>'',
	'saccharomyces_cerevisiae_yjm789_gca_000181435'=>'',
	'saccharomyces_cerevisiae_yjm969_gca_000976395'=>'',
	'saccharomyces_cerevisiae_yjm972_gca_000976425'=>'',
	'saccharomyces_cerevisiae_yjm975_gca_000976455'=>'',
	'saccharomyces_cerevisiae_yjm978_gca_000976485'=>'',
	'saccharomyces_cerevisiae_yjm981_gca_000976515'=>'',
	'saccharomyces_cerevisiae_yjm984_gca_000976545'=>'',
	'saccharomyces_cerevisiae_yjm987_gca_000976575'=>'',
	'saccharomyces_cerevisiae_yjm990_gca_000976605'=>'',
	'saccharomyces_cerevisiae_yjm993_gca_000662435'=>'',
	'saccharomyces_cerevisiae_yjm996_gca_000976665'=>'',
	'saccharomyces_eubayanus_gca_001298625'=>'',
	'saccharomyces_kudriavzevii_ifo_1802_gca_000167075'=>'',
	'saccharomycetaceae_sp_ashbya_aceri__gca_000412225'=>'',
	'saimiri_boliviensis_boliviensis'=>'',
	'saitoella_complicata_nrrl_y_17804_gca_000227095'=>'',
	'saitoella_complicata_nrrl_y_17804_gca_001661265'=>'',
	'salpingoeca_rosetta'=>'',
	'salvator_merianae'=>'',
	'saprolegnia_diclina_vs20'=>'',
	'saprolegnia_parasitica_cbs_223_65'=>'',
	'sarcophilus_harrisii'=>'',
	'sarcoptes_scabiei'=>'',
	'scedosporium_apiospermum_gca_000732125'=>'',
	'scheffersomyces_stipitis_cbs_6054_gca_000209165'=>'',
	'schistosoma_mansoni'=>'',
	'schizophyllum_commune_h4_8_gca_000143185'=>'',
	'schizopora_paradoxa_gca_001020605'=>'',
	'schizosaccharomyces_cryophilus'=>'',
	'schizosaccharomyces_japonicus'=>'',
	'schizosaccharomyces_octosporus'=>'',
	'schizosaccharomyces_pombe'=>'',
	'scleroderma_citrinum_foug_a_gca_000827425'=>'',
	'scleropages_formosus'=>'',
	'sclerotinia_borealis_f_4128_gca_000503235'=>'',
	'sclerotinia_sclerotiorum'=>'',
	'sclerotinia_sclerotiorum_1980_uf_70_gca_001857865'=>'',
	'scophthalmus_maximus'=>'',
	'selaginella_moellendorffii'=>'',
	'serendipita_indica_dsm_11827_gca_000313545'=>'',
	'serendipita_vermifera_maff_305830_gca_000827415'=>'',
	'serinus_canaria'=>'',
	'seriola_dumerili'=>'',
	'seriola_lalandi_dorsalis'=>'',
	'serpula_lacrymans_var_lacrymans_s7_3_gca_000218725'=>'',
	'setaria_italica'=>'',
	'sistotremastrum_niveocremeum_hhb9708_gca_001630475'=>'',
	'sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355'=>'',
	'solanum_lycopersicum'=>'',
	'solanum_tuberosum'=>'',
	'solenopsis_invicta'=>'',
	'sordaria_macrospora_k_hell_gca_000182805'=>'',
	'sorex_araneus'=>'',
	'sorghum_bicolor'=>'',
	'spathaspora_passalidarum_nrrl_y_27907_gca_000223485'=>'',
	'spermophilus_dauricus'=>'',
	'sphaerobolus_stellatus_ss14_gca_000827215'=>'',
	'sphaeroforma_arctica_jp610'=>'',
	'sphaerulina_musiva_so2202_gca_000320565'=>'',
	'sphenodon_punctatus'=>'',
	'spironucleus_salmonicida'=>'',
	'spizellomyces_punctatus_daom_br117_gca_000182565'=>'',
	'sporisorium_reilianum'=>'',
	'sporisorium_reilianum_f_sp_reilianum_gca_900162835'=>'',
	'sporothrix_brasiliensis_5110_gca_000820605'=>'',
	'sporothrix_insectorum_rcef_264_gca_001636815'=>'',
	'sporothrix_schenckii_1099_18_gca_000961545'=>'',
	'sporothrix_schenckii_atcc_58251_gca_000474925'=>'',
	'spraguea_lophii_42_110_gca_000430065'=>'',
	'stachybotrys_chartarum_ibt_40288_gca_000732765'=>'',
	'stachybotrys_chlorohalonata_ibt_40285_gca_000732775'=>'',
	'stagonospora_sp_src1lsm3a_gca_001644525'=>'',
	'stegastes_partitus'=>'',
	'stegodyphus_mimosarum'=>'',
	'stemphylium_lycopersici_gca_001191545'=>'',
	'stereum_hirsutum_fp_91666_ss1_gca_000264905'=>'',
	'strigamia_maritima'=>'',
	'strigomonas_culicis'=>'',
	'strongylocentrotus_purpuratus'=>'',
	'strongyloides_ratti'=>'',
	'sugiyamaella_lignohabitans_gca_001640025'=>'',
	'suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415'=>'',
	'suillus_luteus_uh_slu_lm8_n1_gca_000827255'=>'',
	'sus_scrofa'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/sus_scrofa/proTRAC_normal_testis_generic/clusters.fasta https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/sus_scrofa/proTRAC_normal_ovary_generic/clusters.fasta',
	'syncephalastrum_racemosum_gca_002105135'=>'',
	'taeniopygia_guttata'=>'',
	'takifugu_rubripes'=>'',
	'talaromyces_cellulolyticus_gca_000829775'=>'',
	'talaromyces_islandicus_gca_000985935'=>'',
	'talaromyces_marneffei_atcc_18224_gca_000001985'=>'',
	'talaromyces_marneffei_pm1_gca_000750115'=>'',
	'talaromyces_stipitatus_atcc_10500_gca_000003125'=>'',
	'teleopsis_dalmanni'=>'',
	'termitomyces_sp_j132_gca_001263195'=>'',
	'testicularia_cyperi_gca_003144125'=>'',
	'tetrahymena_thermophila'=>'',
	'tetranychus_urticae'=>'',
	'tetraodon_nigroviridis'=>'',
	'tetrapisispora_blattae_cbs_6284_gca_000315915'=>'',
	'tetrapisispora_phaffii_cbs_4417_gca_000236905'=>'',
	'thalassiosira_oceanica'=>'',
	'thalassiosira_oceanica_ccmp1005'=>'',
	'thalassiosira_pseudonana'=>'',
	'thecamonas_trahens_atcc_50062'=>'',
	'theileria_annulata'=>'',
	'theileria_equi_strain_wa'=>'',
	'theileria_orientalis_strain_shintoku'=>'',
	'theileria_parva'=>'',
	'thelohanellus_kitauei'=>'',
	'theobroma_cacao'=>'',
	'thermothelomyces_thermophila_atcc_42464_gca_000226095'=>'',
	'theropithecus_gelada'=>'',
	'thielavia_terrestris_nrrl_8126_gca_000226115'=>'',
	'thielaviopsis_punctulata_gca_000968615'=>'',
	'tilletia_controversa_gca_001645045'=>'',
	'tilletia_walkeri_gca_001645055'=>'',
	'tilletiaria_anomala_ubc_951_gca_000711695'=>'',
	'tilletiopsis_washingtonensis_gca_003144115'=>'',
	'tolypocladium_ophioglossoides_cbs_100239_gca_001189435'=>'',
	'torrubiella_hemipterigena_gca_000825705'=>'',
	'tortispora_caseinolytica_nrrl_y_17796_gca_001661475'=>'',
	'torulaspora_delbrueckii_gca_000243375'=>'',
	'toxoplasma_gondii'=>'',
	'toxoplasma_gondii_ari'=>'',
	'toxoplasma_gondii_fou'=>'',
	'toxoplasma_gondii_gab2_2007_gal_dom2'=>'',
	'toxoplasma_gondii_gt1'=>'',
	'toxoplasma_gondii_mas'=>'',
	'toxoplasma_gondii_p89'=>'',
	'toxoplasma_gondii_rub'=>'',
	'toxoplasma_gondii_tgcatprc2'=>'',
	'toxoplasma_gondii_vand'=>'',
	'toxoplasma_gondii_veg'=>'',
	'trachipleistophora_hominis_gca_000316135'=>'',
	'trametes_cinnabarina_gca_000765035'=>'',
	'trametes_coccinea_brfm310_gca_002092935'=>'',
	'trametes_pubescens_gca_001895945'=>'',
	'trametes_versicolor_fp_101664_ss1_gca_000271585'=>'',
	'tremella_mesenterica_dsm_1558_gca_000271645'=>'',
	'tribolium_castaneum'=>'',
	'trichoderma_asperellum_cbs_433_97_gca_003025105'=>'',
	'trichoderma_atroviride_imi_206040_gca_000171015'=>'',
	'trichoderma_citrinoviride_gca_003025115'=>'',
	'trichoderma_harzianum_cbs_226_95_gca_003025095'=>'',
	'trichoderma_harzianum_gca_000988865'=>'',
	'trichoderma_longibrachiatum_atcc_18648_gca_003025155'=>'',
	'trichoderma_reesei'=>'',
	'trichoderma_reesei_rut_c_30_gca_000513815'=>'',
	'trichoderma_virens'=>'',
	'trichomonas_vaginalis_g3'=>'',
	'trichophyton_benhamiae_cbs_112371_gca_000151125'=>'',
	'trichophyton_equinum_cbs_127_97_gca_000151175'=>'',
	'trichophyton_interdigitale_h6_gca_000616785'=>'',
	'trichophyton_interdigitale_mr816_gca_000622975'=>'',
	'trichophyton_mentagrophytes_gca_003118255'=>'',
	'trichophyton_rubrum_cbs_100081_gca_000616805'=>'',
	'trichophyton_rubrum_cbs_118892_gca_000151425'=>'',
	'trichophyton_rubrum_cbs_202_88_gca_000616985'=>'',
	'trichophyton_rubrum_cbs_288_86_gca_000616825'=>'',
	'trichophyton_rubrum_cbs_289_86_gca_000616845'=>'',
	'trichophyton_rubrum_cbs_735_88_gca_000616965'=>'',
	'trichophyton_rubrum_d6_gca_000622995'=>'',
	'trichophyton_rubrum_mr1448_gca_000616905'=>'',
	'trichophyton_rubrum_mr1459_gca_000616945'=>'',
	'trichophyton_rubrum_mr850_gca_000616765'=>'',
	'trichophyton_soudanense_cbs_452_61_gca_000616865'=>'',
	'trichophyton_tonsurans_cbs_112818_gca_000151455'=>'',
	'trichophyton_verrucosum_hki_0517_gca_000151505'=>'',
	'trichoplax_adhaerens'=>'',
	'trichosporon_asahii_var_asahii_cbs_2479_gca_000293215'=>'',
	'trichosporon_asahii_var_asahii_cbs_8904_gca_000299215'=>'',
	'trifolium_pratense'=>'',
	'triticum_aestivum'=>'',
	'triticum_dicoccoides'=>'',
	'triticum_urartu'=>'',
	'trypanosoma_brucei'=>'',
	'trypanosoma_cruzi'=>'',
	'trypanosoma_cruzi_dm28c'=>'',
	'tsuchiyaea_wingfieldii_cbs_7118_gca_001720155'=>'',
	'tuber_borchii_gca_003070745'=>'',
	'tuber_magnatum_gca_003182015'=>'',
	'tuber_melanosporum'=>'',
	'tulasnella_calospora_mut_4182_gca_000827465'=>'',
	'tupaia_belangeri'=>'https://www.smallrnagroup.uni-mainz.de/piRNAclusterDB/tupaia_belangeri/proTRAC_SRR1184429/clusters.fasta',
	'tursiops_truncatus'=>'',
	'uncinocarpus_reesii_1704_gca_000003515'=>'',
	'urocitellus_parryii'=>'',
	'ursus_americanus'=>'',
	'ursus_maritimus'=>'',
	'ustilaginoidea_virens_gca_000687475'=>'',
	'ustilaginomycotina_sp_sa_807_gca_003144235'=>'',
	'ustilago_bromivora_gca_900080155'=>'',
	'ustilago_hordei_gca_000286035'=>'',
	'ustilago_maydis'=>'',
	'valsa_mali_gca_000818155'=>'',
	'valsa_mali_var_pyri_gca_000813385'=>'',
	'vanderwaltozyma_polyspora_dsm_70294_gca_000150035'=>'',
	'vavraia_culicis_subsp_floridensis_gca_000192795'=>'',
	'verruconis_gallopava_gca_000836295'=>'',
	'verticillium_alfalfae_vams_102_gca_000150825'=>'',
	'verticillium_dahliae'=>'',
	'verticillium_dahliaejr2'=>'',
	'verticillium_longisporum_gca_001268145'=>'',
	'vicugna_pacos'=>'',
	'vigna_angularis'=>'',
	'vigna_radiata'=>'',
	'vitis_vinifera'=>'',
	'vitrella_brassicaformis_ccmp3155'=>'',
	'vittaforma_corneae_atcc_50505_gca_000231115'=>'',
	'vulpes_vulpes'=>'',
	'wallemia_ichthyophaga_exf_994_gca_000400465'=>'',
	'wallemia_mellicola_cbs_633_66_gca_000263375'=>'',
	'wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255'=>'',
	'wickerhamomyces_ciferrii_gca_000313485'=>'',
	'wolfiporia_cocos_md_104_ss10_gca_000344635'=>'',
	'xenopus_tropicalis'=>'',
	'xiphophorus_couchianus'=>'',
	'xiphophorus_maculatus'=>'',
	'xylona_heveae_tc161_gca_001619985'=>'',
	'yarrowia_lipolytica'=>'',
	'yarrowia_lipolytica_gca_001761485'=>'',
	'yarrowia_lipolytica_gca_900087985'=>'',
	'zea_mays'=>'',
	'zonotrichia_albicollis'=>'',
	'zootermopsis_nevadensis'=>'',
	'zygosaccharomyces_bailii_clib_213_gca_000442885'=>'',
	'zygosaccharomyces_bailii_gca_900162805'=>'',
	'zygosaccharomyces_bailii_isa1307_gca_000530735'=>'',
	'zygosaccharomyces_parabailii_gca_001984395'=>'',
	'zygosaccharomyces_rouxii_gca_000026365'=>'',
	'zygosaccharomyces_rouxii_gca_001972345'=>'',
	'zymoseptoria_brevis_gca_000966595'=>'',
	'zymoseptoria_tritici'=>'',
	);### PIRCLDB URLS END ###
				}
			if(length$piRclDB_urls{$species}>0)
				{
				if(!$silent){print"\nDownload piRNA cluster data for $species{$species} from piRNA cluster database...";}

				@piURLs=split(' ',$piRclDB_urls{$species});
				$i=0;
				$success=0;
				foreach$piURL(@piURLs)
					{
					$i++;
					$request = HTTP::Request->new(GET => "$piURL");
					$response = $useragent->request($request);
					$content = $response->content;
					open(my $out, '>', "UNITAS_refdump_$species/piRclDB_$i");
					print $out "$content";
					close $out;
					if(-e"UNITAS_refdump_$species/piRclDB_$i")
						{
						$success++;
						}
					else
						{
						if(!$silent){print" FAILED: $status!";}
						}
					}
				if($success==$i&&!$silent)
					{
					print" $success/$i OK. Database date: $mday.$mon.$year (dd.mm.yyyy)";
					}
				elsif(!$silent)
					{
					print" Warning: Downloaded $success/$i files!\n\n";
					}
				$db_versions_info.="piRNA cluster database version/date: ...... $mday.$mon.$year (dd.mm.yyyy)\n";

				if(!$silent){print"\nProcess data from piRNA cluster database...";}
				open(PIRCLDB_ALL,">UNITAS_refdump_$species/piRclDB")||die print"\nERROR! Unable to create piRNA cluster sequence file UNITAS_refdump_$species/piRclDB.\n$!\n\n";
				$i_cl=0;
				foreach$id(1..$i)
					{
					open(PIRCLDB,"UNITAS_refdump_$species/piRclDB_$id")||print"\nERROR! Unable to read UNITAS_refdump_$species/piRclDB_$id.\n$!\n\n";
					while(<PIRCLDB>)
						{
						$_=~s/\s*$//;
						if($_=~/^>/)
							{
							$i_cl++;
							$_=~s/ \(.*$//;
							print PIRCLDB_ALL"$_\n";
							}
						else
							{
							$piRcl_seq="";
							while(1)
								{
								$_=~s/.{1,60}//;
								$piRcl_seq.="$&\n";
								last if(length$_==0);
								}
							print PIRCLDB_ALL$piRcl_seq;
							}
						}
					close PIRCLDB;
					unlink("UNITAS_refdump_$species/piRclDB_$id");
					}
				close PIRCLDB_ALL;
				if(!$silent){print" OK ($i_cl piRNA cluster sequences collected).";}
				}
			else
				{
				if(!$silent){print"\nNo piRNA cluster data for $species{$species} at piRNA cluster database -> SKIP.";}
				$db_versions_info.="piRNA cluster database version/date: ...... $mday.$mon.$year (dd.mm.yyyy)\n";
				}
			}

		# download and process ensembl ncRNA data
		if(!$silent){print"\nCheck latest Ensembl database release...";}
		$ensembl_home="";
		$ensembl_home=get("http://www.ensembl.org");
		if(length$ensembl_home>0)
			{
			$ensembl_home=~s/Ensembl genome browser \d+//;
			$ensembl_title=$&;
			$ensembl_title=~s/\d+//;
			$ensembl_release=$&;
			if($ensembl_release!~/^\d+$/)
				{
				if(!$silent){print" FAILED!\nServer error? Assuming release 88 to be the latest.\n\n";}
				$ensembl_release=88;
				}
			elsif(!$silent){print" $ensembl_release.";}
			}
		else
			{
			if(!$silent){print" FAILED!\nServer error? Assuming release 88 to be the latest.\n\n";}
			$ensembl_release=88;
			}

		if(!$silent){print"\nCheck latest EnsemblGenomes database release...";}
		$ensemblgenomes_home="";
		$ensemblgenomes_home=get("http://ensemblgenomes.org/");
		if(length$ensemblgenomes_home>0)
			{
			$ensemblgenomes_home=~s/<option value="\d+">Release \d+ \(//;
			$ensemblgenomes_title=$&;
			$ensemblgenomes_title=~s/\d+//;
			$ensemblgenomes_release=$&;
			if($ensemblgenomes_release!~/^\d+$/)
				{
				if(!$silent){print" FAILED!\nServer error? Assuming release 35 to be the latest.\n\n";}
				$ensemblgenomes_release=35;
				}
			elsif(!$silent){print" $ensemblgenomes_release.";}
			}
		else
			{
			if(!$silent){print" FAILED!\nServer error? Assuming release 35 to be the latest.\n\n";}
			$ensemblgenomes_release=35;
			}

		$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/URLs/Ensembl_ncRNA');
		$response = $useragent->request($request);
		$eval_URL="";
		$eval_URL = $response->content;
		if(length$eval_URL>0)
			{
			eval$eval_URL;
			}
		else
			{
			if(!$silent){print"\nFailed to load current Ensembl URLs for ncRNA data from smallRNAgroup-SERVER. Use internal list.";}
### NCRNA URLS START ###
%ensembl_ncRNA=
	(
	'_candida_arabinofermentans_nrrl_yb_2248_gca_001661425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_arabinofermentans_nrrl_yb_2248_gca_001661425/ncrna/_candida_arabinofermentans_nrrl_yb_2248_gca_001661425.Canar1.ncrna.fa.gz',
	'_candida_auris_gca_001189475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_auris_gca_001189475/ncrna/_candida_auris_gca_001189475.ASM118947v1.ncrna.fa.gz',
	'_candida_auris_gca_002759435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_auris_gca_002759435/ncrna/_candida_auris_gca_002759435.Cand_auris_B8441_V2.ncrna.fa.gz',
	'_candida_auris_gca_002775015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_auris_gca_002775015/ncrna/_candida_auris_gca_002775015.Cand_auris_B11221_V1.ncrna.fa.gz',
	'_candida_auris_gca_003013715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_auris_gca_003013715/ncrna/_candida_auris_gca_003013715.Cand_auris_B11220.ncrna.fa.gz',
	'_candida_auris_gca_003014415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_auris_gca_003014415/ncrna/_candida_auris_gca_003014415.Cand_auris_B11243.ncrna.fa.gz',
	'_candida_duobushaemulonis_gca_002926085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_duobushaemulonis_gca_002926085/ncrna/_candida_duobushaemulonis_gca_002926085.CanDuoHae_v1.0.ncrna.fa.gz',
	'_candida_glabrata_gca_000002545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/_candida_glabrata_gca_000002545/ncrna/_candida_glabrata_gca_000002545.ASM254v2.ncrna.fa.gz',
	'_candida_glabrata_gca_002219185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_glabrata_gca_002219185/ncrna/_candida_glabrata_gca_002219185.ASM221918v1.ncrna.fa.gz',
	'_candida_glabrata_gca_002219195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_glabrata_gca_002219195/ncrna/_candida_glabrata_gca_002219195.ASM221919v1.ncrna.fa.gz',
	'_candida_haemulonis_gca_002926055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_haemulonis_gca_002926055/ncrna/_candida_haemulonis_gca_002926055.CanHae_1.0.ncrna.fa.gz',
	'_candida_pseudohaemulonis_gca_003013735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_pseudohaemulonis_gca_003013735/ncrna/_candida_pseudohaemulonis_gca_003013735.Cand_pseudohaemulonii_B12108.ncrna.fa.gz',
	'_candida_tenuis_atcc_10573_gca_000223465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/_candida_tenuis_atcc_10573_gca_000223465/ncrna/_candida_tenuis_atcc_10573_gca_000223465.Candida_tenuis_v1.0.ncrna.fa.gz',
	'absidia_glauca_gca_900079185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/absidia_glauca_gca_900079185/ncrna/Absidia_glauca_gca_900079185.AG_v1.ncrna.fa.gz',
	'absidia_repens_gca_002105175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/absidia_repens_gca_002105175/ncrna/Absidia_repens_gca_002105175.Absrep1.ncrna.fa.gz',
	'acanthochromis_polyacanthus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/acanthochromis_polyacanthus/ncrna/Acanthochromis_polyacanthus.ASM210954v1.ncrna.fa.gz',
	'acaromyces_ingoldii_gca_003144295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/acaromyces_ingoldii_gca_003144295/ncrna/Acaromyces_ingoldii_gca_003144295.Acain1.ncrna.fa.gz',
	'acidomyces_richmondensis_bfw_gca_001592465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/acidomyces_richmondensis_bfw_gca_001592465/ncrna/Acidomyces_richmondensis_bfw_gca_001592465.Acidomyces_richmondensis_BFW_v1.0.ncrna.fa.gz',
	'acremonium_chrysogenum_atcc_11550_gca_000769265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/acremonium_chrysogenum_atcc_11550_gca_000769265/ncrna/Acremonium_chrysogenum_atcc_11550_gca_000769265.ASM76926v1.ncrna.fa.gz',
	'acyrthosiphon_pisum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/acyrthosiphon_pisum/ncrna/Acyrthosiphon_pisum.Acyr_2.0.ncrna.fa.gz',
	'adineta_vaga'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/adineta_vaga/ncrna/Adineta_vaga.AMS_PRJEB1171_v1.ncrna.fa.gz',
	'aedes_aegypti'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/aedes_aegypti/ncrna/Aedes_aegypti.AaegL3.ncrna.fa.gz',
	'aegilops_tauschii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/aegilops_tauschii/ncrna/Aegilops_tauschii.Aet_v4.0.ncrna.fa.gz',
	'agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555/ncrna/Agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555.Agabi_varbur_1.ncrna.fa.gz',
	'ailuropoda_melanoleuca'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ailuropoda_melanoleuca/ncrna/Ailuropoda_melanoleuca.ailMel1.ncrna.fa.gz',
	'albugo_laibachii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/albugo_laibachii/ncrna/Albugo_laibachii.ENA1.ncrna.fa.gz',
	'allomyces_macrogynus_atcc_38327_gca_000151295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_blastocladiomycota1_collection/allomyces_macrogynus_atcc_38327_gca_000151295/ncrna/Allomyces_macrogynus_atcc_38327_gca_000151295.A_macrogynus_V3.ncrna.fa.gz',
	'alternaria_alternata_gca_001642055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/alternaria_alternata_gca_001642055/ncrna/Alternaria_alternata_gca_001642055.Altal1.ncrna.fa.gz',
	'amanita_muscaria_koide_bx008_gca_000827485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/amanita_muscaria_koide_bx008_gca_000827485/ncrna/Amanita_muscaria_koide_bx008_gca_000827485.Amanita_muscaria_Koide_BX008_v1.0.ncrna.fa.gz',
	'amborella_trichopoda'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/amborella_trichopoda/ncrna/Amborella_trichopoda.AMTR1.0.ncrna.fa.gz',
	'amphiamblys_sp_wsbs2006_gca_001875675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/amphiamblys_sp_wsbs2006_gca_001875675/ncrna/Amphiamblys_sp_wsbs2006_gca_001875675.ASM187567v1.ncrna.fa.gz',
	'amphilophus_citrinellus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/amphilophus_citrinellus/ncrna/Amphilophus_citrinellus.Midas_v5.ncrna.fa.gz',
	'amphimedon_queenslandica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/amphimedon_queenslandica/ncrna/Amphimedon_queenslandica.Aqu1.ncrna.fa.gz',
	'amphiprion_ocellaris'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/amphiprion_ocellaris/ncrna/Amphiprion_ocellaris.AmpOce1.0.ncrna.fa.gz',
	'amphiprion_percula'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/amphiprion_percula/ncrna/Amphiprion_percula.Nemo_v1.ncrna.fa.gz',
	'anabas_testudineus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anabas_testudineus/ncrna/Anabas_testudineus.fAnaTes1.1.ncrna.fa.gz',
	'anaeromyces_robustus_gca_002104895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_neocallimastigomycota1_collection/anaeromyces_robustus_gca_002104895/ncrna/Anaeromyces_robustus_gca_002104895.Anaeromyces_sp._S4_v1.0.ncrna.fa.gz',
	'anas_platyrhynchos_platyrhynchos'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anas_platyrhynchos_platyrhynchos/ncrna/Anas_platyrhynchos_platyrhynchos.CAU_duck1.0.ncrna.fa.gz',
	'angomonas_deanei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/angomonas_deanei/ncrna/Angomonas_deanei.Angomonas_deanei_Genome.ncrna.fa.gz',
	'anncaliia_algerae_pra109_gca_000385855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/anncaliia_algerae_pra109_gca_000385855/ncrna/Anncaliia_algerae_pra109_gca_000385855.Annc_alge_PRA109_V4.ncrna.fa.gz',
	'anncaliia_algerae_pra339_gca_000385875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/anncaliia_algerae_pra339_gca_000385875/ncrna/Anncaliia_algerae_pra339_gca_000385875.Annc_alge_insect_USDA_JJB_V2.ncrna.fa.gz',
	'anolis_carolinensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anolis_carolinensis/ncrna/Anolis_carolinensis.AnoCar2.0.ncrna.fa.gz',
	'anopheles_darlingi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/anopheles_darlingi/ncrna/Anopheles_darlingi.AdarC3.ncrna.fa.gz',
	'anopheles_gambiae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/anopheles_gambiae/ncrna/Anopheles_gambiae.AgamP4.ncrna.fa.gz',
	'anoplophora_glabripennis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/anoplophora_glabripennis/ncrna/Anoplophora_glabripennis.Agla_1.0.ncrna.fa.gz',
	'anser_brachyrhynchus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anser_brachyrhynchus/ncrna/Anser_brachyrhynchus.ASM259213v1.ncrna.fa.gz',
	'anthracocystis_flocculosa_pf_1_gca_000417875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/anthracocystis_flocculosa_pf_1_gca_000417875/ncrna/Anthracocystis_flocculosa_pf_1_gca_000417875.Pflocc_1.0.ncrna.fa.gz',
	'aotus_nancymaae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/aotus_nancymaae/ncrna/Aotus_nancymaae.Anan_2.0.ncrna.fa.gz',
	'aphanomyces_astaci'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/aphanomyces_astaci/ncrna/Aphanomyces_astaci.Apha_asta_APO3_V1.ncrna.fa.gz',
	'aphanomyces_invadans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/aphanomyces_invadans/ncrna/Aphanomyces_invadans.Apha_inva_NJM9701_V1.ncrna.fa.gz',
	'apis_mellifera'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/apis_mellifera/ncrna/Apis_mellifera.Amel_4.5.ncrna.fa.gz',
	'apteryx_haastii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/apteryx_haastii/ncrna/Apteryx_haastii.aptHaa1.ncrna.fa.gz',
	'apteryx_owenii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/apteryx_owenii/ncrna/Apteryx_owenii.aptOwe1.ncrna.fa.gz',
	'apteryx_rowi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/apteryx_rowi/ncrna/Apteryx_rowi.aptRow1.ncrna.fa.gz',
	'arabidopsis_halleri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/arabidopsis_halleri/ncrna/Arabidopsis_halleri.Ahal2.2.ncrna.fa.gz',
	'arabidopsis_lyrata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/arabidopsis_lyrata/ncrna/Arabidopsis_lyrata.v.1.0.ncrna.fa.gz',
	'arabidopsis_thaliana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/arabidopsis_thaliana/ncrna/Arabidopsis_thaliana.TAIR10.ncrna.fa.gz',
	'armillaria_gallica_gca_002307695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/armillaria_gallica_gca_002307695/ncrna/Armillaria_gallica_gca_002307695.Armga1.ncrna.fa.gz',
	'armillaria_ostoyae_gca_900157425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/armillaria_ostoyae_gca_900157425/ncrna/Armillaria_ostoyae_gca_900157425.version_2.ncrna.fa.gz',
	'armillaria_solidipes_gca_002307675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/armillaria_solidipes_gca_002307675/ncrna/Armillaria_solidipes_gca_002307675.Armost1.ncrna.fa.gz',
	'arthrobotrys_oligospora_atcc_24927_gca_000225545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/arthrobotrys_oligospora_atcc_24927_gca_000225545/ncrna/Arthrobotrys_oligospora_atcc_24927_gca_000225545.AOL24927_1.0.ncrna.fa.gz',
	'ascochyta_rabiei_gca_001630375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/ascochyta_rabiei_gca_001630375/ncrna/Ascochyta_rabiei_gca_001630375.AscRab1.0.ncrna.fa.gz',
	'ascoidea_rubescens_dsm_1968_gca_001661345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/ascoidea_rubescens_dsm_1968_gca_001661345/ncrna/Ascoidea_rubescens_dsm_1968_gca_001661345.Ascru1.ncrna.fa.gz',
	'ashbya_gossypii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/ashbya_gossypii/ncrna/Ashbya_gossypii.ASM9102v1.ncrna.fa.gz',
	'aspergillus_aculeatinus_cbs_121060_gca_003184765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_aculeatinus_cbs_121060_gca_003184765/ncrna/Aspergillus_aculeatinus_cbs_121060_gca_003184765.Aspacu1.ncrna.fa.gz',
	'aspergillus_aculeatus_atcc_16872_gca_001890905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_aculeatus_atcc_16872_gca_001890905/ncrna/Aspergillus_aculeatus_atcc_16872_gca_001890905.Aspac1.ncrna.fa.gz',
	'aspergillus_brasiliensis_cbs_101740_gca_001889945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_brasiliensis_cbs_101740_gca_001889945/ncrna/Aspergillus_brasiliensis_cbs_101740_gca_001889945.Aspbr1.ncrna.fa.gz',
	'aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695/ncrna/Aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695.Aspbru1.ncrna.fa.gz',
	'aspergillus_calidoustus_gca_001511075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_calidoustus_gca_001511075/ncrna/Aspergillus_calidoustus_gca_001511075.Acal_Allpaths_LG.ncrna.fa.gz',
	'aspergillus_campestris_ibt_28561_gca_002847485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_campestris_ibt_28561_gca_002847485/ncrna/Aspergillus_campestris_ibt_28561_gca_002847485.Aspcam1.ncrna.fa.gz',
	'aspergillus_candidus_gca_002847045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_candidus_gca_002847045/ncrna/Aspergillus_candidus_gca_002847045.Aspcand1.ncrna.fa.gz',
	'aspergillus_carbonarius_item_5010_gca_001990825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_carbonarius_item_5010_gca_001990825/ncrna/Aspergillus_carbonarius_item_5010_gca_001990825.Aspca3.ncrna.fa.gz',
	'aspergillus_clavatus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_clavatus/ncrna/Aspergillus_clavatus.ASM271v1.ncrna.fa.gz',
	'aspergillus_cristatus_gca_001717485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_cristatus_gca_001717485/ncrna/Aspergillus_cristatus_gca_001717485.ASM171748v1.ncrna.fa.gz',
	'aspergillus_ellipticus_cbs_707_79_gca_003184645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_ellipticus_cbs_707_79_gca_003184645/ncrna/Aspergillus_ellipticus_cbs_707_79_gca_003184645.Aspell1.ncrna.fa.gz',
	'aspergillus_eucalypticola_cbs_122712_gca_003184535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_eucalypticola_cbs_122712_gca_003184535/ncrna/Aspergillus_eucalypticola_cbs_122712_gca_003184535.Aspeuc1.ncrna.fa.gz',
	'aspergillus_flavus_af70_gca_000952835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_flavus_af70_gca_000952835/ncrna/Aspergillus_flavus_af70_gca_000952835.ASM95283v1.ncrna.fa.gz',
	'aspergillus_fumigatus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_fumigatus/ncrna/Aspergillus_fumigatus.ASM265v1.ncrna.fa.gz',
	'aspergillus_fumigatus_gca_002234985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_fumigatus_gca_002234985/ncrna/Aspergillus_fumigatus_gca_002234985.ASM223498v1.ncrna.fa.gz',
	'aspergillus_fumigatusa1163'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_fumigatusa1163/ncrna/Aspergillus_fumigatusa1163.ASM15014v1.ncrna.fa.gz',
	'aspergillus_glaucus_cbs_516_65_gca_001890805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_glaucus_cbs_516_65_gca_001890805/ncrna/Aspergillus_glaucus_cbs_516_65_gca_001890805.Aspgl1.ncrna.fa.gz',
	'aspergillus_heteromorphus_cbs_117_55_gca_003184545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_heteromorphus_cbs_117_55_gca_003184545/ncrna/Aspergillus_heteromorphus_cbs_117_55_gca_003184545.Asphet1.ncrna.fa.gz',
	'aspergillus_indologenus_cbs_114_80_gca_003184685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_indologenus_cbs_114_80_gca_003184685/ncrna/Aspergillus_indologenus_cbs_114_80_gca_003184685.Aspind1.ncrna.fa.gz',
	'aspergillus_japonicus_cbs_114_51_gca_003184785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_japonicus_cbs_114_51_gca_003184785/ncrna/Aspergillus_japonicus_cbs_114_51_gca_003184785.Aspjap1.ncrna.fa.gz',
	'aspergillus_lacticoffeatus_cbs_101883_gca_003184595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_lacticoffeatus_cbs_101883_gca_003184595/ncrna/Aspergillus_lacticoffeatus_cbs_101883_gca_003184595.Asplac1.ncrna.fa.gz',
	'aspergillus_lentulus_gca_001445615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_lentulus_gca_001445615/ncrna/Aspergillus_lentulus_gca_001445615.Alt_assembly01.ncrna.fa.gz',
	'aspergillus_luchuensis_cbs_106_47_gca_001890685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_luchuensis_cbs_106_47_gca_001890685/ncrna/Aspergillus_luchuensis_cbs_106_47_gca_001890685.Aspfo1.ncrna.fa.gz',
	'aspergillus_luchuensis_gca_001602395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_luchuensis_gca_001602395/ncrna/Aspergillus_luchuensis_gca_001602395.AspL_2604.ncrna.fa.gz',
	'aspergillus_neoniger_cbs_115656_gca_003184625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_neoniger_cbs_115656_gca_003184625/ncrna/Aspergillus_neoniger_cbs_115656_gca_003184625.Aspneo1.ncrna.fa.gz',
	'aspergillus_nidulans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_nidulans/ncrna/Aspergillus_nidulans.ASM1142v1.ncrna.fa.gz',
	'aspergillus_nidulans_fgsc_a4_gca_000149205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_nidulans_fgsc_a4_gca_000149205/ncrna/Aspergillus_nidulans_fgsc_a4_gca_000149205.ASM14920v2.ncrna.fa.gz',
	'aspergillus_niger'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_niger/ncrna/Aspergillus_niger.ASM285v2.ncrna.fa.gz',
	'aspergillus_niger_atcc_1015_gca_000230395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aspergillus_niger_atcc_1015_gca_000230395/ncrna/Aspergillus_niger_atcc_1015_gca_000230395.ASPNI_v3.0.ncrna.fa.gz',
	'aspergillus_niger_gca_001515345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_niger_gca_001515345/ncrna/Aspergillus_niger_gca_001515345.ASM151534v1.ncrna.fa.gz',
	'aspergillus_nomius_nrrl_13137_gca_001204775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_nomius_nrrl_13137_gca_001204775/ncrna/Aspergillus_nomius_nrrl_13137_gca_001204775.ASM120477v1.ncrna.fa.gz',
	'aspergillus_novofumigatus_ibt_16806_gca_002847465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_novofumigatus_ibt_16806_gca_002847465/ncrna/Aspergillus_novofumigatus_ibt_16806_gca_002847465.Aspnov1.ncrna.fa.gz',
	'aspergillus_ochraceoroseus_ibt_24754_gca_002846915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_ochraceoroseus_ibt_24754_gca_002846915/ncrna/Aspergillus_ochraceoroseus_ibt_24754_gca_002846915.Aspergillus_ochraceoroseus_IBT_24754_v1.1.ncrna.fa.gz',
	'aspergillus_oryzae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_oryzae/ncrna/Aspergillus_oryzae.ASM18445v3.ncrna.fa.gz',
	'aspergillus_oryzae_3_042_gca_000269785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aspergillus_oryzae_3_042_gca_000269785/ncrna/Aspergillus_oryzae_3_042_gca_000269785.AspOry3042.ncrna.fa.gz',
	'aspergillus_parasiticus_su_1_gca_000956085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_parasiticus_su_1_gca_000956085/ncrna/Aspergillus_parasiticus_su_1_gca_000956085.ASM95608v1.ncrna.fa.gz',
	'aspergillus_piperis_cbs_112811_gca_003184755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_piperis_cbs_112811_gca_003184755/ncrna/Aspergillus_piperis_cbs_112811_gca_003184755.Asppip1.ncrna.fa.gz',
	'aspergillus_rambellii_gca_000986645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_rambellii_gca_000986645/ncrna/Aspergillus_rambellii_gca_000986645.ASM98664v1.ncrna.fa.gz',
	'aspergillus_ruber_cbs_135680_gca_000600275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aspergillus_ruber_cbs_135680_gca_000600275/ncrna/Aspergillus_ruber_cbs_135680_gca_000600275.Eurhe1.ncrna.fa.gz',
	'aspergillus_saccharolyticus_jop_1030_1_gca_003184585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_saccharolyticus_jop_1030_1_gca_003184585/ncrna/Aspergillus_saccharolyticus_jop_1030_1_gca_003184585.Aspsac1.ncrna.fa.gz',
	'aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635/ncrna/Aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635.Aspscle1.ncrna.fa.gz',
	'aspergillus_sclerotioniger_cbs_115572_gca_003184525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_sclerotioniger_cbs_115572_gca_003184525/ncrna/Aspergillus_sclerotioniger_cbs_115572_gca_003184525.Aspscl1.ncrna.fa.gz',
	'aspergillus_steynii_ibt_23096_gca_002849105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_steynii_ibt_23096_gca_002849105/ncrna/Aspergillus_steynii_ibt_23096_gca_002849105.Aspste1.ncrna.fa.gz',
	'aspergillus_sydowii_cbs_593_65_gca_001890705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_sydowii_cbs_593_65_gca_001890705/ncrna/Aspergillus_sydowii_cbs_593_65_gca_001890705.Aspsy1.ncrna.fa.gz',
	'aspergillus_taichungensis_gca_002850765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_taichungensis_gca_002850765/ncrna/Aspergillus_taichungensis_gca_002850765.Asptaic1.ncrna.fa.gz',
	'aspergillus_terreus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_terreus/ncrna/Aspergillus_terreus.ASM14961v1.ncrna.fa.gz',
	'aspergillus_thermomutatus_gca_002237265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_thermomutatus_gca_002237265/ncrna/Aspergillus_thermomutatus_gca_002237265.ASM223726v1.ncrna.fa.gz',
	'aspergillus_tubingensis_cbs_134_48_gca_001890745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_tubingensis_cbs_134_48_gca_001890745/ncrna/Aspergillus_tubingensis_cbs_134_48_gca_001890745.Asptu1.ncrna.fa.gz',
	'aspergillus_turcosus_gca_002234965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_turcosus_gca_002234965/ncrna/Aspergillus_turcosus_gca_002234965.ASM223496v1.ncrna.fa.gz',
	'aspergillus_turcosus_gca_002234975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_turcosus_gca_002234975/ncrna/Aspergillus_turcosus_gca_002234975.ASM223497v1.ncrna.fa.gz',
	'aspergillus_udagawae_gca_001078395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_udagawae_gca_001078395/ncrna/Aspergillus_udagawae_gca_001078395.Aud_assembly01.ncrna.fa.gz',
	'aspergillus_ustus_gca_000812125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_ustus_gca_000812125/ncrna/Aspergillus_ustus_gca_000812125.Austus1.0.ncrna.fa.gz',
	'aspergillus_uvarum_cbs_121591_gca_003184745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_uvarum_cbs_121591_gca_003184745/ncrna/Aspergillus_uvarum_cbs_121591_gca_003184745.Aspuva1.ncrna.fa.gz',
	'aspergillus_vadensis_cbs_113365_gca_003184925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_vadensis_cbs_113365_gca_003184925/ncrna/Aspergillus_vadensis_cbs_113365_gca_003184925.Aspvad1.ncrna.fa.gz',
	'aspergillus_versicolor_cbs_583_65_gca_001890125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_versicolor_cbs_583_65_gca_001890125/ncrna/Aspergillus_versicolor_cbs_583_65_gca_001890125.Aspve1.ncrna.fa.gz',
	'aspergillus_violaceofuscus_cbs_115571_gca_003184705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_violaceofuscus_cbs_115571_gca_003184705/ncrna/Aspergillus_violaceofuscus_cbs_115571_gca_003184705.Aspvio1.ncrna.fa.gz',
	'aspergillus_wentii_dto_134e9_gca_001890725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_wentii_dto_134e9_gca_001890725/ncrna/Aspergillus_wentii_dto_134e9_gca_001890725.Aspwe1.ncrna.fa.gz',
	'astatotilapia_calliptera'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/astatotilapia_calliptera/ncrna/Astatotilapia_calliptera.fAstCal1.2.ncrna.fa.gz',
	'astyanax_mexicanus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/astyanax_mexicanus/ncrna/Astyanax_mexicanus.Astyanax_mexicanus-2.0.ncrna.fa.gz',
	'atta_cephalotes'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/atta_cephalotes/ncrna/Atta_cephalotes.Attacep1.0.ncrna.fa.gz',
	'aureobasidium_melanogenum_cbs_110374_gca_000721775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_melanogenum_cbs_110374_gca_000721775/ncrna/Aureobasidium_melanogenum_cbs_110374_gca_000721775.Aureobasidium_pullulans_var._melanogenum_CBS_110374_v1.0.ncrna.fa.gz',
	'aureobasidium_namibiae_cbs_147_97_gca_000721765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_namibiae_cbs_147_97_gca_000721765/ncrna/Aureobasidium_namibiae_cbs_147_97_gca_000721765.Aureobasidium_pullulans_var._namibiae_CBS_147.97_v1.0.ncrna.fa.gz',
	'aureobasidium_pullulans_exf_150_gca_000721785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_pullulans_exf_150_gca_000721785/ncrna/Aureobasidium_pullulans_exf_150_gca_000721785.Aureobasidium_pullulans_var._pullulans_EXF-150_assembly_version_1.0.ncrna.fa.gz',
	'aureobasidium_subglaciale_exf_2481_gca_000721755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_subglaciale_exf_2481_gca_000721755/ncrna/Aureobasidium_subglaciale_exf_2481_gca_000721755.Aureobasidium_pullulans_var._subglaciale_EXF-2481_v1.0.ncrna.fa.gz',
	'babesia_bovis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/babesia_bovis/ncrna/Babesia_bovis.ASM16539v1.ncrna.fa.gz',
	'babesia_microti_strain_ri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/babesia_microti_strain_ri/ncrna/Babesia_microti_strain_ri.ASM69194v1.ncrna.fa.gz',
	'babjeviella_inositovora_nrrl_y_12698_gca_001661335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/babjeviella_inositovora_nrrl_y_12698_gca_001661335/ncrna/Babjeviella_inositovora_nrrl_y_12698_gca_001661335.Babin1.ncrna.fa.gz',
	'basidiobolus_meristosporus_cbs_931_73_gca_002104905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_entomophthoromycota1_collection/basidiobolus_meristosporus_cbs_931_73_gca_002104905/ncrna/Basidiobolus_meristosporus_cbs_931_73_gca_002104905.Basme2finSC.ncrna.fa.gz',
	'batrachochytrium_dendrobatidis_jam81_gca_000203795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/batrachochytrium_dendrobatidis_jam81_gca_000203795/ncrna/Batrachochytrium_dendrobatidis_jam81_gca_000203795.v1.0.ncrna.fa.gz',
	'batrachochytrium_dendrobatidis_jel423_gca_000149865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/batrachochytrium_dendrobatidis_jel423_gca_000149865/ncrna/Batrachochytrium_dendrobatidis_jel423_gca_000149865.BD_JEL423.ncrna.fa.gz',
	'batrachochytrium_salamandrivorans_gca_002006685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/batrachochytrium_salamandrivorans_gca_002006685/ncrna/Batrachochytrium_salamandrivorans_gca_002006685.Batr_sala_BS_V1.ncrna.fa.gz',
	'baudoinia_panamericana_uamh_10762_gca_000338955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/baudoinia_panamericana_uamh_10762_gca_000338955/ncrna/Baudoinia_panamericana_uamh_10762_gca_000338955.Bauco1.ncrna.fa.gz',
	'beauveria_bassiana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/beauveria_bassiana/ncrna/Beauveria_bassiana.ASM168263v1.ncrna.fa.gz',
	'beauveria_bassiana_arsef_2860_gca_000280675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/beauveria_bassiana_arsef_2860_gca_000280675/ncrna/Beauveria_bassiana_arsef_2860_gca_000280675.ASM28067v1.ncrna.fa.gz',
	'beauveria_bassiana_d1_5_gca_000770705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/beauveria_bassiana_d1_5_gca_000770705/ncrna/Beauveria_bassiana_d1_5_gca_000770705.BBA1.0.ncrna.fa.gz',
	'beauveria_bassiana_gca_002871155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/beauveria_bassiana_gca_002871155/ncrna/Beauveria_bassiana_gca_002871155.ASM287115v1.ncrna.fa.gz',
	'belgica_antarctica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/belgica_antarctica/ncrna/Belgica_antarctica.ASM77530v1.ncrna.fa.gz',
	'beta_vulgaris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/beta_vulgaris/ncrna/Beta_vulgaris.RefBeet-1.2.2.ncrna.fa.gz',
	'bifiguratus_adelaidae_gca_002261195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/bifiguratus_adelaidae_gca_002261195/ncrna/Bifiguratus_adelaidae_gca_002261195.ASM226119v1.ncrna.fa.gz',
	'bigelowiella_natans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/bigelowiella_natans/ncrna/Bigelowiella_natans.Bigna1.ncrna.fa.gz',
	'bigelowiella_natans_gca_000002455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/bigelowiella_natans_gca_000002455/ncrna/Bigelowiella_natans_gca_000002455.ASM245v1.ncrna.fa.gz',
	'bipolaris_maydis_atcc_48331_gca_000354255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_maydis_atcc_48331_gca_000354255/ncrna/Bipolaris_maydis_atcc_48331_gca_000354255.CocheC4_1.ncrna.fa.gz',
	'bipolaris_maydis_c5_gca_000338975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_maydis_c5_gca_000338975/ncrna/Bipolaris_maydis_c5_gca_000338975.CocheC5_3.ncrna.fa.gz',
	'bipolaris_sorokiniana_nd90pr_gca_000338995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_sorokiniana_nd90pr_gca_000338995/ncrna/Bipolaris_sorokiniana_nd90pr_gca_000338995.Cocsa1.ncrna.fa.gz',
	'bipolaris_zeicola_26_r_13_gca_000523435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_zeicola_26_r_13_gca_000523435/ncrna/Bipolaris_zeicola_26_r_13_gca_000523435.Cochliobolus_carbonum_v1.0.ncrna.fa.gz',
	'bison_bison_bison'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/bison_bison_bison/ncrna/Bison_bison_bison.Bison_UMD1.0.ncrna.fa.gz',
	'blastomyces_dermatitidis_atcc_18188_gca_000151595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_dermatitidis_atcc_18188_gca_000151595/ncrna/Blastomyces_dermatitidis_atcc_18188_gca_000151595.BD_ATCC18188_V1.ncrna.fa.gz',
	'blastomyces_dermatitidis_atcc_26199_gca_000166155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_dermatitidis_atcc_26199_gca_000166155/ncrna/Blastomyces_dermatitidis_atcc_26199_gca_000166155.BD_ATCC26199_V2.ncrna.fa.gz',
	'blastomyces_dermatitidis_er_3_gca_000003525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_dermatitidis_er_3_gca_000003525/ncrna/Blastomyces_dermatitidis_er_3_gca_000003525.BD_ER3_V1.ncrna.fa.gz',
	'blastomyces_gilchristii_slh14081_gca_000003855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_gilchristii_slh14081_gca_000003855/ncrna/Blastomyces_gilchristii_slh14081_gca_000003855.BD_SLH14081_V1.ncrna.fa.gz',
	'blastomyces_percursus_gca_001883805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/blastomyces_percursus_gca_001883805/ncrna/Blastomyces_percursus_gca_001883805.Blas_perc_EI222_V1.ncrna.fa.gz',
	'blumeria_graminis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/blumeria_graminis/ncrna/Blumeria_graminis.EF2.ncrna.fa.gz',
	'blumeria_graminis_f_sp_tritici_96224_gca_000418435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blumeria_graminis_f_sp_tritici_96224_gca_000418435/ncrna/Blumeria_graminis_f_sp_tritici_96224_gca_000418435.Bgt_454_newbler_assembly.ncrna.fa.gz',
	'bodo_saltans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/bodo_saltans/ncrna/Bodo_saltans.BSAL.ncrna.fa.gz',
	'bombus_impatiens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/bombus_impatiens/ncrna/Bombus_impatiens.BIMP_2.0.ncrna.fa.gz',
	'bombus_terrestris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/bombus_terrestris/ncrna/Bombus_terrestris.Bter_1.0.ncrna.fa.gz',
	'bombyx_mori'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/bombyx_mori/ncrna/Bombyx_mori.ASM15162v1.ncrna.fa.gz',
	'bos_mutus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/bos_mutus/ncrna/Bos_mutus.BosGru_v2.0.ncrna.fa.gz',
	'bos_taurus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/bos_taurus/ncrna/Bos_taurus.ARS-UCD1.2.ncrna.fa.gz',
	'botryobasidium_botryosum_fd_172_ss1_gca_000697705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/botryobasidium_botryosum_fd_172_ss1_gca_000697705/ncrna/Botryobasidium_botryosum_fd_172_ss1_gca_000697705.Botbo1.ncrna.fa.gz',
	'botrytis_cinerea'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/botrytis_cinerea/ncrna/Botrytis_cinerea.ASM83294v1.ncrna.fa.gz',
	'botrytis_cinerea_bcdw1_gca_000349525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/botrytis_cinerea_bcdw1_gca_000349525/ncrna/Botrytis_cinerea_bcdw1_gca_000349525.BcDW1.ncrna.fa.gz',
	'botrytis_cinerea_t4_gca_000227075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/botrytis_cinerea_t4_gca_000227075/ncrna/Botrytis_cinerea_t4_gca_000227075.BotFuc_Mar2011.ncrna.fa.gz',
	'brachypodium_distachyon'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brachypodium_distachyon/ncrna/Brachypodium_distachyon.Brachypodium_distachyon_v3.0.ncrna.fa.gz',
	'brassica_napus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brassica_napus/ncrna/Brassica_napus.AST_PRJEB5043_v1.ncrna.fa.gz',
	'brassica_oleracea'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brassica_oleracea/ncrna/Brassica_oleracea.BOL.ncrna.fa.gz',
	'brassica_rapa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brassica_rapa/ncrna/Brassica_rapa.Brapa_1.0.ncrna.fa.gz',
	'brettanomyces_bruxellensis_awri1499_gca_000259595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/brettanomyces_bruxellensis_awri1499_gca_000259595/ncrna/Brettanomyces_bruxellensis_awri1499_gca_000259595.AWRI1499_v1.0.ncrna.fa.gz',
	'brugia_malayi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/brugia_malayi/ncrna/Brugia_malayi.Bmal-4.0.ncrna.fa.gz',
	'byssochlamys_spectabilis_no_5_gca_000497085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/byssochlamys_spectabilis_no_5_gca_000497085/ncrna/Byssochlamys_spectabilis_no_5_gca_000497085.PVAR5_assembly01.ncrna.fa.gz',
	'cadophora_sp_dse1049_gca_003073865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/cadophora_sp_dse1049_gca_003073865/ncrna/Cadophora_sp_dse1049_gca_003073865.Cadsp1.ncrna.fa.gz',
	'caenorhabditis_brenneri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_brenneri/ncrna/Caenorhabditis_brenneri.C_brenneri-6.0.1b.ncrna.fa.gz',
	'caenorhabditis_briggsae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_briggsae/ncrna/Caenorhabditis_briggsae.CB4.ncrna.fa.gz',
	'caenorhabditis_elegans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_elegans/ncrna/Caenorhabditis_elegans.WBcel235.ncrna.fa.gz',
	'caenorhabditis_japonica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_japonica/ncrna/Caenorhabditis_japonica.C_japonica-7.0.1.ncrna.fa.gz',
	'caenorhabditis_remanei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_remanei/ncrna/Caenorhabditis_remanei.C_remanei-15.0.1.ncrna.fa.gz',
	'calidris_pugnax'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/calidris_pugnax/ncrna/Calidris_pugnax.ASM143184v1.ncrna.fa.gz',
	'calidris_pygmaea'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/calidris_pygmaea/ncrna/Calidris_pygmaea.ASM369795v1.ncrna.fa.gz',
	'callithrix_jacchus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/callithrix_jacchus/ncrna/Callithrix_jacchus.ASM275486v1.ncrna.fa.gz',
	'calocera_cornea_hhb12733_gca_001632435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/calocera_cornea_hhb12733_gca_001632435/ncrna/Calocera_cornea_hhb12733_gca_001632435.Calco1.ncrna.fa.gz',
	'calocera_viscosa_tufc12733_gca_001630345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/calocera_viscosa_tufc12733_gca_001630345/ncrna/Calocera_viscosa_tufc12733_gca_001630345.Calvi1.ncrna.fa.gz',
	'candida_albicans_12c_gca_000773845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_12c_gca_000773845/ncrna/Candida_albicans_12c_gca_000773845.Cand_albi_12C_V2.ncrna.fa.gz',
	'candida_albicans_19f_gca_000775445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_19f_gca_000775445/ncrna/Candida_albicans_19f_gca_000775445.Cand_albi_19F_V2.ncrna.fa.gz',
	'candida_albicans_ca6_gca_000784695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_ca6_gca_000784695/ncrna/Candida_albicans_ca6_gca_000784695.Cand_albi_Ca6_V2.ncrna.fa.gz',
	'candida_albicans_gc75_gca_000773735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_gc75_gca_000773735/ncrna/Candida_albicans_gc75_gca_000773735.Cand_albi_GC75_V1.ncrna.fa.gz',
	'candida_albicans_l26_gca_000775455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_l26_gca_000775455/ncrna/Candida_albicans_l26_gca_000775455.Cand_albi_L26_V2.ncrna.fa.gz',
	'candida_albicans_p34048_gca_000775465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p34048_gca_000775465/ncrna/Candida_albicans_p34048_gca_000775465.Cand_albi_P34048_V1.ncrna.fa.gz',
	'candida_albicans_p37005_gca_000773745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p37005_gca_000773745/ncrna/Candida_albicans_p37005_gca_000773745.Cand_albi_P37005_V2.ncrna.fa.gz',
	'candida_albicans_p37037_gca_000773825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p37037_gca_000773825/ncrna/Candida_albicans_p37037_gca_000773825.Cand_albi_P37037_V1.ncrna.fa.gz',
	'candida_albicans_p37039_gca_000784515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p37039_gca_000784515/ncrna/Candida_albicans_p37039_gca_000784515.Cand_albi_P37039_V1.ncrna.fa.gz',
	'candida_albicans_p57055_gca_000775505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p57055_gca_000775505/ncrna/Candida_albicans_p57055_gca_000775505.Cand_albi_P57055_V1.ncrna.fa.gz',
	'candida_albicans_p57072_gca_000773805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p57072_gca_000773805/ncrna/Candida_albicans_p57072_gca_000773805.Cand_albi_P57072_V1.ncrna.fa.gz',
	'candida_albicans_p60002_gca_000784525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p60002_gca_000784525/ncrna/Candida_albicans_p60002_gca_000784525.Cand_albi_P60002_V1.ncrna.fa.gz',
	'candida_albicans_p75010_gca_000784575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p75010_gca_000784575/ncrna/Candida_albicans_p75010_gca_000784575.Cand_albi_P75010_V1.ncrna.fa.gz',
	'candida_albicans_p75016_gca_000784595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p75016_gca_000784595/ncrna/Candida_albicans_p75016_gca_000784595.Cand_albi_P75016_V1.ncrna.fa.gz',
	'candida_albicans_p75063_gca_000775525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p75063_gca_000775525/ncrna/Candida_albicans_p75063_gca_000775525.Cand_albi_P75063_V1.ncrna.fa.gz',
	'candida_albicans_p76055_gca_000784505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p76055_gca_000784505/ncrna/Candida_albicans_p76055_gca_000784505.Cand_albi_P76055_V1.ncrna.fa.gz',
	'candida_albicans_p76067_gca_000784495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p76067_gca_000784495/ncrna/Candida_albicans_p76067_gca_000784495.Cand_albi_P76067_V1.ncrna.fa.gz',
	'candida_albicans_p78042_gca_000784615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p78042_gca_000784615/ncrna/Candida_albicans_p78042_gca_000784615.Cand_albi_P78042_V1.ncrna.fa.gz',
	'candida_albicans_p78048_gca_000773725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p78048_gca_000773725/ncrna/Candida_albicans_p78048_gca_000773725.Cand_albi_P78048_V1.ncrna.fa.gz',
	'candida_albicans_p87_gca_000774085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p87_gca_000774085/ncrna/Candida_albicans_p87_gca_000774085.Cand_albi_P87_V2.ncrna.fa.gz',
	'candida_albicans_p94015_gca_000773755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p94015_gca_000773755/ncrna/Candida_albicans_p94015_gca_000773755.Cand_albi_P94015_V1.ncrna.fa.gz',
	'candida_albicans_sc5314_gca_000784635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_sc5314_gca_000784635/ncrna/Candida_albicans_sc5314_gca_000784635.Cand_albi_SC5314_V4.ncrna.fa.gz',
	'candida_albicans_sc5314_gca_000784655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_sc5314_gca_000784655/ncrna/Candida_albicans_sc5314_gca_000784655.Cand_albi_SC5314_V3.ncrna.fa.gz',
	'candida_albicans_wo_1_gca_000149445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_wo_1_gca_000149445/ncrna/Candida_albicans_wo_1_gca_000149445.ASM14944v2.ncrna.fa.gz',
	'candida_dubliniensis_cd36_gca_000026945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_dubliniensis_cd36_gca_000026945/ncrna/Candida_dubliniensis_cd36_gca_000026945.ASM2694v1.ncrna.fa.gz',
	'candida_maltosa_xu316_gca_000344705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_maltosa_xu316_gca_000344705/ncrna/Candida_maltosa_xu316_gca_000344705.Cmaltosa_v01.ncrna.fa.gz',
	'candida_orthopsilosis_co_90_125_gca_000315875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_orthopsilosis_co_90_125_gca_000315875/ncrna/Candida_orthopsilosis_co_90_125_gca_000315875.ASM31587v1.ncrna.fa.gz',
	'candida_parapsilosis_gca_000182765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/candida_parapsilosis_gca_000182765/ncrna/Candida_parapsilosis_gca_000182765.ASM18276v2.ncrna.fa.gz',
	'candida_tropicalis_mya_3404_gca_000006335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_tropicalis_mya_3404_gca_000006335/ncrna/Candida_tropicalis_mya_3404_gca_000006335.ASM633v3.ncrna.fa.gz',
	'canis_familiaris'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/canis_familiaris/ncrna/Canis_familiaris.CanFam3.1.ncrna.fa.gz',
	'canis_lupus_dingo'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/canis_lupus_dingo/ncrna/Canis_lupus_dingo.ASM325472v1.ncrna.fa.gz',
	'capitella_teleta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/capitella_teleta/ncrna/Capitella_teleta.Capitella_teleta_v1.0.ncrna.fa.gz',
	'capra_hircus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/capra_hircus/ncrna/Capra_hircus.ARS1.ncrna.fa.gz',
	'capronia_coronata_cbs_617_96_gca_000585585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/capronia_coronata_cbs_617_96_gca_000585585/ncrna/Capronia_coronata_cbs_617_96_gca_000585585.Capr_coro_CBS_617_96_V1.ncrna.fa.gz',
	'capronia_epimyces_cbs_606_96_gca_000585565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/capronia_epimyces_cbs_606_96_gca_000585565/ncrna/Capronia_epimyces_cbs_606_96_gca_000585565.Capr_epim_CBS_606_96_V1.ncrna.fa.gz',
	'capsaspora_owczarzaki_atcc_30864'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_ichthyosporea1_collection/capsaspora_owczarzaki_atcc_30864/ncrna/Capsaspora_owczarzaki_atcc_30864.C_owczarzaki_V2.ncrna.fa.gz',
	'carlito_syrichta'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/carlito_syrichta/ncrna/Carlito_syrichta.Tarsius_syrichta-2.0.1.ncrna.fa.gz',
	'castor_canadensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/castor_canadensis/ncrna/Castor_canadensis.C.can_genome_v1.0.ncrna.fa.gz',
	'catenaria_anguillulae_pl171_gca_002102555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_blastocladiomycota1_collection/catenaria_anguillulae_pl171_gca_002102555/ncrna/Catenaria_anguillulae_pl171_gca_002102555.Catan2.ncrna.fa.gz',
	'cavia_aperea'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cavia_aperea/ncrna/Cavia_aperea.CavAp1.0.ncrna.fa.gz',
	'cavia_porcellus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cavia_porcellus/ncrna/Cavia_porcellus.Cavpor3.0.ncrna.fa.gz',
	'cebus_capucinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cebus_capucinus/ncrna/Cebus_capucinus.Cebus_imitator-1.0.ncrna.fa.gz',
	'ceraceosorus_bombacis_gca_900000165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ceraceosorus_bombacis_gca_900000165/ncrna/Ceraceosorus_bombacis_gca_900000165.Ceraceosorus_bombacis.ncrna.fa.gz',
	'ceraceosorus_guamensis_gca_003144195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ceraceosorus_guamensis_gca_003144195/ncrna/Ceraceosorus_guamensis_gca_003144195.Cersp1.ncrna.fa.gz',
	'ceratocystis_platani_gca_000978885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/ceratocystis_platani_gca_000978885/ncrna/Ceratocystis_platani_gca_000978885.ASM97888v1.ncrna.fa.gz',
	'cercocebus_atys'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cercocebus_atys/ncrna/Cercocebus_atys.Caty_1.0.ncrna.fa.gz',
	'chaetomium_globosum_cbs_148_51_gca_000143365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/chaetomium_globosum_cbs_148_51_gca_000143365/ncrna/Chaetomium_globosum_cbs_148_51_gca_000143365.ASM14336v1.ncrna.fa.gz',
	'chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225/ncrna/Chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225.CTHT_3.0.ncrna.fa.gz',
	'chelonoidis_abingdonii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chelonoidis_abingdonii/ncrna/Chelonoidis_abingdonii.ASM359739v1.ncrna.fa.gz',
	'chinchilla_lanigera'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chinchilla_lanigera/ncrna/Chinchilla_lanigera.ChiLan1.0.ncrna.fa.gz',
	'chlamydomonas_reinhardtii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/chlamydomonas_reinhardtii/ncrna/Chlamydomonas_reinhardtii.Chlamydomonas_reinhardtii_v5.5.ncrna.fa.gz',
	'chlorocebus_sabaeus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chlorocebus_sabaeus/ncrna/Chlorocebus_sabaeus.ChlSab1.1.ncrna.fa.gz',
	'choanephora_cucurbitarum_gca_001683725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/choanephora_cucurbitarum_gca_001683725/ncrna/Choanephora_cucurbitarum_gca_001683725.ASM168372v1.ncrna.fa.gz',
	'choloepus_hoffmanni'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/choloepus_hoffmanni/ncrna/Choloepus_hoffmanni.choHof1.ncrna.fa.gz',
	'chondrus_crispus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/chondrus_crispus/ncrna/Chondrus_crispus.ASM35022v2.ncrna.fa.gz',
	'chroomonas_mesostigmatica_ccmp1168'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_cryptophyta1_collection/chroomonas_mesostigmatica_ccmp1168/ncrna/Chroomonas_mesostigmatica_ccmp1168.ASM28609v1.ncrna.fa.gz',
	'chrysemys_picta_bellii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chrysemys_picta_bellii/ncrna/Chrysemys_picta_bellii.Chrysemys_picta_bellii-3.0.3.ncrna.fa.gz',
	'ciona_intestinalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ciona_intestinalis/ncrna/Ciona_intestinalis.KH.ncrna.fa.gz',
	'ciona_savignyi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ciona_savignyi/ncrna/Ciona_savignyi.CSAV2.0.ncrna.fa.gz',
	'cladophialophora_bantiana_cbs_173_52_gca_000835475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/cladophialophora_bantiana_cbs_173_52_gca_000835475/ncrna/Cladophialophora_bantiana_cbs_173_52_gca_000835475.Clad_bant_CBS_173_52_V1.ncrna.fa.gz',
	'cladophialophora_carrionii_cbs_160_54_gca_000365165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cladophialophora_carrionii_cbs_160_54_gca_000365165/ncrna/Cladophialophora_carrionii_cbs_160_54_gca_000365165.Clad_carr_CBS_160_54_V1.ncrna.fa.gz',
	'cladophialophora_carrionii_gca_001700775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cladophialophora_carrionii_gca_001700775/ncrna/Cladophialophora_carrionii_gca_001700775.ASM170077v1.ncrna.fa.gz',
	'cladophialophora_immunda_gca_000835495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/cladophialophora_immunda_gca_000835495/ncrna/Cladophialophora_immunda_gca_000835495.Clad_immu_CBS83496_V1.ncrna.fa.gz',
	'cladophialophora_psammophila_cbs_110553_gca_000585535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cladophialophora_psammophila_cbs_110553_gca_000585535/ncrna/Cladophialophora_psammophila_cbs_110553_gca_000585535.Clad_psam_CBS_110553_V1.ncrna.fa.gz',
	'cladophialophora_yegresii_cbs_114405_gca_000585515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cladophialophora_yegresii_cbs_114405_gca_000585515/ncrna/Cladophialophora_yegresii_cbs_114405_gca_000585515.Clad_yegr_CBS_114405_V1.ncrna.fa.gz',
	'claviceps_purpurea_20_1_gca_000347355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/claviceps_purpurea_20_1_gca_000347355/ncrna/Claviceps_purpurea_20_1_gca_000347355.ASM34735v1.ncrna.fa.gz',
	'clavispora_lusitaniae_atcc_42720_gca_000003835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/clavispora_lusitaniae_atcc_42720_gca_000003835/ncrna/Clavispora_lusitaniae_atcc_42720_gca_000003835.ASM383v1.ncrna.fa.gz',
	'clavispora_lusitaniae_gca_001673695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/clavispora_lusitaniae_gca_001673695/ncrna/Clavispora_lusitaniae_gca_001673695.ASM167369v2.ncrna.fa.gz',
	'clohesyomyces_aquaticus_gca_002105025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/clohesyomyces_aquaticus_gca_002105025/ncrna/Clohesyomyces_aquaticus_gca_002105025.Cloaq1.ncrna.fa.gz',
	'coccidioides_immitis_h538_4_gca_000149815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/coccidioides_immitis_h538_4_gca_000149815/ncrna/Coccidioides_immitis_h538_4_gca_000149815.ASM14981v1.ncrna.fa.gz',
	'coccidioides_immitis_rmscc_2394_gca_000149895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/coccidioides_immitis_rmscc_2394_gca_000149895/ncrna/Coccidioides_immitis_rmscc_2394_gca_000149895.ASM14989v1.ncrna.fa.gz',
	'coccidioides_immitis_rmscc_3703_gca_000150085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/coccidioides_immitis_rmscc_3703_gca_000150085/ncrna/Coccidioides_immitis_rmscc_3703_gca_000150085.ASM15008v1.ncrna.fa.gz',
	'coccidioides_immitis_rs_gca_000149335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coccidioides_immitis_rs_gca_000149335/ncrna/Coccidioides_immitis_rs_gca_000149335.ASM14933v2.ncrna.fa.gz',
	'coccidioides_posadasii_c735_delta_sowgp_gca_000151335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coccidioides_posadasii_c735_delta_sowgp_gca_000151335/ncrna/Coccidioides_posadasii_c735_delta_sowgp_gca_000151335.JCVI-cpa1-1.0.ncrna.fa.gz',
	'coccidioides_posadasii_rmscc_3488_gca_000150055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/coccidioides_posadasii_rmscc_3488_gca_000150055/ncrna/Coccidioides_posadasii_rmscc_3488_gca_000150055.ASM15005v1.ncrna.fa.gz',
	'coccidioides_posadasii_str_silveira_gca_000170175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coccidioides_posadasii_str_silveira_gca_000170175/ncrna/Coccidioides_posadasii_str_silveira_gca_000170175.CPS2.ncrna.fa.gz',
	'colletotrichum_fioriniae_pj7_gca_000582985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/colletotrichum_fioriniae_pj7_gca_000582985/ncrna/Colletotrichum_fioriniae_pj7_gca_000582985.CFIO01.ncrna.fa.gz',
	'colletotrichum_gloeosporioides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_gloeosporioides/ncrna/Colletotrichum_gloeosporioides.GCA_000319635.1.ncrna.fa.gz',
	'colletotrichum_gloeosporioides_cg_14_gca_000446055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/colletotrichum_gloeosporioides_cg_14_gca_000446055/ncrna/Colletotrichum_gloeosporioides_cg_14_gca_000446055.Colletotrichum_gloeosporioides1.ncrna.fa.gz',
	'colletotrichum_graminicola'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_graminicola/ncrna/Colletotrichum_graminicola.C_graminicola_M1_001_V1.ncrna.fa.gz',
	'colletotrichum_higginsianum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_higginsianum/ncrna/Colletotrichum_higginsianum.GCA_000313795.2.ncrna.fa.gz',
	'colletotrichum_incanum_gca_001625285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_incanum_gca_001625285/ncrna/Colletotrichum_incanum_gca_001625285.Ci238704v1.ncrna.fa.gz',
	'colletotrichum_nymphaeae_sa_01_gca_001563115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_nymphaeae_sa_01_gca_001563115/ncrna/Colletotrichum_nymphaeae_sa_01_gca_001563115.CNYM01.ncrna.fa.gz',
	'colletotrichum_orbiculare'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_orbiculare/ncrna/Colletotrichum_orbiculare.GCA_000350065.1.ncrna.fa.gz',
	'colletotrichum_salicis_gca_001563125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_salicis_gca_001563125/ncrna/Colletotrichum_salicis_gca_001563125.CSAL01.ncrna.fa.gz',
	'colletotrichum_sublineola_gca_000696135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/colletotrichum_sublineola_gca_000696135/ncrna/Colletotrichum_sublineola_gca_000696135.ASM69613v1.ncrna.fa.gz',
	'colletotrichum_tofieldiae_gca_001625265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_tofieldiae_gca_001625265/ncrna/Colletotrichum_tofieldiae_gca_001625265.Ct0861v1.ncrna.fa.gz',
	'colobus_angolensis_palliatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/colobus_angolensis_palliatus/ncrna/Colobus_angolensis_palliatus.Cang.pa_1.0.ncrna.fa.gz',
	'conidiobolus_coronatus_nrrl_28638_gca_001566745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_entomophthoromycota1_collection/conidiobolus_coronatus_nrrl_28638_gca_001566745/ncrna/Conidiobolus_coronatus_nrrl_28638_gca_001566745.Conidiobolus_coronatus_NRRL28638.ncrna.fa.gz',
	'coniella_lustricola_gca_003019895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/coniella_lustricola_gca_003019895/ncrna/Coniella_lustricola_gca_003019895.Coniella_lustricola_v1.0.ncrna.fa.gz',
	'coniochaeta_ligniaria_nrrl_30616_gca_001879275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/coniochaeta_ligniaria_nrrl_30616_gca_001879275/ncrna/Coniochaeta_ligniaria_nrrl_30616_gca_001879275.Conlig_v1.0.ncrna.fa.gz',
	'coniophora_puteana_rwd_64_598_ss2_gca_000271625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/coniophora_puteana_rwd_64_598_ss2_gca_000271625/ncrna/Coniophora_puteana_rwd_64_598_ss2_gca_000271625.Conpu1.ncrna.fa.gz',
	'coniosporium_apollinis_cbs_100218_gca_000281105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coniosporium_apollinis_cbs_100218_gca_000281105/ncrna/Coniosporium_apollinis_cbs_100218_gca_000281105.Coni_apol_CBS100218__V1.ncrna.fa.gz',
	'coprinopsis_cinerea_okayama7_130_gca_000182895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/coprinopsis_cinerea_okayama7_130_gca_000182895/ncrna/Coprinopsis_cinerea_okayama7_130_gca_000182895.CC3.ncrna.fa.gz',
	'corchorus_capsularis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/corchorus_capsularis/ncrna/Corchorus_capsularis.CCACVL1_1.0.ncrna.fa.gz',
	'cordyceps_brongniartii_rcef_3172_gca_001636735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cordyceps_brongniartii_rcef_3172_gca_001636735/ncrna/Cordyceps_brongniartii_rcef_3172_gca_001636735.BBO_1.0.ncrna.fa.gz',
	'cordyceps_confragosa_rcef_1005_gca_001636795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cordyceps_confragosa_rcef_1005_gca_001636795/ncrna/Cordyceps_confragosa_rcef_1005_gca_001636795.LEL_1.0.ncrna.fa.gz',
	'cordyceps_fumosorosea_arsef_2679_gca_001636725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cordyceps_fumosorosea_arsef_2679_gca_001636725/ncrna/Cordyceps_fumosorosea_arsef_2679_gca_001636725.ISF_1.0.ncrna.fa.gz',
	'cordyceps_militaris_cm01_gca_000225605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cordyceps_militaris_cm01_gca_000225605/ncrna/Cordyceps_militaris_cm01_gca_000225605.CmilitarisCM01_v01.ncrna.fa.gz',
	'corynespora_cassiicola_philippines_gca_003016335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/corynespora_cassiicola_philippines_gca_003016335/ncrna/Corynespora_cassiicola_philippines_gca_003016335.Corynespora_cassiicola_v1.0.ncrna.fa.gz',
	'coturnix_japonica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/coturnix_japonica/ncrna/Coturnix_japonica.Coturnix_japonica_2.0.ncrna.fa.gz',
	'crassostrea_gigas'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/crassostrea_gigas/ncrna/Crassostrea_gigas.oyster_v9.ncrna.fa.gz',
	'cricetulus_griseus_chok1gshd'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cricetulus_griseus_chok1gshd/ncrna/Cricetulus_griseus_chok1gshd.CHOK1GS_HDv1.ncrna.fa.gz',
	'cricetulus_griseus_crigri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cricetulus_griseus_crigri/ncrna/Cricetulus_griseus_crigri.CriGri_1.0.ncrna.fa.gz',
	'cricetulus_griseus_picr'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cricetulus_griseus_picr/ncrna/Cricetulus_griseus_picr.CriGri-PICR.ncrna.fa.gz',
	'crocodylus_porosus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/crocodylus_porosus/ncrna/Crocodylus_porosus.CroPor_comp1.ncrna.fa.gz',
	'cryptococcus_amylolentus_cbs_6039_gca_001720205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_amylolentus_cbs_6039_gca_001720205/ncrna/Cryptococcus_amylolentus_cbs_6039_gca_001720205.Cryp_amyl_CBS6039_V3.ncrna.fa.gz',
	'cryptococcus_amylolentus_cbs_6273_gca_001720235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_amylolentus_cbs_6273_gca_001720235/ncrna/Cryptococcus_amylolentus_cbs_6273_gca_001720235.Cryp_amyl_CBS6273_V2.ncrna.fa.gz',
	'cryptococcus_depauperatus_cbs_7841_gca_001720195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_depauperatus_cbs_7841_gca_001720195/ncrna/Cryptococcus_depauperatus_cbs_7841_gca_001720195.Filo_depa_CBS7841_V1.ncrna.fa.gz',
	'cryptococcus_depauperatus_cbs_7855_gca_001720245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_depauperatus_cbs_7855_gca_001720245/ncrna/Cryptococcus_depauperatus_cbs_7855_gca_001720245.Filo_depa_CBS7855_V2.ncrna.fa.gz',
	'cryptococcus_gattii_ca1280_gca_000836335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ca1280_gca_000836335/ncrna/Cryptococcus_gattii_ca1280_gca_000836335.Cryp_gatt_CA1280_V1.ncrna.fa.gz',
	'cryptococcus_gattii_ca1873_gca_000855695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ca1873_gca_000855695/ncrna/Cryptococcus_gattii_ca1873_gca_000855695.Cryp_gatt_CA1873_V1.ncrna.fa.gz',
	'cryptococcus_gattii_e566_gca_000875815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_e566_gca_000875815/ncrna/Cryptococcus_gattii_e566_gca_000875815.Cryp_gatt_E566_V1.ncrna.fa.gz',
	'cryptococcus_gattii_ejb2_gca_000835745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ejb2_gca_000835745/ncrna/Cryptococcus_gattii_ejb2_gca_000835745.Cryp_gatt_EJB2_V1.ncrna.fa.gz',
	'cryptococcus_gattii_nt_10_gca_000935105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_nt_10_gca_000935105/ncrna/Cryptococcus_gattii_nt_10_gca_000935105.Cryp_gatt_NT-10_V1.ncrna.fa.gz',
	'cryptococcus_gattii_ru294_gca_000836355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ru294_gca_000836355/ncrna/Cryptococcus_gattii_ru294_gca_000836355.Cryp_gatt_Ru294_V1.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_2001_935_1_gca_000835815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_2001_935_1_gca_000835815/ncrna/Cryptococcus_gattii_vgii_2001_935_1_gca_000835815.Cryp_gatt_2001_935-1_V1.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_99_473_gca_000836455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_99_473_gca_000836455/ncrna/Cryptococcus_gattii_vgii_99_473_gca_000836455.Cryp_gatt_99_473_V1.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_ca1014_gca_000875795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_ca1014_gca_000875795/ncrna/Cryptococcus_gattii_vgii_ca1014_gca_000875795.Cryp_gatt_CA1014_V1.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_cbs_10090_gca_000835765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_cbs_10090_gca_000835765/ncrna/Cryptococcus_gattii_vgii_cbs_10090_gca_000835765.Cryp_gatt_CBS10090_V1.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_la55_gca_000836315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_la55_gca_000836315/ncrna/Cryptococcus_gattii_vgii_la55_gca_000836315.Cryp_gatt_LA55_V1.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_mmrl2647_gca_000875855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_mmrl2647_gca_000875855/ncrna/Cryptococcus_gattii_vgii_mmrl2647_gca_000875855.Cryp_gatt_MMRL2647_V2.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_r265_gca_000149475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_r265_gca_000149475/ncrna/Cryptococcus_gattii_vgii_r265_gca_000149475.CNB2.ncrna.fa.gz',
	'cryptococcus_gattii_vgii_ram5_gca_000836375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_ram5_gca_000836375/ncrna/Cryptococcus_gattii_vgii_ram5_gca_000836375.Cryp_gatt_Ram5_V1.ncrna.fa.gz',
	'cryptococcus_gattii_vgiv_ind107_gca_000835755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgiv_ind107_gca_000835755/ncrna/Cryptococcus_gattii_vgiv_ind107_gca_000835755.Cryp_gatt_IND107_V2.ncrna.fa.gz',
	'cryptococcus_gattii_wm276_gca_000185945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_wm276_gca_000185945/ncrna/Cryptococcus_gattii_wm276_gca_000185945.ASM18594v1.ncrna.fa.gz',
	'cryptococcus_neoformans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/cryptococcus_neoformans/ncrna/Cryptococcus_neoformans.ASM9104v1.ncrna.fa.gz',
	'cryptococcus_neoformans_var_grubii_h99_gca_000149245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99_gca_000149245/ncrna/Cryptococcus_neoformans_var_grubii_h99_gca_000149245.CNA3.ncrna.fa.gz',
	'cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385/ncrna/Cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385.ASM14938v1.ncrna.fa.gz',
	'cryptomonas_paramecium'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_cryptophyta1_collection/cryptomonas_paramecium/ncrna/Cryptomonas_paramecium.ASM19445v1.ncrna.fa.gz',
	'cryptosporidium_muris_rn66'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cryptosporidium_muris_rn66/ncrna/Cryptosporidium_muris_rn66.JCVI-cmg-v1.0.ncrna.fa.gz',
	'cryptosporidium_parvum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cryptosporidium_parvum/ncrna/Cryptosporidium_parvum.ASM20969v1.ncrna.fa.gz',
	'cryptosporidium_parvum_iowa_ii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cryptosporidium_parvum_iowa_ii/ncrna/Cryptosporidium_parvum_iowa_ii.ASM16534v1.ncrna.fa.gz',
	'cucumis_sativus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/cucumis_sativus/ncrna/Cucumis_sativus.ASM407v2.ncrna.fa.gz',
	'culex_quinquefasciatus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/culex_quinquefasciatus/ncrna/Culex_quinquefasciatus.CpipJ2.ncrna.fa.gz',
	'cutaneotrichosporon_oleaginosum_gca_001027345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cutaneotrichosporon_oleaginosum_gca_001027345/ncrna/Cutaneotrichosporon_oleaginosum_gca_001027345.Triol1.ncrna.fa.gz',
	'cyanistes_caeruleus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cyanistes_caeruleus/ncrna/Cyanistes_caeruleus.cyaCae2.ncrna.fa.gz',
	'cyberlindnera_fabianii_gca_003205855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/cyberlindnera_fabianii_gca_003205855/ncrna/Cyberlindnera_fabianii_gca_003205855.ASM320585v1.ncrna.fa.gz',
	'cyberlindnera_jadinii_gca_001245095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cyberlindnera_jadinii_gca_001245095/ncrna/Cyberlindnera_jadinii_gca_001245095.cj1.ncrna.fa.gz',
	'cyberlindnera_jadinii_nrrl_y_1542_gca_001661405'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cyberlindnera_jadinii_nrrl_y_1542_gca_001661405/ncrna/Cyberlindnera_jadinii_nrrl_y_1542_gca_001661405.Cybja1.ncrna.fa.gz',
	'cyclospora_cayetanensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cyclospora_cayetanensis/ncrna/Cyclospora_cayetanensis.ASM76915v2.ncrna.fa.gz',
	'cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385/ncrna/Cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385.Cylto1.ncrna.fa.gz',
	'cynoglossus_semilaevis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cynoglossus_semilaevis/ncrna/Cynoglossus_semilaevis.Cse_v1.0.ncrna.fa.gz',
	'cyphellophora_europaea_cbs_101466_gca_000365145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cyphellophora_europaea_cbs_101466_gca_000365145/ncrna/Cyphellophora_europaea_cbs_101466_gca_000365145.Phia_euro_CBS_101466_V1.ncrna.fa.gz',
	'cyprinodon_variegatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cyprinodon_variegatus/ncrna/Cyprinodon_variegatus.C_variegatus-1.0.ncrna.fa.gz',
	'dacryopinax_primogenitus_gca_000292625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/dacryopinax_primogenitus_gca_000292625/ncrna/Dacryopinax_primogenitus_gca_000292625.Dacryopinax_sp._DJM_731_SSP1_v1.0.ncrna.fa.gz',
	'dactylellina_haptotyla_cbs_200_50_gca_000441935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/dactylellina_haptotyla_cbs_200_50_gca_000441935/ncrna/Dactylellina_haptotyla_cbs_200_50_gca_000441935.MHA_v2.ncrna.fa.gz',
	'daedalea_quercina_l_15889_gca_001632345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/daedalea_quercina_l_15889_gca_001632345/ncrna/Daedalea_quercina_l_15889_gca_001632345.Daequ1.ncrna.fa.gz',
	'daldinia_sp_ec12_gca_002120325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/daldinia_sp_ec12_gca_002120325/ncrna/Daldinia_sp_ec12_gca_002120325.DalEC12_v.1.0.ncrna.fa.gz',
	'danaus_plexippus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/danaus_plexippus/ncrna/Danaus_plexippus.Dpv3.ncrna.fa.gz',
	'danio_rerio'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/danio_rerio/ncrna/Danio_rerio.GRCz11.ncrna.fa.gz',
	'daphnia_magna'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/daphnia_magna/ncrna/Daphnia_magna.daphmag2.4.ncrna.fa.gz',
	'daphnia_pulex'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/daphnia_pulex/ncrna/Daphnia_pulex.V1.0.ncrna.fa.gz',
	'dasypus_novemcinctus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/dasypus_novemcinctus/ncrna/Dasypus_novemcinctus.Dasnov3.0.ncrna.fa.gz',
	'daucus_carota'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/daucus_carota/ncrna/Daucus_carota.ASM162521v1.ncrna.fa.gz',
	'debaryomyces_fabryi_gca_001447935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/debaryomyces_fabryi_gca_001447935/ncrna/Debaryomyces_fabryi_gca_001447935.debFab1.0.ncrna.fa.gz',
	'debaryomyces_hansenii_cbs767_gca_000006445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/debaryomyces_hansenii_cbs767_gca_000006445/ncrna/Debaryomyces_hansenii_cbs767_gca_000006445.ASM644v2.ncrna.fa.gz',
	'dendroctonus_ponderosae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/dendroctonus_ponderosae/ncrna/Dendroctonus_ponderosae.DendPond_male_1.0.ncrna.fa.gz',
	'diaporthe_ampelina_gca_001006365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/diaporthe_ampelina_gca_001006365/ncrna/Diaporthe_ampelina_gca_001006365.UCDDA912v1.0.ncrna.fa.gz',
	'dichomitus_squalens_lyad_421_ss1_gca_000275845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/dichomitus_squalens_lyad_421_ss1_gca_000275845/ncrna/Dichomitus_squalens_lyad_421_ss1_gca_000275845.Dichomitus_squalens_v1.0.ncrna.fa.gz',
	'dictyostelium_discoideum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/dictyostelium_discoideum/ncrna/Dictyostelium_discoideum.dicty_2.7.ncrna.fa.gz',
	'dictyostelium_fasciculatum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/dictyostelium_fasciculatum/ncrna/Dictyostelium_fasciculatum.DFas_2.0.ncrna.fa.gz',
	'dictyostelium_lacteum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/dictyostelium_lacteum/ncrna/Dictyostelium_lacteum.ASM160615v1.ncrna.fa.gz',
	'dioscorea_rotundata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/dioscorea_rotundata/ncrna/Dioscorea_rotundata.TDr96_F1_Pseudo_Chromosome_v1.0.ncrna.fa.gz',
	'diplodia_seriata_gca_001006355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/diplodia_seriata_gca_001006355/ncrna/Diplodia_seriata_gca_001006355.UCDDS831v0.1.ncrna.fa.gz',
	'dipodomys_ordii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/dipodomys_ordii/ncrna/Dipodomys_ordii.Dord_2.0.ncrna.fa.gz',
	'dothistroma_septosporum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/dothistroma_septosporum/ncrna/Dothistroma_septosporum.GCA_000340195.1.ncrna.fa.gz',
	'drechmeria_coniospora_gca_001625195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/drechmeria_coniospora_gca_001625195/ncrna/Drechmeria_coniospora_gca_001625195.ASM162519v1.ncrna.fa.gz',
	'drechslerella_stenobrocha_248_gca_000525045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/drechslerella_stenobrocha_248_gca_000525045/ncrna/Drechslerella_stenobrocha_248_gca_000525045.DREv1.ncrna.fa.gz',
	'dromaius_novaehollandiae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/dromaius_novaehollandiae/ncrna/Dromaius_novaehollandiae.droNov1.ncrna.fa.gz',
	'drosophila_ananassae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_ananassae/ncrna/Drosophila_ananassae.dana_caf1.ncrna.fa.gz',
	'drosophila_erecta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_erecta/ncrna/Drosophila_erecta.dere_caf1.ncrna.fa.gz',
	'drosophila_grimshawi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_grimshawi/ncrna/Drosophila_grimshawi.dgri_caf1.ncrna.fa.gz',
	'drosophila_melanogaster'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_melanogaster/ncrna/Drosophila_melanogaster.BDGP6.22.ncrna.fa.gz',
	'drosophila_mojavensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_mojavensis/ncrna/Drosophila_mojavensis.dmoj_caf1.ncrna.fa.gz',
	'drosophila_persimilis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_persimilis/ncrna/Drosophila_persimilis.dper_caf1.ncrna.fa.gz',
	'drosophila_pseudoobscura'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_pseudoobscura/ncrna/Drosophila_pseudoobscura.Dpse_3.0.ncrna.fa.gz',
	'drosophila_sechellia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_sechellia/ncrna/Drosophila_sechellia.dsec_caf1.ncrna.fa.gz',
	'drosophila_simulans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_simulans/ncrna/Drosophila_simulans.ASM75419v3.ncrna.fa.gz',
	'drosophila_virilis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_virilis/ncrna/Drosophila_virilis.dvir_caf1.ncrna.fa.gz',
	'drosophila_willistoni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_willistoni/ncrna/Drosophila_willistoni.dwil_caf1.ncrna.fa.gz',
	'drosophila_yakuba'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_yakuba/ncrna/Drosophila_yakuba.dyak_caf1.ncrna.fa.gz',
	'echinops_telfairi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/echinops_telfairi/ncrna/Echinops_telfairi.TENREC.ncrna.fa.gz',
	'edhazardia_aedis_usnm_41457_gca_000230595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/edhazardia_aedis_usnm_41457_gca_000230595/ncrna/Edhazardia_aedis_usnm_41457_gca_000230595.Edha_aedis_V4b.ncrna.fa.gz',
	'eimeria_brunetti'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_brunetti/ncrna/Eimeria_brunetti.EBH001.ncrna.fa.gz',
	'eimeria_maxima'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_maxima/ncrna/Eimeria_maxima.EMW001.ncrna.fa.gz',
	'eimeria_mitis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_mitis/ncrna/Eimeria_mitis.EMH001.ncrna.fa.gz',
	'eimeria_praecox'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_praecox/ncrna/Eimeria_praecox.EPH001.ncrna.fa.gz',
	'eimeria_tenella'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_tenella/ncrna/Eimeria_tenella.ASM283v1.ncrna.fa.gz',
	'eimeria_tenella_gca_000499545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_tenella_gca_000499545/ncrna/Eimeria_tenella_gca_000499545.ETH001.ncrna.fa.gz',
	'emergomyces_pasteuriana_ep9510_gca_001883825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/emergomyces_pasteuriana_ep9510_gca_001883825/ncrna/Emergomyces_pasteuriana_ep9510_gca_001883825.Emmo_past_UAMH9510_V1.ncrna.fa.gz',
	'emiliania_huxleyi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/emiliania_huxleyi/ncrna/Emiliania_huxleyi.Emiliana_huxleyi_CCMP1516_main_genome_assembly_v1.0.ncrna.fa.gz',
	'emmonsia_crescens_uamh_3008_gca_001008285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/emmonsia_crescens_uamh_3008_gca_001008285/ncrna/Emmonsia_crescens_uamh_3008_gca_001008285.ASM100828v1.ncrna.fa.gz',
	'emmonsia_parva_uamh_139_gca_001014755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/emmonsia_parva_uamh_139_gca_001014755/ncrna/Emmonsia_parva_uamh_139_gca_001014755.ASM101475v1.ncrna.fa.gz',
	'emmonsia_sp_cac_2015a_gca_001660665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/emmonsia_sp_cac_2015a_gca_001660665/ncrna/Emmonsia_sp_cac_2015a_gca_001660665.Emmo_afri_EA111.ncrna.fa.gz',
	'encephalitozoon_cuniculi_ecuniii_l_gca_001078035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_cuniculi_ecuniii_l_gca_001078035/ncrna/Encephalitozoon_cuniculi_ecuniii_l_gca_001078035.ECIIIL.ncrna.fa.gz',
	'encephalitozoon_hellem_atcc_50504_gca_000277815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_hellem_atcc_50504_gca_000277815/ncrna/Encephalitozoon_hellem_atcc_50504_gca_000277815.ASM27781v3.ncrna.fa.gz',
	'encephalitozoon_intestinalis_atcc_50506_gca_000146465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_intestinalis_atcc_50506_gca_000146465/ncrna/Encephalitozoon_intestinalis_atcc_50506_gca_000146465.ASM14646v1.ncrna.fa.gz',
	'encephalitozoon_romaleae_sj_2008_gca_000280035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_romaleae_sj_2008_gca_000280035/ncrna/Encephalitozoon_romaleae_sj_2008_gca_000280035.ASM28003v2.ncrna.fa.gz',
	'endocarpon_pusillum_z07020_gca_000464535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/endocarpon_pusillum_z07020_gca_000464535/ncrna/Endocarpon_pusillum_z07020_gca_000464535.EPUS.ncrna.fa.gz',
	'entamoeba_histolytica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/entamoeba_histolytica/ncrna/Entamoeba_histolytica.JCVI-ESG2-1.0.ncrna.fa.gz',
	'entamoeba_histolytica_gca_001662325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_histolytica_gca_001662325/ncrna/Entamoeba_histolytica_gca_001662325.Ehistolytica_2016.ncrna.fa.gz',
	'entamoeba_histolytica_hm_3_imss'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_histolytica_hm_3_imss/ncrna/Entamoeba_histolytica_hm_3_imss.EHA.strHM3_v1.ncrna.fa.gz',
	'entamoeba_histolytica_ku27'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_histolytica_ku27/ncrna/Entamoeba_histolytica_ku27.EHA_ku27_v1.ncrna.fa.gz',
	'entamoeba_nuttalli_p19'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_nuttalli_p19/ncrna/Entamoeba_nuttalli_p19.ENU1_v1.ncrna.fa.gz',
	'enterocytozoon_bieneusi_h348_gca_000209485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/enterocytozoon_bieneusi_h348_gca_000209485/ncrna/Enterocytozoon_bieneusi_h348_gca_000209485.ASM20948v1.ncrna.fa.gz',
	'epicoccum_nigrum_gca_002116315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/epicoccum_nigrum_gca_002116315/ncrna/Epicoccum_nigrum_gca_002116315.ASM211631v1.ncrna.fa.gz',
	'eptatretus_burgeri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/eptatretus_burgeri/ncrna/Eptatretus_burgeri.Eburgeri_3.2.ncrna.fa.gz',
	'equus_asinus_asinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/equus_asinus_asinus/ncrna/Equus_asinus_asinus.ASM303372v1.ncrna.fa.gz',
	'equus_caballus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/equus_caballus/ncrna/Equus_caballus.EquCab3.0.ncrna.fa.gz',
	'eremothecium_cymbalariae_dbvpg_7215_gca_000235365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/eremothecium_cymbalariae_dbvpg_7215_gca_000235365/ncrna/Eremothecium_cymbalariae_dbvpg_7215_gca_000235365.ASM23536v1.ncrna.fa.gz',
	'eremothecium_gossypii_fdag1_gca_000968835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/eremothecium_gossypii_fdag1_gca_000968835/ncrna/Eremothecium_gossypii_fdag1_gca_000968835.ASM96883v1.ncrna.fa.gz',
	'erinaceus_europaeus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/erinaceus_europaeus/ncrna/Erinaceus_europaeus.HEDGEHOG.ncrna.fa.gz',
	'erysiphe_necator_gca_000798715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/erysiphe_necator_gca_000798715/ncrna/Erysiphe_necator_gca_000798715.ASM79871v1.ncrna.fa.gz',
	'escovopsis_weberi_gca_001278495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/escovopsis_weberi_gca_001278495/ncrna/Escovopsis_weberi_gca_001278495.ASM127849v1.ncrna.fa.gz',
	'esox_lucius'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/esox_lucius/ncrna/Esox_lucius.Eluc_V3.ncrna.fa.gz',
	'eutypa_lata_ucrel1_gca_000349385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/eutypa_lata_ucrel1_gca_000349385/ncrna/Eutypa_lata_ucrel1_gca_000349385.UCREL1V03.ncrna.fa.gz',
	'exidia_glandulosa_hhb12029_gca_001632375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/exidia_glandulosa_hhb12029_gca_001632375/ncrna/Exidia_glandulosa_hhb12029_gca_001632375.Exigl1.ncrna.fa.gz',
	'exophiala_aquamarina_cbs_119918_gca_000709125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/exophiala_aquamarina_cbs_119918_gca_000709125/ncrna/Exophiala_aquamarina_cbs_119918_gca_000709125.Exop_aqua_CBS_119918_V1.ncrna.fa.gz',
	'exophiala_dermatitidis_nih_ut8656_gca_000230625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/exophiala_dermatitidis_nih_ut8656_gca_000230625/ncrna/Exophiala_dermatitidis_nih_ut8656_gca_000230625.Exop_derm_V1.ncrna.fa.gz',
	'exophiala_mesophila_gca_000836275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_mesophila_gca_000836275/ncrna/Exophiala_mesophila_gca_000836275.Exop_meso_CBS40295_V1.ncrna.fa.gz',
	'exophiala_oligosperma_gca_000835515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_oligosperma_gca_000835515/ncrna/Exophiala_oligosperma_gca_000835515.Exop_olig_CBS72588_V1.ncrna.fa.gz',
	'exophiala_sideris_gca_000835395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_sideris_gca_000835395/ncrna/Exophiala_sideris_gca_000835395.Exop_side_CBS121828_V1.ncrna.fa.gz',
	'exophiala_spinifera_gca_000836115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_spinifera_gca_000836115/ncrna/Exophiala_spinifera_gca_000836115.Exop_spin_CBS89968_V1.ncrna.fa.gz',
	'exophiala_xenobiotica_gca_000835505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_xenobiotica_gca_000835505/ncrna/Exophiala_xenobiotica_gca_000835505.Exop_xeno_CBS118157_V1.ncrna.fa.gz',
	'exserohilum_turcica_et28a_gca_000359705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/exserohilum_turcica_et28a_gca_000359705/ncrna/Exserohilum_turcica_et28a_gca_000359705.Setospaeria_trucica_Et28A_v1.0.ncrna.fa.gz',
	'felis_catus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/felis_catus/ncrna/Felis_catus.Felis_catus_9.0.ncrna.fa.gz',
	'fibroporia_radiculosa_gca_000313525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fibroporia_radiculosa_gca_000313525/ncrna/Fibroporia_radiculosa_gca_000313525.ASM31352v1.ncrna.fa.gz',
	'fibularhizoctonia_sp_cbs_109695_gca_001630335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fibularhizoctonia_sp_cbs_109695_gca_001630335/ncrna/Fibularhizoctonia_sp_cbs_109695_gca_001630335.Fibsp1.ncrna.fa.gz',
	'ficedula_albicollis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ficedula_albicollis/ncrna/Ficedula_albicollis.FicAlb_1.4.ncrna.fa.gz',
	'fistulina_hepatica_atcc_64428_gca_000934395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fistulina_hepatica_atcc_64428_gca_000934395/ncrna/Fistulina_hepatica_atcc_64428_gca_000934395.Fishe1.ncrna.fa.gz',
	'fomitiporia_mediterranea_mf3_22_gca_000271605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fomitiporia_mediterranea_mf3_22_gca_000271605/ncrna/Fomitiporia_mediterranea_mf3_22_gca_000271605.Fomme1.ncrna.fa.gz',
	'fomitopsis_pinicola_fp_58527_ss1_gca_000344655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fomitopsis_pinicola_fp_58527_ss1_gca_000344655/ncrna/Fomitopsis_pinicola_fp_58527_ss1_gca_000344655.Fompi3.ncrna.fa.gz',
	'fonsecaea_erecta_gca_001651985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_erecta_gca_001651985/ncrna/Fonsecaea_erecta_gca_001651985.ASM165198v1.ncrna.fa.gz',
	'fonsecaea_monophora_gca_001642475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_monophora_gca_001642475/ncrna/Fonsecaea_monophora_gca_001642475.ASM164247v1.ncrna.fa.gz',
	'fonsecaea_multimorphosa_cbs_102226_gca_000836435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fonsecaea_multimorphosa_cbs_102226_gca_000836435/ncrna/Fonsecaea_multimorphosa_cbs_102226_gca_000836435.Fons_mult_CBS_102226_V1.ncrna.fa.gz',
	'fonsecaea_multimorphosa_gca_001646985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_multimorphosa_gca_001646985/ncrna/Fonsecaea_multimorphosa_gca_001646985.ASM164698v1.ncrna.fa.gz',
	'fonsecaea_nubica_gca_001646965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_nubica_gca_001646965/ncrna/Fonsecaea_nubica_gca_001646965.ASM164696v1.ncrna.fa.gz',
	'fonsecaea_pedrosoi_cbs_271_37_gca_000835455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fonsecaea_pedrosoi_cbs_271_37_gca_000835455/ncrna/Fonsecaea_pedrosoi_cbs_271_37_gca_000835455.Fons_pedr_CBS_271_37_V1.ncrna.fa.gz',
	'fonticula_alba_gca_000388065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_nucleariidaeandfonticulagroup1_collection/fonticula_alba_gca_000388065/ncrna/Fonticula_alba_gca_000388065.Font_alba_ATCC_38817_V2.ncrna.fa.gz',
	'fukomys_damarensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/fukomys_damarensis/ncrna/Fukomys_damarensis.DMR_v1.0.ncrna.fa.gz',
	'fundulus_heteroclitus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/fundulus_heteroclitus/ncrna/Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.ncrna.fa.gz',
	'fusarium_culmorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_culmorum/ncrna/Fusarium_culmorum.EF1.ncrna.fa.gz',
	'fusarium_fujikuroi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_fujikuroi/ncrna/Fusarium_fujikuroi.EF1.ncrna.fa.gz',
	'fusarium_fujikuroi_gca_001023035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fusarium_fujikuroi_gca_001023035/ncrna/Fusarium_fujikuroi_gca_001023035.ASM102303v1.ncrna.fa.gz',
	'fusarium_fujikuroi_gca_001023045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_001023045/ncrna/Fusarium_fujikuroi_gca_001023045.ASM102304v1.ncrna.fa.gz',
	'fusarium_fujikuroi_gca_001023065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fusarium_fujikuroi_gca_001023065/ncrna/Fusarium_fujikuroi_gca_001023065.ASM102306v1.ncrna.fa.gz',
	'fusarium_fujikuroi_gca_900096505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_900096505/ncrna/Fusarium_fujikuroi_gca_900096505.version_1.ncrna.fa.gz',
	'fusarium_fujikuroi_gca_900096685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_900096685/ncrna/Fusarium_fujikuroi_gca_900096685.version_1.ncrna.fa.gz',
	'fusarium_fujikuroi_gca_900096705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_900096705/ncrna/Fusarium_fujikuroi_gca_900096705.version_1.ncrna.fa.gz',
	'fusarium_graminearum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_graminearum/ncrna/Fusarium_graminearum.RR1.ncrna.fa.gz',
	'fusarium_graminearum_gca_000240135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_graminearum_gca_000240135/ncrna/Fusarium_graminearum_gca_000240135.ASM24013v3.ncrna.fa.gz',
	'fusarium_langsethiae_gca_001292635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_langsethiae_gca_001292635/ncrna/Fusarium_langsethiae_gca_001292635.ASM129263v1.ncrna.fa.gz',
	'fusarium_mangiferae_gca_900044065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_mangiferae_gca_900044065/ncrna/Fusarium_mangiferae_gca_900044065.Genome_assembly_version_1.ncrna.fa.gz',
	'fusarium_oxysporum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_oxysporum/ncrna/Fusarium_oxysporum.FO2.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215/ncrna/Fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215.FO_PHW808_V1.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345/ncrna/Fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345.Foc1_1.0.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365/ncrna/Fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365.Foc4_1.0.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195/ncrna/Fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195.FO_II5_V1.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955/ncrna/Fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955.ASM14995v2.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975/ncrna/Fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975.FO_MN25_V1.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_melonis_26406_gca_000260495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_melonis_26406_gca_000260495/ncrna/Fusarium_oxysporum_f_sp_melonis_26406_gca_000260495.FO_melonis_V1.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075/ncrna/Fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075.FO_HDV247_V1.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155/ncrna/Fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155.FO_CL57_V1.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_raphani_54005_gca_000260235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_raphani_54005_gca_000260235/ncrna/Fusarium_oxysporum_f_sp_raphani_54005_gca_000260235.FO_PHW815_V1.ncrna.fa.gz',
	'fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175/ncrna/Fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175.FO_Cotton_V1.ncrna.fa.gz',
	'fusarium_oxysporum_fo47_gca_000271705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_fo47_gca_000271705/ncrna/Fusarium_oxysporum_fo47_gca_000271705.FO_Fo47_V1.ncrna.fa.gz',
	'fusarium_oxysporum_fosc_3_a_gca_000271745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_fosc_3_a_gca_000271745/ncrna/Fusarium_oxysporum_fosc_3_a_gca_000271745.FO_FOSC_3_a_V1.ncrna.fa.gz',
	'fusarium_oxysporum_gca_900096695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_oxysporum_gca_900096695/ncrna/Fusarium_oxysporum_gca_900096695.version_1.ncrna.fa.gz',
	'fusarium_poae_gca_001675295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_poae_gca_001675295/ncrna/Fusarium_poae_gca_001675295.FPOA1.0.ncrna.fa.gz',
	'fusarium_proliferatum_et1_gca_900067095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_proliferatum_et1_gca_900067095/ncrna/Fusarium_proliferatum_et1_gca_900067095.F._proliferatum_ET1_version_1.ncrna.fa.gz',
	'fusarium_proliferatum_gca_900029915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_proliferatum_gca_900029915/ncrna/Fusarium_proliferatum_gca_900029915.F._proliferatum_NRRL62905_version_1.ncrna.fa.gz',
	'fusarium_pseudograminearum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_pseudograminearum/ncrna/Fusarium_pseudograminearum.GCA_000303195.1.ncrna.fa.gz',
	'fusarium_solani'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_solani/ncrna/Fusarium_solani.v2.0.ncrna.fa.gz',
	'fusarium_venenatum_gca_900007375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/fusarium_venenatum_gca_900007375/ncrna/Fusarium_venenatum_gca_900007375.ASM90000737v1.ncrna.fa.gz',
	'fusarium_verticillioides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_verticillioides/ncrna/Fusarium_verticillioides.ASM14955v1.ncrna.fa.gz',
	'gadus_morhua'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gadus_morhua/ncrna/Gadus_morhua.gadMor1.ncrna.fa.gz',
	'gaeumannomyces_graminis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/gaeumannomyces_graminis/ncrna/Gaeumannomyces_graminis.Gae_graminis_V2.ncrna.fa.gz',
	'galdieria_sulphuraria'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/galdieria_sulphuraria/ncrna/Galdieria_sulphuraria.ASM34128v1.ncrna.fa.gz',
	'galerina_marginata_cbs_339_88_gca_000697645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/galerina_marginata_cbs_339_88_gca_000697645/ncrna/Galerina_marginata_cbs_339_88_gca_000697645.Galma1.ncrna.fa.gz',
	'gallus_gallus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gallus_gallus/ncrna/Gallus_gallus.GRCg6a.ncrna.fa.gz',
	'gambusia_affinis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gambusia_affinis/ncrna/Gambusia_affinis.ASM309773v1.ncrna.fa.gz',
	'gasterosteus_aculeatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gasterosteus_aculeatus/ncrna/Gasterosteus_aculeatus.BROADS1.ncrna.fa.gz',
	'gelatoporia_subvermispora_b_gca_000320605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/gelatoporia_subvermispora_b_gca_000320605/ncrna/Gelatoporia_subvermispora_b_gca_000320605.Ceriporiopsis_subvermispora_B_v1.0.ncrna.fa.gz',
	'giardia_intestinalis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis/ncrna/Giardia_intestinalis.ASM49873v1.ncrna.fa.gz',
	'giardia_intestinalis_assemblage_b'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis_assemblage_b/ncrna/Giardia_intestinalis_assemblage_b.G_duodenalis_AssB_BAH15c1.ncrna.fa.gz',
	'giardia_intestinalis_atcc_50581'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis_atcc_50581/ncrna/Giardia_intestinalis_atcc_50581.ASM18240v1.ncrna.fa.gz',
	'giardia_intestinalis_gca_000498715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis_gca_000498715/ncrna/Giardia_intestinalis_gca_000498715.ASM49871v1.ncrna.fa.gz',
	'giardia_lamblia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/giardia_lamblia/ncrna/Giardia_lamblia.GL2.ncrna.fa.gz',
	'giardia_lamblia_p15'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_lamblia_p15/ncrna/Giardia_lamblia_p15.ASM18266v1.ncrna.fa.gz',
	'glarea_lozoyensis_atcc_20868_gca_000409485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/glarea_lozoyensis_atcc_20868_gca_000409485/ncrna/Glarea_lozoyensis_atcc_20868_gca_000409485.GLAREA.ncrna.fa.gz',
	'gloeophyllum_trabeum_atcc_11539_gca_000344685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/gloeophyllum_trabeum_atcc_11539_gca_000344685/ncrna/Gloeophyllum_trabeum_atcc_11539_gca_000344685.Glotr1_1.ncrna.fa.gz',
	'glycine_max'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/glycine_max/ncrna/Glycine_max.Glycine_max_v2.1.ncrna.fa.gz',
	'gonapodya_prolifera_jel478_gca_001574975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/gonapodya_prolifera_jel478_gca_001574975/ncrna/Gonapodya_prolifera_jel478_gca_001574975.Ganpr1.ncrna.fa.gz',
	'gopherus_agassizii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gopherus_agassizii/ncrna/Gopherus_agassizii.ASM289641v1.ncrna.fa.gz',
	'gorilla_gorilla'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gorilla_gorilla/ncrna/Gorilla_gorilla.gorGor4.ncrna.fa.gz',
	'gossypium_raimondii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/gossypium_raimondii/ncrna/Gossypium_raimondii.Graimondii2_0.ncrna.fa.gz',
	'gregarina_niphandrodes'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/gregarina_niphandrodes/ncrna/Gregarina_niphandrodes.GNI3.ncrna.fa.gz',
	'grifola_frondosa_gca_001683735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/grifola_frondosa_gca_001683735/ncrna/Grifola_frondosa_gca_001683735.ASM168373v1.ncrna.fa.gz',
	'grosmannia_clavigera_kw1407_gca_000143105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/grosmannia_clavigera_kw1407_gca_000143105/ncrna/Grosmannia_clavigera_kw1407_gca_000143105.Sanger-454-IlluminaPA_2.0.ncrna.fa.gz',
	'guillardia_theta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/guillardia_theta/ncrna/Guillardia_theta.Guith1.ncrna.fa.gz',
	'guillardia_theta_gca_000002975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_cryptophyta1_collection/guillardia_theta_gca_000002975/ncrna/Guillardia_theta_gca_000002975.ASM297v1.ncrna.fa.gz',
	'gymnopus_luxurians_fd_317_m1_gca_000827265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/gymnopus_luxurians_fd_317_m1_gca_000827265/ncrna/Gymnopus_luxurians_fd_317_m1_gca_000827265.Gymlu1.ncrna.fa.gz',
	'hammondia_hammondi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/hammondia_hammondi/ncrna/Hammondia_hammondi.HHA1_v02.ncrna.fa.gz',
	'hanseniaspora_guilliermondii_gca_900119595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_guilliermondii_gca_900119595/ncrna/Hanseniaspora_guilliermondii_gca_900119595.version_1.ncrna.fa.gz',
	'hanseniaspora_opuntiae_gca_001749795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_opuntiae_gca_001749795/ncrna/Hanseniaspora_opuntiae_gca_001749795.ASM174979v1.ncrna.fa.gz',
	'hanseniaspora_osmophila_gca_001747045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_osmophila_gca_001747045/ncrna/Hanseniaspora_osmophila_gca_001747045.ASM174704v1.ncrna.fa.gz',
	'hanseniaspora_uvarum_dsm_2768_gca_000968475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/hanseniaspora_uvarum_dsm_2768_gca_000968475/ncrna/Hanseniaspora_uvarum_dsm_2768_gca_000968475.Huva2.0.ncrna.fa.gz',
	'hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025/ncrna/Hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025.Hanva1_1.ncrna.fa.gz',
	'haplochromis_burtoni'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/haplochromis_burtoni/ncrna/Haplochromis_burtoni.AstBur1.0.ncrna.fa.gz',
	'hebeloma_cylindrosporum_h7_gca_000827355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hebeloma_cylindrosporum_h7_gca_000827355/ncrna/Hebeloma_cylindrosporum_h7_gca_000827355.Hebeloma_cylindrosporum_h7_v2.0.ncrna.fa.gz',
	'helianthus_annuus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/helianthus_annuus/ncrna/Helianthus_annuus.HanXRQr1.0.ncrna.fa.gz',
	'heliconius_melpomene'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/heliconius_melpomene/ncrna/Heliconius_melpomene.Hmel1.ncrna.fa.gz',
	'helobdella_robusta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/helobdella_robusta/ncrna/Helobdella_robusta.Helro1.ncrna.fa.gz',
	'hesseltinella_vesiculosa_gca_002104935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/hesseltinella_vesiculosa_gca_002104935/ncrna/Hesseltinella_vesiculosa_gca_002104935.Hesve2.ncrna.fa.gz',
	'heterobasidion_irregulare_tc_32_1_gca_000320585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/heterobasidion_irregulare_tc_32_1_gca_000320585/ncrna/Heterobasidion_irregulare_tc_32_1_gca_000320585.Heterobasidion_irregulare_v2.0.ncrna.fa.gz',
	'heterocephalus_glaber_female'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/heterocephalus_glaber_female/ncrna/Heterocephalus_glaber_female.HetGla_female_1.0.ncrna.fa.gz',
	'heterocephalus_glaber_male'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/heterocephalus_glaber_male/ncrna/Heterocephalus_glaber_male.HetGla_1.0.ncrna.fa.gz',
	'hippocampus_comes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/hippocampus_comes/ncrna/Hippocampus_comes.H_comes_QL1_v1.ncrna.fa.gz',
	'hirsutella_minnesotensis_3608_gca_000956045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/hirsutella_minnesotensis_3608_gca_000956045/ncrna/Hirsutella_minnesotensis_3608_gca_000956045.ASM95604v1.ncrna.fa.gz',
	'histoplasma_capsulatum_g186ar_gca_000150115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_g186ar_gca_000150115/ncrna/Histoplasma_capsulatum_g186ar_gca_000150115.ASM15011v1.ncrna.fa.gz',
	'histoplasma_capsulatum_h143_gca_000151035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_h143_gca_000151035/ncrna/Histoplasma_capsulatum_h143_gca_000151035.ASM15103v1.ncrna.fa.gz',
	'histoplasma_capsulatum_h88_gca_000151005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_h88_gca_000151005/ncrna/Histoplasma_capsulatum_h88_gca_000151005.ASM15100v2.ncrna.fa.gz',
	'histoplasma_capsulatum_nam1_gca_000149585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_nam1_gca_000149585/ncrna/Histoplasma_capsulatum_nam1_gca_000149585.ASM14958v1.ncrna.fa.gz',
	'homo_sapiens'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz',
	'hordeum_vulgare'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/hordeum_vulgare/ncrna/Hordeum_vulgare.IBSC_v2.ncrna.fa.gz',
	'hortaea_werneckii_exf_2000_gca_002127715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hortaea_werneckii_exf_2000_gca_002127715/ncrna/Hortaea_werneckii_exf_2000_gca_002127715.HwerPB1.ncrna.fa.gz',
	'hyaloperonospora_arabidopsidis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/hyaloperonospora_arabidopsidis/ncrna/Hyaloperonospora_arabidopsidis.HyaAraEmoy2_2.0.ncrna.fa.gz',
	'hydnomerulius_pinastri_md_312_gca_000827185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hydnomerulius_pinastri_md_312_gca_000827185/ncrna/Hydnomerulius_pinastri_md_312_gca_000827185.Hydpi2.ncrna.fa.gz',
	'hypholoma_sublateritium_fd_334_ss_4_gca_000827495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hypholoma_sublateritium_fd_334_ss_4_gca_000827495/ncrna/Hypholoma_sublateritium_fd_334_ss_4_gca_000827495.Hypsu1.ncrna.fa.gz',
	'hyphopichia_burtonii_nrrl_y_1933_gca_001661395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hyphopichia_burtonii_nrrl_y_1933_gca_001661395/ncrna/Hyphopichia_burtonii_nrrl_y_1933_gca_001661395.Hypbu1.ncrna.fa.gz',
	'hypoxylon_sp_ci_4a_gca_002120315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hypoxylon_sp_ci_4a_gca_002120315/ncrna/Hypoxylon_sp_ci_4a_gca_002120315.HypCI4A_v1.0.ncrna.fa.gz',
	'hypoxylon_sp_co27_5_gca_002120305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/hypoxylon_sp_co27_5_gca_002120305/ncrna/Hypoxylon_sp_co27_5_gca_002120305.HypCO275_v1.0.ncrna.fa.gz',
	'hypoxylon_sp_ec38_gca_002120335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hypoxylon_sp_ec38_gca_002120335/ncrna/Hypoxylon_sp_ec38_gca_002120335.HypEC38_v3.0.ncrna.fa.gz',
	'hypsizygus_marmoreus_gca_001605315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hypsizygus_marmoreus_gca_001605315/ncrna/Hypsizygus_marmoreus_gca_001605315.ASM160531v1.ncrna.fa.gz',
	'ichthyophthirius_multifiliis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/ichthyophthirius_multifiliis/ncrna/Ichthyophthirius_multifiliis.JCVI-IMG1-V.1.ncrna.fa.gz',
	'ictalurus_punctatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ictalurus_punctatus/ncrna/Ictalurus_punctatus.IpCoco_1.2.ncrna.fa.gz',
	'ictidomys_tridecemlineatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ictidomys_tridecemlineatus/ncrna/Ictidomys_tridecemlineatus.SpeTri2.0.ncrna.fa.gz',
	'ixodes_scapularis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/ixodes_scapularis/ncrna/Ixodes_scapularis.IscaW1.ncrna.fa.gz',
	'jaapia_argillacea_mucl_33604_gca_000697665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/jaapia_argillacea_mucl_33604_gca_000697665/ncrna/Jaapia_argillacea_mucl_33604_gca_000697665.Jaaar1.ncrna.fa.gz',
	'jaculus_jaculus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/jaculus_jaculus/ncrna/Jaculus_jaculus.JacJac1.0.ncrna.fa.gz',
	'jaminaea_rosea_gca_003144245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/jaminaea_rosea_gca_003144245/ncrna/Jaminaea_rosea_gca_003144245.Jamsp1.ncrna.fa.gz',
	'junco_hyemalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/junco_hyemalis/ncrna/Junco_hyemalis.ASM382977v1.ncrna.fa.gz',
	'kalmanozyma_brasiliensis_ghg001_gca_000497045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kalmanozyma_brasiliensis_ghg001_gca_000497045/ncrna/Kalmanozyma_brasiliensis_ghg001_gca_000497045.PSEUBRA1.ncrna.fa.gz',
	'kazachstania_africana_cbs_2517_gca_000304475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kazachstania_africana_cbs_2517_gca_000304475/ncrna/Kazachstania_africana_cbs_2517_gca_000304475.Ka_CBS2517.ncrna.fa.gz',
	'kazachstania_naganishii_cbs_8797_gca_000348985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kazachstania_naganishii_cbs_8797_gca_000348985/ncrna/Kazachstania_naganishii_cbs_8797_gca_000348985.ASM34898v1.ncrna.fa.gz',
	'kluyveromyces_dobzhanskii_cbs_2104_gca_000820885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/kluyveromyces_dobzhanskii_cbs_2104_gca_000820885/ncrna/Kluyveromyces_dobzhanskii_cbs_2104_gca_000820885.KLDO_01.ncrna.fa.gz',
	'kluyveromyces_lactis_gca_000002515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kluyveromyces_lactis_gca_000002515/ncrna/Kluyveromyces_lactis_gca_000002515.ASM251v1.ncrna.fa.gz',
	'kluyveromyces_marxianus_dmku3_1042_gca_001417885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/kluyveromyces_marxianus_dmku3_1042_gca_001417885/ncrna/Kluyveromyces_marxianus_dmku3_1042_gca_001417885.Kmar_1.0.ncrna.fa.gz',
	'kluyveromyces_marxianus_gca_001417835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/kluyveromyces_marxianus_gca_001417835/ncrna/Kluyveromyces_marxianus_gca_001417835.KM1777_03.ncrna.fa.gz',
	'kockovaella_imperatae_gca_002102565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kockovaella_imperatae_gca_002102565/ncrna/Kockovaella_imperatae_gca_002102565.Kocim1.ncrna.fa.gz',
	'komagataella_pastoris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/komagataella_pastoris/ncrna/Komagataella_pastoris.GCA_000027005.1.ncrna.fa.gz',
	'komagataella_pastoris_gca_001708105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/komagataella_pastoris_gca_001708105/ncrna/Komagataella_pastoris_gca_001708105.ASM170810v1.ncrna.fa.gz',
	'komagataella_phaffii_cbs_7435_gca_000223565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/komagataella_phaffii_cbs_7435_gca_000223565/ncrna/Komagataella_phaffii_cbs_7435_gca_000223565.PicPas_Mar2011.ncrna.fa.gz',
	'komagataella_phaffii_gca_001708085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/komagataella_phaffii_gca_001708085/ncrna/Komagataella_phaffii_gca_001708085.ASM170808v1.ncrna.fa.gz',
	'komagataella_phaffii_gs115_gca_001746955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/komagataella_phaffii_gs115_gca_001746955/ncrna/Komagataella_phaffii_gs115_gca_001746955.ASM174695v1.ncrna.fa.gz',
	'kryptolebias_marmoratus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/kryptolebias_marmoratus/ncrna/Kryptolebias_marmoratus.ASM164957v1.ncrna.fa.gz',
	'kuraishia_capsulata_cbs_1993_gca_000576695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kuraishia_capsulata_cbs_1993_gca_000576695/ncrna/Kuraishia_capsulata_cbs_1993_gca_000576695.AUH_PRJEB4427_v1.ncrna.fa.gz',
	'kwoniella_bestiolae_cbs_10118_gca_000512585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_bestiolae_cbs_10118_gca_000512585/ncrna/Kwoniella_bestiolae_cbs_10118_gca_000512585.Cryp_best_CBS10118_V1.ncrna.fa.gz',
	'kwoniella_dejecticola_cbs_10117_gca_000512565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_dejecticola_cbs_10117_gca_000512565/ncrna/Kwoniella_dejecticola_cbs_10117_gca_000512565.Cryp_deje_CBS10117_V1.ncrna.fa.gz',
	'kwoniella_heveanensis_bcc8398_gca_000507405'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_heveanensis_bcc8398_gca_000507405/ncrna/Kwoniella_heveanensis_bcc8398_gca_000507405.Cryp_heve_BCC8398_V1.ncrna.fa.gz',
	'kwoniella_heveanensis_cbs_569_gca_000507425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_heveanensis_cbs_569_gca_000507425/ncrna/Kwoniella_heveanensis_cbs_569_gca_000507425.Cryp_heve_CBS569_V2.ncrna.fa.gz',
	'kwoniella_mangroviensis_cbs_10435_gca_000507885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_mangroviensis_cbs_10435_gca_000507885/ncrna/Kwoniella_mangroviensis_cbs_10435_gca_000507885.Kwon_mang_CBS10435_V2.ncrna.fa.gz',
	'kwoniella_mangroviensis_cbs_8507_gca_000507465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_mangroviensis_cbs_8507_gca_000507465/ncrna/Kwoniella_mangroviensis_cbs_8507_gca_000507465.Kwon_mang_CBS8507_V2.ncrna.fa.gz',
	'kwoniella_mangroviensis_cbs_8886_gca_000507485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_mangroviensis_cbs_8886_gca_000507485/ncrna/Kwoniella_mangroviensis_cbs_8886_gca_000507485.Kwon_mang_CBS8886_V1.ncrna.fa.gz',
	'kwoniella_pini_cbs_10737_gca_000512605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_pini_cbs_10737_gca_000512605/ncrna/Kwoniella_pini_cbs_10737_gca_000512605.Cryp_pinu_CBS10737_V1.ncrna.fa.gz',
	'labrus_bergylta'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/labrus_bergylta/ncrna/Labrus_bergylta.BallGen_V1.ncrna.fa.gz',
	'laccaria_amethystina_laam_08_1_gca_000827195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/laccaria_amethystina_laam_08_1_gca_000827195/ncrna/Laccaria_amethystina_laam_08_1_gca_000827195.Laccaria_amethystina_LaAM-08-1_v1.0.ncrna.fa.gz',
	'laccaria_bicolor_s238n_h82_gca_000143565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/laccaria_bicolor_s238n_h82_gca_000143565/ncrna/Laccaria_bicolor_s238n_h82_gca_000143565.V1.0.ncrna.fa.gz',
	'lachancea_dasiensis_cbs_10888_gca_900074725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_dasiensis_cbs_10888_gca_900074725/ncrna/Lachancea_dasiensis_cbs_10888_gca_900074725.LADA0.ncrna.fa.gz',
	'lachancea_fermentati_gca_900074765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_fermentati_gca_900074765/ncrna/Lachancea_fermentati_gca_900074765.LAFE0.ncrna.fa.gz',
	'lachancea_lanzarotensis_gca_000938715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/lachancea_lanzarotensis_gca_000938715/ncrna/Lachancea_lanzarotensis_gca_000938715.LALA0.ncrna.fa.gz',
	'lachancea_meyersii_cbs_8951_gca_900074715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_meyersii_cbs_8951_gca_900074715/ncrna/Lachancea_meyersii_cbs_8951_gca_900074715.LAME0.ncrna.fa.gz',
	'lachancea_mirantina_gca_900074745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_mirantina_gca_900074745/ncrna/Lachancea_mirantina_gca_900074745.LAMI0.ncrna.fa.gz',
	'lachancea_nothofagi_cbs_11611_gca_900074755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_nothofagi_cbs_11611_gca_900074755/ncrna/Lachancea_nothofagi_cbs_11611_gca_900074755.LANO0.ncrna.fa.gz',
	'lachancea_quebecensis_gca_002900925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/lachancea_quebecensis_gca_002900925/ncrna/Lachancea_quebecensis_gca_002900925.ASM290092v1.ncrna.fa.gz',
	'lachancea_sp_cbs_6924_gca_900074735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_sp_cbs_6924_gca_900074735/ncrna/Lachancea_sp_cbs_6924_gca_900074735.LAFA0.ncrna.fa.gz',
	'lachancea_thermotolerans_cbs_6340_gca_000142805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/lachancea_thermotolerans_cbs_6340_gca_000142805/ncrna/Lachancea_thermotolerans_cbs_6340_gca_000142805.ASM14280v1.ncrna.fa.gz',
	'laetiporus_sulphureus_93_53_gca_001632365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/laetiporus_sulphureus_93_53_gca_001632365/ncrna/Laetiporus_sulphureus_93_53_gca_001632365.Laesu1.ncrna.fa.gz',
	'latimeria_chalumnae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/latimeria_chalumnae/ncrna/Latimeria_chalumnae.LatCha1.ncrna.fa.gz',
	'leersia_perrieri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/leersia_perrieri/ncrna/Leersia_perrieri.Lperr_V1.4.ncrna.fa.gz',
	'leishmania_braziliensis_mhom_br_75_m2904'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_braziliensis_mhom_br_75_m2904/ncrna/Leishmania_braziliensis_mhom_br_75_m2904.ASM284v2.ncrna.fa.gz',
	'leishmania_donovani_bpk282a1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_donovani_bpk282a1/ncrna/Leishmania_donovani_bpk282a1.ASM22713v2.ncrna.fa.gz',
	'leishmania_infantum_jpcm5'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_infantum_jpcm5/ncrna/Leishmania_infantum_jpcm5.ASM287v2.ncrna.fa.gz',
	'leishmania_major'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/leishmania_major/ncrna/Leishmania_major.ASM272v2.ncrna.fa.gz',
	'leishmania_mexicana_mhom_gt_2001_u1103'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_mexicana_mhom_gt_2001_u1103/ncrna/Leishmania_mexicana_mhom_gt_2001_u1103.ASM23466v4.ncrna.fa.gz',
	'leishmania_panamensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_panamensis/ncrna/Leishmania_panamensis.ASM75516v1.ncrna.fa.gz',
	'lepeophtheirus_salmonis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lepeophtheirus_salmonis/ncrna/Lepeophtheirus_salmonis.LSalAtl2s.ncrna.fa.gz',
	'lepidothrix_coronata'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/lepidothrix_coronata/ncrna/Lepidothrix_coronata.Lepidothrix_coronata-1.0.ncrna.fa.gz',
	'lepisosteus_oculatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/lepisosteus_oculatus/ncrna/Lepisosteus_oculatus.LepOcu1.ncrna.fa.gz',
	'leptomonas_pyrrhocoris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leptomonas_pyrrhocoris/ncrna/Leptomonas_pyrrhocoris.ASM129339v1.ncrna.fa.gz',
	'leptomonas_seymouri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leptomonas_seymouri/ncrna/Leptomonas_seymouri.ASM129953v1.ncrna.fa.gz',
	'leptosphaeria_maculans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/leptosphaeria_maculans/ncrna/Leptosphaeria_maculans.ASM23037v1.ncrna.fa.gz',
	'leucoagaricus_sp_symc_cos_gca_001563735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/leucoagaricus_sp_symc_cos_gca_001563735/ncrna/Leucoagaricus_sp_symc_cos_gca_001563735.ASM156373v1.ncrna.fa.gz',
	'leucosporidium_creatinivorum_gca_002105055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/leucosporidium_creatinivorum_gca_002105055/ncrna/Leucosporidium_creatinivorum_gca_002105055.Leucr1.ncrna.fa.gz',
	'lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665/ncrna/Lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665.454Minimus.ncrna.fa.gz',
	'lingula_anatina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lingula_anatina/ncrna/Lingula_anatina.LinAna1.0.ncrna.fa.gz',
	'lipomyces_starkeyi_nrrl_y_11557_gca_001661325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lipomyces_starkeyi_nrrl_y_11557_gca_001661325/ncrna/Lipomyces_starkeyi_nrrl_y_11557_gca_001661325.Lipst1_1.ncrna.fa.gz',
	'loa_loa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/loa_loa/ncrna/Loa_loa.Loa_loa_V3.ncrna.fa.gz',
	'lobosporangium_transversale_gca_002105155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/lobosporangium_transversale_gca_002105155/ncrna/Lobosporangium_transversale_gca_002105155.Lobtra1.ncrna.fa.gz',
	'lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685/ncrna/Lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685.ASM14968v1.ncrna.fa.gz',
	'lomentospora_prolificans_gca_002276285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/lomentospora_prolificans_gca_002276285/ncrna/Lomentospora_prolificans_gca_002276285.Lprolificans_pilon.ncrna.fa.gz',
	'lonchura_striata_domestica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/lonchura_striata_domestica/ncrna/Lonchura_striata_domestica.LonStrDom1.ncrna.fa.gz',
	'lotharella_oceanica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/lotharella_oceanica/ncrna/Lotharella_oceanica.ASM69843v2.ncrna.fa.gz',
	'lottia_gigantea'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lottia_gigantea/ncrna/Lottia_gigantea.Lotgi1.ncrna.fa.gz',
	'loxodonta_africana'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/loxodonta_africana/ncrna/Loxodonta_africana.loxAfr3.ncrna.fa.gz',
	'lucilia_cuprina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lucilia_cuprina/ncrna/Lucilia_cuprina.ASM118794v1.ncrna.fa.gz',
	'lupinus_angustifolius'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/lupinus_angustifolius/ncrna/Lupinus_angustifolius.LupAngTanjil_v1.0.ncrna.fa.gz',
	'macaca_fascicularis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/macaca_fascicularis/ncrna/Macaca_fascicularis.Macaca_fascicularis_5.0.ncrna.fa.gz',
	'macaca_mulatta'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/macaca_mulatta/ncrna/Macaca_mulatta.Mmul_8.0.1.ncrna.fa.gz',
	'macaca_nemestrina'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/macaca_nemestrina/ncrna/Macaca_nemestrina.Mnem_1.0.ncrna.fa.gz',
	'macrophomina_phaseolina_ms6_gca_000302655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/macrophomina_phaseolina_ms6_gca_000302655/ncrna/Macrophomina_phaseolina_ms6_gca_000302655.MphMS6_1.0.ncrna.fa.gz',
	'madurella_mycetomatis_gca_001275765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/madurella_mycetomatis_gca_001275765/ncrna/Madurella_mycetomatis_gca_001275765.ASM127576v2.ncrna.fa.gz',
	'magnaporthe_oryzae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/magnaporthe_oryzae/ncrna/Magnaporthe_oryzae.MG8.ncrna.fa.gz',
	'magnaporthe_poae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/magnaporthe_poae/ncrna/Magnaporthe_poae.Mag_poae_ATCC_64411_V1.ncrna.fa.gz',
	'malassezia_globosa_cbs_7966_gca_000181695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_globosa_cbs_7966_gca_000181695/ncrna/Malassezia_globosa_cbs_7966_gca_000181695.ASM18169v1.ncrna.fa.gz',
	'malassezia_pachydermatis_gca_001278385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_pachydermatis_gca_001278385/ncrna/Malassezia_pachydermatis_gca_001278385.MalaPachy.ncrna.fa.gz',
	'malassezia_sympodialis_atcc_42132_gca_000349305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_sympodialis_atcc_42132_gca_000349305/ncrna/Malassezia_sympodialis_atcc_42132_gca_000349305.ASM34930v2.ncrna.fa.gz',
	'malassezia_sympodialis_atcc_42132_gca_900149145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_sympodialis_atcc_42132_gca_900149145/ncrna/Malassezia_sympodialis_atcc_42132_gca_900149145.Msy_ATCC_42132_Feb2016.ncrna.fa.gz',
	'malassezia_vespertilionis_gca_002818225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_vespertilionis_gca_002818225/ncrna/Malassezia_vespertilionis_gca_002818225.ASM281822v1.ncrna.fa.gz',
	'manacus_vitellinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/manacus_vitellinus/ncrna/Manacus_vitellinus.ASM171598v2.ncrna.fa.gz',
	'mandrillus_leucophaeus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mandrillus_leucophaeus/ncrna/Mandrillus_leucophaeus.Mleu.le_1.0.ncrna.fa.gz',
	'manihot_esculenta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/manihot_esculenta/ncrna/Manihot_esculenta.Manihot_esculenta_v6.ncrna.fa.gz',
	'marmota_marmota_marmota'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/marmota_marmota_marmota/ncrna/Marmota_marmota_marmota.marMar2.1.ncrna.fa.gz',
	'marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775/ncrna/Marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775.ASM29877v1.ncrna.fa.gz',
	'mastacembelus_armatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mastacembelus_armatus/ncrna/Mastacembelus_armatus.fMasArm1.1.ncrna.fa.gz',
	'mayetiola_destructor'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/mayetiola_destructor/ncrna/Mayetiola_destructor.Mdes_1.0.ncrna.fa.gz',
	'maylandia_zebra'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/maylandia_zebra/ncrna/Maylandia_zebra.M_zebra_UMD2a.ncrna.fa.gz',
	'medicago_truncatula'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/medicago_truncatula/ncrna/Medicago_truncatula.MedtrA17_4.0.ncrna.fa.gz',
	'megaselia_scalaris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/megaselia_scalaris/ncrna/Megaselia_scalaris.Msca1.ncrna.fa.gz',
	'meira_miltonrushii_gca_003144205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/meira_miltonrushii_gca_003144205/ncrna/Meira_miltonrushii_gca_003144205.Meimi1.ncrna.fa.gz',
	'melampsora_laricipopulina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/melampsora_laricipopulina/ncrna/Melampsora_laricipopulina.GCA_000204055.1.ncrna.fa.gz',
	'meleagris_gallopavo'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/meleagris_gallopavo/ncrna/Meleagris_gallopavo.Turkey_2.01.ncrna.fa.gz',
	'meliniomyces_bicolor_e_gca_002865645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/meliniomyces_bicolor_e_gca_002865645/ncrna/Meliniomyces_bicolor_e_gca_002865645.Melbi2.ncrna.fa.gz',
	'meliniomyces_variabilis_f_gca_002865655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/meliniomyces_variabilis_f_gca_002865655/ncrna/Meliniomyces_variabilis_f_gca_002865655.Melva1.ncrna.fa.gz',
	'melitaea_cinxia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/melitaea_cinxia/ncrna/Melitaea_cinxia.MelCinx1.0.ncrna.fa.gz',
	'melopsittacus_undulatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/melopsittacus_undulatus/ncrna/Melopsittacus_undulatus.Melopsittacus_undulatus_6.3.ncrna.fa.gz',
	'meriones_unguiculatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/meriones_unguiculatus/ncrna/Meriones_unguiculatus.MunDraft-v1.0.ncrna.fa.gz',
	'mesocricetus_auratus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mesocricetus_auratus/ncrna/Mesocricetus_auratus.MesAur1.0.ncrna.fa.gz',
	'metarhizium_acridum_cqma_102_gca_000187405'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_acridum_cqma_102_gca_000187405/ncrna/Metarhizium_acridum_cqma_102_gca_000187405.MetAcr_May2010.ncrna.fa.gz',
	'metarhizium_album_arsef_1941_gca_000804445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/metarhizium_album_arsef_1941_gca_000804445/ncrna/Metarhizium_album_arsef_1941_gca_000804445.MAM_1.0_for_version_1_of_the_Metarhizium_album_genome.ncrna.fa.gz',
	'metarhizium_anisopliae_arsef_23_gca_000187425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_anisopliae_arsef_23_gca_000187425/ncrna/Metarhizium_anisopliae_arsef_23_gca_000187425.MAA_2.0.ncrna.fa.gz',
	'metarhizium_anisopliae_brip_53293_gca_000426965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/metarhizium_anisopliae_brip_53293_gca_000426965/ncrna/Metarhizium_anisopliae_brip_53293_gca_000426965.ASM42696v1.ncrna.fa.gz',
	'metarhizium_anisopliae_gca_000739145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_anisopliae_gca_000739145/ncrna/Metarhizium_anisopliae_gca_000739145.Metarhizium_anisopliae.ncrna.fa.gz',
	'metarhizium_majus_arsef_297_gca_000814945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/metarhizium_majus_arsef_297_gca_000814945/ncrna/Metarhizium_majus_arsef_297_gca_000814945.MAJ_1.0.ncrna.fa.gz',
	'metarhizium_robertsii_gca_000591435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_robertsii_gca_000591435/ncrna/Metarhizium_robertsii_gca_000591435.MR2010_11_27.ncrna.fa.gz',
	'metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035/ncrna/Metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035.Metbi1.ncrna.fa.gz',
	'meyerozyma_guilliermondii_atcc_6260_gca_000149425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/meyerozyma_guilliermondii_atcc_6260_gca_000149425/ncrna/Meyerozyma_guilliermondii_atcc_6260_gca_000149425.ASM14942v1.ncrna.fa.gz',
	'microbotryum_violaceum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/microbotryum_violaceum/ncrna/Microbotryum_violaceum.GCA_000166175.1.ncrna.fa.gz',
	'microcebus_murinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/microcebus_murinus/ncrna/Microcebus_murinus.Mmur_3.0.ncrna.fa.gz',
	'microdochium_bolleyi_gca_001566295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/microdochium_bolleyi_gca_001566295/ncrna/Microdochium_bolleyi_gca_001566295.Microdochium_bolleyi_v1.0.ncrna.fa.gz',
	'microsporum_canis_cbs_113480_gca_000151145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/microsporum_canis_cbs_113480_gca_000151145/ncrna/Microsporum_canis_cbs_113480_gca_000151145.ASM15114v1.ncrna.fa.gz',
	'microtus_ochrogaster'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/microtus_ochrogaster/ncrna/Microtus_ochrogaster.MicOch1.0.ncrna.fa.gz',
	'millerozyma_farinosa_cbs_7064_gca_000315895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/millerozyma_farinosa_cbs_7064_gca_000315895/ncrna/Millerozyma_farinosa_cbs_7064_gca_000315895.ASM31589v1.ncrna.fa.gz',
	'mitosporidium_daphniae_gca_000760515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/mitosporidium_daphniae_gca_000760515/ncrna/Mitosporidium_daphniae_gca_000760515.UGP1.0.ncrna.fa.gz',
	'mixia_osmundae_iam_14324_gca_000241205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/mixia_osmundae_iam_14324_gca_000241205/ncrna/Mixia_osmundae_iam_14324_gca_000241205.Mosmundae_2.0.ncrna.fa.gz',
	'mnemiopsis_leidyi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/mnemiopsis_leidyi/ncrna/Mnemiopsis_leidyi.MneLei_Aug2011.ncrna.fa.gz',
	'moelleriella_libera_rcef_2490_gca_001636675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/moelleriella_libera_rcef_2490_gca_001636675/ncrna/Moelleriella_libera_rcef_2490_gca_001636675.AAL_1.0.ncrna.fa.gz',
	'moesziomyces_antarcticus_gca_000747765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moesziomyces_antarcticus_gca_000747765/ncrna/Moesziomyces_antarcticus_gca_000747765.ASM74776v1.ncrna.fa.gz',
	'moesziomyces_antarcticus_t_34_gca_000334475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moesziomyces_antarcticus_t_34_gca_000334475/ncrna/Moesziomyces_antarcticus_t_34_gca_000334475.Pant_1.0.ncrna.fa.gz',
	'mola_mola'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mola_mola/ncrna/Mola_mola.ASM169857v1.ncrna.fa.gz',
	'moniliophthora_perniciosa_fa553_gca_000183025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moniliophthora_perniciosa_fa553_gca_000183025/ncrna/Moniliophthora_perniciosa_fa553_gca_000183025.ASM18302v1.ncrna.fa.gz',
	'moniliophthora_roreri_gca_001466705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moniliophthora_roreri_gca_001466705/ncrna/Moniliophthora_roreri_gca_001466705.ASM146670v1.ncrna.fa.gz',
	'moniliophthora_roreri_mca_2997_gca_000488995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moniliophthora_roreri_mca_2997_gca_000488995/ncrna/Moniliophthora_roreri_mca_2997_gca_000488995.M_roreri_MCA_2997_v1.ncrna.fa.gz',
	'monodelphis_domestica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/monodelphis_domestica/ncrna/Monodelphis_domestica.monDom5.ncrna.fa.gz',
	'monopterus_albus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/monopterus_albus/ncrna/Monopterus_albus.M_albus_1.0.ncrna.fa.gz',
	'mortierella_elongata_ag_77_gca_001651415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mortierella_elongata_ag_77_gca_001651415/ncrna/Mortierella_elongata_ag_77_gca_001651415.Mortierella_elongata_AG-77_v2.0.ncrna.fa.gz',
	'mortierella_verticillata_nrrl_6337_gca_000739165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mortierella_verticillata_nrrl_6337_gca_000739165/ncrna/Mortierella_verticillata_nrrl_6337_gca_000739165.Mort_vert_NRRL_6337_V1.ncrna.fa.gz',
	'mucor_ambiguus_gca_000950595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mucor_ambiguus_gca_000950595/ncrna/Mucor_ambiguus_gca_000950595.ASM95059v1.ncrna.fa.gz',
	'mucor_circinelloides_f_circinelloides_1006phl_gca_000401635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mucor_circinelloides_f_circinelloides_1006phl_gca_000401635/ncrna/Mucor_circinelloides_f_circinelloides_1006phl_gca_000401635.Muco_sp_1006Ph_V1.ncrna.fa.gz',
	'mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945/ncrna/Mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945.Mucci2.ncrna.fa.gz',
	'mus_caroli'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_caroli/ncrna/Mus_caroli.CAROLI_EIJ_v1.1.ncrna.fa.gz',
	'mus_musculus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus/ncrna/Mus_musculus.GRCm38.ncrna.fa.gz',
	'mus_musculus_129s1svimj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_129s1svimj/ncrna/Mus_musculus_129s1svimj.129S1_SvImJ_v1.ncrna.fa.gz',
	'mus_musculus_aj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_aj/ncrna/Mus_musculus_aj.A_J_v1.ncrna.fa.gz',
	'mus_musculus_akrj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_akrj/ncrna/Mus_musculus_akrj.AKR_J_v1.ncrna.fa.gz',
	'mus_musculus_balbcj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_balbcj/ncrna/Mus_musculus_balbcj.BALB_cJ_v1.ncrna.fa.gz',
	'mus_musculus_c3hhej'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_c3hhej/ncrna/Mus_musculus_c3hhej.C3H_HeJ_v1.ncrna.fa.gz',
	'mus_musculus_c57bl6nj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_c57bl6nj/ncrna/Mus_musculus_c57bl6nj.C57BL_6NJ_v1.ncrna.fa.gz',
	'mus_musculus_casteij'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_casteij/ncrna/Mus_musculus_casteij.CAST_EiJ_v1.ncrna.fa.gz',
	'mus_musculus_cbaj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_cbaj/ncrna/Mus_musculus_cbaj.CBA_J_v1.ncrna.fa.gz',
	'mus_musculus_dba2j'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_dba2j/ncrna/Mus_musculus_dba2j.DBA_2J_v1.ncrna.fa.gz',
	'mus_musculus_fvbnj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_fvbnj/ncrna/Mus_musculus_fvbnj.FVB_NJ_v1.ncrna.fa.gz',
	'mus_musculus_lpj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_lpj/ncrna/Mus_musculus_lpj.LP_J_v1.ncrna.fa.gz',
	'mus_musculus_nodshiltj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_nodshiltj/ncrna/Mus_musculus_nodshiltj.NOD_ShiLtJ_v1.ncrna.fa.gz',
	'mus_musculus_nzohlltj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_nzohlltj/ncrna/Mus_musculus_nzohlltj.NZO_HlLtJ_v1.ncrna.fa.gz',
	'mus_musculus_pwkphj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_pwkphj/ncrna/Mus_musculus_pwkphj.PWK_PhJ_v1.ncrna.fa.gz',
	'mus_musculus_wsbeij'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_wsbeij/ncrna/Mus_musculus_wsbeij.WSB_EiJ_v1.ncrna.fa.gz',
	'mus_pahari'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_pahari/ncrna/Mus_pahari.PAHARI_EIJ_v1.1.ncrna.fa.gz',
	'mus_spicilegus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_spicilegus/ncrna/Mus_spicilegus.MUSP714.ncrna.fa.gz',
	'mus_spretus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_spretus/ncrna/Mus_spretus.SPRET_EiJ_v1.ncrna.fa.gz',
	'musa_acuminata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/musa_acuminata/ncrna/Musa_acuminata.ASM31385v1.ncrna.fa.gz',
	'mustela_putorius_furo'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mustela_putorius_furo/ncrna/Mustela_putorius_furo.MusPutFur1.0.ncrna.fa.gz',
	'mycosphaerella_eumusae_gca_001578235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/mycosphaerella_eumusae_gca_001578235/ncrna/Mycosphaerella_eumusae_gca_001578235.ASM157823v1.ncrna.fa.gz',
	'myotis_lucifugus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/myotis_lucifugus/ncrna/Myotis_lucifugus.Myoluc2.0.ncrna.fa.gz',
	'nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315/ncrna/Nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315.Nadfu1.ncrna.fa.gz',
	'naegleria_gruberi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_heterolobosea1_collection/naegleria_gruberi/ncrna/Naegleria_gruberi.V1.0.ncrna.fa.gz',
	'naematelia_encephala_gca_002105065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/naematelia_encephala_gca_002105065/ncrna/Naematelia_encephala_gca_002105065.Treen1.ncrna.fa.gz',
	'nannizzia_gypsea_cbs_118893_gca_000150975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/nannizzia_gypsea_cbs_118893_gca_000150975/ncrna/Nannizzia_gypsea_cbs_118893_gca_000150975.MS_CBS118893.ncrna.fa.gz',
	'nannochloropsis_gaditana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/nannochloropsis_gaditana/ncrna/Nannochloropsis_gaditana.NagaB31_1.0.ncrna.fa.gz',
	'nannospalax_galili'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/nannospalax_galili/ncrna/Nannospalax_galili.S.galili_v1.0.ncrna.fa.gz',
	'nasonia_vitripennis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/nasonia_vitripennis/ncrna/Nasonia_vitripennis.Nvit_2.1.ncrna.fa.gz',
	'naumovozyma_castellii_cbs_4309_gca_000237345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/naumovozyma_castellii_cbs_4309_gca_000237345/ncrna/Naumovozyma_castellii_cbs_4309_gca_000237345.ASM23734v1.ncrna.fa.gz',
	'naumovozyma_dairenensis_cbs_421_gca_000227115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/naumovozyma_dairenensis_cbs_421_gca_000227115/ncrna/Naumovozyma_dairenensis_cbs_421_gca_000227115.ASM22711v2.ncrna.fa.gz',
	'nematocida_parisii_ertm1_gca_000250985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_parisii_ertm1_gca_000250985/ncrna/Nematocida_parisii_ertm1_gca_000250985.Nema_parisii_ERTm1_V3.ncrna.fa.gz',
	'nematocida_parisii_ertm3_gca_000190615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_parisii_ertm3_gca_000190615/ncrna/Nematocida_parisii_ertm3_gca_000190615.Nema_parisii_ERTm3_V1.ncrna.fa.gz',
	'nematocida_sp_1_ertm2_gca_000250695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_sp_1_ertm2_gca_000250695/ncrna/Nematocida_sp_1_ertm2_gca_000250695.Nema_parisii_ERTm2_V1.ncrna.fa.gz',
	'nematocida_sp_1_ertm6_gca_000738915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_sp_1_ertm6_gca_000738915/ncrna/Nematocida_sp_1_ertm6_gca_000738915.Nema_sp_1_ERTm6_V2.ncrna.fa.gz',
	'nematocida_sp_ertm5_gca_001642415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_sp_ertm5_gca_001642415/ncrna/Nematocida_sp_ertm5_gca_001642415.ASM164241v1.ncrna.fa.gz',
	'nematostella_vectensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/nematostella_vectensis/ncrna/Nematostella_vectensis.ASM20922v1.ncrna.fa.gz',
	'neocallimastix_californiae_gca_002104975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_neocallimastigomycota1_collection/neocallimastix_californiae_gca_002104975/ncrna/Neocallimastix_californiae_gca_002104975.Neocallimastix_sp._G1_v1.0.ncrna.fa.gz',
	'neofusicoccum_parvum_ucrnp2_gca_000385595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/neofusicoccum_parvum_ucrnp2_gca_000385595/ncrna/Neofusicoccum_parvum_ucrnp2_gca_000385595.UCRNP2V3.ncrna.fa.gz',
	'neolamprologus_brichardi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/neolamprologus_brichardi/ncrna/Neolamprologus_brichardi.NeoBri1.0.ncrna.fa.gz',
	'neolecta_irregularis_dah_3_gca_001929475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/neolecta_irregularis_dah_3_gca_001929475/ncrna/Neolecta_irregularis_dah_3_gca_001929475.NeoIirr1.0.ncrna.fa.gz',
	'neolentinus_lepideus_hhb14362_ss_1_gca_001632425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/neolentinus_lepideus_hhb14362_ss_1_gca_001632425/ncrna/Neolentinus_lepideus_hhb14362_ss_1_gca_001632425.Neole1.ncrna.fa.gz',
	'neonectria_ditissima_gca_001306435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/neonectria_ditissima_gca_001306435/ncrna/Neonectria_ditissima_gca_001306435.R0905_v2.0.ncrna.fa.gz',
	'neosartorya_fischeri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/neosartorya_fischeri/ncrna/Neosartorya_fischeri.ASM14964v1.ncrna.fa.gz',
	'neovison_vison'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/neovison_vison/ncrna/Neovison_vison.NNQGG.v01.ncrna.fa.gz',
	'neurospora_crassa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/neurospora_crassa/ncrna/Neurospora_crassa.NC12.ncrna.fa.gz',
	'neurospora_crassa_gca_000786625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/neurospora_crassa_gca_000786625/ncrna/Neurospora_crassa_gca_000786625.Neucr_trp3_1.ncrna.fa.gz',
	'neurospora_tetrasperma_fgsc_2509_gca_000213195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/neurospora_tetrasperma_fgsc_2509_gca_000213195/ncrna/Neurospora_tetrasperma_fgsc_2509_gca_000213195.v1.0.ncrna.fa.gz',
	'nicotiana_attenuata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/nicotiana_attenuata/ncrna/Nicotiana_attenuata.NIATTr2.ncrna.fa.gz',
	'nomascus_leucogenys'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/nomascus_leucogenys/ncrna/Nomascus_leucogenys.Nleu_3.0.ncrna.fa.gz',
	'nosema_apis_brl_01_gca_000447185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_apis_brl_01_gca_000447185/ncrna/Nosema_apis_brl_01_gca_000447185.NapisBRLv01.ncrna.fa.gz',
	'nosema_bombycis_cq1_gca_000383075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_bombycis_cq1_gca_000383075/ncrna/Nosema_bombycis_cq1_gca_000383075.NosBomCQ1_v1.0.ncrna.fa.gz',
	'nosema_ceranae_brl01_gca_000182985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_ceranae_brl01_gca_000182985/ncrna/Nosema_ceranae_brl01_gca_000182985.ASM18298v1.ncrna.fa.gz',
	'nosema_ceranae_gca_000988165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_ceranae_gca_000988165/ncrna/Nosema_ceranae_gca_000988165.ASM98816v1.ncrna.fa.gz',
	'notamacropus_eugenii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/notamacropus_eugenii/ncrna/Notamacropus_eugenii.Meug_1.0.ncrna.fa.gz',
	'notechis_scutatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/notechis_scutatus/ncrna/Notechis_scutatus.TS10Xv2-PRI.ncrna.fa.gz',
	'nothoprocta_perdicaria'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/nothoprocta_perdicaria/ncrna/Nothoprocta_perdicaria.notPer1.ncrna.fa.gz',
	'numida_meleagris'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/numida_meleagris/ncrna/Numida_meleagris.NumMel1.0.ncrna.fa.gz',
	'ochotona_princeps'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ochotona_princeps/ncrna/Ochotona_princeps.OchPri2.0-Ens.ncrna.fa.gz',
	'octodon_degus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/octodon_degus/ncrna/Octodon_degus.OctDeg1.0.ncrna.fa.gz',
	'octopus_bimaculoides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/octopus_bimaculoides/ncrna/Octopus_bimaculoides.PRJNA270931.ncrna.fa.gz',
	'ogataea_parapolymorpha_dl_1_gca_000187245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ogataea_parapolymorpha_dl_1_gca_000187245/ncrna/Ogataea_parapolymorpha_dl_1_gca_000187245.Hansenula_2.ncrna.fa.gz',
	'ogataea_polymorpha_gca_001664045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/ogataea_polymorpha_gca_001664045/ncrna/Ogataea_polymorpha_gca_001664045.Hanpo2.ncrna.fa.gz',
	'oidiodendron_maius_zn_gca_000827325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/oidiodendron_maius_zn_gca_000827325/ncrna/Oidiodendron_maius_zn_gca_000827325.Oidiodendron_maius_Zn_v1.0.ncrna.fa.gz',
	'onchocerca_volvulus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/onchocerca_volvulus/ncrna/Onchocerca_volvulus.ASM49940v2.ncrna.fa.gz',
	'ophiocordyceps_sinensis_co18_gca_000448365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ophiocordyceps_sinensis_co18_gca_000448365/ncrna/Ophiocordyceps_sinensis_co18_gca_000448365.ASM44836v1.ncrna.fa.gz',
	'ophiostoma_piceae_uamh_11346_gca_000410735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ophiostoma_piceae_uamh_11346_gca_000410735/ncrna/Ophiostoma_piceae_uamh_11346_gca_000410735.Opiceae_UAMHv01.ncrna.fa.gz',
	'ordospora_colligata_oc4_gca_000803265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/ordospora_colligata_oc4_gca_000803265/ncrna/Ordospora_colligata_oc4_gca_000803265.ASM80326v1.ncrna.fa.gz',
	'oreochromis_niloticus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oreochromis_niloticus/ncrna/Oreochromis_niloticus.Orenil1.0.ncrna.fa.gz',
	'ornithorhynchus_anatinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ornithorhynchus_anatinus/ncrna/Ornithorhynchus_anatinus.OANA5.ncrna.fa.gz',
	'oryctolagus_cuniculus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryctolagus_cuniculus/ncrna/Oryctolagus_cuniculus.OryCun2.0.ncrna.fa.gz',
	'oryza_barthii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_barthii/ncrna/Oryza_barthii.O.barthii_v1.ncrna.fa.gz',
	'oryza_brachyantha'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_brachyantha/ncrna/Oryza_brachyantha.Oryza_brachyantha.v1.4b.ncrna.fa.gz',
	'oryza_glaberrima'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_glaberrima/ncrna/Oryza_glaberrima.Oryza_glaberrima_V1.ncrna.fa.gz',
	'oryza_glumipatula'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_glumipatula/ncrna/Oryza_glumipatula.Oryza_glumaepatula_v1.5.ncrna.fa.gz',
	'oryza_indica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_indica/ncrna/Oryza_indica.ASM465v1.ncrna.fa.gz',
	'oryza_longistaminata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_longistaminata/ncrna/Oryza_longistaminata.O_longistaminata_v1.0.ncrna.fa.gz',
	'oryza_meridionalis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_meridionalis/ncrna/Oryza_meridionalis.Oryza_meridionalis_v1.3.ncrna.fa.gz',
	'oryza_nivara'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_nivara/ncrna/Oryza_nivara.Oryza_nivara_v1.0.ncrna.fa.gz',
	'oryza_punctata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_punctata/ncrna/Oryza_punctata.Oryza_punctata_v1.2.ncrna.fa.gz',
	'oryza_rufipogon'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_rufipogon/ncrna/Oryza_rufipogon.OR_W1943.ncrna.fa.gz',
	'oryza_sativa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_sativa/ncrna/Oryza_sativa.IRGSP-1.0.ncrna.fa.gz',
	'oryzias_latipes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_latipes/ncrna/Oryzias_latipes.ASM223467v1.ncrna.fa.gz',
	'oryzias_latipes_hni'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_latipes_hni/ncrna/Oryzias_latipes_hni.ASM223471v1.ncrna.fa.gz',
	'oryzias_latipes_hsok'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_latipes_hsok/ncrna/Oryzias_latipes_hsok.ASM223469v1.ncrna.fa.gz',
	'oryzias_melastigma'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_melastigma/ncrna/Oryzias_melastigma.Om_v0.7.RACA.ncrna.fa.gz',
	'ostreococcus_lucimarinus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/ostreococcus_lucimarinus/ncrna/Ostreococcus_lucimarinus.ASM9206v1.ncrna.fa.gz',
	'otolemur_garnettii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/otolemur_garnettii/ncrna/Otolemur_garnettii.OtoGar3.ncrna.fa.gz',
	'ovis_aries'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ovis_aries/ncrna/Ovis_aries.Oar_v3.1.ncrna.fa.gz',
	'pachysolen_tannophilus_nrrl_y_2460_gca_001661245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pachysolen_tannophilus_nrrl_y_2460_gca_001661245/ncrna/Pachysolen_tannophilus_nrrl_y_2460_gca_001661245.Pacta1_2.ncrna.fa.gz',
	'pan_paniscus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pan_paniscus/ncrna/Pan_paniscus.panpan1.1.ncrna.fa.gz',
	'pan_troglodytes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pan_troglodytes/ncrna/Pan_troglodytes.Pan_tro_3.0.ncrna.fa.gz',
	'panthera_pardus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/panthera_pardus/ncrna/Panthera_pardus.PanPar1.0.ncrna.fa.gz',
	'panthera_tigris_altaica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/panthera_tigris_altaica/ncrna/Panthera_tigris_altaica.PanTig1.0.ncrna.fa.gz',
	'papio_anubis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/papio_anubis/ncrna/Papio_anubis.Panu_3.0.ncrna.fa.gz',
	'paracoccidioides_brasiliensis_gca_001713645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/paracoccidioides_brasiliensis_gca_001713645/ncrna/Paracoccidioides_brasiliensis_gca_001713645.ASM171364v1.ncrna.fa.gz',
	'paracoccidioides_brasiliensis_gca_001713695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/paracoccidioides_brasiliensis_gca_001713695/ncrna/Paracoccidioides_brasiliensis_gca_001713695.Paracocci_br_PbCNH.ncrna.fa.gz',
	'paracoccidioides_brasiliensis_pb03_gca_000150475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/paracoccidioides_brasiliensis_pb03_gca_000150475/ncrna/Paracoccidioides_brasiliensis_pb03_gca_000150475.Paracocci_br_Pb03_V2.ncrna.fa.gz',
	'paracoccidioides_brasiliensis_pb18_gca_000150735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/paracoccidioides_brasiliensis_pb18_gca_000150735/ncrna/Paracoccidioides_brasiliensis_pb18_gca_000150735.Paracocci_br_Pb18_V2.ncrna.fa.gz',
	'paracoccidioides_sp_lutzii_pb01_gca_000150705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/paracoccidioides_sp_lutzii_pb01_gca_000150705/ncrna/Paracoccidioides_sp_lutzii_pb01_gca_000150705.Paracocci_br_Pb01_V2.ncrna.fa.gz',
	'paramecium_tetraurelia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/paramecium_tetraurelia/ncrna/Paramecium_tetraurelia.ASM16542v1.ncrna.fa.gz',
	'paramicrosporidium_saccamoebae_gca_002794465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_rozellomycota1_collection/paramicrosporidium_saccamoebae_gca_002794465/ncrna/Paramicrosporidium_saccamoebae_gca_002794465.KSL3_v6.ncrna.fa.gz',
	'paramormyrops_kingsleyae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/paramormyrops_kingsleyae/ncrna/Paramormyrops_kingsleyae.PKINGS_0.1.ncrna.fa.gz',
	'paraphaeosphaeria_sporulosa_gca_001642045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/paraphaeosphaeria_sporulosa_gca_001642045/ncrna/Paraphaeosphaeria_sporulosa_gca_001642045.Parsp1.ncrna.fa.gz',
	'parus_major'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/parus_major/ncrna/Parus_major.Parus_major1.1.ncrna.fa.gz',
	'paulinella_chromatophora'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/paulinella_chromatophora/ncrna/Paulinella_chromatophora.ASM1792v1.ncrna.fa.gz',
	'paxillus_involutus_atcc_200175_gca_000827475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/paxillus_involutus_atcc_200175_gca_000827475/ncrna/Paxillus_involutus_atcc_200175_gca_000827475.Paxillus_involutus_ATCC_200175_v1.0.ncrna.fa.gz',
	'paxillus_rubicundulus_ve08_2h10_gca_000827395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/paxillus_rubicundulus_ve08_2h10_gca_000827395/ncrna/Paxillus_rubicundulus_ve08_2h10_gca_000827395.Paxillus_rubicundulus_Ve08.2h10_v1.0.ncrna.fa.gz',
	'pediculus_humanus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/pediculus_humanus/ncrna/Pediculus_humanus.PhumU2.ncrna.fa.gz',
	'pelodiscus_sinensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pelodiscus_sinensis/ncrna/Pelodiscus_sinensis.PelSin_1.0.ncrna.fa.gz',
	'penicilliopsis_zonata_cbs_506_65_gca_001890105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicilliopsis_zonata_cbs_506_65_gca_001890105/ncrna/Penicilliopsis_zonata_cbs_506_65_gca_001890105.Aspzo1.ncrna.fa.gz',
	'penicillium_brasilianum_gca_001048715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/penicillium_brasilianum_gca_001048715/ncrna/Penicillium_brasilianum_gca_001048715.Pbras_Allpaths-LG.ncrna.fa.gz',
	'penicillium_brasilianum_gca_002016555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_brasilianum_gca_002016555/ncrna/Penicillium_brasilianum_gca_002016555.ASM201655v1.ncrna.fa.gz',
	'penicillium_camemberti_fm_013_gca_000513335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/penicillium_camemberti_fm_013_gca_000513335/ncrna/Penicillium_camemberti_fm_013_gca_000513335.PCAMFM013_20131106.ncrna.fa.gz',
	'penicillium_chrysogenum_gca_000710275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_chrysogenum_gca_000710275/ncrna/Penicillium_chrysogenum_gca_000710275.ASM71027v1.ncrna.fa.gz',
	'penicillium_digitatum_pd1_gca_000315645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_digitatum_pd1_gca_000315645/ncrna/Penicillium_digitatum_pd1_gca_000315645.PdigPd1_v1.ncrna.fa.gz',
	'penicillium_digitatum_phi26_gca_000315665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_digitatum_phi26_gca_000315665/ncrna/Penicillium_digitatum_phi26_gca_000315665.PdigPHI26_v1.ncrna.fa.gz',
	'penicillium_expansum_gca_000769745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_expansum_gca_000769745/ncrna/Penicillium_expansum_gca_000769745.ASM76974v1.ncrna.fa.gz',
	'penicillium_freii_gca_001513925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_freii_gca_001513925/ncrna/Penicillium_freii_gca_001513925.ASM151392v1.ncrna.fa.gz',
	'penicillium_griseofulvum_gca_001561935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_griseofulvum_gca_001561935/ncrna/Penicillium_griseofulvum_gca_001561935.ASM156193v1.ncrna.fa.gz',
	'penicillium_italicum_gca_000769765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_italicum_gca_000769765/ncrna/Penicillium_italicum_gca_000769765.ASM76976v1.ncrna.fa.gz',
	'penicillium_nordicum_gca_001278595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_nordicum_gca_001278595/ncrna/Penicillium_nordicum_gca_001278595.ASM127859v1.ncrna.fa.gz',
	'penicillium_oxalicum_114_2_gca_000346795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_oxalicum_114_2_gca_000346795/ncrna/Penicillium_oxalicum_114_2_gca_000346795.pde_v1.0.ncrna.fa.gz',
	'penicillium_roqueforti_fm164_gca_000513255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_roqueforti_fm164_gca_000513255/ncrna/Penicillium_roqueforti_fm164_gca_000513255.PROQFM164_20130607.ncrna.fa.gz',
	'penicillium_rubens_wisconsin_54_1255_gca_000226395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_rubens_wisconsin_54_1255_gca_000226395/ncrna/Penicillium_rubens_wisconsin_54_1255_gca_000226395.PenChr_Nov2007.ncrna.fa.gz',
	'penicillium_subrubescens_gca_001908125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_subrubescens_gca_001908125/ncrna/Penicillium_subrubescens_gca_001908125.ASM190812v1.ncrna.fa.gz',
	'peniophora_sp_cont_gca_001632445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/peniophora_sp_cont_gca_001632445/ncrna/Peniophora_sp_cont_gca_001632445.Ricme1.ncrna.fa.gz',
	'periconia_macrospinosa_gca_003073855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/periconia_macrospinosa_gca_003073855/ncrna/Periconia_macrospinosa_gca_003073855.Perma1.ncrna.fa.gz',
	'periophthalmus_magnuspinnatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/periophthalmus_magnuspinnatus/ncrna/Periophthalmus_magnuspinnatus.PM.fa.ncrna.fa.gz',
	'perkinsus_marinus_atcc_50983'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/perkinsus_marinus_atcc_50983/ncrna/Perkinsus_marinus_atcc_50983.JCVI_PMG_1.0.ncrna.fa.gz',
	'peromyscus_maniculatus_bairdii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/peromyscus_maniculatus_bairdii/ncrna/Peromyscus_maniculatus_bairdii.HU_Pman_2.1.ncrna.fa.gz',
	'pestalotiopsis_fici_w106_1_gca_000516985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pestalotiopsis_fici_w106_1_gca_000516985/ncrna/Pestalotiopsis_fici_w106_1_gca_000516985.PFICI.ncrna.fa.gz',
	'petromyzon_marinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/petromyzon_marinus/ncrna/Petromyzon_marinus.Pmarinus_7.0.ncrna.fa.gz',
	'pezoloma_ericae_gca_002865625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pezoloma_ericae_gca_002865625/ncrna/Pezoloma_ericae_gca_002865625.Rhier1.ncrna.fa.gz',
	'phaeoacremonium_minimum_ucrpa7_gca_000392275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/phaeoacremonium_minimum_ucrpa7_gca_000392275/ncrna/Phaeoacremonium_minimum_ucrpa7_gca_000392275.UCRPA7V03.ncrna.fa.gz',
	'phaeodactylum_tricornutum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phaeodactylum_tricornutum/ncrna/Phaeodactylum_tricornutum.ASM15095v2.ncrna.fa.gz',
	'phaeomoniella_chlamydospora_gca_001006345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phaeomoniella_chlamydospora_gca_001006345/ncrna/Phaeomoniella_chlamydospora_gca_001006345.UCRPA7V1.0.ncrna.fa.gz',
	'phaeosphaeria_nodorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/phaeosphaeria_nodorum/ncrna/Phaeosphaeria_nodorum.ASM14691v1.ncrna.fa.gz',
	'phanerochaete_carnosa_hhb_10118_sp_gca_000300595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/phanerochaete_carnosa_hhb_10118_sp_gca_000300595/ncrna/Phanerochaete_carnosa_hhb_10118_sp_gca_000300595.Phanerochaete_carnosa_HHB-10118-Sp_v1.0.ncrna.fa.gz',
	'phascolarctos_cinereus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/phascolarctos_cinereus/ncrna/Phascolarctos_cinereus.phaCin_unsw_v4.1.ncrna.fa.gz',
	'phaseolus_vulgaris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/phaseolus_vulgaris/ncrna/Phaseolus_vulgaris.PhaVulg1_0.ncrna.fa.gz',
	'phialocephala_scopiformis_gca_001500285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phialocephala_scopiformis_gca_001500285/ncrna/Phialocephala_scopiformis_gca_001500285.Phisc1.ncrna.fa.gz',
	'phialocephala_subalpina_gca_900073065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phialocephala_subalpina_gca_900073065/ncrna/Phialocephala_subalpina_gca_900073065.PAC_version_1.ncrna.fa.gz',
	'phialophora_americana_gca_000835435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/phialophora_americana_gca_000835435/ncrna/Phialophora_americana_gca_000835435.Capr_semi_CBS27337_V1.ncrna.fa.gz',
	'phialophora_attae_gca_001299255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phialophora_attae_gca_001299255/ncrna/Phialophora_attae_gca_001299255.ASM129925v1.ncrna.fa.gz',
	'phlebia_centrifuga_gca_001913855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/phlebia_centrifuga_gca_001913855/ncrna/Phlebia_centrifuga_gca_001913855.ASM191385v2.ncrna.fa.gz',
	'phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265/ncrna/Phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265.Phlgi1.ncrna.fa.gz',
	'physcomitrella_patens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/physcomitrella_patens/ncrna/Physcomitrella_patens.Phypa_V3.ncrna.fa.gz',
	'phytophthora_infestans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_infestans/ncrna/Phytophthora_infestans.ASM14294v1.ncrna.fa.gz',
	'phytophthora_kernoviae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_kernoviae/ncrna/Phytophthora_kernoviae.PhyKer238_432v1.ncrna.fa.gz',
	'phytophthora_lateralis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_lateralis/ncrna/Phytophthora_lateralis.MPF4_v1.0.ncrna.fa.gz',
	'phytophthora_parasitica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_parasitica/ncrna/Phytophthora_parasitica.Phyt_para_P1569_V1.ncrna.fa.gz',
	'phytophthora_parasitica_cj01a1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_cj01a1/ncrna/Phytophthora_parasitica_cj01a1.Phyt_para_CJ01A1_V1.ncrna.fa.gz',
	'phytophthora_parasitica_gca_000509465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509465/ncrna/Phytophthora_parasitica_gca_000509465.Phyt_para_CJ02B3_V1.ncrna.fa.gz',
	'phytophthora_parasitica_gca_000509485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509485/ncrna/Phytophthora_parasitica_gca_000509485.Phyt_para_CJ05E6_V1.ncrna.fa.gz',
	'phytophthora_parasitica_gca_000509505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509505/ncrna/Phytophthora_parasitica_gca_000509505.Phyt_para_CHvinca01_V1.ncrna.fa.gz',
	'phytophthora_parasitica_gca_000509525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509525/ncrna/Phytophthora_parasitica_gca_000509525.Phyt_para_IAC_01_95_V1.ncrna.fa.gz',
	'phytophthora_parasitica_inra_310'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_inra_310/ncrna/Phytophthora_parasitica_inra_310.PP_INRA-310_V2.ncrna.fa.gz',
	'phytophthora_parasitica_p10297'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_p10297/ncrna/Phytophthora_parasitica_p10297.Phyt_para_P10297_V1.ncrna.fa.gz',
	'phytophthora_parasitica_p1976'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_p1976/ncrna/Phytophthora_parasitica_p1976.Phyt_para_P1976_V1.ncrna.fa.gz',
	'phytophthora_ramorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_ramorum/ncrna/Phytophthora_ramorum.ASM14973v1.ncrna.fa.gz',
	'phytophthora_sojae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_sojae/ncrna/Phytophthora_sojae.P_sojae_V3_0.ncrna.fa.gz',
	'pichia_kudriavzevii_gca_000764455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pichia_kudriavzevii_gca_000764455/ncrna/Pichia_kudriavzevii_gca_000764455.ASM76445v1.ncrna.fa.gz',
	'pichia_kudriavzevii_gca_002166775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pichia_kudriavzevii_gca_002166775/ncrna/Pichia_kudriavzevii_gca_002166775.Cand_krus_Ckrusei653_V21.ncrna.fa.gz',
	'pichia_kudriavzevii_gca_003054445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pichia_kudriavzevii_gca_003054445/ncrna/Pichia_kudriavzevii_gca_003054445.ASM305444v1.ncrna.fa.gz',
	'pichia_membranifaciens_gca_001950575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pichia_membranifaciens_gca_001950575/ncrna/Pichia_membranifaciens_gca_001950575.P_membrenifaciens-1.0.ncrna.fa.gz',
	'pichia_membranifaciens_nrrl_y_2026_gca_001661235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pichia_membranifaciens_nrrl_y_2026_gca_001661235/ncrna/Pichia_membranifaciens_nrrl_y_2026_gca_001661235.Picme2.ncrna.fa.gz',
	'piliocolobus_tephrosceles'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/piliocolobus_tephrosceles/ncrna/Piliocolobus_tephrosceles.ASM277652v2.ncrna.fa.gz',
	'piloderma_croceum_f_1598_gca_000827315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/piloderma_croceum_f_1598_gca_000827315/ncrna/Piloderma_croceum_f_1598_gca_000827315.Piloderma_croceum_F_1598_v1.0.ncrna.fa.gz',
	'piromyces_finnis_gca_002104945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_neocallimastigomycota1_collection/piromyces_finnis_gca_002104945/ncrna/Piromyces_finnis_gca_002104945.Piromyces_sp._finnis_v3.0.ncrna.fa.gz',
	'pisolithus_microcarpus_441_gca_000827275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pisolithus_microcarpus_441_gca_000827275/ncrna/Pisolithus_microcarpus_441_gca_000827275.Pisolithus_microcarpus_441_v1.0.ncrna.fa.gz',
	'pisolithus_tinctorius_marx_270_gca_000827335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pisolithus_tinctorius_marx_270_gca_000827335/ncrna/Pisolithus_tinctorius_marx_270_gca_000827335.Pisolithus_tinctorius_Marx_270_v1.0.ncrna.fa.gz',
	'plasmodium_berghei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_berghei/ncrna/Plasmodium_berghei.PBANKA01.ncrna.fa.gz',
	'plasmodium_berghei_gca_000005395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_berghei_gca_000005395/ncrna/Plasmodium_berghei_gca_000005395.ASM539v1.ncrna.fa.gz',
	'plasmodium_berghei_gca_900044335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_berghei_gca_900044335/ncrna/Plasmodium_berghei_gca_900044335.PbK173.ncrna.fa.gz',
	'plasmodium_chabaudi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_chabaudi/ncrna/Plasmodium_chabaudi.PCHAS01.ncrna.fa.gz',
	'plasmodium_chabaudi_chabaudi_gca_000003075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_chabaudi_chabaudi_gca_000003075/ncrna/Plasmodium_chabaudi_chabaudi_gca_000003075.ASM307v1.ncrna.fa.gz',
	'plasmodium_coatneyi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_coatneyi/ncrna/Plasmodium_coatneyi.ASM168000v1.ncrna.fa.gz',
	'plasmodium_cynomolgi_strain_b'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_cynomolgi_strain_b/ncrna/Plasmodium_cynomolgi_strain_b.PcynB_1.0.ncrna.fa.gz',
	'plasmodium_falciparum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_falciparum/ncrna/Plasmodium_falciparum.EPr1.ncrna.fa.gz',
	'plasmodium_falciparum_7g8'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_7g8/ncrna/Plasmodium_falciparum_7g8.Plas_falc_7G8_V1.ncrna.fa.gz',
	'plasmodium_falciparum_camp_malaysia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_camp_malaysia/ncrna/Plasmodium_falciparum_camp_malaysia.Plas_falc_Malayan_Camp_V1.ncrna.fa.gz',
	'plasmodium_falciparum_dd2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_dd2/ncrna/Plasmodium_falciparum_dd2.ASM14979v1.ncrna.fa.gz',
	'plasmodium_falciparum_fch_4'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_fch_4/ncrna/Plasmodium_falciparum_fch_4.Plas_falc_FCH_4_V2.ncrna.fa.gz',
	'plasmodium_falciparum_hb3'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_hb3/ncrna/Plasmodium_falciparum_hb3.ASM14966v2.ncrna.fa.gz',
	'plasmodium_falciparum_igh_cr14'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_igh_cr14/ncrna/Plasmodium_falciparum_igh_cr14.ASM18605v2.ncrna.fa.gz',
	'plasmodium_falciparum_malips096_e11'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_malips096_e11/ncrna/Plasmodium_falciparum_malips096_e11.Plas_falc_MaliPS096_E11_V1.ncrna.fa.gz',
	'plasmodium_falciparum_nf135_5_c10'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_nf135_5_c10/ncrna/Plasmodium_falciparum_nf135_5_c10.Plas_falc_NF135_5_C10_V1.ncrna.fa.gz',
	'plasmodium_falciparum_nf54'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_nf54/ncrna/Plasmodium_falciparum_nf54.Plas_falc_NF4_V1.ncrna.fa.gz',
	'plasmodium_falciparum_palo_alto_uganda'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_palo_alto_uganda/ncrna/Plasmodium_falciparum_palo_alto_uganda.Plas_falc_Uganda_Palo-Alto_FUP_H_V1.ncrna.fa.gz',
	'plasmodium_falciparum_raj116'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_raj116/ncrna/Plasmodium_falciparum_raj116.ASM18602v2.ncrna.fa.gz',
	'plasmodium_falciparum_santa_lucia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_santa_lucia/ncrna/Plasmodium_falciparum_santa_lucia.Plas_falc_Santa_Lucia_Salvador_I_V1.ncrna.fa.gz',
	'plasmodium_falciparum_tanzania_2000708_'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_tanzania_2000708_/ncrna/Plasmodium_falciparum_tanzania_2000708_.Plas_falc_02000708_V1.ncrna.fa.gz',
	'plasmodium_falciparum_ugt5_1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_ugt5_1/ncrna/Plasmodium_falciparum_ugt5_1.Plas_falc_5_1_V1.ncrna.fa.gz',
	'plasmodium_falciparum_vietnam_oak_knoll_fvo_'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_vietnam_oak_knoll_fvo_/ncrna/Plasmodium_falciparum_vietnam_oak_knoll_fvo_.Plas_falc_Vietnam_Oak-Knoll_FVO_V1.ncrna.fa.gz',
	'plasmodium_fragile'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_fragile/ncrna/Plasmodium_fragile.Plas_frag_nilgiri_V1.ncrna.fa.gz',
	'plasmodium_gaboni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_gaboni/ncrna/Plasmodium_gaboni.ASM160202v1.ncrna.fa.gz',
	'plasmodium_inui_san_antonio_1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_inui_san_antonio_1/ncrna/Plasmodium_inui_san_antonio_1.Plas_inui_San_Antonio_1_V1.ncrna.fa.gz',
	'plasmodium_knowlesi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_knowlesi/ncrna/Plasmodium_knowlesi.ASM635v1.ncrna.fa.gz',
	'plasmodium_knowlesi_strain_h_gca_900004885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_knowlesi_strain_h_gca_900004885/ncrna/Plasmodium_knowlesi_strain_h_gca_900004885.PKNA1-C.2.ncrna.fa.gz',
	'plasmodium_malariae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_malariae/ncrna/Plasmodium_malariae.Pmal.ncrna.fa.gz',
	'plasmodium_ovale_curtisi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_ovale_curtisi/ncrna/Plasmodium_ovale_curtisi.Poc1.ncrna.fa.gz',
	'plasmodium_reichenowi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_reichenowi/ncrna/Plasmodium_reichenowi.PREICH001.ncrna.fa.gz',
	'plasmodium_reichenowi_gca_001601855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_reichenowi_gca_001601855/ncrna/Plasmodium_reichenowi_gca_001601855.ASM160185v1.ncrna.fa.gz',
	'plasmodium_vinckei_petteri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vinckei_petteri/ncrna/Plasmodium_vinckei_petteri.Plas_vinc_pett_CR_V1.ncrna.fa.gz',
	'plasmodium_vinckei_vinckei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vinckei_vinckei/ncrna/Plasmodium_vinckei_vinckei.Plas_vinc_vinckei_V1.ncrna.fa.gz',
	'plasmodium_vivax'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_vivax/ncrna/Plasmodium_vivax.ASM241v2.ncrna.fa.gz',
	'plasmodium_vivax_brazil_i'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_brazil_i/ncrna/Plasmodium_vivax_brazil_i.PV_Brazil_I_V1.ncrna.fa.gz',
	'plasmodium_vivax_india_vii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_india_vii/ncrna/Plasmodium_vivax_india_vii.PV_India_VII_V2.ncrna.fa.gz',
	'plasmodium_vivax_mauritania_i'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_mauritania_i/ncrna/Plasmodium_vivax_mauritania_i.PV_Mauritania_I_V1.ncrna.fa.gz',
	'plasmodium_vivax_north_korean'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_north_korean/ncrna/Plasmodium_vivax_north_korean.PV_N_Korean_V1.ncrna.fa.gz',
	'plasmodium_yoelii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii/ncrna/Plasmodium_yoelii.PY17X01.ncrna.fa.gz',
	'plasmodium_yoelii_17x'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii_17x/ncrna/Plasmodium_yoelii_17x.Plas_yoel_17X_V2.ncrna.fa.gz',
	'plasmodium_yoelii_gca_900002395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii_gca_900002395/ncrna/Plasmodium_yoelii_gca_900002395.PYYM01.ncrna.fa.gz',
	'plasmodium_yoelii_yoelii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii_yoelii/ncrna/Plasmodium_yoelii_yoelii.ASM308v2.ncrna.fa.gz',
	'plasmopara_halstedii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/plasmopara_halstedii/ncrna/Plasmopara_halstedii.Plasmopara_halstedii_genome.ncrna.fa.gz',
	'pleurotus_ostreatus_pc15_gca_000697685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pleurotus_ostreatus_pc15_gca_000697685/ncrna/Pleurotus_ostreatus_pc15_gca_000697685.PleosPC15_2.ncrna.fa.gz',
	'plicaturopsis_crispa_fd_325_ss_3_gca_000827205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/plicaturopsis_crispa_fd_325_ss_3_gca_000827205/ncrna/Plicaturopsis_crispa_fd_325_ss_3_gca_000827205.Plicr1.ncrna.fa.gz',
	'pneumocystis_carinii_b80_gca_001477545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pneumocystis_carinii_b80_gca_001477545/ncrna/Pneumocystis_carinii_b80_gca_001477545.Pneu_cari_B80_V3.ncrna.fa.gz',
	'pneumocystis_jirovecii_ru7_gca_001477535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pneumocystis_jirovecii_ru7_gca_001477535/ncrna/Pneumocystis_jirovecii_ru7_gca_001477535.Pneu_jiro_RU7_V2.ncrna.fa.gz',
	'pneumocystis_murina_b123_gca_000349005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pneumocystis_murina_b123_gca_000349005/ncrna/Pneumocystis_murina_b123_gca_000349005.Pneumo_murina_B123_V4.ncrna.fa.gz',
	'pochonia_chlamydosporia_170_gca_001653235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pochonia_chlamydosporia_170_gca_001653235/ncrna/Pochonia_chlamydosporia_170_gca_001653235.ASM165323v2.ncrna.fa.gz',
	'podospora_anserina_s_mat__gca_000226545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/podospora_anserina_s_mat__gca_000226545/ncrna/Podospora_anserina_s_mat__gca_000226545.ASM22654v1.ncrna.fa.gz',
	'poecilia_formosa'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_formosa/ncrna/Poecilia_formosa.PoeFor_5.1.2.ncrna.fa.gz',
	'poecilia_latipinna'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_latipinna/ncrna/Poecilia_latipinna.P_latipinna-1.0.ncrna.fa.gz',
	'poecilia_mexicana'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_mexicana/ncrna/Poecilia_mexicana.P_mexicana-1.0.ncrna.fa.gz',
	'poecilia_reticulata'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_reticulata/ncrna/Poecilia_reticulata.Guppy_female_1.0_MT.ncrna.fa.gz',
	'pogona_vitticeps'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pogona_vitticeps/ncrna/Pogona_vitticeps.pvi1.1.ncrna.fa.gz',
	'polysphondylium_pallidum_pn500'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/polysphondylium_pallidum_pn500/ncrna/Polysphondylium_pallidum_pn500.PolPal_Dec2009.ncrna.fa.gz',
	'pongo_abelii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pongo_abelii/ncrna/Pongo_abelii.PPYG2.ncrna.fa.gz',
	'populus_trichocarpa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/populus_trichocarpa/ncrna/Populus_trichocarpa.Pop_tri_v3.ncrna.fa.gz',
	'postia_placenta_mad_698_r_gca_000006255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/postia_placenta_mad_698_r_gca_000006255/ncrna/Postia_placenta_mad_698_r_gca_000006255.Postia_placenta_V1.0.ncrna.fa.gz',
	'postia_placenta_mad_698_r_sb12_gca_002117355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/postia_placenta_mad_698_r_sb12_gca_002117355/ncrna/Postia_placenta_mad_698_r_sb12_gca_002117355.PosplRSB12_1.ncrna.fa.gz',
	'procavia_capensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/procavia_capensis/ncrna/Procavia_capensis.proCap1.ncrna.fa.gz',
	'prolemur_simus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/prolemur_simus/ncrna/Prolemur_simus.Prosim_1.0.ncrna.fa.gz',
	'propithecus_coquereli'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/propithecus_coquereli/ncrna/Propithecus_coquereli.Pcoq_1.0.ncrna.fa.gz',
	'protomyces_lactucaedebilis_gca_002105105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/protomyces_lactucaedebilis_gca_002105105/ncrna/Protomyces_lactucaedebilis_gca_002105105.Prola1.ncrna.fa.gz',
	'prunus_persica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/prunus_persica/ncrna/Prunus_persica.Prunus_persica_NCBIv2.ncrna.fa.gz',
	'pseudocercospora_fijiensis_cirad86_gca_000340215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudocercospora_fijiensis_cirad86_gca_000340215/ncrna/Pseudocercospora_fijiensis_cirad86_gca_000340215.Mycfi2.ncrna.fa.gz',
	'pseudocercospora_musae_gca_001578225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudocercospora_musae_gca_001578225/ncrna/Pseudocercospora_musae_gca_001578225.ASM157822v1.ncrna.fa.gz',
	'pseudogymnoascus_destructans_20631_21_gca_000184105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudogymnoascus_destructans_20631_21_gca_000184105/ncrna/Pseudogymnoascus_destructans_20631_21_gca_000184105.Geom_destructans_V1.ncrna.fa.gz',
	'pseudogymnoascus_destructans_gca_001641265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_destructans_gca_001641265/ncrna/Pseudogymnoascus_destructans_gca_001641265.ASM164126v1.ncrna.fa.gz',
	'pseudogymnoascus_sp_03vt05_gca_001662645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_03vt05_gca_001662645/ncrna/Pseudogymnoascus_sp_03vt05_gca_001662645.ASM166264v1.ncrna.fa.gz',
	'pseudogymnoascus_sp_05ny08_gca_001662605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_05ny08_gca_001662605/ncrna/Pseudogymnoascus_sp_05ny08_gca_001662605.ASM166260v1.ncrna.fa.gz',
	'pseudogymnoascus_sp_23342_1_i1_gca_001662575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_23342_1_i1_gca_001662575/ncrna/Pseudogymnoascus_sp_23342_1_i1_gca_001662575.ASM166257v1.ncrna.fa.gz',
	'pseudogymnoascus_sp_24mn13_gca_001662595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_24mn13_gca_001662595/ncrna/Pseudogymnoascus_sp_24mn13_gca_001662595.ASM166259v1.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_103_gca_000750895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_103_gca_000750895/ncrna/Pseudogymnoascus_sp_vkm_f_103_gca_000750895.DG30.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_3557_gca_000750665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudogymnoascus_sp_vkm_f_3557_gca_000750665/ncrna/Pseudogymnoascus_sp_vkm_f_3557_gca_000750665.DG3.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_3775_gca_000750715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_3775_gca_000750715/ncrna/Pseudogymnoascus_sp_vkm_f_3775_gca_000750715.DG4.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_3808_gca_000750675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudogymnoascus_sp_vkm_f_3808_gca_000750675/ncrna/Pseudogymnoascus_sp_vkm_f_3808_gca_000750675.DG1.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4246_gca_000750735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4246_gca_000750735/ncrna/Pseudogymnoascus_sp_vkm_f_4246_gca_000750735.DG5.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745/ncrna/Pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745.DG6.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755/ncrna/Pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755.DG7.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805/ncrna/Pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805.DG10.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815/ncrna/Pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815.DG11.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875/ncrna/Pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875.DG13.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925/ncrna/Pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925.DG36.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935/ncrna/Pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935.DG38.ncrna.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995/ncrna/Pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995.DG39.ncrna.fa.gz',
	'pseudogymnoascus_sp_wsf_3629_gca_001662585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_wsf_3629_gca_001662585/ncrna/Pseudogymnoascus_sp_wsf_3629_gca_001662585.ASM166258v1.ncrna.fa.gz',
	'pseudogymnoascus_verrucosus_gca_001662655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_verrucosus_gca_001662655/ncrna/Pseudogymnoascus_verrucosus_gca_001662655.ASM166265v1.ncrna.fa.gz',
	'pseudoloma_neurophilia_gca_001432165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/pseudoloma_neurophilia_gca_001432165/ncrna/Pseudoloma_neurophilia_gca_001432165.ASM143216v1.ncrna.fa.gz',
	'pseudomassariella_vexata_gca_002105095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudomassariella_vexata_gca_002105095/ncrna/Pseudomassariella_vexata_gca_002105095.Pseve2.ncrna.fa.gz',
	'pseudomicrostroma_glucosiphilum_gca_003144135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pseudomicrostroma_glucosiphilum_gca_003144135/ncrna/Pseudomicrostroma_glucosiphilum_gca_003144135.Rhodsp1.ncrna.fa.gz',
	'pseudozyma_hubeiensis_sy62_gca_000403515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pseudozyma_hubeiensis_sy62_gca_000403515/ncrna/Pseudozyma_hubeiensis_sy62_gca_000403515.ASM40351v1.ncrna.fa.gz',
	'pteropus_vampyrus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pteropus_vampyrus/ncrna/Pteropus_vampyrus.pteVam1.ncrna.fa.gz',
	'puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275/ncrna/Puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275.ASM287327v1.ncrna.fa.gz',
	'puccinia_graminis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_graminis/ncrna/Puccinia_graminis.ASM14992v1.ncrna.fa.gz',
	'puccinia_graminisug99'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_graminisug99/ncrna/Puccinia_graminisug99.v1.ncrna.fa.gz',
	'puccinia_sorghi_gca_001263375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/puccinia_sorghi_gca_001263375/ncrna/Puccinia_sorghi_gca_001263375.ASM126337v1.ncrna.fa.gz',
	'puccinia_striiformis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_striiformis/ncrna/Puccinia_striiformis.PST-130_1.0.ncrna.fa.gz',
	'puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645/ncrna/Puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645.P_striiformis_V1.ncrna.fa.gz',
	'puccinia_triticina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_triticina/ncrna/Puccinia_triticina.ASM15152v1.ncrna.fa.gz',
	'punctularia_strigosozonata_hhb_11173_ss5_gca_000264995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/punctularia_strigosozonata_hhb_11173_ss5_gca_000264995/ncrna/Punctularia_strigosozonata_hhb_11173_ss5_gca_000264995.Punctularia_strigosozonata_v1.0.ncrna.fa.gz',
	'pundamilia_nyererei'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pundamilia_nyererei/ncrna/Pundamilia_nyererei.PunNye1.0.ncrna.fa.gz',
	'purpureocillium_lilacinum_gca_001653205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/purpureocillium_lilacinum_gca_001653205/ncrna/Purpureocillium_lilacinum_gca_001653205.ASM165320v1.ncrna.fa.gz',
	'purpureocillium_lilacinum_gca_001653265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/purpureocillium_lilacinum_gca_001653265/ncrna/Purpureocillium_lilacinum_gca_001653265.ASM165326v1.ncrna.fa.gz',
	'pygocentrus_nattereri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pygocentrus_nattereri/ncrna/Pygocentrus_nattereri.Pygocentrus_nattereri-1.0.2.ncrna.fa.gz',
	'pyrenochaeta_sp_ds3say3a_gca_001644535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pyrenochaeta_sp_ds3say3a_gca_001644535/ncrna/Pyrenochaeta_sp_ds3say3a_gca_001644535.Pyrsp1.ncrna.fa.gz',
	'pyrenophora_teres'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/pyrenophora_teres/ncrna/Pyrenophora_teres.GCA_000166005.1.ncrna.fa.gz',
	'pyrenophora_tritici_repentis_gca_003171545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003171545/ncrna/Pyrenophora_tritici_repentis_gca_003171545.PtrARCrossB10.ncrna.fa.gz',
	'pyrenophora_tritici_repentis_gca_003231325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003231325/ncrna/Pyrenophora_tritici_repentis_gca_003231325.Ptr134.ncrna.fa.gz',
	'pyrenophora_tritici_repentis_gca_003231365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003231365/ncrna/Pyrenophora_tritici_repentis_gca_003231365.Ptr239.ncrna.fa.gz',
	'pyrenophora_tritici_repentis_gca_003231425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003231425/ncrna/Pyrenophora_tritici_repentis_gca_003231425.Ptr86-124.ncrna.fa.gz',
	'pyrenophora_triticirepentis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/pyrenophora_triticirepentis/ncrna/Pyrenophora_triticirepentis.ASM14998v1.ncrna.fa.gz',
	'pythium_aphanidermatum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_aphanidermatum/ncrna/Pythium_aphanidermatum.pag1_scaffolds_v1.ncrna.fa.gz',
	'pythium_arrhenomanes'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_arrhenomanes/ncrna/Pythium_arrhenomanes.par_scaffolds_v1.ncrna.fa.gz',
	'pythium_irregulare'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_irregulare/ncrna/Pythium_irregulare.pir_scaffolds_v1.ncrna.fa.gz',
	'pythium_iwayamai'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_iwayamai/ncrna/Pythium_iwayamai.piw_scaffolds_v1.ncrna.fa.gz',
	'pythium_ultimum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_ultimum/ncrna/Pythium_ultimum.pug.ncrna.fa.gz',
	'pythium_vexans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_vexans/ncrna/Pythium_vexans.pve_scaffolds_v1.ncrna.fa.gz',
	'rachicladosporium_antarcticum_gca_002077065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rachicladosporium_antarcticum_gca_002077065/ncrna/Rachicladosporium_antarcticum_gca_002077065.ASM207706v1.ncrna.fa.gz',
	'rachicladosporium_sp_ccfee_5018_gca_002077045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rachicladosporium_sp_ccfee_5018_gca_002077045/ncrna/Rachicladosporium_sp_ccfee_5018_gca_002077045.ASM207704v2.ncrna.fa.gz',
	'ramularia_collo_cygni_gca_900074925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/ramularia_collo_cygni_gca_900074925/ncrna/Ramularia_collo_cygni_gca_900074925.version_1.ncrna.fa.gz',
	'rasamsonia_emersonii_cbs_393_64_gca_000968595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/rasamsonia_emersonii_cbs_393_64_gca_000968595/ncrna/Rasamsonia_emersonii_cbs_393_64_gca_000968595.ASM96859v1.ncrna.fa.gz',
	'rattus_norvegicus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/rattus_norvegicus/ncrna/Rattus_norvegicus.Rnor_6.0.ncrna.fa.gz',
	'reticulomyxa_filosa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/reticulomyxa_filosa/ncrna/Reticulomyxa_filosa.Reti_assembly1.0.ncrna.fa.gz',
	'rhinocladiella_mackenziei_cbs_650_93_gca_000835555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/rhinocladiella_mackenziei_cbs_650_93_gca_000835555/ncrna/Rhinocladiella_mackenziei_cbs_650_93_gca_000835555.Rhin_mack_CBS_650_93_V1.ncrna.fa.gz',
	'rhinopithecus_bieti'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/rhinopithecus_bieti/ncrna/Rhinopithecus_bieti.ASM169854v1.ncrna.fa.gz',
	'rhinopithecus_roxellana'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/rhinopithecus_roxellana/ncrna/Rhinopithecus_roxellana.Rrox_v1.ncrna.fa.gz',
	'rhizoclosmatium_globosum_gca_002104985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/rhizoclosmatium_globosum_gca_002104985/ncrna/Rhizoclosmatium_globosum_gca_002104985.Rhihy1.ncrna.fa.gz',
	'rhizoctonia_solani_123e_gca_000715385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_123e_gca_000715385/ncrna/Rhizoctonia_solani_123e_gca_000715385.Rhizoctonia_solani_123E_v1.ncrna.fa.gz',
	'rhizoctonia_solani_ag_1_ia_gca_000334115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_ag_1_ia_gca_000334115/ncrna/Rhizoctonia_solani_ag_1_ia_gca_000334115.Rhisol_AG1IA_1.0.ncrna.fa.gz',
	'rhizoctonia_solani_ag_1_ib_gca_000350255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_ag_1_ib_gca_000350255/ncrna/Rhizoctonia_solani_ag_1_ib_gca_000350255.ASM35025v1.ncrna.fa.gz',
	'rhizoctonia_solani_ag_8_wac10335_gca_000695385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_ag_8_wac10335_gca_000695385/ncrna/Rhizoctonia_solani_ag_8_wac10335_gca_000695385.RSAG8-1.V1.ncrna.fa.gz',
	'rhizoctonia_solani_gca_001286725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_gca_001286725/ncrna/Rhizoctonia_solani_gca_001286725.AG22IIIB.ncrna.fa.gz',
	'rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155/ncrna/Rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155.Rir_HGAP_ii_V2.ncrna.fa.gz',
	'rhizophagus_irregularis_daom_181602_gca_000439145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_daom_181602_gca_000439145/ncrna/Rhizophagus_irregularis_daom_181602_gca_000439145.ASM43914v3.ncrna.fa.gz',
	'rhizophagus_irregularis_daom_197198w_gca_000597685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_daom_197198w_gca_000597685/ncrna/Rhizophagus_irregularis_daom_197198w_gca_000597685.Rir.ncrna.fa.gz',
	'rhizophagus_irregularis_gca_001593125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593125/ncrna/Rhizophagus_irregularis_gca_001593125.ASM159312v1.ncrna.fa.gz',
	'rhizophagus_irregularis_gca_001593145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593145/ncrna/Rhizophagus_irregularis_gca_001593145.ASM159314v1.ncrna.fa.gz',
	'rhizophagus_irregularis_gca_001593155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593155/ncrna/Rhizophagus_irregularis_gca_001593155.ASM159315v1.ncrna.fa.gz',
	'rhizophagus_irregularis_gca_001593205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593205/ncrna/Rhizophagus_irregularis_gca_001593205.ASM159320v1.ncrna.fa.gz',
	'rhizopogon_vinicolor_am_or11_026_gca_001658105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizopogon_vinicolor_am_or11_026_gca_001658105/ncrna/Rhizopogon_vinicolor_am_or11_026_gca_001658105.Rhivi1.ncrna.fa.gz',
	'rhizopus_delemar_ra_99_880_gca_000149305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_delemar_ra_99_880_gca_000149305/ncrna/Rhizopus_delemar_ra_99_880_gca_000149305.RO3.ncrna.fa.gz',
	'rhizopus_microsporus_atcc_52813_gca_002708625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_atcc_52813_gca_002708625/ncrna/Rhizopus_microsporus_atcc_52813_gca_002708625.Rhimi1_1.ncrna.fa.gz',
	'rhizopus_microsporus_gca_000825725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_gca_000825725/ncrna/Rhizopus_microsporus_gca_000825725.Rmicro_CBS_344.29_Allpaths-LG.ncrna.fa.gz',
	'rhizopus_microsporus_gca_002083735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_gca_002083735/ncrna/Rhizopus_microsporus_gca_002083735.Rhimi_ATCC11559_1.ncrna.fa.gz',
	'rhizopus_microsporus_var_microsporus_gca_002083745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_var_microsporus_gca_002083745/ncrna/Rhizopus_microsporus_var_microsporus_gca_002083745.Rhimi_ATCC52814_1.ncrna.fa.gz',
	'rhodnius_prolixus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/rhodnius_prolixus/ncrna/Rhodnius_prolixus.RproC3.ncrna.fa.gz',
	'rhodotorula_graminis_wp1_gca_001329695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_graminis_wp1_gca_001329695/ncrna/Rhodotorula_graminis_wp1_gca_001329695.Rhoba1_1.ncrna.fa.gz',
	'rhodotorula_sp_jg_1b_gca_001541205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_sp_jg_1b_gca_001541205/ncrna/Rhodotorula_sp_jg_1b_gca_001541205.Rhosp1.ncrna.fa.gz',
	'rhodotorula_taiwanensis_gca_002922495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_taiwanensis_gca_002922495/ncrna/Rhodotorula_taiwanensis_gca_002922495.ASM292249v1.ncrna.fa.gz',
	'rhodotorula_toruloides_atcc_204091_gca_000222205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_toruloides_atcc_204091_gca_000222205/ncrna/Rhodotorula_toruloides_atcc_204091_gca_000222205.R.glutinis2.ncrna.fa.gz',
	'rhodotorula_toruloides_np11_gca_000320785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_toruloides_np11_gca_000320785/ncrna/Rhodotorula_toruloides_np11_gca_000320785.RHOziaDV1.0.ncrna.fa.gz',
	'rhynchosporium_agropyri_gca_900074905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rhynchosporium_agropyri_gca_900074905/ncrna/Rhynchosporium_agropyri_gca_900074905.version_1.ncrna.fa.gz',
	'rhynchosporium_commune_gca_900074885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rhynchosporium_commune_gca_900074885/ncrna/Rhynchosporium_commune_gca_900074885.version_1.ncrna.fa.gz',
	'rhynchosporium_secalis_gca_900074895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rhynchosporium_secalis_gca_900074895/ncrna/Rhynchosporium_secalis_gca_900074895.version_1.ncrna.fa.gz',
	'rosellinia_necatrix_gca_001445595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rosellinia_necatrix_gca_001445595/ncrna/Rosellinia_necatrix_gca_001445595.Rnecatrix_2.0.ncrna.fa.gz',
	'rozella_allomycis_csf55_gca_000442015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_rozellomycota1_collection/rozella_allomycis_csf55_gca_000442015/ncrna/Rozella_allomycis_csf55_gca_000442015.Rozella_k41_t100.ncrna.fa.gz',
	'saccharomyces_arboricola_h_6_gca_000292725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_arboricola_h_6_gca_000292725/ncrna/Saccharomyces_arboricola_h_6_gca_000292725.SacArb1.0.ncrna.fa.gz',
	'saccharomyces_cerevisiae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/saccharomyces_cerevisiae/ncrna/Saccharomyces_cerevisiae.R64-1-1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_ec1118_gca_000218975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_ec1118_gca_000218975/ncrna/Saccharomyces_cerevisiae_ec1118_gca_000218975.ASM21897v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_fostersb_gca_000190255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_fostersb_gca_000190255/ncrna/Saccharomyces_cerevisiae_fostersb_gca_000190255.ScFostersB_v01.ncrna.fa.gz',
	'saccharomyces_cerevisiae_fosterso_gca_000326005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_fosterso_gca_000326005/ncrna/Saccharomyces_cerevisiae_fosterso_gca_000326005.ScFostersO_v01.ncrna.fa.gz',
	'saccharomyces_cerevisiae_gca_001634645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_gca_001634645/ncrna/Saccharomyces_cerevisiae_gca_001634645.GLBRCY22-3.ncrna.fa.gz',
	'saccharomyces_cerevisiae_gca_002804325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_gca_002804325/ncrna/Saccharomyces_cerevisiae_gca_002804325.ASM280432v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_kyokai_no_7_gca_000260735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_kyokai_no_7_gca_000260735/ncrna/Saccharomyces_cerevisiae_kyokai_no_7_gca_000260735.K7_1.0.ncrna.fa.gz',
	'saccharomyces_cerevisiae_lalvin_qa23_gca_000325965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_lalvin_qa23_gca_000325965/ncrna/Saccharomyces_cerevisiae_lalvin_qa23_gca_000325965.ScQA23_v01.ncrna.fa.gz',
	'saccharomyces_cerevisiae_p283_gca_000568295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_p283_gca_000568295/ncrna/Saccharomyces_cerevisiae_p283_gca_000568295.P283.ncrna.fa.gz',
	'saccharomyces_cerevisiae_p301_gca_000568055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_p301_gca_000568055/ncrna/Saccharomyces_cerevisiae_p301_gca_000568055.p301.ncrna.fa.gz',
	'saccharomyces_cerevisiae_r103_gca_000568365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_r103_gca_000568365/ncrna/Saccharomyces_cerevisiae_r103_gca_000568365.R103.ncrna.fa.gz',
	'saccharomyces_cerevisiae_rm11_1a_gca_000149365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_rm11_1a_gca_000149365/ncrna/Saccharomyces_cerevisiae_rm11_1a_gca_000149365.ASM14936v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_vin13_gca_000190215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_vin13_gca_000190215/ncrna/Saccharomyces_cerevisiae_vin13_gca_000190215.ScVIN13_v01.ncrna.fa.gz',
	'saccharomyces_cerevisiae_vl3_gca_000190235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_vl3_gca_000190235/ncrna/Saccharomyces_cerevisiae_vl3_gca_000190235.ScVL3_v01.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1078_gca_000975645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1078_gca_000975645/ncrna/Saccharomyces_cerevisiae_yjm1078_gca_000975645.Sc_YJM1078_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1083_gca_000975675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1083_gca_000975675/ncrna/Saccharomyces_cerevisiae_yjm1083_gca_000975675.Sc_YJM1083_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1129_gca_000975705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1129_gca_000975705/ncrna/Saccharomyces_cerevisiae_yjm1129_gca_000975705.Sc_YJM1129_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1133_gca_000976695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1133_gca_000976695/ncrna/Saccharomyces_cerevisiae_yjm1133_gca_000976695.Sc_YJM1133_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1190_gca_000976725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1190_gca_000976725/ncrna/Saccharomyces_cerevisiae_yjm1190_gca_000976725.Sc_YJM1190_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1199_gca_000976755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1199_gca_000976755/ncrna/Saccharomyces_cerevisiae_yjm1199_gca_000976755.Sc_YJM1199_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1202_gca_000976785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1202_gca_000976785/ncrna/Saccharomyces_cerevisiae_yjm1202_gca_000976785.Sc_YJM1202_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1208_gca_000976815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1208_gca_000976815/ncrna/Saccharomyces_cerevisiae_yjm1208_gca_000976815.Sc_YJM1208_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1242_gca_000976845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1242_gca_000976845/ncrna/Saccharomyces_cerevisiae_yjm1242_gca_000976845.Sc_YJM1242_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1244_gca_000976875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1244_gca_000976875/ncrna/Saccharomyces_cerevisiae_yjm1244_gca_000976875.Sc_YJM1244_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1248_gca_000976905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1248_gca_000976905/ncrna/Saccharomyces_cerevisiae_yjm1248_gca_000976905.Sc_YJM1248_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1250_gca_000976935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1250_gca_000976935/ncrna/Saccharomyces_cerevisiae_yjm1250_gca_000976935.Sc_YJM1250_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1252_gca_000976965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1252_gca_000976965/ncrna/Saccharomyces_cerevisiae_yjm1252_gca_000976965.Sc_YJM1252_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1273_gca_000976995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1273_gca_000976995/ncrna/Saccharomyces_cerevisiae_yjm1273_gca_000976995.Sc_YJM1273_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1304_gca_000977025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1304_gca_000977025/ncrna/Saccharomyces_cerevisiae_yjm1304_gca_000977025.Sc_YJM1304_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1307_gca_000977055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1307_gca_000977055/ncrna/Saccharomyces_cerevisiae_yjm1307_gca_000977055.Sc_YJM1307_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1311_gca_000977085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1311_gca_000977085/ncrna/Saccharomyces_cerevisiae_yjm1311_gca_000977085.Sc_YJM1311_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1326_gca_000977115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1326_gca_000977115/ncrna/Saccharomyces_cerevisiae_yjm1326_gca_000977115.Sc_YJM1326_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1332_gca_000977145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1332_gca_000977145/ncrna/Saccharomyces_cerevisiae_yjm1332_gca_000977145.Sc_YJM1332_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1336_gca_000977175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1336_gca_000977175/ncrna/Saccharomyces_cerevisiae_yjm1336_gca_000977175.Sc_YJM1336_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1338_gca_000977205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1338_gca_000977205/ncrna/Saccharomyces_cerevisiae_yjm1338_gca_000977205.Sc_YJM1338_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1341_gca_000977235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1341_gca_000977235/ncrna/Saccharomyces_cerevisiae_yjm1341_gca_000977235.Sc_YJM1341_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1342_gca_000977265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1342_gca_000977265/ncrna/Saccharomyces_cerevisiae_yjm1342_gca_000977265.Sc_YJM1342_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1355_gca_000977295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1355_gca_000977295/ncrna/Saccharomyces_cerevisiae_yjm1355_gca_000977295.Sc_YJM1355_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1356_gca_000977325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1356_gca_000977325/ncrna/Saccharomyces_cerevisiae_yjm1356_gca_000977325.Sc_YJM1356_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1381_gca_000977355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1381_gca_000977355/ncrna/Saccharomyces_cerevisiae_yjm1381_gca_000977355.Sc_YJM1381_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1383_gca_000977385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1383_gca_000977385/ncrna/Saccharomyces_cerevisiae_yjm1383_gca_000977385.Sc_YJM1383_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1385_gca_000977415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1385_gca_000977415/ncrna/Saccharomyces_cerevisiae_yjm1385_gca_000977415.Sc_YJM1385_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1386_gca_000977445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1386_gca_000977445/ncrna/Saccharomyces_cerevisiae_yjm1386_gca_000977445.Sc_YJM1386_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1387_gca_000977475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1387_gca_000977475/ncrna/Saccharomyces_cerevisiae_yjm1387_gca_000977475.Sc_YJM1387_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1388_gca_000977505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1388_gca_000977505/ncrna/Saccharomyces_cerevisiae_yjm1388_gca_000977505.Sc_YJM1388_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1389_gca_000977535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1389_gca_000977535/ncrna/Saccharomyces_cerevisiae_yjm1389_gca_000977535.Sc_YJM1389_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1399_gca_000977565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1399_gca_000977565/ncrna/Saccharomyces_cerevisiae_yjm1399_gca_000977565.Sc_YJM1399_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1400_gca_000977595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1400_gca_000977595/ncrna/Saccharomyces_cerevisiae_yjm1400_gca_000977595.Sc_YJM1400_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1401_gca_000977625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1401_gca_000977625/ncrna/Saccharomyces_cerevisiae_yjm1401_gca_000977625.Sc_YJM1401_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1402_gca_000977655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1402_gca_000977655/ncrna/Saccharomyces_cerevisiae_yjm1402_gca_000977655.Sc_YJM1402_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1415_gca_000977685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1415_gca_000977685/ncrna/Saccharomyces_cerevisiae_yjm1415_gca_000977685.Sc_YJM1415_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1417_gca_000977715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1417_gca_000977715/ncrna/Saccharomyces_cerevisiae_yjm1417_gca_000977715.Sc_YJM1417_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1418_gca_000977745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1418_gca_000977745/ncrna/Saccharomyces_cerevisiae_yjm1418_gca_000977745.Sc_YJM1418_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1419_gca_000977775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1419_gca_000977775/ncrna/Saccharomyces_cerevisiae_yjm1419_gca_000977775.Sc_YJM1419_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1433_gca_000977805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1433_gca_000977805/ncrna/Saccharomyces_cerevisiae_yjm1433_gca_000977805.Sc_YJM1433_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1434_gca_000977835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1434_gca_000977835/ncrna/Saccharomyces_cerevisiae_yjm1434_gca_000977835.Sc_YJM1434_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1439_gca_000977865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1439_gca_000977865/ncrna/Saccharomyces_cerevisiae_yjm1439_gca_000977865.Sc_YJM1439_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1443_gca_000977895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1443_gca_000977895/ncrna/Saccharomyces_cerevisiae_yjm1443_gca_000977895.Sc_YJM1443_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1444_gca_000977925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1444_gca_000977925/ncrna/Saccharomyces_cerevisiae_yjm1444_gca_000977925.Sc_YJM1444_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1447_gca_000977955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1447_gca_000977955/ncrna/Saccharomyces_cerevisiae_yjm1447_gca_000977955.Sc_YJM1447_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1450_gca_000977985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1450_gca_000977985/ncrna/Saccharomyces_cerevisiae_yjm1450_gca_000977985.Sc_YJM1450_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1460_gca_000978015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1460_gca_000978015/ncrna/Saccharomyces_cerevisiae_yjm1460_gca_000978015.Sc_YJM1460_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1463_gca_000978045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1463_gca_000978045/ncrna/Saccharomyces_cerevisiae_yjm1463_gca_000978045.Sc_YJM1463_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1477_gca_000978075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1477_gca_000978075/ncrna/Saccharomyces_cerevisiae_yjm1477_gca_000978075.Sc_YJM1477_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1478_gca_000978105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1478_gca_000978105/ncrna/Saccharomyces_cerevisiae_yjm1478_gca_000978105.Sc_YJM1478_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1479_gca_000978135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1479_gca_000978135/ncrna/Saccharomyces_cerevisiae_yjm1479_gca_000978135.Sc_YJM1479_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1526_gca_000978165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1526_gca_000978165/ncrna/Saccharomyces_cerevisiae_yjm1526_gca_000978165.Sc_YJM1526_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1527_gca_000978195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1527_gca_000978195/ncrna/Saccharomyces_cerevisiae_yjm1527_gca_000978195.Sc_YJM1527_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1549_gca_000978225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1549_gca_000978225/ncrna/Saccharomyces_cerevisiae_yjm1549_gca_000978225.Sc_YJM1549_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1573_gca_000978255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1573_gca_000978255/ncrna/Saccharomyces_cerevisiae_yjm1573_gca_000978255.Sc_YJM1573_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1574_gca_000978285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1574_gca_000978285/ncrna/Saccharomyces_cerevisiae_yjm1574_gca_000978285.Sc_YJM1574_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1592_gca_000978315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1592_gca_000978315/ncrna/Saccharomyces_cerevisiae_yjm1592_gca_000978315.Sc_YJM1592_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm1615_gca_000978345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1615_gca_000978345/ncrna/Saccharomyces_cerevisiae_yjm1615_gca_000978345.Sc_YJM1615_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm189_gca_000975735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm189_gca_000975735/ncrna/Saccharomyces_cerevisiae_yjm189_gca_000975735.Sc_YJM189_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm193_gca_000975765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm193_gca_000975765/ncrna/Saccharomyces_cerevisiae_yjm193_gca_000975765.Sc_YJM193_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm195_gca_000975585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm195_gca_000975585/ncrna/Saccharomyces_cerevisiae_yjm195_gca_000975585.Sc_YJM195_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm244_gca_000975615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm244_gca_000975615/ncrna/Saccharomyces_cerevisiae_yjm244_gca_000975615.Sc_YJM244_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm248_gca_000975795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm248_gca_000975795/ncrna/Saccharomyces_cerevisiae_yjm248_gca_000975795.Sc_YJM248_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm270_gca_000975825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm270_gca_000975825/ncrna/Saccharomyces_cerevisiae_yjm270_gca_000975825.Sc_YJM270_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm271_gca_000975855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm271_gca_000975855/ncrna/Saccharomyces_cerevisiae_yjm271_gca_000975855.Sc_YJM271_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm320_gca_000975885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm320_gca_000975885/ncrna/Saccharomyces_cerevisiae_yjm320_gca_000975885.Sc_YJM320_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm326_gca_000975915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm326_gca_000975915/ncrna/Saccharomyces_cerevisiae_yjm326_gca_000975915.Sc_YJM326_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm428_gca_000975945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm428_gca_000975945/ncrna/Saccharomyces_cerevisiae_yjm428_gca_000975945.Sc_YJM428_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm450_gca_000975975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm450_gca_000975975/ncrna/Saccharomyces_cerevisiae_yjm450_gca_000975975.Sc_YJM450_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm451_gca_000976005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm451_gca_000976005/ncrna/Saccharomyces_cerevisiae_yjm451_gca_000976005.Sc_YJM451_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm453_gca_000976035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm453_gca_000976035/ncrna/Saccharomyces_cerevisiae_yjm453_gca_000976035.Sc_YJM453_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm456_gca_000976065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm456_gca_000976065/ncrna/Saccharomyces_cerevisiae_yjm456_gca_000976065.Sc_YJM456_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm470_gca_000976095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm470_gca_000976095/ncrna/Saccharomyces_cerevisiae_yjm470_gca_000976095.Sc_YJM470_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm541_gca_000976125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm541_gca_000976125/ncrna/Saccharomyces_cerevisiae_yjm541_gca_000976125.Sc_YJM541_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm554_gca_000976155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm554_gca_000976155/ncrna/Saccharomyces_cerevisiae_yjm554_gca_000976155.Sc_YJM554_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm555_gca_000976185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm555_gca_000976185/ncrna/Saccharomyces_cerevisiae_yjm555_gca_000976185.Sc_YJM555_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm627_gca_000976215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm627_gca_000976215/ncrna/Saccharomyces_cerevisiae_yjm627_gca_000976215.Sc_YJM627_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm681_gca_000976245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm681_gca_000976245/ncrna/Saccharomyces_cerevisiae_yjm681_gca_000976245.Sc_YJM681_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm682_gca_000976275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm682_gca_000976275/ncrna/Saccharomyces_cerevisiae_yjm682_gca_000976275.Sc_YJM682_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm683_gca_000976305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm683_gca_000976305/ncrna/Saccharomyces_cerevisiae_yjm683_gca_000976305.Sc_YJM683_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm689_gca_000976335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm689_gca_000976335/ncrna/Saccharomyces_cerevisiae_yjm689_gca_000976335.Sc_YJM689_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm693_gca_000976365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm693_gca_000976365/ncrna/Saccharomyces_cerevisiae_yjm693_gca_000976365.Sc_YJM693_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm789_gca_000181435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_yjm789_gca_000181435/ncrna/Saccharomyces_cerevisiae_yjm789_gca_000181435.ASM18143v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm969_gca_000976395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm969_gca_000976395/ncrna/Saccharomyces_cerevisiae_yjm969_gca_000976395.Sc_YJM969_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm972_gca_000976425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm972_gca_000976425/ncrna/Saccharomyces_cerevisiae_yjm972_gca_000976425.Sc_YJM972_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm975_gca_000976455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm975_gca_000976455/ncrna/Saccharomyces_cerevisiae_yjm975_gca_000976455.Sc_YJM975_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm978_gca_000976485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm978_gca_000976485/ncrna/Saccharomyces_cerevisiae_yjm978_gca_000976485.Sc_YJM978_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm981_gca_000976515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm981_gca_000976515/ncrna/Saccharomyces_cerevisiae_yjm981_gca_000976515.Sc_YJM981_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm984_gca_000976545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm984_gca_000976545/ncrna/Saccharomyces_cerevisiae_yjm984_gca_000976545.Sc_YJM984_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm987_gca_000976575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm987_gca_000976575/ncrna/Saccharomyces_cerevisiae_yjm987_gca_000976575.Sc_YJM987_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm990_gca_000976605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm990_gca_000976605/ncrna/Saccharomyces_cerevisiae_yjm990_gca_000976605.Sc_YJM990_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm993_gca_000662435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_yjm993_gca_000662435/ncrna/Saccharomyces_cerevisiae_yjm993_gca_000662435.Sc_YJM993_v1.ncrna.fa.gz',
	'saccharomyces_cerevisiae_yjm996_gca_000976665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm996_gca_000976665/ncrna/Saccharomyces_cerevisiae_yjm996_gca_000976665.Sc_YJM996_v1.ncrna.fa.gz',
	'saccharomyces_eubayanus_gca_001298625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_eubayanus_gca_001298625/ncrna/Saccharomyces_eubayanus_gca_001298625.SEUB3.0.ncrna.fa.gz',
	'saccharomyces_kudriavzevii_ifo_1802_gca_000167075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_kudriavzevii_ifo_1802_gca_000167075/ncrna/Saccharomyces_kudriavzevii_ifo_1802_gca_000167075.Saccharomyces_kudriavzevii_strain_IFO1802_v1.0.ncrna.fa.gz',
	'saccharomycetaceae_sp_ashbya_aceri__gca_000412225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomycetaceae_sp_ashbya_aceri__gca_000412225/ncrna/Saccharomycetaceae_sp_ashbya_aceri__gca_000412225.ASM41222v2.ncrna.fa.gz',
	'saimiri_boliviensis_boliviensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/saimiri_boliviensis_boliviensis/ncrna/Saimiri_boliviensis_boliviensis.SaiBol1.0.ncrna.fa.gz',
	'saitoella_complicata_nrrl_y_17804_gca_000227095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saitoella_complicata_nrrl_y_17804_gca_000227095/ncrna/Saitoella_complicata_nrrl_y_17804_gca_000227095.Scomplicata_3.0.ncrna.fa.gz',
	'saitoella_complicata_nrrl_y_17804_gca_001661265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saitoella_complicata_nrrl_y_17804_gca_001661265/ncrna/Saitoella_complicata_nrrl_y_17804_gca_001661265.Saico1.ncrna.fa.gz',
	'salpingoeca_rosetta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_choanoflagellida1_collection/salpingoeca_rosetta/ncrna/Salpingoeca_rosetta.Proterospongia_sp_ATCC50818.ncrna.fa.gz',
	'salvator_merianae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/salvator_merianae/ncrna/Salvator_merianae.HLtupMer3.ncrna.fa.gz',
	'saprolegnia_diclina_vs20'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/saprolegnia_diclina_vs20/ncrna/Saprolegnia_diclina_vs20.Sap_diclina_VS20_V1.ncrna.fa.gz',
	'saprolegnia_parasitica_cbs_223_65'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/saprolegnia_parasitica_cbs_223_65/ncrna/Saprolegnia_parasitica_cbs_223_65.ASM15154v2.ncrna.fa.gz',
	'sarcophilus_harrisii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sarcophilus_harrisii/ncrna/Sarcophilus_harrisii.DEVIL7.0.ncrna.fa.gz',
	'sarcoptes_scabiei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/sarcoptes_scabiei/ncrna/Sarcoptes_scabiei.SscaA1.ncrna.fa.gz',
	'scedosporium_apiospermum_gca_000732125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/scedosporium_apiospermum_gca_000732125/ncrna/Scedosporium_apiospermum_gca_000732125.ScApio1.0.ncrna.fa.gz',
	'scheffersomyces_stipitis_cbs_6054_gca_000209165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/scheffersomyces_stipitis_cbs_6054_gca_000209165/ncrna/Scheffersomyces_stipitis_cbs_6054_gca_000209165.ASM20916v1.ncrna.fa.gz',
	'schistosoma_mansoni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/schistosoma_mansoni/ncrna/Schistosoma_mansoni.ASM23792v2.ncrna.fa.gz',
	'schizophyllum_commune_h4_8_gca_000143185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/schizophyllum_commune_h4_8_gca_000143185/ncrna/Schizophyllum_commune_h4_8_gca_000143185.v1.0.ncrna.fa.gz',
	'schizopora_paradoxa_gca_001020605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/schizopora_paradoxa_gca_001020605/ncrna/Schizopora_paradoxa_gca_001020605.Schpa1.ncrna.fa.gz',
	'schizosaccharomyces_cryophilus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_cryophilus/ncrna/Schizosaccharomyces_cryophilus.GCA_000004155.2.ncrna.fa.gz',
	'schizosaccharomyces_japonicus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_japonicus/ncrna/Schizosaccharomyces_japonicus.GCA_000149845.2.ncrna.fa.gz',
	'schizosaccharomyces_octosporus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_octosporus/ncrna/Schizosaccharomyces_octosporus.GCA_000150505.2.ncrna.fa.gz',
	'schizosaccharomyces_pombe'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_pombe/ncrna/Schizosaccharomyces_pombe.ASM294v2.ncrna.fa.gz',
	'scleroderma_citrinum_foug_a_gca_000827425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/scleroderma_citrinum_foug_a_gca_000827425/ncrna/Scleroderma_citrinum_foug_a_gca_000827425.Scleroderma_citrinum_Foug_A_v1.0.ncrna.fa.gz',
	'scleropages_formosus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/scleropages_formosus/ncrna/Scleropages_formosus.ASM162426v1.ncrna.fa.gz',
	'sclerotinia_borealis_f_4128_gca_000503235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sclerotinia_borealis_f_4128_gca_000503235/ncrna/Sclerotinia_borealis_f_4128_gca_000503235.SBOR_1.ncrna.fa.gz',
	'sclerotinia_sclerotiorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/sclerotinia_sclerotiorum/ncrna/Sclerotinia_sclerotiorum.ASM14694v1.ncrna.fa.gz',
	'sclerotinia_sclerotiorum_1980_uf_70_gca_001857865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/sclerotinia_sclerotiorum_1980_uf_70_gca_001857865/ncrna/Sclerotinia_sclerotiorum_1980_uf_70_gca_001857865.ASM185786v1.ncrna.fa.gz',
	'scophthalmus_maximus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/scophthalmus_maximus/ncrna/Scophthalmus_maximus.ASM318616v1.ncrna.fa.gz',
	'selaginella_moellendorffii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/selaginella_moellendorffii/ncrna/Selaginella_moellendorffii.v1.0.ncrna.fa.gz',
	'serendipita_indica_dsm_11827_gca_000313545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/serendipita_indica_dsm_11827_gca_000313545/ncrna/Serendipita_indica_dsm_11827_gca_000313545.ASM31354v1.ncrna.fa.gz',
	'serendipita_vermifera_maff_305830_gca_000827415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/serendipita_vermifera_maff_305830_gca_000827415/ncrna/Serendipita_vermifera_maff_305830_gca_000827415.Sebacina_vermifera_MAFF_305830_v1.0.ncrna.fa.gz',
	'serinus_canaria'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/serinus_canaria/ncrna/Serinus_canaria.SCA1.ncrna.fa.gz',
	'seriola_dumerili'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/seriola_dumerili/ncrna/Seriola_dumerili.Sdu_1.0.ncrna.fa.gz',
	'seriola_lalandi_dorsalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/seriola_lalandi_dorsalis/ncrna/Seriola_lalandi_dorsalis.Sedor1.ncrna.fa.gz',
	'serpula_lacrymans_var_lacrymans_s7_3_gca_000218725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/serpula_lacrymans_var_lacrymans_s7_3_gca_000218725/ncrna/Serpula_lacrymans_var_lacrymans_s7_3_gca_000218725.v1.0.ncrna.fa.gz',
	'setaria_italica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/setaria_italica/ncrna/Setaria_italica.Setaria_italica_v2.0.ncrna.fa.gz',
	'sistotremastrum_niveocremeum_hhb9708_gca_001630475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sistotremastrum_niveocremeum_hhb9708_gca_001630475/ncrna/Sistotremastrum_niveocremeum_hhb9708_gca_001630475.Sisni1.ncrna.fa.gz',
	'sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355/ncrna/Sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355.Sissu1.ncrna.fa.gz',
	'solanum_lycopersicum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/solanum_lycopersicum/ncrna/Solanum_lycopersicum.SL3.0.ncrna.fa.gz',
	'solanum_tuberosum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/solanum_tuberosum/ncrna/Solanum_tuberosum.SolTub_3.0.ncrna.fa.gz',
	'solenopsis_invicta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/solenopsis_invicta/ncrna/Solenopsis_invicta.Si_gnG.ncrna.fa.gz',
	'sordaria_macrospora_k_hell_gca_000182805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sordaria_macrospora_k_hell_gca_000182805/ncrna/Sordaria_macrospora_k_hell_gca_000182805.ASM18280v2.ncrna.fa.gz',
	'sorex_araneus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sorex_araneus/ncrna/Sorex_araneus.COMMON_SHREW1.ncrna.fa.gz',
	'sorghum_bicolor'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/sorghum_bicolor/ncrna/Sorghum_bicolor.Sorghum_bicolor_NCBIv3.ncrna.fa.gz',
	'spathaspora_passalidarum_nrrl_y_27907_gca_000223485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/spathaspora_passalidarum_nrrl_y_27907_gca_000223485/ncrna/Spathaspora_passalidarum_nrrl_y_27907_gca_000223485.Spathaspora_passalidarum_v2.0.ncrna.fa.gz',
	'spermophilus_dauricus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/spermophilus_dauricus/ncrna/Spermophilus_dauricus.ASM240643v1.ncrna.fa.gz',
	'sphaerobolus_stellatus_ss14_gca_000827215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sphaerobolus_stellatus_ss14_gca_000827215/ncrna/Sphaerobolus_stellatus_ss14_gca_000827215.Sphst1.ncrna.fa.gz',
	'sphaeroforma_arctica_jp610'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_ichthyosporea1_collection/sphaeroforma_arctica_jp610/ncrna/Sphaeroforma_arctica_jp610.Spha_arctica_JP610_V1.ncrna.fa.gz',
	'sphaerulina_musiva_so2202_gca_000320565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sphaerulina_musiva_so2202_gca_000320565/ncrna/Sphaerulina_musiva_so2202_gca_000320565.Septoria_musiva_SO2202_v1.0.ncrna.fa.gz',
	'sphenodon_punctatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sphenodon_punctatus/ncrna/Sphenodon_punctatus.ASM311381v1.ncrna.fa.gz',
	'spironucleus_salmonicida'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/spironucleus_salmonicida/ncrna/Spironucleus_salmonicida.SSK3.0.ncrna.fa.gz',
	'spizellomyces_punctatus_daom_br117_gca_000182565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/spizellomyces_punctatus_daom_br117_gca_000182565/ncrna/Spizellomyces_punctatus_daom_br117_gca_000182565.S_punctatus_V1.ncrna.fa.gz',
	'sporisorium_reilianum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/sporisorium_reilianum/ncrna/Sporisorium_reilianum.GCA_000230245.1.ncrna.fa.gz',
	'sporisorium_reilianum_f_sp_reilianum_gca_900162835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sporisorium_reilianum_f_sp_reilianum_gca_900162835/ncrna/Sporisorium_reilianum_f_sp_reilianum_gca_900162835.version_2.ncrna.fa.gz',
	'sporothrix_brasiliensis_5110_gca_000820605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/sporothrix_brasiliensis_5110_gca_000820605/ncrna/Sporothrix_brasiliensis_5110_gca_000820605.S_brasiliensis_5110_v1.ncrna.fa.gz',
	'sporothrix_insectorum_rcef_264_gca_001636815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/sporothrix_insectorum_rcef_264_gca_001636815/ncrna/Sporothrix_insectorum_rcef_264_gca_001636815.SPI_1.0.ncrna.fa.gz',
	'sporothrix_schenckii_1099_18_gca_000961545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/sporothrix_schenckii_1099_18_gca_000961545/ncrna/Sporothrix_schenckii_1099_18_gca_000961545.S_schenckii_v1.ncrna.fa.gz',
	'sporothrix_schenckii_atcc_58251_gca_000474925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sporothrix_schenckii_atcc_58251_gca_000474925/ncrna/Sporothrix_schenckii_atcc_58251_gca_000474925.Spor_sche_ATCC58251_V1.ncrna.fa.gz',
	'spraguea_lophii_42_110_gca_000430065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/spraguea_lophii_42_110_gca_000430065/ncrna/Spraguea_lophii_42_110_gca_000430065.Sprlop1.0.ncrna.fa.gz',
	'stachybotrys_chartarum_ibt_40288_gca_000732765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/stachybotrys_chartarum_ibt_40288_gca_000732765/ncrna/Stachybotrys_chartarum_ibt_40288_gca_000732765.S40288v1.ncrna.fa.gz',
	'stachybotrys_chlorohalonata_ibt_40285_gca_000732775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/stachybotrys_chlorohalonata_ibt_40285_gca_000732775/ncrna/Stachybotrys_chlorohalonata_ibt_40285_gca_000732775.S40285v1.ncrna.fa.gz',
	'stagonospora_sp_src1lsm3a_gca_001644525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/stagonospora_sp_src1lsm3a_gca_001644525/ncrna/Stagonospora_sp_src1lsm3a_gca_001644525.Stasp1.ncrna.fa.gz',
	'stegastes_partitus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/stegastes_partitus/ncrna/Stegastes_partitus.Stegastes_partitus-1.0.2.ncrna.fa.gz',
	'stegodyphus_mimosarum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/stegodyphus_mimosarum/ncrna/Stegodyphus_mimosarum.Stegodyphus_mimosarum_v1.ncrna.fa.gz',
	'stemphylium_lycopersici_gca_001191545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/stemphylium_lycopersici_gca_001191545/ncrna/Stemphylium_lycopersici_gca_001191545.ASM119154v1.ncrna.fa.gz',
	'stereum_hirsutum_fp_91666_ss1_gca_000264905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/stereum_hirsutum_fp_91666_ss1_gca_000264905/ncrna/Stereum_hirsutum_fp_91666_ss1_gca_000264905.Stehi1.ncrna.fa.gz',
	'strigamia_maritima'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/strigamia_maritima/ncrna/Strigamia_maritima.Smar1.ncrna.fa.gz',
	'strigomonas_culicis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/strigomonas_culicis/ncrna/Strigomonas_culicis.Strigomonas_culicis_Genome.ncrna.fa.gz',
	'strongylocentrotus_purpuratus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/strongylocentrotus_purpuratus/ncrna/Strongylocentrotus_purpuratus.Spur_3.1.ncrna.fa.gz',
	'strongyloides_ratti'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/strongyloides_ratti/ncrna/Strongyloides_ratti.S_ratti_ED321_v5_0_4.ncrna.fa.gz',
	'sugiyamaella_lignohabitans_gca_001640025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/sugiyamaella_lignohabitans_gca_001640025/ncrna/Sugiyamaella_lignohabitans_gca_001640025.ASM164002v2.ncrna.fa.gz',
	'suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415/ncrna/Suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415.Canta1.ncrna.fa.gz',
	'suillus_luteus_uh_slu_lm8_n1_gca_000827255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/suillus_luteus_uh_slu_lm8_n1_gca_000827255/ncrna/Suillus_luteus_uh_slu_lm8_n1_gca_000827255.Suilu1.ncrna.fa.gz',
	'sus_scrofa'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sus_scrofa/ncrna/Sus_scrofa.Sscrofa11.1.ncrna.fa.gz',
	'syncephalastrum_racemosum_gca_002105135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/syncephalastrum_racemosum_gca_002105135/ncrna/Syncephalastrum_racemosum_gca_002105135.Synrac1.ncrna.fa.gz',
	'taeniopygia_guttata'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/taeniopygia_guttata/ncrna/Taeniopygia_guttata.taeGut3.2.4.ncrna.fa.gz',
	'takifugu_rubripes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/takifugu_rubripes/ncrna/Takifugu_rubripes.FUGU5.ncrna.fa.gz',
	'talaromyces_cellulolyticus_gca_000829775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/talaromyces_cellulolyticus_gca_000829775/ncrna/Talaromyces_cellulolyticus_gca_000829775.TcelY94_1.0.ncrna.fa.gz',
	'talaromyces_islandicus_gca_000985935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/talaromyces_islandicus_gca_000985935/ncrna/Talaromyces_islandicus_gca_000985935.PIS.ncrna.fa.gz',
	'talaromyces_marneffei_atcc_18224_gca_000001985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/talaromyces_marneffei_atcc_18224_gca_000001985/ncrna/Talaromyces_marneffei_atcc_18224_gca_000001985.JCVI-PMFA1-2.0.ncrna.fa.gz',
	'talaromyces_marneffei_pm1_gca_000750115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/talaromyces_marneffei_pm1_gca_000750115/ncrna/Talaromyces_marneffei_pm1_gca_000750115.ASM75011v1.ncrna.fa.gz',
	'talaromyces_stipitatus_atcc_10500_gca_000003125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/talaromyces_stipitatus_atcc_10500_gca_000003125/ncrna/Talaromyces_stipitatus_atcc_10500_gca_000003125.JCVI-TSTA1-3.0.ncrna.fa.gz',
	'teleopsis_dalmanni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/teleopsis_dalmanni/ncrna/Teleopsis_dalmanni.Tel_dalmanni_2A_v1.0.ncrna.fa.gz',
	'termitomyces_sp_j132_gca_001263195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/termitomyces_sp_j132_gca_001263195/ncrna/Termitomyces_sp_j132_gca_001263195.TerJ1321.0.ncrna.fa.gz',
	'testicularia_cyperi_gca_003144125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/testicularia_cyperi_gca_003144125/ncrna/Testicularia_cyperi_gca_003144125.Tescy1.ncrna.fa.gz',
	'tetrahymena_thermophila'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/tetrahymena_thermophila/ncrna/Tetrahymena_thermophila.JCVI-TTA1-2.2.ncrna.fa.gz',
	'tetranychus_urticae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/tetranychus_urticae/ncrna/Tetranychus_urticae.ASM23943v1.ncrna.fa.gz',
	'tetraodon_nigroviridis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/tetraodon_nigroviridis/ncrna/Tetraodon_nigroviridis.TETRAODON8.ncrna.fa.gz',
	'tetrapisispora_blattae_cbs_6284_gca_000315915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/tetrapisispora_blattae_cbs_6284_gca_000315915/ncrna/Tetrapisispora_blattae_cbs_6284_gca_000315915.ASM31591v1.ncrna.fa.gz',
	'tetrapisispora_phaffii_cbs_4417_gca_000236905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/tetrapisispora_phaffii_cbs_4417_gca_000236905/ncrna/Tetrapisispora_phaffii_cbs_4417_gca_000236905.ASM23690v1.ncrna.fa.gz',
	'thalassiosira_oceanica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/thalassiosira_oceanica/ncrna/Thalassiosira_oceanica.ThaOc_1.0.ncrna.fa.gz',
	'thalassiosira_oceanica_ccmp1005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/thalassiosira_oceanica_ccmp1005/ncrna/Thalassiosira_oceanica_ccmp1005.ASM19663v1.ncrna.fa.gz',
	'thalassiosira_pseudonana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/thalassiosira_pseudonana/ncrna/Thalassiosira_pseudonana.ASM14940v2.ncrna.fa.gz',
	'thecamonas_trahens_atcc_50062'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_apusozoa1_collection/thecamonas_trahens_atcc_50062/ncrna/Thecamonas_trahens_atcc_50062.TheTra_May2010.ncrna.fa.gz',
	'theileria_annulata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_annulata/ncrna/Theileria_annulata.ASM322v1.ncrna.fa.gz',
	'theileria_equi_strain_wa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_equi_strain_wa/ncrna/Theileria_equi_strain_wa.JCVI-bewag-v1.1.ncrna.fa.gz',
	'theileria_orientalis_strain_shintoku'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_orientalis_strain_shintoku/ncrna/Theileria_orientalis_strain_shintoku.ASM74089v1.ncrna.fa.gz',
	'theileria_parva'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_parva/ncrna/Theileria_parva.ASM16536v1.ncrna.fa.gz',
	'thelohanellus_kitauei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/thelohanellus_kitauei/ncrna/Thelohanellus_kitauei.ASM82789v1.ncrna.fa.gz',
	'theobroma_cacao'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/theobroma_cacao/ncrna/Theobroma_cacao.Theobroma_cacao_20110822.ncrna.fa.gz',
	'thermothelomyces_thermophila_atcc_42464_gca_000226095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/thermothelomyces_thermophila_atcc_42464_gca_000226095/ncrna/Thermothelomyces_thermophila_atcc_42464_gca_000226095.ASM22609v1.ncrna.fa.gz',
	'theropithecus_gelada'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/theropithecus_gelada/ncrna/Theropithecus_gelada.Tgel_1.0.ncrna.fa.gz',
	'thielavia_terrestris_nrrl_8126_gca_000226115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/thielavia_terrestris_nrrl_8126_gca_000226115/ncrna/Thielavia_terrestris_nrrl_8126_gca_000226115.ASM22611v1.ncrna.fa.gz',
	'thielaviopsis_punctulata_gca_000968615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/thielaviopsis_punctulata_gca_000968615/ncrna/Thielaviopsis_punctulata_gca_000968615.ASM96861v1.ncrna.fa.gz',
	'tilletia_controversa_gca_001645045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletia_controversa_gca_001645045/ncrna/Tilletia_controversa_gca_001645045.ASM164504v1.ncrna.fa.gz',
	'tilletia_walkeri_gca_001645055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletia_walkeri_gca_001645055/ncrna/Tilletia_walkeri_gca_001645055.ASM164505v1.ncrna.fa.gz',
	'tilletiaria_anomala_ubc_951_gca_000711695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletiaria_anomala_ubc_951_gca_000711695/ncrna/Tilletiaria_anomala_ubc_951_gca_000711695.Tilletiaria_anomala_UBC_951_v1.0.ncrna.fa.gz',
	'tilletiopsis_washingtonensis_gca_003144115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletiopsis_washingtonensis_gca_003144115/ncrna/Tilletiopsis_washingtonensis_gca_003144115.Tilwa1.ncrna.fa.gz',
	'tolypocladium_ophioglossoides_cbs_100239_gca_001189435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/tolypocladium_ophioglossoides_cbs_100239_gca_001189435/ncrna/Tolypocladium_ophioglossoides_cbs_100239_gca_001189435.Tophv1.0.ncrna.fa.gz',
	'torrubiella_hemipterigena_gca_000825705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/torrubiella_hemipterigena_gca_000825705/ncrna/Torrubiella_hemipterigena_gca_000825705.Allpaths-LG.ncrna.fa.gz',
	'tortispora_caseinolytica_nrrl_y_17796_gca_001661475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/tortispora_caseinolytica_nrrl_y_17796_gca_001661475/ncrna/Tortispora_caseinolytica_nrrl_y_17796_gca_001661475.Canca1.ncrna.fa.gz',
	'torulaspora_delbrueckii_gca_000243375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/torulaspora_delbrueckii_gca_000243375/ncrna/Torulaspora_delbrueckii_gca_000243375.ASM24337v1.ncrna.fa.gz',
	'toxoplasma_gondii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/toxoplasma_gondii/ncrna/Toxoplasma_gondii.TGA4.ncrna.fa.gz',
	'toxoplasma_gondii_ari'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_ari/ncrna/Toxoplasma_gondii_ari.TGARI_v2.ncrna.fa.gz',
	'toxoplasma_gondii_fou'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_fou/ncrna/Toxoplasma_gondii_fou.TGFOU1v02.ncrna.fa.gz',
	'toxoplasma_gondii_gab2_2007_gal_dom2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_gab2_2007_gal_dom2/ncrna/Toxoplasma_gondii_gab2_2007_gal_dom2.TGGABGALDOM2_v2.1.ncrna.fa.gz',
	'toxoplasma_gondii_gt1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_gt1/ncrna/Toxoplasma_gondii_gt1.TGGT1.ncrna.fa.gz',
	'toxoplasma_gondii_mas'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_mas/ncrna/Toxoplasma_gondii_mas.TGMAS1_v2.ncrna.fa.gz',
	'toxoplasma_gondii_p89'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_p89/ncrna/Toxoplasma_gondii_p89.TGP89A_v02.ncrna.fa.gz',
	'toxoplasma_gondii_rub'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_rub/ncrna/Toxoplasma_gondii_rub.TGRUB_v2.ncrna.fa.gz',
	'toxoplasma_gondii_tgcatprc2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_tgcatprc2/ncrna/Toxoplasma_gondii_tgcatprc2.TGCATPRC2_v2.ncrna.fa.gz',
	'toxoplasma_gondii_vand'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_vand/ncrna/Toxoplasma_gondii_vand.TGVAND_v2.ncrna.fa.gz',
	'toxoplasma_gondii_veg'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_veg/ncrna/Toxoplasma_gondii_veg.TGVEG.ncrna.fa.gz',
	'trachipleistophora_hominis_gca_000316135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/trachipleistophora_hominis_gca_000316135/ncrna/Trachipleistophora_hominis_gca_000316135.ASM31613v1.ncrna.fa.gz',
	'trametes_cinnabarina_gca_000765035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_cinnabarina_gca_000765035/ncrna/Trametes_cinnabarina_gca_000765035.BN946.ncrna.fa.gz',
	'trametes_coccinea_brfm310_gca_002092935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_coccinea_brfm310_gca_002092935/ncrna/Trametes_coccinea_brfm310_gca_002092935.Pycco1.ncrna.fa.gz',
	'trametes_pubescens_gca_001895945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_pubescens_gca_001895945/ncrna/Trametes_pubescens_gca_001895945.ASM189594v1.ncrna.fa.gz',
	'trametes_versicolor_fp_101664_ss1_gca_000271585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_versicolor_fp_101664_ss1_gca_000271585/ncrna/Trametes_versicolor_fp_101664_ss1_gca_000271585.Trametes_versicolor_v1.0.ncrna.fa.gz',
	'tremella_mesenterica_dsm_1558_gca_000271645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tremella_mesenterica_dsm_1558_gca_000271645/ncrna/Tremella_mesenterica_dsm_1558_gca_000271645.Treme1.ncrna.fa.gz',
	'tribolium_castaneum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/tribolium_castaneum/ncrna/Tribolium_castaneum.Tcas5.2.ncrna.fa.gz',
	'trichoderma_asperellum_cbs_433_97_gca_003025105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_asperellum_cbs_433_97_gca_003025105/ncrna/Trichoderma_asperellum_cbs_433_97_gca_003025105.Trias_v._1.0.ncrna.fa.gz',
	'trichoderma_atroviride_imi_206040_gca_000171015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichoderma_atroviride_imi_206040_gca_000171015/ncrna/Trichoderma_atroviride_imi_206040_gca_000171015.TRIAT_v2.0.ncrna.fa.gz',
	'trichoderma_citrinoviride_gca_003025115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_citrinoviride_gca_003025115/ncrna/Trichoderma_citrinoviride_gca_003025115.Trici_v4.0.ncrna.fa.gz',
	'trichoderma_harzianum_cbs_226_95_gca_003025095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_harzianum_cbs_226_95_gca_003025095/ncrna/Trichoderma_harzianum_cbs_226_95_gca_003025095.Allpaths-LG_v._01_03_12.ncrna.fa.gz',
	'trichoderma_harzianum_gca_000988865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/trichoderma_harzianum_gca_000988865/ncrna/Trichoderma_harzianum_gca_000988865.ASM98886v1.ncrna.fa.gz',
	'trichoderma_longibrachiatum_atcc_18648_gca_003025155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_longibrachiatum_atcc_18648_gca_003025155/ncrna/Trichoderma_longibrachiatum_atcc_18648_gca_003025155.Trilo3.ncrna.fa.gz',
	'trichoderma_reesei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/trichoderma_reesei/ncrna/Trichoderma_reesei.GCA_000167675.2.ncrna.fa.gz',
	'trichoderma_reesei_rut_c_30_gca_000513815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichoderma_reesei_rut_c_30_gca_000513815/ncrna/Trichoderma_reesei_rut_c_30_gca_000513815.TrireRUTC30_1.ncrna.fa.gz',
	'trichoderma_virens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/trichoderma_virens/ncrna/Trichoderma_virens.ASM17099v1.ncrna.fa.gz',
	'trichomonas_vaginalis_g3'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_parabasalia1_collection/trichomonas_vaginalis_g3/ncrna/Trichomonas_vaginalis_g3.ASM282v1.ncrna.fa.gz',
	'trichophyton_benhamiae_cbs_112371_gca_000151125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_benhamiae_cbs_112371_gca_000151125/ncrna/Trichophyton_benhamiae_cbs_112371_gca_000151125.ASM15112v2.ncrna.fa.gz',
	'trichophyton_equinum_cbs_127_97_gca_000151175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_equinum_cbs_127_97_gca_000151175/ncrna/Trichophyton_equinum_cbs_127_97_gca_000151175.ASM15117v1.ncrna.fa.gz',
	'trichophyton_interdigitale_h6_gca_000616785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_interdigitale_h6_gca_000616785/ncrna/Trichophyton_interdigitale_h6_gca_000616785.Tric_rubr_H6_V2.ncrna.fa.gz',
	'trichophyton_interdigitale_mr816_gca_000622975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_interdigitale_mr816_gca_000622975/ncrna/Trichophyton_interdigitale_mr816_gca_000622975.Tric_inte_MR816_V1.ncrna.fa.gz',
	'trichophyton_mentagrophytes_gca_003118255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichophyton_mentagrophytes_gca_003118255/ncrna/Trichophyton_mentagrophytes_gca_003118255.ABySS_70bp_45k_152cov_v1.ncrna.fa.gz',
	'trichophyton_rubrum_cbs_100081_gca_000616805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_100081_gca_000616805/ncrna/Trichophyton_rubrum_cbs_100081_gca_000616805.Tric_fish_CBS_100081_V1.ncrna.fa.gz',
	'trichophyton_rubrum_cbs_118892_gca_000151425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_118892_gca_000151425/ncrna/Trichophyton_rubrum_cbs_118892_gca_000151425.ASM15142v1.ncrna.fa.gz',
	'trichophyton_rubrum_cbs_202_88_gca_000616985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_202_88_gca_000616985/ncrna/Trichophyton_rubrum_cbs_202_88_gca_000616985.Tric_raub_CBS_202_88_V2.ncrna.fa.gz',
	'trichophyton_rubrum_cbs_288_86_gca_000616825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_288_86_gca_000616825/ncrna/Trichophyton_rubrum_cbs_288_86_gca_000616825.Tric_fish_CBS_288_86_V1.ncrna.fa.gz',
	'trichophyton_rubrum_cbs_289_86_gca_000616845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_289_86_gca_000616845/ncrna/Trichophyton_rubrum_cbs_289_86_gca_000616845.Tric_kane_CBS_289_86_V1.ncrna.fa.gz',
	'trichophyton_rubrum_cbs_735_88_gca_000616965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_735_88_gca_000616965/ncrna/Trichophyton_rubrum_cbs_735_88_gca_000616965.Tric_megn_CBS_735_88_V1.ncrna.fa.gz',
	'trichophyton_rubrum_d6_gca_000622995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_d6_gca_000622995/ncrna/Trichophyton_rubrum_d6_gca_000622995.Tric_rubr_D6_V1.ncrna.fa.gz',
	'trichophyton_rubrum_mr1448_gca_000616905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_mr1448_gca_000616905/ncrna/Trichophyton_rubrum_mr1448_gca_000616905.Tric_rubr_M1448_V1.ncrna.fa.gz',
	'trichophyton_rubrum_mr1459_gca_000616945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_mr1459_gca_000616945/ncrna/Trichophyton_rubrum_mr1459_gca_000616945.Tric_rubr_MR1459_V1.ncrna.fa.gz',
	'trichophyton_rubrum_mr850_gca_000616765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_mr850_gca_000616765/ncrna/Trichophyton_rubrum_mr850_gca_000616765.Tric_rubr_MR-850_V1.ncrna.fa.gz',
	'trichophyton_soudanense_cbs_452_61_gca_000616865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_soudanense_cbs_452_61_gca_000616865/ncrna/Trichophyton_soudanense_cbs_452_61_gca_000616865.Tric_soud_CBS_452_61_V1.ncrna.fa.gz',
	'trichophyton_tonsurans_cbs_112818_gca_000151455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_tonsurans_cbs_112818_gca_000151455/ncrna/Trichophyton_tonsurans_cbs_112818_gca_000151455.ASM15145v1.ncrna.fa.gz',
	'trichophyton_verrucosum_hki_0517_gca_000151505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_verrucosum_hki_0517_gca_000151505/ncrna/Trichophyton_verrucosum_hki_0517_gca_000151505.ASM15150v1.ncrna.fa.gz',
	'trichoplax_adhaerens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/trichoplax_adhaerens/ncrna/Trichoplax_adhaerens.ASM15027v1.ncrna.fa.gz',
	'trichosporon_asahii_var_asahii_cbs_2479_gca_000293215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trichosporon_asahii_var_asahii_cbs_2479_gca_000293215/ncrna/Trichosporon_asahii_var_asahii_cbs_2479_gca_000293215.Trichosporon_asahii_1.ncrna.fa.gz',
	'trichosporon_asahii_var_asahii_cbs_8904_gca_000299215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trichosporon_asahii_var_asahii_cbs_8904_gca_000299215/ncrna/Trichosporon_asahii_var_asahii_cbs_8904_gca_000299215.ASM29921v2.ncrna.fa.gz',
	'trifolium_pratense'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/trifolium_pratense/ncrna/Trifolium_pratense.Trpr.ncrna.fa.gz',
	'triticum_aestivum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/triticum_aestivum/ncrna/Triticum_aestivum.IWGSC.ncrna.fa.gz',
	'triticum_dicoccoides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/triticum_dicoccoides/ncrna/Triticum_dicoccoides.WEWSeq_v.1.0.ncrna.fa.gz',
	'triticum_urartu'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/triticum_urartu/ncrna/Triticum_urartu.ASM34745v1.ncrna.fa.gz',
	'trypanosoma_brucei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/trypanosoma_brucei/ncrna/Trypanosoma_brucei.TryBru_Apr2005_chr11.ncrna.fa.gz',
	'trypanosoma_cruzi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/trypanosoma_cruzi/ncrna/Trypanosoma_cruzi.ASM20906v1.ncrna.fa.gz',
	'trypanosoma_cruzi_dm28c'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/trypanosoma_cruzi_dm28c/ncrna/Trypanosoma_cruzi_dm28c.T.cruzi.Dm28c_v01.ncrna.fa.gz',
	'tsuchiyaea_wingfieldii_cbs_7118_gca_001720155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tsuchiyaea_wingfieldii_cbs_7118_gca_001720155/ncrna/Tsuchiyaea_wingfieldii_cbs_7118_gca_001720155.Tsuc_wing_CBS7118_V1.ncrna.fa.gz',
	'tuber_borchii_gca_003070745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/tuber_borchii_gca_003070745/ncrna/Tuber_borchii_gca_003070745.Tubbor1.ncrna.fa.gz',
	'tuber_magnatum_gca_003182015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/tuber_magnatum_gca_003182015/ncrna/Tuber_magnatum_gca_003182015.ASM318201v1.ncrna.fa.gz',
	'tuber_melanosporum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/tuber_melanosporum/ncrna/Tuber_melanosporum.ASM15164v1.ncrna.fa.gz',
	'tulasnella_calospora_mut_4182_gca_000827465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tulasnella_calospora_mut_4182_gca_000827465/ncrna/Tulasnella_calospora_mut_4182_gca_000827465.Tulasnella_calospora_AL13_4D_v1.0.ncrna.fa.gz',
	'tupaia_belangeri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/tupaia_belangeri/ncrna/Tupaia_belangeri.TREESHREW.ncrna.fa.gz',
	'tursiops_truncatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/tursiops_truncatus/ncrna/Tursiops_truncatus.turTru1.ncrna.fa.gz',
	'uncinocarpus_reesii_1704_gca_000003515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/uncinocarpus_reesii_1704_gca_000003515/ncrna/Uncinocarpus_reesii_1704_gca_000003515.ASM351v2.ncrna.fa.gz',
	'urocitellus_parryii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/urocitellus_parryii/ncrna/Urocitellus_parryii.ASM342692v1.ncrna.fa.gz',
	'ursus_americanus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ursus_americanus/ncrna/Ursus_americanus.ASM334442v1.ncrna.fa.gz',
	'ursus_maritimus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ursus_maritimus/ncrna/Ursus_maritimus.UrsMar_1.0.ncrna.fa.gz',
	'ustilaginoidea_virens_gca_000687475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ustilaginoidea_virens_gca_000687475/ncrna/Ustilaginoidea_virens_gca_000687475.Assembly_for_version_1_of_the_Villosiclava_virens_genome.ncrna.fa.gz',
	'ustilaginomycotina_sp_sa_807_gca_003144235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ustilaginomycotina_sp_sa_807_gca_003144235/ncrna/Ustilaginomycotina_sp_sa_807_gca_003144235.Ustsp1.ncrna.fa.gz',
	'ustilago_bromivora_gca_900080155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ustilago_bromivora_gca_900080155/ncrna/Ustilago_bromivora_gca_900080155.UBRO_v3.ncrna.fa.gz',
	'ustilago_hordei_gca_000286035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ustilago_hordei_gca_000286035/ncrna/Ustilago_hordei_gca_000286035.ASM28603v1.ncrna.fa.gz',
	'ustilago_maydis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/ustilago_maydis/ncrna/Ustilago_maydis.Umaydis521_2.0.ncrna.fa.gz',
	'valsa_mali_gca_000818155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/valsa_mali_gca_000818155/ncrna/Valsa_mali_gca_000818155.ASM81815v1.ncrna.fa.gz',
	'valsa_mali_var_pyri_gca_000813385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/valsa_mali_var_pyri_gca_000813385/ncrna/Valsa_mali_var_pyri_gca_000813385.ASM81338v1.ncrna.fa.gz',
	'vanderwaltozyma_polyspora_dsm_70294_gca_000150035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/vanderwaltozyma_polyspora_dsm_70294_gca_000150035/ncrna/Vanderwaltozyma_polyspora_dsm_70294_gca_000150035.ASM15003v1.ncrna.fa.gz',
	'vavraia_culicis_subsp_floridensis_gca_000192795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/vavraia_culicis_subsp_floridensis_gca_000192795/ncrna/Vavraia_culicis_subsp_floridensis_gca_000192795.Vavr_culi_floridensis_V1.ncrna.fa.gz',
	'verruconis_gallopava_gca_000836295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/verruconis_gallopava_gca_000836295/ncrna/Verruconis_gallopava_gca_000836295.O_gall_CBS43764.ncrna.fa.gz',
	'verticillium_alfalfae_vams_102_gca_000150825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/verticillium_alfalfae_vams_102_gca_000150825/ncrna/Verticillium_alfalfae_vams_102_gca_000150825.ASM15082v1.ncrna.fa.gz',
	'verticillium_dahliae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/verticillium_dahliae/ncrna/Verticillium_dahliae.ASM15067v2.ncrna.fa.gz',
	'verticillium_dahliaejr2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/verticillium_dahliaejr2/ncrna/Verticillium_dahliaejr2.VDAG_JR2v.4.0.ncrna.fa.gz',
	'verticillium_longisporum_gca_001268145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/verticillium_longisporum_gca_001268145/ncrna/Verticillium_longisporum_gca_001268145.vl1.denovo.v1.ncrna.fa.gz',
	'vicugna_pacos'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/vicugna_pacos/ncrna/Vicugna_pacos.vicPac1.ncrna.fa.gz',
	'vigna_angularis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/vigna_angularis/ncrna/Vigna_angularis.Vigan1.1.ncrna.fa.gz',
	'vigna_radiata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/vigna_radiata/ncrna/Vigna_radiata.Vradiata_ver6.ncrna.fa.gz',
	'vitis_vinifera'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/vitis_vinifera/ncrna/Vitis_vinifera.12X.ncrna.fa.gz',
	'vitrella_brassicaformis_ccmp3155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/vitrella_brassicaformis_ccmp3155/ncrna/Vitrella_brassicaformis_ccmp3155.Vbrassicaformis.ncrna.fa.gz',
	'vittaforma_corneae_atcc_50505_gca_000231115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/vittaforma_corneae_atcc_50505_gca_000231115/ncrna/Vittaforma_corneae_atcc_50505_gca_000231115.Vitt_corn_V1.ncrna.fa.gz',
	'vulpes_vulpes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/vulpes_vulpes/ncrna/Vulpes_vulpes.VulVul2.2.ncrna.fa.gz',
	'wallemia_ichthyophaga_exf_994_gca_000400465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/wallemia_ichthyophaga_exf_994_gca_000400465/ncrna/Wallemia_ichthyophaga_exf_994_gca_000400465.Wallemia_ichthyophaga_version_1.0.ncrna.fa.gz',
	'wallemia_mellicola_cbs_633_66_gca_000263375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/wallemia_mellicola_cbs_633_66_gca_000263375/ncrna/Wallemia_mellicola_cbs_633_66_gca_000263375.Wallemia_sebi_v1.0.ncrna.fa.gz',
	'wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255/ncrna/Wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255.Wican1.ncrna.fa.gz',
	'wickerhamomyces_ciferrii_gca_000313485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/wickerhamomyces_ciferrii_gca_000313485/ncrna/Wickerhamomyces_ciferrii_gca_000313485.ASM31348v1.ncrna.fa.gz',
	'wolfiporia_cocos_md_104_ss10_gca_000344635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/wolfiporia_cocos_md_104_ss10_gca_000344635/ncrna/Wolfiporia_cocos_md_104_ss10_gca_000344635.Wolco1.ncrna.fa.gz',
	'xenopus_tropicalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/xenopus_tropicalis/ncrna/Xenopus_tropicalis.JGI_4.2.ncrna.fa.gz',
	'xiphophorus_couchianus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/xiphophorus_couchianus/ncrna/Xiphophorus_couchianus.Xiphophorus_couchianus-4.0.1.ncrna.fa.gz',
	'xiphophorus_maculatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/xiphophorus_maculatus/ncrna/Xiphophorus_maculatus.X_maculatus-5.0-male.ncrna.fa.gz',
	'xylona_heveae_tc161_gca_001619985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/xylona_heveae_tc161_gca_001619985/ncrna/Xylona_heveae_tc161_gca_001619985.Xylona_heveae_TC161_v1.0.ncrna.fa.gz',
	'yarrowia_lipolytica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/yarrowia_lipolytica/ncrna/Yarrowia_lipolytica.GCA_000002525.1.ncrna.fa.gz',
	'yarrowia_lipolytica_gca_001761485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/yarrowia_lipolytica_gca_001761485/ncrna/Yarrowia_lipolytica_gca_001761485.ASM176148v1.ncrna.fa.gz',
	'yarrowia_lipolytica_gca_900087985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/yarrowia_lipolytica_gca_900087985/ncrna/Yarrowia_lipolytica_gca_900087985.YALIA101.ncrna.fa.gz',
	'zea_mays'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/zea_mays/ncrna/Zea_mays.B73_RefGen_v4.ncrna.fa.gz',
	'zonotrichia_albicollis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/zonotrichia_albicollis/ncrna/Zonotrichia_albicollis.Zonotrichia_albicollis-1.0.1.ncrna.fa.gz',
	'zootermopsis_nevadensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/zootermopsis_nevadensis/ncrna/Zootermopsis_nevadensis.ZooNev1.0.ncrna.fa.gz',
	'zygosaccharomyces_bailii_clib_213_gca_000442885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/zygosaccharomyces_bailii_clib_213_gca_000442885/ncrna/Zygosaccharomyces_bailii_clib_213_gca_000442885.ZYBA0.ncrna.fa.gz',
	'zygosaccharomyces_bailii_gca_900162805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/zygosaccharomyces_bailii_gca_900162805/ncrna/Zygosaccharomyces_bailii_gca_900162805.version_1.ncrna.fa.gz',
	'zygosaccharomyces_bailii_isa1307_gca_000530735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/zygosaccharomyces_bailii_isa1307_gca_000530735/ncrna/Zygosaccharomyces_bailii_isa1307_gca_000530735.Z_bailii_ISA1307_version_3.ncrna.fa.gz',
	'zygosaccharomyces_parabailii_gca_001984395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/zygosaccharomyces_parabailii_gca_001984395/ncrna/Zygosaccharomyces_parabailii_gca_001984395.ASM198439v2.ncrna.fa.gz',
	'zygosaccharomyces_rouxii_gca_000026365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/zygosaccharomyces_rouxii_gca_000026365/ncrna/Zygosaccharomyces_rouxii_gca_000026365.ASM2636v1.ncrna.fa.gz',
	'zygosaccharomyces_rouxii_gca_001972345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/zygosaccharomyces_rouxii_gca_001972345/ncrna/Zygosaccharomyces_rouxii_gca_001972345.ZYGR_assembly01.ncrna.fa.gz',
	'zymoseptoria_brevis_gca_000966595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/zymoseptoria_brevis_gca_000966595/ncrna/Zymoseptoria_brevis_gca_000966595.ASM96659v1.ncrna.fa.gz',
	'zymoseptoria_tritici'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/zymoseptoria_tritici/ncrna/Zymoseptoria_tritici.MG2.ncrna.fa.gz',
	);### NCRNA URLS END ###
			}

		if(!$silent){print"\nDownload ncRNA data for $species{$species} from Ensembl database...";}
		$status=getstore("$ensembl_ncRNA{$species}","UNITAS_refdump_$species/ensembl_ncRNA.gz");
		if(is_success($status))
			{
			if(!$silent){print" OK.";}
			$db_versions_info.="Ensembl version/date: ..................... Release $ensembl_release\n";
			$db_versions_info.="EnsemblGenomes version/date: .............. Release $ensemblgenomes_release\n";
			}
		else
			{
			if(!$silent){print" FAILED! $status. Download data from smallRNAgroup-SERVER...";}
			$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/Ensembl/Ensembl-ncRNA_$species.fasta");
			$response = $useragent->request($request);
			$content = $response->content;
			open(my $out, '>', "UNITAS_refdump_$species/ensembl_ncRNA.gz");
			print $out "$content";
			close $out;
			if(-e"UNITAS_refdump_$species/ensembl_ncRNA.gz")
				{
				if(!$silent){print" OK.";}
				$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/Ensembl/release');
				$response = $useragent->request($request);
				$version1 = $response->content;
				$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/Ensembl/releaseEnsemblGenomes');
				$response = $useragent->request($request);
				$version2 = $response->content;
				$db_versions_info.="Ensembl version/date: ..................... Release $version1\n";
				$db_versions_info.="EnsemblGenomes version/date: .............. Release $version2\n";
				}
			else
				{
				if(!$silent){print" FAILED! $status.\n\n";}
				}
			}
		if(!$silent){print"\nExtract file...";}
		$ae=Archive::Extract->new(archive=>"UNITAS_refdump_$species/ensembl_ncRNA.gz");
		$ae->extract(to=>"UNITAS_refdump_$species")||die print"\nERROR! Unable to decompress UNITAS_refdump_$species/ensembl_ncRNA.gz.\n$!\n\n";
		if(!$silent){print" OK.\nRemove compressed file...";}
		unlink"UNITAS_refdump_$species/ensembl_ncRNA.gz"||print"\nERROR! Unable to remove compressed file UNITAS_refdump_$species/ensembl_ncRNA.gz.\n$!\n\n";
		if(!$silent){print" OK.\nRemove miRNA precursor sequences from ncRNA data file...";}

		# remove miRNA sequences from ncRNA file. miRNA annotation will base on miRBase data, not Ensembl data.
		open(NCRNA,"UNITAS_refdump_$species/ensembl_ncRNA")||die print"\nERROR! Unable to read UNITAS_refdump_$species/ensembl_ncRNA.\n$!\n\n";
		open(NCRNA_EDIT,">UNITAS_refdump_$species/ncRNA.fas")||die print"\nERROR! Unable to create UNITAS_refdump_$species/ncRNA.fas.\n$!\n\n";
		%ncRNA=();
		%avoid_redundant_names=();
		while(<NCRNA>)
			{
			$_=~s/\s*$//;
			if($_=~/^>/)
				{
				# extract biotype
				if($_=~s/gene_biotype:[^ ]+//)
					{
					$biotype=$&;
					$biotype=~s/gene_biotype://;
					}
				elsif($_=~s/transcript_biotype:[^ ]+//)
					{
					$biotype=$&;
					$biotype=~s/transcript_biotype://;
					}
				else
					{
					$biotype="unknown";
					}

				# extract gene symbol
				if($_=~s/gene_symbol:[^ ]+//)
					{
					$symbol=$&;
					$symbol=~s/gene_symbol://;
					}
				else
					{
					$symbol=$_;
					$symbol=~s/ .*$//;
					}

				# extract gene ID
				if($_=~s/gene:[^ ]+//)
					{
					$geneID=$&;
					$geneID=~s/gene://;
					}
				else
					{
					$geneID="unknownID";
					}

				# extract anticodon for tRNAs
				if($_=~s/ for anticodon ...//)
					{
					$anticodon=$&;
					$anticodon=~s/ /_/g;
					}
				else
					{
					$anticodon="";
					}

				# check if key for hash has been used before
				$avoid_redundant_names{"$biotype|${symbol}$anticodon-$geneID"}++;
				if($avoid_redundant_names{"$biotype|${symbol}$anticodon-$geneID"}==1)
					{
					$ncRNA_key="$biotype|${symbol}${anticodon}-$geneID";
					}
				else
					{
					$red_ID=$avoid_redundant_names{"$biotype|${symbol}$anticodon-$geneID"};
					$ncRNA_key="$biotype|${symbol}${anticodon}-$geneID_($red_ID)";
					}
				}
			elsif($biotype!~/miRNA/)
				{
				$ncRNA{$ncRNA_key}.=$_;
				}
			}
		close NCRNA;
		foreach$head(sort{$a cmp $b}keys%ncRNA)
			{
			print NCRNA_EDIT">$head\n$ncRNA{$head}\n";
			}
		undef%ncRNA;
		undef%avoid_redundant_names;
		if(!$silent){print" OK.";}

		# download and process ensembl cDNA data
		$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/URLs/Ensembl_cDNA');
		$response = $useragent->request($request);
		$eval_URL="";
		$eval_URL = $response->content;
		if(length$eval_URL>0)
			{
			eval$eval_URL;
			}
		else
			{
			if(!$silent){print"\nFailed to load current Ensembl URLs for cDNA data from smallRNAgroup-SERVER. Use internal list.";}
### CDNA URLS START ###
%ensembl_cDNA=
	(
	'_candida_arabinofermentans_nrrl_yb_2248_gca_001661425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_arabinofermentans_nrrl_yb_2248_gca_001661425/cdna/_candida_arabinofermentans_nrrl_yb_2248_gca_001661425.Canar1.cdna.all.fa.gz',
	'_candida_auris_gca_001189475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_auris_gca_001189475/cdna/_candida_auris_gca_001189475.ASM118947v1.cdna.all.fa.gz',
	'_candida_auris_gca_002759435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_auris_gca_002759435/cdna/_candida_auris_gca_002759435.Cand_auris_B8441_V2.cdna.all.fa.gz',
	'_candida_auris_gca_002775015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_auris_gca_002775015/cdna/_candida_auris_gca_002775015.Cand_auris_B11221_V1.cdna.all.fa.gz',
	'_candida_auris_gca_003013715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_auris_gca_003013715/cdna/_candida_auris_gca_003013715.Cand_auris_B11220.cdna.all.fa.gz',
	'_candida_auris_gca_003014415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_auris_gca_003014415/cdna/_candida_auris_gca_003014415.Cand_auris_B11243.cdna.all.fa.gz',
	'_candida_duobushaemulonis_gca_002926085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_duobushaemulonis_gca_002926085/cdna/_candida_duobushaemulonis_gca_002926085.CanDuoHae_v1.0.cdna.all.fa.gz',
	'_candida_glabrata_gca_000002545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/_candida_glabrata_gca_000002545/cdna/_candida_glabrata_gca_000002545.ASM254v2.cdna.all.fa.gz',
	'_candida_glabrata_gca_002219185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_glabrata_gca_002219185/cdna/_candida_glabrata_gca_002219185.ASM221918v1.cdna.all.fa.gz',
	'_candida_glabrata_gca_002219195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/_candida_glabrata_gca_002219195/cdna/_candida_glabrata_gca_002219195.ASM221919v1.cdna.all.fa.gz',
	'_candida_haemulonis_gca_002926055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_haemulonis_gca_002926055/cdna/_candida_haemulonis_gca_002926055.CanHae_1.0.cdna.all.fa.gz',
	'_candida_pseudohaemulonis_gca_003013735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/_candida_pseudohaemulonis_gca_003013735/cdna/_candida_pseudohaemulonis_gca_003013735.Cand_pseudohaemulonii_B12108.cdna.all.fa.gz',
	'_candida_tenuis_atcc_10573_gca_000223465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/_candida_tenuis_atcc_10573_gca_000223465/cdna/_candida_tenuis_atcc_10573_gca_000223465.Candida_tenuis_v1.0.cdna.all.fa.gz',
	'absidia_glauca_gca_900079185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/absidia_glauca_gca_900079185/cdna/Absidia_glauca_gca_900079185.AG_v1.cdna.all.fa.gz',
	'absidia_repens_gca_002105175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/absidia_repens_gca_002105175/cdna/Absidia_repens_gca_002105175.Absrep1.cdna.all.fa.gz',
	'acanthochromis_polyacanthus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/acanthochromis_polyacanthus/cdna/Acanthochromis_polyacanthus.ASM210954v1.cdna.abinitio.fa.gz',
	'acaromyces_ingoldii_gca_003144295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/acaromyces_ingoldii_gca_003144295/cdna/Acaromyces_ingoldii_gca_003144295.Acain1.cdna.all.fa.gz',
	'acidomyces_richmondensis_bfw_gca_001592465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/acidomyces_richmondensis_bfw_gca_001592465/cdna/Acidomyces_richmondensis_bfw_gca_001592465.Acidomyces_richmondensis_BFW_v1.0.cdna.all.fa.gz',
	'acremonium_chrysogenum_atcc_11550_gca_000769265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/acremonium_chrysogenum_atcc_11550_gca_000769265/cdna/Acremonium_chrysogenum_atcc_11550_gca_000769265.ASM76926v1.cdna.all.fa.gz',
	'acyrthosiphon_pisum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/acyrthosiphon_pisum/cdna/Acyrthosiphon_pisum.Acyr_2.0.cdna.all.fa.gz',
	'adineta_vaga'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/adineta_vaga/cdna/Adineta_vaga.AMS_PRJEB1171_v1.cdna.all.fa.gz',
	'aedes_aegypti'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/aedes_aegypti/cdna/Aedes_aegypti.AaegL3.cdna.all.fa.gz',
	'aegilops_tauschii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/aegilops_tauschii/cdna/Aegilops_tauschii.Aet_v4.0.cdna.all.fa.gz',
	'agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555/cdna/Agaricus_bisporus_var_burnettii_jb137_s8_gca_000300555.Agabi_varbur_1.cdna.all.fa.gz',
	'ailuropoda_melanoleuca'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ailuropoda_melanoleuca/cdna/Ailuropoda_melanoleuca.ailMel1.cdna.abinitio.fa.gz',
	'albugo_laibachii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/albugo_laibachii/cdna/Albugo_laibachii.ENA1.cdna.all.fa.gz',
	'allomyces_macrogynus_atcc_38327_gca_000151295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_blastocladiomycota1_collection/allomyces_macrogynus_atcc_38327_gca_000151295/cdna/Allomyces_macrogynus_atcc_38327_gca_000151295.A_macrogynus_V3.cdna.all.fa.gz',
	'alternaria_alternata_gca_001642055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/alternaria_alternata_gca_001642055/cdna/Alternaria_alternata_gca_001642055.Altal1.cdna.all.fa.gz',
	'amanita_muscaria_koide_bx008_gca_000827485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/amanita_muscaria_koide_bx008_gca_000827485/cdna/Amanita_muscaria_koide_bx008_gca_000827485.Amanita_muscaria_Koide_BX008_v1.0.cdna.all.fa.gz',
	'amborella_trichopoda'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/amborella_trichopoda/cdna/Amborella_trichopoda.AMTR1.0.cdna.all.fa.gz',
	'amphiamblys_sp_wsbs2006_gca_001875675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/amphiamblys_sp_wsbs2006_gca_001875675/cdna/Amphiamblys_sp_wsbs2006_gca_001875675.ASM187567v1.cdna.all.fa.gz',
	'amphilophus_citrinellus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/amphilophus_citrinellus/cdna/Amphilophus_citrinellus.Midas_v5.cdna.abinitio.fa.gz',
	'amphimedon_queenslandica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/amphimedon_queenslandica/cdna/Amphimedon_queenslandica.Aqu1.cdna.all.fa.gz',
	'amphiprion_ocellaris'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/amphiprion_ocellaris/cdna/Amphiprion_ocellaris.AmpOce1.0.cdna.abinitio.fa.gz',
	'amphiprion_percula'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/amphiprion_percula/cdna/Amphiprion_percula.Nemo_v1.cdna.abinitio.fa.gz',
	'anabas_testudineus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anabas_testudineus/cdna/Anabas_testudineus.fAnaTes1.1.cdna.abinitio.fa.gz',
	'anaeromyces_robustus_gca_002104895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_neocallimastigomycota1_collection/anaeromyces_robustus_gca_002104895/cdna/Anaeromyces_robustus_gca_002104895.Anaeromyces_sp._S4_v1.0.cdna.all.fa.gz',
	'anas_platyrhynchos_platyrhynchos'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anas_platyrhynchos_platyrhynchos/cdna/Anas_platyrhynchos_platyrhynchos.CAU_duck1.0.cdna.abinitio.fa.gz',
	'angomonas_deanei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/angomonas_deanei/cdna/Angomonas_deanei.Angomonas_deanei_Genome.cdna.all.fa.gz',
	'anncaliia_algerae_pra109_gca_000385855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/anncaliia_algerae_pra109_gca_000385855/cdna/Anncaliia_algerae_pra109_gca_000385855.Annc_alge_PRA109_V4.cdna.all.fa.gz',
	'anncaliia_algerae_pra339_gca_000385875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/anncaliia_algerae_pra339_gca_000385875/cdna/Anncaliia_algerae_pra339_gca_000385875.Annc_alge_insect_USDA_JJB_V2.cdna.all.fa.gz',
	'anolis_carolinensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anolis_carolinensis/cdna/Anolis_carolinensis.AnoCar2.0.cdna.abinitio.fa.gz',
	'anopheles_darlingi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/anopheles_darlingi/cdna/Anopheles_darlingi.AdarC3.cdna.abinitio.fa.gz',
	'anopheles_gambiae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/anopheles_gambiae/cdna/Anopheles_gambiae.AgamP4.cdna.abinitio.fa.gz',
	'anoplophora_glabripennis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/anoplophora_glabripennis/cdna/Anoplophora_glabripennis.Agla_1.0.cdna.all.fa.gz',
	'anser_brachyrhynchus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/anser_brachyrhynchus/cdna/Anser_brachyrhynchus.ASM259213v1.cdna.abinitio.fa.gz',
	'anthracocystis_flocculosa_pf_1_gca_000417875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/anthracocystis_flocculosa_pf_1_gca_000417875/cdna/Anthracocystis_flocculosa_pf_1_gca_000417875.Pflocc_1.0.cdna.all.fa.gz',
	'aotus_nancymaae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/aotus_nancymaae/cdna/Aotus_nancymaae.Anan_2.0.cdna.abinitio.fa.gz',
	'aphanomyces_astaci'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/aphanomyces_astaci/cdna/Aphanomyces_astaci.Apha_asta_APO3_V1.cdna.all.fa.gz',
	'aphanomyces_invadans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/aphanomyces_invadans/cdna/Aphanomyces_invadans.Apha_inva_NJM9701_V1.cdna.all.fa.gz',
	'apis_mellifera'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/apis_mellifera/cdna/Apis_mellifera.Amel_4.5.cdna.all.fa.gz',
	'apteryx_haastii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/apteryx_haastii/cdna/Apteryx_haastii.aptHaa1.cdna.abinitio.fa.gz',
	'apteryx_owenii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/apteryx_owenii/cdna/Apteryx_owenii.aptOwe1.cdna.abinitio.fa.gz',
	'apteryx_rowi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/apteryx_rowi/cdna/Apteryx_rowi.aptRow1.cdna.abinitio.fa.gz',
	'arabidopsis_halleri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/arabidopsis_halleri/cdna/Arabidopsis_halleri.Ahal2.2.cdna.all.fa.gz',
	'arabidopsis_lyrata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/arabidopsis_lyrata/cdna/Arabidopsis_lyrata.v.1.0.cdna.abinitio.fa.gz',
	'arabidopsis_thaliana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/arabidopsis_thaliana/cdna/Arabidopsis_thaliana.TAIR10.cdna.abinitio.fa.gz',
	'armillaria_gallica_gca_002307695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/armillaria_gallica_gca_002307695/cdna/Armillaria_gallica_gca_002307695.Armga1.cdna.all.fa.gz',
	'armillaria_ostoyae_gca_900157425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/armillaria_ostoyae_gca_900157425/cdna/Armillaria_ostoyae_gca_900157425.version_2.cdna.all.fa.gz',
	'armillaria_solidipes_gca_002307675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/armillaria_solidipes_gca_002307675/cdna/Armillaria_solidipes_gca_002307675.Armost1.cdna.all.fa.gz',
	'arthrobotrys_oligospora_atcc_24927_gca_000225545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/arthrobotrys_oligospora_atcc_24927_gca_000225545/cdna/Arthrobotrys_oligospora_atcc_24927_gca_000225545.AOL24927_1.0.cdna.all.fa.gz',
	'ascochyta_rabiei_gca_001630375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/ascochyta_rabiei_gca_001630375/cdna/Ascochyta_rabiei_gca_001630375.AscRab1.0.cdna.all.fa.gz',
	'ascoidea_rubescens_dsm_1968_gca_001661345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/ascoidea_rubescens_dsm_1968_gca_001661345/cdna/Ascoidea_rubescens_dsm_1968_gca_001661345.Ascru1.cdna.all.fa.gz',
	'ashbya_gossypii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/ashbya_gossypii/cdna/Ashbya_gossypii.ASM9102v1.cdna.all.fa.gz',
	'aspergillus_aculeatinus_cbs_121060_gca_003184765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_aculeatinus_cbs_121060_gca_003184765/cdna/Aspergillus_aculeatinus_cbs_121060_gca_003184765.Aspacu1.cdna.all.fa.gz',
	'aspergillus_aculeatus_atcc_16872_gca_001890905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_aculeatus_atcc_16872_gca_001890905/cdna/Aspergillus_aculeatus_atcc_16872_gca_001890905.Aspac1.cdna.all.fa.gz',
	'aspergillus_brasiliensis_cbs_101740_gca_001889945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_brasiliensis_cbs_101740_gca_001889945/cdna/Aspergillus_brasiliensis_cbs_101740_gca_001889945.Aspbr1.cdna.all.fa.gz',
	'aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695/cdna/Aspergillus_brunneoviolaceus_cbs_621_78_gca_003184695.Aspbru1.cdna.all.fa.gz',
	'aspergillus_calidoustus_gca_001511075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_calidoustus_gca_001511075/cdna/Aspergillus_calidoustus_gca_001511075.Acal_Allpaths_LG.cdna.all.fa.gz',
	'aspergillus_campestris_ibt_28561_gca_002847485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_campestris_ibt_28561_gca_002847485/cdna/Aspergillus_campestris_ibt_28561_gca_002847485.Aspcam1.cdna.all.fa.gz',
	'aspergillus_candidus_gca_002847045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_candidus_gca_002847045/cdna/Aspergillus_candidus_gca_002847045.Aspcand1.cdna.all.fa.gz',
	'aspergillus_carbonarius_item_5010_gca_001990825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_carbonarius_item_5010_gca_001990825/cdna/Aspergillus_carbonarius_item_5010_gca_001990825.Aspca3.cdna.all.fa.gz',
	'aspergillus_clavatus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_clavatus/cdna/Aspergillus_clavatus.ASM271v1.cdna.all.fa.gz',
	'aspergillus_cristatus_gca_001717485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_cristatus_gca_001717485/cdna/Aspergillus_cristatus_gca_001717485.ASM171748v1.cdna.all.fa.gz',
	'aspergillus_ellipticus_cbs_707_79_gca_003184645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_ellipticus_cbs_707_79_gca_003184645/cdna/Aspergillus_ellipticus_cbs_707_79_gca_003184645.Aspell1.cdna.all.fa.gz',
	'aspergillus_eucalypticola_cbs_122712_gca_003184535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_eucalypticola_cbs_122712_gca_003184535/cdna/Aspergillus_eucalypticola_cbs_122712_gca_003184535.Aspeuc1.cdna.all.fa.gz',
	'aspergillus_flavus_af70_gca_000952835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_flavus_af70_gca_000952835/cdna/Aspergillus_flavus_af70_gca_000952835.ASM95283v1.cdna.all.fa.gz',
	'aspergillus_fumigatus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_fumigatus/cdna/Aspergillus_fumigatus.ASM265v1.cdna.all.fa.gz',
	'aspergillus_fumigatus_gca_002234985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_fumigatus_gca_002234985/cdna/Aspergillus_fumigatus_gca_002234985.ASM223498v1.cdna.all.fa.gz',
	'aspergillus_fumigatusa1163'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_fumigatusa1163/cdna/Aspergillus_fumigatusa1163.ASM15014v1.cdna.all.fa.gz',
	'aspergillus_glaucus_cbs_516_65_gca_001890805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_glaucus_cbs_516_65_gca_001890805/cdna/Aspergillus_glaucus_cbs_516_65_gca_001890805.Aspgl1.cdna.all.fa.gz',
	'aspergillus_heteromorphus_cbs_117_55_gca_003184545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_heteromorphus_cbs_117_55_gca_003184545/cdna/Aspergillus_heteromorphus_cbs_117_55_gca_003184545.Asphet1.cdna.all.fa.gz',
	'aspergillus_indologenus_cbs_114_80_gca_003184685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_indologenus_cbs_114_80_gca_003184685/cdna/Aspergillus_indologenus_cbs_114_80_gca_003184685.Aspind1.cdna.all.fa.gz',
	'aspergillus_japonicus_cbs_114_51_gca_003184785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_japonicus_cbs_114_51_gca_003184785/cdna/Aspergillus_japonicus_cbs_114_51_gca_003184785.Aspjap1.cdna.all.fa.gz',
	'aspergillus_lacticoffeatus_cbs_101883_gca_003184595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_lacticoffeatus_cbs_101883_gca_003184595/cdna/Aspergillus_lacticoffeatus_cbs_101883_gca_003184595.Asplac1.cdna.all.fa.gz',
	'aspergillus_lentulus_gca_001445615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_lentulus_gca_001445615/cdna/Aspergillus_lentulus_gca_001445615.Alt_assembly01.cdna.all.fa.gz',
	'aspergillus_luchuensis_cbs_106_47_gca_001890685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_luchuensis_cbs_106_47_gca_001890685/cdna/Aspergillus_luchuensis_cbs_106_47_gca_001890685.Aspfo1.cdna.all.fa.gz',
	'aspergillus_luchuensis_gca_001602395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_luchuensis_gca_001602395/cdna/Aspergillus_luchuensis_gca_001602395.AspL_2604.cdna.all.fa.gz',
	'aspergillus_neoniger_cbs_115656_gca_003184625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_neoniger_cbs_115656_gca_003184625/cdna/Aspergillus_neoniger_cbs_115656_gca_003184625.Aspneo1.cdna.all.fa.gz',
	'aspergillus_nidulans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_nidulans/cdna/Aspergillus_nidulans.ASM1142v1.cdna.all.fa.gz',
	'aspergillus_nidulans_fgsc_a4_gca_000149205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_nidulans_fgsc_a4_gca_000149205/cdna/Aspergillus_nidulans_fgsc_a4_gca_000149205.ASM14920v2.cdna.all.fa.gz',
	'aspergillus_niger'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_niger/cdna/Aspergillus_niger.ASM285v2.cdna.all.fa.gz',
	'aspergillus_niger_atcc_1015_gca_000230395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aspergillus_niger_atcc_1015_gca_000230395/cdna/Aspergillus_niger_atcc_1015_gca_000230395.ASPNI_v3.0.cdna.all.fa.gz',
	'aspergillus_niger_gca_001515345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_niger_gca_001515345/cdna/Aspergillus_niger_gca_001515345.ASM151534v1.cdna.all.fa.gz',
	'aspergillus_nomius_nrrl_13137_gca_001204775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_nomius_nrrl_13137_gca_001204775/cdna/Aspergillus_nomius_nrrl_13137_gca_001204775.ASM120477v1.cdna.all.fa.gz',
	'aspergillus_novofumigatus_ibt_16806_gca_002847465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_novofumigatus_ibt_16806_gca_002847465/cdna/Aspergillus_novofumigatus_ibt_16806_gca_002847465.Aspnov1.cdna.all.fa.gz',
	'aspergillus_ochraceoroseus_ibt_24754_gca_002846915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_ochraceoroseus_ibt_24754_gca_002846915/cdna/Aspergillus_ochraceoroseus_ibt_24754_gca_002846915.Aspergillus_ochraceoroseus_IBT_24754_v1.1.cdna.all.fa.gz',
	'aspergillus_oryzae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_oryzae/cdna/Aspergillus_oryzae.ASM18445v3.cdna.all.fa.gz',
	'aspergillus_oryzae_3_042_gca_000269785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aspergillus_oryzae_3_042_gca_000269785/cdna/Aspergillus_oryzae_3_042_gca_000269785.AspOry3042.cdna.all.fa.gz',
	'aspergillus_parasiticus_su_1_gca_000956085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_parasiticus_su_1_gca_000956085/cdna/Aspergillus_parasiticus_su_1_gca_000956085.ASM95608v1.cdna.all.fa.gz',
	'aspergillus_piperis_cbs_112811_gca_003184755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_piperis_cbs_112811_gca_003184755/cdna/Aspergillus_piperis_cbs_112811_gca_003184755.Asppip1.cdna.all.fa.gz',
	'aspergillus_rambellii_gca_000986645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_rambellii_gca_000986645/cdna/Aspergillus_rambellii_gca_000986645.ASM98664v1.cdna.all.fa.gz',
	'aspergillus_ruber_cbs_135680_gca_000600275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aspergillus_ruber_cbs_135680_gca_000600275/cdna/Aspergillus_ruber_cbs_135680_gca_000600275.Eurhe1.cdna.all.fa.gz',
	'aspergillus_saccharolyticus_jop_1030_1_gca_003184585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_saccharolyticus_jop_1030_1_gca_003184585/cdna/Aspergillus_saccharolyticus_jop_1030_1_gca_003184585.Aspsac1.cdna.all.fa.gz',
	'aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635/cdna/Aspergillus_sclerotiicarbonarius_cbs_121057_gca_003184635.Aspscle1.cdna.all.fa.gz',
	'aspergillus_sclerotioniger_cbs_115572_gca_003184525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_sclerotioniger_cbs_115572_gca_003184525/cdna/Aspergillus_sclerotioniger_cbs_115572_gca_003184525.Aspscl1.cdna.all.fa.gz',
	'aspergillus_steynii_ibt_23096_gca_002849105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_steynii_ibt_23096_gca_002849105/cdna/Aspergillus_steynii_ibt_23096_gca_002849105.Aspste1.cdna.all.fa.gz',
	'aspergillus_sydowii_cbs_593_65_gca_001890705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_sydowii_cbs_593_65_gca_001890705/cdna/Aspergillus_sydowii_cbs_593_65_gca_001890705.Aspsy1.cdna.all.fa.gz',
	'aspergillus_taichungensis_gca_002850765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_taichungensis_gca_002850765/cdna/Aspergillus_taichungensis_gca_002850765.Asptaic1.cdna.all.fa.gz',
	'aspergillus_terreus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/aspergillus_terreus/cdna/Aspergillus_terreus.ASM14961v1.cdna.all.fa.gz',
	'aspergillus_thermomutatus_gca_002237265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_thermomutatus_gca_002237265/cdna/Aspergillus_thermomutatus_gca_002237265.ASM223726v1.cdna.all.fa.gz',
	'aspergillus_tubingensis_cbs_134_48_gca_001890745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_tubingensis_cbs_134_48_gca_001890745/cdna/Aspergillus_tubingensis_cbs_134_48_gca_001890745.Asptu1.cdna.all.fa.gz',
	'aspergillus_turcosus_gca_002234965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_turcosus_gca_002234965/cdna/Aspergillus_turcosus_gca_002234965.ASM223496v1.cdna.all.fa.gz',
	'aspergillus_turcosus_gca_002234975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_turcosus_gca_002234975/cdna/Aspergillus_turcosus_gca_002234975.ASM223497v1.cdna.all.fa.gz',
	'aspergillus_udagawae_gca_001078395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_udagawae_gca_001078395/cdna/Aspergillus_udagawae_gca_001078395.Aud_assembly01.cdna.all.fa.gz',
	'aspergillus_ustus_gca_000812125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/aspergillus_ustus_gca_000812125/cdna/Aspergillus_ustus_gca_000812125.Austus1.0.cdna.all.fa.gz',
	'aspergillus_uvarum_cbs_121591_gca_003184745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_uvarum_cbs_121591_gca_003184745/cdna/Aspergillus_uvarum_cbs_121591_gca_003184745.Aspuva1.cdna.all.fa.gz',
	'aspergillus_vadensis_cbs_113365_gca_003184925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_vadensis_cbs_113365_gca_003184925/cdna/Aspergillus_vadensis_cbs_113365_gca_003184925.Aspvad1.cdna.all.fa.gz',
	'aspergillus_versicolor_cbs_583_65_gca_001890125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_versicolor_cbs_583_65_gca_001890125/cdna/Aspergillus_versicolor_cbs_583_65_gca_001890125.Aspve1.cdna.all.fa.gz',
	'aspergillus_violaceofuscus_cbs_115571_gca_003184705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/aspergillus_violaceofuscus_cbs_115571_gca_003184705/cdna/Aspergillus_violaceofuscus_cbs_115571_gca_003184705.Aspvio1.cdna.all.fa.gz',
	'aspergillus_wentii_dto_134e9_gca_001890725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/aspergillus_wentii_dto_134e9_gca_001890725/cdna/Aspergillus_wentii_dto_134e9_gca_001890725.Aspwe1.cdna.all.fa.gz',
	'astatotilapia_calliptera'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/astatotilapia_calliptera/cdna/Astatotilapia_calliptera.fAstCal1.2.cdna.abinitio.fa.gz',
	'astyanax_mexicanus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/astyanax_mexicanus/cdna/Astyanax_mexicanus.Astyanax_mexicanus-2.0.cdna.abinitio.fa.gz',
	'atta_cephalotes'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/atta_cephalotes/cdna/Atta_cephalotes.Attacep1.0.cdna.all.fa.gz',
	'aureobasidium_melanogenum_cbs_110374_gca_000721775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_melanogenum_cbs_110374_gca_000721775/cdna/Aureobasidium_melanogenum_cbs_110374_gca_000721775.Aureobasidium_pullulans_var._melanogenum_CBS_110374_v1.0.cdna.all.fa.gz',
	'aureobasidium_namibiae_cbs_147_97_gca_000721765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_namibiae_cbs_147_97_gca_000721765/cdna/Aureobasidium_namibiae_cbs_147_97_gca_000721765.Aureobasidium_pullulans_var._namibiae_CBS_147.97_v1.0.cdna.all.fa.gz',
	'aureobasidium_pullulans_exf_150_gca_000721785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_pullulans_exf_150_gca_000721785/cdna/Aureobasidium_pullulans_exf_150_gca_000721785.Aureobasidium_pullulans_var._pullulans_EXF-150_assembly_version_1.0.cdna.all.fa.gz',
	'aureobasidium_subglaciale_exf_2481_gca_000721755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/aureobasidium_subglaciale_exf_2481_gca_000721755/cdna/Aureobasidium_subglaciale_exf_2481_gca_000721755.Aureobasidium_pullulans_var._subglaciale_EXF-2481_v1.0.cdna.all.fa.gz',
	'babesia_bovis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/babesia_bovis/cdna/Babesia_bovis.ASM16539v1.cdna.all.fa.gz',
	'babesia_microti_strain_ri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/babesia_microti_strain_ri/cdna/Babesia_microti_strain_ri.ASM69194v1.cdna.all.fa.gz',
	'babjeviella_inositovora_nrrl_y_12698_gca_001661335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/babjeviella_inositovora_nrrl_y_12698_gca_001661335/cdna/Babjeviella_inositovora_nrrl_y_12698_gca_001661335.Babin1.cdna.all.fa.gz',
	'basidiobolus_meristosporus_cbs_931_73_gca_002104905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_entomophthoromycota1_collection/basidiobolus_meristosporus_cbs_931_73_gca_002104905/cdna/Basidiobolus_meristosporus_cbs_931_73_gca_002104905.Basme2finSC.cdna.all.fa.gz',
	'batrachochytrium_dendrobatidis_jam81_gca_000203795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/batrachochytrium_dendrobatidis_jam81_gca_000203795/cdna/Batrachochytrium_dendrobatidis_jam81_gca_000203795.v1.0.cdna.all.fa.gz',
	'batrachochytrium_dendrobatidis_jel423_gca_000149865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/batrachochytrium_dendrobatidis_jel423_gca_000149865/cdna/Batrachochytrium_dendrobatidis_jel423_gca_000149865.BD_JEL423.cdna.all.fa.gz',
	'batrachochytrium_salamandrivorans_gca_002006685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/batrachochytrium_salamandrivorans_gca_002006685/cdna/Batrachochytrium_salamandrivorans_gca_002006685.Batr_sala_BS_V1.cdna.all.fa.gz',
	'baudoinia_panamericana_uamh_10762_gca_000338955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/baudoinia_panamericana_uamh_10762_gca_000338955/cdna/Baudoinia_panamericana_uamh_10762_gca_000338955.Bauco1.cdna.all.fa.gz',
	'beauveria_bassiana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/beauveria_bassiana/cdna/Beauveria_bassiana.ASM168263v1.cdna.all.fa.gz',
	'beauveria_bassiana_arsef_2860_gca_000280675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/beauveria_bassiana_arsef_2860_gca_000280675/cdna/Beauveria_bassiana_arsef_2860_gca_000280675.ASM28067v1.cdna.all.fa.gz',
	'beauveria_bassiana_d1_5_gca_000770705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/beauveria_bassiana_d1_5_gca_000770705/cdna/Beauveria_bassiana_d1_5_gca_000770705.BBA1.0.cdna.all.fa.gz',
	'beauveria_bassiana_gca_002871155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/beauveria_bassiana_gca_002871155/cdna/Beauveria_bassiana_gca_002871155.ASM287115v1.cdna.all.fa.gz',
	'belgica_antarctica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/belgica_antarctica/cdna/Belgica_antarctica.ASM77530v1.cdna.all.fa.gz',
	'beta_vulgaris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/beta_vulgaris/cdna/Beta_vulgaris.RefBeet-1.2.2.cdna.all.fa.gz',
	'bifiguratus_adelaidae_gca_002261195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/bifiguratus_adelaidae_gca_002261195/cdna/Bifiguratus_adelaidae_gca_002261195.ASM226119v1.cdna.all.fa.gz',
	'bigelowiella_natans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/bigelowiella_natans/cdna/Bigelowiella_natans.Bigna1.cdna.all.fa.gz',
	'bigelowiella_natans_gca_000002455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/bigelowiella_natans_gca_000002455/cdna/Bigelowiella_natans_gca_000002455.ASM245v1.cdna.all.fa.gz',
	'bipolaris_maydis_atcc_48331_gca_000354255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_maydis_atcc_48331_gca_000354255/cdna/Bipolaris_maydis_atcc_48331_gca_000354255.CocheC4_1.cdna.all.fa.gz',
	'bipolaris_maydis_c5_gca_000338975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_maydis_c5_gca_000338975/cdna/Bipolaris_maydis_c5_gca_000338975.CocheC5_3.cdna.all.fa.gz',
	'bipolaris_sorokiniana_nd90pr_gca_000338995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_sorokiniana_nd90pr_gca_000338995/cdna/Bipolaris_sorokiniana_nd90pr_gca_000338995.Cocsa1.cdna.all.fa.gz',
	'bipolaris_zeicola_26_r_13_gca_000523435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/bipolaris_zeicola_26_r_13_gca_000523435/cdna/Bipolaris_zeicola_26_r_13_gca_000523435.Cochliobolus_carbonum_v1.0.cdna.all.fa.gz',
	'bison_bison_bison'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/bison_bison_bison/cdna/Bison_bison_bison.Bison_UMD1.0.cdna.abinitio.fa.gz',
	'blastomyces_dermatitidis_atcc_18188_gca_000151595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_dermatitidis_atcc_18188_gca_000151595/cdna/Blastomyces_dermatitidis_atcc_18188_gca_000151595.BD_ATCC18188_V1.cdna.all.fa.gz',
	'blastomyces_dermatitidis_atcc_26199_gca_000166155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_dermatitidis_atcc_26199_gca_000166155/cdna/Blastomyces_dermatitidis_atcc_26199_gca_000166155.BD_ATCC26199_V2.cdna.all.fa.gz',
	'blastomyces_dermatitidis_er_3_gca_000003525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_dermatitidis_er_3_gca_000003525/cdna/Blastomyces_dermatitidis_er_3_gca_000003525.BD_ER3_V1.cdna.all.fa.gz',
	'blastomyces_gilchristii_slh14081_gca_000003855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blastomyces_gilchristii_slh14081_gca_000003855/cdna/Blastomyces_gilchristii_slh14081_gca_000003855.BD_SLH14081_V1.cdna.all.fa.gz',
	'blastomyces_percursus_gca_001883805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/blastomyces_percursus_gca_001883805/cdna/Blastomyces_percursus_gca_001883805.Blas_perc_EI222_V1.cdna.all.fa.gz',
	'blumeria_graminis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/blumeria_graminis/cdna/Blumeria_graminis.EF2.cdna.all.fa.gz',
	'blumeria_graminis_f_sp_tritici_96224_gca_000418435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/blumeria_graminis_f_sp_tritici_96224_gca_000418435/cdna/Blumeria_graminis_f_sp_tritici_96224_gca_000418435.Bgt_454_newbler_assembly.cdna.all.fa.gz',
	'bodo_saltans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/bodo_saltans/cdna/Bodo_saltans.BSAL.cdna.all.fa.gz',
	'bombus_impatiens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/bombus_impatiens/cdna/Bombus_impatiens.BIMP_2.0.cdna.all.fa.gz',
	'bombus_terrestris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/bombus_terrestris/cdna/Bombus_terrestris.Bter_1.0.cdna.all.fa.gz',
	'bombyx_mori'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/bombyx_mori/cdna/Bombyx_mori.ASM15162v1.cdna.all.fa.gz',
	'bos_mutus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/bos_mutus/cdna/Bos_mutus.BosGru_v2.0.cdna.abinitio.fa.gz',
	'bos_taurus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/bos_taurus/cdna/Bos_taurus.ARS-UCD1.2.cdna.abinitio.fa.gz',
	'botryobasidium_botryosum_fd_172_ss1_gca_000697705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/botryobasidium_botryosum_fd_172_ss1_gca_000697705/cdna/Botryobasidium_botryosum_fd_172_ss1_gca_000697705.Botbo1.cdna.all.fa.gz',
	'botrytis_cinerea'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/botrytis_cinerea/cdna/Botrytis_cinerea.ASM83294v1.cdna.all.fa.gz',
	'botrytis_cinerea_bcdw1_gca_000349525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/botrytis_cinerea_bcdw1_gca_000349525/cdna/Botrytis_cinerea_bcdw1_gca_000349525.BcDW1.cdna.all.fa.gz',
	'botrytis_cinerea_t4_gca_000227075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/botrytis_cinerea_t4_gca_000227075/cdna/Botrytis_cinerea_t4_gca_000227075.BotFuc_Mar2011.cdna.all.fa.gz',
	'brachypodium_distachyon'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brachypodium_distachyon/cdna/Brachypodium_distachyon.Brachypodium_distachyon_v3.0.cdna.all.fa.gz',
	'brassica_napus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brassica_napus/cdna/Brassica_napus.AST_PRJEB5043_v1.cdna.all.fa.gz',
	'brassica_oleracea'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brassica_oleracea/cdna/Brassica_oleracea.BOL.cdna.all.fa.gz',
	'brassica_rapa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/brassica_rapa/cdna/Brassica_rapa.Brapa_1.0.cdna.all.fa.gz',
	'brettanomyces_bruxellensis_awri1499_gca_000259595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/brettanomyces_bruxellensis_awri1499_gca_000259595/cdna/Brettanomyces_bruxellensis_awri1499_gca_000259595.AWRI1499_v1.0.cdna.all.fa.gz',
	'brugia_malayi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/brugia_malayi/cdna/Brugia_malayi.Bmal-4.0.cdna.all.fa.gz',
	'byssochlamys_spectabilis_no_5_gca_000497085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/byssochlamys_spectabilis_no_5_gca_000497085/cdna/Byssochlamys_spectabilis_no_5_gca_000497085.PVAR5_assembly01.cdna.all.fa.gz',
	'cadophora_sp_dse1049_gca_003073865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/cadophora_sp_dse1049_gca_003073865/cdna/Cadophora_sp_dse1049_gca_003073865.Cadsp1.cdna.all.fa.gz',
	'caenorhabditis_brenneri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_brenneri/cdna/Caenorhabditis_brenneri.C_brenneri-6.0.1b.cdna.all.fa.gz',
	'caenorhabditis_briggsae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_briggsae/cdna/Caenorhabditis_briggsae.CB4.cdna.all.fa.gz',
	'caenorhabditis_elegans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_elegans/cdna/Caenorhabditis_elegans.WBcel235.cdna.all.fa.gz',
	'caenorhabditis_japonica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_japonica/cdna/Caenorhabditis_japonica.C_japonica-7.0.1.cdna.all.fa.gz',
	'caenorhabditis_remanei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/caenorhabditis_remanei/cdna/Caenorhabditis_remanei.C_remanei-15.0.1.cdna.all.fa.gz',
	'calidris_pugnax'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/calidris_pugnax/cdna/Calidris_pugnax.ASM143184v1.cdna.abinitio.fa.gz',
	'calidris_pygmaea'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/calidris_pygmaea/cdna/Calidris_pygmaea.ASM369795v1.cdna.abinitio.fa.gz',
	'callithrix_jacchus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/callithrix_jacchus/cdna/Callithrix_jacchus.ASM275486v1.cdna.abinitio.fa.gz',
	'calocera_cornea_hhb12733_gca_001632435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/calocera_cornea_hhb12733_gca_001632435/cdna/Calocera_cornea_hhb12733_gca_001632435.Calco1.cdna.all.fa.gz',
	'calocera_viscosa_tufc12733_gca_001630345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/calocera_viscosa_tufc12733_gca_001630345/cdna/Calocera_viscosa_tufc12733_gca_001630345.Calvi1.cdna.all.fa.gz',
	'candida_albicans_12c_gca_000773845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_12c_gca_000773845/cdna/Candida_albicans_12c_gca_000773845.Cand_albi_12C_V2.cdna.all.fa.gz',
	'candida_albicans_19f_gca_000775445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_19f_gca_000775445/cdna/Candida_albicans_19f_gca_000775445.Cand_albi_19F_V2.cdna.all.fa.gz',
	'candida_albicans_ca6_gca_000784695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_ca6_gca_000784695/cdna/Candida_albicans_ca6_gca_000784695.Cand_albi_Ca6_V2.cdna.all.fa.gz',
	'candida_albicans_gc75_gca_000773735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_gc75_gca_000773735/cdna/Candida_albicans_gc75_gca_000773735.Cand_albi_GC75_V1.cdna.all.fa.gz',
	'candida_albicans_l26_gca_000775455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_l26_gca_000775455/cdna/Candida_albicans_l26_gca_000775455.Cand_albi_L26_V2.cdna.all.fa.gz',
	'candida_albicans_p34048_gca_000775465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p34048_gca_000775465/cdna/Candida_albicans_p34048_gca_000775465.Cand_albi_P34048_V1.cdna.all.fa.gz',
	'candida_albicans_p37005_gca_000773745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p37005_gca_000773745/cdna/Candida_albicans_p37005_gca_000773745.Cand_albi_P37005_V2.cdna.all.fa.gz',
	'candida_albicans_p37037_gca_000773825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p37037_gca_000773825/cdna/Candida_albicans_p37037_gca_000773825.Cand_albi_P37037_V1.cdna.all.fa.gz',
	'candida_albicans_p37039_gca_000784515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p37039_gca_000784515/cdna/Candida_albicans_p37039_gca_000784515.Cand_albi_P37039_V1.cdna.all.fa.gz',
	'candida_albicans_p57055_gca_000775505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p57055_gca_000775505/cdna/Candida_albicans_p57055_gca_000775505.Cand_albi_P57055_V1.cdna.all.fa.gz',
	'candida_albicans_p57072_gca_000773805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p57072_gca_000773805/cdna/Candida_albicans_p57072_gca_000773805.Cand_albi_P57072_V1.cdna.all.fa.gz',
	'candida_albicans_p60002_gca_000784525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p60002_gca_000784525/cdna/Candida_albicans_p60002_gca_000784525.Cand_albi_P60002_V1.cdna.all.fa.gz',
	'candida_albicans_p75010_gca_000784575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p75010_gca_000784575/cdna/Candida_albicans_p75010_gca_000784575.Cand_albi_P75010_V1.cdna.all.fa.gz',
	'candida_albicans_p75016_gca_000784595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p75016_gca_000784595/cdna/Candida_albicans_p75016_gca_000784595.Cand_albi_P75016_V1.cdna.all.fa.gz',
	'candida_albicans_p75063_gca_000775525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p75063_gca_000775525/cdna/Candida_albicans_p75063_gca_000775525.Cand_albi_P75063_V1.cdna.all.fa.gz',
	'candida_albicans_p76055_gca_000784505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p76055_gca_000784505/cdna/Candida_albicans_p76055_gca_000784505.Cand_albi_P76055_V1.cdna.all.fa.gz',
	'candida_albicans_p76067_gca_000784495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p76067_gca_000784495/cdna/Candida_albicans_p76067_gca_000784495.Cand_albi_P76067_V1.cdna.all.fa.gz',
	'candida_albicans_p78042_gca_000784615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p78042_gca_000784615/cdna/Candida_albicans_p78042_gca_000784615.Cand_albi_P78042_V1.cdna.all.fa.gz',
	'candida_albicans_p78048_gca_000773725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_p78048_gca_000773725/cdna/Candida_albicans_p78048_gca_000773725.Cand_albi_P78048_V1.cdna.all.fa.gz',
	'candida_albicans_p87_gca_000774085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p87_gca_000774085/cdna/Candida_albicans_p87_gca_000774085.Cand_albi_P87_V2.cdna.all.fa.gz',
	'candida_albicans_p94015_gca_000773755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_p94015_gca_000773755/cdna/Candida_albicans_p94015_gca_000773755.Cand_albi_P94015_V1.cdna.all.fa.gz',
	'candida_albicans_sc5314_gca_000784635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_sc5314_gca_000784635/cdna/Candida_albicans_sc5314_gca_000784635.Cand_albi_SC5314_V4.cdna.all.fa.gz',
	'candida_albicans_sc5314_gca_000784655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/candida_albicans_sc5314_gca_000784655/cdna/Candida_albicans_sc5314_gca_000784655.Cand_albi_SC5314_V3.cdna.all.fa.gz',
	'candida_albicans_wo_1_gca_000149445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_albicans_wo_1_gca_000149445/cdna/Candida_albicans_wo_1_gca_000149445.ASM14944v2.cdna.all.fa.gz',
	'candida_dubliniensis_cd36_gca_000026945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_dubliniensis_cd36_gca_000026945/cdna/Candida_dubliniensis_cd36_gca_000026945.ASM2694v1.cdna.all.fa.gz',
	'candida_maltosa_xu316_gca_000344705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_maltosa_xu316_gca_000344705/cdna/Candida_maltosa_xu316_gca_000344705.Cmaltosa_v01.cdna.all.fa.gz',
	'candida_orthopsilosis_co_90_125_gca_000315875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_orthopsilosis_co_90_125_gca_000315875/cdna/Candida_orthopsilosis_co_90_125_gca_000315875.ASM31587v1.cdna.all.fa.gz',
	'candida_parapsilosis_gca_000182765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/candida_parapsilosis_gca_000182765/cdna/Candida_parapsilosis_gca_000182765.ASM18276v2.cdna.all.fa.gz',
	'candida_tropicalis_mya_3404_gca_000006335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/candida_tropicalis_mya_3404_gca_000006335/cdna/Candida_tropicalis_mya_3404_gca_000006335.ASM633v3.cdna.all.fa.gz',
	'canis_familiaris'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/canis_familiaris/cdna/Canis_familiaris.CanFam3.1.cdna.abinitio.fa.gz',
	'canis_lupus_dingo'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/canis_lupus_dingo/cdna/Canis_lupus_dingo.ASM325472v1.cdna.abinitio.fa.gz',
	'capitella_teleta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/capitella_teleta/cdna/Capitella_teleta.Capitella_teleta_v1.0.cdna.all.fa.gz',
	'capra_hircus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/capra_hircus/cdna/Capra_hircus.ARS1.cdna.abinitio.fa.gz',
	'capronia_coronata_cbs_617_96_gca_000585585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/capronia_coronata_cbs_617_96_gca_000585585/cdna/Capronia_coronata_cbs_617_96_gca_000585585.Capr_coro_CBS_617_96_V1.cdna.all.fa.gz',
	'capronia_epimyces_cbs_606_96_gca_000585565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/capronia_epimyces_cbs_606_96_gca_000585565/cdna/Capronia_epimyces_cbs_606_96_gca_000585565.Capr_epim_CBS_606_96_V1.cdna.all.fa.gz',
	'capsaspora_owczarzaki_atcc_30864'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_ichthyosporea1_collection/capsaspora_owczarzaki_atcc_30864/cdna/Capsaspora_owczarzaki_atcc_30864.C_owczarzaki_V2.cdna.all.fa.gz',
	'carlito_syrichta'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/carlito_syrichta/cdna/Carlito_syrichta.Tarsius_syrichta-2.0.1.cdna.abinitio.fa.gz',
	'castor_canadensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/castor_canadensis/cdna/Castor_canadensis.C.can_genome_v1.0.cdna.abinitio.fa.gz',
	'catenaria_anguillulae_pl171_gca_002102555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_blastocladiomycota1_collection/catenaria_anguillulae_pl171_gca_002102555/cdna/Catenaria_anguillulae_pl171_gca_002102555.Catan2.cdna.all.fa.gz',
	'cavia_aperea'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cavia_aperea/cdna/Cavia_aperea.CavAp1.0.cdna.abinitio.fa.gz',
	'cavia_porcellus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cavia_porcellus/cdna/Cavia_porcellus.Cavpor3.0.cdna.abinitio.fa.gz',
	'cebus_capucinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cebus_capucinus/cdna/Cebus_capucinus.Cebus_imitator-1.0.cdna.abinitio.fa.gz',
	'ceraceosorus_bombacis_gca_900000165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ceraceosorus_bombacis_gca_900000165/cdna/Ceraceosorus_bombacis_gca_900000165.Ceraceosorus_bombacis.cdna.all.fa.gz',
	'ceraceosorus_guamensis_gca_003144195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ceraceosorus_guamensis_gca_003144195/cdna/Ceraceosorus_guamensis_gca_003144195.Cersp1.cdna.all.fa.gz',
	'ceratocystis_platani_gca_000978885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/ceratocystis_platani_gca_000978885/cdna/Ceratocystis_platani_gca_000978885.ASM97888v1.cdna.all.fa.gz',
	'cercocebus_atys'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cercocebus_atys/cdna/Cercocebus_atys.Caty_1.0.cdna.abinitio.fa.gz',
	'chaetomium_globosum_cbs_148_51_gca_000143365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/chaetomium_globosum_cbs_148_51_gca_000143365/cdna/Chaetomium_globosum_cbs_148_51_gca_000143365.ASM14336v1.cdna.all.fa.gz',
	'chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225/cdna/Chaetomium_thermophilum_var_thermophilum_dsm_1495_gca_000221225.CTHT_3.0.cdna.all.fa.gz',
	'chelonoidis_abingdonii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chelonoidis_abingdonii/cdna/Chelonoidis_abingdonii.ASM359739v1.cdna.abinitio.fa.gz',
	'chinchilla_lanigera'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chinchilla_lanigera/cdna/Chinchilla_lanigera.ChiLan1.0.cdna.abinitio.fa.gz',
	'chlamydomonas_reinhardtii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/chlamydomonas_reinhardtii/cdna/Chlamydomonas_reinhardtii.Chlamydomonas_reinhardtii_v5.5.cdna.all.fa.gz',
	'chlorocebus_sabaeus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chlorocebus_sabaeus/cdna/Chlorocebus_sabaeus.ChlSab1.1.cdna.abinitio.fa.gz',
	'choanephora_cucurbitarum_gca_001683725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/choanephora_cucurbitarum_gca_001683725/cdna/Choanephora_cucurbitarum_gca_001683725.ASM168372v1.cdna.all.fa.gz',
	'choloepus_hoffmanni'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/choloepus_hoffmanni/cdna/Choloepus_hoffmanni.choHof1.cdna.abinitio.fa.gz',
	'chondrus_crispus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/chondrus_crispus/cdna/Chondrus_crispus.ASM35022v2.cdna.all.fa.gz',
	'chroomonas_mesostigmatica_ccmp1168'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_cryptophyta1_collection/chroomonas_mesostigmatica_ccmp1168/cdna/Chroomonas_mesostigmatica_ccmp1168.ASM28609v1.cdna.all.fa.gz',
	'chrysemys_picta_bellii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/chrysemys_picta_bellii/cdna/Chrysemys_picta_bellii.Chrysemys_picta_bellii-3.0.3.cdna.abinitio.fa.gz',
	'ciona_intestinalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ciona_intestinalis/cdna/Ciona_intestinalis.KH.cdna.abinitio.fa.gz',
	'ciona_savignyi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ciona_savignyi/cdna/Ciona_savignyi.CSAV2.0.cdna.abinitio.fa.gz',
	'cladophialophora_bantiana_cbs_173_52_gca_000835475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/cladophialophora_bantiana_cbs_173_52_gca_000835475/cdna/Cladophialophora_bantiana_cbs_173_52_gca_000835475.Clad_bant_CBS_173_52_V1.cdna.all.fa.gz',
	'cladophialophora_carrionii_cbs_160_54_gca_000365165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cladophialophora_carrionii_cbs_160_54_gca_000365165/cdna/Cladophialophora_carrionii_cbs_160_54_gca_000365165.Clad_carr_CBS_160_54_V1.cdna.all.fa.gz',
	'cladophialophora_carrionii_gca_001700775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cladophialophora_carrionii_gca_001700775/cdna/Cladophialophora_carrionii_gca_001700775.ASM170077v1.cdna.all.fa.gz',
	'cladophialophora_immunda_gca_000835495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/cladophialophora_immunda_gca_000835495/cdna/Cladophialophora_immunda_gca_000835495.Clad_immu_CBS83496_V1.cdna.all.fa.gz',
	'cladophialophora_psammophila_cbs_110553_gca_000585535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cladophialophora_psammophila_cbs_110553_gca_000585535/cdna/Cladophialophora_psammophila_cbs_110553_gca_000585535.Clad_psam_CBS_110553_V1.cdna.all.fa.gz',
	'cladophialophora_yegresii_cbs_114405_gca_000585515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cladophialophora_yegresii_cbs_114405_gca_000585515/cdna/Cladophialophora_yegresii_cbs_114405_gca_000585515.Clad_yegr_CBS_114405_V1.cdna.all.fa.gz',
	'claviceps_purpurea_20_1_gca_000347355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/claviceps_purpurea_20_1_gca_000347355/cdna/Claviceps_purpurea_20_1_gca_000347355.ASM34735v1.cdna.all.fa.gz',
	'clavispora_lusitaniae_atcc_42720_gca_000003835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/clavispora_lusitaniae_atcc_42720_gca_000003835/cdna/Clavispora_lusitaniae_atcc_42720_gca_000003835.ASM383v1.cdna.all.fa.gz',
	'clavispora_lusitaniae_gca_001673695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/clavispora_lusitaniae_gca_001673695/cdna/Clavispora_lusitaniae_gca_001673695.ASM167369v2.cdna.all.fa.gz',
	'clohesyomyces_aquaticus_gca_002105025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/clohesyomyces_aquaticus_gca_002105025/cdna/Clohesyomyces_aquaticus_gca_002105025.Cloaq1.cdna.all.fa.gz',
	'coccidioides_immitis_h538_4_gca_000149815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/coccidioides_immitis_h538_4_gca_000149815/cdna/Coccidioides_immitis_h538_4_gca_000149815.ASM14981v1.cdna.all.fa.gz',
	'coccidioides_immitis_rmscc_2394_gca_000149895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/coccidioides_immitis_rmscc_2394_gca_000149895/cdna/Coccidioides_immitis_rmscc_2394_gca_000149895.ASM14989v1.cdna.all.fa.gz',
	'coccidioides_immitis_rmscc_3703_gca_000150085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/coccidioides_immitis_rmscc_3703_gca_000150085/cdna/Coccidioides_immitis_rmscc_3703_gca_000150085.ASM15008v1.cdna.all.fa.gz',
	'coccidioides_immitis_rs_gca_000149335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coccidioides_immitis_rs_gca_000149335/cdna/Coccidioides_immitis_rs_gca_000149335.ASM14933v2.cdna.all.fa.gz',
	'coccidioides_posadasii_c735_delta_sowgp_gca_000151335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coccidioides_posadasii_c735_delta_sowgp_gca_000151335/cdna/Coccidioides_posadasii_c735_delta_sowgp_gca_000151335.JCVI-cpa1-1.0.cdna.all.fa.gz',
	'coccidioides_posadasii_rmscc_3488_gca_000150055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/coccidioides_posadasii_rmscc_3488_gca_000150055/cdna/Coccidioides_posadasii_rmscc_3488_gca_000150055.ASM15005v1.cdna.all.fa.gz',
	'coccidioides_posadasii_str_silveira_gca_000170175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coccidioides_posadasii_str_silveira_gca_000170175/cdna/Coccidioides_posadasii_str_silveira_gca_000170175.CPS2.cdna.all.fa.gz',
	'colletotrichum_fioriniae_pj7_gca_000582985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/colletotrichum_fioriniae_pj7_gca_000582985/cdna/Colletotrichum_fioriniae_pj7_gca_000582985.CFIO01.cdna.all.fa.gz',
	'colletotrichum_gloeosporioides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_gloeosporioides/cdna/Colletotrichum_gloeosporioides.GCA_000319635.1.cdna.all.fa.gz',
	'colletotrichum_gloeosporioides_cg_14_gca_000446055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/colletotrichum_gloeosporioides_cg_14_gca_000446055/cdna/Colletotrichum_gloeosporioides_cg_14_gca_000446055.Colletotrichum_gloeosporioides1.cdna.all.fa.gz',
	'colletotrichum_graminicola'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_graminicola/cdna/Colletotrichum_graminicola.C_graminicola_M1_001_V1.cdna.all.fa.gz',
	'colletotrichum_higginsianum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_higginsianum/cdna/Colletotrichum_higginsianum.GCA_000313795.2.cdna.all.fa.gz',
	'colletotrichum_incanum_gca_001625285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_incanum_gca_001625285/cdna/Colletotrichum_incanum_gca_001625285.Ci238704v1.cdna.all.fa.gz',
	'colletotrichum_nymphaeae_sa_01_gca_001563115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_nymphaeae_sa_01_gca_001563115/cdna/Colletotrichum_nymphaeae_sa_01_gca_001563115.CNYM01.cdna.all.fa.gz',
	'colletotrichum_orbiculare'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/colletotrichum_orbiculare/cdna/Colletotrichum_orbiculare.GCA_000350065.1.cdna.all.fa.gz',
	'colletotrichum_salicis_gca_001563125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_salicis_gca_001563125/cdna/Colletotrichum_salicis_gca_001563125.CSAL01.cdna.all.fa.gz',
	'colletotrichum_sublineola_gca_000696135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/colletotrichum_sublineola_gca_000696135/cdna/Colletotrichum_sublineola_gca_000696135.ASM69613v1.cdna.all.fa.gz',
	'colletotrichum_tofieldiae_gca_001625265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/colletotrichum_tofieldiae_gca_001625265/cdna/Colletotrichum_tofieldiae_gca_001625265.Ct0861v1.cdna.all.fa.gz',
	'colobus_angolensis_palliatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/colobus_angolensis_palliatus/cdna/Colobus_angolensis_palliatus.Cang.pa_1.0.cdna.abinitio.fa.gz',
	'conidiobolus_coronatus_nrrl_28638_gca_001566745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_entomophthoromycota1_collection/conidiobolus_coronatus_nrrl_28638_gca_001566745/cdna/Conidiobolus_coronatus_nrrl_28638_gca_001566745.Conidiobolus_coronatus_NRRL28638.cdna.all.fa.gz',
	'coniella_lustricola_gca_003019895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/coniella_lustricola_gca_003019895/cdna/Coniella_lustricola_gca_003019895.Coniella_lustricola_v1.0.cdna.all.fa.gz',
	'coniochaeta_ligniaria_nrrl_30616_gca_001879275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/coniochaeta_ligniaria_nrrl_30616_gca_001879275/cdna/Coniochaeta_ligniaria_nrrl_30616_gca_001879275.Conlig_v1.0.cdna.all.fa.gz',
	'coniophora_puteana_rwd_64_598_ss2_gca_000271625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/coniophora_puteana_rwd_64_598_ss2_gca_000271625/cdna/Coniophora_puteana_rwd_64_598_ss2_gca_000271625.Conpu1.cdna.all.fa.gz',
	'coniosporium_apollinis_cbs_100218_gca_000281105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/coniosporium_apollinis_cbs_100218_gca_000281105/cdna/Coniosporium_apollinis_cbs_100218_gca_000281105.Coni_apol_CBS100218__V1.cdna.all.fa.gz',
	'coprinopsis_cinerea_okayama7_130_gca_000182895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/coprinopsis_cinerea_okayama7_130_gca_000182895/cdna/Coprinopsis_cinerea_okayama7_130_gca_000182895.CC3.cdna.all.fa.gz',
	'corchorus_capsularis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/corchorus_capsularis/cdna/Corchorus_capsularis.CCACVL1_1.0.cdna.all.fa.gz',
	'cordyceps_brongniartii_rcef_3172_gca_001636735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cordyceps_brongniartii_rcef_3172_gca_001636735/cdna/Cordyceps_brongniartii_rcef_3172_gca_001636735.BBO_1.0.cdna.all.fa.gz',
	'cordyceps_confragosa_rcef_1005_gca_001636795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cordyceps_confragosa_rcef_1005_gca_001636795/cdna/Cordyceps_confragosa_rcef_1005_gca_001636795.LEL_1.0.cdna.all.fa.gz',
	'cordyceps_fumosorosea_arsef_2679_gca_001636725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cordyceps_fumosorosea_arsef_2679_gca_001636725/cdna/Cordyceps_fumosorosea_arsef_2679_gca_001636725.ISF_1.0.cdna.all.fa.gz',
	'cordyceps_militaris_cm01_gca_000225605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cordyceps_militaris_cm01_gca_000225605/cdna/Cordyceps_militaris_cm01_gca_000225605.CmilitarisCM01_v01.cdna.all.fa.gz',
	'corynespora_cassiicola_philippines_gca_003016335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/corynespora_cassiicola_philippines_gca_003016335/cdna/Corynespora_cassiicola_philippines_gca_003016335.Corynespora_cassiicola_v1.0.cdna.all.fa.gz',
	'coturnix_japonica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/coturnix_japonica/cdna/Coturnix_japonica.Coturnix_japonica_2.0.cdna.abinitio.fa.gz',
	'crassostrea_gigas'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/crassostrea_gigas/cdna/Crassostrea_gigas.oyster_v9.cdna.all.fa.gz',
	'cricetulus_griseus_chok1gshd'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cricetulus_griseus_chok1gshd/cdna/Cricetulus_griseus_chok1gshd.CHOK1GS_HDv1.cdna.abinitio.fa.gz',
	'cricetulus_griseus_crigri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cricetulus_griseus_crigri/cdna/Cricetulus_griseus_crigri.CriGri_1.0.cdna.abinitio.fa.gz',
	'cricetulus_griseus_picr'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cricetulus_griseus_picr/cdna/Cricetulus_griseus_picr.CriGri-PICR.cdna.abinitio.fa.gz',
	'crocodylus_porosus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/crocodylus_porosus/cdna/Crocodylus_porosus.CroPor_comp1.cdna.abinitio.fa.gz',
	'cryptococcus_amylolentus_cbs_6039_gca_001720205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_amylolentus_cbs_6039_gca_001720205/cdna/Cryptococcus_amylolentus_cbs_6039_gca_001720205.Cryp_amyl_CBS6039_V3.cdna.all.fa.gz',
	'cryptococcus_amylolentus_cbs_6273_gca_001720235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_amylolentus_cbs_6273_gca_001720235/cdna/Cryptococcus_amylolentus_cbs_6273_gca_001720235.Cryp_amyl_CBS6273_V2.cdna.all.fa.gz',
	'cryptococcus_depauperatus_cbs_7841_gca_001720195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_depauperatus_cbs_7841_gca_001720195/cdna/Cryptococcus_depauperatus_cbs_7841_gca_001720195.Filo_depa_CBS7841_V1.cdna.all.fa.gz',
	'cryptococcus_depauperatus_cbs_7855_gca_001720245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_depauperatus_cbs_7855_gca_001720245/cdna/Cryptococcus_depauperatus_cbs_7855_gca_001720245.Filo_depa_CBS7855_V2.cdna.all.fa.gz',
	'cryptococcus_gattii_ca1280_gca_000836335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ca1280_gca_000836335/cdna/Cryptococcus_gattii_ca1280_gca_000836335.Cryp_gatt_CA1280_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_ca1873_gca_000855695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ca1873_gca_000855695/cdna/Cryptococcus_gattii_ca1873_gca_000855695.Cryp_gatt_CA1873_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_e566_gca_000875815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_e566_gca_000875815/cdna/Cryptococcus_gattii_e566_gca_000875815.Cryp_gatt_E566_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_ejb2_gca_000835745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ejb2_gca_000835745/cdna/Cryptococcus_gattii_ejb2_gca_000835745.Cryp_gatt_EJB2_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_nt_10_gca_000935105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_nt_10_gca_000935105/cdna/Cryptococcus_gattii_nt_10_gca_000935105.Cryp_gatt_NT-10_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_ru294_gca_000836355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_ru294_gca_000836355/cdna/Cryptococcus_gattii_ru294_gca_000836355.Cryp_gatt_Ru294_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_2001_935_1_gca_000835815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_2001_935_1_gca_000835815/cdna/Cryptococcus_gattii_vgii_2001_935_1_gca_000835815.Cryp_gatt_2001_935-1_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_99_473_gca_000836455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_99_473_gca_000836455/cdna/Cryptococcus_gattii_vgii_99_473_gca_000836455.Cryp_gatt_99_473_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_ca1014_gca_000875795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_ca1014_gca_000875795/cdna/Cryptococcus_gattii_vgii_ca1014_gca_000875795.Cryp_gatt_CA1014_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_cbs_10090_gca_000835765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_cbs_10090_gca_000835765/cdna/Cryptococcus_gattii_vgii_cbs_10090_gca_000835765.Cryp_gatt_CBS10090_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_la55_gca_000836315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_la55_gca_000836315/cdna/Cryptococcus_gattii_vgii_la55_gca_000836315.Cryp_gatt_LA55_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_mmrl2647_gca_000875855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_mmrl2647_gca_000875855/cdna/Cryptococcus_gattii_vgii_mmrl2647_gca_000875855.Cryp_gatt_MMRL2647_V2.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_r265_gca_000149475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_r265_gca_000149475/cdna/Cryptococcus_gattii_vgii_r265_gca_000149475.CNB2.cdna.all.fa.gz',
	'cryptococcus_gattii_vgii_ram5_gca_000836375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgii_ram5_gca_000836375/cdna/Cryptococcus_gattii_vgii_ram5_gca_000836375.Cryp_gatt_Ram5_V1.cdna.all.fa.gz',
	'cryptococcus_gattii_vgiv_ind107_gca_000835755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_vgiv_ind107_gca_000835755/cdna/Cryptococcus_gattii_vgiv_ind107_gca_000835755.Cryp_gatt_IND107_V2.cdna.all.fa.gz',
	'cryptococcus_gattii_wm276_gca_000185945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_gattii_wm276_gca_000185945/cdna/Cryptococcus_gattii_wm276_gca_000185945.ASM18594v1.cdna.all.fa.gz',
	'cryptococcus_neoformans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/cryptococcus_neoformans/cdna/Cryptococcus_neoformans.ASM9104v1.cdna.all.fa.gz',
	'cryptococcus_neoformans_var_grubii_h99_gca_000149245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_grubii_h99_gca_000149245/cdna/Cryptococcus_neoformans_var_grubii_h99_gca_000149245.CNA3.cdna.all.fa.gz',
	'cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385/cdna/Cryptococcus_neoformans_var_neoformans_b_3501a_gca_000149385.ASM14938v1.cdna.all.fa.gz',
	'cryptomonas_paramecium'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_cryptophyta1_collection/cryptomonas_paramecium/cdna/Cryptomonas_paramecium.ASM19445v1.cdna.all.fa.gz',
	'cryptosporidium_muris_rn66'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cryptosporidium_muris_rn66/cdna/Cryptosporidium_muris_rn66.JCVI-cmg-v1.0.cdna.all.fa.gz',
	'cryptosporidium_parvum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cryptosporidium_parvum/cdna/Cryptosporidium_parvum.ASM20969v1.cdna.all.fa.gz',
	'cryptosporidium_parvum_iowa_ii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cryptosporidium_parvum_iowa_ii/cdna/Cryptosporidium_parvum_iowa_ii.ASM16534v1.cdna.all.fa.gz',
	'cucumis_sativus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/cucumis_sativus/cdna/Cucumis_sativus.ASM407v2.cdna.all.fa.gz',
	'culex_quinquefasciatus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/culex_quinquefasciatus/cdna/Culex_quinquefasciatus.CpipJ2.cdna.all.fa.gz',
	'cutaneotrichosporon_oleaginosum_gca_001027345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cutaneotrichosporon_oleaginosum_gca_001027345/cdna/Cutaneotrichosporon_oleaginosum_gca_001027345.Triol1.cdna.all.fa.gz',
	'cyanistes_caeruleus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cyanistes_caeruleus/cdna/Cyanistes_caeruleus.cyaCae2.cdna.abinitio.fa.gz',
	'cyberlindnera_fabianii_gca_003205855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/cyberlindnera_fabianii_gca_003205855/cdna/Cyberlindnera_fabianii_gca_003205855.ASM320585v1.cdna.all.fa.gz',
	'cyberlindnera_jadinii_gca_001245095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cyberlindnera_jadinii_gca_001245095/cdna/Cyberlindnera_jadinii_gca_001245095.cj1.cdna.all.fa.gz',
	'cyberlindnera_jadinii_nrrl_y_1542_gca_001661405'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/cyberlindnera_jadinii_nrrl_y_1542_gca_001661405/cdna/Cyberlindnera_jadinii_nrrl_y_1542_gca_001661405.Cybja1.cdna.all.fa.gz',
	'cyclospora_cayetanensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/cyclospora_cayetanensis/cdna/Cyclospora_cayetanensis.ASM76915v2.cdna.all.fa.gz',
	'cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385/cdna/Cylindrobasidium_torrendii_fp15055_ss_10_gca_000934385.Cylto1.cdna.all.fa.gz',
	'cynoglossus_semilaevis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cynoglossus_semilaevis/cdna/Cynoglossus_semilaevis.Cse_v1.0.cdna.abinitio.fa.gz',
	'cyphellophora_europaea_cbs_101466_gca_000365145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/cyphellophora_europaea_cbs_101466_gca_000365145/cdna/Cyphellophora_europaea_cbs_101466_gca_000365145.Phia_euro_CBS_101466_V1.cdna.all.fa.gz',
	'cyprinodon_variegatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/cyprinodon_variegatus/cdna/Cyprinodon_variegatus.C_variegatus-1.0.cdna.abinitio.fa.gz',
	'dacryopinax_primogenitus_gca_000292625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/dacryopinax_primogenitus_gca_000292625/cdna/Dacryopinax_primogenitus_gca_000292625.Dacryopinax_sp._DJM_731_SSP1_v1.0.cdna.all.fa.gz',
	'dactylellina_haptotyla_cbs_200_50_gca_000441935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/dactylellina_haptotyla_cbs_200_50_gca_000441935/cdna/Dactylellina_haptotyla_cbs_200_50_gca_000441935.MHA_v2.cdna.all.fa.gz',
	'daedalea_quercina_l_15889_gca_001632345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/daedalea_quercina_l_15889_gca_001632345/cdna/Daedalea_quercina_l_15889_gca_001632345.Daequ1.cdna.all.fa.gz',
	'daldinia_sp_ec12_gca_002120325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/daldinia_sp_ec12_gca_002120325/cdna/Daldinia_sp_ec12_gca_002120325.DalEC12_v.1.0.cdna.all.fa.gz',
	'danaus_plexippus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/danaus_plexippus/cdna/Danaus_plexippus.Dpv3.cdna.all.fa.gz',
	'danio_rerio'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/danio_rerio/cdna/Danio_rerio.GRCz11.cdna.abinitio.fa.gz',
	'daphnia_magna'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/daphnia_magna/cdna/Daphnia_magna.daphmag2.4.cdna.all.fa.gz',
	'daphnia_pulex'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/daphnia_pulex/cdna/Daphnia_pulex.V1.0.cdna.all.fa.gz',
	'dasypus_novemcinctus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/dasypus_novemcinctus/cdna/Dasypus_novemcinctus.Dasnov3.0.cdna.abinitio.fa.gz',
	'daucus_carota'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/daucus_carota/cdna/Daucus_carota.ASM162521v1.cdna.all.fa.gz',
	'debaryomyces_fabryi_gca_001447935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/debaryomyces_fabryi_gca_001447935/cdna/Debaryomyces_fabryi_gca_001447935.debFab1.0.cdna.all.fa.gz',
	'debaryomyces_hansenii_cbs767_gca_000006445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/debaryomyces_hansenii_cbs767_gca_000006445/cdna/Debaryomyces_hansenii_cbs767_gca_000006445.ASM644v2.cdna.all.fa.gz',
	'dendroctonus_ponderosae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/dendroctonus_ponderosae/cdna/Dendroctonus_ponderosae.DendPond_male_1.0.cdna.all.fa.gz',
	'diaporthe_ampelina_gca_001006365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/diaporthe_ampelina_gca_001006365/cdna/Diaporthe_ampelina_gca_001006365.UCDDA912v1.0.cdna.all.fa.gz',
	'dichomitus_squalens_lyad_421_ss1_gca_000275845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/dichomitus_squalens_lyad_421_ss1_gca_000275845/cdna/Dichomitus_squalens_lyad_421_ss1_gca_000275845.Dichomitus_squalens_v1.0.cdna.all.fa.gz',
	'dictyostelium_discoideum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/dictyostelium_discoideum/cdna/Dictyostelium_discoideum.dicty_2.7.cdna.all.fa.gz',
	'dictyostelium_fasciculatum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/dictyostelium_fasciculatum/cdna/Dictyostelium_fasciculatum.DFas_2.0.cdna.all.fa.gz',
	'dictyostelium_lacteum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/dictyostelium_lacteum/cdna/Dictyostelium_lacteum.ASM160615v1.cdna.all.fa.gz',
	'dioscorea_rotundata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/dioscorea_rotundata/cdna/Dioscorea_rotundata.TDr96_F1_Pseudo_Chromosome_v1.0.cdna.all.fa.gz',
	'diplodia_seriata_gca_001006355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/diplodia_seriata_gca_001006355/cdna/Diplodia_seriata_gca_001006355.UCDDS831v0.1.cdna.all.fa.gz',
	'dipodomys_ordii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/dipodomys_ordii/cdna/Dipodomys_ordii.Dord_2.0.cdna.abinitio.fa.gz',
	'dothistroma_septosporum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/dothistroma_septosporum/cdna/Dothistroma_septosporum.GCA_000340195.1.cdna.all.fa.gz',
	'drechmeria_coniospora_gca_001625195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/drechmeria_coniospora_gca_001625195/cdna/Drechmeria_coniospora_gca_001625195.ASM162519v1.cdna.all.fa.gz',
	'drechslerella_stenobrocha_248_gca_000525045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/drechslerella_stenobrocha_248_gca_000525045/cdna/Drechslerella_stenobrocha_248_gca_000525045.DREv1.cdna.all.fa.gz',
	'dromaius_novaehollandiae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/dromaius_novaehollandiae/cdna/Dromaius_novaehollandiae.droNov1.cdna.abinitio.fa.gz',
	'drosophila_ananassae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_ananassae/cdna/Drosophila_ananassae.dana_caf1.cdna.all.fa.gz',
	'drosophila_erecta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_erecta/cdna/Drosophila_erecta.dere_caf1.cdna.all.fa.gz',
	'drosophila_grimshawi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_grimshawi/cdna/Drosophila_grimshawi.dgri_caf1.cdna.all.fa.gz',
	'drosophila_melanogaster'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_melanogaster/cdna/Drosophila_melanogaster.BDGP6.22.cdna.all.fa.gz',
	'drosophila_mojavensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_mojavensis/cdna/Drosophila_mojavensis.dmoj_caf1.cdna.all.fa.gz',
	'drosophila_persimilis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_persimilis/cdna/Drosophila_persimilis.dper_caf1.cdna.all.fa.gz',
	'drosophila_pseudoobscura'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_pseudoobscura/cdna/Drosophila_pseudoobscura.Dpse_3.0.cdna.all.fa.gz',
	'drosophila_sechellia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_sechellia/cdna/Drosophila_sechellia.dsec_caf1.cdna.all.fa.gz',
	'drosophila_simulans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_simulans/cdna/Drosophila_simulans.ASM75419v3.cdna.all.fa.gz',
	'drosophila_virilis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_virilis/cdna/Drosophila_virilis.dvir_caf1.cdna.all.fa.gz',
	'drosophila_willistoni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_willistoni/cdna/Drosophila_willistoni.dwil_caf1.cdna.all.fa.gz',
	'drosophila_yakuba'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/drosophila_yakuba/cdna/Drosophila_yakuba.dyak_caf1.cdna.all.fa.gz',
	'echinops_telfairi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/echinops_telfairi/cdna/Echinops_telfairi.TENREC.cdna.abinitio.fa.gz',
	'edhazardia_aedis_usnm_41457_gca_000230595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/edhazardia_aedis_usnm_41457_gca_000230595/cdna/Edhazardia_aedis_usnm_41457_gca_000230595.Edha_aedis_V4b.cdna.all.fa.gz',
	'eimeria_brunetti'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_brunetti/cdna/Eimeria_brunetti.EBH001.cdna.all.fa.gz',
	'eimeria_maxima'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_maxima/cdna/Eimeria_maxima.EMW001.cdna.all.fa.gz',
	'eimeria_mitis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_mitis/cdna/Eimeria_mitis.EMH001.cdna.all.fa.gz',
	'eimeria_praecox'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_praecox/cdna/Eimeria_praecox.EPH001.cdna.all.fa.gz',
	'eimeria_tenella'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_tenella/cdna/Eimeria_tenella.ASM283v1.cdna.all.fa.gz',
	'eimeria_tenella_gca_000499545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/eimeria_tenella_gca_000499545/cdna/Eimeria_tenella_gca_000499545.ETH001.cdna.all.fa.gz',
	'emergomyces_pasteuriana_ep9510_gca_001883825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/emergomyces_pasteuriana_ep9510_gca_001883825/cdna/Emergomyces_pasteuriana_ep9510_gca_001883825.Emmo_past_UAMH9510_V1.cdna.all.fa.gz',
	'emiliania_huxleyi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/emiliania_huxleyi/cdna/Emiliania_huxleyi.Emiliana_huxleyi_CCMP1516_main_genome_assembly_v1.0.cdna.all.fa.gz',
	'emmonsia_crescens_uamh_3008_gca_001008285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/emmonsia_crescens_uamh_3008_gca_001008285/cdna/Emmonsia_crescens_uamh_3008_gca_001008285.ASM100828v1.cdna.all.fa.gz',
	'emmonsia_parva_uamh_139_gca_001014755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/emmonsia_parva_uamh_139_gca_001014755/cdna/Emmonsia_parva_uamh_139_gca_001014755.ASM101475v1.cdna.all.fa.gz',
	'emmonsia_sp_cac_2015a_gca_001660665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/emmonsia_sp_cac_2015a_gca_001660665/cdna/Emmonsia_sp_cac_2015a_gca_001660665.Emmo_afri_EA111.cdna.all.fa.gz',
	'encephalitozoon_cuniculi_ecuniii_l_gca_001078035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_cuniculi_ecuniii_l_gca_001078035/cdna/Encephalitozoon_cuniculi_ecuniii_l_gca_001078035.ECIIIL.cdna.all.fa.gz',
	'encephalitozoon_hellem_atcc_50504_gca_000277815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_hellem_atcc_50504_gca_000277815/cdna/Encephalitozoon_hellem_atcc_50504_gca_000277815.ASM27781v3.cdna.all.fa.gz',
	'encephalitozoon_intestinalis_atcc_50506_gca_000146465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_intestinalis_atcc_50506_gca_000146465/cdna/Encephalitozoon_intestinalis_atcc_50506_gca_000146465.ASM14646v1.cdna.all.fa.gz',
	'encephalitozoon_romaleae_sj_2008_gca_000280035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/encephalitozoon_romaleae_sj_2008_gca_000280035/cdna/Encephalitozoon_romaleae_sj_2008_gca_000280035.ASM28003v2.cdna.all.fa.gz',
	'endocarpon_pusillum_z07020_gca_000464535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/endocarpon_pusillum_z07020_gca_000464535/cdna/Endocarpon_pusillum_z07020_gca_000464535.EPUS.cdna.all.fa.gz',
	'entamoeba_histolytica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/entamoeba_histolytica/cdna/Entamoeba_histolytica.JCVI-ESG2-1.0.cdna.all.fa.gz',
	'entamoeba_histolytica_gca_001662325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_histolytica_gca_001662325/cdna/Entamoeba_histolytica_gca_001662325.Ehistolytica_2016.cdna.all.fa.gz',
	'entamoeba_histolytica_hm_3_imss'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_histolytica_hm_3_imss/cdna/Entamoeba_histolytica_hm_3_imss.EHA.strHM3_v1.cdna.all.fa.gz',
	'entamoeba_histolytica_ku27'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_histolytica_ku27/cdna/Entamoeba_histolytica_ku27.EHA_ku27_v1.cdna.all.fa.gz',
	'entamoeba_nuttalli_p19'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/entamoeba_nuttalli_p19/cdna/Entamoeba_nuttalli_p19.ENU1_v1.cdna.all.fa.gz',
	'enterocytozoon_bieneusi_h348_gca_000209485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/enterocytozoon_bieneusi_h348_gca_000209485/cdna/Enterocytozoon_bieneusi_h348_gca_000209485.ASM20948v1.cdna.all.fa.gz',
	'epicoccum_nigrum_gca_002116315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/epicoccum_nigrum_gca_002116315/cdna/Epicoccum_nigrum_gca_002116315.ASM211631v1.cdna.all.fa.gz',
	'eptatretus_burgeri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/eptatretus_burgeri/cdna/Eptatretus_burgeri.Eburgeri_3.2.cdna.abinitio.fa.gz',
	'equus_asinus_asinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/equus_asinus_asinus/cdna/Equus_asinus_asinus.ASM303372v1.cdna.abinitio.fa.gz',
	'equus_caballus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/equus_caballus/cdna/Equus_caballus.EquCab3.0.cdna.abinitio.fa.gz',
	'eremothecium_cymbalariae_dbvpg_7215_gca_000235365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/eremothecium_cymbalariae_dbvpg_7215_gca_000235365/cdna/Eremothecium_cymbalariae_dbvpg_7215_gca_000235365.ASM23536v1.cdna.all.fa.gz',
	'eremothecium_gossypii_fdag1_gca_000968835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/eremothecium_gossypii_fdag1_gca_000968835/cdna/Eremothecium_gossypii_fdag1_gca_000968835.ASM96883v1.cdna.all.fa.gz',
	'erinaceus_europaeus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/erinaceus_europaeus/cdna/Erinaceus_europaeus.HEDGEHOG.cdna.abinitio.fa.gz',
	'erysiphe_necator_gca_000798715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/erysiphe_necator_gca_000798715/cdna/Erysiphe_necator_gca_000798715.ASM79871v1.cdna.all.fa.gz',
	'escovopsis_weberi_gca_001278495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/escovopsis_weberi_gca_001278495/cdna/Escovopsis_weberi_gca_001278495.ASM127849v1.cdna.all.fa.gz',
	'esox_lucius'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/esox_lucius/cdna/Esox_lucius.Eluc_V3.cdna.abinitio.fa.gz',
	'eutypa_lata_ucrel1_gca_000349385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/eutypa_lata_ucrel1_gca_000349385/cdna/Eutypa_lata_ucrel1_gca_000349385.UCREL1V03.cdna.all.fa.gz',
	'exidia_glandulosa_hhb12029_gca_001632375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/exidia_glandulosa_hhb12029_gca_001632375/cdna/Exidia_glandulosa_hhb12029_gca_001632375.Exigl1.cdna.all.fa.gz',
	'exophiala_aquamarina_cbs_119918_gca_000709125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/exophiala_aquamarina_cbs_119918_gca_000709125/cdna/Exophiala_aquamarina_cbs_119918_gca_000709125.Exop_aqua_CBS_119918_V1.cdna.all.fa.gz',
	'exophiala_dermatitidis_nih_ut8656_gca_000230625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/exophiala_dermatitidis_nih_ut8656_gca_000230625/cdna/Exophiala_dermatitidis_nih_ut8656_gca_000230625.Exop_derm_V1.cdna.all.fa.gz',
	'exophiala_mesophila_gca_000836275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_mesophila_gca_000836275/cdna/Exophiala_mesophila_gca_000836275.Exop_meso_CBS40295_V1.cdna.all.fa.gz',
	'exophiala_oligosperma_gca_000835515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_oligosperma_gca_000835515/cdna/Exophiala_oligosperma_gca_000835515.Exop_olig_CBS72588_V1.cdna.all.fa.gz',
	'exophiala_sideris_gca_000835395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_sideris_gca_000835395/cdna/Exophiala_sideris_gca_000835395.Exop_side_CBS121828_V1.cdna.all.fa.gz',
	'exophiala_spinifera_gca_000836115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_spinifera_gca_000836115/cdna/Exophiala_spinifera_gca_000836115.Exop_spin_CBS89968_V1.cdna.all.fa.gz',
	'exophiala_xenobiotica_gca_000835505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/exophiala_xenobiotica_gca_000835505/cdna/Exophiala_xenobiotica_gca_000835505.Exop_xeno_CBS118157_V1.cdna.all.fa.gz',
	'exserohilum_turcica_et28a_gca_000359705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/exserohilum_turcica_et28a_gca_000359705/cdna/Exserohilum_turcica_et28a_gca_000359705.Setospaeria_trucica_Et28A_v1.0.cdna.all.fa.gz',
	'felis_catus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/felis_catus/cdna/Felis_catus.Felis_catus_9.0.cdna.abinitio.fa.gz',
	'fibroporia_radiculosa_gca_000313525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fibroporia_radiculosa_gca_000313525/cdna/Fibroporia_radiculosa_gca_000313525.ASM31352v1.cdna.all.fa.gz',
	'fibularhizoctonia_sp_cbs_109695_gca_001630335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fibularhizoctonia_sp_cbs_109695_gca_001630335/cdna/Fibularhizoctonia_sp_cbs_109695_gca_001630335.Fibsp1.cdna.all.fa.gz',
	'ficedula_albicollis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ficedula_albicollis/cdna/Ficedula_albicollis.FicAlb_1.4.cdna.abinitio.fa.gz',
	'fistulina_hepatica_atcc_64428_gca_000934395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fistulina_hepatica_atcc_64428_gca_000934395/cdna/Fistulina_hepatica_atcc_64428_gca_000934395.Fishe1.cdna.all.fa.gz',
	'fomitiporia_mediterranea_mf3_22_gca_000271605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fomitiporia_mediterranea_mf3_22_gca_000271605/cdna/Fomitiporia_mediterranea_mf3_22_gca_000271605.Fomme1.cdna.all.fa.gz',
	'fomitopsis_pinicola_fp_58527_ss1_gca_000344655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/fomitopsis_pinicola_fp_58527_ss1_gca_000344655/cdna/Fomitopsis_pinicola_fp_58527_ss1_gca_000344655.Fompi3.cdna.all.fa.gz',
	'fonsecaea_erecta_gca_001651985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_erecta_gca_001651985/cdna/Fonsecaea_erecta_gca_001651985.ASM165198v1.cdna.all.fa.gz',
	'fonsecaea_monophora_gca_001642475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_monophora_gca_001642475/cdna/Fonsecaea_monophora_gca_001642475.ASM164247v1.cdna.all.fa.gz',
	'fonsecaea_multimorphosa_cbs_102226_gca_000836435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fonsecaea_multimorphosa_cbs_102226_gca_000836435/cdna/Fonsecaea_multimorphosa_cbs_102226_gca_000836435.Fons_mult_CBS_102226_V1.cdna.all.fa.gz',
	'fonsecaea_multimorphosa_gca_001646985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_multimorphosa_gca_001646985/cdna/Fonsecaea_multimorphosa_gca_001646985.ASM164698v1.cdna.all.fa.gz',
	'fonsecaea_nubica_gca_001646965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fonsecaea_nubica_gca_001646965/cdna/Fonsecaea_nubica_gca_001646965.ASM164696v1.cdna.all.fa.gz',
	'fonsecaea_pedrosoi_cbs_271_37_gca_000835455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fonsecaea_pedrosoi_cbs_271_37_gca_000835455/cdna/Fonsecaea_pedrosoi_cbs_271_37_gca_000835455.Fons_pedr_CBS_271_37_V1.cdna.all.fa.gz',
	'fonticula_alba_gca_000388065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_nucleariidaeandfonticulagroup1_collection/fonticula_alba_gca_000388065/cdna/Fonticula_alba_gca_000388065.Font_alba_ATCC_38817_V2.cdna.all.fa.gz',
	'fukomys_damarensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/fukomys_damarensis/cdna/Fukomys_damarensis.DMR_v1.0.cdna.abinitio.fa.gz',
	'fundulus_heteroclitus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/fundulus_heteroclitus/cdna/Fundulus_heteroclitus.Fundulus_heteroclitus-3.0.2.cdna.abinitio.fa.gz',
	'fusarium_culmorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_culmorum/cdna/Fusarium_culmorum.EF1.cdna.all.fa.gz',
	'fusarium_fujikuroi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_fujikuroi/cdna/Fusarium_fujikuroi.EF1.cdna.all.fa.gz',
	'fusarium_fujikuroi_gca_001023035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fusarium_fujikuroi_gca_001023035/cdna/Fusarium_fujikuroi_gca_001023035.ASM102303v1.cdna.all.fa.gz',
	'fusarium_fujikuroi_gca_001023045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_001023045/cdna/Fusarium_fujikuroi_gca_001023045.ASM102304v1.cdna.all.fa.gz',
	'fusarium_fujikuroi_gca_001023065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/fusarium_fujikuroi_gca_001023065/cdna/Fusarium_fujikuroi_gca_001023065.ASM102306v1.cdna.all.fa.gz',
	'fusarium_fujikuroi_gca_900096505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_900096505/cdna/Fusarium_fujikuroi_gca_900096505.version_1.cdna.all.fa.gz',
	'fusarium_fujikuroi_gca_900096685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_900096685/cdna/Fusarium_fujikuroi_gca_900096685.version_1.cdna.all.fa.gz',
	'fusarium_fujikuroi_gca_900096705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_fujikuroi_gca_900096705/cdna/Fusarium_fujikuroi_gca_900096705.version_1.cdna.all.fa.gz',
	'fusarium_graminearum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_graminearum/cdna/Fusarium_graminearum.RR1.cdna.all.fa.gz',
	'fusarium_graminearum_gca_000240135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_graminearum_gca_000240135/cdna/Fusarium_graminearum_gca_000240135.ASM24013v3.cdna.all.fa.gz',
	'fusarium_langsethiae_gca_001292635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_langsethiae_gca_001292635/cdna/Fusarium_langsethiae_gca_001292635.ASM129263v1.cdna.all.fa.gz',
	'fusarium_mangiferae_gca_900044065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_mangiferae_gca_900044065/cdna/Fusarium_mangiferae_gca_900044065.Genome_assembly_version_1.cdna.all.fa.gz',
	'fusarium_oxysporum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_oxysporum/cdna/Fusarium_oxysporum.FO2.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215/cdna/Fusarium_oxysporum_f_sp_conglutinans_race_2_54008_gca_000260215.FO_PHW808_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345/cdna/Fusarium_oxysporum_f_sp_cubense_race_1_gca_000350345.Foc1_1.0.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365/cdna/Fusarium_oxysporum_f_sp_cubense_race_4_gca_000350365.Foc4_1.0.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195/cdna/Fusarium_oxysporum_f_sp_cubense_tropical_race_4_54006_gca_000260195.FO_II5_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955/cdna/Fusarium_oxysporum_f_sp_lycopersici_4287_gca_000149955.ASM14995v2.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975/cdna/Fusarium_oxysporum_f_sp_lycopersici_mn25_gca_000259975.FO_MN25_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_melonis_26406_gca_000260495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_melonis_26406_gca_000260495/cdna/Fusarium_oxysporum_f_sp_melonis_26406_gca_000260495.FO_melonis_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075/cdna/Fusarium_oxysporum_f_sp_pisi_hdv247_gca_000260075.FO_HDV247_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155/cdna/Fusarium_oxysporum_f_sp_radicis_lycopersici_26381_gca_000260155.FO_CL57_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_raphani_54005_gca_000260235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_raphani_54005_gca_000260235/cdna/Fusarium_oxysporum_f_sp_raphani_54005_gca_000260235.FO_PHW815_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175/cdna/Fusarium_oxysporum_f_sp_vasinfectum_25433_gca_000260175.FO_Cotton_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_fo47_gca_000271705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_fo47_gca_000271705/cdna/Fusarium_oxysporum_fo47_gca_000271705.FO_Fo47_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_fosc_3_a_gca_000271745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/fusarium_oxysporum_fosc_3_a_gca_000271745/cdna/Fusarium_oxysporum_fosc_3_a_gca_000271745.FO_FOSC_3_a_V1.cdna.all.fa.gz',
	'fusarium_oxysporum_gca_900096695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_oxysporum_gca_900096695/cdna/Fusarium_oxysporum_gca_900096695.version_1.cdna.all.fa.gz',
	'fusarium_poae_gca_001675295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_poae_gca_001675295/cdna/Fusarium_poae_gca_001675295.FPOA1.0.cdna.all.fa.gz',
	'fusarium_proliferatum_et1_gca_900067095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_proliferatum_et1_gca_900067095/cdna/Fusarium_proliferatum_et1_gca_900067095.F._proliferatum_ET1_version_1.cdna.all.fa.gz',
	'fusarium_proliferatum_gca_900029915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/fusarium_proliferatum_gca_900029915/cdna/Fusarium_proliferatum_gca_900029915.F._proliferatum_NRRL62905_version_1.cdna.all.fa.gz',
	'fusarium_pseudograminearum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_pseudograminearum/cdna/Fusarium_pseudograminearum.GCA_000303195.1.cdna.all.fa.gz',
	'fusarium_solani'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_solani/cdna/Fusarium_solani.v2.0.cdna.all.fa.gz',
	'fusarium_venenatum_gca_900007375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/fusarium_venenatum_gca_900007375/cdna/Fusarium_venenatum_gca_900007375.ASM90000737v1.cdna.all.fa.gz',
	'fusarium_verticillioides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fusarium_verticillioides/cdna/Fusarium_verticillioides.ASM14955v1.cdna.all.fa.gz',
	'gadus_morhua'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gadus_morhua/cdna/Gadus_morhua.gadMor1.cdna.abinitio.fa.gz',
	'gaeumannomyces_graminis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/gaeumannomyces_graminis/cdna/Gaeumannomyces_graminis.Gae_graminis_V2.cdna.all.fa.gz',
	'galdieria_sulphuraria'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/galdieria_sulphuraria/cdna/Galdieria_sulphuraria.ASM34128v1.cdna.all.fa.gz',
	'galerina_marginata_cbs_339_88_gca_000697645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/galerina_marginata_cbs_339_88_gca_000697645/cdna/Galerina_marginata_cbs_339_88_gca_000697645.Galma1.cdna.all.fa.gz',
	'gallus_gallus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gallus_gallus/cdna/Gallus_gallus.GRCg6a.cdna.abinitio.fa.gz',
	'gambusia_affinis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gambusia_affinis/cdna/Gambusia_affinis.ASM309773v1.cdna.abinitio.fa.gz',
	'gasterosteus_aculeatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gasterosteus_aculeatus/cdna/Gasterosteus_aculeatus.BROADS1.cdna.abinitio.fa.gz',
	'gelatoporia_subvermispora_b_gca_000320605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/gelatoporia_subvermispora_b_gca_000320605/cdna/Gelatoporia_subvermispora_b_gca_000320605.Ceriporiopsis_subvermispora_B_v1.0.cdna.all.fa.gz',
	'giardia_intestinalis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis/cdna/Giardia_intestinalis.ASM49873v1.cdna.all.fa.gz',
	'giardia_intestinalis_assemblage_b'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis_assemblage_b/cdna/Giardia_intestinalis_assemblage_b.G_duodenalis_AssB_BAH15c1.cdna.all.fa.gz',
	'giardia_intestinalis_atcc_50581'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis_atcc_50581/cdna/Giardia_intestinalis_atcc_50581.ASM18240v1.cdna.all.fa.gz',
	'giardia_intestinalis_gca_000498715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_intestinalis_gca_000498715/cdna/Giardia_intestinalis_gca_000498715.ASM49871v1.cdna.all.fa.gz',
	'giardia_lamblia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/giardia_lamblia/cdna/Giardia_lamblia.GL2.cdna.all.fa.gz',
	'giardia_lamblia_p15'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/giardia_lamblia_p15/cdna/Giardia_lamblia_p15.ASM18266v1.cdna.all.fa.gz',
	'glarea_lozoyensis_atcc_20868_gca_000409485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/glarea_lozoyensis_atcc_20868_gca_000409485/cdna/Glarea_lozoyensis_atcc_20868_gca_000409485.GLAREA.cdna.all.fa.gz',
	'gloeophyllum_trabeum_atcc_11539_gca_000344685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/gloeophyllum_trabeum_atcc_11539_gca_000344685/cdna/Gloeophyllum_trabeum_atcc_11539_gca_000344685.Glotr1_1.cdna.all.fa.gz',
	'glycine_max'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/glycine_max/cdna/Glycine_max.Glycine_max_v2.1.cdna.all.fa.gz',
	'gonapodya_prolifera_jel478_gca_001574975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/gonapodya_prolifera_jel478_gca_001574975/cdna/Gonapodya_prolifera_jel478_gca_001574975.Ganpr1.cdna.all.fa.gz',
	'gopherus_agassizii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gopherus_agassizii/cdna/Gopherus_agassizii.ASM289641v1.cdna.abinitio.fa.gz',
	'gorilla_gorilla'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/gorilla_gorilla/cdna/Gorilla_gorilla.gorGor4.cdna.abinitio.fa.gz',
	'gossypium_raimondii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/gossypium_raimondii/cdna/Gossypium_raimondii.Graimondii2_0.cdna.all.fa.gz',
	'gregarina_niphandrodes'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/gregarina_niphandrodes/cdna/Gregarina_niphandrodes.GNI3.cdna.all.fa.gz',
	'grifola_frondosa_gca_001683735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/grifola_frondosa_gca_001683735/cdna/Grifola_frondosa_gca_001683735.ASM168373v1.cdna.all.fa.gz',
	'grosmannia_clavigera_kw1407_gca_000143105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/grosmannia_clavigera_kw1407_gca_000143105/cdna/Grosmannia_clavigera_kw1407_gca_000143105.Sanger-454-IlluminaPA_2.0.cdna.all.fa.gz',
	'guillardia_theta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/guillardia_theta/cdna/Guillardia_theta.Guith1.cdna.all.fa.gz',
	'guillardia_theta_gca_000002975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_cryptophyta1_collection/guillardia_theta_gca_000002975/cdna/Guillardia_theta_gca_000002975.ASM297v1.cdna.all.fa.gz',
	'gymnopus_luxurians_fd_317_m1_gca_000827265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/gymnopus_luxurians_fd_317_m1_gca_000827265/cdna/Gymnopus_luxurians_fd_317_m1_gca_000827265.Gymlu1.cdna.all.fa.gz',
	'hammondia_hammondi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/hammondia_hammondi/cdna/Hammondia_hammondi.HHA1_v02.cdna.all.fa.gz',
	'hanseniaspora_guilliermondii_gca_900119595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_guilliermondii_gca_900119595/cdna/Hanseniaspora_guilliermondii_gca_900119595.version_1.cdna.all.fa.gz',
	'hanseniaspora_opuntiae_gca_001749795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_opuntiae_gca_001749795/cdna/Hanseniaspora_opuntiae_gca_001749795.ASM174979v1.cdna.all.fa.gz',
	'hanseniaspora_osmophila_gca_001747045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_osmophila_gca_001747045/cdna/Hanseniaspora_osmophila_gca_001747045.ASM174704v1.cdna.all.fa.gz',
	'hanseniaspora_uvarum_dsm_2768_gca_000968475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/hanseniaspora_uvarum_dsm_2768_gca_000968475/cdna/Hanseniaspora_uvarum_dsm_2768_gca_000968475.Huva2.0.cdna.all.fa.gz',
	'hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025/cdna/Hanseniaspora_valbyensis_nrrl_y_1626_gca_001664025.Hanva1_1.cdna.all.fa.gz',
	'haplochromis_burtoni'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/haplochromis_burtoni/cdna/Haplochromis_burtoni.AstBur1.0.cdna.abinitio.fa.gz',
	'hebeloma_cylindrosporum_h7_gca_000827355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hebeloma_cylindrosporum_h7_gca_000827355/cdna/Hebeloma_cylindrosporum_h7_gca_000827355.Hebeloma_cylindrosporum_h7_v2.0.cdna.all.fa.gz',
	'helianthus_annuus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/helianthus_annuus/cdna/Helianthus_annuus.HanXRQr1.0.cdna.all.fa.gz',
	'heliconius_melpomene'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/heliconius_melpomene/cdna/Heliconius_melpomene.Hmel1.cdna.all.fa.gz',
	'helobdella_robusta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/helobdella_robusta/cdna/Helobdella_robusta.Helro1.cdna.all.fa.gz',
	'hesseltinella_vesiculosa_gca_002104935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/hesseltinella_vesiculosa_gca_002104935/cdna/Hesseltinella_vesiculosa_gca_002104935.Hesve2.cdna.all.fa.gz',
	'heterobasidion_irregulare_tc_32_1_gca_000320585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/heterobasidion_irregulare_tc_32_1_gca_000320585/cdna/Heterobasidion_irregulare_tc_32_1_gca_000320585.Heterobasidion_irregulare_v2.0.cdna.all.fa.gz',
	'heterocephalus_glaber_female'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/heterocephalus_glaber_female/cdna/Heterocephalus_glaber_female.HetGla_female_1.0.cdna.abinitio.fa.gz',
	'heterocephalus_glaber_male'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/heterocephalus_glaber_male/cdna/Heterocephalus_glaber_male.HetGla_1.0.cdna.abinitio.fa.gz',
	'hippocampus_comes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/hippocampus_comes/cdna/Hippocampus_comes.H_comes_QL1_v1.cdna.abinitio.fa.gz',
	'hirsutella_minnesotensis_3608_gca_000956045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/hirsutella_minnesotensis_3608_gca_000956045/cdna/Hirsutella_minnesotensis_3608_gca_000956045.ASM95604v1.cdna.all.fa.gz',
	'histoplasma_capsulatum_g186ar_gca_000150115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_g186ar_gca_000150115/cdna/Histoplasma_capsulatum_g186ar_gca_000150115.ASM15011v1.cdna.all.fa.gz',
	'histoplasma_capsulatum_h143_gca_000151035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_h143_gca_000151035/cdna/Histoplasma_capsulatum_h143_gca_000151035.ASM15103v1.cdna.all.fa.gz',
	'histoplasma_capsulatum_h88_gca_000151005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_h88_gca_000151005/cdna/Histoplasma_capsulatum_h88_gca_000151005.ASM15100v2.cdna.all.fa.gz',
	'histoplasma_capsulatum_nam1_gca_000149585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/histoplasma_capsulatum_nam1_gca_000149585/cdna/Histoplasma_capsulatum_nam1_gca_000149585.ASM14958v1.cdna.all.fa.gz',
	'homo_sapiens'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.abinitio.fa.gz',
	'hordeum_vulgare'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/hordeum_vulgare/cdna/Hordeum_vulgare.IBSC_v2.cdna.all.fa.gz',
	'hortaea_werneckii_exf_2000_gca_002127715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hortaea_werneckii_exf_2000_gca_002127715/cdna/Hortaea_werneckii_exf_2000_gca_002127715.HwerPB1.cdna.all.fa.gz',
	'hyaloperonospora_arabidopsidis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/hyaloperonospora_arabidopsidis/cdna/Hyaloperonospora_arabidopsidis.HyaAraEmoy2_2.0.cdna.all.fa.gz',
	'hydnomerulius_pinastri_md_312_gca_000827185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hydnomerulius_pinastri_md_312_gca_000827185/cdna/Hydnomerulius_pinastri_md_312_gca_000827185.Hydpi2.cdna.all.fa.gz',
	'hypholoma_sublateritium_fd_334_ss_4_gca_000827495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hypholoma_sublateritium_fd_334_ss_4_gca_000827495/cdna/Hypholoma_sublateritium_fd_334_ss_4_gca_000827495.Hypsu1.cdna.all.fa.gz',
	'hyphopichia_burtonii_nrrl_y_1933_gca_001661395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hyphopichia_burtonii_nrrl_y_1933_gca_001661395/cdna/Hyphopichia_burtonii_nrrl_y_1933_gca_001661395.Hypbu1.cdna.all.fa.gz',
	'hypoxylon_sp_ci_4a_gca_002120315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hypoxylon_sp_ci_4a_gca_002120315/cdna/Hypoxylon_sp_ci_4a_gca_002120315.HypCI4A_v1.0.cdna.all.fa.gz',
	'hypoxylon_sp_co27_5_gca_002120305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/hypoxylon_sp_co27_5_gca_002120305/cdna/Hypoxylon_sp_co27_5_gca_002120305.HypCO275_v1.0.cdna.all.fa.gz',
	'hypoxylon_sp_ec38_gca_002120335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/hypoxylon_sp_ec38_gca_002120335/cdna/Hypoxylon_sp_ec38_gca_002120335.HypEC38_v3.0.cdna.all.fa.gz',
	'hypsizygus_marmoreus_gca_001605315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/hypsizygus_marmoreus_gca_001605315/cdna/Hypsizygus_marmoreus_gca_001605315.ASM160531v1.cdna.all.fa.gz',
	'ichthyophthirius_multifiliis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/ichthyophthirius_multifiliis/cdna/Ichthyophthirius_multifiliis.JCVI-IMG1-V.1.cdna.all.fa.gz',
	'ictalurus_punctatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ictalurus_punctatus/cdna/Ictalurus_punctatus.IpCoco_1.2.cdna.abinitio.fa.gz',
	'ictidomys_tridecemlineatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ictidomys_tridecemlineatus/cdna/Ictidomys_tridecemlineatus.SpeTri2.0.cdna.abinitio.fa.gz',
	'ixodes_scapularis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/ixodes_scapularis/cdna/Ixodes_scapularis.IscaW1.cdna.all.fa.gz',
	'jaapia_argillacea_mucl_33604_gca_000697665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/jaapia_argillacea_mucl_33604_gca_000697665/cdna/Jaapia_argillacea_mucl_33604_gca_000697665.Jaaar1.cdna.all.fa.gz',
	'jaculus_jaculus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/jaculus_jaculus/cdna/Jaculus_jaculus.JacJac1.0.cdna.abinitio.fa.gz',
	'jaminaea_rosea_gca_003144245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/jaminaea_rosea_gca_003144245/cdna/Jaminaea_rosea_gca_003144245.Jamsp1.cdna.all.fa.gz',
	'junco_hyemalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/junco_hyemalis/cdna/Junco_hyemalis.ASM382977v1.cdna.abinitio.fa.gz',
	'kalmanozyma_brasiliensis_ghg001_gca_000497045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kalmanozyma_brasiliensis_ghg001_gca_000497045/cdna/Kalmanozyma_brasiliensis_ghg001_gca_000497045.PSEUBRA1.cdna.all.fa.gz',
	'kazachstania_africana_cbs_2517_gca_000304475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kazachstania_africana_cbs_2517_gca_000304475/cdna/Kazachstania_africana_cbs_2517_gca_000304475.Ka_CBS2517.cdna.all.fa.gz',
	'kazachstania_naganishii_cbs_8797_gca_000348985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kazachstania_naganishii_cbs_8797_gca_000348985/cdna/Kazachstania_naganishii_cbs_8797_gca_000348985.ASM34898v1.cdna.all.fa.gz',
	'kluyveromyces_dobzhanskii_cbs_2104_gca_000820885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/kluyveromyces_dobzhanskii_cbs_2104_gca_000820885/cdna/Kluyveromyces_dobzhanskii_cbs_2104_gca_000820885.KLDO_01.cdna.all.fa.gz',
	'kluyveromyces_lactis_gca_000002515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kluyveromyces_lactis_gca_000002515/cdna/Kluyveromyces_lactis_gca_000002515.ASM251v1.cdna.all.fa.gz',
	'kluyveromyces_marxianus_dmku3_1042_gca_001417885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/kluyveromyces_marxianus_dmku3_1042_gca_001417885/cdna/Kluyveromyces_marxianus_dmku3_1042_gca_001417885.Kmar_1.0.cdna.all.fa.gz',
	'kluyveromyces_marxianus_gca_001417835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/kluyveromyces_marxianus_gca_001417835/cdna/Kluyveromyces_marxianus_gca_001417835.KM1777_03.cdna.all.fa.gz',
	'kockovaella_imperatae_gca_002102565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kockovaella_imperatae_gca_002102565/cdna/Kockovaella_imperatae_gca_002102565.Kocim1.cdna.all.fa.gz',
	'komagataella_pastoris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/komagataella_pastoris/cdna/Komagataella_pastoris.GCA_000027005.1.cdna.all.fa.gz',
	'komagataella_pastoris_gca_001708105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/komagataella_pastoris_gca_001708105/cdna/Komagataella_pastoris_gca_001708105.ASM170810v1.cdna.all.fa.gz',
	'komagataella_phaffii_cbs_7435_gca_000223565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/komagataella_phaffii_cbs_7435_gca_000223565/cdna/Komagataella_phaffii_cbs_7435_gca_000223565.PicPas_Mar2011.cdna.all.fa.gz',
	'komagataella_phaffii_gca_001708085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/komagataella_phaffii_gca_001708085/cdna/Komagataella_phaffii_gca_001708085.ASM170808v1.cdna.all.fa.gz',
	'komagataella_phaffii_gs115_gca_001746955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/komagataella_phaffii_gs115_gca_001746955/cdna/Komagataella_phaffii_gs115_gca_001746955.ASM174695v1.cdna.all.fa.gz',
	'kryptolebias_marmoratus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/kryptolebias_marmoratus/cdna/Kryptolebias_marmoratus.ASM164957v1.cdna.abinitio.fa.gz',
	'kuraishia_capsulata_cbs_1993_gca_000576695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/kuraishia_capsulata_cbs_1993_gca_000576695/cdna/Kuraishia_capsulata_cbs_1993_gca_000576695.AUH_PRJEB4427_v1.cdna.all.fa.gz',
	'kwoniella_bestiolae_cbs_10118_gca_000512585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_bestiolae_cbs_10118_gca_000512585/cdna/Kwoniella_bestiolae_cbs_10118_gca_000512585.Cryp_best_CBS10118_V1.cdna.all.fa.gz',
	'kwoniella_dejecticola_cbs_10117_gca_000512565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_dejecticola_cbs_10117_gca_000512565/cdna/Kwoniella_dejecticola_cbs_10117_gca_000512565.Cryp_deje_CBS10117_V1.cdna.all.fa.gz',
	'kwoniella_heveanensis_bcc8398_gca_000507405'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_heveanensis_bcc8398_gca_000507405/cdna/Kwoniella_heveanensis_bcc8398_gca_000507405.Cryp_heve_BCC8398_V1.cdna.all.fa.gz',
	'kwoniella_heveanensis_cbs_569_gca_000507425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_heveanensis_cbs_569_gca_000507425/cdna/Kwoniella_heveanensis_cbs_569_gca_000507425.Cryp_heve_CBS569_V2.cdna.all.fa.gz',
	'kwoniella_mangroviensis_cbs_10435_gca_000507885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_mangroviensis_cbs_10435_gca_000507885/cdna/Kwoniella_mangroviensis_cbs_10435_gca_000507885.Kwon_mang_CBS10435_V2.cdna.all.fa.gz',
	'kwoniella_mangroviensis_cbs_8507_gca_000507465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_mangroviensis_cbs_8507_gca_000507465/cdna/Kwoniella_mangroviensis_cbs_8507_gca_000507465.Kwon_mang_CBS8507_V2.cdna.all.fa.gz',
	'kwoniella_mangroviensis_cbs_8886_gca_000507485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_mangroviensis_cbs_8886_gca_000507485/cdna/Kwoniella_mangroviensis_cbs_8886_gca_000507485.Kwon_mang_CBS8886_V1.cdna.all.fa.gz',
	'kwoniella_pini_cbs_10737_gca_000512605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/kwoniella_pini_cbs_10737_gca_000512605/cdna/Kwoniella_pini_cbs_10737_gca_000512605.Cryp_pinu_CBS10737_V1.cdna.all.fa.gz',
	'labrus_bergylta'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/labrus_bergylta/cdna/Labrus_bergylta.BallGen_V1.cdna.abinitio.fa.gz',
	'laccaria_amethystina_laam_08_1_gca_000827195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/laccaria_amethystina_laam_08_1_gca_000827195/cdna/Laccaria_amethystina_laam_08_1_gca_000827195.Laccaria_amethystina_LaAM-08-1_v1.0.cdna.all.fa.gz',
	'laccaria_bicolor_s238n_h82_gca_000143565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/laccaria_bicolor_s238n_h82_gca_000143565/cdna/Laccaria_bicolor_s238n_h82_gca_000143565.V1.0.cdna.all.fa.gz',
	'lachancea_dasiensis_cbs_10888_gca_900074725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_dasiensis_cbs_10888_gca_900074725/cdna/Lachancea_dasiensis_cbs_10888_gca_900074725.LADA0.cdna.all.fa.gz',
	'lachancea_fermentati_gca_900074765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_fermentati_gca_900074765/cdna/Lachancea_fermentati_gca_900074765.LAFE0.cdna.all.fa.gz',
	'lachancea_lanzarotensis_gca_000938715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/lachancea_lanzarotensis_gca_000938715/cdna/Lachancea_lanzarotensis_gca_000938715.LALA0.cdna.all.fa.gz',
	'lachancea_meyersii_cbs_8951_gca_900074715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_meyersii_cbs_8951_gca_900074715/cdna/Lachancea_meyersii_cbs_8951_gca_900074715.LAME0.cdna.all.fa.gz',
	'lachancea_mirantina_gca_900074745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_mirantina_gca_900074745/cdna/Lachancea_mirantina_gca_900074745.LAMI0.cdna.all.fa.gz',
	'lachancea_nothofagi_cbs_11611_gca_900074755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_nothofagi_cbs_11611_gca_900074755/cdna/Lachancea_nothofagi_cbs_11611_gca_900074755.LANO0.cdna.all.fa.gz',
	'lachancea_quebecensis_gca_002900925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/lachancea_quebecensis_gca_002900925/cdna/Lachancea_quebecensis_gca_002900925.ASM290092v1.cdna.all.fa.gz',
	'lachancea_sp_cbs_6924_gca_900074735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lachancea_sp_cbs_6924_gca_900074735/cdna/Lachancea_sp_cbs_6924_gca_900074735.LAFA0.cdna.all.fa.gz',
	'lachancea_thermotolerans_cbs_6340_gca_000142805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/lachancea_thermotolerans_cbs_6340_gca_000142805/cdna/Lachancea_thermotolerans_cbs_6340_gca_000142805.ASM14280v1.cdna.all.fa.gz',
	'laetiporus_sulphureus_93_53_gca_001632365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/laetiporus_sulphureus_93_53_gca_001632365/cdna/Laetiporus_sulphureus_93_53_gca_001632365.Laesu1.cdna.all.fa.gz',
	'latimeria_chalumnae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/latimeria_chalumnae/cdna/Latimeria_chalumnae.LatCha1.cdna.abinitio.fa.gz',
	'leersia_perrieri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/leersia_perrieri/cdna/Leersia_perrieri.Lperr_V1.4.cdna.abinitio.fa.gz',
	'leishmania_braziliensis_mhom_br_75_m2904'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_braziliensis_mhom_br_75_m2904/cdna/Leishmania_braziliensis_mhom_br_75_m2904.ASM284v2.cdna.all.fa.gz',
	'leishmania_donovani_bpk282a1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_donovani_bpk282a1/cdna/Leishmania_donovani_bpk282a1.ASM22713v2.cdna.all.fa.gz',
	'leishmania_infantum_jpcm5'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_infantum_jpcm5/cdna/Leishmania_infantum_jpcm5.ASM287v2.cdna.all.fa.gz',
	'leishmania_major'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/leishmania_major/cdna/Leishmania_major.ASM272v2.cdna.all.fa.gz',
	'leishmania_mexicana_mhom_gt_2001_u1103'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_mexicana_mhom_gt_2001_u1103/cdna/Leishmania_mexicana_mhom_gt_2001_u1103.ASM23466v4.cdna.all.fa.gz',
	'leishmania_panamensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leishmania_panamensis/cdna/Leishmania_panamensis.ASM75516v1.cdna.all.fa.gz',
	'lepeophtheirus_salmonis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lepeophtheirus_salmonis/cdna/Lepeophtheirus_salmonis.LSalAtl2s.cdna.all.fa.gz',
	'lepidothrix_coronata'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/lepidothrix_coronata/cdna/Lepidothrix_coronata.Lepidothrix_coronata-1.0.cdna.abinitio.fa.gz',
	'lepisosteus_oculatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/lepisosteus_oculatus/cdna/Lepisosteus_oculatus.LepOcu1.cdna.abinitio.fa.gz',
	'leptomonas_pyrrhocoris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leptomonas_pyrrhocoris/cdna/Leptomonas_pyrrhocoris.ASM129339v1.cdna.all.fa.gz',
	'leptomonas_seymouri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/leptomonas_seymouri/cdna/Leptomonas_seymouri.ASM129953v1.cdna.all.fa.gz',
	'leptosphaeria_maculans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/leptosphaeria_maculans/cdna/Leptosphaeria_maculans.ASM23037v1.cdna.all.fa.gz',
	'leucoagaricus_sp_symc_cos_gca_001563735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/leucoagaricus_sp_symc_cos_gca_001563735/cdna/Leucoagaricus_sp_symc_cos_gca_001563735.ASM156373v1.cdna.all.fa.gz',
	'leucosporidium_creatinivorum_gca_002105055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/leucosporidium_creatinivorum_gca_002105055/cdna/Leucosporidium_creatinivorum_gca_002105055.Leucr1.cdna.all.fa.gz',
	'lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665/cdna/Lichtheimia_corymbifera_jmrc_fsu_9682_gca_000723665.454Minimus.cdna.all.fa.gz',
	'lingula_anatina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lingula_anatina/cdna/Lingula_anatina.LinAna1.0.cdna.all.fa.gz',
	'lipomyces_starkeyi_nrrl_y_11557_gca_001661325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/lipomyces_starkeyi_nrrl_y_11557_gca_001661325/cdna/Lipomyces_starkeyi_nrrl_y_11557_gca_001661325.Lipst1_1.cdna.all.fa.gz',
	'loa_loa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/loa_loa/cdna/Loa_loa.Loa_loa_V3.cdna.all.fa.gz',
	'lobosporangium_transversale_gca_002105155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/lobosporangium_transversale_gca_002105155/cdna/Lobosporangium_transversale_gca_002105155.Lobtra1.cdna.all.fa.gz',
	'lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685/cdna/Lodderomyces_elongisporus_nrrl_yb_4239_gca_000149685.ASM14968v1.cdna.all.fa.gz',
	'lomentospora_prolificans_gca_002276285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/lomentospora_prolificans_gca_002276285/cdna/Lomentospora_prolificans_gca_002276285.Lprolificans_pilon.cdna.all.fa.gz',
	'lonchura_striata_domestica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/lonchura_striata_domestica/cdna/Lonchura_striata_domestica.LonStrDom1.cdna.abinitio.fa.gz',
	'lotharella_oceanica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/lotharella_oceanica/cdna/Lotharella_oceanica.ASM69843v2.cdna.all.fa.gz',
	'lottia_gigantea'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lottia_gigantea/cdna/Lottia_gigantea.Lotgi1.cdna.all.fa.gz',
	'loxodonta_africana'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/loxodonta_africana/cdna/Loxodonta_africana.loxAfr3.cdna.abinitio.fa.gz',
	'lucilia_cuprina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/lucilia_cuprina/cdna/Lucilia_cuprina.ASM118794v1.cdna.all.fa.gz',
	'lupinus_angustifolius'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/lupinus_angustifolius/cdna/Lupinus_angustifolius.LupAngTanjil_v1.0.cdna.all.fa.gz',
	'macaca_fascicularis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/macaca_fascicularis/cdna/Macaca_fascicularis.Macaca_fascicularis_5.0.cdna.abinitio.fa.gz',
	'macaca_mulatta'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/macaca_mulatta/cdna/Macaca_mulatta.Mmul_8.0.1.cdna.abinitio.fa.gz',
	'macaca_nemestrina'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/macaca_nemestrina/cdna/Macaca_nemestrina.Mnem_1.0.cdna.abinitio.fa.gz',
	'macrophomina_phaseolina_ms6_gca_000302655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/macrophomina_phaseolina_ms6_gca_000302655/cdna/Macrophomina_phaseolina_ms6_gca_000302655.MphMS6_1.0.cdna.all.fa.gz',
	'madurella_mycetomatis_gca_001275765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/madurella_mycetomatis_gca_001275765/cdna/Madurella_mycetomatis_gca_001275765.ASM127576v2.cdna.all.fa.gz',
	'magnaporthe_oryzae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/magnaporthe_oryzae/cdna/Magnaporthe_oryzae.MG8.cdna.all.fa.gz',
	'magnaporthe_poae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/magnaporthe_poae/cdna/Magnaporthe_poae.Mag_poae_ATCC_64411_V1.cdna.all.fa.gz',
	'malassezia_globosa_cbs_7966_gca_000181695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_globosa_cbs_7966_gca_000181695/cdna/Malassezia_globosa_cbs_7966_gca_000181695.ASM18169v1.cdna.all.fa.gz',
	'malassezia_pachydermatis_gca_001278385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_pachydermatis_gca_001278385/cdna/Malassezia_pachydermatis_gca_001278385.MalaPachy.cdna.all.fa.gz',
	'malassezia_sympodialis_atcc_42132_gca_000349305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_sympodialis_atcc_42132_gca_000349305/cdna/Malassezia_sympodialis_atcc_42132_gca_000349305.ASM34930v2.cdna.all.fa.gz',
	'malassezia_sympodialis_atcc_42132_gca_900149145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_sympodialis_atcc_42132_gca_900149145/cdna/Malassezia_sympodialis_atcc_42132_gca_900149145.Msy_ATCC_42132_Feb2016.cdna.all.fa.gz',
	'malassezia_vespertilionis_gca_002818225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/malassezia_vespertilionis_gca_002818225/cdna/Malassezia_vespertilionis_gca_002818225.ASM281822v1.cdna.all.fa.gz',
	'manacus_vitellinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/manacus_vitellinus/cdna/Manacus_vitellinus.ASM171598v2.cdna.abinitio.fa.gz',
	'mandrillus_leucophaeus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mandrillus_leucophaeus/cdna/Mandrillus_leucophaeus.Mleu.le_1.0.cdna.abinitio.fa.gz',
	'manihot_esculenta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/manihot_esculenta/cdna/Manihot_esculenta.Manihot_esculenta_v6.cdna.all.fa.gz',
	'marmota_marmota_marmota'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/marmota_marmota_marmota/cdna/Marmota_marmota_marmota.marMar2.1.cdna.abinitio.fa.gz',
	'marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775/cdna/Marssonina_brunnea_f_sp_multigermtubi_mb_m1_gca_000298775.ASM29877v1.cdna.all.fa.gz',
	'mastacembelus_armatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mastacembelus_armatus/cdna/Mastacembelus_armatus.fMasArm1.1.cdna.abinitio.fa.gz',
	'mayetiola_destructor'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/mayetiola_destructor/cdna/Mayetiola_destructor.Mdes_1.0.cdna.all.fa.gz',
	'maylandia_zebra'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/maylandia_zebra/cdna/Maylandia_zebra.M_zebra_UMD2a.cdna.abinitio.fa.gz',
	'medicago_truncatula'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/medicago_truncatula/cdna/Medicago_truncatula.MedtrA17_4.0.cdna.all.fa.gz',
	'megaselia_scalaris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/megaselia_scalaris/cdna/Megaselia_scalaris.Msca1.cdna.all.fa.gz',
	'meira_miltonrushii_gca_003144205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/meira_miltonrushii_gca_003144205/cdna/Meira_miltonrushii_gca_003144205.Meimi1.cdna.all.fa.gz',
	'melampsora_laricipopulina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/melampsora_laricipopulina/cdna/Melampsora_laricipopulina.GCA_000204055.1.cdna.all.fa.gz',
	'meleagris_gallopavo'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/meleagris_gallopavo/cdna/Meleagris_gallopavo.Turkey_2.01.cdna.abinitio.fa.gz',
	'meliniomyces_bicolor_e_gca_002865645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/meliniomyces_bicolor_e_gca_002865645/cdna/Meliniomyces_bicolor_e_gca_002865645.Melbi2.cdna.all.fa.gz',
	'meliniomyces_variabilis_f_gca_002865655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/meliniomyces_variabilis_f_gca_002865655/cdna/Meliniomyces_variabilis_f_gca_002865655.Melva1.cdna.all.fa.gz',
	'melitaea_cinxia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/melitaea_cinxia/cdna/Melitaea_cinxia.MelCinx1.0.cdna.abinitio.fa.gz',
	'melopsittacus_undulatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/melopsittacus_undulatus/cdna/Melopsittacus_undulatus.Melopsittacus_undulatus_6.3.cdna.abinitio.fa.gz',
	'meriones_unguiculatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/meriones_unguiculatus/cdna/Meriones_unguiculatus.MunDraft-v1.0.cdna.abinitio.fa.gz',
	'mesocricetus_auratus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mesocricetus_auratus/cdna/Mesocricetus_auratus.MesAur1.0.cdna.abinitio.fa.gz',
	'metarhizium_acridum_cqma_102_gca_000187405'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_acridum_cqma_102_gca_000187405/cdna/Metarhizium_acridum_cqma_102_gca_000187405.MetAcr_May2010.cdna.all.fa.gz',
	'metarhizium_album_arsef_1941_gca_000804445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/metarhizium_album_arsef_1941_gca_000804445/cdna/Metarhizium_album_arsef_1941_gca_000804445.MAM_1.0_for_version_1_of_the_Metarhizium_album_genome.cdna.all.fa.gz',
	'metarhizium_anisopliae_arsef_23_gca_000187425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_anisopliae_arsef_23_gca_000187425/cdna/Metarhizium_anisopliae_arsef_23_gca_000187425.MAA_2.0.cdna.all.fa.gz',
	'metarhizium_anisopliae_brip_53293_gca_000426965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/metarhizium_anisopliae_brip_53293_gca_000426965/cdna/Metarhizium_anisopliae_brip_53293_gca_000426965.ASM42696v1.cdna.all.fa.gz',
	'metarhizium_anisopliae_gca_000739145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_anisopliae_gca_000739145/cdna/Metarhizium_anisopliae_gca_000739145.Metarhizium_anisopliae.cdna.all.fa.gz',
	'metarhizium_majus_arsef_297_gca_000814945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/metarhizium_majus_arsef_297_gca_000814945/cdna/Metarhizium_majus_arsef_297_gca_000814945.MAJ_1.0.cdna.all.fa.gz',
	'metarhizium_robertsii_gca_000591435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/metarhizium_robertsii_gca_000591435/cdna/Metarhizium_robertsii_gca_000591435.MR2010_11_27.cdna.all.fa.gz',
	'metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035/cdna/Metschnikowia_bicuspidata_var_bicuspidata_nrrl_yb_4993_gca_001664035.Metbi1.cdna.all.fa.gz',
	'meyerozyma_guilliermondii_atcc_6260_gca_000149425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/meyerozyma_guilliermondii_atcc_6260_gca_000149425/cdna/Meyerozyma_guilliermondii_atcc_6260_gca_000149425.ASM14942v1.cdna.all.fa.gz',
	'microbotryum_violaceum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/microbotryum_violaceum/cdna/Microbotryum_violaceum.GCA_000166175.1.cdna.all.fa.gz',
	'microcebus_murinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/microcebus_murinus/cdna/Microcebus_murinus.Mmur_3.0.cdna.abinitio.fa.gz',
	'microdochium_bolleyi_gca_001566295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/microdochium_bolleyi_gca_001566295/cdna/Microdochium_bolleyi_gca_001566295.Microdochium_bolleyi_v1.0.cdna.all.fa.gz',
	'microsporum_canis_cbs_113480_gca_000151145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/microsporum_canis_cbs_113480_gca_000151145/cdna/Microsporum_canis_cbs_113480_gca_000151145.ASM15114v1.cdna.all.fa.gz',
	'microtus_ochrogaster'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/microtus_ochrogaster/cdna/Microtus_ochrogaster.MicOch1.0.cdna.abinitio.fa.gz',
	'millerozyma_farinosa_cbs_7064_gca_000315895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/millerozyma_farinosa_cbs_7064_gca_000315895/cdna/Millerozyma_farinosa_cbs_7064_gca_000315895.ASM31589v1.cdna.all.fa.gz',
	'mitosporidium_daphniae_gca_000760515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/mitosporidium_daphniae_gca_000760515/cdna/Mitosporidium_daphniae_gca_000760515.UGP1.0.cdna.all.fa.gz',
	'mixia_osmundae_iam_14324_gca_000241205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/mixia_osmundae_iam_14324_gca_000241205/cdna/Mixia_osmundae_iam_14324_gca_000241205.Mosmundae_2.0.cdna.all.fa.gz',
	'mnemiopsis_leidyi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/mnemiopsis_leidyi/cdna/Mnemiopsis_leidyi.MneLei_Aug2011.cdna.all.fa.gz',
	'moelleriella_libera_rcef_2490_gca_001636675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/moelleriella_libera_rcef_2490_gca_001636675/cdna/Moelleriella_libera_rcef_2490_gca_001636675.AAL_1.0.cdna.all.fa.gz',
	'moesziomyces_antarcticus_gca_000747765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moesziomyces_antarcticus_gca_000747765/cdna/Moesziomyces_antarcticus_gca_000747765.ASM74776v1.cdna.all.fa.gz',
	'moesziomyces_antarcticus_t_34_gca_000334475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moesziomyces_antarcticus_t_34_gca_000334475/cdna/Moesziomyces_antarcticus_t_34_gca_000334475.Pant_1.0.cdna.all.fa.gz',
	'mola_mola'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mola_mola/cdna/Mola_mola.ASM169857v1.cdna.abinitio.fa.gz',
	'moniliophthora_perniciosa_fa553_gca_000183025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moniliophthora_perniciosa_fa553_gca_000183025/cdna/Moniliophthora_perniciosa_fa553_gca_000183025.ASM18302v1.cdna.all.fa.gz',
	'moniliophthora_roreri_gca_001466705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moniliophthora_roreri_gca_001466705/cdna/Moniliophthora_roreri_gca_001466705.ASM146670v1.cdna.all.fa.gz',
	'moniliophthora_roreri_mca_2997_gca_000488995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/moniliophthora_roreri_mca_2997_gca_000488995/cdna/Moniliophthora_roreri_mca_2997_gca_000488995.M_roreri_MCA_2997_v1.cdna.all.fa.gz',
	'monodelphis_domestica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/monodelphis_domestica/cdna/Monodelphis_domestica.monDom5.cdna.abinitio.fa.gz',
	'monopterus_albus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/monopterus_albus/cdna/Monopterus_albus.M_albus_1.0.cdna.abinitio.fa.gz',
	'mortierella_elongata_ag_77_gca_001651415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mortierella_elongata_ag_77_gca_001651415/cdna/Mortierella_elongata_ag_77_gca_001651415.Mortierella_elongata_AG-77_v2.0.cdna.all.fa.gz',
	'mortierella_verticillata_nrrl_6337_gca_000739165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mortierella_verticillata_nrrl_6337_gca_000739165/cdna/Mortierella_verticillata_nrrl_6337_gca_000739165.Mort_vert_NRRL_6337_V1.cdna.all.fa.gz',
	'mucor_ambiguus_gca_000950595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mucor_ambiguus_gca_000950595/cdna/Mucor_ambiguus_gca_000950595.ASM95059v1.cdna.all.fa.gz',
	'mucor_circinelloides_f_circinelloides_1006phl_gca_000401635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mucor_circinelloides_f_circinelloides_1006phl_gca_000401635/cdna/Mucor_circinelloides_f_circinelloides_1006phl_gca_000401635.Muco_sp_1006Ph_V1.cdna.all.fa.gz',
	'mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945/cdna/Mucor_circinelloides_f_lusitanicus_cbs_277_49_gca_001638945.Mucci2.cdna.all.fa.gz',
	'mus_caroli'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_caroli/cdna/Mus_caroli.CAROLI_EIJ_v1.1.cdna.abinitio.fa.gz',
	'mus_musculus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus/cdna/Mus_musculus.GRCm38.cdna.abinitio.fa.gz',
	'mus_musculus_129s1svimj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_129s1svimj/cdna/Mus_musculus_129s1svimj.129S1_SvImJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_aj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_aj/cdna/Mus_musculus_aj.A_J_v1.cdna.abinitio.fa.gz',
	'mus_musculus_akrj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_akrj/cdna/Mus_musculus_akrj.AKR_J_v1.cdna.abinitio.fa.gz',
	'mus_musculus_balbcj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_balbcj/cdna/Mus_musculus_balbcj.BALB_cJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_c3hhej'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_c3hhej/cdna/Mus_musculus_c3hhej.C3H_HeJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_c57bl6nj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_c57bl6nj/cdna/Mus_musculus_c57bl6nj.C57BL_6NJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_casteij'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_casteij/cdna/Mus_musculus_casteij.CAST_EiJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_cbaj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_cbaj/cdna/Mus_musculus_cbaj.CBA_J_v1.cdna.abinitio.fa.gz',
	'mus_musculus_dba2j'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_dba2j/cdna/Mus_musculus_dba2j.DBA_2J_v1.cdna.abinitio.fa.gz',
	'mus_musculus_fvbnj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_fvbnj/cdna/Mus_musculus_fvbnj.FVB_NJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_lpj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_lpj/cdna/Mus_musculus_lpj.LP_J_v1.cdna.abinitio.fa.gz',
	'mus_musculus_nodshiltj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_nodshiltj/cdna/Mus_musculus_nodshiltj.NOD_ShiLtJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_nzohlltj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_nzohlltj/cdna/Mus_musculus_nzohlltj.NZO_HlLtJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_pwkphj'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_pwkphj/cdna/Mus_musculus_pwkphj.PWK_PhJ_v1.cdna.abinitio.fa.gz',
	'mus_musculus_wsbeij'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_musculus_wsbeij/cdna/Mus_musculus_wsbeij.WSB_EiJ_v1.cdna.abinitio.fa.gz',
	'mus_pahari'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_pahari/cdna/Mus_pahari.PAHARI_EIJ_v1.1.cdna.abinitio.fa.gz',
	'mus_spicilegus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_spicilegus/cdna/Mus_spicilegus.MUSP714.cdna.abinitio.fa.gz',
	'mus_spretus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mus_spretus/cdna/Mus_spretus.SPRET_EiJ_v1.cdna.abinitio.fa.gz',
	'musa_acuminata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/musa_acuminata/cdna/Musa_acuminata.ASM31385v1.cdna.all.fa.gz',
	'mustela_putorius_furo'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/mustela_putorius_furo/cdna/Mustela_putorius_furo.MusPutFur1.0.cdna.abinitio.fa.gz',
	'mycosphaerella_eumusae_gca_001578235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/mycosphaerella_eumusae_gca_001578235/cdna/Mycosphaerella_eumusae_gca_001578235.ASM157823v1.cdna.all.fa.gz',
	'myotis_lucifugus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/myotis_lucifugus/cdna/Myotis_lucifugus.Myoluc2.0.cdna.abinitio.fa.gz',
	'nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315/cdna/Nadsonia_fulvescens_var_elongata_dsm_6958_gca_001661315.Nadfu1.cdna.all.fa.gz',
	'naegleria_gruberi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_heterolobosea1_collection/naegleria_gruberi/cdna/Naegleria_gruberi.V1.0.cdna.all.fa.gz',
	'naematelia_encephala_gca_002105065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/naematelia_encephala_gca_002105065/cdna/Naematelia_encephala_gca_002105065.Treen1.cdna.all.fa.gz',
	'nannizzia_gypsea_cbs_118893_gca_000150975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/nannizzia_gypsea_cbs_118893_gca_000150975/cdna/Nannizzia_gypsea_cbs_118893_gca_000150975.MS_CBS118893.cdna.all.fa.gz',
	'nannochloropsis_gaditana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/nannochloropsis_gaditana/cdna/Nannochloropsis_gaditana.NagaB31_1.0.cdna.all.fa.gz',
	'nannospalax_galili'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/nannospalax_galili/cdna/Nannospalax_galili.S.galili_v1.0.cdna.abinitio.fa.gz',
	'nasonia_vitripennis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/nasonia_vitripennis/cdna/Nasonia_vitripennis.Nvit_2.1.cdna.all.fa.gz',
	'naumovozyma_castellii_cbs_4309_gca_000237345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/naumovozyma_castellii_cbs_4309_gca_000237345/cdna/Naumovozyma_castellii_cbs_4309_gca_000237345.ASM23734v1.cdna.all.fa.gz',
	'naumovozyma_dairenensis_cbs_421_gca_000227115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/naumovozyma_dairenensis_cbs_421_gca_000227115/cdna/Naumovozyma_dairenensis_cbs_421_gca_000227115.ASM22711v2.cdna.all.fa.gz',
	'nematocida_parisii_ertm1_gca_000250985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_parisii_ertm1_gca_000250985/cdna/Nematocida_parisii_ertm1_gca_000250985.Nema_parisii_ERTm1_V3.cdna.all.fa.gz',
	'nematocida_parisii_ertm3_gca_000190615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_parisii_ertm3_gca_000190615/cdna/Nematocida_parisii_ertm3_gca_000190615.Nema_parisii_ERTm3_V1.cdna.all.fa.gz',
	'nematocida_sp_1_ertm2_gca_000250695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_sp_1_ertm2_gca_000250695/cdna/Nematocida_sp_1_ertm2_gca_000250695.Nema_parisii_ERTm2_V1.cdna.all.fa.gz',
	'nematocida_sp_1_ertm6_gca_000738915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_sp_1_ertm6_gca_000738915/cdna/Nematocida_sp_1_ertm6_gca_000738915.Nema_sp_1_ERTm6_V2.cdna.all.fa.gz',
	'nematocida_sp_ertm5_gca_001642415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nematocida_sp_ertm5_gca_001642415/cdna/Nematocida_sp_ertm5_gca_001642415.ASM164241v1.cdna.all.fa.gz',
	'nematostella_vectensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/nematostella_vectensis/cdna/Nematostella_vectensis.ASM20922v1.cdna.all.fa.gz',
	'neocallimastix_californiae_gca_002104975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_neocallimastigomycota1_collection/neocallimastix_californiae_gca_002104975/cdna/Neocallimastix_californiae_gca_002104975.Neocallimastix_sp._G1_v1.0.cdna.all.fa.gz',
	'neofusicoccum_parvum_ucrnp2_gca_000385595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/neofusicoccum_parvum_ucrnp2_gca_000385595/cdna/Neofusicoccum_parvum_ucrnp2_gca_000385595.UCRNP2V3.cdna.all.fa.gz',
	'neolamprologus_brichardi'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/neolamprologus_brichardi/cdna/Neolamprologus_brichardi.NeoBri1.0.cdna.abinitio.fa.gz',
	'neolecta_irregularis_dah_3_gca_001929475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/neolecta_irregularis_dah_3_gca_001929475/cdna/Neolecta_irregularis_dah_3_gca_001929475.NeoIirr1.0.cdna.all.fa.gz',
	'neolentinus_lepideus_hhb14362_ss_1_gca_001632425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/neolentinus_lepideus_hhb14362_ss_1_gca_001632425/cdna/Neolentinus_lepideus_hhb14362_ss_1_gca_001632425.Neole1.cdna.all.fa.gz',
	'neonectria_ditissima_gca_001306435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/neonectria_ditissima_gca_001306435/cdna/Neonectria_ditissima_gca_001306435.R0905_v2.0.cdna.all.fa.gz',
	'neosartorya_fischeri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/neosartorya_fischeri/cdna/Neosartorya_fischeri.ASM14964v1.cdna.all.fa.gz',
	'neovison_vison'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/neovison_vison/cdna/Neovison_vison.NNQGG.v01.cdna.abinitio.fa.gz',
	'neurospora_crassa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/neurospora_crassa/cdna/Neurospora_crassa.NC12.cdna.all.fa.gz',
	'neurospora_crassa_gca_000786625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/neurospora_crassa_gca_000786625/cdna/Neurospora_crassa_gca_000786625.Neucr_trp3_1.cdna.all.fa.gz',
	'neurospora_tetrasperma_fgsc_2509_gca_000213195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/neurospora_tetrasperma_fgsc_2509_gca_000213195/cdna/Neurospora_tetrasperma_fgsc_2509_gca_000213195.v1.0.cdna.all.fa.gz',
	'nicotiana_attenuata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/nicotiana_attenuata/cdna/Nicotiana_attenuata.NIATTr2.cdna.all.fa.gz',
	'nomascus_leucogenys'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/nomascus_leucogenys/cdna/Nomascus_leucogenys.Nleu_3.0.cdna.abinitio.fa.gz',
	'nosema_apis_brl_01_gca_000447185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_apis_brl_01_gca_000447185/cdna/Nosema_apis_brl_01_gca_000447185.NapisBRLv01.cdna.all.fa.gz',
	'nosema_bombycis_cq1_gca_000383075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_bombycis_cq1_gca_000383075/cdna/Nosema_bombycis_cq1_gca_000383075.NosBomCQ1_v1.0.cdna.all.fa.gz',
	'nosema_ceranae_brl01_gca_000182985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_ceranae_brl01_gca_000182985/cdna/Nosema_ceranae_brl01_gca_000182985.ASM18298v1.cdna.all.fa.gz',
	'nosema_ceranae_gca_000988165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/nosema_ceranae_gca_000988165/cdna/Nosema_ceranae_gca_000988165.ASM98816v1.cdna.all.fa.gz',
	'notamacropus_eugenii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/notamacropus_eugenii/cdna/Notamacropus_eugenii.Meug_1.0.cdna.abinitio.fa.gz',
	'notechis_scutatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/notechis_scutatus/cdna/Notechis_scutatus.TS10Xv2-PRI.cdna.abinitio.fa.gz',
	'nothoprocta_perdicaria'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/nothoprocta_perdicaria/cdna/Nothoprocta_perdicaria.notPer1.cdna.abinitio.fa.gz',
	'numida_meleagris'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/numida_meleagris/cdna/Numida_meleagris.NumMel1.0.cdna.abinitio.fa.gz',
	'ochotona_princeps'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ochotona_princeps/cdna/Ochotona_princeps.OchPri2.0-Ens.cdna.abinitio.fa.gz',
	'octodon_degus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/octodon_degus/cdna/Octodon_degus.OctDeg1.0.cdna.abinitio.fa.gz',
	'octopus_bimaculoides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/octopus_bimaculoides/cdna/Octopus_bimaculoides.PRJNA270931.cdna.all.fa.gz',
	'ogataea_parapolymorpha_dl_1_gca_000187245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ogataea_parapolymorpha_dl_1_gca_000187245/cdna/Ogataea_parapolymorpha_dl_1_gca_000187245.Hansenula_2.cdna.all.fa.gz',
	'ogataea_polymorpha_gca_001664045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/ogataea_polymorpha_gca_001664045/cdna/Ogataea_polymorpha_gca_001664045.Hanpo2.cdna.all.fa.gz',
	'oidiodendron_maius_zn_gca_000827325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/oidiodendron_maius_zn_gca_000827325/cdna/Oidiodendron_maius_zn_gca_000827325.Oidiodendron_maius_Zn_v1.0.cdna.all.fa.gz',
	'onchocerca_volvulus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/onchocerca_volvulus/cdna/Onchocerca_volvulus.ASM49940v2.cdna.all.fa.gz',
	'ophiocordyceps_sinensis_co18_gca_000448365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ophiocordyceps_sinensis_co18_gca_000448365/cdna/Ophiocordyceps_sinensis_co18_gca_000448365.ASM44836v1.cdna.all.fa.gz',
	'ophiostoma_piceae_uamh_11346_gca_000410735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ophiostoma_piceae_uamh_11346_gca_000410735/cdna/Ophiostoma_piceae_uamh_11346_gca_000410735.Opiceae_UAMHv01.cdna.all.fa.gz',
	'ordospora_colligata_oc4_gca_000803265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/ordospora_colligata_oc4_gca_000803265/cdna/Ordospora_colligata_oc4_gca_000803265.ASM80326v1.cdna.all.fa.gz',
	'oreochromis_niloticus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oreochromis_niloticus/cdna/Oreochromis_niloticus.Orenil1.0.cdna.abinitio.fa.gz',
	'ornithorhynchus_anatinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ornithorhynchus_anatinus/cdna/Ornithorhynchus_anatinus.OANA5.cdna.abinitio.fa.gz',
	'oryctolagus_cuniculus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryctolagus_cuniculus/cdna/Oryctolagus_cuniculus.OryCun2.0.cdna.abinitio.fa.gz',
	'oryza_barthii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_barthii/cdna/Oryza_barthii.O.barthii_v1.cdna.abinitio.fa.gz',
	'oryza_brachyantha'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_brachyantha/cdna/Oryza_brachyantha.Oryza_brachyantha.v1.4b.cdna.abinitio.fa.gz',
	'oryza_glaberrima'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_glaberrima/cdna/Oryza_glaberrima.Oryza_glaberrima_V1.cdna.abinitio.fa.gz',
	'oryza_glumipatula'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_glumipatula/cdna/Oryza_glumipatula.Oryza_glumaepatula_v1.5.cdna.abinitio.fa.gz',
	'oryza_indica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_indica/cdna/Oryza_indica.ASM465v1.cdna.abinitio.fa.gz',
	'oryza_longistaminata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_longistaminata/cdna/Oryza_longistaminata.O_longistaminata_v1.0.cdna.all.fa.gz',
	'oryza_meridionalis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_meridionalis/cdna/Oryza_meridionalis.Oryza_meridionalis_v1.3.cdna.all.fa.gz',
	'oryza_nivara'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_nivara/cdna/Oryza_nivara.Oryza_nivara_v1.0.cdna.abinitio.fa.gz',
	'oryza_punctata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_punctata/cdna/Oryza_punctata.Oryza_punctata_v1.2.cdna.abinitio.fa.gz',
	'oryza_rufipogon'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_rufipogon/cdna/Oryza_rufipogon.OR_W1943.cdna.all.fa.gz',
	'oryza_sativa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/oryza_sativa/cdna/Oryza_sativa.IRGSP-1.0.cdna.abinitio.fa.gz',
	'oryzias_latipes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_latipes/cdna/Oryzias_latipes.ASM223467v1.cdna.abinitio.fa.gz',
	'oryzias_latipes_hni'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_latipes_hni/cdna/Oryzias_latipes_hni.ASM223471v1.cdna.abinitio.fa.gz',
	'oryzias_latipes_hsok'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_latipes_hsok/cdna/Oryzias_latipes_hsok.ASM223469v1.cdna.abinitio.fa.gz',
	'oryzias_melastigma'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/oryzias_melastigma/cdna/Oryzias_melastigma.Om_v0.7.RACA.cdna.abinitio.fa.gz',
	'ostreococcus_lucimarinus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/ostreococcus_lucimarinus/cdna/Ostreococcus_lucimarinus.ASM9206v1.cdna.all.fa.gz',
	'otolemur_garnettii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/otolemur_garnettii/cdna/Otolemur_garnettii.OtoGar3.cdna.abinitio.fa.gz',
	'ovis_aries'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ovis_aries/cdna/Ovis_aries.Oar_v3.1.cdna.abinitio.fa.gz',
	'pachysolen_tannophilus_nrrl_y_2460_gca_001661245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pachysolen_tannophilus_nrrl_y_2460_gca_001661245/cdna/Pachysolen_tannophilus_nrrl_y_2460_gca_001661245.Pacta1_2.cdna.all.fa.gz',
	'pan_paniscus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pan_paniscus/cdna/Pan_paniscus.panpan1.1.cdna.abinitio.fa.gz',
	'pan_troglodytes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pan_troglodytes/cdna/Pan_troglodytes.Pan_tro_3.0.cdna.abinitio.fa.gz',
	'panthera_pardus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/panthera_pardus/cdna/Panthera_pardus.PanPar1.0.cdna.abinitio.fa.gz',
	'panthera_tigris_altaica'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/panthera_tigris_altaica/cdna/Panthera_tigris_altaica.PanTig1.0.cdna.abinitio.fa.gz',
	'papio_anubis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/papio_anubis/cdna/Papio_anubis.Panu_3.0.cdna.abinitio.fa.gz',
	'paracoccidioides_brasiliensis_gca_001713645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/paracoccidioides_brasiliensis_gca_001713645/cdna/Paracoccidioides_brasiliensis_gca_001713645.ASM171364v1.cdna.all.fa.gz',
	'paracoccidioides_brasiliensis_gca_001713695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/paracoccidioides_brasiliensis_gca_001713695/cdna/Paracoccidioides_brasiliensis_gca_001713695.Paracocci_br_PbCNH.cdna.all.fa.gz',
	'paracoccidioides_brasiliensis_pb03_gca_000150475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/paracoccidioides_brasiliensis_pb03_gca_000150475/cdna/Paracoccidioides_brasiliensis_pb03_gca_000150475.Paracocci_br_Pb03_V2.cdna.all.fa.gz',
	'paracoccidioides_brasiliensis_pb18_gca_000150735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/paracoccidioides_brasiliensis_pb18_gca_000150735/cdna/Paracoccidioides_brasiliensis_pb18_gca_000150735.Paracocci_br_Pb18_V2.cdna.all.fa.gz',
	'paracoccidioides_sp_lutzii_pb01_gca_000150705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/paracoccidioides_sp_lutzii_pb01_gca_000150705/cdna/Paracoccidioides_sp_lutzii_pb01_gca_000150705.Paracocci_br_Pb01_V2.cdna.all.fa.gz',
	'paramecium_tetraurelia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/paramecium_tetraurelia/cdna/Paramecium_tetraurelia.ASM16542v1.cdna.all.fa.gz',
	'paramicrosporidium_saccamoebae_gca_002794465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_rozellomycota1_collection/paramicrosporidium_saccamoebae_gca_002794465/cdna/Paramicrosporidium_saccamoebae_gca_002794465.KSL3_v6.cdna.all.fa.gz',
	'paramormyrops_kingsleyae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/paramormyrops_kingsleyae/cdna/Paramormyrops_kingsleyae.PKINGS_0.1.cdna.abinitio.fa.gz',
	'paraphaeosphaeria_sporulosa_gca_001642045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/paraphaeosphaeria_sporulosa_gca_001642045/cdna/Paraphaeosphaeria_sporulosa_gca_001642045.Parsp1.cdna.all.fa.gz',
	'parus_major'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/parus_major/cdna/Parus_major.Parus_major1.1.cdna.abinitio.fa.gz',
	'paulinella_chromatophora'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/paulinella_chromatophora/cdna/Paulinella_chromatophora.ASM1792v1.cdna.all.fa.gz',
	'paxillus_involutus_atcc_200175_gca_000827475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/paxillus_involutus_atcc_200175_gca_000827475/cdna/Paxillus_involutus_atcc_200175_gca_000827475.Paxillus_involutus_ATCC_200175_v1.0.cdna.all.fa.gz',
	'paxillus_rubicundulus_ve08_2h10_gca_000827395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/paxillus_rubicundulus_ve08_2h10_gca_000827395/cdna/Paxillus_rubicundulus_ve08_2h10_gca_000827395.Paxillus_rubicundulus_Ve08.2h10_v1.0.cdna.all.fa.gz',
	'pediculus_humanus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/pediculus_humanus/cdna/Pediculus_humanus.PhumU2.cdna.all.fa.gz',
	'pelodiscus_sinensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pelodiscus_sinensis/cdna/Pelodiscus_sinensis.PelSin_1.0.cdna.abinitio.fa.gz',
	'penicilliopsis_zonata_cbs_506_65_gca_001890105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicilliopsis_zonata_cbs_506_65_gca_001890105/cdna/Penicilliopsis_zonata_cbs_506_65_gca_001890105.Aspzo1.cdna.all.fa.gz',
	'penicillium_brasilianum_gca_001048715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/penicillium_brasilianum_gca_001048715/cdna/Penicillium_brasilianum_gca_001048715.Pbras_Allpaths-LG.cdna.all.fa.gz',
	'penicillium_brasilianum_gca_002016555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_brasilianum_gca_002016555/cdna/Penicillium_brasilianum_gca_002016555.ASM201655v1.cdna.all.fa.gz',
	'penicillium_camemberti_fm_013_gca_000513335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/penicillium_camemberti_fm_013_gca_000513335/cdna/Penicillium_camemberti_fm_013_gca_000513335.PCAMFM013_20131106.cdna.all.fa.gz',
	'penicillium_chrysogenum_gca_000710275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_chrysogenum_gca_000710275/cdna/Penicillium_chrysogenum_gca_000710275.ASM71027v1.cdna.all.fa.gz',
	'penicillium_digitatum_pd1_gca_000315645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_digitatum_pd1_gca_000315645/cdna/Penicillium_digitatum_pd1_gca_000315645.PdigPd1_v1.cdna.all.fa.gz',
	'penicillium_digitatum_phi26_gca_000315665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_digitatum_phi26_gca_000315665/cdna/Penicillium_digitatum_phi26_gca_000315665.PdigPHI26_v1.cdna.all.fa.gz',
	'penicillium_expansum_gca_000769745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_expansum_gca_000769745/cdna/Penicillium_expansum_gca_000769745.ASM76974v1.cdna.all.fa.gz',
	'penicillium_freii_gca_001513925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_freii_gca_001513925/cdna/Penicillium_freii_gca_001513925.ASM151392v1.cdna.all.fa.gz',
	'penicillium_griseofulvum_gca_001561935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_griseofulvum_gca_001561935/cdna/Penicillium_griseofulvum_gca_001561935.ASM156193v1.cdna.all.fa.gz',
	'penicillium_italicum_gca_000769765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_italicum_gca_000769765/cdna/Penicillium_italicum_gca_000769765.ASM76976v1.cdna.all.fa.gz',
	'penicillium_nordicum_gca_001278595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_nordicum_gca_001278595/cdna/Penicillium_nordicum_gca_001278595.ASM127859v1.cdna.all.fa.gz',
	'penicillium_oxalicum_114_2_gca_000346795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_oxalicum_114_2_gca_000346795/cdna/Penicillium_oxalicum_114_2_gca_000346795.pde_v1.0.cdna.all.fa.gz',
	'penicillium_roqueforti_fm164_gca_000513255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_roqueforti_fm164_gca_000513255/cdna/Penicillium_roqueforti_fm164_gca_000513255.PROQFM164_20130607.cdna.all.fa.gz',
	'penicillium_rubens_wisconsin_54_1255_gca_000226395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/penicillium_rubens_wisconsin_54_1255_gca_000226395/cdna/Penicillium_rubens_wisconsin_54_1255_gca_000226395.PenChr_Nov2007.cdna.all.fa.gz',
	'penicillium_subrubescens_gca_001908125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/penicillium_subrubescens_gca_001908125/cdna/Penicillium_subrubescens_gca_001908125.ASM190812v1.cdna.all.fa.gz',
	'peniophora_sp_cont_gca_001632445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/peniophora_sp_cont_gca_001632445/cdna/Peniophora_sp_cont_gca_001632445.Ricme1.cdna.all.fa.gz',
	'periconia_macrospinosa_gca_003073855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/periconia_macrospinosa_gca_003073855/cdna/Periconia_macrospinosa_gca_003073855.Perma1.cdna.all.fa.gz',
	'periophthalmus_magnuspinnatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/periophthalmus_magnuspinnatus/cdna/Periophthalmus_magnuspinnatus.PM.fa.cdna.abinitio.fa.gz',
	'perkinsus_marinus_atcc_50983'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/perkinsus_marinus_atcc_50983/cdna/Perkinsus_marinus_atcc_50983.JCVI_PMG_1.0.cdna.all.fa.gz',
	'peromyscus_maniculatus_bairdii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/peromyscus_maniculatus_bairdii/cdna/Peromyscus_maniculatus_bairdii.HU_Pman_2.1.cdna.abinitio.fa.gz',
	'pestalotiopsis_fici_w106_1_gca_000516985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pestalotiopsis_fici_w106_1_gca_000516985/cdna/Pestalotiopsis_fici_w106_1_gca_000516985.PFICI.cdna.all.fa.gz',
	'petromyzon_marinus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/petromyzon_marinus/cdna/Petromyzon_marinus.Pmarinus_7.0.cdna.abinitio.fa.gz',
	'pezoloma_ericae_gca_002865625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pezoloma_ericae_gca_002865625/cdna/Pezoloma_ericae_gca_002865625.Rhier1.cdna.all.fa.gz',
	'phaeoacremonium_minimum_ucrpa7_gca_000392275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/phaeoacremonium_minimum_ucrpa7_gca_000392275/cdna/Phaeoacremonium_minimum_ucrpa7_gca_000392275.UCRPA7V03.cdna.all.fa.gz',
	'phaeodactylum_tricornutum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phaeodactylum_tricornutum/cdna/Phaeodactylum_tricornutum.ASM15095v2.cdna.all.fa.gz',
	'phaeomoniella_chlamydospora_gca_001006345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phaeomoniella_chlamydospora_gca_001006345/cdna/Phaeomoniella_chlamydospora_gca_001006345.UCRPA7V1.0.cdna.all.fa.gz',
	'phaeosphaeria_nodorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/phaeosphaeria_nodorum/cdna/Phaeosphaeria_nodorum.ASM14691v1.cdna.all.fa.gz',
	'phanerochaete_carnosa_hhb_10118_sp_gca_000300595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/phanerochaete_carnosa_hhb_10118_sp_gca_000300595/cdna/Phanerochaete_carnosa_hhb_10118_sp_gca_000300595.Phanerochaete_carnosa_HHB-10118-Sp_v1.0.cdna.all.fa.gz',
	'phascolarctos_cinereus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/phascolarctos_cinereus/cdna/Phascolarctos_cinereus.phaCin_unsw_v4.1.cdna.abinitio.fa.gz',
	'phaseolus_vulgaris'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/phaseolus_vulgaris/cdna/Phaseolus_vulgaris.PhaVulg1_0.cdna.all.fa.gz',
	'phialocephala_scopiformis_gca_001500285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phialocephala_scopiformis_gca_001500285/cdna/Phialocephala_scopiformis_gca_001500285.Phisc1.cdna.all.fa.gz',
	'phialocephala_subalpina_gca_900073065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phialocephala_subalpina_gca_900073065/cdna/Phialocephala_subalpina_gca_900073065.PAC_version_1.cdna.all.fa.gz',
	'phialophora_americana_gca_000835435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/phialophora_americana_gca_000835435/cdna/Phialophora_americana_gca_000835435.Capr_semi_CBS27337_V1.cdna.all.fa.gz',
	'phialophora_attae_gca_001299255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/phialophora_attae_gca_001299255/cdna/Phialophora_attae_gca_001299255.ASM129925v1.cdna.all.fa.gz',
	'phlebia_centrifuga_gca_001913855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/phlebia_centrifuga_gca_001913855/cdna/Phlebia_centrifuga_gca_001913855.ASM191385v2.cdna.all.fa.gz',
	'phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265/cdna/Phlebiopsis_gigantea_11061_1_cr5_6_gca_000832265.Phlgi1.cdna.all.fa.gz',
	'physcomitrella_patens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/physcomitrella_patens/cdna/Physcomitrella_patens.Phypa_V3.cdna.all.fa.gz',
	'phytophthora_infestans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_infestans/cdna/Phytophthora_infestans.ASM14294v1.cdna.all.fa.gz',
	'phytophthora_kernoviae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_kernoviae/cdna/Phytophthora_kernoviae.PhyKer238_432v1.cdna.all.fa.gz',
	'phytophthora_lateralis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_lateralis/cdna/Phytophthora_lateralis.MPF4_v1.0.cdna.all.fa.gz',
	'phytophthora_parasitica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_parasitica/cdna/Phytophthora_parasitica.Phyt_para_P1569_V1.cdna.all.fa.gz',
	'phytophthora_parasitica_cj01a1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_cj01a1/cdna/Phytophthora_parasitica_cj01a1.Phyt_para_CJ01A1_V1.cdna.all.fa.gz',
	'phytophthora_parasitica_gca_000509465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509465/cdna/Phytophthora_parasitica_gca_000509465.Phyt_para_CJ02B3_V1.cdna.all.fa.gz',
	'phytophthora_parasitica_gca_000509485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509485/cdna/Phytophthora_parasitica_gca_000509485.Phyt_para_CJ05E6_V1.cdna.all.fa.gz',
	'phytophthora_parasitica_gca_000509505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509505/cdna/Phytophthora_parasitica_gca_000509505.Phyt_para_CHvinca01_V1.cdna.all.fa.gz',
	'phytophthora_parasitica_gca_000509525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_gca_000509525/cdna/Phytophthora_parasitica_gca_000509525.Phyt_para_IAC_01_95_V1.cdna.all.fa.gz',
	'phytophthora_parasitica_inra_310'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_inra_310/cdna/Phytophthora_parasitica_inra_310.PP_INRA-310_V2.cdna.all.fa.gz',
	'phytophthora_parasitica_p10297'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_p10297/cdna/Phytophthora_parasitica_p10297.Phyt_para_P10297_V1.cdna.all.fa.gz',
	'phytophthora_parasitica_p1976'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/phytophthora_parasitica_p1976/cdna/Phytophthora_parasitica_p1976.Phyt_para_P1976_V1.cdna.all.fa.gz',
	'phytophthora_ramorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_ramorum/cdna/Phytophthora_ramorum.ASM14973v1.cdna.all.fa.gz',
	'phytophthora_sojae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/phytophthora_sojae/cdna/Phytophthora_sojae.P_sojae_V3_0.cdna.all.fa.gz',
	'pichia_kudriavzevii_gca_000764455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pichia_kudriavzevii_gca_000764455/cdna/Pichia_kudriavzevii_gca_000764455.ASM76445v1.cdna.all.fa.gz',
	'pichia_kudriavzevii_gca_002166775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pichia_kudriavzevii_gca_002166775/cdna/Pichia_kudriavzevii_gca_002166775.Cand_krus_Ckrusei653_V21.cdna.all.fa.gz',
	'pichia_kudriavzevii_gca_003054445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pichia_kudriavzevii_gca_003054445/cdna/Pichia_kudriavzevii_gca_003054445.ASM305444v1.cdna.all.fa.gz',
	'pichia_membranifaciens_gca_001950575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pichia_membranifaciens_gca_001950575/cdna/Pichia_membranifaciens_gca_001950575.P_membrenifaciens-1.0.cdna.all.fa.gz',
	'pichia_membranifaciens_nrrl_y_2026_gca_001661235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pichia_membranifaciens_nrrl_y_2026_gca_001661235/cdna/Pichia_membranifaciens_nrrl_y_2026_gca_001661235.Picme2.cdna.all.fa.gz',
	'piliocolobus_tephrosceles'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/piliocolobus_tephrosceles/cdna/Piliocolobus_tephrosceles.ASM277652v2.cdna.abinitio.fa.gz',
	'piloderma_croceum_f_1598_gca_000827315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/piloderma_croceum_f_1598_gca_000827315/cdna/Piloderma_croceum_f_1598_gca_000827315.Piloderma_croceum_F_1598_v1.0.cdna.all.fa.gz',
	'piromyces_finnis_gca_002104945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_neocallimastigomycota1_collection/piromyces_finnis_gca_002104945/cdna/Piromyces_finnis_gca_002104945.Piromyces_sp._finnis_v3.0.cdna.all.fa.gz',
	'pisolithus_microcarpus_441_gca_000827275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pisolithus_microcarpus_441_gca_000827275/cdna/Pisolithus_microcarpus_441_gca_000827275.Pisolithus_microcarpus_441_v1.0.cdna.all.fa.gz',
	'pisolithus_tinctorius_marx_270_gca_000827335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pisolithus_tinctorius_marx_270_gca_000827335/cdna/Pisolithus_tinctorius_marx_270_gca_000827335.Pisolithus_tinctorius_Marx_270_v1.0.cdna.all.fa.gz',
	'plasmodium_berghei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_berghei/cdna/Plasmodium_berghei.PBANKA01.cdna.all.fa.gz',
	'plasmodium_berghei_gca_000005395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_berghei_gca_000005395/cdna/Plasmodium_berghei_gca_000005395.ASM539v1.cdna.all.fa.gz',
	'plasmodium_berghei_gca_900044335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_berghei_gca_900044335/cdna/Plasmodium_berghei_gca_900044335.PbK173.cdna.all.fa.gz',
	'plasmodium_chabaudi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_chabaudi/cdna/Plasmodium_chabaudi.PCHAS01.cdna.all.fa.gz',
	'plasmodium_chabaudi_chabaudi_gca_000003075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_chabaudi_chabaudi_gca_000003075/cdna/Plasmodium_chabaudi_chabaudi_gca_000003075.ASM307v1.cdna.all.fa.gz',
	'plasmodium_coatneyi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_coatneyi/cdna/Plasmodium_coatneyi.ASM168000v1.cdna.all.fa.gz',
	'plasmodium_cynomolgi_strain_b'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_cynomolgi_strain_b/cdna/Plasmodium_cynomolgi_strain_b.PcynB_1.0.cdna.all.fa.gz',
	'plasmodium_falciparum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_falciparum/cdna/Plasmodium_falciparum.EPr1.cdna.all.fa.gz',
	'plasmodium_falciparum_7g8'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_7g8/cdna/Plasmodium_falciparum_7g8.Plas_falc_7G8_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_camp_malaysia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_camp_malaysia/cdna/Plasmodium_falciparum_camp_malaysia.Plas_falc_Malayan_Camp_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_dd2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_dd2/cdna/Plasmodium_falciparum_dd2.ASM14979v1.cdna.all.fa.gz',
	'plasmodium_falciparum_fch_4'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_fch_4/cdna/Plasmodium_falciparum_fch_4.Plas_falc_FCH_4_V2.cdna.all.fa.gz',
	'plasmodium_falciparum_hb3'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_hb3/cdna/Plasmodium_falciparum_hb3.ASM14966v2.cdna.all.fa.gz',
	'plasmodium_falciparum_igh_cr14'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_igh_cr14/cdna/Plasmodium_falciparum_igh_cr14.ASM18605v2.cdna.all.fa.gz',
	'plasmodium_falciparum_malips096_e11'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_malips096_e11/cdna/Plasmodium_falciparum_malips096_e11.Plas_falc_MaliPS096_E11_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_nf135_5_c10'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_nf135_5_c10/cdna/Plasmodium_falciparum_nf135_5_c10.Plas_falc_NF135_5_C10_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_nf54'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_nf54/cdna/Plasmodium_falciparum_nf54.Plas_falc_NF4_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_palo_alto_uganda'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_palo_alto_uganda/cdna/Plasmodium_falciparum_palo_alto_uganda.Plas_falc_Uganda_Palo-Alto_FUP_H_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_raj116'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_raj116/cdna/Plasmodium_falciparum_raj116.ASM18602v2.cdna.all.fa.gz',
	'plasmodium_falciparum_santa_lucia'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_santa_lucia/cdna/Plasmodium_falciparum_santa_lucia.Plas_falc_Santa_Lucia_Salvador_I_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_tanzania_2000708_'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_tanzania_2000708_/cdna/Plasmodium_falciparum_tanzania_2000708_.Plas_falc_02000708_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_ugt5_1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_ugt5_1/cdna/Plasmodium_falciparum_ugt5_1.Plas_falc_5_1_V1.cdna.all.fa.gz',
	'plasmodium_falciparum_vietnam_oak_knoll_fvo_'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_falciparum_vietnam_oak_knoll_fvo_/cdna/Plasmodium_falciparum_vietnam_oak_knoll_fvo_.Plas_falc_Vietnam_Oak-Knoll_FVO_V1.cdna.all.fa.gz',
	'plasmodium_fragile'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_fragile/cdna/Plasmodium_fragile.Plas_frag_nilgiri_V1.cdna.all.fa.gz',
	'plasmodium_gaboni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_gaboni/cdna/Plasmodium_gaboni.ASM160202v1.cdna.all.fa.gz',
	'plasmodium_inui_san_antonio_1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_inui_san_antonio_1/cdna/Plasmodium_inui_san_antonio_1.Plas_inui_San_Antonio_1_V1.cdna.all.fa.gz',
	'plasmodium_knowlesi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_knowlesi/cdna/Plasmodium_knowlesi.ASM635v1.cdna.all.fa.gz',
	'plasmodium_knowlesi_strain_h_gca_900004885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_knowlesi_strain_h_gca_900004885/cdna/Plasmodium_knowlesi_strain_h_gca_900004885.PKNA1-C.2.cdna.all.fa.gz',
	'plasmodium_malariae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_malariae/cdna/Plasmodium_malariae.Pmal.cdna.all.fa.gz',
	'plasmodium_ovale_curtisi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_ovale_curtisi/cdna/Plasmodium_ovale_curtisi.Poc1.cdna.all.fa.gz',
	'plasmodium_reichenowi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_reichenowi/cdna/Plasmodium_reichenowi.PREICH001.cdna.all.fa.gz',
	'plasmodium_reichenowi_gca_001601855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_reichenowi_gca_001601855/cdna/Plasmodium_reichenowi_gca_001601855.ASM160185v1.cdna.all.fa.gz',
	'plasmodium_vinckei_petteri'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vinckei_petteri/cdna/Plasmodium_vinckei_petteri.Plas_vinc_pett_CR_V1.cdna.all.fa.gz',
	'plasmodium_vinckei_vinckei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vinckei_vinckei/cdna/Plasmodium_vinckei_vinckei.Plas_vinc_vinckei_V1.cdna.all.fa.gz',
	'plasmodium_vivax'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/plasmodium_vivax/cdna/Plasmodium_vivax.ASM241v2.cdna.all.fa.gz',
	'plasmodium_vivax_brazil_i'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_brazil_i/cdna/Plasmodium_vivax_brazil_i.PV_Brazil_I_V1.cdna.all.fa.gz',
	'plasmodium_vivax_india_vii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_india_vii/cdna/Plasmodium_vivax_india_vii.PV_India_VII_V2.cdna.all.fa.gz',
	'plasmodium_vivax_mauritania_i'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_mauritania_i/cdna/Plasmodium_vivax_mauritania_i.PV_Mauritania_I_V1.cdna.all.fa.gz',
	'plasmodium_vivax_north_korean'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_vivax_north_korean/cdna/Plasmodium_vivax_north_korean.PV_N_Korean_V1.cdna.all.fa.gz',
	'plasmodium_yoelii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii/cdna/Plasmodium_yoelii.PY17X01.cdna.all.fa.gz',
	'plasmodium_yoelii_17x'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii_17x/cdna/Plasmodium_yoelii_17x.Plas_yoel_17X_V2.cdna.all.fa.gz',
	'plasmodium_yoelii_gca_900002395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii_gca_900002395/cdna/Plasmodium_yoelii_gca_900002395.PYYM01.cdna.all.fa.gz',
	'plasmodium_yoelii_yoelii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/plasmodium_yoelii_yoelii/cdna/Plasmodium_yoelii_yoelii.ASM308v2.cdna.all.fa.gz',
	'plasmopara_halstedii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/plasmopara_halstedii/cdna/Plasmopara_halstedii.Plasmopara_halstedii_genome.cdna.all.fa.gz',
	'pleurotus_ostreatus_pc15_gca_000697685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pleurotus_ostreatus_pc15_gca_000697685/cdna/Pleurotus_ostreatus_pc15_gca_000697685.PleosPC15_2.cdna.all.fa.gz',
	'plicaturopsis_crispa_fd_325_ss_3_gca_000827205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/plicaturopsis_crispa_fd_325_ss_3_gca_000827205/cdna/Plicaturopsis_crispa_fd_325_ss_3_gca_000827205.Plicr1.cdna.all.fa.gz',
	'pneumocystis_carinii_b80_gca_001477545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pneumocystis_carinii_b80_gca_001477545/cdna/Pneumocystis_carinii_b80_gca_001477545.Pneu_cari_B80_V3.cdna.all.fa.gz',
	'pneumocystis_jirovecii_ru7_gca_001477535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pneumocystis_jirovecii_ru7_gca_001477535/cdna/Pneumocystis_jirovecii_ru7_gca_001477535.Pneu_jiro_RU7_V2.cdna.all.fa.gz',
	'pneumocystis_murina_b123_gca_000349005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pneumocystis_murina_b123_gca_000349005/cdna/Pneumocystis_murina_b123_gca_000349005.Pneumo_murina_B123_V4.cdna.all.fa.gz',
	'pochonia_chlamydosporia_170_gca_001653235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pochonia_chlamydosporia_170_gca_001653235/cdna/Pochonia_chlamydosporia_170_gca_001653235.ASM165323v2.cdna.all.fa.gz',
	'podospora_anserina_s_mat__gca_000226545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/podospora_anserina_s_mat__gca_000226545/cdna/Podospora_anserina_s_mat__gca_000226545.ASM22654v1.cdna.all.fa.gz',
	'poecilia_formosa'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_formosa/cdna/Poecilia_formosa.PoeFor_5.1.2.cdna.abinitio.fa.gz',
	'poecilia_latipinna'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_latipinna/cdna/Poecilia_latipinna.P_latipinna-1.0.cdna.abinitio.fa.gz',
	'poecilia_mexicana'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_mexicana/cdna/Poecilia_mexicana.P_mexicana-1.0.cdna.abinitio.fa.gz',
	'poecilia_reticulata'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/poecilia_reticulata/cdna/Poecilia_reticulata.Guppy_female_1.0_MT.cdna.abinitio.fa.gz',
	'pogona_vitticeps'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pogona_vitticeps/cdna/Pogona_vitticeps.pvi1.1.cdna.abinitio.fa.gz',
	'polysphondylium_pallidum_pn500'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_amoebozoa1_collection/polysphondylium_pallidum_pn500/cdna/Polysphondylium_pallidum_pn500.PolPal_Dec2009.cdna.all.fa.gz',
	'pongo_abelii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pongo_abelii/cdna/Pongo_abelii.PPYG2.cdna.abinitio.fa.gz',
	'populus_trichocarpa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/populus_trichocarpa/cdna/Populus_trichocarpa.Pop_tri_v3.cdna.all.fa.gz',
	'postia_placenta_mad_698_r_gca_000006255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/postia_placenta_mad_698_r_gca_000006255/cdna/Postia_placenta_mad_698_r_gca_000006255.Postia_placenta_V1.0.cdna.all.fa.gz',
	'postia_placenta_mad_698_r_sb12_gca_002117355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/postia_placenta_mad_698_r_sb12_gca_002117355/cdna/Postia_placenta_mad_698_r_sb12_gca_002117355.PosplRSB12_1.cdna.all.fa.gz',
	'procavia_capensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/procavia_capensis/cdna/Procavia_capensis.proCap1.cdna.abinitio.fa.gz',
	'prolemur_simus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/prolemur_simus/cdna/Prolemur_simus.Prosim_1.0.cdna.abinitio.fa.gz',
	'propithecus_coquereli'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/propithecus_coquereli/cdna/Propithecus_coquereli.Pcoq_1.0.cdna.abinitio.fa.gz',
	'protomyces_lactucaedebilis_gca_002105105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/protomyces_lactucaedebilis_gca_002105105/cdna/Protomyces_lactucaedebilis_gca_002105105.Prola1.cdna.all.fa.gz',
	'prunus_persica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/prunus_persica/cdna/Prunus_persica.Prunus_persica_NCBIv2.cdna.all.fa.gz',
	'pseudocercospora_fijiensis_cirad86_gca_000340215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudocercospora_fijiensis_cirad86_gca_000340215/cdna/Pseudocercospora_fijiensis_cirad86_gca_000340215.Mycfi2.cdna.all.fa.gz',
	'pseudocercospora_musae_gca_001578225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudocercospora_musae_gca_001578225/cdna/Pseudocercospora_musae_gca_001578225.ASM157822v1.cdna.all.fa.gz',
	'pseudogymnoascus_destructans_20631_21_gca_000184105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudogymnoascus_destructans_20631_21_gca_000184105/cdna/Pseudogymnoascus_destructans_20631_21_gca_000184105.Geom_destructans_V1.cdna.all.fa.gz',
	'pseudogymnoascus_destructans_gca_001641265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_destructans_gca_001641265/cdna/Pseudogymnoascus_destructans_gca_001641265.ASM164126v1.cdna.all.fa.gz',
	'pseudogymnoascus_sp_03vt05_gca_001662645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_03vt05_gca_001662645/cdna/Pseudogymnoascus_sp_03vt05_gca_001662645.ASM166264v1.cdna.all.fa.gz',
	'pseudogymnoascus_sp_05ny08_gca_001662605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_05ny08_gca_001662605/cdna/Pseudogymnoascus_sp_05ny08_gca_001662605.ASM166260v1.cdna.all.fa.gz',
	'pseudogymnoascus_sp_23342_1_i1_gca_001662575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_23342_1_i1_gca_001662575/cdna/Pseudogymnoascus_sp_23342_1_i1_gca_001662575.ASM166257v1.cdna.all.fa.gz',
	'pseudogymnoascus_sp_24mn13_gca_001662595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_24mn13_gca_001662595/cdna/Pseudogymnoascus_sp_24mn13_gca_001662595.ASM166259v1.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_103_gca_000750895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_103_gca_000750895/cdna/Pseudogymnoascus_sp_vkm_f_103_gca_000750895.DG30.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_3557_gca_000750665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudogymnoascus_sp_vkm_f_3557_gca_000750665/cdna/Pseudogymnoascus_sp_vkm_f_3557_gca_000750665.DG3.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_3775_gca_000750715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_3775_gca_000750715/cdna/Pseudogymnoascus_sp_vkm_f_3775_gca_000750715.DG4.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_3808_gca_000750675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/pseudogymnoascus_sp_vkm_f_3808_gca_000750675/cdna/Pseudogymnoascus_sp_vkm_f_3808_gca_000750675.DG1.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4246_gca_000750735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4246_gca_000750735/cdna/Pseudogymnoascus_sp_vkm_f_4246_gca_000750735.DG5.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745/cdna/Pseudogymnoascus_sp_vkm_f_4281_fw_2241__gca_000750745.DG6.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755/cdna/Pseudogymnoascus_sp_vkm_f_4513_fw_928__gca_000750755.DG7.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805/cdna/Pseudogymnoascus_sp_vkm_f_4515_fw_2607__gca_000750805.DG10.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815/cdna/Pseudogymnoascus_sp_vkm_f_4516_fw_969__gca_000750815.DG11.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875/cdna/Pseudogymnoascus_sp_vkm_f_4517_fw_2822__gca_000750875.DG13.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925/cdna/Pseudogymnoascus_sp_vkm_f_4518_fw_2643__gca_000750925.DG36.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935/cdna/Pseudogymnoascus_sp_vkm_f_4519_fw_2642__gca_000750935.DG38.cdna.all.fa.gz',
	'pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995/cdna/Pseudogymnoascus_sp_vkm_f_4520_fw_2644__gca_000750995.DG39.cdna.all.fa.gz',
	'pseudogymnoascus_sp_wsf_3629_gca_001662585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_sp_wsf_3629_gca_001662585/cdna/Pseudogymnoascus_sp_wsf_3629_gca_001662585.ASM166258v1.cdna.all.fa.gz',
	'pseudogymnoascus_verrucosus_gca_001662655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudogymnoascus_verrucosus_gca_001662655/cdna/Pseudogymnoascus_verrucosus_gca_001662655.ASM166265v1.cdna.all.fa.gz',
	'pseudoloma_neurophilia_gca_001432165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/pseudoloma_neurophilia_gca_001432165/cdna/Pseudoloma_neurophilia_gca_001432165.ASM143216v1.cdna.all.fa.gz',
	'pseudomassariella_vexata_gca_002105095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pseudomassariella_vexata_gca_002105095/cdna/Pseudomassariella_vexata_gca_002105095.Pseve2.cdna.all.fa.gz',
	'pseudomicrostroma_glucosiphilum_gca_003144135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pseudomicrostroma_glucosiphilum_gca_003144135/cdna/Pseudomicrostroma_glucosiphilum_gca_003144135.Rhodsp1.cdna.all.fa.gz',
	'pseudozyma_hubeiensis_sy62_gca_000403515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/pseudozyma_hubeiensis_sy62_gca_000403515/cdna/Pseudozyma_hubeiensis_sy62_gca_000403515.ASM40351v1.cdna.all.fa.gz',
	'pteropus_vampyrus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pteropus_vampyrus/cdna/Pteropus_vampyrus.pteVam1.cdna.abinitio.fa.gz',
	'puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275/cdna/Puccinia_coronata_var_avenae_f_sp_avenae_gca_002873275.ASM287327v1.cdna.all.fa.gz',
	'puccinia_graminis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_graminis/cdna/Puccinia_graminis.ASM14992v1.cdna.all.fa.gz',
	'puccinia_graminisug99'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_graminisug99/cdna/Puccinia_graminisug99.v1.cdna.all.fa.gz',
	'puccinia_sorghi_gca_001263375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/puccinia_sorghi_gca_001263375/cdna/Puccinia_sorghi_gca_001263375.ASM126337v1.cdna.all.fa.gz',
	'puccinia_striiformis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_striiformis/cdna/Puccinia_striiformis.PST-130_1.0.cdna.all.fa.gz',
	'puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645/cdna/Puccinia_striiformis_f_sp_tritici_pst_78_gca_001191645.P_striiformis_V1.cdna.all.fa.gz',
	'puccinia_triticina'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/puccinia_triticina/cdna/Puccinia_triticina.ASM15152v1.cdna.all.fa.gz',
	'punctularia_strigosozonata_hhb_11173_ss5_gca_000264995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/punctularia_strigosozonata_hhb_11173_ss5_gca_000264995/cdna/Punctularia_strigosozonata_hhb_11173_ss5_gca_000264995.Punctularia_strigosozonata_v1.0.cdna.all.fa.gz',
	'pundamilia_nyererei'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pundamilia_nyererei/cdna/Pundamilia_nyererei.PunNye1.0.cdna.abinitio.fa.gz',
	'purpureocillium_lilacinum_gca_001653205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/purpureocillium_lilacinum_gca_001653205/cdna/Purpureocillium_lilacinum_gca_001653205.ASM165320v1.cdna.all.fa.gz',
	'purpureocillium_lilacinum_gca_001653265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/purpureocillium_lilacinum_gca_001653265/cdna/Purpureocillium_lilacinum_gca_001653265.ASM165326v1.cdna.all.fa.gz',
	'pygocentrus_nattereri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/pygocentrus_nattereri/cdna/Pygocentrus_nattereri.Pygocentrus_nattereri-1.0.2.cdna.abinitio.fa.gz',
	'pyrenochaeta_sp_ds3say3a_gca_001644535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/pyrenochaeta_sp_ds3say3a_gca_001644535/cdna/Pyrenochaeta_sp_ds3say3a_gca_001644535.Pyrsp1.cdna.all.fa.gz',
	'pyrenophora_teres'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/pyrenophora_teres/cdna/Pyrenophora_teres.GCA_000166005.1.cdna.all.fa.gz',
	'pyrenophora_tritici_repentis_gca_003171545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003171545/cdna/Pyrenophora_tritici_repentis_gca_003171545.PtrARCrossB10.cdna.all.fa.gz',
	'pyrenophora_tritici_repentis_gca_003231325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003231325/cdna/Pyrenophora_tritici_repentis_gca_003231325.Ptr134.cdna.all.fa.gz',
	'pyrenophora_tritici_repentis_gca_003231365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003231365/cdna/Pyrenophora_tritici_repentis_gca_003231365.Ptr239.cdna.all.fa.gz',
	'pyrenophora_tritici_repentis_gca_003231425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/pyrenophora_tritici_repentis_gca_003231425/cdna/Pyrenophora_tritici_repentis_gca_003231425.Ptr86-124.cdna.all.fa.gz',
	'pyrenophora_triticirepentis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/pyrenophora_triticirepentis/cdna/Pyrenophora_triticirepentis.ASM14998v1.cdna.all.fa.gz',
	'pythium_aphanidermatum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_aphanidermatum/cdna/Pythium_aphanidermatum.pag1_scaffolds_v1.cdna.all.fa.gz',
	'pythium_arrhenomanes'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_arrhenomanes/cdna/Pythium_arrhenomanes.par_scaffolds_v1.cdna.all.fa.gz',
	'pythium_irregulare'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_irregulare/cdna/Pythium_irregulare.pir_scaffolds_v1.cdna.all.fa.gz',
	'pythium_iwayamai'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_iwayamai/cdna/Pythium_iwayamai.piw_scaffolds_v1.cdna.all.fa.gz',
	'pythium_ultimum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_ultimum/cdna/Pythium_ultimum.pug.cdna.all.fa.gz',
	'pythium_vexans'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/pythium_vexans/cdna/Pythium_vexans.pve_scaffolds_v1.cdna.all.fa.gz',
	'rachicladosporium_antarcticum_gca_002077065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rachicladosporium_antarcticum_gca_002077065/cdna/Rachicladosporium_antarcticum_gca_002077065.ASM207706v1.cdna.all.fa.gz',
	'rachicladosporium_sp_ccfee_5018_gca_002077045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rachicladosporium_sp_ccfee_5018_gca_002077045/cdna/Rachicladosporium_sp_ccfee_5018_gca_002077045.ASM207704v2.cdna.all.fa.gz',
	'ramularia_collo_cygni_gca_900074925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/ramularia_collo_cygni_gca_900074925/cdna/Ramularia_collo_cygni_gca_900074925.version_1.cdna.all.fa.gz',
	'rasamsonia_emersonii_cbs_393_64_gca_000968595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/rasamsonia_emersonii_cbs_393_64_gca_000968595/cdna/Rasamsonia_emersonii_cbs_393_64_gca_000968595.ASM96859v1.cdna.all.fa.gz',
	'rattus_norvegicus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/rattus_norvegicus/cdna/Rattus_norvegicus.Rnor_6.0.cdna.abinitio.fa.gz',
	'reticulomyxa_filosa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_rhizaria1_collection/reticulomyxa_filosa/cdna/Reticulomyxa_filosa.Reti_assembly1.0.cdna.all.fa.gz',
	'rhinocladiella_mackenziei_cbs_650_93_gca_000835555'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/rhinocladiella_mackenziei_cbs_650_93_gca_000835555/cdna/Rhinocladiella_mackenziei_cbs_650_93_gca_000835555.Rhin_mack_CBS_650_93_V1.cdna.all.fa.gz',
	'rhinopithecus_bieti'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/rhinopithecus_bieti/cdna/Rhinopithecus_bieti.ASM169854v1.cdna.abinitio.fa.gz',
	'rhinopithecus_roxellana'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/rhinopithecus_roxellana/cdna/Rhinopithecus_roxellana.Rrox_v1.cdna.abinitio.fa.gz',
	'rhizoclosmatium_globosum_gca_002104985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/rhizoclosmatium_globosum_gca_002104985/cdna/Rhizoclosmatium_globosum_gca_002104985.Rhihy1.cdna.all.fa.gz',
	'rhizoctonia_solani_123e_gca_000715385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_123e_gca_000715385/cdna/Rhizoctonia_solani_123e_gca_000715385.Rhizoctonia_solani_123E_v1.cdna.all.fa.gz',
	'rhizoctonia_solani_ag_1_ia_gca_000334115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_ag_1_ia_gca_000334115/cdna/Rhizoctonia_solani_ag_1_ia_gca_000334115.Rhisol_AG1IA_1.0.cdna.all.fa.gz',
	'rhizoctonia_solani_ag_1_ib_gca_000350255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_ag_1_ib_gca_000350255/cdna/Rhizoctonia_solani_ag_1_ib_gca_000350255.ASM35025v1.cdna.all.fa.gz',
	'rhizoctonia_solani_ag_8_wac10335_gca_000695385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_ag_8_wac10335_gca_000695385/cdna/Rhizoctonia_solani_ag_8_wac10335_gca_000695385.RSAG8-1.V1.cdna.all.fa.gz',
	'rhizoctonia_solani_gca_001286725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizoctonia_solani_gca_001286725/cdna/Rhizoctonia_solani_gca_001286725.AG22IIIB.cdna.all.fa.gz',
	'rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155/cdna/Rhizophagus_irregularis_daom_181602_daom_197198_gca_002897155.Rir_HGAP_ii_V2.cdna.all.fa.gz',
	'rhizophagus_irregularis_daom_181602_gca_000439145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_daom_181602_gca_000439145/cdna/Rhizophagus_irregularis_daom_181602_gca_000439145.ASM43914v3.cdna.all.fa.gz',
	'rhizophagus_irregularis_daom_197198w_gca_000597685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_daom_197198w_gca_000597685/cdna/Rhizophagus_irregularis_daom_197198w_gca_000597685.Rir.cdna.all.fa.gz',
	'rhizophagus_irregularis_gca_001593125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593125/cdna/Rhizophagus_irregularis_gca_001593125.ASM159312v1.cdna.all.fa.gz',
	'rhizophagus_irregularis_gca_001593145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593145/cdna/Rhizophagus_irregularis_gca_001593145.ASM159314v1.cdna.all.fa.gz',
	'rhizophagus_irregularis_gca_001593155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593155/cdna/Rhizophagus_irregularis_gca_001593155.ASM159315v1.cdna.all.fa.gz',
	'rhizophagus_irregularis_gca_001593205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizophagus_irregularis_gca_001593205/cdna/Rhizophagus_irregularis_gca_001593205.ASM159320v1.cdna.all.fa.gz',
	'rhizopogon_vinicolor_am_or11_026_gca_001658105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhizopogon_vinicolor_am_or11_026_gca_001658105/cdna/Rhizopogon_vinicolor_am_or11_026_gca_001658105.Rhivi1.cdna.all.fa.gz',
	'rhizopus_delemar_ra_99_880_gca_000149305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_delemar_ra_99_880_gca_000149305/cdna/Rhizopus_delemar_ra_99_880_gca_000149305.RO3.cdna.all.fa.gz',
	'rhizopus_microsporus_atcc_52813_gca_002708625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_atcc_52813_gca_002708625/cdna/Rhizopus_microsporus_atcc_52813_gca_002708625.Rhimi1_1.cdna.all.fa.gz',
	'rhizopus_microsporus_gca_000825725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_gca_000825725/cdna/Rhizopus_microsporus_gca_000825725.Rmicro_CBS_344.29_Allpaths-LG.cdna.all.fa.gz',
	'rhizopus_microsporus_gca_002083735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_gca_002083735/cdna/Rhizopus_microsporus_gca_002083735.Rhimi_ATCC11559_1.cdna.all.fa.gz',
	'rhizopus_microsporus_var_microsporus_gca_002083745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/rhizopus_microsporus_var_microsporus_gca_002083745/cdna/Rhizopus_microsporus_var_microsporus_gca_002083745.Rhimi_ATCC52814_1.cdna.all.fa.gz',
	'rhodnius_prolixus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/rhodnius_prolixus/cdna/Rhodnius_prolixus.RproC3.cdna.abinitio.fa.gz',
	'rhodotorula_graminis_wp1_gca_001329695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_graminis_wp1_gca_001329695/cdna/Rhodotorula_graminis_wp1_gca_001329695.Rhoba1_1.cdna.all.fa.gz',
	'rhodotorula_sp_jg_1b_gca_001541205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_sp_jg_1b_gca_001541205/cdna/Rhodotorula_sp_jg_1b_gca_001541205.Rhosp1.cdna.all.fa.gz',
	'rhodotorula_taiwanensis_gca_002922495'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_taiwanensis_gca_002922495/cdna/Rhodotorula_taiwanensis_gca_002922495.ASM292249v1.cdna.all.fa.gz',
	'rhodotorula_toruloides_atcc_204091_gca_000222205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_toruloides_atcc_204091_gca_000222205/cdna/Rhodotorula_toruloides_atcc_204091_gca_000222205.R.glutinis2.cdna.all.fa.gz',
	'rhodotorula_toruloides_np11_gca_000320785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/rhodotorula_toruloides_np11_gca_000320785/cdna/Rhodotorula_toruloides_np11_gca_000320785.RHOziaDV1.0.cdna.all.fa.gz',
	'rhynchosporium_agropyri_gca_900074905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rhynchosporium_agropyri_gca_900074905/cdna/Rhynchosporium_agropyri_gca_900074905.version_1.cdna.all.fa.gz',
	'rhynchosporium_commune_gca_900074885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rhynchosporium_commune_gca_900074885/cdna/Rhynchosporium_commune_gca_900074885.version_1.cdna.all.fa.gz',
	'rhynchosporium_secalis_gca_900074895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rhynchosporium_secalis_gca_900074895/cdna/Rhynchosporium_secalis_gca_900074895.version_1.cdna.all.fa.gz',
	'rosellinia_necatrix_gca_001445595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/rosellinia_necatrix_gca_001445595/cdna/Rosellinia_necatrix_gca_001445595.Rnecatrix_2.0.cdna.all.fa.gz',
	'rozella_allomycis_csf55_gca_000442015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_rozellomycota1_collection/rozella_allomycis_csf55_gca_000442015/cdna/Rozella_allomycis_csf55_gca_000442015.Rozella_k41_t100.cdna.all.fa.gz',
	'saccharomyces_arboricola_h_6_gca_000292725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_arboricola_h_6_gca_000292725/cdna/Saccharomyces_arboricola_h_6_gca_000292725.SacArb1.0.cdna.all.fa.gz',
	'saccharomyces_cerevisiae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/saccharomyces_cerevisiae/cdna/Saccharomyces_cerevisiae.R64-1-1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_ec1118_gca_000218975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_ec1118_gca_000218975/cdna/Saccharomyces_cerevisiae_ec1118_gca_000218975.ASM21897v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_fostersb_gca_000190255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_fostersb_gca_000190255/cdna/Saccharomyces_cerevisiae_fostersb_gca_000190255.ScFostersB_v01.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_fosterso_gca_000326005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_fosterso_gca_000326005/cdna/Saccharomyces_cerevisiae_fosterso_gca_000326005.ScFostersO_v01.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_gca_001634645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_gca_001634645/cdna/Saccharomyces_cerevisiae_gca_001634645.GLBRCY22-3.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_gca_002804325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_gca_002804325/cdna/Saccharomyces_cerevisiae_gca_002804325.ASM280432v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_kyokai_no_7_gca_000260735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_kyokai_no_7_gca_000260735/cdna/Saccharomyces_cerevisiae_kyokai_no_7_gca_000260735.K7_1.0.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_lalvin_qa23_gca_000325965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_lalvin_qa23_gca_000325965/cdna/Saccharomyces_cerevisiae_lalvin_qa23_gca_000325965.ScQA23_v01.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_p283_gca_000568295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_p283_gca_000568295/cdna/Saccharomyces_cerevisiae_p283_gca_000568295.P283.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_p301_gca_000568055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_p301_gca_000568055/cdna/Saccharomyces_cerevisiae_p301_gca_000568055.p301.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_r103_gca_000568365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_r103_gca_000568365/cdna/Saccharomyces_cerevisiae_r103_gca_000568365.R103.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_rm11_1a_gca_000149365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_rm11_1a_gca_000149365/cdna/Saccharomyces_cerevisiae_rm11_1a_gca_000149365.ASM14936v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_vin13_gca_000190215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_vin13_gca_000190215/cdna/Saccharomyces_cerevisiae_vin13_gca_000190215.ScVIN13_v01.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_vl3_gca_000190235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_vl3_gca_000190235/cdna/Saccharomyces_cerevisiae_vl3_gca_000190235.ScVL3_v01.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1078_gca_000975645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1078_gca_000975645/cdna/Saccharomyces_cerevisiae_yjm1078_gca_000975645.Sc_YJM1078_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1083_gca_000975675'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1083_gca_000975675/cdna/Saccharomyces_cerevisiae_yjm1083_gca_000975675.Sc_YJM1083_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1129_gca_000975705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1129_gca_000975705/cdna/Saccharomyces_cerevisiae_yjm1129_gca_000975705.Sc_YJM1129_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1133_gca_000976695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1133_gca_000976695/cdna/Saccharomyces_cerevisiae_yjm1133_gca_000976695.Sc_YJM1133_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1190_gca_000976725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1190_gca_000976725/cdna/Saccharomyces_cerevisiae_yjm1190_gca_000976725.Sc_YJM1190_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1199_gca_000976755'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1199_gca_000976755/cdna/Saccharomyces_cerevisiae_yjm1199_gca_000976755.Sc_YJM1199_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1202_gca_000976785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1202_gca_000976785/cdna/Saccharomyces_cerevisiae_yjm1202_gca_000976785.Sc_YJM1202_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1208_gca_000976815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1208_gca_000976815/cdna/Saccharomyces_cerevisiae_yjm1208_gca_000976815.Sc_YJM1208_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1242_gca_000976845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1242_gca_000976845/cdna/Saccharomyces_cerevisiae_yjm1242_gca_000976845.Sc_YJM1242_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1244_gca_000976875'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1244_gca_000976875/cdna/Saccharomyces_cerevisiae_yjm1244_gca_000976875.Sc_YJM1244_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1248_gca_000976905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1248_gca_000976905/cdna/Saccharomyces_cerevisiae_yjm1248_gca_000976905.Sc_YJM1248_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1250_gca_000976935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1250_gca_000976935/cdna/Saccharomyces_cerevisiae_yjm1250_gca_000976935.Sc_YJM1250_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1252_gca_000976965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1252_gca_000976965/cdna/Saccharomyces_cerevisiae_yjm1252_gca_000976965.Sc_YJM1252_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1273_gca_000976995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1273_gca_000976995/cdna/Saccharomyces_cerevisiae_yjm1273_gca_000976995.Sc_YJM1273_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1304_gca_000977025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1304_gca_000977025/cdna/Saccharomyces_cerevisiae_yjm1304_gca_000977025.Sc_YJM1304_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1307_gca_000977055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1307_gca_000977055/cdna/Saccharomyces_cerevisiae_yjm1307_gca_000977055.Sc_YJM1307_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1311_gca_000977085'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1311_gca_000977085/cdna/Saccharomyces_cerevisiae_yjm1311_gca_000977085.Sc_YJM1311_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1326_gca_000977115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1326_gca_000977115/cdna/Saccharomyces_cerevisiae_yjm1326_gca_000977115.Sc_YJM1326_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1332_gca_000977145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1332_gca_000977145/cdna/Saccharomyces_cerevisiae_yjm1332_gca_000977145.Sc_YJM1332_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1336_gca_000977175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1336_gca_000977175/cdna/Saccharomyces_cerevisiae_yjm1336_gca_000977175.Sc_YJM1336_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1338_gca_000977205'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1338_gca_000977205/cdna/Saccharomyces_cerevisiae_yjm1338_gca_000977205.Sc_YJM1338_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1341_gca_000977235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1341_gca_000977235/cdna/Saccharomyces_cerevisiae_yjm1341_gca_000977235.Sc_YJM1341_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1342_gca_000977265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1342_gca_000977265/cdna/Saccharomyces_cerevisiae_yjm1342_gca_000977265.Sc_YJM1342_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1355_gca_000977295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1355_gca_000977295/cdna/Saccharomyces_cerevisiae_yjm1355_gca_000977295.Sc_YJM1355_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1356_gca_000977325'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1356_gca_000977325/cdna/Saccharomyces_cerevisiae_yjm1356_gca_000977325.Sc_YJM1356_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1381_gca_000977355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1381_gca_000977355/cdna/Saccharomyces_cerevisiae_yjm1381_gca_000977355.Sc_YJM1381_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1383_gca_000977385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1383_gca_000977385/cdna/Saccharomyces_cerevisiae_yjm1383_gca_000977385.Sc_YJM1383_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1385_gca_000977415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1385_gca_000977415/cdna/Saccharomyces_cerevisiae_yjm1385_gca_000977415.Sc_YJM1385_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1386_gca_000977445'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1386_gca_000977445/cdna/Saccharomyces_cerevisiae_yjm1386_gca_000977445.Sc_YJM1386_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1387_gca_000977475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1387_gca_000977475/cdna/Saccharomyces_cerevisiae_yjm1387_gca_000977475.Sc_YJM1387_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1388_gca_000977505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1388_gca_000977505/cdna/Saccharomyces_cerevisiae_yjm1388_gca_000977505.Sc_YJM1388_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1389_gca_000977535'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1389_gca_000977535/cdna/Saccharomyces_cerevisiae_yjm1389_gca_000977535.Sc_YJM1389_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1399_gca_000977565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1399_gca_000977565/cdna/Saccharomyces_cerevisiae_yjm1399_gca_000977565.Sc_YJM1399_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1400_gca_000977595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1400_gca_000977595/cdna/Saccharomyces_cerevisiae_yjm1400_gca_000977595.Sc_YJM1400_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1401_gca_000977625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1401_gca_000977625/cdna/Saccharomyces_cerevisiae_yjm1401_gca_000977625.Sc_YJM1401_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1402_gca_000977655'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1402_gca_000977655/cdna/Saccharomyces_cerevisiae_yjm1402_gca_000977655.Sc_YJM1402_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1415_gca_000977685'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1415_gca_000977685/cdna/Saccharomyces_cerevisiae_yjm1415_gca_000977685.Sc_YJM1415_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1417_gca_000977715'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1417_gca_000977715/cdna/Saccharomyces_cerevisiae_yjm1417_gca_000977715.Sc_YJM1417_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1418_gca_000977745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1418_gca_000977745/cdna/Saccharomyces_cerevisiae_yjm1418_gca_000977745.Sc_YJM1418_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1419_gca_000977775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1419_gca_000977775/cdna/Saccharomyces_cerevisiae_yjm1419_gca_000977775.Sc_YJM1419_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1433_gca_000977805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1433_gca_000977805/cdna/Saccharomyces_cerevisiae_yjm1433_gca_000977805.Sc_YJM1433_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1434_gca_000977835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1434_gca_000977835/cdna/Saccharomyces_cerevisiae_yjm1434_gca_000977835.Sc_YJM1434_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1439_gca_000977865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1439_gca_000977865/cdna/Saccharomyces_cerevisiae_yjm1439_gca_000977865.Sc_YJM1439_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1443_gca_000977895'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1443_gca_000977895/cdna/Saccharomyces_cerevisiae_yjm1443_gca_000977895.Sc_YJM1443_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1444_gca_000977925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1444_gca_000977925/cdna/Saccharomyces_cerevisiae_yjm1444_gca_000977925.Sc_YJM1444_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1447_gca_000977955'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1447_gca_000977955/cdna/Saccharomyces_cerevisiae_yjm1447_gca_000977955.Sc_YJM1447_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1450_gca_000977985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1450_gca_000977985/cdna/Saccharomyces_cerevisiae_yjm1450_gca_000977985.Sc_YJM1450_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1460_gca_000978015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1460_gca_000978015/cdna/Saccharomyces_cerevisiae_yjm1460_gca_000978015.Sc_YJM1460_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1463_gca_000978045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1463_gca_000978045/cdna/Saccharomyces_cerevisiae_yjm1463_gca_000978045.Sc_YJM1463_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1477_gca_000978075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1477_gca_000978075/cdna/Saccharomyces_cerevisiae_yjm1477_gca_000978075.Sc_YJM1477_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1478_gca_000978105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1478_gca_000978105/cdna/Saccharomyces_cerevisiae_yjm1478_gca_000978105.Sc_YJM1478_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1479_gca_000978135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1479_gca_000978135/cdna/Saccharomyces_cerevisiae_yjm1479_gca_000978135.Sc_YJM1479_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1526_gca_000978165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1526_gca_000978165/cdna/Saccharomyces_cerevisiae_yjm1526_gca_000978165.Sc_YJM1526_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1527_gca_000978195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1527_gca_000978195/cdna/Saccharomyces_cerevisiae_yjm1527_gca_000978195.Sc_YJM1527_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1549_gca_000978225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1549_gca_000978225/cdna/Saccharomyces_cerevisiae_yjm1549_gca_000978225.Sc_YJM1549_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1573_gca_000978255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1573_gca_000978255/cdna/Saccharomyces_cerevisiae_yjm1573_gca_000978255.Sc_YJM1573_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1574_gca_000978285'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1574_gca_000978285/cdna/Saccharomyces_cerevisiae_yjm1574_gca_000978285.Sc_YJM1574_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1592_gca_000978315'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm1592_gca_000978315/cdna/Saccharomyces_cerevisiae_yjm1592_gca_000978315.Sc_YJM1592_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm1615_gca_000978345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm1615_gca_000978345/cdna/Saccharomyces_cerevisiae_yjm1615_gca_000978345.Sc_YJM1615_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm189_gca_000975735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm189_gca_000975735/cdna/Saccharomyces_cerevisiae_yjm189_gca_000975735.Sc_YJM189_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm193_gca_000975765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm193_gca_000975765/cdna/Saccharomyces_cerevisiae_yjm193_gca_000975765.Sc_YJM193_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm195_gca_000975585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm195_gca_000975585/cdna/Saccharomyces_cerevisiae_yjm195_gca_000975585.Sc_YJM195_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm244_gca_000975615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm244_gca_000975615/cdna/Saccharomyces_cerevisiae_yjm244_gca_000975615.Sc_YJM244_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm248_gca_000975795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm248_gca_000975795/cdna/Saccharomyces_cerevisiae_yjm248_gca_000975795.Sc_YJM248_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm270_gca_000975825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm270_gca_000975825/cdna/Saccharomyces_cerevisiae_yjm270_gca_000975825.Sc_YJM270_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm271_gca_000975855'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm271_gca_000975855/cdna/Saccharomyces_cerevisiae_yjm271_gca_000975855.Sc_YJM271_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm320_gca_000975885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm320_gca_000975885/cdna/Saccharomyces_cerevisiae_yjm320_gca_000975885.Sc_YJM320_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm326_gca_000975915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm326_gca_000975915/cdna/Saccharomyces_cerevisiae_yjm326_gca_000975915.Sc_YJM326_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm428_gca_000975945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm428_gca_000975945/cdna/Saccharomyces_cerevisiae_yjm428_gca_000975945.Sc_YJM428_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm450_gca_000975975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm450_gca_000975975/cdna/Saccharomyces_cerevisiae_yjm450_gca_000975975.Sc_YJM450_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm451_gca_000976005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm451_gca_000976005/cdna/Saccharomyces_cerevisiae_yjm451_gca_000976005.Sc_YJM451_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm453_gca_000976035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm453_gca_000976035/cdna/Saccharomyces_cerevisiae_yjm453_gca_000976035.Sc_YJM453_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm456_gca_000976065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm456_gca_000976065/cdna/Saccharomyces_cerevisiae_yjm456_gca_000976065.Sc_YJM456_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm470_gca_000976095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm470_gca_000976095/cdna/Saccharomyces_cerevisiae_yjm470_gca_000976095.Sc_YJM470_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm541_gca_000976125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm541_gca_000976125/cdna/Saccharomyces_cerevisiae_yjm541_gca_000976125.Sc_YJM541_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm554_gca_000976155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm554_gca_000976155/cdna/Saccharomyces_cerevisiae_yjm554_gca_000976155.Sc_YJM554_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm555_gca_000976185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm555_gca_000976185/cdna/Saccharomyces_cerevisiae_yjm555_gca_000976185.Sc_YJM555_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm627_gca_000976215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm627_gca_000976215/cdna/Saccharomyces_cerevisiae_yjm627_gca_000976215.Sc_YJM627_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm681_gca_000976245'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm681_gca_000976245/cdna/Saccharomyces_cerevisiae_yjm681_gca_000976245.Sc_YJM681_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm682_gca_000976275'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm682_gca_000976275/cdna/Saccharomyces_cerevisiae_yjm682_gca_000976275.Sc_YJM682_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm683_gca_000976305'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm683_gca_000976305/cdna/Saccharomyces_cerevisiae_yjm683_gca_000976305.Sc_YJM683_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm689_gca_000976335'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm689_gca_000976335/cdna/Saccharomyces_cerevisiae_yjm689_gca_000976335.Sc_YJM689_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm693_gca_000976365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm693_gca_000976365/cdna/Saccharomyces_cerevisiae_yjm693_gca_000976365.Sc_YJM693_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm789_gca_000181435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_yjm789_gca_000181435/cdna/Saccharomyces_cerevisiae_yjm789_gca_000181435.ASM18143v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm969_gca_000976395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_cerevisiae_yjm969_gca_000976395/cdna/Saccharomyces_cerevisiae_yjm969_gca_000976395.Sc_YJM969_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm972_gca_000976425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm972_gca_000976425/cdna/Saccharomyces_cerevisiae_yjm972_gca_000976425.Sc_YJM972_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm975_gca_000976455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm975_gca_000976455/cdna/Saccharomyces_cerevisiae_yjm975_gca_000976455.Sc_YJM975_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm978_gca_000976485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm978_gca_000976485/cdna/Saccharomyces_cerevisiae_yjm978_gca_000976485.Sc_YJM978_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm981_gca_000976515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm981_gca_000976515/cdna/Saccharomyces_cerevisiae_yjm981_gca_000976515.Sc_YJM981_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm984_gca_000976545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm984_gca_000976545/cdna/Saccharomyces_cerevisiae_yjm984_gca_000976545.Sc_YJM984_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm987_gca_000976575'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm987_gca_000976575/cdna/Saccharomyces_cerevisiae_yjm987_gca_000976575.Sc_YJM987_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm990_gca_000976605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm990_gca_000976605/cdna/Saccharomyces_cerevisiae_yjm990_gca_000976605.Sc_YJM990_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm993_gca_000662435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_cerevisiae_yjm993_gca_000662435/cdna/Saccharomyces_cerevisiae_yjm993_gca_000662435.Sc_YJM993_v1.cdna.all.fa.gz',
	'saccharomyces_cerevisiae_yjm996_gca_000976665'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saccharomyces_cerevisiae_yjm996_gca_000976665/cdna/Saccharomyces_cerevisiae_yjm996_gca_000976665.Sc_YJM996_v1.cdna.all.fa.gz',
	'saccharomyces_eubayanus_gca_001298625'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saccharomyces_eubayanus_gca_001298625/cdna/Saccharomyces_eubayanus_gca_001298625.SEUB3.0.cdna.all.fa.gz',
	'saccharomyces_kudriavzevii_ifo_1802_gca_000167075'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomyces_kudriavzevii_ifo_1802_gca_000167075/cdna/Saccharomyces_kudriavzevii_ifo_1802_gca_000167075.Saccharomyces_kudriavzevii_strain_IFO1802_v1.0.cdna.all.fa.gz',
	'saccharomycetaceae_sp_ashbya_aceri__gca_000412225'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/saccharomycetaceae_sp_ashbya_aceri__gca_000412225/cdna/Saccharomycetaceae_sp_ashbya_aceri__gca_000412225.ASM41222v2.cdna.all.fa.gz',
	'saimiri_boliviensis_boliviensis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/saimiri_boliviensis_boliviensis/cdna/Saimiri_boliviensis_boliviensis.SaiBol1.0.cdna.abinitio.fa.gz',
	'saitoella_complicata_nrrl_y_17804_gca_000227095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/saitoella_complicata_nrrl_y_17804_gca_000227095/cdna/Saitoella_complicata_nrrl_y_17804_gca_000227095.Scomplicata_3.0.cdna.all.fa.gz',
	'saitoella_complicata_nrrl_y_17804_gca_001661265'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/saitoella_complicata_nrrl_y_17804_gca_001661265/cdna/Saitoella_complicata_nrrl_y_17804_gca_001661265.Saico1.cdna.all.fa.gz',
	'salpingoeca_rosetta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_choanoflagellida1_collection/salpingoeca_rosetta/cdna/Salpingoeca_rosetta.Proterospongia_sp_ATCC50818.cdna.all.fa.gz',
	'salvator_merianae'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/salvator_merianae/cdna/Salvator_merianae.HLtupMer3.cdna.abinitio.fa.gz',
	'saprolegnia_diclina_vs20'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/saprolegnia_diclina_vs20/cdna/Saprolegnia_diclina_vs20.Sap_diclina_VS20_V1.cdna.all.fa.gz',
	'saprolegnia_parasitica_cbs_223_65'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/saprolegnia_parasitica_cbs_223_65/cdna/Saprolegnia_parasitica_cbs_223_65.ASM15154v2.cdna.all.fa.gz',
	'sarcophilus_harrisii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sarcophilus_harrisii/cdna/Sarcophilus_harrisii.DEVIL7.0.cdna.abinitio.fa.gz',
	'sarcoptes_scabiei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/sarcoptes_scabiei/cdna/Sarcoptes_scabiei.SscaA1.cdna.all.fa.gz',
	'scedosporium_apiospermum_gca_000732125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/scedosporium_apiospermum_gca_000732125/cdna/Scedosporium_apiospermum_gca_000732125.ScApio1.0.cdna.all.fa.gz',
	'scheffersomyces_stipitis_cbs_6054_gca_000209165'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/scheffersomyces_stipitis_cbs_6054_gca_000209165/cdna/Scheffersomyces_stipitis_cbs_6054_gca_000209165.ASM20916v1.cdna.all.fa.gz',
	'schistosoma_mansoni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/schistosoma_mansoni/cdna/Schistosoma_mansoni.ASM23792v2.cdna.all.fa.gz',
	'schizophyllum_commune_h4_8_gca_000143185'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/schizophyllum_commune_h4_8_gca_000143185/cdna/Schizophyllum_commune_h4_8_gca_000143185.v1.0.cdna.all.fa.gz',
	'schizopora_paradoxa_gca_001020605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/schizopora_paradoxa_gca_001020605/cdna/Schizopora_paradoxa_gca_001020605.Schpa1.cdna.all.fa.gz',
	'schizosaccharomyces_cryophilus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_cryophilus/cdna/Schizosaccharomyces_cryophilus.GCA_000004155.2.cdna.all.fa.gz',
	'schizosaccharomyces_japonicus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_japonicus/cdna/Schizosaccharomyces_japonicus.GCA_000149845.2.cdna.all.fa.gz',
	'schizosaccharomyces_octosporus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_octosporus/cdna/Schizosaccharomyces_octosporus.GCA_000150505.2.cdna.all.fa.gz',
	'schizosaccharomyces_pombe'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/schizosaccharomyces_pombe/cdna/Schizosaccharomyces_pombe.ASM294v2.cdna.all.fa.gz',
	'scleroderma_citrinum_foug_a_gca_000827425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/scleroderma_citrinum_foug_a_gca_000827425/cdna/Scleroderma_citrinum_foug_a_gca_000827425.Scleroderma_citrinum_Foug_A_v1.0.cdna.all.fa.gz',
	'scleropages_formosus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/scleropages_formosus/cdna/Scleropages_formosus.ASM162426v1.cdna.abinitio.fa.gz',
	'sclerotinia_borealis_f_4128_gca_000503235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sclerotinia_borealis_f_4128_gca_000503235/cdna/Sclerotinia_borealis_f_4128_gca_000503235.SBOR_1.cdna.all.fa.gz',
	'sclerotinia_sclerotiorum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/sclerotinia_sclerotiorum/cdna/Sclerotinia_sclerotiorum.ASM14694v1.cdna.all.fa.gz',
	'sclerotinia_sclerotiorum_1980_uf_70_gca_001857865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/sclerotinia_sclerotiorum_1980_uf_70_gca_001857865/cdna/Sclerotinia_sclerotiorum_1980_uf_70_gca_001857865.ASM185786v1.cdna.all.fa.gz',
	'scophthalmus_maximus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/scophthalmus_maximus/cdna/Scophthalmus_maximus.ASM318616v1.cdna.abinitio.fa.gz',
	'selaginella_moellendorffii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/selaginella_moellendorffii/cdna/Selaginella_moellendorffii.v1.0.cdna.all.fa.gz',
	'serendipita_indica_dsm_11827_gca_000313545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/serendipita_indica_dsm_11827_gca_000313545/cdna/Serendipita_indica_dsm_11827_gca_000313545.ASM31354v1.cdna.all.fa.gz',
	'serendipita_vermifera_maff_305830_gca_000827415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/serendipita_vermifera_maff_305830_gca_000827415/cdna/Serendipita_vermifera_maff_305830_gca_000827415.Sebacina_vermifera_MAFF_305830_v1.0.cdna.all.fa.gz',
	'serinus_canaria'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/serinus_canaria/cdna/Serinus_canaria.SCA1.cdna.abinitio.fa.gz',
	'seriola_dumerili'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/seriola_dumerili/cdna/Seriola_dumerili.Sdu_1.0.cdna.abinitio.fa.gz',
	'seriola_lalandi_dorsalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/seriola_lalandi_dorsalis/cdna/Seriola_lalandi_dorsalis.Sedor1.cdna.abinitio.fa.gz',
	'serpula_lacrymans_var_lacrymans_s7_3_gca_000218725'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/serpula_lacrymans_var_lacrymans_s7_3_gca_000218725/cdna/Serpula_lacrymans_var_lacrymans_s7_3_gca_000218725.v1.0.cdna.all.fa.gz',
	'setaria_italica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/setaria_italica/cdna/Setaria_italica.Setaria_italica_v2.0.cdna.all.fa.gz',
	'sistotremastrum_niveocremeum_hhb9708_gca_001630475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sistotremastrum_niveocremeum_hhb9708_gca_001630475/cdna/Sistotremastrum_niveocremeum_hhb9708_gca_001630475.Sisni1.cdna.all.fa.gz',
	'sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355/cdna/Sistotremastrum_suecicum_hhb10207_ss_3_gca_001632355.Sissu1.cdna.all.fa.gz',
	'solanum_lycopersicum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/solanum_lycopersicum/cdna/Solanum_lycopersicum.SL3.0.cdna.all.fa.gz',
	'solanum_tuberosum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/solanum_tuberosum/cdna/Solanum_tuberosum.SolTub_3.0.cdna.all.fa.gz',
	'solenopsis_invicta'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/solenopsis_invicta/cdna/Solenopsis_invicta.Si_gnG.cdna.all.fa.gz',
	'sordaria_macrospora_k_hell_gca_000182805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sordaria_macrospora_k_hell_gca_000182805/cdna/Sordaria_macrospora_k_hell_gca_000182805.ASM18280v2.cdna.all.fa.gz',
	'sorex_araneus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sorex_araneus/cdna/Sorex_araneus.COMMON_SHREW1.cdna.abinitio.fa.gz',
	'sorghum_bicolor'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/sorghum_bicolor/cdna/Sorghum_bicolor.Sorghum_bicolor_NCBIv3.cdna.all.fa.gz',
	'spathaspora_passalidarum_nrrl_y_27907_gca_000223485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/spathaspora_passalidarum_nrrl_y_27907_gca_000223485/cdna/Spathaspora_passalidarum_nrrl_y_27907_gca_000223485.Spathaspora_passalidarum_v2.0.cdna.all.fa.gz',
	'spermophilus_dauricus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/spermophilus_dauricus/cdna/Spermophilus_dauricus.ASM240643v1.cdna.abinitio.fa.gz',
	'sphaerobolus_stellatus_ss14_gca_000827215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sphaerobolus_stellatus_ss14_gca_000827215/cdna/Sphaerobolus_stellatus_ss14_gca_000827215.Sphst1.cdna.all.fa.gz',
	'sphaeroforma_arctica_jp610'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_ichthyosporea1_collection/sphaeroforma_arctica_jp610/cdna/Sphaeroforma_arctica_jp610.Spha_arctica_JP610_V1.cdna.all.fa.gz',
	'sphaerulina_musiva_so2202_gca_000320565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sphaerulina_musiva_so2202_gca_000320565/cdna/Sphaerulina_musiva_so2202_gca_000320565.Septoria_musiva_SO2202_v1.0.cdna.all.fa.gz',
	'sphenodon_punctatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sphenodon_punctatus/cdna/Sphenodon_punctatus.ASM311381v1.cdna.abinitio.fa.gz',
	'spironucleus_salmonicida'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_fornicata1_collection/spironucleus_salmonicida/cdna/Spironucleus_salmonicida.SSK3.0.cdna.all.fa.gz',
	'spizellomyces_punctatus_daom_br117_gca_000182565'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_chytridiomycota1_collection/spizellomyces_punctatus_daom_br117_gca_000182565/cdna/Spizellomyces_punctatus_daom_br117_gca_000182565.S_punctatus_V1.cdna.all.fa.gz',
	'sporisorium_reilianum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/sporisorium_reilianum/cdna/Sporisorium_reilianum.GCA_000230245.1.cdna.all.fa.gz',
	'sporisorium_reilianum_f_sp_reilianum_gca_900162835'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/sporisorium_reilianum_f_sp_reilianum_gca_900162835/cdna/Sporisorium_reilianum_f_sp_reilianum_gca_900162835.version_2.cdna.all.fa.gz',
	'sporothrix_brasiliensis_5110_gca_000820605'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/sporothrix_brasiliensis_5110_gca_000820605/cdna/Sporothrix_brasiliensis_5110_gca_000820605.S_brasiliensis_5110_v1.cdna.all.fa.gz',
	'sporothrix_insectorum_rcef_264_gca_001636815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/sporothrix_insectorum_rcef_264_gca_001636815/cdna/Sporothrix_insectorum_rcef_264_gca_001636815.SPI_1.0.cdna.all.fa.gz',
	'sporothrix_schenckii_1099_18_gca_000961545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/sporothrix_schenckii_1099_18_gca_000961545/cdna/Sporothrix_schenckii_1099_18_gca_000961545.S_schenckii_v1.cdna.all.fa.gz',
	'sporothrix_schenckii_atcc_58251_gca_000474925'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/sporothrix_schenckii_atcc_58251_gca_000474925/cdna/Sporothrix_schenckii_atcc_58251_gca_000474925.Spor_sche_ATCC58251_V1.cdna.all.fa.gz',
	'spraguea_lophii_42_110_gca_000430065'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/spraguea_lophii_42_110_gca_000430065/cdna/Spraguea_lophii_42_110_gca_000430065.Sprlop1.0.cdna.all.fa.gz',
	'stachybotrys_chartarum_ibt_40288_gca_000732765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/stachybotrys_chartarum_ibt_40288_gca_000732765/cdna/Stachybotrys_chartarum_ibt_40288_gca_000732765.S40288v1.cdna.all.fa.gz',
	'stachybotrys_chlorohalonata_ibt_40285_gca_000732775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/stachybotrys_chlorohalonata_ibt_40285_gca_000732775/cdna/Stachybotrys_chlorohalonata_ibt_40285_gca_000732775.S40285v1.cdna.all.fa.gz',
	'stagonospora_sp_src1lsm3a_gca_001644525'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/stagonospora_sp_src1lsm3a_gca_001644525/cdna/Stagonospora_sp_src1lsm3a_gca_001644525.Stasp1.cdna.all.fa.gz',
	'stegastes_partitus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/stegastes_partitus/cdna/Stegastes_partitus.Stegastes_partitus-1.0.2.cdna.abinitio.fa.gz',
	'stegodyphus_mimosarum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/stegodyphus_mimosarum/cdna/Stegodyphus_mimosarum.Stegodyphus_mimosarum_v1.cdna.all.fa.gz',
	'stemphylium_lycopersici_gca_001191545'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/stemphylium_lycopersici_gca_001191545/cdna/Stemphylium_lycopersici_gca_001191545.ASM119154v1.cdna.all.fa.gz',
	'stereum_hirsutum_fp_91666_ss1_gca_000264905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/stereum_hirsutum_fp_91666_ss1_gca_000264905/cdna/Stereum_hirsutum_fp_91666_ss1_gca_000264905.Stehi1.cdna.all.fa.gz',
	'strigamia_maritima'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/strigamia_maritima/cdna/Strigamia_maritima.Smar1.cdna.all.fa.gz',
	'strigomonas_culicis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/strigomonas_culicis/cdna/Strigomonas_culicis.Strigomonas_culicis_Genome.cdna.all.fa.gz',
	'strongylocentrotus_purpuratus'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/strongylocentrotus_purpuratus/cdna/Strongylocentrotus_purpuratus.Spur_3.1.cdna.abinitio.fa.gz',
	'strongyloides_ratti'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/strongyloides_ratti/cdna/Strongyloides_ratti.S_ratti_ED321_v5_0_4.cdna.all.fa.gz',
	'sugiyamaella_lignohabitans_gca_001640025'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/sugiyamaella_lignohabitans_gca_001640025/cdna/Sugiyamaella_lignohabitans_gca_001640025.ASM164002v2.cdna.all.fa.gz',
	'suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415/cdna/Suhomyces_tanzawaensis_nrrl_y_17324_gca_001661415.Canta1.cdna.all.fa.gz',
	'suillus_luteus_uh_slu_lm8_n1_gca_000827255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/suillus_luteus_uh_slu_lm8_n1_gca_000827255/cdna/Suillus_luteus_uh_slu_lm8_n1_gca_000827255.Suilu1.cdna.all.fa.gz',
	'sus_scrofa'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/sus_scrofa/cdna/Sus_scrofa.Sscrofa11.1.cdna.abinitio.fa.gz',
	'syncephalastrum_racemosum_gca_002105135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_mucoromycota1_collection/syncephalastrum_racemosum_gca_002105135/cdna/Syncephalastrum_racemosum_gca_002105135.Synrac1.cdna.all.fa.gz',
	'taeniopygia_guttata'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/taeniopygia_guttata/cdna/Taeniopygia_guttata.taeGut3.2.4.cdna.abinitio.fa.gz',
	'takifugu_rubripes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/takifugu_rubripes/cdna/Takifugu_rubripes.FUGU5.cdna.abinitio.fa.gz',
	'talaromyces_cellulolyticus_gca_000829775'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/talaromyces_cellulolyticus_gca_000829775/cdna/Talaromyces_cellulolyticus_gca_000829775.TcelY94_1.0.cdna.all.fa.gz',
	'talaromyces_islandicus_gca_000985935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/talaromyces_islandicus_gca_000985935/cdna/Talaromyces_islandicus_gca_000985935.PIS.cdna.all.fa.gz',
	'talaromyces_marneffei_atcc_18224_gca_000001985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/talaromyces_marneffei_atcc_18224_gca_000001985/cdna/Talaromyces_marneffei_atcc_18224_gca_000001985.JCVI-PMFA1-2.0.cdna.all.fa.gz',
	'talaromyces_marneffei_pm1_gca_000750115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/talaromyces_marneffei_pm1_gca_000750115/cdna/Talaromyces_marneffei_pm1_gca_000750115.ASM75011v1.cdna.all.fa.gz',
	'talaromyces_stipitatus_atcc_10500_gca_000003125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/talaromyces_stipitatus_atcc_10500_gca_000003125/cdna/Talaromyces_stipitatus_atcc_10500_gca_000003125.JCVI-TSTA1-3.0.cdna.all.fa.gz',
	'teleopsis_dalmanni'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/teleopsis_dalmanni/cdna/Teleopsis_dalmanni.Tel_dalmanni_2A_v1.0.cdna.all.fa.gz',
	'termitomyces_sp_j132_gca_001263195'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/termitomyces_sp_j132_gca_001263195/cdna/Termitomyces_sp_j132_gca_001263195.TerJ1321.0.cdna.all.fa.gz',
	'testicularia_cyperi_gca_003144125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/testicularia_cyperi_gca_003144125/cdna/Testicularia_cyperi_gca_003144125.Tescy1.cdna.all.fa.gz',
	'tetrahymena_thermophila'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/tetrahymena_thermophila/cdna/Tetrahymena_thermophila.JCVI-TTA1-2.2.cdna.all.fa.gz',
	'tetranychus_urticae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/tetranychus_urticae/cdna/Tetranychus_urticae.ASM23943v1.cdna.all.fa.gz',
	'tetraodon_nigroviridis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/tetraodon_nigroviridis/cdna/Tetraodon_nigroviridis.TETRAODON8.cdna.abinitio.fa.gz',
	'tetrapisispora_blattae_cbs_6284_gca_000315915'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/tetrapisispora_blattae_cbs_6284_gca_000315915/cdna/Tetrapisispora_blattae_cbs_6284_gca_000315915.ASM31591v1.cdna.all.fa.gz',
	'tetrapisispora_phaffii_cbs_4417_gca_000236905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/tetrapisispora_phaffii_cbs_4417_gca_000236905/cdna/Tetrapisispora_phaffii_cbs_4417_gca_000236905.ASM23690v1.cdna.all.fa.gz',
	'thalassiosira_oceanica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/thalassiosira_oceanica/cdna/Thalassiosira_oceanica.ThaOc_1.0.cdna.all.fa.gz',
	'thalassiosira_oceanica_ccmp1005'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_stramenopiles1_collection/thalassiosira_oceanica_ccmp1005/cdna/Thalassiosira_oceanica_ccmp1005.ASM19663v1.cdna.all.fa.gz',
	'thalassiosira_pseudonana'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/thalassiosira_pseudonana/cdna/Thalassiosira_pseudonana.ASM14940v2.cdna.all.fa.gz',
	'thecamonas_trahens_atcc_50062'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_apusozoa1_collection/thecamonas_trahens_atcc_50062/cdna/Thecamonas_trahens_atcc_50062.TheTra_May2010.cdna.all.fa.gz',
	'theileria_annulata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_annulata/cdna/Theileria_annulata.ASM322v1.cdna.all.fa.gz',
	'theileria_equi_strain_wa'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_equi_strain_wa/cdna/Theileria_equi_strain_wa.JCVI-bewag-v1.1.cdna.all.fa.gz',
	'theileria_orientalis_strain_shintoku'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_orientalis_strain_shintoku/cdna/Theileria_orientalis_strain_shintoku.ASM74089v1.cdna.all.fa.gz',
	'theileria_parva'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/theileria_parva/cdna/Theileria_parva.ASM16536v1.cdna.all.fa.gz',
	'thelohanellus_kitauei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/thelohanellus_kitauei/cdna/Thelohanellus_kitauei.ASM82789v1.cdna.all.fa.gz',
	'theobroma_cacao'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/theobroma_cacao/cdna/Theobroma_cacao.Theobroma_cacao_20110822.cdna.all.fa.gz',
	'thermothelomyces_thermophila_atcc_42464_gca_000226095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/thermothelomyces_thermophila_atcc_42464_gca_000226095/cdna/Thermothelomyces_thermophila_atcc_42464_gca_000226095.ASM22609v1.cdna.all.fa.gz',
	'theropithecus_gelada'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/theropithecus_gelada/cdna/Theropithecus_gelada.Tgel_1.0.cdna.abinitio.fa.gz',
	'thielavia_terrestris_nrrl_8126_gca_000226115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/thielavia_terrestris_nrrl_8126_gca_000226115/cdna/Thielavia_terrestris_nrrl_8126_gca_000226115.ASM22611v1.cdna.all.fa.gz',
	'thielaviopsis_punctulata_gca_000968615'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/thielaviopsis_punctulata_gca_000968615/cdna/Thielaviopsis_punctulata_gca_000968615.ASM96861v1.cdna.all.fa.gz',
	'tilletia_controversa_gca_001645045'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletia_controversa_gca_001645045/cdna/Tilletia_controversa_gca_001645045.ASM164504v1.cdna.all.fa.gz',
	'tilletia_walkeri_gca_001645055'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletia_walkeri_gca_001645055/cdna/Tilletia_walkeri_gca_001645055.ASM164505v1.cdna.all.fa.gz',
	'tilletiaria_anomala_ubc_951_gca_000711695'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletiaria_anomala_ubc_951_gca_000711695/cdna/Tilletiaria_anomala_ubc_951_gca_000711695.Tilletiaria_anomala_UBC_951_v1.0.cdna.all.fa.gz',
	'tilletiopsis_washingtonensis_gca_003144115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tilletiopsis_washingtonensis_gca_003144115/cdna/Tilletiopsis_washingtonensis_gca_003144115.Tilwa1.cdna.all.fa.gz',
	'tolypocladium_ophioglossoides_cbs_100239_gca_001189435'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/tolypocladium_ophioglossoides_cbs_100239_gca_001189435/cdna/Tolypocladium_ophioglossoides_cbs_100239_gca_001189435.Tophv1.0.cdna.all.fa.gz',
	'torrubiella_hemipterigena_gca_000825705'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/torrubiella_hemipterigena_gca_000825705/cdna/Torrubiella_hemipterigena_gca_000825705.Allpaths-LG.cdna.all.fa.gz',
	'tortispora_caseinolytica_nrrl_y_17796_gca_001661475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/tortispora_caseinolytica_nrrl_y_17796_gca_001661475/cdna/Tortispora_caseinolytica_nrrl_y_17796_gca_001661475.Canca1.cdna.all.fa.gz',
	'torulaspora_delbrueckii_gca_000243375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/torulaspora_delbrueckii_gca_000243375/cdna/Torulaspora_delbrueckii_gca_000243375.ASM24337v1.cdna.all.fa.gz',
	'toxoplasma_gondii'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/toxoplasma_gondii/cdna/Toxoplasma_gondii.TGA4.cdna.all.fa.gz',
	'toxoplasma_gondii_ari'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_ari/cdna/Toxoplasma_gondii_ari.TGARI_v2.cdna.all.fa.gz',
	'toxoplasma_gondii_fou'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_fou/cdna/Toxoplasma_gondii_fou.TGFOU1v02.cdna.all.fa.gz',
	'toxoplasma_gondii_gab2_2007_gal_dom2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_gab2_2007_gal_dom2/cdna/Toxoplasma_gondii_gab2_2007_gal_dom2.TGGABGALDOM2_v2.1.cdna.all.fa.gz',
	'toxoplasma_gondii_gt1'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_gt1/cdna/Toxoplasma_gondii_gt1.TGGT1.cdna.all.fa.gz',
	'toxoplasma_gondii_mas'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_mas/cdna/Toxoplasma_gondii_mas.TGMAS1_v2.cdna.all.fa.gz',
	'toxoplasma_gondii_p89'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_p89/cdna/Toxoplasma_gondii_p89.TGP89A_v02.cdna.all.fa.gz',
	'toxoplasma_gondii_rub'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_rub/cdna/Toxoplasma_gondii_rub.TGRUB_v2.cdna.all.fa.gz',
	'toxoplasma_gondii_tgcatprc2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_tgcatprc2/cdna/Toxoplasma_gondii_tgcatprc2.TGCATPRC2_v2.cdna.all.fa.gz',
	'toxoplasma_gondii_vand'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_vand/cdna/Toxoplasma_gondii_vand.TGVAND_v2.cdna.all.fa.gz',
	'toxoplasma_gondii_veg'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/toxoplasma_gondii_veg/cdna/Toxoplasma_gondii_veg.TGVEG.cdna.all.fa.gz',
	'trachipleistophora_hominis_gca_000316135'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/trachipleistophora_hominis_gca_000316135/cdna/Trachipleistophora_hominis_gca_000316135.ASM31613v1.cdna.all.fa.gz',
	'trametes_cinnabarina_gca_000765035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_cinnabarina_gca_000765035/cdna/Trametes_cinnabarina_gca_000765035.BN946.cdna.all.fa.gz',
	'trametes_coccinea_brfm310_gca_002092935'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_coccinea_brfm310_gca_002092935/cdna/Trametes_coccinea_brfm310_gca_002092935.Pycco1.cdna.all.fa.gz',
	'trametes_pubescens_gca_001895945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_pubescens_gca_001895945/cdna/Trametes_pubescens_gca_001895945.ASM189594v1.cdna.all.fa.gz',
	'trametes_versicolor_fp_101664_ss1_gca_000271585'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trametes_versicolor_fp_101664_ss1_gca_000271585/cdna/Trametes_versicolor_fp_101664_ss1_gca_000271585.Trametes_versicolor_v1.0.cdna.all.fa.gz',
	'tremella_mesenterica_dsm_1558_gca_000271645'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tremella_mesenterica_dsm_1558_gca_000271645/cdna/Tremella_mesenterica_dsm_1558_gca_000271645.Treme1.cdna.all.fa.gz',
	'tribolium_castaneum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/tribolium_castaneum/cdna/Tribolium_castaneum.Tcas5.2.cdna.all.fa.gz',
	'trichoderma_asperellum_cbs_433_97_gca_003025105'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_asperellum_cbs_433_97_gca_003025105/cdna/Trichoderma_asperellum_cbs_433_97_gca_003025105.Trias_v._1.0.cdna.all.fa.gz',
	'trichoderma_atroviride_imi_206040_gca_000171015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichoderma_atroviride_imi_206040_gca_000171015/cdna/Trichoderma_atroviride_imi_206040_gca_000171015.TRIAT_v2.0.cdna.all.fa.gz',
	'trichoderma_citrinoviride_gca_003025115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_citrinoviride_gca_003025115/cdna/Trichoderma_citrinoviride_gca_003025115.Trici_v4.0.cdna.all.fa.gz',
	'trichoderma_harzianum_cbs_226_95_gca_003025095'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_harzianum_cbs_226_95_gca_003025095/cdna/Trichoderma_harzianum_cbs_226_95_gca_003025095.Allpaths-LG_v._01_03_12.cdna.all.fa.gz',
	'trichoderma_harzianum_gca_000988865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/trichoderma_harzianum_gca_000988865/cdna/Trichoderma_harzianum_gca_000988865.ASM98886v1.cdna.all.fa.gz',
	'trichoderma_longibrachiatum_atcc_18648_gca_003025155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichoderma_longibrachiatum_atcc_18648_gca_003025155/cdna/Trichoderma_longibrachiatum_atcc_18648_gca_003025155.Trilo3.cdna.all.fa.gz',
	'trichoderma_reesei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/trichoderma_reesei/cdna/Trichoderma_reesei.GCA_000167675.2.cdna.all.fa.gz',
	'trichoderma_reesei_rut_c_30_gca_000513815'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichoderma_reesei_rut_c_30_gca_000513815/cdna/Trichoderma_reesei_rut_c_30_gca_000513815.TrireRUTC30_1.cdna.all.fa.gz',
	'trichoderma_virens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/trichoderma_virens/cdna/Trichoderma_virens.ASM17099v1.cdna.all.fa.gz',
	'trichomonas_vaginalis_g3'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_parabasalia1_collection/trichomonas_vaginalis_g3/cdna/Trichomonas_vaginalis_g3.ASM282v1.cdna.all.fa.gz',
	'trichophyton_benhamiae_cbs_112371_gca_000151125'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_benhamiae_cbs_112371_gca_000151125/cdna/Trichophyton_benhamiae_cbs_112371_gca_000151125.ASM15112v2.cdna.all.fa.gz',
	'trichophyton_equinum_cbs_127_97_gca_000151175'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_equinum_cbs_127_97_gca_000151175/cdna/Trichophyton_equinum_cbs_127_97_gca_000151175.ASM15117v1.cdna.all.fa.gz',
	'trichophyton_interdigitale_h6_gca_000616785'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_interdigitale_h6_gca_000616785/cdna/Trichophyton_interdigitale_h6_gca_000616785.Tric_rubr_H6_V2.cdna.all.fa.gz',
	'trichophyton_interdigitale_mr816_gca_000622975'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_interdigitale_mr816_gca_000622975/cdna/Trichophyton_interdigitale_mr816_gca_000622975.Tric_inte_MR816_V1.cdna.all.fa.gz',
	'trichophyton_mentagrophytes_gca_003118255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/trichophyton_mentagrophytes_gca_003118255/cdna/Trichophyton_mentagrophytes_gca_003118255.ABySS_70bp_45k_152cov_v1.cdna.all.fa.gz',
	'trichophyton_rubrum_cbs_100081_gca_000616805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_100081_gca_000616805/cdna/Trichophyton_rubrum_cbs_100081_gca_000616805.Tric_fish_CBS_100081_V1.cdna.all.fa.gz',
	'trichophyton_rubrum_cbs_118892_gca_000151425'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_118892_gca_000151425/cdna/Trichophyton_rubrum_cbs_118892_gca_000151425.ASM15142v1.cdna.all.fa.gz',
	'trichophyton_rubrum_cbs_202_88_gca_000616985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_202_88_gca_000616985/cdna/Trichophyton_rubrum_cbs_202_88_gca_000616985.Tric_raub_CBS_202_88_V2.cdna.all.fa.gz',
	'trichophyton_rubrum_cbs_288_86_gca_000616825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_288_86_gca_000616825/cdna/Trichophyton_rubrum_cbs_288_86_gca_000616825.Tric_fish_CBS_288_86_V1.cdna.all.fa.gz',
	'trichophyton_rubrum_cbs_289_86_gca_000616845'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_289_86_gca_000616845/cdna/Trichophyton_rubrum_cbs_289_86_gca_000616845.Tric_kane_CBS_289_86_V1.cdna.all.fa.gz',
	'trichophyton_rubrum_cbs_735_88_gca_000616965'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_cbs_735_88_gca_000616965/cdna/Trichophyton_rubrum_cbs_735_88_gca_000616965.Tric_megn_CBS_735_88_V1.cdna.all.fa.gz',
	'trichophyton_rubrum_d6_gca_000622995'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_d6_gca_000622995/cdna/Trichophyton_rubrum_d6_gca_000622995.Tric_rubr_D6_V1.cdna.all.fa.gz',
	'trichophyton_rubrum_mr1448_gca_000616905'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_mr1448_gca_000616905/cdna/Trichophyton_rubrum_mr1448_gca_000616905.Tric_rubr_M1448_V1.cdna.all.fa.gz',
	'trichophyton_rubrum_mr1459_gca_000616945'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_mr1459_gca_000616945/cdna/Trichophyton_rubrum_mr1459_gca_000616945.Tric_rubr_MR1459_V1.cdna.all.fa.gz',
	'trichophyton_rubrum_mr850_gca_000616765'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_rubrum_mr850_gca_000616765/cdna/Trichophyton_rubrum_mr850_gca_000616765.Tric_rubr_MR-850_V1.cdna.all.fa.gz',
	'trichophyton_soudanense_cbs_452_61_gca_000616865'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_soudanense_cbs_452_61_gca_000616865/cdna/Trichophyton_soudanense_cbs_452_61_gca_000616865.Tric_soud_CBS_452_61_V1.cdna.all.fa.gz',
	'trichophyton_tonsurans_cbs_112818_gca_000151455'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_tonsurans_cbs_112818_gca_000151455/cdna/Trichophyton_tonsurans_cbs_112818_gca_000151455.ASM15145v1.cdna.all.fa.gz',
	'trichophyton_verrucosum_hki_0517_gca_000151505'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/trichophyton_verrucosum_hki_0517_gca_000151505/cdna/Trichophyton_verrucosum_hki_0517_gca_000151505.ASM15150v1.cdna.all.fa.gz',
	'trichoplax_adhaerens'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/trichoplax_adhaerens/cdna/Trichoplax_adhaerens.ASM15027v1.cdna.all.fa.gz',
	'trichosporon_asahii_var_asahii_cbs_2479_gca_000293215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trichosporon_asahii_var_asahii_cbs_2479_gca_000293215/cdna/Trichosporon_asahii_var_asahii_cbs_2479_gca_000293215.Trichosporon_asahii_1.cdna.all.fa.gz',
	'trichosporon_asahii_var_asahii_cbs_8904_gca_000299215'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/trichosporon_asahii_var_asahii_cbs_8904_gca_000299215/cdna/Trichosporon_asahii_var_asahii_cbs_8904_gca_000299215.ASM29921v2.cdna.all.fa.gz',
	'trifolium_pratense'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/trifolium_pratense/cdna/Trifolium_pratense.Trpr.cdna.all.fa.gz',
	'triticum_aestivum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/triticum_aestivum/cdna/Triticum_aestivum.IWGSC.cdna.all.fa.gz',
	'triticum_dicoccoides'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/triticum_dicoccoides/cdna/Triticum_dicoccoides.WEWSeq_v.1.0.cdna.all.fa.gz',
	'triticum_urartu'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/triticum_urartu/cdna/Triticum_urartu.ASM34745v1.cdna.all.fa.gz',
	'trypanosoma_brucei'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/trypanosoma_brucei/cdna/Trypanosoma_brucei.TryBru_Apr2005_chr11.cdna.all.fa.gz',
	'trypanosoma_cruzi'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/trypanosoma_cruzi/cdna/Trypanosoma_cruzi.ASM20906v1.cdna.all.fa.gz',
	'trypanosoma_cruzi_dm28c'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_euglenozoa1_collection/trypanosoma_cruzi_dm28c/cdna/Trypanosoma_cruzi_dm28c.T.cruzi.Dm28c_v01.cdna.all.fa.gz',
	'tsuchiyaea_wingfieldii_cbs_7118_gca_001720155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tsuchiyaea_wingfieldii_cbs_7118_gca_001720155/cdna/Tsuchiyaea_wingfieldii_cbs_7118_gca_001720155.Tsuc_wing_CBS7118_V1.cdna.all.fa.gz',
	'tuber_borchii_gca_003070745'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/tuber_borchii_gca_003070745/cdna/Tuber_borchii_gca_003070745.Tubbor1.cdna.all.fa.gz',
	'tuber_magnatum_gca_003182015'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/tuber_magnatum_gca_003182015/cdna/Tuber_magnatum_gca_003182015.ASM318201v1.cdna.all.fa.gz',
	'tuber_melanosporum'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/tuber_melanosporum/cdna/Tuber_melanosporum.ASM15164v1.cdna.all.fa.gz',
	'tulasnella_calospora_mut_4182_gca_000827465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/tulasnella_calospora_mut_4182_gca_000827465/cdna/Tulasnella_calospora_mut_4182_gca_000827465.Tulasnella_calospora_AL13_4D_v1.0.cdna.all.fa.gz',
	'tupaia_belangeri'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/tupaia_belangeri/cdna/Tupaia_belangeri.TREESHREW.cdna.abinitio.fa.gz',
	'tursiops_truncatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/tursiops_truncatus/cdna/Tursiops_truncatus.turTru1.cdna.abinitio.fa.gz',
	'uncinocarpus_reesii_1704_gca_000003515'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/uncinocarpus_reesii_1704_gca_000003515/cdna/Uncinocarpus_reesii_1704_gca_000003515.ASM351v2.cdna.all.fa.gz',
	'urocitellus_parryii'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/urocitellus_parryii/cdna/Urocitellus_parryii.ASM342692v1.cdna.abinitio.fa.gz',
	'ursus_americanus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ursus_americanus/cdna/Ursus_americanus.ASM334442v1.cdna.abinitio.fa.gz',
	'ursus_maritimus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/ursus_maritimus/cdna/Ursus_maritimus.UrsMar_1.0.cdna.abinitio.fa.gz',
	'ustilaginoidea_virens_gca_000687475'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/ustilaginoidea_virens_gca_000687475/cdna/Ustilaginoidea_virens_gca_000687475.Assembly_for_version_1_of_the_Villosiclava_virens_genome.cdna.all.fa.gz',
	'ustilaginomycotina_sp_sa_807_gca_003144235'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ustilaginomycotina_sp_sa_807_gca_003144235/cdna/Ustilaginomycotina_sp_sa_807_gca_003144235.Ustsp1.cdna.all.fa.gz',
	'ustilago_bromivora_gca_900080155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ustilago_bromivora_gca_900080155/cdna/Ustilago_bromivora_gca_900080155.UBRO_v3.cdna.all.fa.gz',
	'ustilago_hordei_gca_000286035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/ustilago_hordei_gca_000286035/cdna/Ustilago_hordei_gca_000286035.ASM28603v1.cdna.all.fa.gz',
	'ustilago_maydis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/ustilago_maydis/cdna/Ustilago_maydis.Umaydis521_2.0.cdna.all.fa.gz',
	'valsa_mali_gca_000818155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/valsa_mali_gca_000818155/cdna/Valsa_mali_gca_000818155.ASM81815v1.cdna.all.fa.gz',
	'valsa_mali_var_pyri_gca_000813385'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/valsa_mali_var_pyri_gca_000813385/cdna/Valsa_mali_var_pyri_gca_000813385.ASM81338v1.cdna.all.fa.gz',
	'vanderwaltozyma_polyspora_dsm_70294_gca_000150035'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/vanderwaltozyma_polyspora_dsm_70294_gca_000150035/cdna/Vanderwaltozyma_polyspora_dsm_70294_gca_000150035.ASM15003v1.cdna.all.fa.gz',
	'vavraia_culicis_subsp_floridensis_gca_000192795'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/vavraia_culicis_subsp_floridensis_gca_000192795/cdna/Vavraia_culicis_subsp_floridensis_gca_000192795.Vavr_culi_floridensis_V1.cdna.all.fa.gz',
	'verruconis_gallopava_gca_000836295'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/verruconis_gallopava_gca_000836295/cdna/Verruconis_gallopava_gca_000836295.O_gall_CBS43764.cdna.all.fa.gz',
	'verticillium_alfalfae_vams_102_gca_000150825'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/verticillium_alfalfae_vams_102_gca_000150825/cdna/Verticillium_alfalfae_vams_102_gca_000150825.ASM15082v1.cdna.all.fa.gz',
	'verticillium_dahliae'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/verticillium_dahliae/cdna/Verticillium_dahliae.ASM15067v2.cdna.all.fa.gz',
	'verticillium_dahliaejr2'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/verticillium_dahliaejr2/cdna/Verticillium_dahliaejr2.VDAG_JR2v.4.0.cdna.all.fa.gz',
	'verticillium_longisporum_gca_001268145'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/verticillium_longisporum_gca_001268145/cdna/Verticillium_longisporum_gca_001268145.vl1.denovo.v1.cdna.all.fa.gz',
	'vicugna_pacos'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/vicugna_pacos/cdna/Vicugna_pacos.vicPac1.cdna.abinitio.fa.gz',
	'vigna_angularis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/vigna_angularis/cdna/Vigna_angularis.Vigan1.1.cdna.all.fa.gz',
	'vigna_radiata'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/vigna_radiata/cdna/Vigna_radiata.Vradiata_ver6.cdna.all.fa.gz',
	'vitis_vinifera'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/vitis_vinifera/cdna/Vitis_vinifera.12X.cdna.all.fa.gz',
	'vitrella_brassicaformis_ccmp3155'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/protists/fasta/protists_alveolata1_collection/vitrella_brassicaformis_ccmp3155/cdna/Vitrella_brassicaformis_ccmp3155.Vbrassicaformis.cdna.all.fa.gz',
	'vittaforma_corneae_atcc_50505_gca_000231115'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_microsporidia1_collection/vittaforma_corneae_atcc_50505_gca_000231115/cdna/Vittaforma_corneae_atcc_50505_gca_000231115.Vitt_corn_V1.cdna.all.fa.gz',
	'vulpes_vulpes'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/vulpes_vulpes/cdna/Vulpes_vulpes.VulVul2.2.cdna.abinitio.fa.gz',
	'wallemia_ichthyophaga_exf_994_gca_000400465'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/wallemia_ichthyophaga_exf_994_gca_000400465/cdna/Wallemia_ichthyophaga_exf_994_gca_000400465.Wallemia_ichthyophaga_version_1.0.cdna.all.fa.gz',
	'wallemia_mellicola_cbs_633_66_gca_000263375'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/wallemia_mellicola_cbs_633_66_gca_000263375/cdna/Wallemia_mellicola_cbs_633_66_gca_000263375.Wallemia_sebi_v1.0.cdna.all.fa.gz',
	'wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255/cdna/Wickerhamomyces_anomalus_nrrl_y_366_8_gca_001661255.Wican1.cdna.all.fa.gz',
	'wickerhamomyces_ciferrii_gca_000313485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/wickerhamomyces_ciferrii_gca_000313485/cdna/Wickerhamomyces_ciferrii_gca_000313485.ASM31348v1.cdna.all.fa.gz',
	'wolfiporia_cocos_md_104_ss10_gca_000344635'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_basidiomycota1_collection/wolfiporia_cocos_md_104_ss10_gca_000344635/cdna/Wolfiporia_cocos_md_104_ss10_gca_000344635.Wolco1.cdna.all.fa.gz',
	'xenopus_tropicalis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/xenopus_tropicalis/cdna/Xenopus_tropicalis.JGI_4.2.cdna.abinitio.fa.gz',
	'xiphophorus_couchianus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/xiphophorus_couchianus/cdna/Xiphophorus_couchianus.Xiphophorus_couchianus-4.0.1.cdna.abinitio.fa.gz',
	'xiphophorus_maculatus'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/xiphophorus_maculatus/cdna/Xiphophorus_maculatus.X_maculatus-5.0-male.cdna.abinitio.fa.gz',
	'xylona_heveae_tc161_gca_001619985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/xylona_heveae_tc161_gca_001619985/cdna/Xylona_heveae_tc161_gca_001619985.Xylona_heveae_TC161_v1.0.cdna.all.fa.gz',
	'yarrowia_lipolytica'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/yarrowia_lipolytica/cdna/Yarrowia_lipolytica.GCA_000002525.1.cdna.all.fa.gz',
	'yarrowia_lipolytica_gca_001761485'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/yarrowia_lipolytica_gca_001761485/cdna/Yarrowia_lipolytica_gca_001761485.ASM176148v1.cdna.all.fa.gz',
	'yarrowia_lipolytica_gca_900087985'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/yarrowia_lipolytica_gca_900087985/cdna/Yarrowia_lipolytica_gca_900087985.YALIA101.cdna.all.fa.gz',
	'zea_mays'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/plants/fasta/zea_mays/cdna/Zea_mays.B73_RefGen_v4.cdna.all.fa.gz',
	'zonotrichia_albicollis'=>'ftp://ftp.ensembl.org/pub/release-'.$ensembl_release.'/fasta/zonotrichia_albicollis/cdna/Zonotrichia_albicollis.Zonotrichia_albicollis-1.0.1.cdna.abinitio.fa.gz',
	'zootermopsis_nevadensis'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/metazoa/fasta/zootermopsis_nevadensis/cdna/Zootermopsis_nevadensis.ZooNev1.0.cdna.all.fa.gz',
	'zygosaccharomyces_bailii_clib_213_gca_000442885'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/zygosaccharomyces_bailii_clib_213_gca_000442885/cdna/Zygosaccharomyces_bailii_clib_213_gca_000442885.ZYBA0.cdna.all.fa.gz',
	'zygosaccharomyces_bailii_gca_900162805'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota4_collection/zygosaccharomyces_bailii_gca_900162805/cdna/Zygosaccharomyces_bailii_gca_900162805.version_1.cdna.all.fa.gz',
	'zygosaccharomyces_bailii_isa1307_gca_000530735'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/zygosaccharomyces_bailii_isa1307_gca_000530735/cdna/Zygosaccharomyces_bailii_isa1307_gca_000530735.Z_bailii_ISA1307_version_3.cdna.all.fa.gz',
	'zygosaccharomyces_parabailii_gca_001984395'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/zygosaccharomyces_parabailii_gca_001984395/cdna/Zygosaccharomyces_parabailii_gca_001984395.ASM198439v2.cdna.all.fa.gz',
	'zygosaccharomyces_rouxii_gca_000026365'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota1_collection/zygosaccharomyces_rouxii_gca_000026365/cdna/Zygosaccharomyces_rouxii_gca_000026365.ASM2636v1.cdna.all.fa.gz',
	'zygosaccharomyces_rouxii_gca_001972345'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota3_collection/zygosaccharomyces_rouxii_gca_001972345/cdna/Zygosaccharomyces_rouxii_gca_001972345.ZYGR_assembly01.cdna.all.fa.gz',
	'zymoseptoria_brevis_gca_000966595'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/fungi_ascomycota2_collection/zymoseptoria_brevis_gca_000966595/cdna/Zymoseptoria_brevis_gca_000966595.ASM96659v1.cdna.all.fa.gz',
	'zymoseptoria_tritici'=>'ftp://ftp.ensemblgenomes.org/pub/release-'.$ensemblgenomes_release.'/fungi/fasta/zymoseptoria_tritici/cdna/Zymoseptoria_tritici.MG2.cdna.all.fa.gz',
	);### CDNA URLS END ###
			}

		if(!$silent){print"\nDownload cDNA data for $species{$species} from Ensembl database...";}
		$status=getstore("$ensembl_cDNA{$species}","UNITAS_refdump_$species/ensembl_cDNA.gz");
		if(is_success($status))
			{
			if(!$silent){print" OK.";}
			}
		else
			{
			if(!$silent){print" FAILED! $status. Download data from smallRNAgroup-SERVER...";}
			$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/Ensembl/Ensembl-cDNA.all_$species.fasta");
			$response = $useragent->request($request);
			$content = $response->content;
			open(my $out, '>', "UNITAS_refdump_$species/ensembl_cDNA.gz");
			print $out "$content";
			close $out;
			if(-e"UNITAS_refdump_$species/ensembl_cDNA.gz")
				{
				if(!$silent){print" OK.";}
				}
			else
				{
				if(!$silent){print" FAILED! $status.\n\n";}
				}
			}
		if(!$silent){print"\nExtract file...";}
		$ae=Archive::Extract->new(archive=>"UNITAS_refdump_$species/ensembl_cDNA.gz");
		$ae->extract(to=>"UNITAS_refdump_$species")||die print"\nERROR! Unable to decompress UNITAS_refdump_$species/ensembl_cDNA.gz.\n$!\n\n";
		if(!$silent){print" OK.\nRemove compressed file...";}
		unlink"UNITAS_refdump_$species/ensembl_cDNA.gz"||print"\nERROR! Unable to remove compressed file UNITAS_refdump_$species/ensembl_cDNA.gz.\n$!\n\n";
		if(!$silent){print" OK.\nProcess cDNA data from Ensembl...";}
		open(CDNA,"UNITAS_refdump_$species/ensembl_cDNA")||die print"\nERROR! Unable to open UNITAS_refdump_$species/ensembl_cDNA.\n$!\n\n";
		open(CDNA_EDIT,">UNITAS_refdump_$species/cDNA.fas")||die print"\nERROR! Unable to create UNITAS_refdump_$species/cDNA.fas.\n$!\n\n";
		%cDNA=();
		$cDNA_seq_ID=0;
		while(<CDNA>)
			{
			$_=~s/\s*$//;
			if($_=~/^>/)
				{
				$cDNA_seq_ID++;
				# extract biotype
				if($_=~s/transcript_biotype:[^ ]+//)
					{
					$biotype=$&;
					$biotype=~s/transcript_biotype://;
					}
				elsif($_=~s/gene_biotype:[^ ]+//)
					{
					$biotype=$&;
					$biotype=~s/gene_biotype://;
					}
				else
					{
					$biotype="unknown";
					}

				# extract gene symbol (gene if no gene symbol is provided)
				if($_=~s/gene_symbol:[^ ]+//)
					{
					$symbol=$&;
					$symbol=~s/gene_symbol://;
					}
				elsif($_=~s/gene:[^ ]+//)
					{
					$symbol=$&;
					$symbol=~s/gene://;
					}
				elsif($_=~s/GENSCAN\d+//)
					{
					$symbol=$&;
					}
				elsif($_=~s/SNAP\d+//)
					{
					$symbol=$&;
					}
				else
					{
					$symbol=$cDNA_seq_ID;
					}
				}
			else
				{
				$cDNA{"$biotype|$symbol"}.=$_;
				}
			}
		close CDNA;
		foreach$head(keys%cDNA)
			{
			print CDNA_EDIT">$head\n$cDNA{$head}\n";
			}
		close CDNA_EDIT;
		undef%cDNA;
		if(!$silent){print" OK.";}

		# add tRNA data from gtRNAdb
		foreach$id(keys%gtRNAdb)
			{
			$i++;
			print NCRNA_EDIT">tRNA|$id\n$gtRNAdb{$id}\n";
			}
		undef%gtRNAdb;

		# download tRF-1 sequences -> add to ncRNA.fas
		if(!$silent){print"\nDownload tRF-1 sequences for $species{$species}...";}
		$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/tRF-1/$species.tRF-1.fas");
		$response = $useragent->request($request);
		$content = $response->content;
		open(my $out, '>', "UNITAS_refdump_$species/tRF-1.fas");
		print $out "$content";
		close $out;
		if(-e"UNITAS_refdump_$species/tRF-1.fas")
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/tRF-1/release');
			$response = $useragent->request($request);
			$version = $response->content;
			$db_versions_info.="tRF-1 sequence data version/date: ......... $version (dd.mm.yyyy)\n";
			if(!$silent){print" OK. $version (dd.mm.yyyy).";}
			}
		else
			{
			if(!$silent){print" FAILED! $status\n\n";}
			}
		if(-e"UNITAS_refdump_$species/tRF-1.fas")
			{
			open(TRF1,"UNITAS_refdump_$species/tRF-1.fas");
			$tRF1_add_count=0;
			while(<TRF1>)
				{
				$_=~s/\s*$//;
				if($_=~/^>/)
					{
					$tRF1_add_count++;
					$head=$_;
					}
				else
					{
					print NCRNA_EDIT"$head\n$_\n";
					}
				}
			close TRF1;
			print"\ntRF-1 sequences: $tRF1_add_count.";
			}
		$i+=$tRF1_add_count;

		# download tRNA-leader sequences -> add to ncRNA.fas
		if(!$silent){print"\nDownload tRNA-leader sequences for $species{$species}...";}
		$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/tRNA_leader/$species.tRNA_leader.fas");
		$response = $useragent->request($request);
		$content = $response->content;
		open(my $out, '>', "UNITAS_refdump_$species/tRNA_leader.fas");
		print $out "$content";
		close $out;
		if(-e"UNITAS_refdump_$species/tRNA_leader.fas")
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/tRNA_leader/release');
			$response = $useragent->request($request);
			$version = $response->content;
			$db_versions_info.="tRNA-leader sequence data version/date: ... $version (dd.mm.yyyy)\n";
			if(!$silent){print" OK. $version (dd.mm.yyyy).";}
			}
		else
			{
			if(!$silent){print" FAILED! $status\n\n";}
			}
		if(-e"UNITAS_refdump_$species/tRNA_leader.fas")
			{
			open(TRNA_LEADER,"UNITAS_refdump_$species/tRNA_leader.fas");
			$tRNA_leader_add_count=0;
			while(<TRNA_LEADER>)
				{
				$_=~s/\s*$//;
				if($_=~/^>/)
					{
					$tRNA_leader_add_count++;
					$head=$_;
					}
				else
					{
					print NCRNA_EDIT"$head\n$_\n";
					}
				}
			close TRNA_LEADER;
			print"\ntRNA-leader sequences: $tRNA_leader_add_count.";
			}
		$i+=$tRNA_leader_add_count;

		# download silva rRNA data -> add to ncRNA.fas
		if(!$silent){print"\nDownload SILVA SSU rRNA data for $species{$species}...";}
		$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/SILVA_rRNA/SILVA_SSU_$species.fasta");
		$response = $useragent->request($request);
		$content = $response->content;
		open(my $out, '>', "UNITAS_refdump_$species/SILVA_rRNA.SSU");
		print $out "$content";
		close $out;
		if(-e"UNITAS_refdump_$species/SILVA_rRNA.SSU")
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/SILVA_rRNA/release');
			$response = $useragent->request($request);
			$version = $response->content;
			$db_versions_info.="SILVA rRNA (SSU) database version/date: ... Release $version\n";
			if(!$silent){print" OK. Release $version.";}
			}
		else
			{
			if(!$silent){print" FAILED! $status\n\n";}
			}
		if(!$silent){print"\nDownload SILVA LSU rRNA data for $species{$species}...";}
		$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/SILVA_rRNA/SILVA_LSU_$species.fasta");
		$response = $useragent->request($request);
		$content = $response->content;
		open(my $out, '>', "UNITAS_refdump_$species/SILVA_rRNA.LSU");
		print $out "$content";
		close $out;
		if(is_success($status))
			{
			$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/SILVA_rRNA/release');
			$response = $useragent->request($request);
			$version = $response->content;
			$db_versions_info.="SILVA rRNA (LSU) database version/date: ... Release $version\n";
			if(!$silent){print" OK. Release $version.";}
			}
		else
			{
			if(!$silent){print" FAILED! $status\n\n";}
			}
		open(SSU,"UNITAS_refdump_$species/SILVA_rRNA.SSU")||die print"\nERROR! Unable to read UNITAS_refdump_$species/SILVA_rRNA.SSU.\n$!\n\n";
		open(LSU,"UNITAS_refdump_$species/SILVA_rRNA.LSU")||die print"\nERROR! Unable to read UNITAS_refdump_$species/SILVA_rRNA.LSU.\n$!\n\n";
		$i_ssu=0;
		$i_lsu=0;
		while(<SSU>)
			{
			if($_=~/^>/)
				{
				$i_ssu++;
				}
			print NCRNA_EDIT$_;
			}
		while(<LSU>)
			{
			if($_=~/^>/)
				{
				$i_lsu++;
				}
			print NCRNA_EDIT$_;
			}
		close SSU;
		close LSU;
		close NCRNA_EDIT;
		$i+=$i_lsu+$i_ssu;
		if(!$silent){print"\nrRNA sequences: (SSU=$i_ssu LSU=$i_lsu).\nTOTAL: $i ncRNA sequences stored in UNITAS_refdump_$species/ncRNA.fas.";}

		# download miRBase data
		$miRBase_mature='ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.gz';
		$miRBase_hairpin='ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz';
		if(!$silent){print"\nDownload mature miRNA sequences from miRBase...";}
		$status=getstore($miRBase_mature,"UNITAS_refdump_$species/miRBase_mature.gz");
		if(is_success($status))
			{
			$version=get('ftp://mirbase.org/pub/mirbase/CURRENT/README');
			$version=~s/Release \d+//;
			$version=$&;
			$db_versions_info.="miRBase database version/date: ............ $version\n";
			if(!$silent){print" OK. $version.";}
			}
		else
			{
			if(!$silent){print" FAILED! $status. Download data from smallRNAgroup-SERVER...";}
			$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/miRBase/mature.fa.gz");
			$response = $useragent->request($request);
			$content = $response->content;
			open(my $out, '>', "UNITAS_refdump_$species/miRBase_mature.gz");
			print $out "$content";
			close $out;
			if(-e"UNITAS_refdump_$species/miRBase_mature.gz")
				{
				$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/miRBase/release');
				$response = $useragent->request($request);
				$version = $response->content;
				if(!$silent){print" OK. Release $version.";}
				$db_versions_info.="miRBase database version/date: ........ Release $version\n";
				}
			else
				{
				die print" FAILED! $status\n\n";
				}
			}
		if(!$silent){print"\nExtract file...";}
		$ae=Archive::Extract->new(archive=>"UNITAS_refdump_$species/miRBase_mature.gz");
		$ae->extract(to=>"UNITAS_refdump_$species")||die print"\nERROR! Unable to decompress UNITAS_refdump_$species/miRBase_mature.gz.\n$!\n\n";
		if(!$silent){print" OK.\nRemove compressed file...";}
		unlink"UNITAS_refdump_$species/miRBase_mature.gz"||print"\nERROR! Unable to remove compressed file UNITAS_refdump_$species/miRBase_mature.gz.\n$!\n\n";
		if(!$silent){print" OK.\nDownload hairpin miRNA sequences from miRBase...";}
		$status=getstore($miRBase_hairpin,"UNITAS_refdump_$species/miRBase_hairpin.gz");
		if(is_success($status))
			{
			$version=get('ftp://mirbase.org/pub/mirbase/CURRENT/README');
			$version=~s/Release \d+//;
			$version=$&;
			if(!$silent){print" OK. $version.";}
			}
		else
			{
			if(!$silent){print" FAILED! $status. Download data from smallRNAgroup-SERVER...";}
			$request = HTTP::Request->new(GET => "https://www.smallrnagroup.uni-mainz.de/data/UNITAS/miRBase/hairpin.fa.gz");
			$response = $useragent->request($request);
			$content = $response->content;
			open(my $out, '>', "UNITAS_refdump_$species/miRBase_hairpin.gz");
			print $out "$content";
			close $out;
			if(-e"UNITAS_refdump_$species/miRBase_hairpin.gz")
				{
				$request = HTTP::Request->new(GET => 'https://www.smallrnagroup.uni-mainz.de/data/UNITAS/miRBase/release');
				$response = $useragent->request($request);
				$version = $response->content;
				if(!$silent){print" OK. Release $version.";}
				}
			else
				{
				die print" FAILED! $status\n\n";
				}
			}
		if(!$silent){print"\nExtract file...";}
		$ae=Archive::Extract->new(archive=>"UNITAS_refdump_$species/miRBase_hairpin.gz");
		$ae->extract(to=>"UNITAS_refdump_$species")||die print"\nERROR! Unable to decompress UNITAS_refdump_$species/miRBase_hairpin.gz.\n$!\n\n";
		if(!$silent){print" OK.\nRemove compressed file...";}
		unlink"UNITAS_refdump_$species/miRBase_hairpin.gz"||print"\nERROR! Unable to remove compressed file UNITAS_refdump_$species/miRBase_hairpin.\n$!\n\n";
		if(!$silent){print" OK.";}


		# split hairpin miRNA sequences to species/non-species files
		if(!$silent){print"\nSplit hairpin miRNA sequences into $species{$species}/non-$species{$species} files...";}
		open(MIR_HAIRPIN,"UNITAS_refdump_$species/miRBase_hairpin")||die print"\nERROR! Unable to open miRBase file UNITAS_refdump_$species/miRBase_hairpin.\n$!\n\n";
		open(MIR_HAIRPIN_SPECIES,">UNITAS_refdump_$species/miRBase_hairpin_$species{$species}.fas")||die print"\nERROR! Unable to create file UNITAS_refdump_$species/miRBase_hairpin_$species{$species}.fas.\n$!\n\n";
		open(MIR_HAIRPIN_NONSPECIES,">UNITAS_refdump_$species/miRBase_hairpin_non-$species{$species}.fas")||die print"\nERROR! Unable to create file UNITAS_refdump_$species/miRBase_hairpin_non-$species{$species}.fas.\n$!\n\n";
		%miRhairpin_species=();
		%miRhairpin_nonspecies=();
		$i=0;
		@head=();
		while(<MIR_HAIRPIN>)
			{
			$_=~s/\s*$//;
			if($_=~s/^>//)
				{
				if(lc$head[2].'_'.$head[3]eq$species)
					{
					$i++;
					$miRhairpin_species{$hairpin_seq}=$head[4].'_'.$head[5];
					}
				@head=split(' ',$_);
				$hairpin_seq="";
				}
			else
				{
				$_=~s/U/T/g;
				$hairpin_seq.=$_;
				}
			}
		if(lc$head[2].'_'.$head[3]eq$species)
			{
			$i++;
			$miRhairpin_species{$hairpin_seq}=$head[4].'_'.$head[5];
			}
		close MIR_HAIRPIN;
		if(!$silent){print" $species{$species}: $i.";}

		open(MIR_HAIRPIN,"UNITAS_refdump_$species/miRBase_hairpin")||die print"\nERROR! Unable to open miRBase file UNITAS_refdump_$species/miRBase_hairpin.\n$!\n\n";
		$i=0;
		@head=();
		while(<MIR_HAIRPIN>)
			{
			$_=~s/\s*$//;
			if($_=~s/^>//)
				{
				if(lc$head[2].'_'.$head[3]ne$species&&!$miRhairpin_species{$hairpin_seq}) # species is different and sequence is not in species
					{
					$i++;
					$miRhairpin_nonspecies{$hairpin_seq}.="$head[0]_stem-loop ";
					}
				@head=split(' ',$_);
				$hairpin_seq="";
				}
			else
				{
				$_=~s/U/T/g;
				$hairpin_seq.=$_;
				}
			}
		if(lc$head[2].'_'.$head[3]ne$species&&!$miRhairpin_species{$hairpin_seq})
			{
			$i++;
			$miRhairpin_nonspecies{$hairpin_seq}.="$head[0]_stem-loop ";
			}
		close MIR_HAIRPIN;
		foreach(keys%miRhairpin_nonspecies)
			{
			$miRhairpin_nonspecies{$_}=~s/ $//;
			}
		if(!$silent){print" non-$species{$species}: $i.";}

		#sort keys to make reproducable files
		@keyshairpin_species=keys%miRhairpin_species;
		@keyshairpin_species=sort@keyshairpin_species;
		foreach$miR(@keyshairpin_species)
			{
			print MIR_HAIRPIN_SPECIES">$miRhairpin_species{$miR}\n$miR\n";
			}
		@keyshairpin_nonspecies=keys%miRhairpin_nonspecies;
		@keyshairpin_nonspecies=sort@keyshairpin_nonspecies;
		foreach$miR(@keyshairpin_nonspecies)
			{
			print MIR_HAIRPIN_NONSPECIES">$miRhairpin_nonspecies{$miR}\n$miR\n";
			}
		undef%miRhairpin_species;
		undef%miRhairpin_nonspecies;
		undef@keyshairpin_species;
		undef@keyshairpin_nonspecies;
		close MIR_HAIRPIN_SPECIES;
		close MIR_HAIRPIN_NONSPECIES;
		if(!$silent){print" OK.\n";}

		open(DBVERSIONS,">UNITAS_refdump_$species/db_versions.info")||print"\n$!\n";
		print DBVERSIONS$db_versions_info;
		close DBVERSIONS;
		}
	}

##################################
# DIFFERENTIAL EXPRESSION MODULE #
##################################

sub diff_expr
	{
	print"


                 unitas: DIFFERENTIAL EXPRESSION MODULE

    |     1a
  E |   XXXXXX              1c
  X |   XXXXXX     1b     XXXXXX
  P |   XXXXXX   XXXXXX   XXXXXX
  R |   XXXXXX   XXXXXX   XXXXXX
  E |   XXXXXX   XXXXXX   XXXXXX
  S |   XXXXXX   XXXXXX   XXXXXX                      2b
  S |   XXXXXX   XXXXXX   XXXXXX             2a     OOOOOO
  I |   XXXXXX   XXXXXX   XXXXXX           OOOOOO   OOOOOO
  O |   XXXXXX   XXXXXX   XXXXXX           OOOOOO   OOOOOO     2c
  N |   XXXXXX   XXXXXX   XXXXXX           OOOOOO   OOOOOO   OOOOOO
    |   XXXXXX   XXXXXX   XXXXXX           OOOOOO   OOOOOO   OOOOOO
    |___XXXXXX___XXXXXX___XXXXXX___________OOOOOO___OOOOOO___OOOOOO_____
               condition 1                        condition 2

";
	if(@diffex_g1==0||@diffex_g2==0)
		{
		print"\n\nYou must define unitas output folders for each group with -g1 and -g2.\nThe command should look like this:\n\nperl unitas.pl -diffexpr -g1 folderA -g1 folderB -g2 folderC -g2 folderD\n\n";
		exit;
		}
	if($standard_Ttest) # load the Statistics::Distributions module
		{
		$load_module=eval{require Statistics::Distributions;Statistics::Distributions->import();1;};
		if(!$load_module){
		print"

Perl module Statistics::Distributions is required to perform a t-test, yet it
is not installed on this machine! You can download it here:

https://metacpan.org/pod/Statistics::Distributions.

";
		exit;}
		}


	# remove redundant folders from groups
	%diffex_g1=();
	print"\n\nGROUP 1:";
	foreach(sort@diffex_g1)
		{
		$diffex_g1{$_}=1;
		print"\n $_";
		}
	%diffex_g2=();
	print"\n\nGROUP 2:";
	foreach(sort@diffex_g2)
		{
		$diffex_g2{$_}=1;
		print"\n $_";
		}

	# check number of replicates per group
	if(keys%diffex_g1<2||keys%diffex_g2<2)
		{
		if($standard_Ttest&&keys%diffex_g1==1&&keys%diffex_g2==1)
			{
			print"\n\nNOTE: You cannot perform a t-test with only one replicate per group!\nUnitas will perform an alternative SAND test with fictional variance.\nHandle results with caution!";
			undef$standard_Ttest;
			}
		$print_estimated_deviation=$estimated_deviation*100;
		print"\n\nWARNING! There should be at least two replicates in each of the two groups.";
		if($estimated_deviation>0)
			{
			print"\nVariance will be estimated by adding two virtual replicates with rpm values\n+- $print_estimated_deviation%.\n";
			}
		else
			{
			print"\nIf it is not possible to use more replicates, use the option\n-diffexpr_estdev [0..1] to estimate standard deviation.\n";
			}
		}

	# print which test will be performed
	$test_to_perform="'SAND test' (shared area of two normal distributions)";
	if($standard_Ttest)
		{
		$test_to_perform="standard t-test (unpaired, two-sided)";
		}
	print"

unitas will perform a $test_to_perform
with critical value for p<=$diff_expr_p to find differentially expressed small RNAs.

";



	# check for existence of each folder
	foreach(keys%diffex_g1,keys%diffex_g2)
		{
		if(!-d$_)
			{
			print"\nERROR! Folder $_ does not exist.\n\n";
			exit;
			}
		}

	#check for occurence of any folder in both groups
	foreach$fld(keys%diffex_g1)
		{
		if($diffex_g2{$fld})
			{
			print"\nERROR! A replicate must not occur in both groups.\n$f1 was assigned to group 1 AND group 2.\n\n";
			exit;
			}
		}

	# collect sequences and calculate rpm values
	print"\n\nCollect sequences and calculate rpm values...";
	%rpm=();
	%abs_reads=();
	$total_reads_group1=0;
	$total_reads_group2=0;
	foreach$fld(keys%diffex_g1,keys%diffex_g2)
		{
		open(SUMMARY,"$fld/$outfile_prefix.annotation_summary.txt")||die print"\nERROR! Unable to open $fld/$outfile_prefix.annotation_summary.txt.\n$!\n\n";
		print"\nRead data from $fld...";
		while(<SUMMARY>)
			{
			next if($_=~/^low_complexity/);
			next if($_=~/^ /);
			$_=~s/\s*$//;
			@d=split("\t",$_);
			$total_reads{$fld}+=$d[1];
			if($diffex_g1{$fld})
				{
				$total_reads_group1+=$d[1];
				}
			elsif($diffex_g2{$fld})
				{
				$total_reads_group2+=$d[1];
				}
			last if($_=~/^\s*$/);
			}
		close SUMMARY;

		if(-d"$fld/fasta") # since version 1.6.0 unitas puts all the FASTA files into a sub-folder
			{
			opendir(DIR,"$fld/fasta")||die print"\nERROR! Unable to open directory $fld/fasta.\n$!\n\n";
			$fasta_dir="$fld/fasta";
			}
		else
			{
			opendir(DIR,$fld)||die print"\nERROR! Unable to open directory $fld.\n$!\n\n";
			$fasta_dir=$fld;
			}
		while($file=readdir(DIR))
			{
			# open unitas fasta files but not those with special format of fasta headers (where header not or not only refers to sequence read counts)
			if($file=~/^$outfile_prefix\..+\.fas$/&&$file!~/^$outfile_prefix\.non-miR\.fas$/&&$file!~/^$outfile_prefix\.piRcandidates\.fas$/&&$file!~/^$outfile_prefix\.phasiRNA\.sorted\.fas$/)
				{
				$class=$file;
				$class=~s/^$outfile_prefix\.//;
				$class=~s/\.fas$//;
				open(FAS,"$fasta_dir/$file")||die print"\nERROR! Unable to open fasta file $fld/$file.\n$!\n\n";
				while(<FAS>)
					{
					$_=~s/\s*$//;
					if($_=~s/^>//)
						{
						$reads=$_;
						}
					else
						{
						$rpm{$_}{$fld}=$reads/$total_reads{$fld}*1000000;
						$abs_reads{$_}{$fld}=$reads;
						$annotation{$_}{$class}=1;
						}
					}
				close FAS;
				}
			}
		closedir DIR;
		$print_collected_reads=int($total_reads{$fld}+0.5);
		print" OK.\nTotal reads: $print_collected_reads\n";
		}
	$seqs_to_analyze=keys%rpm;
	print"

 Calculate raw p values for differential expression of $seqs_to_analyze sequences
 0%                                  50%                                  100%
 |------------------------------------|------------------------------------|
 ";

	# perform the statistical test for each sequence
	$sig=0;
	$nsig=0;
	$skipped_tests=0;
	$performed_tests=0;
	%uncorrected_p_value=();
	%higher_expression_in_group=();
	$progress=0;
	$progress_dots=0;
	foreach$seq(keys%rpm)
		{
		$progress++;
		if($progress>=$seqs_to_analyze/75)
			{
			print".";
			$progress=0;
			$progress_dots++;
			}
		@i1=();
		@i2=();
		$present_in_all_probes=1;
		foreach$fld_g1(keys%diffex_g1)
			{
			if(!$rpm{$seq}{$fld_g1})
				{
				$rpm{$seq}{$fld_g1}=0;
				$present_in_all_probes=0;
				}
			push(@i1,$rpm{$seq}{$fld_g1});
			}
		foreach$fld_g2(keys%diffex_g2)
			{
			if(!$rpm{$seq}{$fld_g2})
				{
				$rpm{$seq}{$fld_g2}=0;
				$present_in_all_probes=0;
				}
			push(@i2,$rpm{$seq}{$fld_g2});
			}

		if($present_in_all_probes==0&&$diffexpr_nostrays==1)
			{
			$skipped_tests++;
			}
		else
			{
			$performed_tests++;

			# perform a t-test (two-sided)
			if($standard_Ttest)
				{
				# calculate mean for each group
				$avg1=calc_avg(@i1);
				$avg2=calc_avg(@i2);
				$avg_table_out="\t$avg1\t$avg2";

				# test which group has higher avg expression
				$higher_expression_in_group{$seq}=2;
				if($avg1>$avg2){$higher_expression_in_group{$seq}=1;}

				# calculate common variance
				$sum_diff_squares1=0;
				$sum_diff_squares2=0;
				foreach$x(@i1){$sum_diff_squares1+=($x-$avg1)**2;}
				foreach$x(@i2){$sum_diff_squares2+=($x-$avg2)**2;}

				$s_square=(($sum_diff_squares1+$sum_diff_squares2))/((@i1+@i2)-2);
				$t_value=(abs($avg1-$avg2))/(sqrt(($s_square/@i1)+($s_square/@i2)));
				$dof=(@i1+@i2)-2;
				$p_Ttest=Statistics::Distributions::tprob($dof,$t_value);
				$uncorrected_p_value{$seq}=$p_Ttest*2; # p for upper AND lower (two-sided) end of the distribution
				$p_shared=$p_Ttest*2;
				record_test_resultst();
				}

			# calculate shared area of two normal distributions as a proxy for p
			else
				{
				calc_shared_area();
				$uncorrected_p_value{$seq}=$p_shared;
				}
			}
		}
	foreach($progress_dots..74)
		{
		print".";
		}
	print"\n\n$performed_tests tests performed: $sig tests with p<=$diff_expr_p.";

	# Boferroni (1) / Bonferroni-Holm (2) correction of alpha error
	$sig=0;
	%corrected_significant=();
	if($alpha_correction==1)
		{
		print"\nApply Bonferroni-Holm correction of alpha error...";
		$i=-1;
		foreach$seq(sort{$uncorrected_p_value{$a}<=>$uncorrected_p_value{$b}}keys%uncorrected_p_value)
			{
			$i++;
			if($uncorrected_p_value{$seq}*($performed_tests-$i)<=$diff_expr_p||$diff_expr_p>=1)
				{
				$sig++;
				if($uncorrected_p_value{$seq}>1){$uncorrected_p_value{$seq}=1;}
				$corrected_significant{$seq}=$uncorrected_p_value{$seq};
				}
			else
				{
				last;
				}
			}
		foreach$seq(keys%corrected_significant)
			{
			$corrected_significant{$seq}=$corrected_significant{$seq}*($performed_tests-$i);
			}

		print" done. -> $sig significant results.";
		}
	elsif($alpha_correction==2)
		{
		print"\nApply strict Bonferroni correction of alpha error...";
		foreach$seq(keys%uncorrected_p_value)
			{
			$p_corrected=$uncorrected_p_value{$seq}*$performed_tests;
			if($p_corrected<=$diff_expr_p||$diff_expr_p>=1)
				{
				$sig++;
				if($p_corrected>1){$p_corrected=1;}
				$corrected_significant{$seq}=$p_corrected;
				}
			}
		print" done. -> $sig significant results.";
		}
	else
		{
		print"\nCollect results without correction of alpha error...";
		foreach$seq(keys%uncorrected_p_value)
			{
			if($uncorrected_p_value{$seq}<=$diff_expr_p||$diff_expr_p>=1)
				{
				$sig++;
				$corrected_significant{$seq}=$uncorrected_p_value{$seq}; # use uncorrected p values for output
				}
			}
		print" done.";
		}

	# print significant results to output file
	print"\nPrint results to outfiles...";
	print DIFFEX_OUT_1"sequence";
	print DIFFEX_OUT_2"sequence";
	foreach$fld(@diffex_g1,@diffex_g2)
		{
		print DIFFEX_OUT_1"\tabs_$fld";
		print DIFFEX_OUT_2"\tabs_$fld";
		}
	foreach$fld(@diffex_g1,@diffex_g2)
		{
		print DIFFEX_OUT_1"\trpm_$fld";
		print DIFFEX_OUT_2"\trpm_$fld";
		}
	print DIFFEX_OUT_1"\tavg_group1\tavg_group2\tp-value\tcorr_p-value\tclass\n";
	print DIFFEX_OUT_2"\tavg_group1\tavg_group2\tp-value\tcorr_p-value\tclass\n";

	foreach$seq(sort{$a cmp $b}keys%corrected_significant)
		{
		# group 1 > group 2
		if($higher_expression_in_group{$seq}==1)
			{
			print DIFFEX_OUT_1"${seq}$diffexpr_table_out{$seq}\t$uncorrected_p_value{$seq}\t$corrected_significant{$seq}";
			foreach$class(sort{$a cmp $b}keys%{$annotation{$seq}})
				{
				print DIFFEX_OUT_1"\t$class";
				}
			print DIFFEX_OUT_1"\n";
			}
		# group 2 > group 1
		elsif($higher_expression_in_group{$seq}==2)
			{
			print DIFFEX_OUT_2"${seq}$diffexpr_table_out{$seq}\t$uncorrected_p_value{$seq}\t$corrected_significant{$seq}";
			foreach$class(sort{$a cmp $b}keys%{$annotation{$seq}})
				{
				print DIFFEX_OUT_2"\t$class";
				}
			print DIFFEX_OUT_2"\n";
			}
		}
	close DIFFEX_OUT_1;
	close DIFFEX_OUT_2;
	print" done.\n\n\n";

	sub calc_shared_area
		{
		avg_and_dev();
		sub avg_and_dev
			{
			# calculate average and standard deviation
			$avg1=calc_avg(@i1);
			$avg2=calc_avg(@i2);
			$avg_table_out="\t$avg1\t$avg2";
			if(@i1==1&&$estimated_deviation>0) # simulate variance if there is only one replicate
				{
				push(@i1,$i1[0]*(1-$estimated_deviation));
				push(@i1,$i1[0]*(1+$estimated_deviation));
				}
			if(@i2==1&&$estimated_deviation>0) # simulate variance if there is only one replicate
				{
				push(@i2,$i2[0]*(1-$estimated_deviation));
				push(@i2,$i2[0]*(1+$estimated_deviation));
				}
			$avg=$avg1; $stddev1=calc_stddev(@i1);
			$avg=$avg2; $stddev2=calc_stddev(@i2);

			# inverse group1 and group2 (group1 must have lower avg)
			$higher_expression_in_group{$seq}=2;
			if($avg1>$avg2)
				{
				($avg1,$avg2)=($avg2,$avg1);
				($stddev1,$stddev2)=($stddev2,$stddev1);
				$higher_expression_in_group{$seq}=1;
				}

			# use minimal expected standard deviation (corresponds to expected standard deviation in technical replicates -> e.g. resequencing the same library. Depends on rpm and library size)
			if($avg1==0)
				{
				$avg1=1/10**10;
				}
			if($avg2==0)
				{
				$avg2=1/10**10;
				}

			$minimum_relative_stddev1=(1/sqrt($total_reads_group1/1000000))/sqrt($avg1);
			$minimum_relative_stddev2=(1/sqrt($total_reads_group2/1000000))/sqrt($avg2);
			if($stddev1<$minimum_relative_stddev1)
				{
				$stddev1=$minimum_relative_stddev1;
				}
			if($stddev2<$minimum_relative_stddev2)
				{
				$stddev2=$minimum_relative_stddev2;
				}
			}

		# calculate the point of intersection ($xi) of both normal distributions
		if(abs($stddev1-$stddev2)<0.00001)
			{
			$xi=($avg1+$avg2)/2;
			}
		else
			{
			$xi=((($stddev2**2)*$avg1)-(($stddev1**2)*$avg2)+(($stddev1*$stddev2)*(sqrt((($avg1-$avg2)**2)+((2*(($stddev2**2)-($stddev1**2)))*(log($stddev2/$stddev2)))))))/(($stddev2**2)-($stddev1**2));
			}

		# calculate the area spanned by both graphs (=error probability p)
		$A_shared=2*(1-(integral($xi,$avg1,$stddev1)));
		$p_shared=$A_shared/($A_shared+(2*(1-$A_shared)));
		record_test_resultst();
		sub record_test_resultst
			{
			if($p_shared<=$diff_expr_p||$diff_expr_p>=1)
				{
				$sig++;
				foreach$fld(@diffex_g1,@diffex_g2)
					{
					if(!$abs_reads{$seq}{$fld})
						{
						$abs_reads{$seq}{$fld}=0;
						}
					$diffexpr_table_out{$seq}.="\t$abs_reads{$seq}{$fld}";
					}
				foreach$fld(@diffex_g1,@diffex_g2)
					{
					$diffexpr_table_out{$seq}.="\t$rpm{$seq}{$fld}";
					}
				$diffexpr_table_out{$seq}.=$avg_table_out;
				}
			else
				{
				$nsig++;
				}
			}
		}

	sub calc_avg
		{
		$avg=0;
		foreach(@_)
			{
			$avg+=$_;
			}
		return ($avg/@_);
		}
	sub calc_stddev
		{
		$var=0;
		foreach(@_)
			{
			$var+=($_-$avg)**2;
			}
		$var=$var/@_;
		return (sqrt($var));
		}
	sub density
		{
		($x,$m,$s)=(0,0,1);
		$x=shift if@_;
		$m=shift if@_;
		$s=shift if@_;
		$z=($x-$m)/$s;
		return exp(-0.5*$z*$z)/(2.506628274631*$s);
		}
	sub integral
		{
		($x,$m,$s)=(0,0,1);
		$x=shift if@_;
		$m=shift if@_;
		$s=shift if@_;
		$z=($x-$m)/$s;
		$t=1.0/(1+0.2316419*abs($z));
		$y=$t*(0.319381530+$t*(-0.356563782+$t*(1.781477937+$t*(-1.821255978+$t*1.330274429))));
		if($z>0)
			{
			return 1-density($z)*$y;
			}
		else
			{
			return density($z)*$y;
			}
		}
	}
}
